---
title: "Human ATRT scRNAseq data analysis"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
    fig_caption: yes
    fig_height: 8
    fig_width: 9
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
  word_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r paths}

thisProject                  <- "cellOrigin_ATRT/"

# Paths
pathToRhabdoid_U830_projects <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/"
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.6/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.5/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToThisProject            <- paste0(pathToANALYSIS_PROJECTS,thisProject)
pathToData                   <- paste0(pathToThisProject,"data/")
pathToFigures                <- paste0(pathToThisProject,"svn/analyse/script/figures/")
pathToReport                 <- paste0(pathToThisProject,"svn/analyse/report/")
pathToRNApip_RESULT          <- paste0(pathToThisProject,"RNApip_RESULT/")
pathToFeatureCounts          <- paste0(pathToThisProject,"FeatureCounts_RESULT/")

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r manage_renv, eval=FALSE}

# Initialize Renv
#renv::init()

```


```{r load_library, eval=FALSE}

# Load RNAseq library
source("load_scRNAseq_library.R")

```

```{r loading_RData, eval=FALSE}

# Load saved RData
load("cellOrigin_ATRT_scRNAseq_Jessa2019_ATRT5.RData")

#save.image(file="cellOrigin_ATRT_scRNAseq_Jessa2019_ATRT5.RData")

```

```{r load_plot_function}

# Load external files
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")
source("scRNAseq_functions.R")

```

```{r delete_all_objects}

# Delete all objects
#rm(list = ls())

```

# Jessa2019_ATRT5 scRNA-seq

## Load data and create Seurat object

```{r load_data}

pathTo_Jessa2019_ATRT5_cellRangerCount <- "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/data/Jessa2019_scRNAseq_data/ATRT5/"

# Load 10X data
scRNAseq_data_Jessa2019_ATRT5_raw <- Read10X(data.dir         = pathTo_Jessa2019_ATRT5_cellRangerCount,
                                             gene.column      = 2,
                                             unique.features  = TRUE
                                             )

# Create Seurat object
scRNAseq_Seurat_Jessa2019_ATRT5_raw <- CreateSeuratObject(counts   = scRNAseq_data_Jessa2019_ATRT5_raw,
                                                          project  = "Jessa2019_ATRT5"
                                                          )

# Check Seurat object 
scRNAseq_Seurat_Jessa2019_ATRT5_raw

# Export meta data for all cells (before filtering)
write.table(x      = scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data,
            file   = "scRNAseq_Seurat_Jessa2019_ATRT5_raw_metadata.txt",
            quote  = FALSE,
            sep    = ","
            )

```

## Cell quality control and filtering

### Adding QC metrics

```{r mito_ribo_reads_fraction}

# Log10 number of UMIs
scRNAseq_Seurat_Jessa2019_ATRT5_raw[["log10_nCount_RNA"]] <- log10(scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data$nCount_RNA)

# Log10 number of UMIs
scRNAseq_Seurat_Jessa2019_ATRT5_raw[["nCount_per_Feature"]] <- scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data$nCount_RNA/scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data$nFeature_RNA

# Ribosomal protein reads fraction
scRNAseq_Seurat_Jessa2019_ATRT5_raw[["percent.riboProt"]] <- PercentageFeatureSet(object  = scRNAseq_Seurat_Jessa2019_ATRT5_raw,
                                                                         pattern = "^RP[LS].*",
                                                                         assay   = "RNA"
                                                                         )
# Mitocondrial genes percentage
scRNAseq_Seurat_Jessa2019_ATRT5_raw[["percent.mito"]] <- PercentageFeatureSet(object   = scRNAseq_Seurat_Jessa2019_ATRT5_raw,
                                                                     pattern  = "^MT-",
                                                                     assay    = "RNA"
                                                                     )

# Check meta data
head(scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data)

```

### Number of detected genes and number of RNA (UMIs)

```{r number_of_genes_per_cell}

# Set minimum number of genes
min_nFeature <- round(quantile(scRNAseq_Seurat_Jessa2019_ATRT5_raw@meta.data$nFeature_RNA,
                               probs = 0.05
                               )
                      )
min_nFeature

# Plot histogram of RNA count
#pdf(file = paste0(pathToFigures,"histogram_scRNAseq_Jessa2019_ATRT5_nbRNA.pdf"), width = 12, height = 12)
plot_histogram_nbRNA(scRNAseq_Seurat_Jessa2019_ATRT5_raw, log = 10, binwidth=0.05)
#dev.off()

# Plot density distribution of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_Jessa2019_ATRT5_nbRNA.pdf"), width = 12, height = 12)
plot_density_nbRNA(scRNAseq_Seurat_Jessa2019_ATRT5_raw, log = 10)
#dev.off()

# Plot histogram of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_Jessa2019_ATRT5_nbGenes.pdf"), width = 12, height = 12)
plot_histogram_nbGenes(scRNAseq_Seurat_Jessa2019_ATRT5_raw, log = 10, binwidth=0.05)
#dev.off()

# Plot density distribution of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_Jessa2019_ATRT5_nbGenes.pdf"), width = 12, height = 12)
plot_density_nbGenes(scRNAseq_Seurat_Jessa2019_ATRT5_raw, log = 10)
#dev.off()

# Plot scatter plot nCount_RNA and nFeature_RNA
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_Jessa2019_ATRT5_nbRNA_vs_nbGene.pdf"), width = 12, height = 12)
plot_nbRNA_vs_nbGenes(scRNAseq_Seurat_Jessa2019_ATRT5_raw, min_nbGene = min_nFeature)
#dev.off()

# Density distribution of the number of UMIs per gene
#pdf(file = paste0(pathToFigures,"histogram_scRNAseq_Jessa2019_ATRT5_nbRNA_per_gene.pdf"), width = 12, height = 12)
plot_distribution_nbRNA_per_Gene(scRNAseq_Seurat_Jessa2019_ATRT5_raw)
#dev.off()

```

### Fraction of reads mapped to mitochondrial genes

Since the data is from a single nuclei RNA-seq, we do note expect mitochondrial gene UMIs.

```{r mitochondrial_gene}

# Set percentage of count falling in mitochondrial genes
maxPercentMito <- 100

# Scatter plot of nb of RNA and percent of mitochondrial RNA
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_Jessa2019_ATRT5_nbRNA_vs_percMito.pdf"), width = 12, height = 12)
plot_nbRNA_vs_percentMito(scRNAseq_Seurat_Jessa2019_ATRT5_raw,
                          maxPercentMito = maxPercentMito
                          )
#dev.off()

```

### Fraction of reads mapped to ribosomal protein genes

```{r ribosomal_gene}

# Scatter plot nb of genes and fraction of reads map to ribosomal proteins
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_Jessa2019_ATRT5_percRiboProt_vs_nbGene.pdf"), width = 12, height = 12)
plot_percentRibo_vs_nbGene(scRNAseq_Seurat_Jessa2019_ATRT5_raw, min_nbGene = min_nFeature)
#dev.off()

#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_Jessa2019_ATRT5_percRiboProt_vs_percMito.pdf"), width = 12, height = 12)
plot_percentRibo_vs_percentMito(scRNAseq_Seurat_Jessa2019_ATRT5_raw, min_nbGene = min_nFeature, max_percMito = maxPercentMito)
#dev.off()

```

## Cell filtering

```{r cell_filtering}

# Subset "good" cells
scRNAseq_Seurat_Jessa2019_ATRT5 <- subset(x      = scRNAseq_Seurat_Jessa2019_ATRT5_raw,
                                          subset = nFeature_RNA > min_nFeature
                                          )

# Plot histogram of nb of cells after each filtering step
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Jessa2019_ATRT5_cellFiltering_nbGenes_percMito.pdf"), width = 12, height = 8)
barplot_cellfiltering(scRNAseq_Seurat_Jessa2019_ATRT5_raw, maxPercentMito = NULL, min_nFeature = min_nFeature, ylim = 10000)
#dev.off()

```

## Data normalization

```{r data_normalization}

# Normalization
scRNAseq_Seurat_Jessa2019_ATRT5 <- NormalizeData(object                = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                 normalization.method  = "LogNormalize",
                                                 scale.factor          = 10000
                                                 )

```

# Assign cell-cycle scores

```{r assign_cellCycle_scores}

# Cell cycle gene markers (present in Seurat)
str(cc.genes)
length(cc.genes$s.genes)
length(cc.genes$g2m.genes)

cc.genes$g2m.genes
cc.genes$s.genes

# Compute cell cycle scoring  
scRNAseq_Seurat_Jessa2019_ATRT5 <- CellCycleScoring(object       = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                    s.features   = cc.genes$s.genes,
                                                    g2m.features = cc.genes$g2m.genes
                                                    )

# Check meta data
head(scRNAseq_Seurat_Jessa2019_ATRT5@meta.data)

#pdf(file = paste0(pathToFigures,"plots_cellCyclPhase_scRNAseq_Jessa2019_ATRT5.pdf"), width = 16, height = 12)
plot_cellCyclePhase(scRNAseq_Seurat_Jessa2019_ATRT5)
#dev.off()

```


## Dimensionality reduction (PCA)

```{r feature_selection}

# Highly variable genes
nFeatures <- 2000
variableFeatures_method <- "vst"
scRNAseq_Seurat_Jessa2019_ATRT5 <- FindVariableFeatures(object            = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                        selection.method  = variableFeatures_method,
                                                        nfeatures         = nFeatures
                                                        )

# Mean-variance scatter plot
#pdf(file = paste0(pathToFigures,"variableFeaturePlot_scRNAseq_Jessa2019_ATRT5.pdf"), width = 12, height = 12)
VariableFeaturePlot(scRNAseq_Seurat_Jessa2019_ATRT5) +
    xlab("\nAverage Expression") +
    ylab("Standard Deviation\n") +
        theme_bw() +
        theme(text = element_text(size = 30),
              axis.title = element_text(size = 30, face = "bold"),
              legend.position = c(0.9,0.8)
              )
#dev.off()

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5 <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5,
                                             features = rownames(scRNAseq_Seurat_Jessa2019_ATRT5)
                                             )

# PCA
scRNAseq_Seurat_Jessa2019_ATRT5 <- RunPCA(object = scRNAseq_Seurat_Jessa2019_ATRT5)

# PCs elbow plot
PCmin <- 12
#pdf(file = paste0(pathToFigures,"elbowPlot_scRNAseq_Jessa2019_ATRT5_PCA.pdf"), width = 12, height = 12)
ElbowPlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
          ndims = 50
          ) +
    geom_vline(xintercept = PCmin, lty = 2, color = "blue") +
    xlab("\nPC") +
    ylab("Standard Deviation\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Heatmap of each PC
#pdf(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_Jessa2019_ATRT5_PCA.pdf"), width = 15, height = 11)
DimHeatmap(object = scRNAseq_Seurat_Jessa2019_ATRT5,
           dims = 1:15,
           ncol = 5
           )
#dev.off()

# Plot PCA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA.pdf"), width = 12, height = 12)
DimPlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
        pt.size = 2,
        reduction = "pca",
        cols = rep("black",length(levels(Idents(scRNAseq_Seurat_Jessa2019_ATRT5))))
        ) +
    xlab("\nPC_1") +
    ylab("PC_2\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

```

## Plot PCA and covariates

```{r PCA_and_covariates}

# Extract scaled matrix
scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                               slot = "scale.data"
                                                               )
                                                  )
head(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix[,1:3])

# Get HVF
scRNAseq_Seurat_Jessa2019_ATRT5_mvGenes <- HVFInfo(object = scRNAseq_Seurat_Jessa2019_ATRT5,
                                          selection.method = "vst"
                                          )
head(scRNAseq_Seurat_Jessa2019_ATRT5_mvGenes)
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_mvGenes)

# Get most variable genes
scRNAseq_Seurat_Jessa2019_ATRT5_mvGenes[order(scRNAseq_Seurat_Jessa2019_ATRT5_mvGenes$variance.standardized, decreasing = TRUE),] %>% .[1:nFeatures,] -> mvGenes

# Subset matrix
scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes <- scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix[as.vector(rownames(mvGenes)),]
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)

# PCA analysis
PC        <- prcomp(data.frame(t(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)

pci       <- data.frame(PC$x,scRNAseq_Seurat_Jessa2019_ATRT5@meta.data)

# Plot eigen value
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Jessa2019_ATRT5_PCA_eigValPer.pdf"), width = 12, height = 12)
ggplot() +
    geom_bar(data = data.frame(PC = 1:50, eigValPer = eigValPer[1:50]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity"
             ) +
    geom_bar(data = data.frame(PC = 1:PCmin, eigValPer = eigValPer[1:PCmin]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity",
             fill = "blue"
             ) +
    xlab(label = "\nPC") +
    ylab(label = "% variability (eigen value)\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Plot PCA and nb of UMIs
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_nbUMIs.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "log10_nCount_RNA", legendTitle = "log10(nbUMIs)")
#dev.off()

# Plot PCA and nb of genes
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_nbGenes.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nFeature_RNA", legendTitle = "nbGenes")
#dev.off()

# Plot PCA and nb RNA per gene
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_nbRNA_per_gene.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nCount_per_Feature", legendTitle = "UMIs/gene")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_percMito.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.mito", legendTitle = "% mito")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_percRiboProt.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.riboProt", legendTitle = "% riboProt")
#dev.off()

# Plot PCA and cell cycle phase
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_Jessa2019_ATRT5_PCA_cellCyclePhase.pdf"), width = 12, height = 12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = Phase),
               size    = 1#,
               #pch     = 21
               ) +
    #scale_fill_continuous(type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```


# Cell clustering

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_Jessa2019_ATRT5 <- FindNeighbors(object = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                 dims = 1:PCmin
                                                 )

clustering_resolutions <- seq(from = 0, to = 2, by = 0.1)

# Clustering by modularity optimization technique
scRNAseq_Seurat_Jessa2019_ATRT5 <- FindClusters(object     = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                resolution = clustering_resolutions
                                                )      

# Save seurat object
saveRDS(object = scRNAseq_Seurat_Jessa2019_ATRT5,
        file   = paste0(pathToReport,"scRNAseq_Seurat_Jessa2019_ATRT5.rds"
                        )
        )
    
```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_Jessa2019_ATRT5.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data,
         prefix          = "RNA_snn_res.",
         #node_colour     = "sc3_stability",
         #layout          = "sugiyama",
         use_core_edges  = TRUE,
         color           = "red"
         )
#dev.off()

choosenResolution <- "RNA_snn_res.0.5"

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_Jessa2019_ATRT5_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(choosenResolution)),
                             fill  = Phase
                             )
               ) +
    geom_text(mapping  = aes(x     = eval(as.name(choosenResolution)),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_Jessa2019_ATRT5@meta.data)

write.table(x      = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data,
            file   = "scRNAseq_Seurat_Jessa2019_ATRT5_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP visualization

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_Jessa2019_ATRT5 <- RunUMAP(object       = scRNAseq_Seurat_Jessa2019_ATRT5,
                                           dims         = 1:PCmin,
                                           n.neighbors  = 30
                                           )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_Jessa2019_ATRT5,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = choosenResolution,
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_nbUMIs.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "nFeature_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = "log10(nbUMIs)") +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_nbGenes.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "log10_nCount_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "nCount_per_Feature",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_percMito.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "percent.mito",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of ribosomal proteine
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_percRiboprot.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "percent.riboProt",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_clust_cellCyclePhase.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_Jessa2019_ATRT5,
        reduction   = "umap",
        pt.size     = 2,
        group.by    = "Phase",
        label       = FALSE,
        label.size  = 15
        ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

```


## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_Jessa2019_ATRT5_clust_nbUMIs.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_Jessa2019_ATRT5, clustering = choosenResolution, covariate = "log10_nCount_RNA", ylab = "log10(UMI count)\n")
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_Jessa2019_ATRT5_clust_nbGenes.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_Jessa2019_ATRT5, clustering = choosenResolution, covariate = "nFeature_RNA", ylab = "nbGenes")
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_Jessa2019_ATRT5_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_Jessa2019_ATRT5, clustering = choosenResolution, covariate = "nCount_per_Feature", ylab = "UMIs/gene")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_Jessa2019_ATRT5_clust_percMito.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_Jessa2019_ATRT5, clustering = choosenResolution, covariate = "percent.mito", ylab = "% mito")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_Jessa2019_ATRT5_clust_percRiboProt.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_Jessa2019_ATRT5, clustering = choosenResolution, covariate = "percent.riboProt", ylab = "% riboProt")
#dev.off()

```


## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.25
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Set Idents to choosen resolution
Idents(scRNAseq_Seurat_Jessa2019_ATRT5) <- choosenResolution

levels(Idents(scRNAseq_Seurat_Jessa2019_ATRT5))

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_Jessa2019_ATRT5.markers <- FindAllMarkers(object            = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                          min.pct           = min_pct,
                                                          logfc.threshold   = min_logfc,
                                                          min.diff.pct      = min_diffPct,
                                                          only.pos          = TRUE,
                                                          min.cells.feature = min_cellsFeature
                                                          )

head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Jessa2019_ATRT5_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_Jessa2019_ATRT5.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10#,
              #fontface = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_Jessa2019_ATRT5.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_Jessa2019_ATRT5.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_Jessa2019_ATRT5_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
          features = scRNAseq_Seurat_Jessa2019_ATRT5.markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_Jessa2019_ATRT5_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
          features = nTop_markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=20))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_Jessa2019_ATRT5_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_Jessa2019_ATRT5.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_Jessa2019_ATRT5.markers,
            file        = paste0(pathToReport,"scRNAseq_Jessa2019_ATRT5_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

## Expression of genes/gene sets of interest

### SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = "SMARCB1",
            pt.size  = 1,
            cols = c("gray","yellow","brown1","brown4")
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_SMARCB1_nebulosa.pdf"), width = 12, height = 10)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = "SMARCB1",
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

##### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_SHH_genes_nebulosa.pdf"), width = 20, height = 12)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = geneSet_ATRT_Torchia2016_SHH$Gene,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT MYC signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select MYC gene signatures
geneSet_ATRT_Torchia2016_MYC <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "MYC"
                                       )
geneSet_ATRT_Torchia2016_MYC$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = geneSet_ATRT_Torchia2016_MYC$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_MYC_genes_nebulosa.pdf"), width = 20, height = 12)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = geneSet_ATRT_Torchia2016_MYC$Gene,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = geneSet_ATRT_Torchia2016_MYC$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT SHH specific TFs (Johann et al, 2016: Table S6)

```{r SHH_markers_expression_Torchia2016}

# Select SHH TFs
geneSet_ATRT_Johann2016_SHH_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "SHH"
                                          )
geneSet_ATRT_Johann2016_SHH_TFs$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_SHH_TFs_nebulosa.pdf"), width = 20, height = 12)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT MYC specific TFs (Johann et al, 2016: Table S6)

```{r MYC_markers_expression_Torchia2016}

# Select MYC TFs
geneSet_ATRT_Johann2016_MYC_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "MYC"
                                          )
geneSet_ATRT_Johann2016_MYC_TFs$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 1,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_MYC_TFs_nebulosa.pdf"), width = 24, height = 8)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 1
        )
#dev.off()

```

##### ATRT SHH markers found heterogeneously expressed in integrated INI254, INI255 and INI267 data 

```{r heterATRTSHHmarkers}

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 24, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5","INSM1"),
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_nebulosa.pdf"), width = 20, height = 12)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5","INSM1"),
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 24, height = 12)
VlnPlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
        features = c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5","INSM1"),
        slot     = "data",
        pt.size  = 0,
        ncol     = 4
        )
#dev.off()

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_min.pdf"), width = 26, height = 08)
FeaturePlot(object   = scRNAseq_Seurat_Jessa2019_ATRT5,
            features = c("STMN4","DLL3","TTYH1"),
            pt.size  = 1,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#devtools::install_github("powellgenomicslab/Nebulosa")
library(Nebulosa)

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_min_nebulosa.pdf"), width = 26, height = 08)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = c("STMN4","DLL3","TTYH1"),
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

?plot_density

# Density plot of undifferentiated markers
undifferentiatedGeneMarkers <- c("SOX2","FABP7","OTX2","HES5","ID4","TTYH1")
densityPlot_undifferentiatedGeneMarkers_list <-  plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                              features  = undifferentiatedGeneMarkers,
                                                              slot      = "data",
                                                              method    = "wkde",
                                                              pal       = "plasma",
                                                              size      = 2,
                                                              joint     = FALSE,
                                                              adjust    = 2
                                                              )
#densityPlot_undifferentiatedGeneMarkers_list

head(densityPlot_undifferentiatedGeneMarkers_list[[1]]$data)

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[1]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_SOX2.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[2]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_FABP7.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[3]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_OTX2.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[4]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_HES5.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[5]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_ID4.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_undifferentiatedGeneMarkers_list[[6]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_TTYH1.csv",
          row.names = TRUE
          )

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_undifferentiatedGeneMarkers_nebulosa.pdf"), width = 14, height = 8)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = undifferentiatedGeneMarkers,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

# Density plot of commited markers
commitedGeneMarkers <- c("STMN4","DLL3","DCX")
densityPlot_commitedGeneMarkers_list <-  plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                      features  = commitedGeneMarkers,
                                                      slot      = "data",
                                                      method    = "wkde",
                                                      pal       = "plasma",
                                                      size      = 2,
                                                      joint     = FALSE,
                                                      adjust    = 2,
                                                      combine = FALSE
                                                      )
#densityPlot_commitedGeneMarkers_list

# Export table for source data
write.csv(x = densityPlot_commitedGeneMarkers_list[[1]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_STMN4.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_commitedGeneMarkers_list[[2]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_DLL3.csv",
          row.names = TRUE
          )

# Export table for source data
write.csv(x = densityPlot_commitedGeneMarkers_list[[3]]$data,
          file = "LobonIglesias_sourceData/SuppFig9_DCX.csv",
          row.names = TRUE
          )

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_commitedGeneMarkers_nebulosa.pdf"), width = 14, height = 4)
plot_density(object    = scRNAseq_Seurat_Jessa2019_ATRT5,
             features  = commitedGeneMarkers,
             slot      = "data",
             method    = "wkde",
             pal       = "plasma",
             size      = 2,
             joint     = FALSE,
             adjust    = 2
             )
#dev.off()

```

```{r supplementaryFig}

library(patchwork)

suppFig7bis_Aa <- wrap_plots(densityPlot_undifferentiatedGeneMarkers_list[[1]],
                             densityPlot_undifferentiatedGeneMarkers_list[[2]],
                             densityPlot_undifferentiatedGeneMarkers_list[[3]],
                             densityPlot_undifferentiatedGeneMarkers_list[[4]],
                             densityPlot_undifferentiatedGeneMarkers_list[[5]],
                             densityPlot_undifferentiatedGeneMarkers_list[[6]],
                             ncol = 3
                             ) +
    plot_annotation(title = "Undifferentiated markers"
                    ) &
    theme(text = element_text(size = 6),
          plot.title = element_text(face = "bold"),
          legend.position = "right",
          legend.key.height = unit(0.75,"cm"),
          legend.key.width = unit(0.25,"cm")
          )
#suppFig7bis_Aa

suppFig7bis_Ab <- wrap_plots(densityPlot_commitedGeneMarkers_list[[1]],
                            densityPlot_commitedGeneMarkers_list[[2]],
                            densityPlot_commitedGeneMarkers_list[[3]],
                            ncol = 3
                            ) +
    plot_annotation(title = "Committed markers") &
    theme(text = element_text(size = 6),
          plot.title = element_text(face = "bold"),
          legend.position = "right",
          legend.key.height = unit(0.75,"cm"),
          legend.key.width = unit(0.25,"cm")
          )
#suppFig7bis_Ab

suppFig7bis_A <- wrap_plots(suppFig7bis_Aa,
                          suppFig7bis_Ab,
                          #plot_spacer(),
                          heights = c(0.7,0.3)
                          ) 
#suppFig7bis_A

library(jpeg)
library(tiff)

SuppFig7bis_B <- readTIFF(source = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/info/SuppFig9_B.tiff",
                          native = TRUE
                          )
SuppFig7bis_B <- rasterGrob(image        = SuppFig7bis_B,
                             interpolate  = TRUE,
                             width        = 1.999999999,
                             height       = 1.5
                             )


suppFig7bis <- wrap_plots(wrap_elements(suppFig7bis_A) + labs(tag = "A"),
                          wrap_plots(plot_spacer(),
                                     wrap_plots(SuppFig7bis_B) + labs(tag = "B"),
                                     plot_spacer(),
                                     widths = c(0.333,0.333,0.333)
                                     ),
                          nrow = 2,
                          heights = c(0.75,0.25)
                          )  &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Supplementary Fig. 9\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )
#suppFig7bis

# Save figure into pdf file
ggsave(plot   = suppFig7bis,
       file   = paste0(pathToFigures,"SuppFig9new_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig9new_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

# Add Jessa et al ATRT5 metadata

```{r addMetaData}

# Import Jessa et al. ATRT5 metadata
Jessa2019_ATRT5_metadata <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/data/Jessa2019_scRNAseq_data/ATRT5/ATRT5.metadata.tsv",
                                       header = TRUE
                                       )

head(Jessa2019_ATRT5_metadata)
nrow(Jessa2019_ATRT5_metadata)

Jessa2019_ATRT5_metadata$Cluster <- factor(Jessa2019_ATRT5_metadata$Cluster)

rownames(Jessa2019_ATRT5_metadata) <- paste0(Jessa2019_ATRT5_metadata$Cell,"-1")

head(Jessa2019_ATRT5_metadata)

ggplot(data = Jessa2019_ATRT5_metadata
       ) +
    geom_point(mapping = aes(x = tSNE_1, y = tSNE_2, color = Cluster)
               ) +
    theme_bw()

scRNAseq_Seurat_Jessa2019_ATRT5_plus <- scRNAseq_Seurat_Jessa2019_ATRT5

scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data <- merge(x = scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data,
                                                        y = Jessa2019_ATRT5_metadata[,c("Cluster","PC1","PC2","tSNE_1","tSNE_2")],
                                                        by = 0,
                                                        all.x = TRUE,
                                                        all.y = FALSE
                                                        )

colnames(scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data)

scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data <- column_to_rownames(scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data, var = "Row.names")

colnames(scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data)

head(scRNAseq_Seurat_Jessa2019_ATRT5_plus@meta.data)

# Plot UMAP
pdf(file = paste0(pathToFigures,"umap_scRNAseq_Jessa2019_ATRT5_publishedClust.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_Jessa2019_ATRT5_plus,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "Cluster",
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
dev.off()

```

# Cerebro analysis (for visualization)

## Copy seurat object

```{r copy_seurat_object}

stopifnot(require(cerebroApp))

scRNAseq_Seurat_Jessa2019_ATRT5
scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- scRNAseq_Seurat_Jessa2019_ATRT5
head(scRNAseq_Seurat_Jessa2019_ATRT5_cerebro@meta.data)

```

## Clusters relationship trees

```{r clusters_relationship}

# Create vectors of clustering resolutions
groups <- paste0('RNA_snn_res.',seq(0,2,0.1))
groups

# Build phylogenetic tree of clusters
for (res in groups[-1])
{
    print(res)
    Idents(scRNAseq_Seurat_Jessa2019_ATRT5_cerebro) <- res
    scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- BuildClusterTree(object           = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                                                dims             = 1:PCmin,
                                                reorder          = FALSE,
                                                reorder.numeric  = FALSE
                                                )
    scRNAseq_Seurat_Jessa2019_ATRT5_cerebro@misc$trees[[res]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cerebro@tools$BuildClusterTree
}

```

## Compute percentage of mitochondrial and ribosomal transcripts

```{r cerebro_mito_ribo}

# % of mito and ribo 
scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- addPercentMtRibo(object             = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                                                            organism           = 'hg',
                                                            gene_nomenclature  = 'name'
                                                            )

```

## Get most expressed genes

```{r cerebro_most_expressed_genes}

# Most expressed genes
scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- getMostExpressedGenes(object  = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                                                                 assay   = 'RNA',
                                                                 groups  = groups
                                                                 )

```

## Get marker genes

```{r cerebro_marker_genes}

# Mark genes
scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- getMarkerGenes(object    = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                                                          assay     = 'RNA',
                                                          organism  = 'hg',
                                                          groups    = groups,
                                                          name      = 'scRNAseq_Jessa2019_ATRT5',
                                                          only_pos  = TRUE
                                                          )

```

## Perform pathway enrichment on marker genes

```{r cerebro_pathway_enrichment_on_markers}

# Pathway analysis
scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- getEnrichedPathways(object              = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                                                               marker_genes_input  = 'scRNAseq_Jessa2019_ATRT5',
                                                               adj_p_cutoff        = 0.01,
                                                               max_terms           = 100
                                                               )

```

## Gene set enrichment

```{r cerebro_gene_set_enrichment}

# Pathway analysis
#scRNAseq_Seurat_Jessa2019_ATRT5_cerebro <- performGeneSetEnrichmentAnalysis(object    = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
 #                                                           assay     = "RNA",
  #                                                          GMT_file  = "xx/xx/msigdb.v5.2.symbols_mouse.gmt",
   #                                                         groups    = groups
    #                                                        )

```


## Export to cerebro format

```{r export_to_cerebro_format}

colnames(scRNAseq_Seurat_Jessa2019_ATRT5_cerebro@meta.data)

# Export to cerebro format
exportFromSeurat(object             = scRNAseq_Seurat_Jessa2019_ATRT5_cerebro,
                 assay              = 'RNA',
                 slot               = 'data',
                 file               = paste0(pathToReport,'scRNAseq_Seurat_Jessa2019_ATRT5.crb'),
                 experiment_name    = 'scRNAseq_Jessa2019_ATRT5',
                 organism           = 'mm',
                 groups             = groups,
                 nUMI               = 'nCount_RNA',
                 nGene              = 'nFeature_RNA',
                 add_all_meta_data  = TRUE,
                 verbose            = FALSE
                 )

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToReport,'scRNAseq_Seurat_Jessa2019_ATRT5.crb'),
              zerkaloDir = 'mandrian_cellOrigin_ATRT'
              )

```

# Trajectory Inference Analysis

## ElpiGraph method

### Trajectory reconstruction for all clusters

#### Prepare matrix

```{r cellCycle}

# Scaled matrix of all genes
head(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix[,1:3])
dim(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix[,1:3])

# Scaled matrix of highly variable genes
head(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes[,1:3])
dim(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes[,1:3])

# Scaled matrix of DEGs
scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs <- unique(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix[scRNAseq_Seurat_Jessa2019_ATRT5.markers$gene,])
dim(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs)

```

##### Principal tree

```{r cellCycle}

computeElasticPrincipalTree_func <- function(matrix)
{
    computeElasticPrincipalTree(X                       = t(matrix),
                                NumNodes                = 30,
                                Lambda                  = .03,
                                Mu                      = 0.1,
                                Do_PCA                  = TRUE,
                                CenterData              = TRUE,
                                drawAccuracyComplexity  = FALSE,
                                drawPCAView             = FALSE,
                                drawEnergy              = FALSE
                                )
}


# Construct a principal Tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)

# Check elpigraph object
str(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_all_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal Tree
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs)

# Check elpigraph object
str(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_all_DEGs_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree[[length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

computeElasticPrincipalCurve_func <- function(matrix)
{
    computeElasticPrincipalCurve(X                      = t(matrix),
                                 NumNodes               = 30,
                                 Lambda                 = .03,
                                 Mu                     = 0.1,
                                 Do_PCA                 = TRUE,
                                 CenterData             = TRUE,
                                 drawAccuracyComplexity = FALSE,
                                 drawPCAView            = FALSE,
                                 drawEnergy             = FALSE
                                 )
}

# Construct a principal Curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)

# Check elpigraph object
str(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_all_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal Curve
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs)

# Check elpigraph object
str(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_all_DEGs_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve[[length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 0, 1, 5, 7 and 11

#### Extract cluster 0, 1, 5, 7 and 11 seurat object and matrix

```{r extract_cluster0135_object}

# Select cluster 0, 1, 5, 7 and 11 marker genes
head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

cluster_015711_DEG <- rownames(filter(scRNAseq_Seurat_Jessa2019_ATRT5.markers,
                                      cluster %in% c(0,1,5,7,11)))
head(cluster_015711_DEG)
length(cluster_015711_DEG)

# Subset Seurat object with cluster markers
scRNAseq_Seurat_Jessa2019_ATRT5_015711 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                            RNA_snn_res.0.8 %in% c(0,1,5,7,11),
                                            features = cluster_015711_DEG
                                            )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_015711)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_015711,
                                                                   assay = "RNA",#"integrated",
                                                                   slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix)

length(rownames(scRNAseq_Seurat_Jessa2019_ATRT5))

# Subset Seurat object with all genes
scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                     RNA_snn_res.0.8 %in% c(0,1,5,7,11)#,
                                        #features = cluster_015711_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes <- FindVariableFeatures(scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes,
                                                                   selection.method = "vst",
                                                                   nfeatures         = nFeatures
                                                                   )

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "scale.data"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711 <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_015711_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711_mvGenes <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_015711_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_015711_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

#### Principal Curve

```{r 015711_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_015711_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711_mvGenes <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_015711_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_015711_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 0, 1, 7 and 11

#### Extract cluster 0, 1, 7 and 11 seurat object and matrix

```{r extract_cluster01711_object}

# Select cluster 0, 1, 7 and 11 marker genes
head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

cluster_01711_DEG <- rownames(filter(scRNAseq_Seurat_Jessa2019_ATRT5.markers,
                                      cluster %in% c(0,1,7,11)))
head(cluster_01711_DEG)
length(cluster_01711_DEG)

# Subset Seurat object
scRNAseq_Seurat_Jessa2019_ATRT5_01711 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                            RNA_snn_res.0.8 %in% c(0,1,7,11),
                                            features = cluster_01711_DEG
                                            )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01711)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01711,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                     RNA_snn_res.0.8 %in% c(0,1,7,11)#,
                                        #features = cluster_015711_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes <- FindVariableFeatures(scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes,
                                                                   selection.method = "vst",
                                                                   nfeatures         = nFeatures
                                                                   )

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01711_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "scale.data"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711 <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01711_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711_mvGenes <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01711_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01711_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

```

#### Principal Curve

```{r 01711_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01711_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711_mvGenes <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01711_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01711_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

```

### Trajectory reconstruction for cluster 0, 1, 5, 7, 9 and 11

#### Extract cluster 0, 1, 5, 7, 9 and 11 seurat object and matrix

```{r extract_cluster0135_object}

# Select cluster 0, 1, 5, 7, 9 and 11 marker genes
head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

cluster_0157911_DEG <- rownames(filter(scRNAseq_Seurat_Jessa2019_ATRT5.markers,
                                      cluster %in% c(0,1,5,7,9,11)))
head(cluster_0157911_DEG)
length(cluster_0157911_DEG)

# Subset Seurat object
scRNAseq_Seurat_Jessa2019_ATRT5_0157911 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                             RNA_snn_res.0.8 %in% c(0,1,5,7,9,11),
                                             features = cluster_0157911_DEG
                                             )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_0157911)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_0157911,
                                                                    assay = "RNA",#"integrated",
                                                                    slot = "scale.data"#"scale.data"
                                                                    )
                                                       )

nrow(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                     RNA_snn_res.0.8 %in% c(0,1,5,7,9,11)#,
                                        #features = cluster_015711_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes <- FindVariableFeatures(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes,
                                                                   selection.method = "vst",
                                                                   nfeatures         = nFeatures
                                                                   )

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "scale.data"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911 <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_0157911_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911_mvGenes <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_0157911_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_0157911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

#### Principal Curve

```{r 0157911_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_0157911_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911_mvGenes <- computeElasticPrincipalCurve_func(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_0157911_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_0157911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 2, 7, 9 and 11

#### Extract cluster 2, 7, 9 and 11 seurat object and matrix

```{r extract_cluster29711_object}

# Select cluster 2, 7, 9 and 11 marker genes
head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

cluster_27911_DEG <- rownames(filter(scRNAseq_Seurat_Jessa2019_ATRT5.markers,
                                      cluster %in% c(2,7,9,11)))
head(cluster_27911_DEG)
length(cluster_27911_DEG)

# Subset Seurat object
scRNAseq_Seurat_Jessa2019_ATRT5_27911 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                           RNA_snn_res.0.8 %in% c(2,7,9,11),
                                           features = cluster_29711_DEG
                                           )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_27911)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_27911,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "scale.data"#"scale.data"
                                                                    )
                                                       )

nrow(scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                     RNA_snn_res.0.8 %in% c(2,7,9,11)#,
                                        #features = cluster_015711_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes <- FindVariableFeatures(scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes,
                                                                  selection.method = "vst",
                                                                  nfeatures         = nFeatures
                                                                  )

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_27911_allGenes,
                                                                          assay = "RNA",#"integrated",
                                                                          slot = "scale.data"#"scale.data"
                                                                          )
                                                              )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911 <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_27911_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_27911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911_mvGenes <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_27911_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_27911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_29711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

#### Principal Curve

```{r 29711_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_29711 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_29711_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_27911_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_29711_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_29711[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_29711)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_29711[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_29711)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_29711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_27911_mvGenes <- computeElasticPrincipalCurve_func(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_27911_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_27911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_27911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_27911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_27911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_29711@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 0, 1, 2, 5, 7, 9 and 11

#### Extract cluster 0, 1, 2, 5, 7, 9 and 11 seurat object and matrix

```{r extract_cluster01257911_object}

# Select cluster 0, 1, 2, 5, 7, 9 and 11 marker genes
head(scRNAseq_Seurat_Jessa2019_ATRT5.markers)

cluster_01257911_DEG <- rownames(filter(scRNAseq_Seurat_Jessa2019_ATRT5.markers,
                                      cluster %in% c(0,1,2,5,7,9,11)))
head(cluster_01257911_DEG)
length(cluster_01257911_DEG)

# Subset Seurat object
scRNAseq_Seurat_Jessa2019_ATRT5_01257911 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                           RNA_snn_res.0.8 %in% c(0,1,2,5,7,9,11),
                                           features = cluster_01257911_DEG
                                           )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01257911)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01257911,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "scale.data"#"scale.data"
                                                                    )
                                                       )

nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                       RNA_snn_res.0.8 %in% c(0,1,2,5,7,9,11)#,
                                        #features = cluster_015711_DEG
                                                       )

# Highly variable features
scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes <- FindVariableFeatures(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes,
                                                                  selection.method = "vst",
                                                                  nfeatures         = nFeatures
                                                                  )

# Data scaling 
scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes <- ScaleData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_01257911_allGenes,
                                                                             assay = "RNA",#"integrated",
                                                                             slot = "scale.data"#"scale.data"
                                                                             )
                                                                )
nrow(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix)

```
   
#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911 <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01257911_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911_mvGenes <- computeElasticPrincipalTree_func(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01257911_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_tree_01257911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

#### Principal Curve

```{r 01257911_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01257911_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal curve
scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911_mvGenes <- computeElasticPrincipalCurve_func(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_Jessa2019_ATRT5_01257911_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911_mvGenes[[length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911_mvGenes)]],
                BootPG    = scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911_mvGenes[1:(length(scRNAseq_Seurat_Jessa2019_ATRT5_ElPiGraph_curve_01257911_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Monocle version 3 method

#### Monocle on all clusters (using mvGenes)

```{r monocle_method}

head(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes[,1:4])
dim(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes),
                                                                               row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_mvGenes)
                                                                               )
                                                    )


nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_Jessa2019_ATRT5_cds <- learn_graph(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds,
                                              close_loop          = TRUE,
                                              learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                         prune_graph               = TRUE,
                                                                         rann.k                    = NULL
                                                                         )
                                              )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_list[[i]],
                                                  color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_list)[i])
scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot
}

library(gridExtra)

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_plot_list, ncol = 3))
#dev.off()

```

#### Monocle on all clusters (using DEGs)

```{r monocle_method}

head(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs[,1:4])
dim(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs)

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs),
                                                                               row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_scaled_matrix_DEGs)
                                                                               )
                                                    )


nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds,
                                                     reduction_method  = "UMAP"
                                                     )
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds <- learn_graph(cds = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds,
                                                   close_loop          = TRUE,
                                                   learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                              prune_graph               = TRUE,
                                                                              rann.k                    = NULL
                                                                              )
                                                   )
    scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_branch_points = FALSE,
                                                       label_cell_groups = FALSE,
                                                       label_leaves = FALSE
                                                       ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_list)[i])
scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_allClusters_DEGs_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_DEGs_cds_plot_list, ncol = 3))
#dev.off()

```


#### Monocle on cluster 0, 1, 5, 7 and 11 (using DEG)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_015711_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- learn_graph(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust015711_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust015711.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = TRUE,
           label_leaves                   = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7 and 11 (using mvGenes)

```{r monocle_method}

scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix,
                                                                   cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_015711_allGenes@meta.data),
                                                                   gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix),
                                                                                              row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_matrix)
                                                                                              )
                                                                   )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes,
                                                                num_dim       = nbPC,
                                                                scaling       = TRUE,
                                                                method        = "PCA",
                                                                norm_method   = "none",
                                                                pseudo_count  = 0
                                                                )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes,
                                                                  max_components    = 2,
                                                                  reduction_method  = "UMAP"
                                                                  )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes,
                                                                   reduction_method  = "UMAP"
                                                                   )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes <- learn_graph(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes,
                                                                 close_loop          = FALSE,
                                                                 learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                            prune_graph               = TRUE,
                                                                                            rann.k                    = NULL
                                                                                            )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust015711_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust015711_mvGenes_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           cell_size                      = 2,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

#### Monocle on cluster 0, 1, 7 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01711_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711 <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust01711_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 7 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust01711_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust0157911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust0157911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911 <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust0157911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_0157911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust0157911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_0157911_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 2, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_27911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_27911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911 <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust27911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 2, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_27911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_27911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust27911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_27911_mvGenes_plot_list, ncol = 2))
#dev.off()

```


#### Monocle on cluster 0, 1, 2, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911 <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust01257911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 2, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_Jessa2019_ATRT5_01257911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_Jessa2019_ATRT5_01257911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- align_cds(cds = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_list)[i])
    scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot_list[[length(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_Jessa2019_ATRT5_clust01257911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_Jessa2019_ATRT5_cds_01257911_mvGenes_plot_list, ncol = 2))
#dev.off()

```

# RNA velocity

## Extract meta-data for scVelo 

### Extract all cluster meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_Jessa2019_ATRT5)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_Jessa2019_ATRT5)),
          file       = "scRNAseq_Seurat_Jessa2019_ATRT5_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_Jessa2019_ATRT5, reduction = "umap"),
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_Jessa2019_ATRT5@meta.data,
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_metaData.csv"
          )

```

#### Extract cluster  0,1,5,7,11 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_Jessa2019_ATRT5_015711)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_Jessa2019_ATRT5_015711)),
          file       = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x     = Embeddings(scRNAseq_Seurat_Jessa2019_ATRT5_015711, reduction = "umap"),
          file  = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_UMAP.csv",
          )

# Extract other mata data
write.csv(x     = scRNAseq_Seurat_Jessa2019_ATRT5_015711@meta.data,
          file  = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_metaData.csv"
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_Jessa2019_ATRT5_015711, reduction = "umap"),
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_UMAP.csv",
          )

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_015711 = cluster_015711_DEG),
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_DEG.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_015711_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_015711_mvGenes_monocle3_UMAP_metaData.csv"
          )


```


#### Extract cluster  0,1,7,11 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_Jessa2019_ATRT5_01711)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_Jessa2019_ATRT5_01711)),
          file       = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x     = Embeddings(scRNAseq_Seurat_Jessa2019_ATRT5_01711, reduction = "umap"),
          file  = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_UMAP.csv",
          )

# Extract other mata data
write.csv(x     = scRNAseq_Seurat_Jessa2019_ATRT5_01711@meta.data,
          file  = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_metaData.csv"
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_Jessa2019_ATRT5_01711, reduction = "umap"),
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_UMAP.csv",
          )

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_01711 = cluster_01711_DEG),
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_DEG.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_Jessa2019_ATRT5_cds_01711_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_Jessa2019_ATRT5_01711_mvGenes_monocle3_UMAP_metaData.csv"
          )


```


scVelo analysis is performed in a python script (cellOrigin_ATRT_scRNAseq_integrated_INI254_INI255_INI267_scVelo.py), see you there



# SCENIC analysis

```{r scenic}

# Extract Seurat object for cluster 0,1,2,5,6,7,8,9,10,11,12
scRNAseq_Seurat_Jessa2019_ATRT5_012567891011 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                  RNA_snn_res.0.8 %in% c(0,1,2,5,6,7,8,9,10,11)
                                                  )

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_Jessa2019_ATRT5_012567891011)),
          file       = "scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_cellIDs.csv",
          row.names  = FALSE
          )


```

# Pseudo-bulk analysis

```{r PseudoBulk}

# Extract Seurat object for cluster 0,1,2,5,6,7,8,9,10,11,12
scRNAseq_Seurat_Jessa2019_ATRT5_012567891011 <- subset(x = scRNAseq_Seurat_Jessa2019_ATRT5,
                                                  RNA_snn_res.0.8 %in% c(0,1,2,5,6,7,8,9,10,11)
                                                  )

str(scRNAseq_Seurat_Jessa2019_ATRT5_012567891011)

# Extract expression matrix
scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_counts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_Jessa2019_ATRT5_012567891011,
                                                                                assay = "RNA",#"integrated",
                                                                                slot = "counts"#"scale.data"
                                                                                )
                                                                  )

head(scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_counts_matrix[,1:3])

# Create pseudo-bulk by summing the raw counts of all cells
scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix <- data.frame(Jessa2019_ATRT5 = rowSums(scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_counts_matrix))
scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix <- rownames_to_column(.data = scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix,
                                                                                  var = "gene"
                                                                                  )

nrow(filter(scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix, Jessa2019_ATRT5 != 0))

# Check the density distribution of pseudo counts
ggplot(data = scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix
       ) +
    geom_density(mapping = aes(x = log2(Jessa2019_ATRT5 + 1)),
                 color = "blue",
                 size = 1.5
                 ) +
    theme_bw()

# Extract cell IDs
write.csv(x          = scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix,
          file       = "scRNAseq_Seurat_Jessa2019_ATRT5_012567891011_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )

```


# Session info

```{r sessionInfo}

# Session info
sessionInfo()

```
