---
title: "Human ATRT RNAseq data"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
    fig_caption: yes
    fig_height: 8
    fig_width: 9
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
  word_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r paths}

thisProject                  <- "cellOrigin_ATRT/"

# Paths
pathToRhabdoid_U830_projects <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/"
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.6/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.5/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToThisProject            <- paste0(pathToANALYSIS_PROJECTS,thisProject)
pathToData                   <- paste0(pathToThisProject,"data/")
pathToFigures                <- paste0(pathToThisProject,"svn/analyse/script/figures/")
pathToReport                 <- paste0(pathToThisProject,"svn/analyse/report/")
pathToRNApip_RESULT          <- paste0(pathToThisProject,"RNApip_RESULT/")
pathToFeatureCounts          <- paste0(pathToThisProject,"FeatureCounts_RESULT/")

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r manage_renv, eval=FALSE}

# Initialize Renv
#renv::init()

```


```{r load_library, eval=FALSE}

# Load RNAseq library
source("load_scRNAseq_library.R")

```

```{r loading_RData, eval=FALSE}

# Load saved RData
load("cellOrigin_ATRT_scRNAseq_INI287.RData")

#save.image(file="cellOrigin_ATRT_scRNAseq_INI287.RData")

```

```{r load_plot_function}

# Load external files
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")
source("scRNAseq_functions.R")

```

```{r delete_all_objects}

# Delete all objects
#rm(list = ls())

```

# INI287 scRNA-seq

## Load data and create Seurat object

```{r load_data}

pathTo_INI287_cellRangerCount <- "/data/kdi_prod/dataset_all/2008905/export/user/count_Tumor-cells-from-Tumor_homo_sapiens/filtered_feature_bc_matrix"

# Load 10X data
scRNAseq_data_INI287_raw <- Read10X(data.dir         = pathTo_INI287_cellRangerCount,
                                    gene.column      = 2,
                                    unique.features  = TRUE
                                    )

# Create Seurat object
scRNAseq_Seurat_INI287_raw <- CreateSeuratObject(counts   = scRNAseq_data_INI287_raw,
                                                 project  = "INI287"
                                                 )

# Check Seurat object 
scRNAseq_Seurat_INI287_raw

# Export meta data for all cells (before filtering)
write.table(x      = scRNAseq_Seurat_INI287_raw@meta.data,
            file   = "scRNAseq_Seurat_INI287_raw_metadata.txt",
            quote  = FALSE,
            sep    = ","
            )

```

## Cell quality control and filtering

### Adding additional QC metrics

```{r mito_ribo_reads_fraction}

# Log10 number of UMIs
scRNAseq_Seurat_INI287_raw[["log10_nCount_RNA"]] <- log10(scRNAseq_Seurat_INI287_raw@meta.data$nCount_RNA)

# Log10 number of UMIs
scRNAseq_Seurat_INI287_raw[["nCount_per_Feature"]] <- scRNAseq_Seurat_INI287_raw@meta.data$nCount_RNA/scRNAseq_Seurat_INI287_raw@meta.data$nFeature_RNA

# Ribosomal protein reads fraction
scRNAseq_Seurat_INI287_raw[["percent.riboProt"]] <- PercentageFeatureSet(object  = scRNAseq_Seurat_INI287_raw,
                                                                         pattern = "^RP[LS].*",
                                                                         assay   = "RNA"
                                                                         )

# Mitocondrial genes percentage
scRNAseq_Seurat_INI287_raw[["percent.mito"]] <- PercentageFeatureSet(object   = scRNAseq_Seurat_INI287_raw,
                                                                     pattern  = "^MT-",
                                                                     assay    = "RNA"
                                                                     )

# Check meta data
head(scRNAseq_Seurat_INI287_raw@meta.data)

summary(scRNAseq_Seurat_INI287_raw@meta.data$percent.riboProt)

```

### Number of detected genes and number of RNA (UMIs)

```{r number_of_genes_per_cell}

# Set minimum number of genes
min_nFeature <- round(quantile(scRNAseq_Seurat_INI287_raw@meta.data$nFeature_RNA,
         probs = 0.05
         ))
min_nFeature

# Plot histogram of RNA count
#pdf(file = paste0(pathToFigures,"histogram_scRNAseq_INI287_nbRNA.pdf"), width = 12, height = 12)
plot_histogram_nbRNA(scRNAseq_Seurat_INI287_raw, log = 10, binwidth=0.05)
#dev.off()

# Plot density distribution of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_INI287_nbRNA.pdf"), width = 12, height = 12)
plot_density_nbRNA(scRNAseq_Seurat_INI287_raw, log = 10)
#dev.off()

# Plot histogram of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_INI287_nbGenes.pdf"), width = 12, height = 12)
plot_histogram_nbGenes(scRNAseq_Seurat_INI287_raw, log = 10, binwidth=0.05)
#dev.off()

# Plot density distribution of total RNA count
#pdf(file = paste0(pathToFigures,"density_scRNAseq_INI287_nbGenes.pdf"), width = 12, height = 12)
plot_density_nbGenes(scRNAseq_Seurat_INI287_raw, log = 10)
#dev.off()

# Plot scatter plot nCount_RNA and nFeature_RNA
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_INI287_nbRNA_vs_nbGene.pdf"), width = 12, height = 12)
plot_nbRNA_vs_nbGenes(scRNAseq_Seurat_INI287_raw, min_nbGene = min_nFeature)
#dev.off()

# Density distribution of the number of UMIs per gene
#pdf(file = paste0(pathToFigures,"histogram_scRNAseq_INI287_nbRNA_per_gene.pdf"), width = 12, height = 12)
plot_distribution_nbRNA_per_Gene(scRNAseq_Seurat_INI287_raw)
#dev.off()

```

### Fraction of reads mapped to mitochondrial genes

```{r mitochondrial_gene}

# Set percentage of count falling in mitochondrial genes
maxPercentMito <- 20

# Scatter plot of nb of RNA and percent of mitochondrial RNA
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_INI287_nbRNA_vs_percMito.pdf"), width = 12, height = 12)
plot_nbRNA_vs_percentMito(scRNAseq_Seurat_INI287_raw,
                          maxPercentMito = maxPercentMito
                          )
#dev.off()

```

### Fraction of reads mapped to ribosomal protein genes

```{r ribosomal_gene}

# Scatter plot nb of genes and fraction of reads map to ribosomal proteins
#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_INI287_percRiboProt_vs_nbGene.pdf"), width = 12, height = 12)
plot_percentRibo_vs_nbGene(scRNAseq_Seurat_INI287_raw, min_nbGene = min_nFeature)
#dev.off()

#pdf(file = paste0(pathToFigures,"scatterPlot_scRNAseq_INI287_percRiboProt_vs_percMito.pdf"), width = 12, height = 12)
plot_percentRibo_vs_percentMito(scRNAseq_Seurat_INI287_raw, min_nbGene = min_nFeature, max_percMito = maxPercentMito)
#dev.off()

```

## Cell filtering

```{r cell_filtering}

# Subset "good" cells
scRNAseq_Seurat_INI287 <- subset(x      = subset(x      = scRNAseq_Seurat_INI287_raw,
                                                 subset = nFeature_RNA > min_nFeature
                                                 ),
                                 subset = percent.mito < maxPercentMito
                                 )

# Plot histogram of nb of cells after each filtering step
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_cellFiltering_nbGenes_percMito.pdf"), width = 12, height = 8)
barplot_cellfiltering(scRNAseq_Seurat_INI287_raw, maxPercentMito = maxPercentMito, min_nFeature = min_nFeature)
#dev.off()

```

## Data normalization

```{r data_normalization}

# Normalization
scRNAseq_Seurat_INI287 <- NormalizeData(object                = scRNAseq_Seurat_INI287,
                                        normalization.method  = "LogNormalize",
                                        scale.factor          = 10000
                                        )

```

# Assign cell-cycle scores

```{r assign_cellCycle_scores}

# Cell cycle gene markers (present in Seurat)
str(cc.genes)
length(cc.genes$s.genes)
length(cc.genes$g2m.genes)

cc.genes$g2m.genes
cc.genes$s.genes

# Compute cell cycle scoring  
scRNAseq_Seurat_INI287 <- CellCycleScoring(object       = scRNAseq_Seurat_INI287,
                                           s.features   = cc.genes$s.genes,
                                           g2m.features = cc.genes$g2m.genes
                                           )

# Check meta data
head(scRNAseq_Seurat_INI287@meta.data)

#pdf(file = paste0(pathToFigures,"plots_cellCyclPhase_scRNAseq_INI287.pdf"), width = 16, height = 12)
plot_cellCyclePhase(scRNAseq_Seurat_INI287)
#dev.off()

```


## Dimensionality reduction (PCA)

```{r feature_selection}

# Highly variable genes
nFeatures <- 2000
variableFeatures_method <- "vst"
scRNAseq_Seurat_INI287 <- FindVariableFeatures(object            = scRNAseq_Seurat_INI287,
                                               selection.method  = variableFeatures_method,
                                               nfeatures         = nFeatures
                                               )

# Mean-variance scatter plot
#pdf(file = paste0(pathToFigures,"variableFeaturePlot_scRNAseq_INI287.pdf"), width = 12, height = 12)
VariableFeaturePlot(scRNAseq_Seurat_INI287) +
    xlab("\nAverage Expression") +
    ylab("Standard Deviation\n") +
    theme_bw() +
    theme(text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = c(0.25,0.85)
          )
#dev.off()

# Data scaling 
scRNAseq_Seurat_INI287 <- ScaleData(object = scRNAseq_Seurat_INI287,
                                    features = rownames(scRNAseq_Seurat_INI287)
                                    )

# PCA
scRNAseq_Seurat_INI287 <- RunPCA(object = scRNAseq_Seurat_INI287)

# PCs elbow plot
PCmin <- 13
#pdf(file = paste0(pathToFigures,"elbowPlot_scRNAseq_INI287_PCA.pdf"), width = 12, height = 12)
ElbowPlot(object = scRNAseq_Seurat_INI287,
          ndims = 50
          ) +
    geom_vline(xintercept = PCmin, lty = 2, color = "blue") +
    xlab("\nPC") +
    ylab("Standard Deviation\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Heatmap of each PC
#pdf(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_INI287_PCA.pdf"), width = 15, height = 11)
DimHeatmap(object = scRNAseq_Seurat_INI287,
           dims = 1:15,
           ncol = 5
           )
#dev.off()

# Plot PCA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA.pdf"), width = 12, height = 12)
DimPlot(object = scRNAseq_Seurat_INI287,
        pt.size = 2,
        group.by = NULL,
        cols = rep("black",length(levels(Idents(scRNAseq_Seurat_INI287)))),
        reduction = "pca"
        ) +
    xlab("\nPC_1") +
    ylab("PC_2\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = "none"
          )
#dev.off()

```

## Plot PCA and covariates

```{r PCA_and_covariates}

# Extract scaled matrix
scRNAseq_Seurat_INI287_scaled_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_INI287,
                                                               slot = "scale.data"
                                                               )
                                                  )
head(scRNAseq_Seurat_INI287_scaled_matrix[,1:3])

# Get HVF
scRNAseq_Seurat_INI287_mvGenes <- HVFInfo(object = scRNAseq_Seurat_INI287,
                                          selection.method = "vst"
                                          )
head(scRNAseq_Seurat_INI287_mvGenes)
nrow(scRNAseq_Seurat_INI287_mvGenes)

# Get most variable genes
scRNAseq_Seurat_INI287_mvGenes[order(scRNAseq_Seurat_INI287_mvGenes$variance.standardized, decreasing = TRUE),] %>% .[1:nFeatures,] -> mvGenes

# Subset matrix
scRNAseq_Seurat_INI287_scaled_matrix_mvGenes <- scRNAseq_Seurat_INI287_scaled_matrix[as.vector(rownames(mvGenes)),]
nrow(scRNAseq_Seurat_INI287_scaled_matrix_mvGenes)

# PCA analysis
PC        <- prcomp(data.frame(t(scRNAseq_Seurat_INI287_scaled_matrix_mvGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)

pci       <- data.frame(PC$x,scRNAseq_Seurat_INI287@meta.data)

# Plot eigen value
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_PCA_eigValPer.pdf"), width = 12, height = 12)
ggplot() +
    geom_bar(data = data.frame(PC = 1:50, eigValPer = eigValPer[1:50]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity"
             ) +
    geom_bar(data = data.frame(PC = 1:PCmin, eigValPer = eigValPer[1:PCmin]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity",
             fill = "blue"
             ) +
    xlab(label = "\nPC") +
    ylab(label = "% variability (eigen value)\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Plot PCA and nb of UMIs
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_nbUMIs.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "log10_nCount_RNA", legendTitle = "log10(nbUMIs)", title = "")
#dev.off()

# Plot PCA and nb of genes
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_nbGenes.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nFeature_RNA", legendTitle = "nbGenes", title = "")
#dev.off()

# Plot PCA and nb RNA per gene
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_nbRNA_per_gene.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nCount_per_Feature", legendTitle = "UMIs/gene", title = "")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_percMito.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.mito", legendTitle = "% mito", title = "")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_percRiboProt.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.riboProt", legendTitle = "% riboProt", title = "")
#dev.off()

# Plot PCA and cell cycle phase
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_INI287_PCA_cellCyclePhase.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = Phase),
               size    = 1#,
               #pch     = 21
               ) +
    #scale_fill_continuous(type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```


# Cell clustering

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_INI287 <- FindNeighbors(object = scRNAseq_Seurat_INI287,
                                        dims = 1:PCmin
                                        )

clustering_resolutions <- seq(from = 0, to = 2, by = 0.1)

# Clustering by modularity optimization technique
scRNAseq_Seurat_INI287 <- FindClusters(object     = scRNAseq_Seurat_INI287,
                                       resolution = clustering_resolutions
                                       )      

# Save seurat object
saveRDS(object = scRNAseq_Seurat_INI287,
        file   = paste0(pathToReport,"scRNAseq_Seurat_INI287.rds"
                        )
        )
    
```

## Choose the best resolution of clustering

```{r choose_resolution}


# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_INI287.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_INI287@meta.data,
         prefix          = "RNA_snn_res.",
         #node_colour     = "sc3_stability",
         layout          = "sugiyama",
         use_core_edges  = TRUE,
         color           = "red"
         )
#dev.off()

choosenResolution <- "RNA_snn_res.0.2"

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_INI287_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_INI287@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(choosenResolution)),
                             fill  = Phase
                             )
               ) +
    geom_text(mapping  = aes(x     = eval(as.name(choosenResolution)),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_INI287@meta.data)

write.table(x      = scRNAseq_Seurat_INI287@meta.data,
            file   = "scRNAseq_Seurat_INI287_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP visualization

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_INI287 <- RunUMAP(object       = scRNAseq_Seurat_INI287,
                                  dims         = 1:PCmin,
                                  n.neighbors  = 30
                                  )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_INI287,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = choosenResolution,
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_nbUMIs.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_INI287,
            features = "nFeature_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_nbGenes.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_INI287,
            features = "log10_nCount_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_INI287,
            features = "nCount_per_Feature",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_percMito.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_INI287,
            features = "percent.mito",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of ribosomal proteine
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_percRiboprot.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_INI287,
            features = "percent.riboProt",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_clust_cellCyclePhase.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_INI287,
        reduction   = "umap",
        pt.size     = 2,
        group.by    = "Phase",
        label       = FALSE,
        label.size  = 15
        ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

```


## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_INI287_clust_nbUMIs.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_INI287, clustering = choosenResolution, covariate = "log10_nCount_RNA", ylab = "log10(UMI count)\n")
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_INI287_clust_nbGenes.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_INI287, clustering = choosenResolution, covariate = "nFeature_RNA", ylab = "nbGenes")
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_INI287_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_INI287, clustering = choosenResolution, covariate = "nCount_per_Feature", ylab = "UMIs/gene")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_INI287_clust_percMito.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_INI287, clustering = choosenResolution, covariate = "percent.mito", ylab = "% mito")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_INI287_clust_percRiboProt.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_INI287, clustering = choosenResolution, covariate = "percent.riboProt", ylab = "% riboProt")
#dev.off()

```


## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.50
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Set Idents to choosen resolution
Idents(scRNAseq_Seurat_INI287) <- choosenResolution

levels(Idents(scRNAseq_Seurat_INI287))

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_INI287.markers <- FindAllMarkers(object            = scRNAseq_Seurat_INI287,
                                                 min.pct           = min_pct,
                                                 logfc.threshold   = min_logfc,
                                                 min.diff.pct      = min_diffPct,
                                                 only.pos          = TRUE,
                                                 min.cells.feature = min_cellsFeature
                                                 )

head(scRNAseq_Seurat_INI287.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_INI287.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10#,
              #fontface = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_INI287.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_INI287.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_INI287_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_INI287,
          features = scRNAseq_Seurat_INI287.markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_INI287_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_INI287,
          features = nTop_markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=20))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_INI287_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_INI287.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_INI287.markers,
            file        = paste0(pathToReport,"scRNAseq_INI287_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

# Expression of genes/gene sets of interest

## SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = "SMARCB1",
            pt.size  = 1,
            cols     = c("gray","yellow","brown1","brown4")
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

##### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT MYC signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select MYC gene signatures
geneSet_ATRT_Torchia2016_MYC <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "MYC"
                                       )
geneSet_ATRT_Torchia2016_MYC$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = geneSet_ATRT_Torchia2016_MYC$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = geneSet_ATRT_Torchia2016_MYC$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT SHH specific TFs (Johann et al, 2016: Table S6)

```{r SHH_markers_expression_Torchia2016}

# Select SHH TFs
geneSet_ATRT_Johann2016_SHH_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "SHH"
                                          )
geneSet_ATRT_Johann2016_SHH_TFs$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### ATRT MYC specific TFs (Johann et al, 2016: Table S6)

```{r MYC_markers_expression_Torchia2016}

# Select MYC TFs
geneSet_ATRT_Johann2016_MYC_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "MYC"
                                          )
geneSet_ATRT_Johann2016_MYC_TFs$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 1,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 1
        )
#dev.off()

```

## Cerebellum marker genes expression

### Import and explore cell types database

```{r importCellTypeMarkers}

# Import cannonical and published cell type markers
cellType_markers <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/annotation_file/cellType_markers.csv",
                               header  = TRUE,
                               sep     = "\t",
                               check.names = FALSE
                               )
head(cellType_markers)

# Gene markers per cell type
group_by(.data = filter(cellType_markers,
                        Group == "cerebellum"
                        )
         #,Cell.type
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### Cerebellum marke genes expression from Wizeman et al., 2019

#### Select Wizeman cerebellar cell types marker

```{r importCellTypeMarkers}

cellType_markers_Wizeman2019 <- filter(.data  = cellType_markers,
                                       Group  == "cerebellum",
                                       Source == "Wizeman et al., 2019"
                                       )

# Gene markers per cell type
group_by(.data = cellType_markers_Wizeman2019
         ,`Cell type`
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes  = c("hgnc_symbol"),
                             mart        = human,
                             attributesL = "mgi_symbol",
                             martL       = mouse,
                             uniqueRows  = TRUE
                             )

head(symbols_conversion)

```

#### Convert murine symbol to human symbol

```{r human_mouse_conversion}

# Convert murine symbol to human symbol
head(cellType_markers_Wizeman2019)

cellType_markers_Wizeman2019_plus <- merge(x      = cellType_markers_Wizeman2019,
                                           y      = symbols_conversion,
                                           by.x   = "Gene",
                                           by.y   = "MGI.symbol",
                                           all.x  = TRUE,
                                           all.y  = FALSE
                                           )
head(cellType_markers_Wizeman2019_plus)

```

#### Expression of cerebellar marker genes (Wezman et al., 2019)

##### Roof Plate

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RoofPlate <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Roof Plate"
                                                  )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 14)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```


##### Rhombic Lip

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RhombicLip <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Rhombic Lip"
                                                  )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### Ventricular Zone

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_VentricularZone <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                       `Cell type` == "Ventricular Zone"
                                                       )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Bergmann Glia

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_BergamannGlia <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                     `Cell type` == "Bergmann Glia"
                                                     )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### GABAergic

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_GABAergic <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                 `Cell type` == "GABAergic"
                                                 )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

### Glutamatergic marker genes expression from Carter et al., 2018

#### Select Carter et al. glutamatergic cell markers

```{r importCellTypeMarkers}

cellType_markers_Carter2018 <- filter(.data  = cellType_markers,
                                       Group  == "glutamatergic cells",
                                       Source == "Carter et al., 2018"
                                       )

# Gene markers per cell type
group_by(.data = cellType_markers_Carter2018
         ,`Cell type`
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Convert murine symbol to human symbol

```{r human_mouse_conversion}

# Convert murine symbol to human symbol
head(cellType_markers_Carter2018)

cellType_markers_Carter2018_plus <- merge(x      = cellType_markers_Carter2018,
                                           y      = symbols_conversion,
                                           by.x   = "Gene",
                                           by.y   = "MGI.symbol",
                                           all.x  = TRUE,
                                           all.y  = FALSE
                                           )
head(cellType_markers_Carter2018_plus)

```

#### Expression of glutamatergic cell markers (Carter et al., 2018)

##### Glutamatergic CN (1st cluster of fig 4C)

```{r glutamatergic_CN}

# Select glutamatergic CN markers 
cellType_markers_Carter2018_glutamatergicCN <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster1 - Glutamatergic CN"
                                                      )

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_glutamatergicCN_Carter2018_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_glutamatergicCN$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_glutamatergicCN_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_glutamatergicCN$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Glutamatergic Root (1st cluster of fig 4C)

```{r glutamatergic_Root}

# Select glutamatergic CN markers 
cellType_markers_Carter2018_glutamatergicRoot <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster2 - Root"
                                                      )
nrow(cellType_markers_Carter2018_glutamatergicRoot)

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_glutamatergicRoot_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_glutamatergicRoot$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_glutamatergicRoot_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_glutamatergicRoot$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 3rd gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_Root}

# Select Carter et al.. 2018 fig4 cluster 3 genes 
cellType_markers_Carter2018_fig4C_cluster3 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster3"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster3)

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_fig4C_cluster3_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_fig4C_cluster3$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_fig4C_cluster3_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_fig4C_cluster3$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 4th gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_cluster4}

# Select Carter et al.. 2018 fig4 cluster 4 genes 
cellType_markers_Carter2018_fig4C_cluster4 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster4"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster4)

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_fig4C_cluster4_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_fig4C_cluster4$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_fig4C_cluster4_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_fig4C_cluster4$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 5th gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_cluster5}

# Select Carter et al.. 2018 fig4 cluster 4 genes 
cellType_markers_Carter2018_fig4C_cluster5 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster5"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster5)

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_fig4C_cluster5_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_fig4C_cluster5$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_fig4C_cluster5_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_fig4C_cluster5$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### GNP (1st cluster of fig 4C)

```{r GNP}

# Select GNP markers 
cellType_markers_Carter2018_GNP <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster6 - GNP"
                                                      )
nrow(cellType_markers_Carter2018_GNP)

# Change default assay to RNA (some genes cannot find in INI287
DefaultAssay(scRNAseq_Seurat_INI287) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_GNP_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = cellType_markers_Carter2018_GNP$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_GNP_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = cellType_markers_Carter2018_GNP$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```
















# Run IKAP (scaling, PCA, clustering, differential analysis)

```{r integratedData_ikap}

# Loa IKAP functions
source("/data/users/mandrian/rhabdoid_U830_projects/programmes/IKAP/Seurat3_code/IKAP_Seurat3.R")

# Launch IKAP
scRNAseq_Seurat_INI287 <- IKAP(sobj        = scRNAseq_Seurat_INI287,
                               out.dir     = "./IKAP_INI287",
                               r.kmax.est  = 2
                               )

# Save seurat object
saveRDS(object = scRNAseq_Seurat_INI287,
        file   = paste0(pathToReport,"scRNAseq_INI287_IKAP.rds"
                        )
        )

```




## Optimum clustering and visualization

```{r import_IKAP_results_markers}

# Choosen clustering
choosenClustering <- "PC10K11"

choosenDim <- 1:10

# Change default ident
Idents(scRNAseq_Seurat_INI287) <- scRNAseq_Seurat_INI287@meta.data[[choosenClustering]]

# Run UMAP
scRNAseq_Seurat_INI287 <- RunUMAP(object      = scRNAseq_Seurat_INI287,
                                  dims        = choosenDim,
                                  n.neighbors = 30
                                  )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_INI287,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = choosenClustering,
        label       = TRUE,
        label.size  = 10
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 40),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## Number of cell per cluster

```{r numberOfCellPerCluster}

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_IKAP_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_INI287@meta.data
       ) +
    stat_count(mapping = aes(x     = PC10K11,
                             fill  = orig.ident
                             )
               ) +
    geom_text(mapping  = aes(x     = PC10K11,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              face     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

```

###  Average expression matrix

```{r average_expression_matrix}

str(scRNAseq_Seurat_INI287, max.level=3)

#Extract scaled integrated
matrix_INI287_IKAP <- AverageExpression(object     = scRNAseq_Seurat_INI287,
                                                                        #features   = icicicicicicicici,
                                                                        assays     = "RNA"#,
                                                                        #slot  = "scale.data"
                                                                        )$RNA

head(matrix_INI287_IKAP)

nrow(matrix_INI287_IKAP)

# Export matrix
write.table(x      = matrix_INI287_IKAP,
            file   = paste0(pathToReport,"matrix_INI287_IKAP.txt"),
            sep    = "\t",
            quote  = FALSE
            )

```

## Differential analysis results

```{r import_IKAP_results_markers}

# Import IKAP result of marker genes
IKAP_markers_INI287 <- readRDS(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/IKAP_INI287/markers.all.rds")

head(IKAP_markers_INI287[[choosenClustering]])

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_IKAP_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = IKAP_markers_INI287[[choosenClustering]],
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10,
              face = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Export marker table
write.table(x           = IKAP_markers_INI287[[choosenClustering]],
            file        = paste0(pathToReport,"scRNAseq_INI287_cluster_geneMarkers_IKAP.txt"),
            sep         = "\t",
            quote       = FALSE
            )

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_INI287_IKAP_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_INI287,
          features = IKAP_markers_INI287[[choosenClustering]]$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
topMarkers <- IKAP_markers_INI287[[choosenClustering]] %>%
    group_by(cluster) %>%
    top_n(n = nTop, wt = avg_logFC)
topMarkers

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_INI287_IKAP_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_INI287,
          features = topMarkers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=20))
#dev.off()

```

# Cell type identification

## Tumor cells identification

### Import and explore cell types database

```{r importCellTypeMarkers}

# Import cannonical and published cell type markers
cellType_markers <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/annotation_file/cellType_markers.csv",
                               header  = TRUE,
                               sep     = "\t"
                               )
head(cellType_markers)

# Gene markers per cell type
group_by(.data = filter(cellType_markers,
                        Cell.type == "Microglia"
                        )
         #,Cell.type
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### Import and explore geneSet database

```{r importCellTypeMarkers}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                               header  = TRUE,
                               sep     = ";"
                               )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### ATRT SHH signature expression

```{r SHH_markers_expression}

# Select SHH gene signatures
geneSet_ATRT_SHH <- filter(.data = geneSet,
                           Group == "ATRT_SHH(1)"
                           )

nrow(geneSet_ATRT_SHH)

as.vector(geneSet_ATRT_SHH$Gene)

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP_ATRT_SHH_genes.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = as.vector(geneSet_ATRT_SHH$Gene),
            pt.size  = 0.25
            ) #+
    #ggtitle(label = "SMARCB1") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_IKAP_ATRT_SHH_genes.pdf"), width = 16, height = 8)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = as.vector(geneSet_ATRT_SHH$Gene)
        ) #+
    #ggtitle(label = "SMARCB1") +
    #xlab("\nCluster") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
#dev.off()

```

### ATRT MYC signature expression

```{r MYC_markers_expression}

# Select MYC gene signatures
geneSet_ATRT_MYC <- filter(.data = geneSet,
                           Group == "ATRT_MYC(2b)"
                           )

nrow(geneSet_ATRT_MYC)

as.vector(geneSet_ATRT_MYC$Gene)

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP_ATRT_MYC_genes.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = as.vector(geneSet_ATRT_MYC$Gene),
            pt.size  = 0.25
            ) #+
    #ggtitle(label = "SMARCB1") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_IKAP_ATRT_MYC_genes.pdf"), width = 16, height = 8)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = as.vector(geneSet_ATRT_MYC$Gene)
        ) #+
    #ggtitle(label = "SMARCB1") +
    #xlab("\nCluster") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
#dev.off()

```

### SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = "SMARCB1",
            pt.size  = 1
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_IKAP_ATRT_SHH_VermisAnterior_SMARCB1.pdf"), width = 16, height = 8)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = "SMARCB1"#,
        #pt.size  = 2
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

### Canonical markers expression

#### Immune cells canonical markers

```{r immuneCells}

# Select immune cells canonical markers
cellType_markers_immuneCell_canonical <- filter(.data = cellType_markers,
                                                Source == "canonical marker",
                                                Group == "immune cells",
                                                Organism == "human"
                                                )


as.vector(cellType_markers_immuneCell_canonical$Gene)


#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP_immuneCells_canonicalMarkers.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = unique(as.vector(cellType_markers_immuneCell_canonical$Gene)),
            pt.size  = 0.25
            ) #+
    #ggtitle(label = "SMARCB1") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
#dev.off()

```

# Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes  = c("hgnc_symbol"),
                             mart        = human,
                             attributesL = "mgi_symbol",
                             martL       = mouse,
                             uniqueRows  = TRUE
                             )
head(symbols_conversion)

# Add mouse orthologous gene symbol
IKAP_markers_INI287_choosenClustering <- merge(x      = IKAP_markers_INI287[[choosenClustering]],
                                                       y      = symbols_conversion,
                                                       by.x   = "gene",
                                                       by.y   = "HGNC.symbol",
                                                       all.x  = TRUE,
                                                       all.y  = FALSE
                                                       )
head(IKAP_markers_INI287_choosenClustering)

```

#### Microglia gene set from Jessa2019

```{r immuneCells}

# Select immune cells canonical markers
cellType_markers_Microglia_Jessa2019 <- filter(.data = cellType_markers,
                                               Source == "Jessa et al., 2019",
                                               Cell.type == "Microglia",
                                               Organism == "murine"
                                               )



head(cellType_markers_Microglia_Jessa2019$Gene)

cluster <- vector()
k <- vector()
K <- vector()
pVal <- vector()
for (clust in levels(IKAP_markers_INI287_choosenClustering$cluster))
{
    g <- filter(.data    = IKAP_markers_INI287_choosenClustering,
                cluster  == clust
                )$MGI.symbol
    K <- c(K,length(g))
    G <- as.vector(cellType_markers_Microglia_Jessa2019$Gene)
    # Intersect cluster gene markers and cell type markers
    gG <- intersect(g,G)
    k <- c(k,length(gG))
    # Hypergeometric test
    p <- computeHypergeometrix_pval(x = g,
                                    y = G,
                                    N = nrow(scRNAseq_Seurat_INI287)
                                    )
    pVal <- c(pVal,p)
    cluster <- c(cluster,clust)
}
overlap_INI287_markers_Microglia_Jessa2019 <- data.frame(cluster,k,K,pVal)

# Compute q values
overlap_INI287_markers_Microglia_Jessa2019$qVal <- qvalue(p = overlap_INI287_markers_Microglia_Jessa2019$pVal)$qvalues
overlap_INI287_markers_Microglia_Jessa2019

# Keep cluster order for ggplot
overlap_INI287_markers_Microglia_Jessa2019$cluster <- factor(overlap_INI287_markers_Microglia_Jessa2019$cluster,
                                                                     levels = overlap_INI287_markers_Microglia_Jessa2019$cluster
                                                                     )

# Plot barplot
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_IKAP_markers_Microglia_Jessa2019.pdf"), width = 16, height = 8)
ggplot(data    = overlap_INI287_markers_Microglia_Jessa2019,
       mapping = aes(x = cluster,
                     y = -log10(qVal)
                     )
       ) +
    geom_bar(stat = "identity",
             fill = "brown"
             ) +
    #coord_flip() +
    xlab("\nCluster") +
    ylab("-log10(q-value)\n") +
    #ylim(0,xlim) +
    #geom_text(mapping = aes(label=paste0("(",k,"/",K,")")),
     #         size = 10,
      #        hjust = -0.1
       #       ) +
    theme_bw() +
    theme(text         = element_text(size=40, color = "black"),
          panel.border = element_blank(),
          axis.text.x  = element_text(color = "black")
          )
#dev.off()

```

### Microglia marker Soussa2017

```{r SMARCB1_expression}

# Select immune cells canonical markers
cellType_markers_Microglia_Soussa2017 <- filter(.data = cellType_markers,
                                                Source == "Sousa et al., 2017",
                                                Cell.type == "Microglia",
                                                Organism == "murine"
                                                )


# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_IKAP_Microglia_Soussa2017.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_INI287,
            features = symbols_conversion[symbols_conversion$MGI.symbol == cellType_markers_Microglia_Soussa2017$Gene,"HGNC.symbol"],
            pt.size  = 1
            ) +
    #ggtitle(label = "") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_INI287_IKAP_Microglia_Soussa2017.pdf"), width = 16, height = 8)
VlnPlot(object   = scRNAseq_Seurat_INI287,
        features = symbols_conversion[symbols_conversion$MGI.symbol == cellType_markers_Microglia_Soussa2017$Gene,"HGNC.symbol"]
        ) +
    #ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

#### Microglia gene set from Zhang2014

```{r immuneCells}

# Select immune cells canonical markers
cellType_markers_Microglia_Zhang2014 <- filter(.data = cellType_markers,
                                               Source == "Zhang et al., 2014",
                                               Cell.type == "Microglia",
                                               Organism == "murine"
                                               )


length(cellType_markers_Microglia_Zhang2014$Gene)

cluster <- vector()
k <- vector()
K <- vector()
pVal <- vector()
for (clust in levels(IKAP_markers_INI287_choosenClustering$cluster))
{
    g <- filter(.data    = IKAP_markers_INI287_choosenClustering,
                cluster  == clust
                )$MGI.symbol
    K <- c(K,length(g))
    G <- as.vector(cellType_markers_Microglia_Zhang2014$Gene)
    # Intersect cluster gene markers and cell type markers
    gG <- intersect(g,G)
    k <- c(k,length(gG))
    # Hypergeometric test
    p <- computeHypergeometrix_pval(x = g,
                                    y = G,
                                    N = nrow(scRNAseq_Seurat_INI287)
                                    )
    pVal <- c(pVal,p)
    cluster <- c(cluster,clust)
}
overlap_INI287_markers_Microglia_Zhang2014 <- data.frame(cluster,k,K,pVal)

# Compute q values
overlap_INI287_markers_Microglia_Zhang2014$qVal <- qvalue(p = overlap_INI287_markers_Microglia_Zhang2014$pVal)$qvalues

overlap_INI287_markers_Microglia_Zhang2014

# Keep cluster order for ggplot
overlap_INI287_markers_Microglia_Zhang2014$cluster <- factor(overlap_INI287_markers_Microglia_Zhang2014$cluster,
                                                                     levels = overlap_INI287_markers_Microglia_Zhang2014$cluster
                                                                     )

# Plot barplot
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_INI287_IKAP_markers_Microglia_Zhang2014.pdf"), width = 16, height = 10)
ggplot(data    = overlap_INI287_markers_Microglia_Zhang2014,
       mapping = aes(x = cluster,
                     y = -log10(qVal)
                     )
       ) +
    geom_bar(stat = "identity",
             fill = "brown"
             ) +
    #coord_flip() +
    xlab("\nCluster") +
    ylab("-log10(q-value)\n") +
    #ylim(0,xlim) +
    #geom_text(mapping = aes(label=paste0("(",k,"/",K,")")),
     #         size = 10,
      #        hjust = -0.1
       #       ) +
    theme_bw() +
    theme(text         = element_text(size=40, color = "black"),
          panel.border = element_blank(),
          axis.text.x  = element_text(color = "black")
          )
#dev.off()

```



# Cerebro analysis (for visualization)

## Copy seurat object

```{r copy_seurat_object}

stopifnot(require(cerebroApp))

scRNAseq_Seurat_INI287

scRNAseq_Seurat_INI287_cerebro <- scRNAseq_Seurat_INI287
head(scRNAseq_Seurat_INI287_cerebro@meta.data)

```

## Clusters relationship trees

```{r clusters_relationship}

# Create vectors of clustering resolutions
groups <- paste0('RNA_snn_res.',seq(0,2,0.1))
groups

# Build phylogenetic tree of clusters
for (res in groups[-1])
{
    print(res)
    Idents(scRNAseq_Seurat_INI287_cerebro) <- res
    scRNAseq_Seurat_INI287_cerebro <- BuildClusterTree(object           = scRNAseq_Seurat_INI287_cerebro,
                                                       dims             = 1:PCmin,
                                                       reorder          = FALSE,
                                                       reorder.numeric  = FALSE
                                                       )
    scRNAseq_Seurat_INI287_cerebro@misc$trees[[res]] <- scRNAseq_Seurat_INI287_cerebro@tools$BuildClusterTree
}

```

## Compute percentage of mitochondrial and ribosomal transcripts

```{r cerebro_mito_ribo}

# % of mito and ribo 
scRNAseq_Seurat_INI287_cerebro <- addPercentMtRibo(object             = scRNAseq_Seurat_INI287_cerebro,
                                                   organism           = 'hg',
                                                   gene_nomenclature  = 'name'
                                                   )

```

## Get most expressed genes

```{r cerebro_most_expressed_genes}

# Most expressed genes
scRNAseq_Seurat_INI287_cerebro <- getMostExpressedGenes(object  = scRNAseq_Seurat_INI287_cerebro,
                                                        assay   = 'RNA',
                                                        groups  = groups
                                                        )

```

## Get marker genes

```{r cerebro_marker_genes}

# Mark genes
scRNAseq_Seurat_INI287_cerebro <- getMarkerGenes(object    = scRNAseq_Seurat_INI287_cerebro,
                                                 assay     = 'RNA',
                                                 organism  = 'hg',
                                                 groups    = groups,
                                                 name      = 'scRNAseq_INI287',
                                                 only_pos  = TRUE
                                                 )

```

## Perform pathway enrichment on marker genes

```{r cerebro_pathway_enrichment_on_markers}

# Pathway analysis
scRNAseq_Seurat_INI287_cerebro <- getEnrichedPathways(object              = scRNAseq_Seurat_INI287_cerebro,
                                                      marker_genes_input  = 'scRNAseq_INI287',
                                                      adj_p_cutoff        = 0.01,
                                                      max_terms           = 100
                                                      )

```

## Gene set enrichment

```{r cerebro_gene_set_enrichment}

# Pathway analysis
#scRNAseq_Seurat_INI287_cerebro <- performGeneSetEnrichmentAnalysis(object    = scRNAseq_Seurat_INI287_cerebro,
 #                                                           assay     = "RNA",
  #                                                          GMT_file  = "xx/xx/msigdb.v5.2.symbols_mouse.gmt",
   #                                                         groups    = groups
    #                                                        )

```


## Export to cerebro format

```{r export_to_cerebro_format}

colnames(scRNAseq_Seurat_INI287_cerebro@meta.data)

# Export to cerebro format
exportFromSeurat(object             = scRNAseq_Seurat_INI287_cerebro,
                 assay              = 'RNA',
                 slot               = 'data',
                 file               = paste0(pathToReport,'scRNAseq_Seurat_INI287.crb'),
                 experiment_name    = 'scRNAseq_INI287',
                 organism           = 'mm',
                 groups             = groups,
                 nUMI               = 'nCount_RNA',
                 nGene              = 'nFeature_RNA',
                 add_all_meta_data  = TRUE,
                 verbose            = FALSE
                 )

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToReport,'scRNAseq_Seurat_INI287.crb'),
              zerkaloDir = 'mandrian_cellOrigin_ATRT'
              )

```


# CelliD analysis

## Compute MCA on INI287 data

```{r MCA}

scRNAseq_Seurat_INI287_cellID <- scRNAseq_Seurat_INI287

# Compute MCA
scRNAseq_Seurat_INI287_cellID <- RunMCA(X     = scRNAseq_Seurat_INI287_cellID,
                                        nmcs  = 30,
                                        slot  = "data"#"scale.data"
                                        )

# Plot MCA
DimPlotMC(X        = scRNAseq_Seurat_INI287_cellID,
          group.by = choosenResolution
          )

```

## Automatic cell type prediction using pre-established marker lists

### Import MSigDB C8 gene signatures

```{r sessionInfo}

# Extract C8 MSigDB gene set collection
msigdb_hsapiens_c8 <- msigdbr(species = "Homo sapiens",
                              category = "C8"
                              )
head(msigdb_hsapiens_c8)                                   

msigdb_hsapiens_c8_min <- msigdb_hsapiens_c8[,c("gene_symbol","gs_name")]

```

### Overlap with MSigDB C8 FAN_EMBRYONIC gene sets (Fan et al, 2018)

```{r FAN_EMBRYONIC}

# Select FAN_EMBRYONIC gene sets
msigdb_hsapiens_c8_FAN_EMBRYONIC <- filter(.data = msigdb_hsapiens_c8_min,
                                            grepl("FAN_EMBRYONIC_",gs_name)
                                            )
msigdb_hsapiens_c8_FAN_EMBRYONIC

# Convert data frame to list
msigdb_hsapiens_c8_FAN_EMBRYONIC_list <- as.list(unstack(msigdb_hsapiens_c8_FAN_EMBRYONIC))

# Run hypergeometric test
scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC <- RunCellHGT(X          = scRNAseq_Seurat_INI287_cellID,
                                                            pathways   = msigdb_hsapiens_c8_FAN_EMBRYONIC_list,
                                                            reduction  = "mca",
                                                            dim        = 30
                                                            )

dim(scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC)
head(scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_FAN_EMBRYONIC <- rownames(scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC)[apply(scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC, 2, which.max)]

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif <- ifelse(apply(scRNAseq_Seurat_INI287_msigdbC8_FAN_EMBRYONIC, 2, max) > hgt_thrs,
                                                     yes = cellTypes_by_msigdbC8_FAN_EMBRYONIC,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_INI287_cellID) == names(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_INI287_cellID <- AddMetaData(object    = scRNAseq_Seurat_INI287_cellID,
                                             metadata  = cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif,
                                             col.name  = "msigdb_C8_FAN_EMBRYONIC"
                                             )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_cellID_msigdb_C8_FAN_EMBRYONIC.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_INI287_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_FAN_EMBRYONIC",
        pt.size    = 3,
        cols = c("red", #astrocyte
                 "red", #astrocyte
                 "violet", #endothelial
                 "black", #immune
                 "mediumblue", #cajal_retzius
                 "blue", #excitatory_neuron
                 "darkgreen", #glial
                 "palegoldenrod", #microglia
                 "cyan", #B_cell
                 "cyan", #T_cell
                 "violet", #endothelial
                 "violet", #endothelial
                 "yellow", #myeloid
                 "cyan", #naiv_like_T_cell
                 "blue", #excitatory_neuron
                 "palegoldenrod", #microglia
                 "palegoldenrod", #microglia
                 "darkblue", #NSC_2
                 "snow3", #OLIG,
                 "brown",#OPC
                 "gray" #OPC
                 ),
        label      = FALSE
        ) #+
    #theme(legend.position = "none")
#dev.off()

```

### Overlap with MSigDB C8 DESCARTES_FETAL_CEREBELLUM gene sets (Cao et al, 2020)

```{r DESCARTES_FETAL_CEREBELLUM}

# Select DESCARTES_FETAL_CEREBELLUM gene sets
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM <- filter(.data = msigdb_hsapiens_c8_min,
                                                        grepl("DESCARTES_FETAL_CEREBELLUM_",gs_name)
                                                        )
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM

# Convert data frame to list
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list <- as.list(unstack(msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM))

# Run hypergeometric test
scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- RunCellHGT(X          = scRNAseq_Seurat_INI287_cellID,
                                                                             pathways   = msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list,
                                                                             reduction  = "mca",
                                                                             dims        = 30
                                                                             )

dim(scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM)
head(scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- rownames(scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM)[apply(scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, which.max)]

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif <- ifelse(apply(scRNAseq_Seurat_INI287_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, max) > hgt_thrs,
                                                                  yes = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM,
                                                                  no = "unassigned"
                                                                  )
head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_INI287_cellID) == names(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_INI287_cellID <- AddMetaData(object    = scRNAseq_Seurat_INI287_cellID,
                                                 metadata  = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif,
                                                 col.name  = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM"
                                                 )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_cellID_msigdb_C8_DESCARTES_FETAL_CEREBELLUM.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_INI287_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM",
        pt.size    = 2,
        cols = c("red", #astrocyte
                 "blue", #granule_neurons
                 "darkblue", #interneurons
                 "palegoldenrod", #microglia
                 "black", #oligodendrocytes,
                 "cyan", #purkinje_neurons
                 "yellow", #SLC24A4 PEX5L positive cells
                 "darkgreen", #unipolar brush cells
                 "violet", #endothelial
                 "grey" #unassigned
                 ),
        label      = FALSE
        )
#dev.off()

```

## Comparing integrated data (INI254,INI255,INI267) to INI287 using Cell-ID

### Compute MCA on integrated dataset

```{r MCA_integrated}

# Import integrated Seurat object
scRNAseq_Seurat_integrated <- readRDS(file = paste0(pathToReport,"scRNAseq_Seurat_integrated.rds")
                                      )
scRNAseq_Seurat_integrated

# Compute MCA
scRNAseq_Seurat_integrated <- RunMCA(X     = scRNAseq_Seurat_integrated,
                                     nmcs  = 30,
                                     slot  = "data"#"scale.data"
                                     )

# Plot MCA
DimPlotMC(X        = scRNAseq_Seurat_integrated,
          group.by = "integrated_snn_res.0.2"
          )

```

### Extract gene signatures from integrated data

```{r MCA_integrated}

nbSignatures <- 200

# Extracting per-cluster gene signatures
integrated_cluster_signatures <- GetGroupGeneSet(X         = scRNAseq_Seurat_integrated,
                                                dims       = 1:30,
                                                n.features = nbSignatures,
                                                group.by   = "integrated_snn_res.0.2"
                                                )
str(integrated_cluster_signatures)
head(integrated_cluster_signatures)

# Extracting per-cell gene signatures
integrated_cell_signatures <- GetCellGeneSet(X           = scRNAseq_Seurat_integrated,
                                             dims        = 1:30,
                                             n.features  = nbSignatures
                                             )
str(integrated_cell_signatures)
head(integrated_cell_signatures)

```

### Match integrated (INI254,INI255,INI267) cluster to INI287 cells

```{r group_to_cell_matching}

# Group-to-cell matching
cellID_INI287cells_vs_integratedClusters <- RunCellHGT(X         = scRNAseq_Seurat_INI287_cellID,
                                                       pathways  = integrated_cluster_signatures,
                                                       dims      = 30
                                                       )

str(cellID_INI287cells_vs_integratedClusters)
head(cellID_INI287cells_vs_integratedClusters[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_INI287cells_vs_integratedClusters)))
enrichment_thrs <- 0.25

# Density distribution of enrichement score
melt(as.matrix(cellID_INI287cells_vs_integratedClusters)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing INI287 cell to integrated cluster according to max enrichment score
cellID_INI287cells_vs_integratedClusters_prediction <- rownames(cellID_INI287cells_vs_integratedClusters)[apply(cellID_INI287cells_vs_integratedClusters, 2, which.max)]
head(cellID_INI287cells_vs_integratedClusters_prediction)
levels(factor(cellID_INI287cells_vs_integratedClusters_prediction))

# Assing INI287 cell to integrated cluster according to max enrichment score with a significant thresold
cellID_INI287cells_vs_integratedClusters_prediction_sign <- ifelse(test = apply(cellID_INI287cells_vs_integratedClusters, 2, max) > enrichment_thrs,
                                                                       yes = cellID_INI287cells_vs_integratedClusters_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_INI287cells_vs_integratedClusters_prediction_sign)

# Convert into data frame
cellID_INI287cells_vs_integratedClusters_prediction_sign_df <- data.frame(integratedCluster = cellID_INI287cells_vs_integratedClusters_prediction_sign)

cellID_INI287cells_vs_integratedClusters_prediction_sign_df <- rownames_to_column(.data = cellID_INI287cells_vs_integratedClusters_prediction_sign_df,
                                                                                      var = "INI287_cell"
                                                                                      )

head(cellID_INI287cells_vs_integratedClusters_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_INI287_cellID@meta.data) == cellID_INI287cells_vs_integratedClusters_prediction_sign_df$INI287_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_INI287_cellID <- AddMetaData(object    = scRNAseq_Seurat_INI287_cellID,
                                             metadata  = cellID_INI287cells_vs_integratedClusters_prediction_sign_df$integratedCluster,
                                             col.name  = "cellID_integratedCluster"
                                             )

colnames(scRNAseq_Seurat_INI287@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_INI287_cellID_integratedCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_INI287_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = cellID_integratedCluster,
                             fill  = eval(as.name(choosenResolution))
                             )
               ) +
    labs(fill = "INI287\ncluster") +
    xlab("\nintegrated matching cluster") +
    geom_text(mapping  = aes(x     = cellID_integratedCluster,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# CAL-ATRT cluster colors
cell_cluster_col <- c("2"  = "violet",
                      "4"  = "violetred",
                      "9"  = "blue4",
                      "3"  = "springgreen4",
                      "5"  = "springgreen",
                      "6"  = "sienna1",
                      "7"  = "sienna3",
                      "8"  = "sienna4",
                      "0"  = "skyblue",
                      "1"  = "skyblue4"
                      )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_cellID_integratedCluster.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_INI287_cellID,
        reduction   = "umap",
        pt.size     = 2,
        group.by    = "cellID_integratedCluster",
        label       = FALSE,
        label.size  = 15,
        cols        = cell_cluster_col[levels(factor(scRNAseq_Seurat_INI287@meta.data$RNA_snn_res.0.8))]
        ) +
    labs(color = "\nintegrated\nmatching\ncluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

### Match integrated (INI254,INI255,INI267) cell to INI287 cells

```{r cell_to_cell_matching}

# Group-to-cell matching
cellID_INI287cells_vs_integratedCells <- RunCellHGT(X         = scRNAseq_Seurat_INI287_cellID,
                                                    pathways  = integrated_cell_signatures,
                                                    dims      = 30
                                                    )

str(cellID_INI287cells_vs_integratedCells)
head(cellID_INI287cells_vs_integratedCells[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_INI287cells_vs_integratedCells)))
enrichment_thrs <- 0.1

# Density distribution of enrichement score
melt(as.matrix(cellID_INI287cells_vs_integratedCells)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing INI287 cell to integrated cluster according to max enrichment score
cellID_INI287cells_vs_integratedCells_prediction <- rownames(cellID_INI287cells_vs_integratedCells)[apply(cellID_INI287cells_vs_integratedCells, 2, which.max)]
head(cellID_INI287cells_vs_integratedCells_prediction)

# Assing INI287 cell to integrated cluster according to max enrichment score with a significant thresold
cellID_INI287cells_vs_integratedCells_prediction_sign <- ifelse(test = apply(cellID_INI287cells_vs_integratedCells, 2, max) > enrichment_thrs,
                                                                       yes = cellID_INI287cells_vs_integratedCells_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_INI287cells_vs_integratedCells_prediction_sign)

# Convert into data frame
cellID_INI287cells_vs_integratedCells_prediction_sign_df <- data.frame(integratedCell = cellID_INI287cells_vs_integratedCells_prediction_sign)

cellID_INI287cells_vs_integratedCells_prediction_sign_df <- rownames_to_column(.data = cellID_INI287cells_vs_integratedCells_prediction_sign_df,
                                                                                      var = "INI287_cell"
                                                                                      )

head(cellID_INI287cells_vs_integratedCells_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_INI287@meta.data) == cellID_INI287cells_vs_integratedCells_prediction_sign_df$INI287_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_INI287_cellID <- AddMetaData(object    = scRNAseq_Seurat_INI287_cellID,
                                                 metadata  = cellID_INI287cells_vs_integratedCells_prediction_sign_df$integratedCell,
                                                 col.name  = "cellID_integratedCell"
                                                 )

colnames(scRNAseq_Seurat_INI287_cellID@meta.data)

# Extract integrated cell cluster
scRNAseq_Seurat_integrated_cellCluster <- data.frame(integratedCluster = scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"],
                                                     row.names = rownames(scRNAseq_Seurat_integrated@meta.data)
                                                     )

scRNAseq_Seurat_integrated_cellCluster <- rownames_to_column(.data = scRNAseq_Seurat_integrated_cellCluster,
                                                             var = "integratedCell"
                                                             )

head(scRNAseq_Seurat_integrated_cellCluster)

# Merge integrated cell cluster to meta data
scRNAseq_Seurat_INI287_cellID@meta.data <- left_join(x      = rownames_to_column(scRNAseq_Seurat_INI287_cellID@meta.data),
                                                           y      = scRNAseq_Seurat_integrated_cellCluster,
                                                           by   = c("cellID_integratedCell" = "integratedCell")
                                                           )

rownames(scRNAseq_Seurat_INI287_cellID@meta.data) <- scRNAseq_Seurat_INI287_cellID@meta.data$rowname
scRNAseq_Seurat_INI287_cellID@meta.data$rowname <- NULL

head(scRNAseq_Seurat_INI287_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_INI287_cellID_integratedCell.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_INI287_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(choosenResolution)),
                             fill  = integratedCluster
                             )
               ) +
    labs(fill = "\nintegrated\nmatching\ncell cluster") +
    xlab("INI287 cluster") +
    geom_text(mapping  = aes(x     = eval(as.name(choosenResolution)),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_INI287_cellID_integratedCell.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_INI287_cellID,
        reduction   = "umap",
        pt.size     = 2,
        group.by    = "integratedCluster",
        label       = FALSE,
        label.size  = 15,
        cols        = cell_cluster_col
        ) +
    labs(color = "\nintegrated\nmatching\ncell cluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```



# Session info

```{r sessionInfo}

# Session info
sessionInfo()

```
