---
title: "scRNAseq ATRT-SHH (Cerebellar Ant. Lobe) - integrated analysis"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
    fig_caption: yes
    fig_height: 8
    fig_width: 9
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
  word_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r paths}

thisProject                  <- "cellOrigin_ATRT/"

# Paths
pathToRhabdoid_U830_projects <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/"
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToThisProject            <- paste0(pathToANALYSIS_PROJECTS,thisProject)
pathToData                   <- paste0(pathToThisProject,"data/")
pathToFigures                <- paste0(pathToThisProject,"svn/analyse/script/figures/")
pathToReport                 <- paste0(pathToThisProject,"svn/analyse/report/")
pathToInfo                   <- paste0(pathToThisProject,"svn/analyse/info/")
pathToRNApip_RESULT          <- paste0(pathToThisProject,"RNApip_RESULT/")
pathToFeatureCounts          <- paste0(pathToThisProject,"FeatureCounts_RESULT/")

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r manage_renv, eval=FALSE}

# Initialize Renv
#renv::init()

```


```{r load_library, eval=FALSE}

# Load RNAseq library
source("load_scRNAseq_library.R")

```

```{r reticulate, eval=FALSE}

library(reticulate)

#use_python("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/conda_env_cytoTRACE/bin/python3.6")

#use_condaenv("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/conda_env_cytoTRACE")

#use_python("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/conda_env_cytoTRACE/lib/python3.6/site-packages")

#detach("package:CytoTRACE", unload=TRUE)

library(CytoTRACE)

```

```{r loading_RData, eval=FALSE}

# Load saved RData
load("cellOrigin_ATRT_scRNAseq_integrated_INI254_INI255_INI267.RData")

#save.image(file="cellOrigin_ATRT_scRNAseq_integrated_INI254_INI255_INI267.RData")

```

```{r load_plot_function}

# Load external files
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")
source("scRNAseq_functions.R")

```

```{r delete_all_objects}

# List objects ordered by their size
source("/data/users/mandrian/programmes/lsos.R")

lsos()

# Delete all objects
#rm(list = ls())

```

# Load normalized Seurat object

## INI254

```{r load_data}

scRNAseq_Seurat_INI254 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_INI254.rds")
                                  )

```

## INI255

```{r load_data}

scRNAseq_Seurat_INI255 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_INI255.rds")
                                  )

```

## INI267

```{r load_data}

scRNAseq_Seurat_INI267 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_INI267.rds")
                                  )
```

# Integrate seurat object

```{r integrated_seurat_object}

# Create named list of seurat object
seurat_object_list <- list(INI254 = scRNAseq_Seurat_INI254,
                           INI255 = scRNAseq_Seurat_INI255,
                           INI267 = scRNAseq_Seurat_INI267
                           )
#rm(seurat_object_list)

# Find anchors
max_dim <- 30
scRNAseq_Seurat_anchors <- FindIntegrationAnchors(object.list           = seurat_object_list,
                                                  assay                 = NULL,
                                                  reference             = NULL,
                                                  anchor.features       = 2000,
                                                  scale                 = TRUE,
                                                  normalization.method  = "LogNormalize",
                                                  sct.clip.range        = NULL,
                                                  reduction             = "cca",
                                                  l2.norm               = TRUE,
                                                  dims                  = 1:max_dim,
                                                  k.anchor              = 5,
                                                  k.filter              = 200,
                                                  k.score               = 30,
                                                  max.features          = 200,
                                                  nn.method             = "rann",
                                                  eps                   = 0,
                                                  verbose               = TRUE
                                                  )

# Integrated seurat objects
scRNAseq_Seurat_integrated <- IntegrateData(anchorset              = scRNAseq_Seurat_anchors,
                                            new.assay.name         = "integrated",
                                            normalization.method   = "LogNormalize",
                                            features               = NULL,
                                            features.to.integrate  = NULL,
                                            dims                   = 1:max_dim,
                                            k.weight               = 100,
                                            weight.reduction       = NULL,
                                            sd.weight              = 1,
                                            sample.tree            = NULL,
                                            preserve.order         = FALSE,
                                            do.cpp                 = TRUE,
                                            eps                    = 0,
                                            verbose                = TRUE
                                            )

```

# Cell cylcle phase identification (Using Andrey's method)

This methode is based on pseudo-time approach.

```{r cellCycle}

# Highly variable genes
nFeatures <- 2000
variableFeatures_method <- "vst"
scRNAseq_Seurat_integrated_ICA <- FindVariableFeatures(object            = scRNAseq_Seurat_integrated,
                                                       selection.method  = variableFeatures_method,
                                                       nfeatures         = nFeatures
                                                       )

# Data scaling 
scRNAseq_Seurat_integrated_ICA <- ScaleData(object = scRNAseq_Seurat_integrated_ICA,
                                            features = rownames(scRNAseq_Seurat_integrated_ICA)
                                            )


# Run ICA
scRNAseq_Seurat_integrated_ICA <- RunICA(object           = scRNAseq_Seurat_integrated_ICA,
                                         ica.function     = "icafast",
                                         nics             = 50,
                                         ndims.print      = 1:20,
                                         nfeatures.print  = 30,
                                         reduction.name   = "ica",
                                         reduction.key    = "ica_",
                                         seed.use         = 1234
                                         )

# Save object into RDS file then delete it (too large)
saveRDS(object = scRNAseq_Seurat_integrated_ICA, file = "scRNAseq_Seurat_integrated_ICA.RDS")
rm(scRNAseq_Seurat_integrated_ICA)

# Plot ICA
#pdf(file = paste0(pathToFigures,"ica_scRNAseq_integrated.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_ICA,
        reduction   = "ica",
        dims        = c(1,2), #14
        pt.size     = 1#,
        #group.by    = paste0("RNA_snn_res.",choosenResolution),
        #label       = TRUE,
        #label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

str(scRNAseq_Seurat_integrated_ICA, levels=3)

head(scRNAseq_Seurat_integrated_ICA@reductions$ica@cell.embeddings)

library(gg3D)

# Plot 3D ICA
g_angle1 <- ggplot(data    = data.frame(scRNAseq_Seurat_integrated_ICA@reductions$ica@cell.embeddings),
                   mapping = aes(x = ica_1,
                                 y = ica_2,
                                 z = ica_5
                                 )
                   ) +
    axes_3D(theta = 60, phi = 10) +
    stat_3D(theta = 60, phi = 10) +           
    labs_3D(labs  = c("ICA1","ICA2","ICA5"),
            hjust = c(0,1,1),
            vjust = c(-0.5,-0.5,-0.5),
            angle = c(120,10,95),
            theta = 60,
            phi   = 10
            ) +
    theme_void()
#
g_angle2 <- ggplot(data    = data.frame(scRNAseq_Seurat_integrated_ICA@reductions$ica@cell.embeddings),
                   mapping = aes(x = ica_1,
                                 y = ica_5,
                                 z = ica_2
                                 )
                   ) +
    axes_3D(theta = 120, phi = 10) +
    stat_3D(theta = 120, phi = 10) +
    labs_3D(labs  = c("ICA1","ICA5","ICA2"),
            hjust = c(0,1,1),
            vjust = c(-0.5,-0.5,-0.5),
            angle = c(40,-20,90),
            theta = 120,
            phi   = 10
            ) +
    theme_void()
#
g_angle3 <- ggplot(data    = data.frame(scRNAseq_Seurat_integrated_ICA@reductions$ica@cell.embeddings),
                   mapping = aes(x = ica_5,
                                 y = ica_2,
                                 z = ica_1
                                 )
                   ) +
    axes_3D(theta = 180, phi = 40) +
    stat_3D(theta = 180, phi = 40) +           
    labs_3D(labs  = c("ICA5","ICA2","ICA1"),
            hjust = c(0,1,1),
            vjust = c(-0.5,-0.5,-0.5),
            angle = c(0,-80,80),
            theta = 180,
            phi   = 40
            ) +
    theme_void()
#
#pdf(file = paste0(pathToFigures,"ica_3d_scRNAseq_integrated.pdf"), width = 20, height = 6)
plot_grid(g_angle1,
          g_angle2,
          g_angle3,
          ncol = 3
          )
#dev.off()

# Build matrix from selected compontents
scRNAseq_Seurat_integrated_ICA_matrix <- scRNAseq_Seurat_integrated_ICA@reductions$ica@cell.embeddings[,c("ica_1","ica_2","ica_5")]

head(scRNAseq_Seurat_integrated_ICA_matrix)

```

## Compute pseudotime

```{r cellCycle}

#devtools::install_github("Albluca/distutils")
#devtools::install_github("Albluca/ElPiGraph.R")

library(ElPiGraph.R)

# Construct a principal graph
scRNAseq_Seurat_integrated_ICA_ElPiGraph <- computeElasticPrincipalCircle(X          = scRNAseq_Seurat_integrated_ICA_matrix,
                                                                          NumNodes   = 50,
                                                                          Do_PCA     = FALSE,
                                                                          nReps      = 10,
                                                                          ProbPoint  = 0.8#,
                                        #n.cores   = 8
                                                                          )

scRNAseq_Seurat_integrated_ICA_ElPiGraph

# Plot principal graph
#pdf(file = paste0(pathToFigures,"epg_scRNAseq_integrated.pdf"), width = 14, height = 12)
PlotPG(X         = scRNAseq_Seurat_integrated_ICA_matrix,
       TargetPG  = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[length(scRNAseq_Seurat_integrated_ICA_ElPiGraph)]],
       BootPG    = scRNAseq_Seurat_integrated_ICA_ElPiGraph[1:(length(scRNAseq_Seurat_integrated_ICA_ElPiGraph)-1)],
       PointSize = 1.5,
       p.alpha = 0.5
       )
#dev.off()

# Get igraph network object (to use in getPseudotime function below)
scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph <- ConstructGraph(PrintGraph = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]])

scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph

# Get the substructure of interest (here the whole nodes)
scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle <- GetSubGraph(Net        = scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph,
                                                               Structure  = "circle",
                                                               Circular   = TRUE,
                                                               KeepEnds   = TRUE,
                                                               Nodes      = 50
                                                               )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle

# Project point onto graph
scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection <- project_point_onto_graph(X              = scRNAseq_Seurat_integrated_ICA_matrix,
                                                                                NodePositions  = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]]$NodePositions,
                                                                                Edges          = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]]$Edges$Edges
                                                                                )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection

# Compute pseudotime
scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime <- getPseudotime(ProjStruct = scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection,
                                                                     NodeSeq = scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle$Circle_1
                                                                     )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime

# Plot pseudotime distribution
ggplot(data = data.frame(Pt = scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime$Pt)
       ) +
    geom_histogram(mapping = aes(x = Pt),
                   fill = "grey",
                   color = "black"
                   ) +
    theme_bw()
    
```


# Linear dimensional reduction

```{r feature_selection}

# Data scaling 
scRNAseq_Seurat_integrated <- ScaleData(object = scRNAseq_Seurat_integrated,
                                        features = rownames(scRNAseq_Seurat_integrated)
                                        )

# PCA
scRNAseq_Seurat_integrated <- RunPCA(object = scRNAseq_Seurat_integrated)

# PCs elbow plot
PCmin <- 12
#pdf(file = paste0(pathToFigures,"elbowPlot_scRNAseq_integrated_PCA.pdf"), width = 12, height = 12)
ElbowPlot(object = scRNAseq_Seurat_integrated,
          ndims = 50
          ) +
    geom_vline(xintercept = PCmin, lty = 2, color = "blue") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Heatmap of each PC
#pdf(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_integrated_PCA.pdf"), width = 15, height = 12)
#png(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_integrated_PCA.png"), width = 1000, height = 800)
DimHeatmap(object = scRNAseq_Seurat_integrated,
           dims   = 1:15,
           ncol   = 5,
           raster = FALSE,
           assay  = "integrated"
           )
#dev.off()

# Plot PCA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA.pdf"), width = 12, height = 12)
DimPlot(object = scRNAseq_Seurat_integrated,
        reduction = "pca",
        group.by = "orig.ident"
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(text = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "right"
          )
#dev.off()

```

## Plot PCA and covariates

```{r PCA_and_covariates}

# Extract matrix
scRNAseq_Seurat_integrated_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated,
                                                            slot = "scale.data"                                              )                                                                 )

head(scRNAseq_Seurat_integrated_matrix[,1:3])
dim(scRNAseq_Seurat_integrated_matrix)

# PCA analysis
PC        <- prcomp(data.frame(t(scRNAseq_Seurat_integrated_matrix)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)

pci       <- data.frame(PC$x,scRNAseq_Seurat_integrated@meta.data)

# Plot eigen value
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_PCA_eigValPer.pdf"), width = 10, height = 10)
ggplot() +
    geom_bar(data = data.frame(PC = 1:50, eigValPer = eigValPer[1:50]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity"
             ) +
    geom_bar(data = data.frame(PC = 1:PCmin, eigValPer = eigValPer[1:PCmin]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity",
             fill = "blue"
             ) +
    xlab(label = "\nPC") +
    ylab(label = "% variability (eigen value)\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Plot PCA and sample
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_sample.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = orig.ident),
               size    = 1
               ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb of UMIs
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_nbUMIs.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = nCount_RNA),
               size    = 1
               ) +
    scale_colour_continuous(name = "nb of UMIs",
                            type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb of genes
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_nbGenes.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = nFeature_RNA),
               size    = 1
               ) +
    scale_colour_continuous(name = "nb of genes",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb RNA per gene
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_nbRNA_per_gene.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(colour = nCount_per_Feature),
               size    = 1
               ) +
    scale_colour_continuous(name = "UMIs/gene",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_percMito.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = percent.mito),
               size    = 1
               ) +
    scale_colour_continuous(name = "% mito",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and cell cycle phase
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_PCA_cellCyclePhase.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = Phase),
               size    = 1#,
               #pch     = 21
               ) +
    #scale_fill_continuous(type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```

## Clustering

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_integrated <- FindNeighbors(object = scRNAseq_Seurat_integrated,
                                            dims   = 1:PCmin
                                            )

# Clustering by modularity optimization technique
scRNAseq_Seurat_integrated <- FindClusters(object     = scRNAseq_Seurat_integrated,
                                           resolution = seq(from  = 0,
                                                            to    = 2,
                                                            by    = 0.1
                                                            )
                                           )      

```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_integrated.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_integrated@meta.data,
         prefix          = "integrated_snn_res.",
         #node_colour     = "sc3_stability",
         layout          = "sugiyama",
         use_core_edges  = TRUE
         )
#dev.off()

choosenResolution <- 0.2

# Clustering by modularity optimization technique
#scRNAseq_Seurat_integrated_choosenResolution <- FindClusters(object     = scRNAseq_Seurat_integrated,
 #                                                            resolution = choosenResolution
  #                                                           )      


# Set default ident
Idents(scRNAseq_Seurat_integrated) <- scRNAseq_Seurat_integrated@meta.data[["integrated_snn_res.0.2"]]

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated@meta.data
       ) +
    stat_count(mapping = aes(x     = integrated_snn_res.0.2,
                             fill  = orig.ident
                             )
               ) +
    geom_text(mapping  = aes(x     = integrated_snn_res.0.2,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              face     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
scale_fill_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_integrated@meta.data)

write.table(x      = scRNAseq_Seurat_integrated@meta.data,
            file   = "scRNAseq_Seurat_integrated_PCmin_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_integrated <- RunUMAP(object       = scRNAseq_Seurat_integrated,
                                      dims         = 1:PCmin,
                                      n.neighbors  = 30
                                      )

# Plot UMAP and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(#legend.position = "none",
        text = element_text(size = 20)
    )
#dev.off()

# Plot UMAP and sample
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_sample.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "orig.ident"
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# Save seurat object
saveRDS(object = scRNAseq_Seurat_integrated,
        file   = paste0(pathToReport,"scRNAseq_Seurat_integrated.rds"
                        )
        )

# Save as singleCellExperiment object
scRNAseq_integrated_sce <- as.SingleCellExperiment(x = scRNAseq_Seurat_integrated)

saveRDS(object = scRNAseq_integrated_sce,
        file   = paste0("scRNAseq_integrated_sce.rds"
                        )
        )

# Save as loom object
#as.loom(x         = scRNAseq_Seurat_integrated,
 #       filename  = "scRNAseq_integrated.loom"
  #      )


```

## UMAP and cell type annotation

```{r UMAP}


#scRNAseq_Seurat_integrated_cellAnnot <- scRNAseq_Seurat_integrated

# Import cell type annotation (see e-mail from Maria on Feb. 18, 2021)
#cluster_cellType <- read.csv2(file = paste0(pathToInfo,"Clustering_synthesis_2.csv"),
 #                             header = TRUE
  #                            )[,1:4]
#head(cluster_cellType)

# Import cell type annotation (see e-mail from Maria on Feb. 18, 2021)
#cluster_cellType <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_4.txt"),
 #                              header = TRUE,
  #                             sep = "\t"
   #                           )
#head(cluster_cellType)

# Import cell type annotation (see e-mail from Maria on March 31st, 2021)
#cluster_cellType <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_5bis.csv"),
 #                              header = TRUE,
  #                             sep = ";",
   #                            check.names = FALSE
    #                          )
#head(cluster_cellType)

# Import cell type annotation (see e-mail from Maria on March 31st, 2021)
#cluster_cellType <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_6.csv"),
 #                              header = TRUE,
  #                             sep = ";",
   #                            check.names = FALSE
    #                          )
#head(cluster_cellType)

#cluster_cellType <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_8_plus.csv"),
 #                              header = TRUE,
  #                             sep = ";",
   #                            check.names = FALSE
    #                          )

#cluster_cellType <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_9.csv"),
 #                              header = TRUE,
  #                             sep = ";",
   #                            check.names = FALSE
    #                          )

cluster_cellType <- read.table(file         = paste0(pathToInfo,"Clustering_synthesis_10.csv"),
                               header       = TRUE,
                               sep          = ";",
                               check.names  = FALSE
                              )

cluster_cellType

# Select clustering methods (here CCA PC MIN for integrated data and PCmin clustering)
cluster_cellType_integratedPCmin <- filter(.data = cluster_cellType,
                                           method == "CCA PC MIN",
                                           !is.na(cluster)
                                           )

# Convert cluster number to factor (otherwise join will not work correctly)
cluster_cellType_integratedPCmin$cluster <- as.factor(cluster_cellType_integratedPCmin$cluster)

# Create a new meta.data with cell type name
metaData_cellAnnot <- left_join(x = scRNAseq_Seurat_integrated@meta.data,
                                y = cluster_cellType_integratedPCmin,
                                by = c("integrated_snn_res.0.2" = "cluster")
                                )

head(metaData_cellAnnot[,c("cell_type","cell_subtype","integrated_snn_res.0.2")])

# Add cell_type and cell_subtype to Seurat object
scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object   = scRNAseq_Seurat_integrated,
                                                    metadata = metaData_cellAnnot$cell_type,
                                                    col.name = "cell_type"
                                                    )

scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object   = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata = metaData_cellAnnot$cell_subtype,
                                                    col.name = "cell_subtype"
                                                    )

scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object   = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata = metaData_cellAnnot$cell_subtype_name,
                                                    col.name = "cell_subtype_name"
                                                    )

scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object   = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata = metaData_cellAnnot$cell_subtype_shortName,
                                                    col.name = "cell_subtype_shortName"
                                                    )

scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object   = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata = metaData_cellAnnot$cell_type_shortName,
                                                    col.name = "cell_type_shortName"
                                                    )

colnames(scRNAseq_Seurat_integrated_cellAnnot@meta.data)

head(scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_subtype)

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellType.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_cellAnnot,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cell_type",
        label       = FALSE,
        label.size  = 15
        ) +
    #scale_color_manual(values = c("red","blue","green","yellow","violet","grey","brown","cyan","forestgreen","black")) +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellSubType.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_cellAnnot,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cell_subtype_name",
        label       = TRUE,
        label.size  = 5
        ) +
    #scale_color_manual(values = c("red","blue","green","yellow","violet","grey","brown","cyan","forestgreen","black")) +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellSubType_name.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_cellAnnot,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cell_subtype_shortName",
        label       = FALSE,
        label.size  = 5
        ) +
    #scale_color_manual(values = c("red","blue","green","yellow","violet","grey","brown","cyan","forestgreen","black")) +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## UMAP for presentation

```{r umap_for_presentation}

levels(factor(scRNAseq_Seurat_integrated@meta.data$integrated_snn_res.0.2))

# Plot UMAP of all clusters and cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_presentation.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15,
        cols        = c("grey40","grey40","darkred","darkgreen","darkred","blue","hotpink","hotpink","orange4","blueviolet"),
        label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
        label.box  = TRUE
        ) +
    #theme_dark() +
    theme(legend.position = "none",
        text = element_text(size = 20),
        panel.background = element_rect(fill = "black")
    )
#dev.off()

# Plot UMAP of all clusters and cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_thesisDefense.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15,
        cols        = c("skyblue","skyblue4","violet","springgreen4","violetred","springgreen","sienna1","sienna3","sienna4","blue4"),
        label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
        label.box  = TRUE
        ) +
    #theme_dark() +
    theme(legend.position = "none",
        text = element_text(size = 20),
        panel.background = element_rect(fill = "black")
    )
#dev.off()


# Plot UMAP of cluster 0,1,3 and 5 and cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_presentation_subset.pdf"), width = 12, height = 10)
DimPlot(object      = subset(scRNAseq_Seurat_integrated,
                             idents = c(0,1,3,5)
                             ),
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15,
        cols        = c("grey40","grey40","darkgreen","blue"),
        label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
        label.box  = TRUE
        ) +
    #theme_dark() +
    theme(legend.position = "none",
        text = element_text(size = 20),
        panel.background = element_rect(fill = "black")
    )
#dev.off()

# Create legend for presentation
scRNAseq_Seurat_integrated@meta.data %>%
    group_by(integrated_snn_res.0.2) %>%
    summarize(number = n())

g <- data.frame(cell_type = c("Undifferentiated","Cycling cells","Neuronal Progenitor Like 1","Neuronal Progenitor Like 2","Non-Neuronal","hypoxic-inflammatory response (HIR)"),
                number = c(7705, 2521, 1309,1090, 1696, 284)
                )

cell_type_col <- c("Undifferentiated cells"                     = "skyblue3",
                   "Cycling cells"                        = "violetred1",
                   "Neuronal progenitor like 1 (NPL1)"           = "springgreen4",
                   "Neuronal progenitor like 2 (NPL2)"           = "springgreen",
                   "Non-neuronal cells"                         = "sienna3",
                   "Hypoxia and Inflammatory Reactive cells (HIR)"  = "blue4"
                   )

cell_type_shortName_col <- c("UD"             = "skyblue3",
                             "Cycling cells"  = "violetred1",
                             "NPL1"           = "springgreen4",
                             "NPL2"           = "springgreen",
                             "Non-neuronal"   = "sienna3",
                             "HIR"            = "blue4"
                             )

p <- ggplot(data = g
            ) +
    geom_bar(mapping = aes(x = cell_type,
                           y = number,
                           fill = cell_type
                           ),
             stat = 'identity'
             ) +
    scale_fill_manual(name = "Cell Type", values = cell_type_col[levels(factor(g$cell_type))]
                      ) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black"),
          legend.key.height = unit(1.5,"cm"),
          legend.key.width = unit(1.5,"cm")#,
          #legend.direction = "horizontal"
          )

leg <- get_legend(p)

library(ggpubr)

#pdf(file = paste0(pathToFigures,"legend_scRNAseq_integrated_clust_presentation.pdf"), width = 10, height = 16)
as_ggplot(leg)
#dev.off()
  
```

## UMAP for figure

```{r umap_for_figure}

levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_subtype_name))

cell_subtype_name_col <- c("Cycling cells (G1S)"                            = "violet",
                           "Cycling cells (G2M)"                            = "violetred",
                           "Hypoxia and Inflammation Reactive cells (HIR)"  = "blue4",
                           "Neuronal progenitor like 1 (NPL1)"              = "springgreen4",
                           "Neuronal progenitor like 2 (NPL2)"              = "springgreen",
                           "Non-neuronal cells 1"                           = "sienna1",
                           "Non-neuronal cells 2"                           = "sienna3",
                           "Non-neuronal cells 3"                           = "sienna4",
                           "Undifferentiated cells 0 (UD1)"                 = "skyblue",
                           "Undifferentiated cells 1 (UD2)"                 = "skyblue4"
                           )

cell_subtype_shortName_col <- c("G1S"             = "violet",
                                "G2M"             = "violetred",
                                "HIR"             = "blue4",
                                "NPL1"            = "springgreen4",
                                "NPL2"            = "springgreen",
                                "Non-neuronal 1"  = "sienna1",
                                "Non-neuronal 2"  = "sienna3",
                                "Non-neuronal 3"  = "sienna4",
                                "UD1"             = "skyblue",
                                "UD2"             = "skyblue4"
                                )

cell_cluster_col <- c("2"  = "violet",
                      "4"  = "violetred",
                      "9"  = "blue4",
                      "3"  = "springgreen4",
                      "5"  = "springgreen",
                      "6"  = "sienna1",
                      "7"  = "sienna3",
                      "8"  = "sienna4",
                      "0"  = "skyblue",
                      "1"  = "skyblue4"
                      )

cluster_number_col <- c("2"  = "black",
                        "4"    = "black",
                        "9"    = "grey",
                        "3"    = "black",
                        "5"    = "black",
                        "6"    = "black",
                        "7"    = "black",
                        "8"    = "black",
                        "0"    = "black",
                        "1"    = "black"
                        )

# Plot UMAP of all clusters and cell type
umap_scRNAseq_clust_cellTypes <- DimPlot(object      = scRNAseq_Seurat_integrated_cellAnnot,
                                         reduction   = "umap",
                                         pt.size     = 0.25,
                                         group.by    = "cell_subtype_name",#paste0("integrated_snn_res.",choosenResolution),
                                         #group.by    = "cell_subtype_name",
                                         label       = TRUE,
                                         label.size  = paperFig_fontsize/3,
                                         #cols        = c("skyblue","skyblue4","violet","springgreen4","violetred","springgreen","sienna1","sienna3","sienna4","blue4"),
                                         cols = cell_subtype_name_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_subtype_name))],
                                         label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
                                         label.box  = TRUE
                                         ) +
    xlab("") +
    ylab("") +
    theme(legend.position = "none",
          text = element_text(size = paperFig_fontsize),
          axis.text = element_text(size = paperFig_fontsize),
          axis.ticks = element_line(size = 0.25),
          axis.line = element_line(size = 0.25),
          panel.background = element_rect(fill = "white")
          )

umap_scRNAseq_clust_cellTypes <- DimPlot(object      = scRNAseq_Seurat_integrated_cellAnnot,
                                         reduction   = "umap",
                                         pt.size     = 0.25,
                                         group.by    = "integrated_snn_res.0.2",
                                         label       = TRUE,
                                         label.size  = paperFig_fontsize/3,
                                         cols        = cell_cluster_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))],
                                         label.color = "white",
                                         label.box = TRUE
                                         #label.color = cluster_number_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))]
                                         ) +
    xlab("") +
    ylab("") +
    theme(legend.position   = "none",
          text              = element_text(size = paperFig_fontsize),
          axis.text         = element_text(size = paperFig_fontsize),
          axis.ticks        = element_line(size = 0.25),
          axis.line         = element_line(size = 0.25),
          panel.background  = element_rect(fill = "white")
          )

umap_scRNAseq_clust_cellTypes

cluster_number_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))]

# Create legend for figure
scRNAseq_Seurat_integrated_cellAnnot@meta.data %>%
    group_by(cell_subtype_name) %>%
    summarize(number = n()) ->
    cell_type_name_nb
cell_type_name_nb

cell_type_name_nb$cell_subtype_name <- factor(cell_type_name_nb$cell_subtype_name,
                                              levels = c("Neuronal progenitor like 1 (NPL1)",
                                                         "Neuronal progenitor like 2 (NPL2)",
                                                         "Undifferentiated cells 0",
                                                         "Undifferentiated cells 1",
                                                         "Cycling cells (G1S)",
                                                         "Cycling cells (G2M)",
                                                         "Hypoxia and Inflammation Reactive cells (HIR)",
                                                         "Non-neuronal cells 1",
                                                         "Non-neuronal cells 2",
                                                         "Non-neuronal cells 3"
                                                         )
                                              )

p <- ggplot(data = cell_type_name_nb
            ) +
    geom_bar(mapping = aes(x = cell_subtype_name,
                           y = number,
                           fill = cell_subtype_name
                           ),
             stat = 'identity'
             ) +
    scale_fill_manual(name = "Cell Type", values = cell_subtype_name_col[levels(factor(cell_type_name_nb$cell_subtype_name))]
                      ) +
    guides(fill = guide_legend(ncol=1)) +
    theme(text = element_text(size = paperFig_fontsize*1.5),
          panel.background = element_rect(fill = "black"),
          legend.key.height = unit(0.35,"cm"),
          legend.key.width = unit(0.25,"cm")#,
          #legend.direction = "horizontal"
          )

library(cowplot)
legend_cellTypes <- get_legend(p)

```


## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_nbUMIs.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated,
            features = "nFeature_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_nbGenes.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated,
            features = "nCount_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_nbRNA_per_gene.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated,
            features = "nCount_per_Feature",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_percMito.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated,
            features = "percent.mito",
            reduction = "umap"
            )  +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_cellCyclePhase.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "Phase",
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_clust_nbUMIs.pdf"), width = 12, height = 12)
 ggplot(data = scRNAseq_Seurat_integrated@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = log10(nCount_RNA),
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = log10(nCount_RNA)
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "log10(UMI count)\n")+
    ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_clust_nbGenes.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nFeature_RNA,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nFeature_RNA
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "# genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nCount_per_Feature,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nCount_per_Feature
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "UMIs per gene\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_clust_percMito.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = percent.mito,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = percent.mito
                               ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "% mito genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.50
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_choosenResolution.markers <- FindAllMarkers(object            = scRNAseq_Seurat_integrated,
                                                                       assay             = "RNA",
                                                                       min.pct           = min_pct,
                                                                       logfc.threshold   = min_logfc,
                                                                       min.diff.pct      = min_diffPct,
                                                                       only.pos          = TRUE,
                                                                       min.cells.feature = min_cellsFeature
                                                                       )

head(scRNAseq_Seurat_integrated_choosenResolution.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_choosenResolution.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10,
              face = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_integrated_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_integrated_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_integrated_choosenResolution,
          features = scRNAseq_Seurat_integrated_choosenResolution.markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_integrated_choosenResolution,
          features = nTop_markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=15))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_integrated_choosenResolution.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_integrated_choosenResolution.markers,
            file        = paste0(pathToReport,"scRNAseq_integrated_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

### cluster 0,1 versus all others

```{r cluster01_vs_others}

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_markers_01_vs_others <- FindMarkers(object            = scRNAseq_Seurat_integrated_choosenResolution,
                                                               assay             = "RNA",
                                                               min.pct           = min_pct,
                                                               logfc.threshold   = min_logfc,
                                                               min.diff.pct      = min_diffPct,
                                                               only.pos          = TRUE,
                                                               min.cells.feature = min_cellsFeature,
                                                               ident.1 = c(0,1)
                                                               )


head(scRNAseq_Seurat_integrated_markers_01_vs_others)

dim(scRNAseq_Seurat_integrated_markers_01_vs_others)

scRNAseq_Seurat_integrated_markers_01_vs_others$gene <- rownames(scRNAseq_Seurat_integrated_markers_01_vs_others)

scRNAseq_Seurat_integrated_markers_01_vs_others_plus <- mutate(scRNAseq_Seurat_integrated_markers_01_vs_others,
                                                               pct1_pct2 = pct.1 - pct.2
                                                               )
head(scRNAseq_Seurat_integrated_markers_01_vs_others_plus)

scRNAseq_Seurat_integrated_markers_01_vs_others_Plus <- scRNAseq_Seurat_integrated_markers_01_vs_others_plus[order(-scRNAseq_Seurat_integrated_markers_01_vs_others_plus$p_val_adj,
                                                                                                                   scRNAseq_Seurat_integrated_markers_01_vs_others_plus$pct1_pct2,
                                                                                                                   scRNAseq_Seurat_integrated_markers_01_vs_others_plus$avg_logFC
                                                                                                                   ),
                                                                                                             ]

dim(scRNAseq_Seurat_integrated_markers_01_vs_others_Plus)

scRNAseq_Seurat_integrated_markers_01_vs_others_Plus$gene

```

### SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = "SMARCB1",
            pt.size  = 1
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_choosenResolution,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}


# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

##### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "integrated"#"RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

# Select SHH genes with heterogene expression
geneSet_ATRT_Torchia2016_SHH_genes_heteroExp <- c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5")#"HES1"

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes_heteroExp.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 3
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes_heteroExp_vertical.pdf"), width = 08, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = c("STMN4","DLL3","TTYH1"),
            pt.size  = 2,
            slot     = "data",
            ncol     = 1,
            cols     = c("gray","yellow","brown1","brown4")
            ) +
    theme(text       = element_text(size = 15),
          plot.title = element_text(size = 20, face = "bold")
          )
#dev.off()


```

##### ATRT SHH specific TFs (Johann et al, 2016: Table S6)

```{r SHH_markers_expression_Torchia2016}

# Select SHH TFs
geneSet_ATRT_Johann2016_SHH_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "SHH"
                                          )
geneSet_ATRT_Johann2016_SHH_TFs$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

# Select SHH genes with heterogene expression
geneSet_ATRT_Johann2016_SHH_TFs_heteroExp <- "INSM1"

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs_heteroExp.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = geneSet_ATRT_Johann2016_SHH_TFs_heteroExp,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 1
            )
#dev.off()

```

##### ATRT SHH signature genes from Torchia et al., 2016 and Johann et al., 2016 to show intratumoral heterogeneity of ATRT-SHH expression profile

```{r SHH_markers_expression_heterogeneity}

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 22, height = 8)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = c(geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,geneSet_ATRT_Johann2016_SHH_TFs_heteroExp),
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 4
            )
#dev.off()

umap_Torchia2016_Johann2016_SHH_genes_heteroExp <- FeaturePlot(object   = scRNAseq_Seurat_integrated,
                                                               features = c(geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,geneSet_ATRT_Johann2016_SHH_TFs_heteroExp),
                                                               pt.size  = 0.1,
                                                               slot     = "data",
                                                               cols     = c("gray","yellow","brown1","brown4"),
                                                               ncol     = 4
                                                               ) +
    plot_layout(guides = "collect") &
    theme(text = element_text(size = paperFig_fontsize),
          axis.text = element_text(size = paperFig_fontsize),
          plot.title = element_text(size = paperFig_fontsize*1.5, face = 'bold')
          )
umap_Torchia2016_Johann2016_SHH_genes_heteroExp

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_min.pdf"), width = 26, height = 8)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = c("STMN4","DLL3","TTYH1"),
            pt.size  = 1,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 4
            )
#dev.off()


```

##### IRX1, OTX2, EN1 genes

```{r SHH_markers_expression_heterogeneity}

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_IRX1_OTX2_EN1.pdf"), width = 22, height = 8)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = c("IRX1","OTX2","EN1"),
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 3
            )
#dev.off()

```

##### Function for plotting heatmaps of genes of interest

```{r SHH_markers_expression_heterogeneity}

# Extract expression matrix for cluster 0135 and all genes
scRNAseq_Seurat_integrated_allGenes_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated,
                                                                                 assay = "RNA",#"integrated",
                                                                                 slot = "data"#"scale.data"
                                                                                 )
                                                                    ),
                                                 Class = "matrix"
                                                 )
nrow(scRNAseq_Seurat_integrated_allGenes_matrix)

str(scRNAseq_Seurat_integrated)

# Interesting genes 
interestingGenes <- c("IRX1","OTX2","EN1")

interestingGenes[!is.element(interestingGenes, rownames(scRNAseq_Seurat_integrated_allGenes_matrix))]

scRNAseq_Seurat_integrated_umap <- scRNAseq_Seurat_integrated@reductions$umap@cell.embeddings

# Create a list of plots
plotList_heatmap_umap_interesingGenes <- list()
for (gene in interestingGenes)
{
    # Select genes
    scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix <- scRNAseq_Seurat_integrated_allGenes_matrix[gene,]
    # Melt matrix
    scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix_melt <- melt(scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix)  
    # Merge matrix and metadata
    scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix_melt_plus <- merge(x = scRNAseq_Seurat_integrated_umap,
                                                                                  y = scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix_melt,
                                                                                  by = 0
                                                                                  )
    # Plot 
    p <- ggplot(data = scRNAseq_Seurat_integrated_allGenes_interestingGene_matrix_melt_plus,
                mapping = aes(x = UMAP_1,
                              y = UMAP_2
                              )
                ) +
        geom_point(mapping = aes(color = value),
                   size = 0.01
                   ) +
        ggtitle(label = gene) +
        scale_color_gradientn(name = "count", colours = c("grey","yellow","red","darkred")
                              ) +
        theme_void() +
        theme(legend.position = "none",
              plot.title = element_text(hjust = 0.5, size = paperFig_fontsize*1.5),
              text = element_text(size = paperFig_fontsize)
              )
    plotList_heatmap_umap_interesingGenes[[length(plotList_heatmap_umap_interesingGenes)+1]] <- p
}
names(plotList_heatmap_umap_interesingGenes) <- interestingGenes

```


## Cerebellum marker genes expression

### Import and explore cell types database

```{r importCellTypeMarkers}

# Import cannonical and published cell type markers
cellType_markers <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/annotation_file/cellType_markers.csv",
                               header  = TRUE,
                               sep     = "\t",
                               check.names = FALSE
                               )
head(cellType_markers)

# Gene markers per cell type
group_by(.data = filter(cellType_markers,
                        Group == "cerebellum"
                        )
         #,Cell.type
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### Cerebellum marke genes expression from Wizeman et al., 2019

#### Select Wizeman cerebellar cell types marker

```{r importCellTypeMarkers}

cellType_markers_Wizeman2019 <- filter(.data  = cellType_markers,
                                       Group  == "cerebellum",
                                       Source == "Wizeman et al., 2019"
                                       )

# Gene markers per cell type
group_by(.data = cellType_markers_Wizeman2019
         ,`Cell type`
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl", host = "dec2021.archive.ensembl.org")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl", host = "dec2021.archive.ensembl.org")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes  = c("hgnc_symbol"),
                             mart        = human,
                             attributesL = "mgi_symbol",
                             martL       = mouse,
                             uniqueRows  = TRUE
                             )

head(symbols_conversion)

```

#### Convert murine symbol to human symbol

```{r human_mouse_conversion}

# Convert murine symbol to human symbol
head(cellType_markers_Wizeman2019)

cellType_markers_Wizeman2019_plus <- merge(x      = cellType_markers_Wizeman2019,
                                           y      = symbols_conversion,
                                           by.x   = "Gene",
                                           by.y   = "MGI.symbol",
                                           all.x  = TRUE,
                                           all.y  = FALSE
                                           )
head(cellType_markers_Wizeman2019_plus)

```

#### Expression of cerebellar marker genes (Wezman et al., 2019)

##### Roof Plate

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RoofPlate <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Roof Plate"
                                                  )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 14)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```


##### Rhombic Lip

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RhombicLip <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Rhombic Lip"
                                                  )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### Ventricular Zone

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_VentricularZone <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                       `Cell type` == "Ventricular Zone"
                                                       )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Bergmann Glia

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_BergamannGlia <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                     `Cell type` == "Bergmann Glia"
                                                     )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### GABAergic

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_GABAergic <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                 `Cell type` == "GABAergic"
                                                 )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

### Glutamatergic marker genes expression from Carter et al., 2018

#### Select Carter et al. glutamatergic cell markers

```{r importCellTypeMarkers}

cellType_markers_Carter2018 <- filter(.data  = cellType_markers,
                                       Group  == "glutamatergic cells",
                                       Source == "Carter et al., 2018"
                                       )

# Gene markers per cell type
group_by(.data = cellType_markers_Carter2018
         ,`Cell type`
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Convert murine symbol to human symbol

```{r human_mouse_conversion}

# Convert murine symbol to human symbol
head(cellType_markers_Carter2018)

cellType_markers_Carter2018_plus <- merge(x      = cellType_markers_Carter2018,
                                           y      = symbols_conversion,
                                           by.x   = "Gene",
                                           by.y   = "MGI.symbol",
                                           all.x  = TRUE,
                                           all.y  = FALSE
                                           )
head(cellType_markers_Carter2018_plus)

```

#### Expression of glutamatergic cell markers (Carter et al., 2018)

##### Glutamatergic CN (1st cluster of fig 4C)

```{r glutamatergic_CN}

# Select glutamatergic CN markers 
cellType_markers_Carter2018_glutamatergicCN <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster1 - Glutamatergic CN"
                                                      )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_glutamatergicCN_Carter2018_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_glutamatergicCN$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_glutamatergicCN_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_glutamatergicCN$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Glutamatergic Root (1st cluster of fig 4C)

```{r glutamatergic_Root}

# Select glutamatergic CN markers 
cellType_markers_Carter2018_glutamatergicRoot <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster2 - Root"
                                                      )
nrow(cellType_markers_Carter2018_glutamatergicRoot)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_glutamatergicRoot_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_glutamatergicRoot$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_glutamatergicRoot_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_glutamatergicRoot$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 3rd gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_Root}

# Select Carter et al.. 2018 fig4 cluster 3 genes 
cellType_markers_Carter2018_fig4C_cluster3 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster3"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster3)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_fig4C_cluster3_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_fig4C_cluster3$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_fig4C_cluster3_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_fig4C_cluster3$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 4th gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_cluster4}

# Select Carter et al.. 2018 fig4 cluster 4 genes 
cellType_markers_Carter2018_fig4C_cluster4 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster4"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster4)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_fig4C_cluster4_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_fig4C_cluster4$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_fig4C_cluster4_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_fig4C_cluster4$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### 5th gene cluster of fig 4C (Carter et al., 2018)

```{r glutamatergic_cluster5}

# Select Carter et al.. 2018 fig4 cluster 4 genes 
cellType_markers_Carter2018_fig4C_cluster5 <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster5"
                                                      )
nrow(cellType_markers_Carter2018_fig4C_cluster5)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_fig4C_cluster5_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_fig4C_cluster5$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_fig4C_cluster5_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_fig4C_cluster5$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### GNP (1st cluster of fig 4C)

```{r GNP}

# Select GNP markers 
cellType_markers_Carter2018_GNP <- filter(.data       = cellType_markers_Carter2018_plus,
                                                      `Cell type` == "cluster6 - GNP"
                                                      )
nrow(cellType_markers_Carter2018_GNP)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_GNP_Carter2018_genes.pdf"), width = 10, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = cellType_markers_Carter2018_GNP$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_GNP_Carter2018_genes.pdf"), width = 10, height = 16)
VlnPlot(object   = scRNAseq_Seurat_integrated,
        features = cellType_markers_Carter2018_GNP$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```



## Cluster marker genes identified by Maria (see e-mail from Maria on March 3, 2021)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes <- c("CNPY1","EPHB1","RUNX1T1","EN2","IRX1","IRX2","OTX2","HES3","FGF8","FGF19","FZD10","WNT3A")

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clusterMarkersMaria_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated,
            features = clusterMarkerGenes,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","brown1","brown4"),
            combine  = TRUE
            ) #+
    #theme(axis.line = element_blank(),
     #     axis.text.x = element_blank(),
      #    axis.text.y = element_blank(),
       #   axis.ticks = element_blank(),
        #  axis.title.x = element_blank(),
         # axis.title.y = element_blank()
          #)
#dev.off()

scRNAseq_Seurat_integrated_cellAnnot

```

## Stacked violin plot for marker genes identified by Maria for each cell type (see e-mail from Maria on March 31, 2021 in the Clustering_synthesis_5.xlsx file)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes2 <- c("PCNA","TOP2A","BIRC5","NUSAP1","NSG1","STMN2","LHX9","GAP43","DLL3","TAGLN3","ASCL1","NEUROG1","DLX5","CD82","EDNRB","ZFP36","SPARCL1","GJA1","CYR61")
clusterMarkerGenes3 <- c("FOS","SOCS3","SOX4","SOX11", #stemness
                         "NRG1","CD82","NKAIN4", #astro like ,"GJA11","EDNRB1","EDNRB2"
                         "DLL3","PCP4","SYT1","ELAVL4","DLX6",#neuronal ,"ELAVL31","ELAV41"
                         "DNER","JAG1",#active notch pathway
                         "MCM3","MCM3","MCM3","MCM3","CDC20","CCNB1","DLGAP5"#cell cylce
                         )

# Import marker genes for cell type 
cell_type_genes <- read.table(file = paste0(pathToInfo,"Clustering_synthesis_9_cell_type_genes.csv"),
                              header = TRUE,
                              sep = ";"
                              ) %>%
    filter(include == "yes")

head(cell_type_genes)

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_cellAnnot) <- "RNA"

str(scRNAseq_Seurat_integrated_cellAnnot, max.level=3)

scRNAseq_Seurat_integrated_cellAnnot

# Set default ident
Idents(scRNAseq_Seurat_integrated_cellAnnot) <- scRNAseq_Seurat_integrated_cellAnnot@meta.data[["cell_type"]]

# Stacked violon plot
#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_genes.pdf"), width = 8, height = 14)
StackedVlnPlot(obj = scRNAseq_Seurat_integrated_cellAnnot,
               #features = c(clusterMarkerGenes2,clusterMarkerGenes3),
               features = cell_type_genes$genes,
               #plot.margin = unit(c(0,0,0,0), "cm"),
               group.by = "cell_type"
               )
#dev.off()

# Subset seurat object
scRNAseq_Seurat_integrated_cellAnnot_NPL1_NPL2_UD <- subset(scRNAseq_Seurat_integrated_cellAnnot,
                                                            cell_type %in% c("Neuronal progenitor like 1 (NPL1)","Neuronal progenitor like 2 (NPL2)","Undifferentiated cells")
                                                            )


# Subset marker genes
cell_type_genes_NPL1NPL2UD <- filter(.data = cell_type_genes,
                                     cell_type %in% c("Neuronal","Neuronal Progenitor Like 1","Neuronal Progenitor Like 2","Undifferentiated")
                                     )

cell_type_genes_NPL1NPL2UD

# Stacked violon plot
#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_NPL1NPL2UD_genes.pdf"), width = 10, height = 12)
StackedVlnPlot(obj      = scRNAseq_Seurat_integrated_cellAnnot_NPL1_NPL2_UD,
               features = cell_type_genes_NPL1NPL2UD$genes,
               group.by = "cell_type"
               )
#dev.off()

# Stacked violon plot for figure
StackedVlnPlot_cellTypes_0135 <- StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_cellAnnot_NPL1_NPL2_UD,
                                                features  = cell_type_genes_NPL1NPL2UD$genes,
                                                group.by  = "cell_type_shortName",
                                                cols      = cell_type_shortName_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot_NPL1_NPL2_UD@meta.data$cell_type_shortName))],
                                                fontsize  = paperFig_fontsize
                                                )

StackedVlnPlot_cellTypes_0135

# Subset marker genes
cell_type_genes_CCNNHIR <- filter(.data = cell_type_genes,
                                  cell_type %in% c("Cycling cells","HIR","Non-Neuronal")
                                  )

# Import cell subtype (cluster) marker genes of interest
cell_subtype_genes <- read.table(file   = paste0(pathToInfo,"Clustering_synthesis_cell_subtype_genes.csv"),
                                 header = TRUE,
                                 sep    = ";"
                                 ) %>%
    filter(include == "yes")

head(cell_subtype_genes)

# Subset marker genes of Cycling cells, HIR and Non-Neuronal cell type
cell_subtype_genes_CCNNHIR <- filter(.data = cell_subtype_genes,
                                     cell_type %in% c("Cycling cells","HIR","Non-Neuronal")
                                     )

cluster_cellType

# Subset seurat object
scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR <- subset(scRNAseq_Seurat_integrated_cellAnnot,
                                                         cell_subtype_shortName %in% c("G1S","G2M","HIR","Non-neuronal 1","Non-neuronal 2","Non-neuronal 3")
                                                         )


scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR@meta.data$cell_subtype_shortName <- factor(scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR@meta.data$cell_subtype_shortName,
                                                                                          levels = c("Non-neuronal 1","Non-neuronal 2","Non-neuronal 3","HIR","G1S","G2M")
                                                                                          )

# Stacked violon plot
#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_CCNNHIR_genes.pdf"), width = 8, height = 14)
StackedVlnPlot(obj = scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR,
               features = cell_subtype_genes_CCNNHIR$genes,
               group.by = "cell_subtype_shortName",
               cols      = cell_subtype_shortName_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR@meta.data$cell_subtype_shortName))],
               fontsize  = 20
               )
#dev.off()

stackedVlnPlot_CCNNHIR_markerGenes <- StackedVlnPlot(obj = scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR,
                                                                                       features = cell_subtype_genes_CCNNHIR$genes,
                                                                                       group.by = "cell_subtype_shortName",
                                                                                       cols      = cell_subtype_shortName_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot_CC_NN_HIR@meta.data$cell_subtype_shortName))],
                                                                                       fontsize  = paperFig_fontsize
                                                                                       )

```

```{r stackedViolinplotForPresentation}

#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_genes_presentation.pdf"), width = 14, height = 14)
StackedVlnPlot(obj           = scRNAseq_Seurat_integrated_cellAnnot,
               features      = cell_type_genes$genes,
               group.by      = "cell_type",
               cols          = c("violetred2","blue4","springgreen4","springgreen","sienna","skyblue2"),
               fontsize      = 20
               )
#dev.off()

```

```{r stackedViolinplot_cyclingCells_markers}

head(scRNAseq_Seurat_integrated_cellAnnot@meta.data)

# Subset cycling cells clusters
scRNAseq_Seurat_integrated_cellAnnot_cyclingCells <- subset(scRNAseq_Seurat_integrated_cellAnnot,
                                                            cell_type == "Cycling cells"
                                                            )

cyclingCells_clusterMarkers <- c("GINS2","MCM3","MCM6","TOP2A","BIRC5","NUSAP1","CDC20","CCNB1","PLK1","DLGAP5")

#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_cyclingCells.pdf"), width = 8, height = 16)
StackedVlnPlot(obj           = scRNAseq_Seurat_integrated_cellAnnot_cyclingCells,
               #features     = c(clusterMarkerGenes2,clusterMarkerGenes3),
               features      = cyclingCells_clusterMarkers,
               #plot.margin  = unit(c(0,0,0,0), "cm"),
               group.by      = "cell_subtype_name",
               cols          = c("violet","violetred2"),
               fontsize      = 30
               )
#dev.off()

```

```{r stackedViolinplot_nonNeuronal_markers}

head(scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_type)

# Subset cycling cells clusters
scRNAseq_Seurat_integrated_cellAnnot_nonNeuronal <- subset(scRNAseq_Seurat_integrated_cellAnnot,
                                                           cell_type == "Non-Neuronal"
                                                           )

nonNeuronal_clusterMarkers <- c("IFITM3","SERPINF1","ATP1A2","CD82","NKAIN4")

#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_nonNeuronal.pdf"), width = 8, height = 16)
StackedVlnPlot(obj           = scRNAseq_Seurat_integrated_cellAnnot_nonNeuronal,
               #features     = c(clusterMarkerGenes2,clusterMarkerGenes3),
               features      = nonNeuronal_clusterMarkers,
               #plot.margin  = unit(c(0,0,0,0), "cm"),
               group.by      = "cell_subtype_name",
               cols          = c("sienna1","sienna3","sienna4"),
               fontsize      = 30
               )
#dev.off()

```

```{r stackedViolinplotForPaper}

paperFig_fontsize <- 6

cell_type_col_fig

#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_clusterMarkersMaria_genes_fig.pdf"), width = 14, height = 10)
StackedVlnPlot_cellTypes <- StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_cellAnnot,
               #features                             = c(clusterMarkerGenes2,clusterMarkerGenes3),
                                           features  = cell_type_genes$genes,
               #plot.margin                          = unit(c(0,0,0,0), "cm"),
                                           group.by  = "cell_type",
                                           cols      = c("darkred","orange4","darkgreen","blue","darkgreen","violet","black"),
                                           fontsize  = paperFig_fontsize
                                           )
#dev.off()

```

## UMAP showing marker genes identified by Maria for each cell type (see e-mail from Maria on March 31, 2021 in the Clustering_synthesis_5.xlsx file)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes2_min <- c("PCNA","TOP2A","STMN2","DLL3","SPARCL1","ZFP36","CYR61","EDNRB")

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_cellAnnot) <- "RNA"

# Cell type marker gene gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clusterMarkerGenes2_min.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_cellAnnot,
            features = clusterMarkerGenes2_min,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","brown1","brown4"),
            combine  = TRUE
            ) #+
    #theme(axis.line = element_blank(),
     #     axis.text.x = element_blank(),
      #    axis.text.y = element_blank(),
       #   axis.ticks = element_blank(),
        #  axis.title.x = element_blank(),
         # axis.title.y = element_blank()
          #)
#dev.off()

```


## SCENIC Analysis (PCmin)

```{r import_SCENIC_AUCscore}

# Path to SCENIC result
pathToSCENICresult <- paste0(pathToThisProject,"svn/analyse/script/SCENIC_integrated/")

```

### Gene Regulatory Network

#### Import GRN data from SCENIC analysis

```{r import_SCENIC_GRN}

# Import GNR table
scRNAseq_integrated_scenic_GNR <- read.table(file = paste0(pathToSCENICresult,"scRNASeq_integrated_scenic_GRN.tsv"),
                                             header = TRUE,
                                             sep = "\t",
                                             check.names = TRUE,
                                             skip = 2
                                             )
colnames(scRNAseq_integrated_scenic_GNR) <- c("TF","MotifID","AUC","Annotation","Context","MotifSimilarityQvalue","NES","OrthologousIdentity","RankAtMax","TargetGenes")
head(scRNAseq_integrated_scenic_GNR)

```

#### Select interesting GRN

##### Notch signaling GRN: ASCL1, HES4, HES5, HES7

```{r import_SCENIC_GRN}

# Select Notch signaling regulons
scRNAseq_integrated_scenic_GNR_NotchSignaling <- filter(.data = scRNAseq_integrated_scenic_GNR,
                                                        TF %in% c("ASCL1","HES4","HES5","HES7")
                                                        )


# Remove []
scRNAseq_integrated_scenic_GNR_NotchSignaling$TargetGenes2 <- gsub(pattern ="[\\[\\]]", replacement = "", x =  scRNAseq_integrated_scenic_GNR_NotchSignaling$TargetGenes, perl=TRUE)

scRNAseq_integrated_scenic_GNR_NotchSignaling

# Separate target genes
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep <- separate(data = scRNAseq_integrated_scenic_GNR_NotchSignaling,
                                                              col = TargetGenes2,
                                                              sep = "\\),\\ \\(",
                                                              into = c(letters,paste0(letters,"1")),
                                                              fill = "right"
                                                              ) %>%
    # remove columns with all na
    dplyr::select(where(function(x) any(!is.na(x))))
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep

# Melt
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt <- melt(data=scRNAseq_integrated_scenic_GNR_NotchSignaling_sep,
     id.vars = c("TF","MotifID","AUC","Annotation","Context","MotifSimilarityQvalue","NES","OrthologousIdentity","RankAtMax","TargetGenes"),
     measure.vars = c("a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z","a1","b1","c1","d1","e1","f1","g1","h1","i1","j1","k1","l1","m1","n1","o1")
       )

head(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt)
nrow(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt)

# Remove NA
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA <- scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt[complete.cases(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt),]
nrow(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA)

# Remove columns
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA$variable <- NULL
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA$TargetGenes <- NULL

# Separate column
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep <- separate(data = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA,
                                                                            col = value,
                                                                            into = c("target","value"),
                                                                            sep = ",\\ ",
                                                                            remove = TRUE
                                                                            )

head(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep)
nrow(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep)

# Remove remaining "()"
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep$target <- gsub(pattern = "\\(",
                                                                                replacement="",
                                                                                x=scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep$target
                                                                                )

scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep$value <- gsub(pattern = "\\)",
                                                                                replacement="",
                                                                                x=scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep$value
                                                                                )

head(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep)

# Create interaction column
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep$Interaction <- "trans"

nrow(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep)

head(scRNAseq_Seurat_integrated_choosenResolution.markers)

# Merge with cluster markers
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_plus <- merge(x = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep,
                                                                              y = scRNAseq_Seurat_integrated_choosenResolution.markers,
                                                                              by.x = "target",
                                                                              by.y = "gene",
                                                                              all.x = TRUE,
                                                                              all.y = FALSE
                                                                              )

# Export table to cytoscape
write.table(x = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_plus,
            file = paste0(pathToReport,"cellOrigin_scRNAseq_integrated_scenic_GNR_NotchSignaling.txt"),
            sep ="\t",
            quote = FALSE,
            row.names = FALSE
            )

# Keep only cluster 0,1,3,5 markers
scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_deg0135 <- filter(.data = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep,
                                                                                  target %in% cluster0135_DEG
                                                                                  )

head(scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_deg0135)

# Export table to cytoscape
write.table(x = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_deg0135,
            file = paste0(pathToReport,"cellOrigin_scRNAseq_integrated_scenic_GNR_NotchSignaling_deg0135.txt"),
            sep ="\t",
            quote = FALSE,
            row.names = FALSE
            )

# Export table to cytoscape
write.table(x = scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_deg0135$target,
            file = paste0(pathToReport,"scRNAseq_integrated_scenic_GNR_NotchSignaling_sep_melt_noNA_sep_deg0135_targetGene.txt"),
            sep ="\n",
            quote = FALSE,
            row.names = FALSE
            )


```

##### Add STRING PPI

```{r string}

# Import PPI from ST
string_detailed_ppi <- read_delim(file = paste0(pathToData,"STRING/9606.protein.links.detailed.v11.0.txt.gz"),
                                  delim = " "
                                  )

head(string_detailed_ppi)

nrow(string_detailed_ppi)

# Keep only experimental ppi
string_detailed_ppi_experimental <- filter(.data = string_detailed_ppi,
                                           experimental > 0
                                           )
nrow(string_detailed_ppi_experimental)

# Import PPI info
string_ppi_info <- read_delim(file = paste0(pathToData,"STRING/9606.protein.info.v11.0.txt.gz"),
                              delim = "\t"
                              )
head(string_ppi_info)

colnames(string_ppi_info)

# Merge PPI info and PPI link
string_detailed_ppi_experimental_plus <- merge(x     = string_detailed_ppi_experimental,
                                               y     = string_ppi_info,
                                               by.x  = "protein1",
                                               by.y  = "protein_external_id",
                                               all   = FALSE
                                               )
head(string_detailed_ppi_experimental_plus)

colnames(string_detailed_ppi_experimental_plus)[11] <- "protein1_name"
colnames(string_detailed_ppi_experimental_plus)[12] <- "protein1_size"
colnames(string_detailed_ppi_experimental_plus)[13] <- "protein1_annotation"

string_detailed_ppi_experimental_Plus <- merge(x     = string_detailed_ppi_experimental_plus,
                                               y     = string_ppi_info,
                                               by.x  = "protein2",
                                               by.y  = "protein_external_id",
                                               all   = FALSE
                                               )

colnames(string_detailed_ppi_experimental_Plus)

colnames(string_detailed_ppi_experimental_Plus)[14] <- "protein2_name"
colnames(string_detailed_ppi_experimental_Plus)[15] <- "protein2_size"
colnames(string_detailed_ppi_experimental_Plus)[16] <- "protein2_annotation"

head(string_detailed_ppi_experimental_Plus)

# Export table to cytoscape
write.table(x = string_detailed_ppi_experimental_Plus,
            file = paste0(pathToReport,"string_detailed_ppi_experimental.txt"),
            sep ="\t",
            quote = FALSE,
            row.names = FALSE
            )

```

### Cellular enrichment of regulons using AUC score

#### Import AUC score matrix

```{r import_SCENIC_AUCscore}

# Import SCENIC AUC score
scRNAseq_integrated_scenic_auc_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_auc_matrix.csv"))
head(scRNAseq_integrated_scenic_auc_matrix)
dim(scRNAseq_integrated_scenic_auc_matrix)

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_auc_matrix.csv"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

# Transpose AUC matrix
scRNAseq_integrated_scenic_auc_matrix_t <- t(scRNAseq_integrated_scenic_auc_matrix)

head(scRNAseq_integrated_scenic_auc_matrix_t)

dim(scRNAseq_integrated_scenic_auc_matrix_t)

# Density distribution
scRNAseq_integrated_scenic_auc_matrix_t_melt <- melt(scRNAseq_integrated_scenic_auc_matrix_t)
head(scRNAseq_integrated_scenic_auc_matrix_t_melt)

AUCthres <- 0.01
ggplot(data = scRNAseq_integrated_scenic_auc_matrix_t_melt) +
    geom_density(mapping = aes(x = value
                               ),
                 fill = "grey"
                 ) +
    geom_vline(xintercept=AUCthres, color = "red") +
    theme_bw()

# Filtering
scRNAseq_integrated_scenic_auc_matrix_t_filt <- expFilter(data      = scRNAseq_integrated_scenic_auc_matrix_t,
                                                          threshold = AUCthres,
                                                          p         = 0.05,
                                                          graph     = TRUE
                                                          )

dim(scRNAseq_integrated_scenic_auc_matrix_t_filt)

```

#### Heatmap of AUC score

```{r SCENIC_AUCscore_heatmap}

# Most variable genes
mvRegulons <- genes.selection(data      = scRNAseq_integrated_scenic_auc_matrix_t_filt,
                              thres.num = 50
                              )

# Subset matrix
scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons <- scRNAseq_integrated_scenic_auc_matrix_t_filt[mvRegulons,]

# Check order
all(rownames(scRNAseq_Seurat_integrated@meta.data) == colnames(scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons))

# Top annotation
topAnnot_PCmin <- HeatmapAnnotation(df                      = data.frame(Cluster = as.character(scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"]), check.names = FALSE),
                                    col                     = list(Cluster = cluster_n10_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(2,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Bottom annotation
bottomAnnot_PCmin <- HeatmapAnnotation(df                      = data.frame(Patient_ID = as.character(scRNAseq_Seurat_integrated@meta.data[,"orig.ident"]), check.names = FALSE),
                                    col                     = list(Patient_ID = patient_id_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(1,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Define heatmap color
col_fun_AUCscore = colorRamp2(breaks = c(0,0.2,0.4,0.6), colors = c("cyan","wheat1","red","darkred"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             bottom_annotation           = bottomAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_split                = 6,
             row_split                   = 14,
             col                         = col_fun_AUCscore,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "AUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_SCENIC_AUCscore.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```


#### UMAP of cells based on AUC score

```{r AUCscore_umap}

# Compute UMAP
scRNAseq_integrated_scenic_auc_umap <- data.frame(umap(t(scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons)))
head(scRNAseq_integrated_scenic_auc_umap)

colnames(scRNAseq_integrated_scenic_auc_umap) <- c("UMAP1","UMAP2")
head(scRNAseq_integrated_scenic_auc_umap)

# Bind meta data
scRNAseq_integrated_scenic_auc_umap_plus <- cbind(scRNAseq_integrated_scenic_auc_umap,
                                              cluster = as.character(scRNAseq_Seurat_integrated@meta.data$integrated_snn_res.0.2)
                                              )

head(scRNAseq_integrated_scenic_auc_umap_plus)

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_SCENIC_AUCscore.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_integrated_scenic_auc_umap_plus,
       mapping = aes(x = UMAP1, y = UMAP2)
       ) +
    geom_point(mapping = aes(color = cluster
                             ),
               size = 2
               )+
    xlab("UMAP1") +
    ylab("UMAP2") +
    ggtitle(NULL) +
    theme_bw()+
    theme(text               = element_text(size = 20),
          legend.position    = "none"
          #legend.title       = element_text(face = "bold"),
          #legend.key.height  = unit(1.5,"cm")
          )
#dev.off()

```	

### Cellular enrichment of regulons using binarized AUC score

#### Import binarized AUC score matrix

```{r import_SCENIC_bin_AUCscore}

# Import SCENIC AUC score
scRNAseq_integrated_scenic_auc_matrix_bin <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_auc_matrix_bin.csv"))

head(scRNAseq_integrated_scenic_auc_matrix_bin)
dim(scRNAseq_integrated_scenic_auc_matrix_bin)

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_auc_matrix_bin.csv"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

# Transpose AUC_BIN matrix
scRNAseq_integrated_scenic_auc_matrix_bin_t <- t(scRNAseq_integrated_scenic_auc_matrix_bin)
head(scRNAseq_integrated_scenic_auc_matrix_bin_t)

dim(scRNAseq_integrated_scenic_auc_matrix_bin_t)

# Density distribution
scRNAseq_integrated_scenic_auc_matrix_bin_t_melt <- melt(scRNAseq_integrated_scenic_auc_matrix_bin_t)
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_melt)

scRNAseq_integrated_scenic_auc_matrix_bin_t_melt <- mutate(.data = scRNAseq_integrated_scenic_auc_matrix_bin_t_melt,
                                                       value = as.character(value)
                                                       )

ggplot(data = scRNAseq_integrated_scenic_auc_matrix_bin_t_melt) +
    geom_bar(mapping = aes(x = value
                                 ),
                   stat = "count",
                   fill = "grey"
             ) +
    xlab("\nbinarized AUC score") +
    ylab("Count\n") +
    theme_bw()

```

#### Filtering

```{r AUC_bin_filtering}

# Remove all lines where counts are all equal to zero
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ <- scRNAseq_integrated_scenic_auc_matrix_bin_t[apply(X = scRNAseq_integrated_scenic_auc_matrix_bin_t,
                                                                                            MARGIN = 1,
                                                                                            FUN = function(x) !all(x==0)),
                                                                                      ]

dim(scRNAseq_integrated_scenic_auc_matrix_bin_t)
dim(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ)

# Filtering
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt <- expFilter(data      = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ,
                                                             threshold = 0.5,
                                                             p         = 0.05,
                                                             graph     = TRUE
                                                             )

dim(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt)

```

#### Heatmap of AUC_BIN score

```{r SCENIC_AUC_BINscore_heatmap}

# Most variable genes
mvRegulons_auc_bin <- genes.selection(data      = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt,
                                      thres.num = 50
                                      )

# Subset matrix
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons <- scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt[mvRegulons_auc_bin,]

# Check order
all(rownames(scRNAseq_Seurat_integrated@meta.data) == colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons))

# Define heatmap color
col_fun_AUCbinScore = colorRamp2(breaks = c(0,1), colors = c("white","black"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons,
             clustering_distance_rows    = "binary",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_row_dend               = FALSE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             bottom_annotation           = bottomAnnot_PCmin,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"])),
             column_split                = factor(scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"]),
             #row_split                   = 2,
             col                         = col_fun_AUCbinScore,
             show_heatmap_legend         = FALSE
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_SCENIC_AUCbinScore.pdf"), width = 12, height = 12)
draw(h)
#dev.off()

```

#### Heatmap of the percentage of cells with activated regulon

```{r heatmap_perc_active_regulon}

# Melt matrix
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt <- melt(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons)
colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt) <- c("regulon","cellID","AUCbin")
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt)

# Merge matrix and meta data
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus <- merge(x     = scRNAseq_Seurat_integrated@meta.data["integrated_snn_res.0.2"],
                                                                              y     = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt,
                                                                              by.x  = 0,
                                                                              by.y  = "cellID",
                                                                              all   = FALSE
                                                                              )
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus)

# Create table of cluster, regulon and number of cells
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells <- group_by(.data = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus,
                                                                                          integrated_snn_res.0.2,
                                                                                          regulon
                                                                                          ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells) = c("cluster","regulon","nbCell")
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells)

# Create table of cluster, regulon and number of cells in which regulon is activated
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells <- filter(.data = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus, AUCbin == 1) %>%
    group_by(integrated_snn_res.0.2,
             regulon
             ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells) = c("cluster","regulon","nbCell_AUCbin1")
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells)

# Create a table of total number of cells and cells in which regulon is activated
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells <- merge(x = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells,
                                                                                                   y = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells,
                                                                                                   by = c("cluster","regulon"),
                                                                                                   all = FALSE
                                                                                                   )
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Compute percentage of cells with active regulon
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$activeRegulon <- scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell_AUCbin1*100/scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell

head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Transform table to matrix (for ComplexHeatmap function)
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix <- spread(data = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells[,c("cluster","regulon","activeRegulon")],
                                                                                                 key = cluster,
                                                                                                 value = activeRegulon
                                                                                                 )
rownames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix) <- scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon
scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon <- NULL
head(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix)



# Top annotation
topAnnot_cluster <- HeatmapAnnotation(cluster                  = colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
                                      text                     = anno_text(x= colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix), rot = 0),
                                      col                      = list(cluster = cluster_n10_col),
                                      gap                      = unit(4,"mm"),
                                      show_annotation_name     = FALSE,
                                      annotation_name_gp       = gpar(fontsize = 20),
                                      annotation_legend_param  = list(fontsize   = 20,
                                                                      title_gp    = gpar(fontsize = 20, font = 2),
                                                                      labels_gp   = gpar(fontsize = 20),
                                                                      grid_width  = unit(12,"mm"),
                                                                      grid_height = unit(12,"mm"),
                                                                      ncol        = 1
                                                                      ),
                                      simple_anno_size        = unit(2,"cm"),
                                      na_col                  = "white",
                                      show_legend             = FALSE
                                      )

# Heatmap
h <- Heatmap(matrix                    = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix,
             cluster_columns           = FALSE,
             clustering_distance_rows  = "euclidean",
             clustering_method_rows    = "average",
             top_annotation            = topAnnot_cluster,
             show_row_dend             = FALSE,
             column_title              = "cluster\n",
             row_title                 = "Top 50 most variable regulons",
             column_split              = colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
             show_column_names         = FALSE,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(2,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n% of cells with active regulon",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter",
                                                direction = "horizontal"
                                                )
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_SCENIC_AUCbinScore_perc.pdf"), width = 12, height = 12)
draw(h,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Cell type specific regulators

#### Import SCENIC RSS score

```{r import_SCENIC_AUCscore}

# Import SCENIC RSS score
scRNAseq_integrated_scenic_rss_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_rss_matrix.csv"),
                                                  check.names = FALSE
                                                  )
scRNAseq_integrated_scenic_rss_matrix[,1:4]
dim(scRNAseq_integrated_scenic_rss_matrix)

# Create column of cluster
scRNAseq_integrated_scenic_rss_matrix$cluster <- rownames(scRNAseq_integrated_scenic_rss_matrix)
head(scRNAseq_integrated_scenic_rss_matrix)

# Melt table
scRNAseq_integrated_scenic_rss_matrix_melt <- melt(scRNAseq_integrated_scenic_rss_matrix)
head(scRNAseq_integrated_scenic_rss_matrix_melt)

# Density distribution
ggplot(data = scRNAseq_integrated_scenic_auc_matrix_bin_t_melt) +
    geom_density(mapping = aes(x = value),
                   fill = "grey"
             ) +
    xlab("\nRSS score") +
    ylab("density\n") +
    theme_bw()

# Transpose RSS matrix (to be copied to zerkalo and then explored by Maria)
scRNAseq_integrated_scenic_rss_matrix_t <- t(data.matrix(scRNAseq_integrated_scenic_rss_matrix))
head(scRNAseq_integrated_scenic_rss_matrix_t)
scRNAseq_integrated_scenic_rss_matrix_t <- data.frame(scRNAseq_integrated_scenic_rss_matrix_t,
                                                      check.names = FALSE
                                                      )
scRNAseq_integrated_scenic_rss_matrix_t$TF <- rownames(scRNAseq_integrated_scenic_rss_matrix_t)
head(scRNAseq_integrated_scenic_rss_matrix_t)

# Export (to be copied to zerkalo)
write.table(x = scRNAseq_integrated_scenic_rss_matrix_t,
            file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_rss_matrix_t.txt"),
            quote = FALSE,
            sep = "\t",
            row.names = FALSE
            )
            
# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_integrated_scenic_rss_matrix_t.txt"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

#### Plot top 5 RSS score

```{r import_SCENIC_AUCscore}

# Convert variable column (regulon) into factor
scRNAseq_integrated_scenic_rss_matrix_melt$variable <- factor(scRNAseq_integrated_scenic_rss_matrix_melt$variable)
head(scRNAseq_integrated_scenic_rss_matrix_melt)

levels(factor(scRNAseq_integrated_scenic_rss_matrix_melt$cluster))

scRNAseq_integrated_scenic_rss_matrix_melt[which(scRNAseq_integrated_scenic_rss_matrix_melt$cluster == "9"),] %>%
    .[order(.$value, decreasing = TRUE),] %>%
    .[1:5,"variable"]

# Plot RSS per cluster
scRNAseq_integrated_scenic_rss_plot_list = list()
all_top5 <- data.frame(regulon = character(0), cluster = character(0))
for (cluster in levels(factor(scRNAseq_integrated_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_integrated_scenic_rss_matrix_melt[which(scRNAseq_integrated_scenic_rss_matrix_melt$cluster == cluster),]
    # Pick top 5 regulons
    top5 <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    #all_top5 <- c(all_top5, as.vector(top5))
    all_top5 <- rbind(all_top5,
                      data.frame(regulon = top5, cluster = cluster)
                      )
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% top5),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% top5),
                        mapping = aes(label = variable),
                        color = "red",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_integrated_scenic_rss_plot_list[[length(scRNAseq_integrated_scenic_rss_plot_list)+1]] <- p
}

library(ggpubr)

#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_SCENIC_RSS.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_integrated_scenic_rss_plot_list, ncol=5, nrow=2)
#dev.off()

all_top5

all_top5[duplicated(all_top5$regulon),]

```

#### Plot top interested regulons RSS (selected by Maria)

```{r plot_RSS_interested_regulons}

# Import interested regulons
interestedRegulons <- read.table(file = paste0(pathToInfo,"LISTE_regulons_plus_importantes_par_cluster.csv"),
                                 header = TRUE,
                                 colClasses = c("character","character","numeric","character")
                                 ) %>%
    filter(Include == "yes")
head(interestedRegulons)

# Plot RSS per cluster
scRNAseq_integrated_scenic_rss_interestedRegulons_plot_list = list()
for (cluster in levels(factor(scRNAseq_integrated_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_integrated_scenic_rss_matrix_melt[which(scRNAseq_integrated_scenic_rss_matrix_melt$cluster == cluster),]
    # Interested regulons
    intReg <- paste0(dplyr::filter(.data = interestedRegulons, Cluster == cluster)$Gene,"(+)")
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% intReg),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% intReg),
                        mapping = aes(label = variable),
                        segment.colour = "grey",
                        color = "darkgreen",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_integrated_scenic_rss_interestedRegulons_plot_list[[length(scRNAseq_integrated_scenic_rss_interestedRegulons_plot_list)+1]] <- p
}

#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_SCENIC_RSS_interestedRegulons.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_integrated_scenic_rss_interestedRegulons_plot_list, ncol=5, nrow=2)
#dev.off()

```

#### Plot top interested regulons RSS (for paper figure)

```{r plot_RSS_interested_regulons}

# Import interested regulons
interestedRegulons <- read.table(file = paste0(pathToInfo,"LISTE_regulons_plus_importantes_par_cluster.csv"),
                                 header = TRUE,
                                 colClasses = c("character","character","numeric","character")
                                 ) %>%
    filter(Include == "yes")
head(interestedRegulons)

head(scRNAseq_integrated_scenic_rss_matrix_melt)

source("reorder_within.R")

# Create function for plotting regulons of interest per cell type
plotInterestedReg <- function(clusters, cell_subtype_shortName)
{
    f <- scRNAseq_integrated_scenic_rss_matrix_melt[which(scRNAseq_integrated_scenic_rss_matrix_melt$cluster %in% clusters),]
    intReg <- paste0(dplyr::filter(.data = interestedRegulons, Cluster %in% clusters)$Gene,"(+)")
    p <- ggplot(data = f,
                mapping = aes(x = reorder_within(variable,-value,cluster),
                              y = value
                              )
                ) +
        facet_wrap(facets = ~ cluster) +
        ggtitle(cell_subtype_shortName) +
        xlab("") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "grey",
                   size = 0.5
                   ) +
        geom_point(data = filter(f, variable %in% intReg),
                   color = "red",
                   size = 0.5
                   ) + 
        geom_text_repel(data            = filter(f, variable %in% intReg),
                        mapping         = aes(label = variable),
                        segment.colour  = "black",
                        min.segment.length    = 0.25,
                        segment.size    = 0.1,
                        color           = "black",
                        size            = paperFig_fontsize/4,
                        force           = 50
                        ) + 
        theme_bw() +
        theme(text = element_text(size = paperFig_fontsize),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              strip.background = element_rect(fill = cell_subtype_shortName_col[cell_subtype_shortName]),
              strip.text = element_text(size = paperFig_fontsize*1.5, colour = "white", face = "bold"),
              plot.title = element_text(hjust = 0.5, face = "bold", size = paperFig_fontsize*1.5)
              )
}

# Plot interested regulons for cluster 0 (undifferentiated)
interestedRegulons_clust0 <- plotInterestedReg(clusters = "0",
                                               cell_subtype_shortName = "UD1"
                                               )

# Plot interested regulons for cluster 1 (undifferentiated)
interestedRegulons_clust1 <- plotInterestedReg(clusters = "1",
                                                cell_subtype_shortName = "UD2"
                                                )

# Plot interested regulons for cluster 2 (cycling cells)
interestedRegulons_clust2 <- plotInterestedReg(clusters = "2",
                                               cell_subtype_shortName= "G1S"
                                               )

# Plot interested regulons for cluster 3 (neuronal prog)
interestedRegulons_clust3 <- plotInterestedReg(clusters = "3",
                                               cell_subtype_shortName = "NPL1"
                                               )

# Plot interested regulons for cluster 4 (cycling cells)
interestedRegulons_clust4 <- plotInterestedReg(clusters = "4",
                                               cell_subtype_shortName = "G2M"
                                               )

# Plot interested regulons for cluster 5 (neuronal commit)
interestedRegulons_clust5 <- plotInterestedReg(clusters = "5",
                                               cell_subtype_shortName = "NPL2"
                                                )

# Plot interested regulons for cluster 6 (non-neuronal)
interestedRegulons_clust6 <- plotInterestedReg(clusters = "6",
                                               cell_subtype_shortName = "Non-neuronal 1"
                                               )

# Plot interested regulons for cluster 7 (non-neuronal)
interestedRegulons_clust7 <- plotInterestedReg(clusters = "7",
                                               cell_subtype_shortName = "Non-neuronal 2"
                                               )

# Plot interested regulons for cluster 8 (neuronal commit)
interestedRegulons_clust8 <- plotInterestedReg(clusters = "8",
                                                cell_subtype_shortName = "Non-neuronal 3"
                                                )

# Plot interested regulons for cluster 9 (neuronal commit)
interestedRegulons_clust9 <- plotInterestedReg(clusters = "9",
                                               cell_subtype_shortName = "HIR"
                                               )

# Plot interested regulons for cluster 2 and 4 (cycling cells)
#interestedRegulons_clust24 <- plotInterestedReg(clusters = c("2","4"),
 #                                               cellType = "cycling cells"
  #                                              )
interestedRegulons_clust24 <- grid.arrange(arrangeGrob(interestedRegulons_clust2 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank(),
                                                             axis.text.y = element_blank()
                                                             ),
                                                       interestedRegulons_clust4 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("Cycling Cells",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                        #               rot = 90,
                                                         #              vjust = 1,
                                                          #            gp = gpar(fontsize = paperFig_fontsize)
                                                           #            )
                                                       )
                                           )


# Plot interested regulons for cluster 0 and 1 (cycling cells)
#interestedRegulons_clust01 <- plotInterestedReg(clusters = c("0","1"),
 #                                               cellType = "undifferentiated"
  #                                              )
interestedRegulons_clust01 <- grid.arrange(arrangeGrob(interestedRegulons_clust0 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank(),
                                                             axis.text.y = element_blank()
                                                             ),
                                                       interestedRegulons_clust1 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("UD cells",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",
                                                                                fontsize = paperFig_fontsize*1.5
                                                                                )
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                                       #rot = 90,
                                                                       #vjust = 1,
                                                                      #gp = gpar(fontsize = paperFig_fontsize)
                                                                       #)
                                                       )
                                           )

interestedRegulons_clust35 <- grid.arrange(arrangeGrob(interestedRegulons_clust3 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank()#,
                                                             #axis.text.y = element_blank()
                                                             ),
                                                       interestedRegulons_clust5 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       widths = c(0.53,0.47),
                                                       top = textGrob("NPL cells",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",
                                                                                fontsize = paperFig_fontsize*1.5
                                                                                )
                                                                      ),
                                                       left = textGrob("RSS score",
                                                                       rot = 90,
                                                                       vjust = 1,
                                                                      gp = gpar(fontsize = paperFig_fontsize)
                                                                       )
                                                       )
                                           )

# Plot interested regulons for cluster 6 and 7 (cycling cells)
#interestedRegulons_clust67 <- plotInterestedReg(clusters = c("6","7"),
 #                                               cellType = "non-neuronal"
  #                                              )
interestedRegulons_clust78 <- grid.arrange(arrangeGrob(interestedRegulons_clust6 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_clust7 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("non-neuronal",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      ),
                                                       left = textGrob("RSS score",
                                                                       rot = 90,
                                                                       vjust = 1,
                                                                      gp = gpar(fontsize = paperFig_fontsize)
                                                                       )
                                                       )
                                           )


interestedRegulons_clust678 <- grid.arrange(arrangeGrob(interestedRegulons_clust6 +
                                                       theme(axis.title.y = element_blank(),
axis.text.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_clust7 +
                                                       theme(axis.title.y = element_blank(),
axis.text.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_clust8 +
                                                       theme(axis.title.y = element_blank(),
axis.text.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("Non-Neuronal",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                        #               rot = 90,
                                                         #              vjust = 1,
                                                          #            gp = gpar(fontsize = paperFig_fontsize)
                                                           #            )
                                                       )
                                           )


```


#### Heatmap (AUC score) of top 5 RSS score regulons

```{r import_SCENIC_AUCscore}

# Heatmap of all top5
scRNAseq_integrated_scenic_auc_matrix_t_allTop5 <- scRNAseq_integrated_scenic_auc_matrix_t[as.vector(all_top5$regulon),]
head(scRNAseq_integrated_scenic_auc_matrix_t_allTop5[,1:4])

# Density of scaled UCS score
plot(density(t(scale(t(scRNAseq_integrated_scenic_auc_matrix_t_allTop5)))))

col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0), colors = c("cyan","white","red","darkred"))
#col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0,5), colors = c("blue","cyan","white","yellow","red"))
# Heatmap
h <- Heatmap(matrix                      = t(scale(t(scRNAseq_integrated_scenic_auc_matrix_t_allTop5))),
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = FALSE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Cluster specific regulons",
             top_annotation              = topAnnot_PCmin,
             bottom_annotation           = bottomAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"])),
             column_split                = factor(scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"]),
             row_split                   = all_top5$cluster, #rep(c("0","1","3","4","5","6","7","8","9"), each = 5),
             col                         = col_fun_AUCscore_scaled,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "scaled\nAUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_SCENIC_AUCscore_scaled_allTop5.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

### Figure 8 patchwork

```{r figure8_patchwork}

figure8ab <- wrap_plots(a       = umap_scRNAseq_clust_cellTypes,
                        b       = legend_cellTypes,
                        c       = StackedVlnPlot_cellTypes,
                        widths  = c(0.35, 0.15,0.5)
                        )

figure8c1 <- wrap_plots(interestedRegulons_clust3,
                        interestedRegulons_clust5,
                        interestedRegulons_clust01,
                       #interestedRegulons_clust8,
                       #interestedRegulons_clust9,
                       #interestedRegulons_clust24,
                       nrow = 1,
                       widths = c(0.25,0.25,0.5)
                       )

figure8c2 <- wrap_plots(#interestedRegulons_clust24,
                                        #interestedRegulons_clust01,
    interestedRegulons_clust24,
    interestedRegulons_clust678,
    interestedRegulons_clust9,
    nrow = 1,
    widths = c(0.2,0.3,0.1)
                        )

figure8 <- wrap_plots(a = figure8ab,
                      b = figure8c1,
                      c = figure8c2,
                      heights = c(0.4,0.3,0.4)
                      )

# Save image as pdf
ggsave(plot = figure8,
       file = paste0(pathToFigures,"figure8.pdf"),
       width = 210, #full a4 210
       height = 297*2/3, # full a4 297
       units = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figure8.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

### Figure 8bis patchwork (figure7_LobonIglesias2022)

```{r figure8bis_patchwork}

# UMAP and clustering
figure8bis_A <- wrap_elements(umap_scRNAseq_clust_cellTypes) + labs(tag = 'A')
    
figure8bis_B <- wrap_elements(StackedVlnPlot_cellTypes_0135)  + labs(tag = 'B')
    
figure8bis_AB <- wrap_plots(figure8bis_A,
                            legend_cellTypes,
                            figure8bis_B,
                            widths = c(0.45,0.4,0.25)
                            )

figure8bis_C <- wrap_plots(plotList_heatmap_umap_interesingGenes[["IRX1"]] + labs(tag = 'C')  + theme(plot.tag = element_text(vjust = 3)),
                           plotList_heatmap_umap_interesingGenes[["OTX2"]],
                           plotList_heatmap_umap_interesingGenes[["EN1"]]
                           )

figure8bis_D <- wrap_plots(wrap_plots(interestedRegulons_clust35) + labs(tag = 'D') + theme(plot.tag = element_text(vjust = 3)),
                           interestedRegulons_clust01,
                           widths = c(0.525,0.475)
                           )

figure8bis_CD <- wrap_plots(figure8bis_C,
                            figure8bis_D,
                            widths = c(0.5,0.5)
                            )

figure8bis_E <- wrap_plots(trajectory_inference_monocle3_clust0135) + labs(tag = 'E')

library(magick)
library(ggpubr)

scVelo_clust0135_DEG_monocle3 <- image_read("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/figures/scvelo_scRNAseq_integrated_embeddingStream_clust0135_DEG_monocle3_dynamical_thesisDefense.png")
scVelo_clust0135_DEG_monocle3 <- ggplot()+
    background_image(scVelo_clust0135_DEG_monocle3)


figure8bis_F <- wrap_elements(scVelo_clust0135_DEG_monocle3) + labs(tag = 'F')

figure8bis_EF <- wrap_plots(figure8bis_E,
                            figure8bis_F,
                            widths = c(0.5,0.5)
                            )

figure8bis_G <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["DCX"]]  +
    labs(tag = 'G'),
    plotList_heatmap_monocle3_interesingGenes[["ELAVL4"]],
    plotList_heatmap_monocle3_interesingGenes[["SOX4"]],
    plotList_heatmap_monocle3_interesingGenes[["SOX11"]],
    plotList_heatmap_monocle3_interesingGenes[["REST"]],
    plotList_heatmap_monocle3_interesingGenes[["ID4"]],
    plotList_heatmap_monocle3_interesingGenes[["SOX2"]],
    plotList_heatmap_monocle3_interesingGenes[["ACTL6A"]],
    plotList_heatmap_monocle3_interesingGenes[["ACTL6B"]],
    nrow = 3
    )

figure8bis_H <- wrap_elements(dotplot_cellphonedb_NOTCH) +
    labs(tag = 'H')

figure8bis_GH <- wrap_plots(figure8bis_G,
                            figure8bis_H,
                            widths = c(0.5,0.5)
                            )

figure8bis_H_lgd <- wrap_plots(plot_spacer(),
                               legend_dotplot_cellphonedb_NOTCH,
                               widths = c(0.5,0.5)
                               )

# Patchwork
figure8bis <- wrap_plots(figure8bis_AB,
                         figure8bis_CD,
                         figure8bis_EF,
                         #figure8bis_G,
                         #figure8bis_H,
                         figure8bis_GH,
                         figure8bis_H_lgd,
                         heights = c(0.25,0.15,0.2,0.3,0.05)
                         ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Figure 7\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Rename figure
figure7_LobonIglesias2022 <- figure8bis

# Save figure into pdf file
ggsave(plot   = figure7_LobonIglesias2022,
       file   = paste0(pathToFigures,"figure7_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figure7_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

### Supplementary Fig 7 (LobonIglesias2022)

```{r figure8bis_patchwork}

# UMAP and heatmap
SuppFig7_A <- wrap_elements(umap_Torchia2016_Johann2016_SHH_genes_heteroExp) + labs(tag = 'A')

# Stacked violin plot for cycling cells and non-neuronal clusters
SuppFig7_B <- wrap_elements(stackedVlnPlot_CCNNHIR_markerGenes) + labs(tag = 'B')

# RSS
SuppFig7_C <- wrap_plots(interestedRegulons_clust9,
                        interestedRegulons_clust24,
                        interestedRegulons_clust678,
                        widths = c(0.15,0.375,0.575)
                        )

SuppFig7_C_tag <- wrap_elements(plot_spacer() + labs(tag='C') +
                                theme(text = element_text(face = 'bold',size = 9),
                                      plot.tag = element_text(size = 20, face = 'bold')
                                      )

# ElPiGraph
SuppFig7_D <- wrap_plots(elpiGraph_scRNAseq_integrated_0135_curve_cellType) + labs(tag = 'D')

# RSS and ElPiGraph
SuppFig7_CD <- wrap_plots(SuppFig7_C_tag,
                          SuppFig7_C,
                          SuppFig7_D,
                          heights = c(0.01,0.5,0.5)
                          )

# Stacked violin plot & RSS & ElphiGraph
SuppFig7_BCD <- wrap_plots(SuppFig7_B,
                           SuppFig7_CD,
                           widths = c(0.333,0.666)
                           )

# Pseudotime
SuppFig7_E <- wrap_plots(pseudotime_monocle3_clust0135) + labs(tag = 'E')

# Latent time
#library(figpatch)
#SuppFig7_F <- fig(path = paste0(pathToFigures, "scvelo_scRNAseq_integrated_clust0135_DEG_monocle3_latentTime_fig.png")
 #                 )# +
    #labs(tag = "A")
SuppFig_F <- wrap_plots(scVelo_clust0135_DEG_monocle3_latentTime) + labs(tag = 'F')


# Pseudotime & latent time
SuppFig7_EF <- wrap_plots(SuppFig7_E,
                          SuppFig_F,
                          heights = c(0.5,0.5)
                          )

# SWI/WNF
SuppFig7_G <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["DLL3"]]  +
                         labs(tag = 'G'),
                         plotList_heatmap_monocle3_interesingGenes[["DLL1"]],
                         plotList_heatmap_monocle3_interesingGenes[["HES1"]],
                         plotList_heatmap_monocle3_interesingGenes[["HES4"]],
                         nrow = 2
                         )

# Pseudotime & latent time & SWI/SNF
SuppFig7_EFG <- wrap_plots(SuppFig7_EF,
                           SuppFig7_G,
                           plot_spacer(),
                           widths = c(0.3,0.5,0.2)
                          )


# Patchwork
SuppFig7 <- wrap_plots(SuppFig7_A,
                       SuppFig7_BCD,
                       SuppFig7_EFG,
                       heights = c(0.4,0.4,0.2)
                       ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Supplementary Fig. 7\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Save figure into pdf file
ggsave(plot   = SuppFig7,
       file   = paste0(pathToFigures,"SuppFig7_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig7_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```


### Supplementary Fig 8 (LobonIglesias2022)

```{r figure8bis_patchwork}

######################
# nb Genes filtering #
######################

# Import individual raw data sample meta data
INI254_metadata <- read.table(file = "scRNAseq_Seurat_INI254_metadata.txt",
                              sep = ",",
                              header = TRUE
                              )
head(INI254_metadata)
#
INI255_metadata <- read.table(file = "scRNAseq_Seurat_INI255_metadata.txt",
                              sep = ",",
                              header = TRUE
                              )
head(INI255_metadata)
#
INI267_metadata <- read.table(file = "scRNAseq_Seurat_INI267_metadata.txt",
                              sep = ",",
                              header = TRUE
                              )
head(INI267_metadata)

# Function to plot nbUM vs nbGenes
plot_nbUMI_nbGenes_func <- function(metadata, min_nFeature)
    {
        p <- ggplot() +
            geom_point(data = filter(.data = metadata,
                                     nFeature_RNA >= min_nFeature
                                     ),
                       mapping = aes(x = nCount_RNA,
                                     y = nFeature_RNA
                                     ),
                       size = 0.1,
                       pch = 21,
                       color = "darkgreen"
                       ) +
            geom_point(data = filter(.data = metadata,
                                     nFeature_RNA < min_nFeature
                                     ),
                       mapping = aes(x = nCount_RNA,
                                     y = nFeature_RNA
                                     ),
                       size = 0.1,
                       pch = 21,
                       color = "grey"
                       ) +
            xlab(label = "\nRNA count (UMI)") +
            ylab(label = "nb of genes\n") +
            ylim(0,10000) +
            theme_bw() +
            geom_hline(yintercept = min_nFeature,
                       color = "red",
                       lty = 2,
                       lwd = 0.25
                       ) +
            theme(axis.text = element_text(size = paperFig_fontsize),
                  axis.title = element_text(size = paperFig_fontsize, face = "bold")
                  ) +
            geom_text(mapping = aes(x = (max(metadata$nCount_RNA) - min(metadata$nCount_RNA))/2,
                                    y = min_nFeature
                                    ),
                      label = min_nFeature,
                      color = "red",
                      size = paperFig_fontsize/3,
                      nudge_y = max(metadata$nFeature_RNA)/20
                      )
        # Add marginal density
        p <- ggMarginal(p        = p,
                        type     = "density",
                        margins  = "both",
                        xparams  = list(fill = "yellow"),
                        yparams  = list(fill = "blue")
                        )
        return(p)
    }

# Plot nbUMI vs nbGenes for individual samples
INI254_nbUMI_nbGenes <- plot_nbUMI_nbGenes_func(INI254_metadata, 410)
INI255_nbUMI_nbGenes <- plot_nbUMI_nbGenes_func(INI255_metadata, 491)
INI267_nbUMI_nbGenes <- plot_nbUMI_nbGenes_func(INI267_metadata, 656)

SuppFig8_A <- wrap_plots(wrap_elements(INI254_nbUMI_nbGenes) + labs(tag='A'),
                         wrap_elements(INI255_nbUMI_nbGenes),
                         wrap_elements(INI267_nbUMI_nbGenes),
                         widths = c(0.333,0.333,0.333)
                         )


####################
# Mito % filtering #
####################

# Import individual raw data sample meta data
INI254_nbGenes_metadata <- read.table(file = "scRNAseq_Seurat_INI254_nbGenes_metadata.txt",
                              sep = ",",
                              header = TRUE
                              )
head(INI254_nbGenes_metadata)
#
INI255_nbGenes_metadata <- read.table(file = "scRNAseq_Seurat_INI255_nbGenes_metadata.txt",
                                      sep = ",",
                                      header = TRUE
                                      )
head(INI255_metadata)
#
INI267_nbGenes_metadata <- read.table(file = "scRNAseq_Seurat_INI267_nbGenes_metadata.txt",
                              sep = ",",
                              header = TRUE
                              )
head(INI267_metadata)

plot_nbUMI_percMito_func <- function(metadata, max_percOfMitoGenes)
{
    p <- ggplot() +
        geom_point(data = metadata,
                   mapping = aes(x = nCount_RNA,
                                 y = percent.mito
                                 ),
                   size = 0.1,
                   pch = 21,
                   color = "grey"
                   ) +
        geom_point(data = filter(.data = metadata,
                                 percent.mito < max_percOfMitoGenes
                                 ),
                   mapping = aes(x = nCount_RNA,
                                 y = percent.mito
                                 ),
                   size = 0.1,
                   pch = 21,
                   color = "blue"
                   ) +
        ylim(0,65) +
        xlab(label = "\nnb of RNA (UMI count)") +
        ylab(label = "\n% mito. genes\n") +
        geom_hline(yintercept = max_percOfMitoGenes,
                   color = "red",
                   lty = 2
                   ) +
        theme_bw() +
        theme(axis.text = element_text(size = paperFig_fontsize),
              axis.title = element_text(size = paperFig_fontsize,
                                        face = "bold"
                                        )
              ) +
        geom_text(mapping = aes(x = (max(metadata$nCount_RNA) - min(metadata$nCount_RNA))/2,
                                y = max_percOfMitoGenes
                                ),
                  label = paste0(max_percOfMitoGenes," %"),
                  color = "red",
                  size = paperFig_fontsize/3,
                  nudge_y = 2
                  )
    p <- ggMarginal(p        = p,
                    type     = "density",
                    margins  = "y",
                    fill     = "violet"
                    )
}

# Plot nbUMI vs %Mito for individual samples
INI254_nbUMI_percMito <- plot_nbUMI_percMito_func(INI254_nbGenes_metadata, 5)
INI255_nbUMI_percMito <- plot_nbUMI_percMito_func(INI255_nbGenes_metadata, 5)
INI267_nbUMI_percMito <- plot_nbUMI_percMito_func(INI267_nbGenes_metadata, 10)

SuppFig8_B <- wrap_plots(wrap_elements(INI254_nbUMI_percMito) + labs(tag='B'),
                         wrap_elements(INI255_nbUMI_percMito),
                         wrap_elements(INI267_nbUMI_percMito),
                         widths = c(0.333,0.333,0.333)
                         )

#########################################
# nb of cells after each filtering step #
#########################################

INI254_nbOfCells <- data.frame(filtering = c("no filt.","nb Genes filt.","% mito filt."),
                               nbOfCells = c(nrow(INI254_metadata),nrow(INI254_nbGenes_metadata),nrow(filter(INI254_nbGenes_metadata,percent.mito < 5))),
                               check.names = FALSE
                               )
#
INI255_nbOfCells <- data.frame(filtering = c("no filt.","nb Genes filt.","% mito filt."),
                               nbOfCells = c(nrow(INI255_metadata),nrow(INI255_nbGenes_metadata),nrow(filter(INI255_nbGenes_metadata,percent.mito < 5))),
                               check.names = FALSE
                               )
#
INI267_nbOfCells <- data.frame(filtering = c("no filt.","nb Genes filt.","% mito filt."),
                               nbOfCells = c(nrow(INI267_metadata),nrow(INI267_nbGenes_metadata),nrow(filter(INI267_nbGenes_metadata,percent.mito < 5))),
                               check.names = FALSE
                               )

plot_nbOfCells_func <- function(table)
{
    table$filtering <- factor(table$filtering, levels = c("no filt.","nb Genes filt.","% mito filt."))
    ggplot(data = table,
           mapping = aes(x = filtering,
                         y = nbOfCells,
                         fill = filtering
                         ),
           ) +
        geom_bar(stat = "identity"
                 ) +
        ylim(0,10000) +
        scale_fill_manual(values = c("black","darkgreen", "green")) +
        xlab(label = "\nfiltering step") +
        ylab(label = "nb of cells\n") +
        geom_text(aes(label = nbOfCells), size = paperFig_fontsize/2.5, fontface = "bold", vjust = -0.25) +
        theme_bw() +
        theme(axis.text = element_text(size = paperFig_fontsize),
              axis.title = element_text(size = paperFig_fontsize, face = "bold"),
              legend.position = "none"
              )
}

barplot_nbOfCells_INI254 <- plot_nbOfCells_func(INI254_nbOfCells)
barplot_nbOfCells_INI255 <- plot_nbOfCells_func(INI255_nbOfCells)
barplot_nbOfCells_INI267 <- plot_nbOfCells_func(INI267_nbOfCells)

SuppFig8_C <- wrap_plots(wrap_elements(barplot_nbOfCells_INI254) + labs(tag='C'),
                         wrap_elements(barplot_nbOfCells_INI255),
                         wrap_elements(barplot_nbOfCells_INI267),
                         widths = c(0.333,0.333,0.333)
                         )

##########################################
# PCA of merged (not integrated) samples #
##########################################

seurat_object_list <- list(INI254 = scRNAseq_Seurat_INI254,
                           INI255 = scRNAseq_Seurat_INI255,
                           INI267 = scRNAseq_Seurat_INI267
                           )

# Rename cells using object name as prefix
for (obj in names(seurat_object_list))
{
    seurat_object_list[[obj]] <- RenameCells(object = seurat_object_list[[obj]],
                                             add.cell.id = obj
                                             )
}

# Merge seurat objects
#scRNAseq_Seurat_merge <- reduce(.x         = seurat_object_list,
 #                               .f         = merge,
  #                              merge.data = FALSE
   #                             )

scRNAseq_Seurat_merge <- merge(merge(seurat_object_list[["INI254"]],
                                     seurat_object_list[["INI255"]],
                                     merge.data = FALSE
                                     ),
                               seurat_object_list[["INI267"]],
                               merge.data = FALSE
                               )

# Normalization
scRNAseq_Seurat_merge <- NormalizeData(object                = scRNAseq_Seurat_merge,
                                       normalization.method  = "LogNormalize",
                                       scale.factor          = 10000
                                       )

# Compute cell cycle scoring  
scRNAseq_Seurat_merge <- CellCycleScoring(object        = scRNAseq_Seurat_merge,
                                          s.features    = cc.genes$s.genes,
                                          g2m.features  = cc.genes$g2m.genes
                                          )


# Highly variable genes
scRNAseq_Seurat_merge <- FindVariableFeatures(object            = scRNAseq_Seurat_merge,
                                              selection.method  = variableFeatures_method,
                                              nfeatures         = nFeatures
                                              )

# Data scaling 
scRNAseq_Seurat_merge <- ScaleData(object = scRNAseq_Seurat_merge#,
                                   #features = rownames(scRNAseq_Seurat_merge)
                                   )

# Extract matrix
scRNAseq_Seurat_merge_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_merge,
                                                       slot = "scale.data"
                                                       )
                                          )

dim(scRNAseq_Seurat_merge_matrix)

# PCA analysis
scRNAseq_Seurat_merge_PC        <- prcomp(data.frame(t(scRNAseq_Seurat_merge_matrix)))
scRNAseq_Seurat_merge_eigVal    <- get_eigenvalue(scRNAseq_Seurat_merge_PC)
scRNAseq_Seurat_merge_eigValPer <- round(scRNAseq_Seurat_merge_eigVal$variance.percent, digits = 2)
scRNAseq_Seurat_merge_pci       <- data.frame(scRNAseq_Seurat_merge_PC$x,scRNAseq_Seurat_merge@meta.data)

plot_pca_func <- function(pci,eigValPer,covariate,cols,name)
{
    ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = !!enquo(covariate)),
               size    = 0.25
               ) +
    scale_colour_manual(name = name, values = cols) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    #ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = paperFig_fontsize),
          legend.title      = element_text(face = "bold"),
          legend.key.height = unit(0.5,"cm")
          )
}

scRNAseq_Seurat_merge_PCA_orig.ident <- plot_pca_func(scRNAseq_Seurat_merge_pci,scRNAseq_Seurat_merge_eigValPer,orig.ident,c("red","green","blue"), "Sample")
scRNAseq_Seurat_merge_PCA_Phase <- plot_pca_func(scRNAseq_Seurat_merge_pci,scRNAseq_Seurat_merge_eigValPer,Phase,c("grey","brown","yellow"),"Phase")

scRNAseq_Seurat_integrated_eigValPer <- eigValPer
scRNAseq_Seurat_integrated_pci <- pci

scRNAseq_Seurat_integrated_PCA_orig.ident <- plot_pca_func(scRNAseq_Seurat_integrated_pci,scRNAseq_Seurat_merge_eigValPer,orig.ident,c("red","green","blue"),"Sample")
scRNAseq_Seurat_integrated_PCA_Phase <- plot_pca_func(scRNAseq_Seurat_integrated_pci,scRNAseq_Seurat_merge_eigValPer,Phase,c("grey","brown","yellow"),"Phase")

SuppFig8_D <- wrap_plots(wrap_elements(scRNAseq_Seurat_merge_PCA_orig.ident) + labs(tag='D'),
                         wrap_elements(scRNAseq_Seurat_integrated_PCA_orig.ident),
                         wrap_elements(scRNAseq_Seurat_merge_PCA_Phase),
                         wrap_elements(scRNAseq_Seurat_integrated_PCA_Phase)
                         ) +
    plot_layout(nrow = 2)

# Patchwork
SuppFig8 <- wrap_plots(SuppFig8_A,
                       SuppFig8_B,
                       SuppFig8_C,
                       SuppFig8_D,
                       heights = c(0.2,0.2,0.15,0.45)
                       ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Supplementary Fig. 8\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Save figure into pdf file
ggsave(plot   = SuppFig8,
       file   = paste0(pathToFigures,"SuppFig8_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig8_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```


## Deconvolution of scRNAseq using mouse cell clusters from Vladoiu et al, 2019

###  Average expression matrix

```{r average_expression_matrix}

#Extract scaled integrated
matrix_scRNAseq_Seurat_integrated <- AverageExpression(object     = scRNAseq_Seurat_integrated,
                                                       assays     = "RNA"
                                                       )$RNA

head(matrix_scRNAseq_Seurat_integrated)
nrow(matrix_scRNAseq_Seurat_integrated)

# Export matrix
write.table(x      = matrix_scRNAseq_Seurat_integrated,
            file   = paste0(pathToReport,"matrix_scRNAseq_Seurat_integrated.txt"),
            sep    = "\t",
            quote  = FALSE
            )

```

### Prepare mixture file

CIBERSORT need tab-delimited format (see CIBERSORT tutorial for more details).

```{r mixture_file}

# Create column "Gene" from rownames
matrix_scRNAseq_Seurat_integrated_mixMat <- add_column(.data    = matrix_scRNAseq_Seurat_integrated,
                                                       Gene     = rownames(matrix_scRNAseq_Seurat_integrated),
                                                       .before  = 1
                                                       )
rownames(matrix_scRNAseq_Seurat_integrated_mixMat) <- NULL
head(matrix_scRNAseq_Seurat_integrated_mixMat)

# Export mixture file for CIBERSORT
write.table(x          = as.matrix(matrix_scRNAseq_Seurat_integrated_mixMat),
            file       = paste0(pathToReport,"CIBERSORT/matrix_scRNAseq_Seurat_integrated_mixMat.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```

### Import mouse signature matrix

```{r import_mouse_signature_matrix}

# Import mouse signature matrix
mouseSignature_valdoiu2019 <- read.table(file         = paste0(pathToData,"mouseSignatures_vladoiu2019.csv"),
                                          header       = TRUE,
                                          sep          = ";",
                                          dec          = ",",
                                          check.names  = FALSE
                                          )
head(mouseSignature_valdoiu2019)
ncol(mouseSignature_valdoiu2019)
nrow(mouseSignature_valdoiu2019)

```

### Prepare signature matrix

```{r signature_matrix}

# Import cell_type num, name and lineage
vladoiu_cellTypes <- read.table(file    = paste0(pathToData,"vladoiu2019_cellTypes.csv"),
                                header  = TRUE,
                                sep     = ";"
                                )
head(vladoiu_cellTypes)
nrow(vladoiu_cellTypes)

# Select cluster in mouse signature matrix
vladoiu_cellTypes_cibersort <- filter(.data = vladoiu_cellTypes,
                                      cluster_num %in% colnames(mouseSignature_valdoiu2019)
                                      )
head(vladoiu_cellTypes_cibersort)
nrow(vladoiu_cellTypes_cibersort)

# Check order
colnames(mouseSignature_valdoiu2019) == c("Gene", vladoiu_cellTypes_cibersort$cluster_num)

# Put cluster name as colnames
mouseSignature_valdoiu2019_cibersort <- mouseSignature_valdoiu2019
colnames(mouseSignature_valdoiu2019_cibersort) <- c("Gene", as.vector(vladoiu_cellTypes_cibersort$cluster_name))

head(mouseSignature_valdoiu2019_cibersort)
nrow(mouseSignature_valdoiu2019_cibersort)

# Export signature matrix
write.table(x          = mouseSignature_valdoiu2019_cibersort,
            file       = paste0(pathToReport,"CIBERSORT/mouseSignature_valdoiu2019_cibersort.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```


### Run CIBERSORT Fraction

For the moment CIBERSORT fraction is run from the website platform.
The result (fraction) is plotted as heatmap below.

```{r geneSubset}

# Import CIBERSORT Fraction results
CIBERSORTx_Job32_scRNAseq_Seurat_integrated <- read.table(file         = paste0(pathToReport,"CIBERSORT/CIBERSORTx_Job32_scRNAseq_Seurat_integrated.txt"),
                                                          sep          = "\t",
                                                          header       = TRUE,
                                                          check.names  = FALSE
                                                          )
head(CIBERSORTx_Job32_scRNAseq_Seurat_integrated)

# Put Mixture column (Sample name) as rownames and remove P-value, Correlation and RMSE columns
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min <- CIBERSORTx_Job32_scRNAseq_Seurat_integrated
rownames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min)   <- CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min$Mixture
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min$Mixture     <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min$`P-value`   <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min$Correlation <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min$RMSE        <- NULL
head(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min)

# Density
melt(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min) %>%
    ggplot(mapping = aes(x = value)
           ) +
    geom_density() +
    geom_density(mapping = aes(x = -log10(value)),
                 color = "blue"
                 )

rownames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min)

# Check column order
vladoiu_cellTypes_cibersort$cluster_name == colnames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min)

# Define heatmap color
col_fun_mouseVladoiu2019 = colorRamp2(breaks = c(0,0.1,0.6), colors = c("lightskyblue","navajowhite","red"))

# Plot heatmap
h <- Heatmap(matrix                       = CIBERSORTx_Job32_scRNAseq_Seurat_integrated_min,
             cluster_rows                 = TRUE,
             cluster_columns              = TRUE,
             show_row_names               = TRUE,
             show_column_names            = TRUE,
             show_row_dend                = TRUE,
             show_column_dend             = FALSE,
             clustering_distance_rows     = "euclidean",
             clustering_method_rows       = "ward",
             clustering_distance_columns  = "euclidean",
             clustering_method_columns    = "ward",
             #row_order                   = order(sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup),
             #row_split                    = sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup,
             column_split                 = vladoiu_cellTypes_cibersort$cluster_lineage,
             cluster_row_slices           = TRUE,
             cluster_column_slices        = FALSE,
             column_names_gp              = gpar(fontsize = 20),
             column_names_max_height      = unit(15,"cm"),
             row_names_gp                 = gpar(fontsize = 20),
             show_heatmap_legend          = FALSE,
             row_split                    = 3,
             column_title_gp              = gpar(fontsize = 20, fontface="bold"),
             row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
             col                          = col_fun_mouseVladoiu2019,
             row_title                    = c("A","B","C")
             )
#pdf(file = paste0(pathToFigures,"heatmap_CIBERSORT_fraction_scRNAseq_integrated.pdf"), width = 20, height =10) 
draw(h)
#dev.off()

# Legend of relative abundance
legend_fraction <- Legend(col_fun         = col_fun_mouseVladoiu2019,
                          title           = "relative abundance\n",
                          grid_height     = unit(1,"cm"),
                          legend_width    = unit(10,"cm"),
                          title_gp        = gpar(fontsize = 20, fontface = "bold"),
                          title_position  = "topcenter",
                          direction       = "horizontal",
                          labels_gp       = gpar(fontsize=15)
                          )

# Combine legends
legends <- packLegend(list        = list(legend_fraction),
                      direction   ="horizontal",
                      column_gap  = unit(2,"cm")
                      )

#pdf(file = paste0(pathToFigures,"legend_heatmap_CIBERSORT_fraction_scRNAseq_integrated.pdf"), width = 20, height =10) 
draw(legends)
#dev.off()

```


## Too-many-cells

### Extract matrix of cells

```{r extract_cell_matrix}

# Extract integrated assay table
scRNAseq_Seurat_integrated_matrix <- scRNAseq_Seurat_integrated_cellAnnot@assays$integrated@data
head(scRNAseq_Seurat_integrated_matrix[,1:3])

# Export table
write.table(x      = scRNAseq_Seurat_integrated_matrix,
            file   = paste0(pathToReport,"scRNAseq_Seurat_integrated_matrix.csv"),
            sep    = ",",
            quote  = FALSE
            )

```

### Export data for figshare (Lobon-Iglesias et al., 2022)

```{r export_data}

# Export table (for figshare, Lobon-Iglesias et al., 2022)
write.table(x      = scRNAseq_Seurat_integrated_matrix,
            file   = paste0(pathToReport,"scRNAseq_INI254_INI255_INI267_integrated_matrix.csv"),
            sep    = ",",
            quote  = FALSE
            )

colnames(scRNAseq_Seurat_integrated@meta.data)

head(scRNAseq_Seurat_integrated@meta.data)

# Prepare meta data for figshare
scRNAseq_INI254_INI255_INI267_integrated_metadata <- scRNAseq_Seurat_integrated@meta.data
#
scRNAseq_INI254_INI255_INI267_integrated_metadata <- rownames_to_column(.data = scRNAseq_INI254_INI255_INI267_integrated_metadata,
                                                                        var = "cell_barcode"
                                                                        )
#
scRNAseq_INI254_INI255_INI267_integrated_metadata <- dplyr::select(scRNAseq_INI254_INI255_INI267_integrated_metadata,
                                                                   cell_barcode,
                                                                   orig.ident,
                                                                   nCount_RNA,
                                                                   nFeature_RNA,
                                                                   percent.mito,
                                                                   S.Score,
                                                                   G2M.Score,
                                                                   Phase,
                                                                   integrated_snn_res.0.2
                                                                   ) %>%
    mutate(patient_ID = orig.ident,
           cluster = integrated_snn_res.0.2,
           .keep = "unused"
           )

head(scRNAseq_INI254_INI255_INI267_integrated_metadata)

# Export table (for figshare, Lobon-Iglesias et al., 2022)
write.table(x      = scRNAseq_INI254_INI255_INI267_integrated_metadata,
            file   = paste0(pathToReport,"scRNAseq_INI254_INI255_INI267_integrated_metadata.csv"),
            sep    = ",",
            quote  = FALSE
            )

```

### Extract meta data

```{r extract_metadata}

# Extract meta data
scRNAseq_Seurat_integrated_metadata <- scRNAseq_Seurat_integrated@meta.data

scRNAseq_Seurat_integrated_metadata <- rownames_to_column(.data = scRNAseq_Seurat_integrated_metadata,
                                                          var   = "item"
                                                          )

# Export table
write.table(x          = scRNAseq_Seurat_integrated_metadata,
            file       = paste0(pathToReport,"scRNAseq_Seurat_integrated_metadata.csv"),
            sep        = ",",
            quote      = FALSE,
            row.names  = FALSE
            )

```

### Launch too-many-cells

```{r launch_too_many_cells}

pathToReport

system2(command = "/bioinfo/local/build/singularity/singularity-3.5.2/bin/singularity",
        args = "run /bioinfo/local/Singularity/too-many-cells/too-many-cells_0.2.2.0.simg make-tree --matrix-path /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_Seurat_integrated_matrix.csv --normalization NoneNorm  --labels-file /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_Seurat_integrated_metadata.csv --output /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/tooManyCells_INI254"
        )

```

## Trajectory reconstruction

### ElpiGraph method

#### TO REMOVE: elpiGraph tutorial

```{r cellCycle}

##################
# sincellTE tuto #
##################

library(reticulate)

sk <- import("sklearn.manifold")

# MLLE projection
MLLE.Model <- sk$LocallyLinearEmbedding(n_neighbors=as.integer(ceiling(nrow(scRNAseq_Seurat_integrated_matrix)*.1)),
                                        n_components = as.integer(3),
                                        method = 'modified',
                                        eigen_solver = 'dense',
                                        # n_jobs = as.integer(4),
                                        random_state = as.integer(10),
                                        neighbors_algorithm = 'kd_tree')

MLLE.Fit <- MLLE.Model$fit(ExpMat)
MLLE.Emb <- MLLE.Fit$embedding_







# Get igraph network object (to use in getPseudotime function below)
scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph <- ConstructGraph(PrintGraph = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]])

scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph

# Get the substructure of interest (here the whole nodes)
scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle <- GetSubGraph(Net        = scRNAseq_Seurat_integrated_ICA_ElPiGraph_iGraph,
                                                               Structure  = "circle",
                                                               Circular   = TRUE,
                                                               KeepEnds   = TRUE,
                                                               Nodes      = 50
                                                               )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle

# Project point onto graph
scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection <- project_point_onto_graph(X              = scRNAseq_Seurat_integrated_ICA_matrix,
                                                                                NodePositions  = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]]$NodePositions,
                                                                                Edges          = scRNAseq_Seurat_integrated_ICA_ElPiGraph[[11]]$Edges$Edges
                                                                                )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection

# Compute pseudotime
scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime <- getPseudotime(ProjStruct = scRNAseq_Seurat_integrated_ICA_ElPiGraph_projection,
                                                                     NodeSeq = scRNAseq_Seurat_integrated_ICA_ElPiGraph_circle$Circle_1
                                                                     )

scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime

# Plot pseudotime distribution
ggplot(data = data.frame(Pt = scRNAseq_Seurat_integrated_ICA_ElPiGraph_pseudotime$Pt)
       ) +
    geom_histogram(mapping = aes(x = Pt),
                   fill = "grey",
                   color = "black"
                   ) +
    theme_bw()

```

#### Trajectory reconstruction for all clusters

##### Prepare matrix

```{r cellCycle}

# Extract expression matrix
scRNAseq_Seurat_integrated_cellAnnot_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot,
                                                                      assay = "integrated",
                                                                      slot = "scale.data"
                                                                      )
                                                         )
dim(scRNAseq_Seurat_integrated_cellAnnot_matrix)
head(scRNAseq_Seurat_integrated_cellAnnot_matrix[,1:4])

```

##### Principal tree

```{r cellCycle}

# Construct a principal Tree
scRNAseq_Seurat_integrated_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                                                                         NumNodes    = 30,
                                                                         Lambda      = .03,
                                                                         Mu          = 0.1,
                                                                         Do_PCA      = TRUE,
                                                                         CenterData  = TRUE,
                                                                         drawAccuracyComplexity = FALSE,
                                                                         drawPCAView = FALSE,
                                                                         drawEnergy  = FALSE
                                                                         )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_all_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot@meta.data$orig.ident
                )
#dev.off()

```

#### Trajectory reconstruction for cluster 0, 1, 3 and 5

##### Extract cluster 0, 1, 3 and 5 seurat object and matrix

```{r extract_cluster0135_object}

# Select cluster 0, 1, 3 and 5 marker genes
head(scRNAseq_Seurat_integrated_choosenResolution.markers)

cluster0135_DEG <- rownames(filter(scRNAseq_Seurat_integrated_choosenResolution.markers,
       cluster %in% c(0,1,3,5)))
#cluster0135_DEG <- c(cluster0135_DEG,"TTYH1","SMARCB1")
head(cluster0135_DEG)
length(cluster0135_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_cellAnnot_0135 <- subset(x = scRNAseq_Seurat_integrated_cellAnnot,
                                                    integrated_snn_res.0.2 %in% c(0,1,3,5),
                                                    features = cluster0135_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_cellAnnot_0135)

# Extract expression matrix
scRNAseq_Seurat_integrated_cellAnnot_0135_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot_0135,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "sparseMatrix"
   )
nrow(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)

```

##### Principal Tree

```{r cellCycle}

# See below for matrix subsetting
dim(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)

# Construct a principal tree
scRNAseq_Seurat_integrated_ElPiGraph_tree_0135 <- computeElasticPrincipalTree(X                       = t(as.matrix(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)),
                                                                              NumNodes                = 30,
                                                                              Lambda                  = .03,
                                                                              Mu                      = .01,
                                                                              Do_PCA                  = TRUE,
                                                                              CenterData              = TRUE,
                                                                              drawAccuracyComplexity  = FALSE,
                                                                              drawPCAView             = FALSE,
                                                                              drawEnergy              = FALSE
                                                                              )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cluster0135_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_integrated_ElPiGraph_curve_0135 <- computeElasticPrincipalCurve(X                       = t(as.matrix(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)),
                                                                                NumNodes                = 30,
                                                                                Lambda                  = .001,
                                                                                Mu                      = .1,
                                                                                Do_PCA                  = TRUE,
                                                                                CenterData              = TRUE,
                                                                                drawAccuracyComplexity  = FALSE,
                                                                                drawPCAView             = FALSE,
                                                                                drawEnergy              = FALSE
                                                                                )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$orig.ident
                )
#dev.off()


########## for seminar presentation ##########

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cellType_presentation.pdf"), width = 16, height = 8)
PlotPG_modified2(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                PGCol = "yellow",
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_type
                ) #+
    #scale_fill_manual(name = "Cell type", values = c("blue","darkgreen","grey40"))
#dev.off()

# Plot principal graph with cell type without legend
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cellType_presentation_noLegend.pdf"), width = 14, height = 8)
PlotPG_modified3(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                PGCol = "yellow",
                                        #PointSize = 1.5,
                p.alpha = 1,
                colors = c("blue","darkgreen","grey40"),
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_type
                )
#dev.off()


########## for Maria's thesis defense ##########

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_0135_curve_cellType_thesisDefense.pdf"), width = 16, height = 8)
PlotPG_modified3(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                pathCol = "red",
                                        #PointSize = 2,
                pointSize = 2,
                p.alpha = 1,
                panel.background = "black",
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_subtype,
                colors = c("springgreen","springgreen4","skyblue","skyblue4")
                )
#dev.off()

elpiGraph_scRNAseq_integrated_0135_curve_cellType <- PlotPG_modified3(X         = t(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_0135)-1)],
                pathCol = "red",
                pointSize = 0.5,
                p.alpha = 1,
                panel.background = "white",
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data$cell_subtype,
                colors = c("springgreen","springgreen4","skyblue","skyblue4")
                )

```

#### Trajectory reconstruction for cluster 0, 1, 3, 5, 6 and 7

##### Extract cluster 0, 1, 3, 5, 6 and 7 seurat object and matrix

```{r extract_cluster013567_object}

# Select cluster marker genes
head(scRNAseq_Seurat_integrated_choosenResolution.markers)

cluster013567_DEG <- rownames(filter(scRNAseq_Seurat_integrated_choosenResolution.markers,
       cluster %in% c(0,1,3,5,6,7)))
#cluster013567_DEG <- c(cluster013567_DEG,"TTYH1","SMARCB1")
head(cluster013567_DEG)
length(cluster013567_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_cellAnnot_013567 <- subset(x = scRNAseq_Seurat_integrated_cellAnnot,
                                                    integrated_snn_res.0.2 %in% c(0,1,3,5,6,7),
                                                    features = cluster013567_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_cellAnnot_013567)

# Extract expression matrix
scRNAseq_Seurat_integrated_cellAnnot_013567_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot_013567,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "sparseMatrix"
   )
nrow(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)

```

##### Principal Tree

```{r cellCycle}

# See below for matrix subsetting
dim(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)

# Construct a principal tree
scRNAseq_Seurat_integrated_ElPiGraph_tree_013567 <- computeElasticPrincipalTree(X                       = t(as.matrix(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)),
                                                                              NumNodes                = 30,
                                                                              Lambda                  = .03,
                                                                              Mu                      = .01,
                                                                              Do_PCA                  = TRUE,
                                                                              CenterData              = TRUE,
                                                                              drawAccuracyComplexity  = FALSE,
                                                                              drawPCAView             = FALSE,
                                                                              drawEnergy              = FALSE
                                                                              )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cluster013567_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_integrated_ElPiGraph_curve_013567 <- computeElasticPrincipalCurve(X                       = t(as.matrix(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)),
                                                                                NumNodes                = 30,
                                                                                Lambda                  = .001,
                                                                                Mu                      = .1,
                                                                                Do_PCA                  = TRUE,
                                                                                CenterData              = TRUE,
                                                                                drawAccuracyComplexity  = FALSE,
                                                                                drawPCAView             = FALSE,
                                                                                drawEnergy              = FALSE
                                                                                )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$integrated_snn_res.0.2
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_013567_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data$orig.ident
                )
#dev.off()

```


### Monocle version 3 method

#### Monocle on all clusters

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_cellAnnot_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_cellAnnot@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_matrix)
                                                                               )
                                                    )


#nbPCs <- c(12,seq(10,100,10))
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_cds
                                              )
    scRNAseq_Seurat_integrated_cds_list[[length(scRNAseq_Seurat_integrated_cds_list)+1]] <- scRNAseq_Seurat_integrated_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_list[[i]],
                                                  color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_cds_list)[i])
scRNAseq_Seurat_integrated_cds_plot_list[[length(scRNAseq_Seurat_integrated_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_cds_plot
}

library(gridExtra)

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_cds_plot_list, ncol = 3))
#dev.off()

```

#### Monocle on cluster 0, 1, 3 and 5

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_cds_0135 <- new_cell_data_set(expression_data = scRNAseq_Seurat_integrated_cellAnnot_0135_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)
                                                                                    )
                                                         )

nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_cds_0135_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_cds_0135 <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_cds_0135,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
scRNAseq_Seurat_integrated_cds_0135 <- align_cds(cds = scRNAseq_Seurat_integrated_cds_0135,
                                                 alignment_group = "orig.ident"
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_cds_0135 <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_cds_0135,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_integrated_cds_0135 <- cluster_cells(cds               = scRNAseq_Seurat_integrated_cds_0135,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_integrated_cds_0135 <- learn_graph(cds = scRNAseq_Seurat_integrated_cds_0135)
    scRNAseq_Seurat_integrated_cds_0135_list[[length(scRNAseq_Seurat_integrated_cds_0135_list)+1]] <- scRNAseq_Seurat_integrated_cds_0135
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_cds_0135_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_cds_0135_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_cds_0135_list))
{
# Plot
scRNAseq_Seurat_integrated_cds_0135_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[[i]],
                                                       color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_cds_0135_list)[i])
    scRNAseq_Seurat_integrated_cds_0135_plot_list[[length(scRNAseq_Seurat_integrated_cds_0135_plot_list)+1]] <- scRNAseq_Seurat_integrated_cds_0135_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_cds_0135_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = TRUE,
           label_leaves                   = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

##### Monocle3 plot for seminar's presentation

```{r plot_monocle_for_presentation}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_presentation.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "yellow",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type", values = c("blue","darkgreen","grey40")) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black")
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_presentation_noLegend.pdf"), width = 14, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "yellow",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type", values = c("blue","darkgreen","grey40")) +
    theme(text = element_text(size = 30),
          panel.background = element_rect(fill = "black"),
          legend.position = "none"
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_presentation2.pdf"), width = 18, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "grey",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type", values = c("blue","darkgreen","grey40")) +
    geom_segment(aes(x = -5, xend = 0, y = 0.5, yend = 0.5), arrow = arrow(ends = "last"), colour = "red", lwd = 3) +
    geom_text(aes(x = -6.5, y = 0.5), label = "NOTCH\nLIGANDS", fontface = "bold", colour = "yellow", size = 10) +
    geom_text(aes(x = 2, y = 0.5), label = "NOTCH\nRECEPTORS", fontface = "bold", colour = "yellow", size = 10) +
    geom_text(aes(x = -11, y = -2), label = "3", fontface = "bold", colour = "darkgreen", size = 20) +
    geom_text(aes(x = -6, y = 2), label = "5", fontface = "bold", colour = "blue", size = 20) +
    geom_text(aes(x = 1, y = 2), label = "0-1", fontface = "bold", colour = "grey40", size = 20) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black"),
          legend.position = "none"
          )
#dev.off()

```


##### Monocle3 plot for Maria's thesis presentation

```{r plot_monocle_for_presentation}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_thesisDefense.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 2,
           trajectory_graph_color         = "red",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type", values = c("springgreen","springgreen4","skyblue","skyblue4")) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black"),
          legend.position = "none"
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_thesisDefense2.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 2,
           trajectory_graph_color         = "grey",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type", values = c("springgreen","springgreen4","skyblue","skyblue4")) +
    geom_segment(aes(x = -5, xend = 0, y = 0.5, yend = 0.5), arrow = arrow(ends = "last"), colour = "red", lwd = 3) +
    geom_text(aes(x = -6.5, y = 0.5), label = "LIGANDS", fontface = "bold", colour = "yellow", size = 15) +
    geom_text(aes(x = 2, y = 0.5), label = "RECEPTORS", fontface = "bold", colour = "yellow", size = 15) +
    geom_text(aes(x = -11, y = -2), label = "3", fontface = "bold", colour = "springgreen4", size = 20) +
    geom_text(aes(x = -6, y = 2), label = "5", fontface = "bold", colour = "springgreen", size = 20) +
    geom_text(aes(x = 1, y = 2), label = "0-1", fontface = "bold", colour = "skyblue2", size = 20) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black"),
          legend.position = "none"
          )
#dev.off()

pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_thesisDefense2.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 2,
           trajectory_graph_color         = "red",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    scale_colour_manual(name = "Cell type",
                        values = c("springgreen","springgreen4","skyblue","skyblue4"),
                        labels = c("NPL2","NPL1","UD0","UD1")
                        ) +
    theme(text = element_text(size = 40),
          #panel.background = element_rect(fill = "black"),
          legend.position = "right"
          )
dev.off()

```

```{r plot_monocle_for_figure}

trajectory_inference_monocle3_clust0135 <- plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
                                                      color_cells_by                 = "integrated_snn_res.0.2",
                                                      cell_size                      = 0.25,
                                                      trajectory_graph_color         = "red",
                                                      trajectory_graph_segment_size  = 0.75,
                                                      label_cell_groups              = FALSE,
                                                      label_branch_points            = FALSE,
                                                      label_roots                    = FALSE,
                                                      label_leaves                   = FALSE
                                                      ) +
    scale_colour_manual(name = "Cell type",
                        values = cell_cluster_col[levels(factor(scRNAseq_Seurat_integrated_cds_0135_list$PCmin@colData$integrated_snn_res.0.2))]
                        ) +
    theme_void() +
    theme(text = element_text(size = paperFig_fontsize),
          panel.background = element_rect(colour = "white", fill = "white"),
          legend.position = 'none'
          )

trajectory_inference_monocle3_clust0135

```

```{r plot_monocle_without_celltype}

# TO REMOVE: For meeting with Kornelius team to avoid giving more info
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellType_noName.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40),
          legend.position = "none"
          )
#dev.off()

```

##### Plot monocle and covariates

```{r monocle_and_covariates}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cluster.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "integrated_snn_res.0.2",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_patientID.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "orig.ident",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_cellSubType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

##### Pseudotime (monocle)

```{r pseudotime_analysis}

# a helper function to identify the root principal points:
get_earliest_principal_node <- function(cds, cell_type="Neuronal progenitor"){
    cell_ids <- which(colData(cds)[, "cell_type"] == cell_type)

    closest_vertex <-
        cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
    closest_vertex <- as.matrix(closest_vertex[colnames(cds), ])
    root_pr_nodes <-
        igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
        (which.max(table(closest_vertex[cell_ids,]))))]

    root_pr_nodes
}

# Order cells
scRNAseq_Seurat_integrated_cds_0135_pseudotime <- order_cells(cds = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
                                                              reduction_method = "UMAP",
                                                              root_pr_nodes = get_earliest_principal_node(scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]])
                                                              )

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_pseudotime.pdf"), width = 20, height = 10)
plot_cells(cds                                           = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           color_cells_by                                = "pseudotime",
                                        #group_cells_by  = "cell_type",
           cell_size                                     = 1,
           trajectory_graph_color                        = "black",
           trajectory_graph_segment_size                 = 2,
           label_cell_groups                             = FALSE,
           label_branch_points                           = FALSE,
           label_roots                                   = FALSE,
           label_leaves                                  = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

# Plot for paper
pseudotime_monocle3_clust0135 <- plot_cells(cds                                           = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           color_cells_by                                = "pseudotime",
                                        #group_cells_by  = "cell_type",
           cell_size                                     = 0.25,
           trajectory_graph_color                        = "red",
           trajectory_graph_segment_size                 = 0.75,
           label_cell_groups                             = FALSE,
           label_branch_points                           = FALSE,
           label_roots                                   = FALSE,
           label_leaves                                  = FALSE
           ) +
        theme_void() +
    theme(text = element_text(size = paperFig_fontsize),
          panel.background = element_rect(colour = "white", fill = "white"),
legend.key.size = unit(0.25,'cm')
          )
#pseudotime_monocle3_clust0135

```

##### Plot genes of interest on monocle embedding

```{r genes_of_interest_on_monocle_plot}

lineage_genes <- c("STMN4","DLL3","HES6")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_lineageGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = lineage_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

NOTCH_pathway_genes <- c("HES1","HES4","HES6","DLL1","DLL3","STAT3","YAP1","DNER","JAG1","RBPJ","MAML2","ASCL1","ROBO1","ROBO2","ZMIZ1","MAPT","HDAC1","HDAC2","TLE1","NOTCH2","NOTCH3","PDCD10","PBX1","WWC1","CCND1","SAP30","TCF7L2","NCSTN")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_notchGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = NOTCH_pathway_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

NOTCH_pathway_selected_genes <- c("HES6","DLL3","ASCL1","HDAC2")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_notchSelectedGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = NOTCH_pathway_selected_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_HES6.pdf"), width = 14, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = "HES6",
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_DLL3.pdf"), width = 14, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = "DLL3",
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_ASCL1.pdf"), width = 14, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = "ASCL1",
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust0135_HDAC2.pdf"), width = 14, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_pseudotime,
           genes = "HDAC2",
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

#### Monocle on cluster 0, 1, 3, 5, 6 and 7

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_cds_013567 <- new_cell_data_set(expression_data = scRNAseq_Seurat_integrated_cellAnnot_013567_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)
                                                                                    )
                                                         )

nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_cds_013567_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_cds_013567 <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_cds_013567,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
scRNAseq_Seurat_integrated_cds_013567 <- align_cds(cds = scRNAseq_Seurat_integrated_cds_013567,
                                                 alignment_group = "orig.ident"
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_cds_013567 <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_cds_013567,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_integrated_cds_013567 <- cluster_cells(cds               = scRNAseq_Seurat_integrated_cds_013567,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_integrated_cds_013567 <- learn_graph(cds = scRNAseq_Seurat_integrated_cds_013567)
    scRNAseq_Seurat_integrated_cds_013567_list[[length(scRNAseq_Seurat_integrated_cds_013567_list)+1]] <- scRNAseq_Seurat_integrated_cds_013567
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_cds_013567_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_cds_013567_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_cds_013567_list))
{
# Plot
scRNAseq_Seurat_integrated_cds_013567_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_list[[i]],
                                                       color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_cds_013567_list)[i])
    scRNAseq_Seurat_integrated_cds_013567_plot_list[[length(scRNAseq_Seurat_integrated_cds_013567_plot_list)+1]] <- scRNAseq_Seurat_integrated_cds_013567_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_cds_013567_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_cluster.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]],
           color_cells_by                 = "integrated_snn_res.0.2",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_patientID.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]],
           color_cells_by                 = "orig.ident",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_cellSubType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

# a helper function to identify the root principal points:
get_earliest_principal_node <- function(cds, cell_type="Neuronal progenitor"){
    cell_ids <- which(colData(cds)[, "cell_type"] == cell_type)
    closest_vertex <-
        cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
    closest_vertex <- as.matrix(closest_vertex[colnames(cds), ])
    root_pr_nodes <-
        igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
        (which.max(table(closest_vertex[cell_ids,]))))]
    root_pr_nodes
}

# Order cells
scRNAseq_Seurat_integrated_cds_013567_pseudotime <- order_cells(cds = scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]],
                                                              reduction_method = "UMAP",
                                                              root_pr_nodes = get_earliest_principal_node(scRNAseq_Seurat_integrated_cds_013567_list[["PCmin"]])
                                                              )

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_pseudotime.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_pseudotime,
           color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

lineage_genes <- c("STMN4","DLL3","HES6")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_clust013567_lineageGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_013567_pseudotime,
           genes = lineage_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

### Monocle version 2 method

#### Monocle2 on all clusters

```{r monocle_method}

library(monocle)

# Create monocle2 cds object
scRNAseq_Seurat_integrated_cds_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_cellAnnot_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_cellAnnot@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )


# Create Monocle2 cds object
#scRNAseq_Seurat_integrated_cds_mncl2 <- as.CellDataSet(x = scRNAseq_Seurat_integrated_cellAnnot,
 #                                                      assay = "RNA"
  #                                                     )

# Choose genes that define a cell's progress
HVF <- VariableFeatures(object = scRNAseq_Seurat_integrated_cellAnnot,
                        assay = "integrated"
                        )

scRNAseq_Seurat_integrated_cds_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                                                          ordering_genes = HVF
                                                          )

# Dimensionality reduction
scRNAseq_Seurat_integrated_cds_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                                                        max_components = 2,
                                                        norm_method = "none",
                                                        relative_expr = TRUE,
                                                        scaling = TRUE,
                                                        residualModelFormulaStr = "~ orig.ident",
                                                        reduction_method = "DDRTree"
                                                        )

# Order cells along trajectory
scRNAseq_Seurat_integrated_cds_mncl2 <- orderCells(scRNAseq_Seurat_integrated_cds_mncl2)

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_allClusters_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_allClusters_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                     color_by = "integrated_snn_res.0.2"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_allClusters_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_allClusters_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

```

#### Monocle2 on cluster 0,1,3 and 5

```{r monocle_method}

# Create monocle2 cds object
scRNAseq_Seurat_integrated_cds_0135_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_cellAnnot_0135_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_0135_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )



# Choose genes that define a cell's progress
head(cluster0135_DEG)
length(cluster0135_DEG)


test <- scRNAseq_Seurat_integrated_cds_0135_mncl2[c("ACTL6B","STMN4","ELAVL4","DCX","GAP43","ID4"),]

plot_pseudotime_heatmap(cds_subset     = test,
                        show_rownames  = TRUE,
                        num_clusters   = 3
                        )

#scRNAseq_Seurat_integrated_cds_0135_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
 #                                                              ordering_genes = cluster0135_DEG
  #                                                             )

# Dimensionality reduction
scRNAseq_Seurat_integrated_cds_0135_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
                                                             max_components = 2,
                                                             norm_method = "none",
                                                             relative_expr = TRUE,
                                                             scaling = TRUE,
                                                             residualModelFormulaStr = "~ orig.ident",
                                                             reduction_method = "DDRTree"
                                                             )

# Order cells along trajectory
scRNAseq_Seurat_integrated_cds_0135_mncl2 <- orderCells(scRNAseq_Seurat_integrated_cds_0135_mncl2)

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust0135_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust0135_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
                     color_by = "integrated_snn_res.0.2"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust0135_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust0135_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_0135_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

```

#### Monocle2 on cluster 0,1,3,5,6,7

```{r monocle_method}

# Create monocle2 cds object
scRNAseq_Seurat_integrated_cds_013567_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_cellAnnot_013567_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_cellAnnot_013567@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_cellAnnot_013567_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )



# Choose genes that define a cell's progress
head(cluster013567_DEG)
length(cluster013567_DEG)

#scRNAseq_Seurat_integrated_cds_013567_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
 #                                                              ordering_genes = cluster013567_DEG
  #                                                             )

# Dimensionality reduction
scRNAseq_Seurat_integrated_cds_013567_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
                                                             max_components = 2,
                                                             norm_method = "none",
                                                             relative_expr = TRUE,
                                                             scaling = TRUE,
                                                             residualModelFormulaStr = "~ orig.ident",
                                                             reduction_method = "DDRTree"
                                                             )

# Order cells along trajectory
scRNAseq_Seurat_integrated_cds_013567_mncl2 <- orderCells(scRNAseq_Seurat_integrated_cds_013567_mncl2)

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust013567_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust013567_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
                     color_by = "integrated_snn_res.0.2"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust013567_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_clust013567_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_cds_013567_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

```


### CytoTRACE method

CytoTRACE predicts the differentiation state of cells using gene Counts and expression.

#### CytoTRACE all clusters

```{r cytoTrace}

# Extract raw count expression matrix
scRNAseq_Seurat_integrated_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot,
                                                                      assay  = "RNA",#"integrated",
                                                                      slot   = "counts"#"scale.data"
                                                                      )
                                                         )
head(scRNAseq_Seurat_integrated_rawCounts_matrix[,1:4])
nrow(scRNAseq_Seurat_integrated_rawCounts_matrix)

# Run CytoTRACE
scRNAseq_Seurat_integrated_cytoTrace <- CytoTRACE(mat            = scRNAseq_Seurat_integrated_rawCounts_matrix,
                                                  batch          = scRNAseq_Seurat_integrated_cellAnnot@meta.data$orig.ident,
                                                  enableFast     = TRUE,
                                                  ncores         = 1,
                                                  subsamplesize  = 1000
                                                  )
str(scRNAseq_Seurat_integrated_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_integrated@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_integrated_cytoTrace$CytoTRACE)))

scRNAseq_Seurat_integrated_allGenes <- scRNAseq_Seurat_integrated

# Append metadata with CytoTRACE
scRNAseq_Seurat_integrated_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_allGenes,
                                                         metadata = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_cytoTrace$CytoTRACE)
                                                         )

colnames(scRNAseq_Seurat_integrated_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust_CytoTRACE.pdf"), width = 16, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_allGenes,
            pt.size = 1,
            features = "CytoTRACE",
            reduction = "umap",
            cols     = c("red","yellow","grey","cyan","blue"),
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_integrated_allGenes,
                              clustering  = paste0("integrated_snn_res.",choosenResolution),
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "mean"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_integrated_cytoGenes <- data.frame(corr = scRNAseq_Seurat_integrated_cytoTrace$cytoGenes)
scRNAseq_Seurat_integrated_cytoGenes <- rownames_to_column(scRNAseq_Seurat_integrated_cytoGenes, var = "gene")
head(scRNAseq_Seurat_integrated_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_integrated_cytoGenes)-5):nrow(scRNAseq_Seurat_integrated_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```


#### CytoTRACE 0135

```{r cytoTrace}

# Subset cluster 0,1,3 and 5 with all clusters
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes <- subset(x = scRNAseq_Seurat_integrated_cellAnnot,
                                                             integrated_snn_res.0.2 %in% c(0,1,3,5)#,
                                                             #features = cluster0135_DEG
                                                             )
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes


# Extract expression matrix
scRNAseq_Seurat_integrated_0135_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "counts"#"scale.data"
                                                                           )
                                                                )
dim(scRNAseq_Seurat_integrated_0135_rawCounts_matrix)
head(scRNAseq_Seurat_integrated_0135_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_integrated_0135_cytoTrace <- CytoTRACE(mat            = scRNAseq_Seurat_integrated_0135_rawCounts_matrix,
                                                       batch          = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data$orig.ident,
                                                       enableFast     = TRUE,
                                                       ncores         = 1,
                                                       subsamplesize  = 1000
                                                       )
str(scRNAseq_Seurat_integrated_0135_cytoTrace)


# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_0135_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_0135_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_integrated_0135_cytoTrace$CytoTRACE)))


# Append metadata with CytoTRACE
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes,
                                                                  metadata = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_0135_cytoTrace$CytoTRACE)
                                                                  )

colnames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data)

# Check order
all(rownames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data) == rownames(scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData))

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes,
                                                                  metadata = scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                                  )

colnames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellAnnot_0135_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellAnnot_0135_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = integrated_snn_res.0.2
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_cellAnnot_0135_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes,
                              clustering  = paste0("integrated_snn_res.",choosenResolution),
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "mean"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_integrated_0135_cytoGenes <- data.frame(corr = scRNAseq_Seurat_integrated_0135_cytoTrace$cytoGenes)
scRNAseq_Seurat_integrated_0135_cytoGenes <- rownames_to_column(scRNAseq_Seurat_integrated_0135_cytoGenes, var = "gene")
head(scRNAseq_Seurat_integrated_0135_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_cellAnnot_0135_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_0135_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_integrated_0135_cytoGenes)-5):nrow(scRNAseq_Seurat_integrated_0135_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```


## RNA velocity

### Extract meta-data for scVelo 

#### Extract all cluster meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_cellAnnot)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_cellAnnot)),
          file = "scRNAseq_Seurat_integrated_cellAnnot_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_cellAnnot, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_cellAnnot_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_cellAnnot@meta.data,
          file = "scRNAseq_Seurat_integrated_cellAnnot_metaData.csv"
          )

```

#### Extract cluster  0,1,3,5 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_cellAnnot_0135)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_cellAnnot_0135)),
          file = "scRNAseq_Seurat_integrated_cellAnnot_0135_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_cellAnnot_0135, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_cellAnnot_0135_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_cellAnnot_0135@meta.data,
          file = "scRNAseq_Seurat_integrated_cellAnnot_0135_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_cds_0135_list[["PCmin"]],
                color_cells_by                 = "cell_subtype",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]

head(scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_cellAnnot_0135)

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_0135=scRNAseq_Seurat_integrated_cellAnnot_0135@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_cellAnnot_0135_DEG.csv"
          )

```

scVelo analysis is performed in a python script (cellOrigin_ATRT_scRNAseq_integrated_INI254_INI255_INI267_scVelo.py), see you there




# Expression of interested genes in the UMAP of 0,1,3,5 clusters produced by monocle 3

## Extract matrix

```{r heatmap_gene_on_monocle3_UMAP}

# Subset Seurat object
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes <- subset(x = scRNAseq_Seurat_integrated_cellAnnot,
                                                             integrated_snn_res.0.2 %in% c(0,1,3,5)#,
                                                    #features = 
                                                             )
nrow(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes)

# Extract expression matrix for cluster 0135 and all genes
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "matrix"
   )
nrow(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix)

dim(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix)

dim(scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData)

# Save object to RDS file then delete it (too large)
saveRDS(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix,
        file = "scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix.RDS"
        )

# Recover RDS object
scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix <- readRDS("scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix.RDS")

# Remove object
#rm(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix)

# Check order
all(colnames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix) == rownames(scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData))

```

## Heatmap of interesting genes on monocle 3 embedding

```{r notch_interesting_genes}

# Interesting genes 
interestingGenes <- c("HES1","HES4","HES5","HES6","DLL1","ZMIZ1","FBXW7","ROBO1","MAPT","TLE1","STAT3","HDAC1","ASCL1","DLL3","HDAC2","MLLT11","NREP","ENC1","INSM1","SO 11","ELAVL4","DCX","RBFOX2","NRN1","CBLN2","STMN2","STMN4","GAP43","MEG3","NSG1","NEUROD1","NEUROG1","TAGLN3", "ACTL6B","RCOR2","LEF1","REST","SOX1","ID4","ID3","NANOG","SOX2","SOX4","JAG1","GADD45G","DCC","FGF8","FGF17","FGF18","WNT1","GBX2","LMX1B","OTX2","EN1","EN2","IRX1","IRX2","IRX3","PAX5","PAX2","SMARCA4","SMARCB1","SMARCA2","SMARCC1","SMARCC2","ARID1A","ARID1B","ARID2","BRD7","PBRM1","ACTL6A","SMARCD1","SMARCD2","SMARCD3","SMARCE1")

#interestingGenes[duplicated(interestingGenes)]

# Plot heatmap of genes of interest on monocle3 embedding
for (gene in interestingGenes)
{
    # Select genes
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix <- scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix[gene,]
    # Melt matrix
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt <- melt(scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix)
    colnames(scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt) <- gene
    # Merge matrix and metadata
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt_plus <- merge(x = scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData,
                                                                                         y = scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt,
                                                                                         by = 0
                                                                                         )
    # Plot 
    p <- ggplot(data = scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt_plus,
                mapping = aes(x = data_dim_1,
                              y = data_dim_2
                              )
                ) +
        geom_point(mapping = aes(color = eval(as.name(gene)))
                   ) +
        ggtitle(label = gene) +
        scale_color_gradientn(name = "count", colours = c("grey","yellow","red","darkred")
                              ) +
        theme_void() +
        theme(plot.title = element_text(hjust = 0.5, face = "bold"),
              text = element_text(size = 30)
              )
    # Feed plot list
    pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_clust0135_monocle3UMAP_",gene,".pdf"), width = 14, height = 10)
    print(p)
    dev.off()
}

scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix <- scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix["WNT1",]

scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt <- melt(scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix)

```

## Heatmap of interesting genes on monocle 3 embedding (for figure)

```{r notch_interesting_genes}

# Create a list of plots
plotList_heatmap_monocle3_interesingGenes <- list()
for (gene in interestingGenes)
{
    # Select genes
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix <- scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix[gene,]
    # Melt matrix
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt <- melt(scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix)  
    # Merge matrix and metadata
    scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt_plus <- merge(x = scRNAseq_Seurat_integrated_cellAnnot_0135_monocle3_UMAP_metaData,
                                                                                         y = scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt,
                                                                                         by = 0
                                                                                        )
    # Plot 
    p <- ggplot(data = scRNAseq_Seurat_integrated_cellAnnot_0135_interestingGene_matrix_melt_plus,
                mapping = aes(x = data_dim_1,
                              y = data_dim_2
                              )
                ) +
        geom_point(mapping = aes(color = value),
                   size = 0.25
                   ) +
        ggtitle(label = gene) +
        scale_color_gradientn(name = "count", colours = c("grey","yellow","red","darkred")
                              ) +
        theme_void() +
        theme(legend.position = "none",
              plot.title = element_text(hjust = 0.5),
              text = element_text(size = paperFig_fontsize)
              )
    plotList_heatmap_monocle3_interesingGenes[[length(plotList_heatmap_monocle3_interesingGenes)+1]] <- p
}
names(plotList_heatmap_monocle3_interesingGenes) <- interestingGenes

plotList_heatmap_monocle3_interesingGenes[["HES1"]]

```

# Ligand-receptor interaction analysis with cellPhonDB

## cellphonedb analysis on cluster 0,1,3,5

### Export metadata and count matrix

```{r cellphonedb}

head(data.frame(Idents(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes)))

scRNAseq_Seurat_integrated_0135_cellID_cluster <- data.frame(Idents(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes))
head(scRNAseq_Seurat_integrated_0135_cellID_cluster)

scRNAseq_Seurat_integrated_0135_cellID_cluster$cellID <- rownames(scRNAseq_Seurat_integrated_0135_cellID_cluster)
head(scRNAseq_Seurat_integrated_0135_cellID_cluster)

colnames(scRNAseq_Seurat_integrated_0135_cellID_cluster) <- c("cluster","cellID") 
head(scRNAseq_Seurat_integrated_0135_cellID_cluster)

scRNAseq_Seurat_integrated_0135_cellID_cluster <- scRNAseq_Seurat_integrated_0135_cellID_cluster[,c(2,1)]
head(scRNAseq_Seurat_integrated_0135_cellID_cluster)

write.table(x      = scRNAseq_Seurat_integrated_0135_cellID_cluster,
            file   = "scRNAseq_Seurat_integrated_0135_cellID_cluster.txt",
            quote  = FALSE,
            sep    = "\t",
            row.names = FALSE,
            col.names = TRUE
            )


all(colnames(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix) == scRNAseq_Seurat_integrated_0135_cellID_cluster$cellID)

scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df <- data.frame(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_matrix, check.names=FALSE)

# Save object to RDS file then delete it (too large)
saveRDS(object = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df,
        file = "scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df.RDS"
        )

# Recover RDS object
#scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df <- readRDS("scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df.RDS")

# Remove object
#rm(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df)

head(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df[,1:3])

scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df <- rownames_to_column(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df,"Gene")

head(scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df[,1:3])

write.table(x      = scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df,
            file   = "scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df.txt",
            quote  = FALSE,
            sep    = "\t",
            row.names = FALSE,
            col.names = TRUE
            )

```

CellPhoneDB was launched with the following command line:
'cellphonedb method statistical_analysis --output-path ./cellPhoneDB --counts-data hgnc_symbol scRNAseq_Seurat_integrated_0135_cellID_cluster.txt scRNAseq_Seurat_integrated_cellAnnot_0135_allGenes_df.txt'


### Importing results

```{r import_cellphonedb_result}

# Import mean results
scRNAseq_integrated_0135_cellphonedb_mean <- read.table(file         = "cellPhoneDB/means.txt",
                                                        header       = TRUE,
                                                        sep          = "\t",
                                                        check.names  = FALSE
                                                        )
head(scRNAseq_integrated_0135_cellphonedb_mean)
dim(scRNAseq_integrated_0135_cellphonedb_mean)

scRNAseq_integrated_0135_cellphonedb_mean_moins <- dplyr::select(.data = scRNAseq_integrated_0135_cellphonedb_mean,
                                                                 -id_cp_interaction,
                                                                 -partner_a,
                                                                 -partner_b,
                                                                 -gene_a,
                                                                 -gene_b,
                                                                 -secreted,
                                                                 -receptor_a,
                                                                 -receptor_b,
                                                                 -annotation_strategy,
                                                                 -is_integrin
                                                                 )
head(scRNAseq_integrated_0135_cellphonedb_mean_moins)

scRNAseq_integrated_0135_cellphonedb_mean_melt <- melt(scRNAseq_integrated_0135_cellphonedb_mean_moins)
head(scRNAseq_integrated_0135_cellphonedb_mean_melt)

colnames(scRNAseq_integrated_0135_cellphonedb_mean_melt) <- c("interacting_pair","clusters","mean")


# Import pvalue results
scRNAseq_integrated_0135_cellphonedb_pvalues <- read.table(file = "cellPhoneDB/pvalues.txt",
                                                           header = TRUE,
                                                           sep = "\t",
                                                           check.names = FALSE
                                                           )
head(scRNAseq_integrated_0135_cellphonedb_pvalues)
dim(scRNAseq_integrated_0135_cellphonedb_pvalues)

scRNAseq_integrated_0135_cellphonedb_pvalues_moins <- dplyr::select(.data = scRNAseq_integrated_0135_cellphonedb_pvalues,
                                                                 -id_cp_interaction,
                                                                 -partner_a,
                                                                 -partner_b,
                                                                 -gene_a,
                                                                 -gene_b,
                                                                 -secreted,
                                                                 -receptor_a,
                                                                 -receptor_b,
                                                                 -annotation_strategy,
                                                                 -is_integrin
                                                                 )
head(scRNAseq_integrated_0135_cellphonedb_pvalues_moins)

scRNAseq_integrated_0135_cellphonedb_pvalues_melt <- melt(scRNAseq_integrated_0135_cellphonedb_pvalues_moins)
head(scRNAseq_integrated_0135_cellphonedb_pvalues_melt)

colnames(scRNAseq_integrated_0135_cellphonedb_pvalues_melt) <- c("interacting_pair","clusters","pvalues")

head(scRNAseq_integrated_0135_cellphonedb_pvalues_melt)

# Merge pvalue and mean
scRNAseq_integrated_0135_cellphonedb <- merge(x = scRNAseq_integrated_0135_cellphonedb_mean_melt,
                                              y = scRNAseq_integrated_0135_cellphonedb_pvalues_melt,
                                              by = c("interacting_pair","clusters")
                                              )
head(scRNAseq_integrated_0135_cellphonedb)

```


### Filtering results

```{r filtering_and_ploting}

# Threshold
mean_thrs <- 0.075
pvalue_thrs <- 0.05

# Density distribution of means
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_0135_cellphonedb_means.pdf"), width=14, height=12)
ggplot(data = scRNAseq_integrated_0135_cellphonedb
       ) +
    geom_density(mapping = aes(x = mean),
                 fill = "blue"
                 ) +
    geom_vline(xintercept = mean_thrs, lty=2, color='red') +
    theme_bw() +
    theme(text = element_text(size=30)
          )
#dev.off()

# Density distribution of pvalues
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_0135_cellphonedb_pvalues.pdf"), width=14, height=12)
ggplot(data = scRNAseq_integrated_0135_cellphonedb
       ) +
    geom_density(mapping = aes(x = pvalues),
                 fill = "darkgreen"
                 ) +
    geom_vline(xintercept = pvalue_thrs, lty=2, color='red') +
    theme_bw() +
    theme(text = element_text(size=30)
          )
#dev.off()

nrow(scRNAseq_integrated_0135_cellphonedb)

# Filter interaction results
scRNAseq_integrated_0135_cellphonedb_filt <- filter(.data = scRNAseq_integrated_0135_cellphonedb,
                                                    mean > mean_thrs,
                                                    pvalues < pvalue_thrs
                                                    )

head(scRNAseq_integrated_0135_cellphonedb_filt)
nrow(scRNAseq_integrated_0135_cellphonedb_filt)

# Export filtered results
write.table(x      = scRNAseq_integrated_0135_cellphonedb_filt,
            file   = paste0(pathToReport,"scRNAseq_integrated_0135_cellphonedb_filt.txt"),
            quote  = FALSE,
            sep    = "\t",
            row.names = FALSE,
            col.names = TRUE
            )

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToReport,"scRNAseq_integrated_0135_cellphonedb_filt.txt"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

### Plot results

```{r plot_results}

# Volcano plot
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_0135_cellphonedb.pdf"), width=16, height=12)
ggplot(data = scRNAseq_integrated_0135_cellphonedb
       ) +
    geom_point(mapping = aes(x = mean,
                             y = -log10(pvalues)
                             ),
               color = 'grey'
               ) +
    geom_point(data = scRNAseq_integrated_0135_cellphonedb_filt,
               mapping = aes(x = mean,
                             y = -log10(pvalues)
                             ),
               color = "red"
               ) +
        geom_point(data = filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                  grepl("NOTCH", interacting_pair)
                                  ),
               mapping = aes(x = mean,
                             y = -log10(pvalues)
                             ),
               color = "darkgreen",
               size = 3
               ) +
    geom_text_repel(data = filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                  grepl("NOTCH", interacting_pair)
                                  ),
                    mapping = aes(x = mean,
                                  y = -log10(pvalues),
                                  label = interacting_pair
                                  ),
                    size = 5
                    ) +
    facet_wrap(~clusters) +
    theme_bw() +
    theme(text = element_text(size = 30)
          )
#dev.off()

summary(-log10(scRNAseq_integrated_0135_cellphonedb_filt$pvalues))

summary(scRNAseq_integrated_0135_cellphonedb_filt$pvalues)

# Plot dot plot
#pdf(file = paste0(pathToFigures,"dotPlot_scRNAseq_integrated_0135_cellphonedb.pdf"), width=16, height=10)
ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt
       ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = 1-pvalues,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    theme_bw() +
    theme(text = element_text(size = 30),
          axis.text.y = element_text(size = 8),
          )
#dev.off()

head(scRNAseq_integrated_0135_cellphonedb_filt)

# Select result involving NOTCH
scRNAseq_integrated_0135_cellphonedb_filt_NOTCH <- filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                                          grepl("*NOTCH*",interacting_pair )
                                                          )

# Plot dot plot for NOTCH
#pdf(file = paste0(pathToFigures,"dotPlot_scRNAseq_integrated_0135_cellphonedb_NOTCH.pdf"), width=16, height=10)
ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_NOTCH
       ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = 30),
          axis.text.y = element_text(size = 30),
          legend.position = "none"
          )
#dev.off()

# Plot dot plot (for figure)
dotplot_cellphonedb_NOTCH <- ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_NOTCH
                                    ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    #scale_color_gradient(low = "orange",high="darkred") +
    scale_color_gradient(low = "olivedrab2",high="darkgreen") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = paperFig_fontsize),
          axis.text.x = element_text(size = paperFig_fontsize, face = "bold"),
          axis.text.y = element_text(size = paperFig_fontsize),
          legend.position = "none",
          legend.box = "horizontal"
          )
dotplot_cellphonedb_NOTCH

library(cowplot)
legend_dotplot_cellphonedb_NOTCH <- get_legend(ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_NOTCH
                                                      ) +
                                               geom_point(mapping = aes(x = clusters,
                                                                        y = interacting_pair,
                                                                        size = (1-pvalues)/100,
                                                                        color = mean
                                                                        )
                                                          ) +
                                               scale_color_gradient(low = "olivedrab2",
                                                                    high="darkgreen"
                                                                    ) +
                                               theme_bw() +
                                               ylab("") +
                                               theme(text = element_text(size = paperFig_fontsize),
                                                     axis.text.x = element_text(size = paperFig_fontsize, face = "bold"),
                                                     axis.text.y = element_text(size = paperFig_fontsize),
                                                     legend.position = "bottom",
                                                     legend.box = "vertical"
                                                     )
                                               )

# Plot dot plot (for presentation)
pdf(file = paste0(pathToFigures,"dotPlot_scRNAseq_integrated_0135_cellphonedb_NOTCH_pres.pdf"), width=18, height=12)
ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_NOTCH
                                    ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = 1-pvalues,
                             color = mean
                             )
               ) +
    scale_size_continuous(range = c(1,20)) +
    scale_color_gradient(low = "olivedrab2",high="darkgreen") +
#    guide_colourbar(barheigth =  
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = 40),
          axis.text.x = element_text(size = 40, face = "bold"),
          axis.text.y = element_text(size = 40),
          legend.position = "bottom",
          legend.box = "vertical",
          legend.key.width = unit(2,"cm"),
          legend.key.height = unit(1,"cm")
          )
dev.off()

# Select result involving EPHB
scRNAseq_integrated_0135_cellphonedb_filt_EPHB <- filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                                          grepl("^EPHB.+",interacting_pair )
                                                          )

# Plot dot plot for EPHB
#

# Plot dot plot (for figure)
dotplot_cellphonedb_EPHB <- ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_EPHB
                                    ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    #scale_color_gradient(low = "olivedrab2",high="darkgreen") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = paperFig_fontsize),
          axis.text.x = element_text(size = paperFig_fontsize, face = "bold"),
          axis.text.y = element_text(size = paperFig_fontsize),
          legend.position = "right",
          legend.box = "horizontal"
          )
#dotplot_cellphonedb_EPHB

# Select result involving FGFR
scRNAseq_integrated_0135_cellphonedb_filt_FGFR <- filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                                          grepl("^FGFR.+",interacting_pair )
                                                          )

# Plot dot plot for FGFR
#pdf(file = paste0(pathToFigures,"dotPlot_scRNAseq_integrated_0135_cellphonedb_FGFR.pdf"), width=16, height=10)
ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_FGFR
       ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = 30),
          axis.text.y = element_text(size = 30),
          legend.position = "none"
          )
#dev.off()

# Plot dot plot (for figure)
dotplot_cellphonedb_FGFR <- ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_FGFR
                                    ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    #scale_color_gradient(low = "olivedrab2",high="darkgreen") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = paperFig_fontsize),
          axis.text.x = element_text(size = paperFig_fontsize, face = "bold"),
          axis.text.y = element_text(size = paperFig_fontsize),
          legend.position = "right",
          legend.box = "horizontal"
          )
#dotplot_cellphonedb_FGFR

# Select result involving BMPR
scRNAseq_integrated_0135_cellphonedb_filt_BMPR <- filter(.data = scRNAseq_integrated_0135_cellphonedb_filt,
                                                          grepl("^BMPR.+",interacting_pair )
                                                          )

# Plot dot plot for BMPR
#pdf(file = paste0(pathToFigures,"dotPlot_scRNAseq_integrated_0135_cellphonedb_BMPR.pdf"), width=16, height=10)
ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_BMPR
       ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = 30),
          axis.text.y = element_text(size = 30),
          legend.position = "none"
          )
#dev.off()

# Plot dot plot (for figure)
dotplot_cellphonedb_BMPR <- ggplot(data = scRNAseq_integrated_0135_cellphonedb_filt_BMPR
                                    ) +
    geom_point(mapping = aes(x = clusters,
                             y = interacting_pair,
                             size = (1-pvalues)/100,
                             color = mean
                             )
               ) +
    scale_color_gradient(low = "orange",high="darkred") +
    #scale_color_gradient(low = "olivedrab2",high="darkgreen") +
    theme_bw() +
    ylab("") +
    theme(text = element_text(size = paperFig_fontsize),
          axis.text.x = element_text(size = paperFig_fontsize, face = "bold"),
          axis.text.y = element_text(size = paperFig_fontsize),
          legend.position = "right",
          legend.box = "horizontal"
          )
#dotplot_cellphonedb_BMPR

```

### Figure 9 patchwork

```{r figure9_patchwork}

library(magick) # image_read
library(ggpubr) # background_image()

# Import scVelo stream
scVelo_clust0135_DEG_monocle3 <- image_read("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/figures/scvelo_scRNAseq_integrated_embeddingStream_clust0135_DEG_monocle3_dynamical_fig.png")
scVelo_clust0135_DEG_monocle3 <- ggplot()+
    background_image(scVelo_clust0135_DEG_monocle3)

# Import scVelo latent time
scVelo_clust0135_DEG_monocle3_latentTime <- image_read("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/figures/scvelo_scRNAseq_integrated_clust0135_DEG_monocle3_latentTime_fig.png")
scVelo_clust0135_DEG_monocle3_latentTime <- ggplot()+
    background_image(scVelo_clust0135_DEG_monocle3_latentTime)


figure9ab <- wrap_plots(a = trajectory_inference_monocle3_clust0135,
                        b = scVelo_clust0135_DEG_monocle3,
                        nrow = 1,
                        widths = c(0.525,0.475)
                        )

figure9cd <- wrap_plots(pseudotime_monocle3_clust0135,
                        scVelo_clust0135_DEG_monocle3_latentTime,
                        nrow = 1
                        )

figure9e <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["SOX4"]],
                       plotList_heatmap_monocle3_interesingGenes[["SOX11"]],
                       nrow = 1
                       )

figure9f <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["REST"]]
                       )

figure9ef <- wrap_plots(figure9e,
                        figure9f,
                        nrow = 1
                        )

figure9g <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["ACTL6B"]],
                       plotList_heatmap_monocle3_interesingGenes[["STMN4"]],
                       plotList_heatmap_monocle3_interesingGenes[["ELAVL4"]],
                       plotList_heatmap_monocle3_interesingGenes[["DCX"]],
                       plotList_heatmap_monocle3_interesingGenes[["GAP43"]],
                       nrow = 2
                       )

figure9h <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["ID3"]],
                       plotList_heatmap_monocle3_interesingGenes[["ID4"]],
                       plotList_heatmap_monocle3_interesingGenes[["SOX2"]],
                       nrow = 2
                       )

figure9gh <- wrap_plots(a = figure9g,
                        b = figure9h,
                        nrow = 1
                        )


figure9i <- wrap_plots(a = plotList_heatmap_monocle3_interesingGenes[["DLL3"]],
                       b = plotList_heatmap_monocle3_interesingGenes[["HES1"]],
                       c = plotList_heatmap_monocle3_interesingGenes[["DLL1"]],
                       d = plotList_heatmap_monocle3_interesingGenes[["HES4"]],
                       nrow = 2
                       )

figure9ij <- wrap_plots(a = figure9i,
                        b = dotplot_cellphonedb_NOTCH,
                        nrow = 1
                        )

figure9 <- wrap_plots(figure9ab,
                      figure9cd,
                      figure9ef,
                      figure9gh,
                      figure9ij,
                      heights = c(0.175,0.175,0.175,0.225,0.25)
                      )

# Save image as pdf
ggsave(plot    = figure9,
       file    = paste0(pathToFigures,"figure9.pdf"),
       width   = 210, #full a4 210
       height  = 297, # full a4 297
       units   = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figure9.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

### Supplementary figure 

```{r suppFig_patchwork}

suppfig9abc <- wrap_plots(dotplot_cellphonedb_EPHB,
                          dotplot_cellphonedb_FGFR,
                          dotplot_cellphonedb_BMPR,
                          nrow = 3
                          )

suppFig <- wrap_plots(suppfig9abc)

# Save image as pdf
ggsave(plot = suppFig,
       file = paste0(pathToFigures,"suppFig.pdf"),
       width = 210, #full a4 210
       height = 297, # full a4 297
       units = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"suppFig.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```


### Significant mean

```{r significant_mean}

# Import significant mean results
scRNAseq_integrated_0135_cellphonedb_smean <- read.table(file = "cellPhoneDB/significant_means.txt",
                                                        header = TRUE,
                                                        sep = "\t",
                                                        check.names = FALSE
                                                        )
head(scRNAseq_integrated_0135_cellphonedb_smean)
dim(scRNAseq_integrated_0135_cellphonedb_smean)

scRNAseq_integrated_0135_cellphonedb_smean_moins <- dplyr::select(.data = scRNAseq_integrated_0135_cellphonedb_smean
                                                                 ,-id_cp_interaction
                                                                 ,-partner_a
                                                                 ,-partner_b
                                                                 #,gene_a
                                                                 #,gene_b
                                                                 ,-secreted
                                                                 ,-receptor_a
                                                                 ,-receptor_b
                                                                 ,-annotation_strategy
                                                                 ,-is_integrin
                                                                 ,-rank
                                                                 )
head(scRNAseq_integrated_0135_cellphonedb_smean_moins)

scRNAseq_integrated_0135_cellphonedb_smean_melt <- melt(scRNAseq_integrated_0135_cellphonedb_smean_moins,
                                                        id.vars = c("interacting_pair","gene_a","gene_b")
                                                        )

head(scRNAseq_integrated_0135_cellphonedb_smean_melt)

colnames(scRNAseq_integrated_0135_cellphonedb_smean_melt) <- c("interacting_pair","gene_a","gene_b","clusters","sig_mean")
head(scRNAseq_integrated_0135_cellphonedb_smean_melt)

scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA <- scRNAseq_integrated_0135_cellphonedb_smean_melt[!is.na(scRNAseq_integrated_0135_cellphonedb_smean_melt$sig_mean),]

nrow(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA)

scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA <- filter(.data = scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA,
                                                               gene_a != "",
                                                               gene_b != ""
                                                               )

nrow(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA)

head(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA)

scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus <- dplyr::select(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA,
                                                                            gene_a,
                                                                            gene_b
                                                                            )

head(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus)

scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast <- dcast(data           = scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus,
                                                                          formula        = gene_a ~ gene_b,
                                                                          fun.aggregate  = length,
                                                                          drop           = c(TRUE,TRUE)
                                                                          )

head(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast)


rownames(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast) <- scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast$gene_a

scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast$gene_a <- NULL

head(scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast)

# Heatmap
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_0135_cellphonedb.pdf"), width=14, height=12)
Heatmap(matrix                       = scRNAseq_integrated_0135_cellphonedb_smean_melt_noNA_minus_dcast,
        name                         = "number of interactions",
        cluster_columns              = TRUE,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "complete",
        clustering_method_columns    = "complete",
        row_names_side               = "left",
        column_names_side            = "top",
        show_row_dend                = FALSE,
        show_column_dend             = FALSE,
        col = colorRamp2(breaks = c(0,2.5,7.5,10), colors = c("grey88","yellow","red","darkred"))
        )
#dev.off()

```


# CelliD analysis

## Compute MCA

```{r MCA}

scRNAseq_Seurat_integrated_cellAnnot

library(CellID)

# Compute MCA
scRNAseq_Seurat_integrated_cellAnnot <- RunMCA(X = scRNAseq_Seurat_integrated_cellAnnot,
                                               nmcs = 30
                                               )

# Plot MCA
DimPlotMC(X = scRNAseq_Seurat_integrated_cellAnnot,
          group.by = "RNA_snn_res.0.2"
          )

```

## Automatic cell type prediction using pre-established marker lists

### Import MSigDB C8 gene signatures

```{r sessionInfo}

library(msigdbr)

# Extract C8 MSigDB gene set collection
msigdb_hsapiens_c8 <- msigdbr(species = "Homo sapiens",
                              category = "C8"
                              )

head(msigdb_hsapiens_c8)                                   

msigdb_hsapiens_c8_min <- msigdb_hsapiens_c8[,c("gene_symbol","gs_name")]

```

### Overlap with MSigDB C8 FAN_EMBRYONIC gene sets (Fan et al, 2018)

```{r FAN_EMBRYONIC}

# Select FAN_EMBRYONIC gene sets
msigdb_hsapiens_c8_FAN_EMBRYONIC <- filter(.data = msigdb_hsapiens_c8_min,
                                            grepl("FAN_EMBRYONIC_",gs_name)
                                            )

msigdb_hsapiens_c8_FAN_EMBRYONIC

# Convert data frame to list
msigdb_hsapiens_c8_FAN_EMBRYONIC_list <- as.list(unstack(msigdb_hsapiens_c8_FAN_EMBRYONIC))

# Run hypergeometric test
scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC <- RunCellHGT(X          = scRNAseq_Seurat_integrated_cellAnnot,
                                                                          pathways   = msigdb_hsapiens_c8_FAN_EMBRYONIC_list,
                                                                          reduction  = "mca",
                                                                          dim        = 30
                                                                          )

dim(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC)
head(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_FAN_EMBRYONIC <- rownames(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC)[apply(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC, 2, which.max)]

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif <- ifelse(apply(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_FAN_EMBRYONIC, 2, max) > hgt_thrs,
                                                     yes = cellTypes_by_msigdbC8_FAN_EMBRYONIC,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_integrated_cellAnnot) == names(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object    = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata  = cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif,
                                                    col.name  = "msigdb_C8_FAN_EMBRYONIC"
                                                    )

# Plot UMAP with annotated cell type
DimPlot(object     = scRNAseq_Seurat_integrated_cellAnnot,
        reduction  = "umap",
        group.by   = "msigdb_C8_FAN_EMBRYONIC",
        pt.size    = 3,
        cols = c("red","red3","violet","cyan","blue","yellow","black","palegoldenrod","green","violet","violet","darkgreen","blue","mediumblue","palegoldenrod","darkblue","darkblue","snow3"),
        label      = FALSE
        ) #+
    #theme(legend.position = "none")

```

### Overlap with MSigDB C8 DESCARTES_FETAL_CEREBELLUM gene sets (Cao et al, 2020)

```{r DESCARTES_FETAL_CEREBELLUM}

# Select DESCARTES_FETAL_CEREBELLUM gene sets
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM <- filter(.data = msigdb_hsapiens_c8_min,
                                            grepl("DESCARTES_FETAL_CEREBELLUM_",gs_name)
                                            )

msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM

# Convert data frame to list
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list <- as.list(unstack(msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM))

# Run hypergeometric test
scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- RunCellHGT(X          = scRNAseq_Seurat_integrated_cellAnnot,
                                                                          pathways   = msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list,
                                                                          reduction  = "mca",
                                                                          dim        = 30
                                                                          )

dim(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)
head(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- rownames(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)[apply(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, which.max)]

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif <- ifelse(apply(scRNAseq_Seurat_integrated_cellAnnot_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, max) > hgt_thrs,
                                                     yes = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_integrated_cellAnnot) == names(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_integrated_cellAnnot <- AddMetaData(object    = scRNAseq_Seurat_integrated_cellAnnot,
                                                    metadata  = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif,
                                                    col.name  = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM"
                                                    )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellID_msigdb_C8_DESCARTES_FETAL_CEREBELLUM.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_integrated_cellAnnot,
        reduction  = "umap",
        group.by   = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM",
        pt.size    = 2,
        cols = c("red","blue","mediumblue","yellow","black","darkblue","tan1","violet","darkgreen","snow3"),
        label      = FALSE
        )
#dev.off()

```


## Comparing integrated data of CTX,LGE,MGE,CGE (Lee et al., 2022) to integrated R26 tumor cells using Cell-ID

### Compute MCA on integrated dataset

```{r MCA_integrated}

# Import integrated Seurat object
scRNAseq_Seurat_Lee2022_integrated <- readRDS(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/report/scRNAseq_Seurat_Lee2022_integrated.RDS")
                                              )
scRNAseq_Seurat_Lee2022_integrated

# Compute MCA on integrated data
scRNAseq_Seurat_Lee2022_integrated <- RunMCA(X     = scRNAseq_Seurat_Lee2022_integrated,
                                             nmcs  = 30,
                                             slot  = "data"#"scale.data"
                                             )

# Plot MCA of integrated dataset
#pdf(file = paste0(pathToFigures,"mca_cellID_scRNAseq_Lee2022_integrated.pdf"), width = 14, height = 10)
DimPlotMC(X        = scRNAseq_Seurat_Lee2022_integrated,
          group.by = "orig.ident",
          dim      = c(1,2),
          cols        = c("violet","yellow","blue","red","grey"),
          )
#dev.off()

```

### Extract gene signatures from integrated data

```{r MCA_integrated}

nbSignatures <- 200

# Extracting per-cluster gene signatures
Lee2022_integrated_sample_signatures <- GetGroupGeneSet(X         = scRNAseq_Seurat_Lee2022_integrated,
                                                        dims       = 1:30,
                                                        n.features = nbSignatures,
                                                        group.by   = "orig.ident"
                                                        )

str(Lee2022_integrated_sample_signatures)
head(Lee2022_integrated_sample_signatures)

# Extracting per-cell gene signatures
Lee2022_integrated_cell_signatures <- GetCellGeneSet(X           = scRNAseq_Seurat_Lee2022_integrated,
                                                     dims        = 1:30,
                                                     n.features  = nbSignatures
                                                     )
str(Lee2022_integrated_cell_signatures)
head(Lee2022_integrated_cell_signatures)

```

### Convert human gene name to mouse gene name

```{r convert_cluster_signature}

# Convert list to data frame
Lee2022_integrated_sample_signatures_df <- data.frame(Lee2022_integrated_sample_signatures, check.names=FALSE)
head(Lee2022_integrated_sample_signatures_df)

# Melt data frame
Lee2022_integrated_sample_signatures_df_melt <- melt(Lee2022_integrated_sample_signatures_df, id.vars = 0)
head(Lee2022_integrated_sample_signatures_df_melt)
nrow(Lee2022_integrated_sample_signatures_df_melt)

head(symbols_conversion)

# Merge melted data frame with mouse/human gene name conversion table
Lee2022_integrated_sample_signatures_df_melt_humanSymbol <- merge(x = Lee2022_integrated_sample_signatures_df_melt,
                                                                  y = symbols_conversion,
                                                                  by.x = "value",
                                                                  by.y = "MGI.symbol",
                                                                  all = FALSE
                                                                  )

head(Lee2022_integrated_sample_signatures_df_melt_humanSymbol)
nrow(Lee2022_integrated_sample_signatures_df_melt_humanSymbol)

Lee2022_integrated_sample_signatures_df_melt_humanSymbol$value <- NULL

# Convert melted data frame to list (go back to list format)
Lee2022_integrated_sample_signatures_humanSymbol <- aggregate(formula = HGNC.symbol ~ variable,
                                                              data = Lee2022_integrated_sample_signatures_df_melt_humanSymbol,
                                                              FUN = 'c'
                                                              ) %>%
    spread(key = variable,
           value = HGNC.symbol
           ) %>%
    as.list(all.names=TRUE) %>%
    unlist(recursive=FALSE)

```

```{r convert_cluster_signature}

# Convert list to data frame
Lee2022_integrated_cell_signatures_df <- data.frame(Lee2022_integrated_cell_signatures, check.names=FALSE)
head(Lee2022_integrated_cell_signatures_df[,1:4])

# Melt data frame
Lee2022_integrated_cell_signatures_df_melt <- melt(Lee2022_integrated_cell_signatures_df, id.vars = 0)
head(Lee2022_integrated_cell_signatures_df_melt)
nrow(Lee2022_integrated_cell_signatures_df_melt)

# Merge melted data frame with mouse/human gene name conversion table
Lee2022_integrated_cell_signatures_df_melt_humanSymbol <- merge(x     = Lee2022_integrated_cell_signatures_df_melt,
                                                                y     = symbols_conversion,
                                                                by.x  = "value",
                                                                by.y  = "MGI.symbol",
                                                                all   = FALSE
                                                                )

head(Lee2022_integrated_cell_signatures_df_melt_humanSymbol)
nrow(Lee2022_integrated_cell_signatures_df_melt_humanSymbol)

Lee2022_integrated_cell_signatures_df_melt_humanSymbol$value <- NULL

# Convert melted data frame to list (go back to list format)
Lee2022_integrated_cell_signatures_humanSymbol <- aggregate(formula = HGNC.symbol ~ variable,
                                                            data = Lee2022_integrated_cell_signatures_df_melt_humanSymbol,
                                                            FUN = 'c'
                                                            ) %>%
    spread(key = variable,
           value = HGNC.symbol
           ) %>%
    as.list(all.names=TRUE) %>%
    unlist(recursive=FALSE)

```


### Match Lee et al, 2022 samples (CTX,LGE,MGE,CGE) to integrated_R26_tumCells cells

```{r group_to_cell_matching}

scRNAseq_Seurat_integrated_cellID <- scRNAseq_Seurat_integrated_cellAnnot

# Group-to-cell matching
cellID_integratedCells_vs_Lee2022sample <- RunCellHGT(X         = scRNAseq_Seurat_integrated_cellID,
                                                      pathways  = Lee2022_integrated_sample_signatures_humanSymbol,
                                                      dims      = 30
                                                      )

str(cellID_integratedCells_vs_Lee2022sample)
head(cellID_integratedCells_vs_Lee2022sample[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_integratedCells_vs_Lee2022sample)))
enrichment_thrs <- 2

# Density distribution of enrichement score
melt(as.matrix(cellID_integratedCells_vs_Lee2022sample)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integratedCells_vs_Lee2022sample_prediction <- rownames(cellID_integratedCells_vs_Lee2022sample)[apply(cellID_integratedCells_vs_Lee2022sample, 2, which.max)]
head(cellID_integratedCells_vs_Lee2022sample_prediction)
levels(factor(cellID_integratedCells_vs_Lee2022sample_prediction))

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integratedCells_vs_Lee2022sample_prediction_sign <- ifelse(test = apply(cellID_integratedCells_vs_Lee2022sample, 2, max) > enrichment_thrs,
                                                                       yes = cellID_integratedCells_vs_Lee2022sample_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_integratedCells_vs_Lee2022sample_prediction_sign)

# Convert into data frame
cellID_integratedCells_vs_Lee2022sample_prediction_sign_df <- data.frame(Lee2022_sample = cellID_integratedCells_vs_Lee2022sample_prediction_sign)

cellID_integratedCells_vs_Lee2022sample_prediction_sign_df <- rownames_to_column(.data = cellID_integratedCells_vs_Lee2022sample_prediction_sign_df,
                                                                                 var = "integratedCAL_cell"
                                                                                 )

head(cellID_integratedCells_vs_Lee2022sample_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_cellID@meta.data) == cellID_integratedCells_vs_Lee2022sample_prediction_sign_df$integratedCAL_cell)

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_integrated_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_cellID,
                                                              metadata  = cellID_integratedCells_vs_Lee2022sample_prediction_sign_df$Lee2022_sample,
                                                              col.name  = "cellID_Lee2022_sample"
                                                              )

colnames(scRNAseq_Seurat_integrated_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_cellID_Lee2022_sample.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = cellID_Lee2022_sample,
                             fill  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                             )
               ) +
    labs(fill = "integrated\nCAL-SHH\ncluster") +
    xlab("\nLee et al., 2022 E12.5 sample") +
    scale_fill_manual(values = cell_cluster_col[levels(factor(scRNAseq_Seurat_integrated_cellID@meta.data$integrated_snn_res.0.2))],
                      na.value = "grey"
                      ) +
    geom_text(mapping  = aes(x     = cellID_Lee2022_sample,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellID_Lee2022_sample.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_cellID,
        reduction   = "umap",
        pt.size     = 1,
        cols        = c("violet","yellow","blue","red","grey"),
        group.by    = "cellID_Lee2022_sample",
        label       = FALSE,
        label.size  = 15
        ) +
    labs(color = "Lee et al. (2022)\nmatching\nsample") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

### Match Lee et al, 2022 cells (annotated CTX,LGE,MGE,CGE) to integrated_R26_tumCells cells

```{r cell_to_cell_matching}

# Cell-to-cell matching
cellID_integratedCells_vs_Lee2022cells <- RunCellHGT(X         = scRNAseq_Seurat_integrated_cellID,
                                                     pathways  = Lee2022_integrated_cell_signatures_humanSymbol,
                                                     dims      = 30
                                                     )

str(cellID_integratedCells_vs_Lee2022cells)
head(cellID_integratedCells_vs_Lee2022cells[,1:4])

# Enrichment score threshold
#summary(melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCells)))
enrichment_thrs <- 0.2

# Density distribution of enrichement score
melt(as.matrix(cellID_integratedCells_vs_Lee2022cells)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integratedCells_vs_Lee2022cells_prediction <- rownames(cellID_integratedCells_vs_Lee2022cells)[apply(cellID_integratedCells_vs_Lee2022cells, 2, which.max)]
head(cellID_integratedCells_vs_Lee2022cells_prediction)

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integratedCells_vs_Lee2022cells_prediction_sign <- ifelse(test = apply(cellID_integratedCells_vs_Lee2022cells, 2, max) > enrichment_thrs,
                                                                       yes = cellID_integratedCells_vs_Lee2022cells_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_integratedCells_vs_Lee2022cells_prediction_sign)

# Convert into data frame
cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df <- data.frame(Lee2022_cells = cellID_integratedCells_vs_Lee2022cells_prediction_sign)

head(cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df)

cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df <- rownames_to_column(.data = cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df,
                                                                                           var = "integratedCAL_cell"
                                                                                           )

head(cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_cellID@meta.data) == cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df$integratedCAL_cell)

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_integrated_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_cellID,
                                                 metadata  = cellID_integratedCells_vs_Lee2022cells_prediction_prediction_sign_df$Lee2022_cells,
                                                 col.name  = "cellID_Lee2022_cell"
                                                 )

colnames(scRNAseq_Seurat_integrated_cellID@meta.data)

# Extract Lee et al, 2022 integrated cell sample
scRNAseq_Seurat_Lee2022_integrated_cellSample <- data.frame(Lee2022_sample = scRNAseq_Seurat_Lee2022_integrated@meta.data[,"orig.ident"],
                                                            row.names = rownames(scRNAseq_Seurat_Lee2022_integrated@meta.data)
                                                            )

head(scRNAseq_Seurat_Lee2022_integrated_cellSample)

scRNAseq_Seurat_Lee2022_integrated_cellSample <- rownames_to_column(.data = scRNAseq_Seurat_Lee2022_integrated_cellSample,
                                                                     var = "Lee2022_cell"
                                                                     )

head(scRNAseq_Seurat_Lee2022_integrated_cellSample)

# Delete scRNAseq_Seurat_integrated object
rm(scRNAseq_Seurat_Lee2022_integrated)

head(scRNAseq_Seurat_integrated_cellID@meta.data)

# Merge integrated cell cluster to meta data
scRNAseq_Seurat_integrated_cellID@meta.data <- left_join(x      = rownames_to_column(scRNAseq_Seurat_integrated_cellID@meta.data),
                                                         y      = scRNAseq_Seurat_Lee2022_integrated_cellSample,
                                                         by   = c("cellID_Lee2022_cell" = "Lee2022_cell")
                                                         )

# Rename column
scRNAseq_Seurat_integrated_cellID@meta.data <- mutate(scRNAseq_Seurat_integrated_cellID@meta.data,
                                                      cellID_Lee2022_cellSample = Lee2022_sample,
                                                      .keep = "unused"
                                                      )

# Put rownames (cell barcode)
rownames(scRNAseq_Seurat_integrated_cellID@meta.data) <- scRNAseq_Seurat_integrated_cellID@meta.data$rowname
scRNAseq_Seurat_integrated_cellID@meta.data$rowname <- NULL

head(scRNAseq_Seurat_integrated_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_cellID_Lee2022_cellSample.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             fill  = cellID_Lee2022_cellSample
                             )
               ) +
    labs(fill = "Lee et al, 2002\nsample") +
    xlab("\n integrated CAL-SHH cluster") +
    scale_fill_manual(values = c("violet","yellow","blue","red"),
                      na.value = "grey"
                      ) +
    geom_text(mapping  = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_cellID_Lee2022_cellSample.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cellID_Lee2022_cellSample",
        label       = FALSE,
        label.size  = 15,
        cols        = c("violet","yellow","blue","red")
        ) +
    labs(color = "Lee et al, 2022\nsample") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

















## Clustering (Using PCmin=50)

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_integrated_PC50 <- FindNeighbors(object = scRNAseq_Seurat_integrated,
                                                 dims   = 1:50
                                                 )

# Clustering by modularity optimization technique
scRNAseq_Seurat_integrated_PC50 <- FindClusters(object     = scRNAseq_Seurat_integrated_PC50,
                                                resolution = seq(from  = 0,
                                                                 to    = 2,
                                                                 by    = 0.1
                                                                 )
                                                )      

```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_integrated_PC50.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_integrated_PC50@meta.data,
         prefix          = "integrated_snn_res.",
         #node_colour     = "sc3_stability",
         #layout          = "sugiyama",
         use_core_edges  = TRUE
         )
#dev.off()

choosenResolution <- 0.3

# Clustering by modularity optimization technique
#scRNAseq_Seurat_integrated_PC50_choosenResolution <- FindClusters(object     = scRNAseq_Seurat_integrated_PC50,
 #                                                                 resolution = choosenResolution
  #                                                                )      

# Set default ident
Idents(scRNAseq_Seurat_integrated_PC50) <- scRNAseq_Seurat_integrated_PC50@meta.data[["integrated_snn_res.0.3"]]

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_PC50_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_PC50@meta.data
       ) +
    stat_count(mapping = aes(x     = integrated_snn_res.0.3,
                             fill  = orig.ident
                             )
               ) +
    geom_text(mapping  = aes(x     = integrated_snn_res.0.3,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              face     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
scale_fill_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_integrated_PC50@meta.data)

write.table(x      = scRNAseq_Seurat_integrated_PC50@meta.data,
            file   = "scRNAseq_Seurat_integrated_PC50_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_integrated_PC50 <- RunUMAP(object       = scRNAseq_Seurat_integrated_PC50,
                                           dims         = 1:50,
                                           n.neighbors  = 30
                                           )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_sample.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "orig.ident",
        label       = FALSE
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# Save seurat object
saveRDS(object = scRNAseq_Seurat_integrated_PC50,
        file   = paste0(pathToReport,"scRNAseq_Seurat_integrated_PC50.rds"
                        )
        )

```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_nbUMIs.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_PC50,
            features = "nFeature_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_nbGenes.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_PC50,
            features = "nCount_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_nbRNA_per_gene.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_PC50,
            features = "nCount_per_Feature",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_percMito.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_PC50,
            features = "percent.mito",
            reduction = "umap"
            )  +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_clust_cellCyclePhase.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "Phase",
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_PC50_clust_nbUMIs.pdf"), width = 12, height = 12)
 ggplot(data = scRNAseq_Seurat_integrated_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = log10(nCount_RNA),
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = log10(nCount_RNA)
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "log10(UMI count)\n")+
    ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_PC50_clust_nbGenes.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nFeature_RNA,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nFeature_RNA
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "# genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_PC50_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nCount_per_Feature,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nCount_per_Feature
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "UMIs per gene\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_PC50_clust_percMito.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = percent.mito,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = percent.mito
                               ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "% mito genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```


## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.50
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_PC50_choosenResolution.markers <- FindAllMarkers(object            = scRNAseq_Seurat_integrated_PC50,
                                                                            assay             = "RNA",
                                                                            min.pct           = min_pct,
                                                                            logfc.threshold   = min_logfc,
                                                                            min.diff.pct      = min_diffPct,
                                                                            only.pos          = TRUE,
                                                                            min.cells.feature = min_cellsFeature
                                                                            )

head(scRNAseq_Seurat_integrated_PC50_choosenResolution.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_PC50_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_PC50_choosenResolution.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_integrated_PC50_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_integrated_PC50_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_integrated_PC50_choosenResolution,
          features = scRNAseq_Seurat_integrated_PC50_choosenResolution.markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_integrated_PC50_choosenResolution,
          features = nTop_markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=15))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_PC50_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_integrated_PC50_choosenResolution.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_integrated_PC50_choosenResolution.markers,
            file        = paste0(pathToReport,"scRNAseq_integrated_PC50_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

## SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = "SMARCB1",
            pt.size  = 1
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## Expression of ATRT bulk signature genes

#### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

#### Expression of cerebellar marker genes (Wezman et al., 2019)

##### Roof Plate

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 14)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```


##### Rhombic Lip

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### Ventricular Zone

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Bergmann Glia

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### GABAergic

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_PC50,
            features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_PC50_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_PC50,
        features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

## SCENIC Analysis (PC50)

### Cellular enrichment of regulons using AUC score

#### Heatmap of AUC score

```{r SCENIC_AUCscore_heatmap}

# Check order
all(rownames(scRNAseq_Seurat_integrated@meta.data) == colnames(scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons))

# Top annotation
topAnnot_PC50 <- HeatmapAnnotation(df                      = data.frame(Cluster = as.character(scRNAseq_Seurat_integrated_PC50@meta.data[,"integrated_snn_res.0.3"]), check.names = FALSE),
                                   col                     = list(Cluster = cluster_n12_col),
                                   gap                     = unit(2,"mm"),
                                   show_annotation_name    = TRUE,
                                   annotation_name_gp      = gpar(fontsize    = 20),
                                   annotation_legend_param = list(fontsize    = 20,
                                                                  title_gp    = gpar(fontsize = 20, font = 2),
                                                                  labels_gp   = gpar(fontsize = 20),
                                                                  grid_width  = unit(12,"mm"),
                                                                  grid_height = unit(12,"mm"),
                                                                  ncol        = 1
                                                                  ),
                                   simple_anno_size        = unit(2,"cm"),
                                   na_col                  = "white",
                                   show_legend             = TRUE
                                   )

# Bottom annotation
bottomAnnot_PC50 <- HeatmapAnnotation(df                      = data.frame(Patient_ID = as.character(scRNAseq_Seurat_integrated_PC50@meta.data[,"orig.ident"]), check.names = FALSE),
                                    col                     = list(Patient_ID = patient_id_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(1,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,0.2,0.4,0.6), colors = c("cyan","wheat1","red","darkred"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_scenic_auc_matrix_t_filt_mvRegulons,
             clustering_distance_columns = "pearson",
        clustering_method_columns   = "ward",
        clustering_distance_rows    = "euclidean",
        clustering_method_rows      = "ward",
        cluster_rows                = TRUE,
        show_row_names              = TRUE,
        show_column_names           = FALSE,
        column_title                = "cells",
        row_title                   = "Top 50 most variable regulons",
        top_annotation              = topAnnot_PC50,
        bottom_annotation           = bottomAnnot_PC50,
        show_heatmap_legend         = TRUE,
        column_dend_height          = unit(40, "mm"),
        column_split                = 6,
        row_split                   = 14,
        col                         = col_fun,
heatmap_legend_param        = list(color_bar      = "continuous",
                                   grid_width     = unit(0.75,"cm"),
                                   grid_height    = unit(2,"cm"),
                                   legend_width   = unit(4,"cm"),
                                   title          = "AUC\n score\n",
                                   title_gp       = gpar(fontsize = 20),
                                   legend_gp      = gpar(fontsize = 20),
                                   title_position = "topcenter"
                                   )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_SCENIC_AUCscore.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

#### UMAP of cells based on AUC score

```{r AUCscore_umap}

# Bind meta data
scRNAseq_integrated_PC50_scenic_auc_umap_plus <- cbind(scRNAseq_integrated_scenic_auc_umap,
                                              cluster = as.character(scRNAseq_Seurat_integrated_PC50@meta.data$integrated_snn_res.0.3)
                                              )
head(scRNAseq_integrated_PC50_scenic_auc_umap_plus)

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_PC50_SCENIC_AUCscore.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_integrated_PC50_scenic_auc_umap_plus,
       mapping = aes(x = UMAP1, y = UMAP2)
       ) +
    geom_point(mapping = aes(color = cluster
                             ),
               size = 2
               )+
    xlab("UMAP1") +
    ylab("UMAP2") +
    ggtitle(NULL) +
    theme_bw()+
    theme(text               = element_text(size = 20),
          legend.position    = "none"
          #legend.title       = element_text(face = "bold"),
          #legend.key.height  = unit(1.5,"cm")
          )
#dev.off()

```	

### Cellular enrichment of regulons using binarized AUC score

#### Heatmap of AUC_BIN score

```{r SCENIC_AUC_BINscore_heatmap}

# Check order
all(rownames(scRNAseq_Seurat_integrated_PC50@meta.data) == colnames(scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons))

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,1), colors = c("white","black"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons,
             clustering_distance_rows    = "binary",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_row_dend               = FALSE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PC50,
             bottom_annotation           = bottomAnnot_PC50,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated_PC50@meta.data[,"integrated_snn_res.0.3"])),
             column_split                = factor(scRNAseq_Seurat_integrated_PC50@meta.data[,"integrated_snn_res.0.3"]),
             #row_split                   = 2,
             col                         = col_fun,
             show_heatmap_legend         = FALSE
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_SCENIC_AUCbinScore.pdf"), width = 12, height = 12)
draw(h)
#dev.off()

```

#### Heatmap of the percentage of cells with activated regulon

```{r heatmap_perc_active_regulon}

# Merge matrix and meta data
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus <- merge(x     = scRNAseq_Seurat_integrated_PC50@meta.data["integrated_snn_res.0.3"],
                                                                                   y     = scRNAseq_integrated_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt,
                                                                                   by.x  = 0,
                                                                                   by.y  = "cellID",
                                                                                   all   = FALSE
                                                                                   )
head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus)

# Create table of cluster, regulon and number of cells
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells <- group_by(.data = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus,
                                                                                                   integrated_snn_res.0.3,
                                                                                                   regulon
                                                                                                   ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells) = c("cluster","regulon","nbCell")
head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells)

# Create table of cluster, regulon and number of cells in which regulon is activated
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells <- filter(.data = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus, AUCbin == 1) %>%
    group_by(integrated_snn_res.0.3,
             regulon
             ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells) = c("cluster","regulon","nbCell_AUCbin1")
head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells)

# Create a table of total number of cells and cells in which regulon is activated
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells <- merge(x = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells,
                                                                                                   y = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells,
                                                                                                   by = c("cluster","regulon"),
                                                                                                   all = FALSE
                                                                                                   )
head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Compute percentage of cells with active regulon
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$activeRegulon <- scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell_AUCbin1*100/scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell

head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Transform table to matrix (for ComplexHeatmap function)
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix <- spread(data = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells[,c("cluster","regulon","activeRegulon")],
                                                                                                 key = cluster,
                                                                                                 value = activeRegulon
                                                                                                 )
rownames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix) <- scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon
scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon <- NULL
head(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix)



# Top annotation
topAnnot_cluster_PC50 <- HeatmapAnnotation(cluster                  = colnames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
                                           text                     = anno_text(x= colnames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix), rot = 0),
                                           col                      = list(cluster = cluster_n12_col),
                                           gap                      = unit(4,"mm"),
                                           show_annotation_name     = FALSE,
                                           annotation_name_gp       = gpar(fontsize = 20),
                                           annotation_legend_param  = list(fontsize   = 20,
                                                                           title_gp    = gpar(fontsize = 20, font = 2),
                                                                           labels_gp   = gpar(fontsize = 20),
                                                                           grid_width  = unit(12,"mm"),
                                                                           grid_height = unit(12,"mm"),
                                                                           ncol        = 1
                                                                           ),
                                           simple_anno_size        = unit(2,"cm"),
                                           na_col                  = "white",
                                           show_legend             = FALSE
                                           )

# Heatmap
h <- Heatmap(matrix                    = scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix,
             cluster_columns           = FALSE,
             clustering_distance_rows  = "euclidean",
             clustering_method_rows    = "average",
             top_annotation            = topAnnot_cluster_PC50,
             show_row_dend             = FALSE,
             column_title              = "cluster\n",
             row_title                 = "Top 50 most variable regulons",
             column_split              = colnames(scRNAseq_integrated_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
             show_column_names         = FALSE,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(2,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n% of cells with active regulon",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter",
                                                direction = "horizontal"
                                                )
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_SCENIC_AUCbinScore_perc.pdf"), width = 12, height = 12)
draw(h,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Cell type specific regulators

#### Import SCENIC RSS score

```{r import_SCENIC_AUCscore}

# Import SCENIC RSS score
scRNAseq_integrated_PC50_scenic_rss_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_PC50_scenic_rss_matrix.csv"))
head(scRNAseq_integrated_PC50_scenic_rss_matrix)
dim(scRNAseq_integrated_PC50_scenic_rss_matrix)

# Create column of cluster
scRNAseq_integrated_PC50_scenic_rss_matrix$cluster <- rownames(scRNAseq_integrated_PC50_scenic_rss_matrix)
head(scRNAseq_integrated_PC50_scenic_rss_matrix)

# Density distribution
scRNAseq_integrated_PC50_scenic_rss_matrix_melt <- melt(scRNAseq_integrated_PC50_scenic_rss_matrix)
head(scRNAseq_integrated_PC50_scenic_rss_matrix_melt)

ggplot(data = scRNAseq_integrated_PC50_scenic_rss_matrix_melt) +
    geom_density(mapping = aes(x = value),
                   fill = "grey"
             ) +
    xlab("\nRSS score") +
    ylab("density\n") +
    theme_bw()

# Convert variable column (regulon) into factor
scRNAseq_integrated_PC50_scenic_rss_matrix_melt$variable <- factor(scRNAseq_integrated_PC50_scenic_rss_matrix_melt$variable)
head(scRNAseq_integrated_PC50_scenic_rss_matrix_melt)

# Plot RSS per cluster
scRNAseq_integrated_PC50_scenic_rss_plot_list = list()
for (cluster in levels(factor(scRNAseq_integrated_PC50_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_integrated_PC50_scenic_rss_matrix_melt[which(scRNAseq_integrated_PC50_scenic_rss_matrix_melt$cluster == cluster),]
    # Pick top 5 regulons
    top5 <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% top5),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% top5),
                        mapping = aes(label = variable),
                        color = "red",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_integrated_PC50_scenic_rss_plot_list[[length(scRNAseq_integrated_PC50_scenic_rss_plot_list)+1]] <- p
}

#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_PC50_SCENIC_RSS.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_integrated_PC50_scenic_rss_plot_list)
#dev.off()

```

## Deconvolution of scRNAseq using mouse cell clusters from Vladoiu et al, 2019 (PC50)

###  Average expression matrix

```{r average_expression_matrix}

#Extract scaled integrated
matrix_scRNAseq_Seurat_integrated_PC50 <- AverageExpression(object     = scRNAseq_Seurat_integrated_PC50,
                                                       assays     = "RNA"
                                                       )$RNA

head(matrix_scRNAseq_Seurat_integrated_PC50)
nrow(matrix_scRNAseq_Seurat_integrated_PC50)

# Export matrix
write.table(x      = matrix_scRNAseq_Seurat_integrated_PC50,
            file   = paste0(pathToReport,"matrix_scRNAseq_Seurat_integrated_PC50.txt"),
            sep    = "\t",
            quote  = FALSE
            )

```

### Prepare mixture file

CIBERSORT need tab-delimited format (see CIBERSORT tutorial for more details).

```{r mixture_file}

# Create column "Gene" from rownames
matrix_scRNAseq_Seurat_integrated_PC50_mixMat <- add_column(.data    = matrix_scRNAseq_Seurat_integrated_PC50,
                                                       Gene     = rownames(matrix_scRNAseq_Seurat_integrated_PC50),
                                                       .before  = 1
                                                       )
rownames(matrix_scRNAseq_Seurat_integrated_PC50_mixMat) <- NULL
head(matrix_scRNAseq_Seurat_integrated_PC50_mixMat)

# Export mixture file for CIBERSORT
write.table(x          = as.matrix(matrix_scRNAseq_Seurat_integrated_PC50_mixMat),
            file       = paste0(pathToReport,"CIBERSORT/matrix_scRNAseq_Seurat_integrated_PC50_mixMat.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```


### Run CIBERSORT Fraction

For the moment CIBERSORT fraction is run from the website platform.
The result (fraction) is plotted as heatmap below.

```{r geneSubset}

# Import CIBERSORT Fraction results
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50 <- read.table(file         = paste0(pathToReport,"CIBERSORT/CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50.txt"),
                                                              sep          = "\t",
                                                              header       = TRUE,
                                                              check.names  = FALSE
                                                              )
head(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50)

# Put Mixture column (Sample name) as rownames and remove P-value, Correlation and RMSE columns
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min <- CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50
rownames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min)   <- CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min$Mixture
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min$Mixture     <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min$`P-value`   <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min$Correlation <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min$RMSE        <- NULL
head(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min)

# Density
melt(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min) %>%
    ggplot(mapping = aes(x = value)
           ) +
    geom_density() +
    geom_density(mapping = aes(x = -log10(value)),
                 color = "blue"
                 )

rownames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min)

# Check column order
vladoiu_cellTypes_cibersort$cluster_name == colnames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min)

# Define heatmap color
col_fun_mouseVladoiu2019 = colorRamp2(breaks = c(0,0.1,0.6), colors = c("lightskyblue","navajowhite","red"))

# Plot heatmap
h <- Heatmap(matrix                       = CIBERSORTx_Job33_scRNAseq_Seurat_integrated_PC50_min,
             cluster_rows                 = TRUE,
             cluster_columns              = TRUE,
             show_row_names               = TRUE,
             show_column_names            = TRUE,
             show_row_dend                = TRUE,
             show_column_dend             = FALSE,
             clustering_distance_rows     = "euclidean",
             clustering_method_rows       = "ward",
             clustering_distance_columns  = "euclidean",
             clustering_method_columns    = "ward",
             #row_order                   = order(sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup),
             #row_split                    = sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup,
             column_split                 = vladoiu_cellTypes_cibersort$cluster_lineage,
             cluster_row_slices           = TRUE,
             cluster_column_slices        = FALSE,
             column_names_gp              = gpar(fontsize = 20),
             column_names_max_height      = unit(15,"cm"),
             row_names_gp                 = gpar(fontsize = 20),
             show_heatmap_legend          = FALSE,
             row_split                    = 3,
             column_title_gp              = gpar(fontsize = 20, fontface="bold"),
             row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
             col                          = col_fun_mouseVladoiu2019,
             row_title                    = c("A","B","C")
             )
#pdf(file = paste0(pathToFigures,"heatmap_CIBERSORT_fraction_scRNAseq_integrated_PC50.pdf"), width = 20, height =10) 
draw(h)
#dev.off()

# Legend of relative abundance
legend_fraction <- Legend(col_fun         = col_fun_mouseVladoiu2019,
                          title           = "relative abundance\n",
                          grid_height     = unit(1,"cm"),
                          legend_width    = unit(10,"cm"),
                          title_gp        = gpar(fontsize = 20, fontface = "bold"),
                          title_position  = "topcenter",
                          direction       = "horizontal",
                          labels_gp       = gpar(fontsize=15)
                          )

# Combine legends
legends <- packLegend(list        = list(legend_fraction),
                      direction   ="horizontal",
                      column_gap  = unit(2,"cm")
                      )

#pdf(file = paste0(pathToFigures,"legend_heatmap_CIBERSORT_fraction_scRNAseq_integrated_PC50.pdf"), width = 20, height =10) 
draw(legends)
#dev.off()

```

# Session info

```{r sessionInfo}

# Session info
sessionInfo()

```
