---
title: "Human scRNAseq analysis of InnoRT003 CAL ATRT"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
    fig_caption: yes
    fig_height: 8
    fig_width: 9
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
  word_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r paths}

thisProject                  <- "cellOrigin_ATRT/"

# Paths
pathToRhabdoid_U830_projects <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/"
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.6/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.5/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToThisProject            <- paste0(pathToANALYSIS_PROJECTS,thisProject)
pathToData                   <- paste0(pathToThisProject,"data/")
pathToFigures                <- paste0(pathToThisProject,"svn/analyse/script/figures/")
pathToReport                 <- paste0(pathToThisProject,"svn/analyse/report/")
pathToRNApip_RESULT          <- paste0(pathToThisProject,"RNApip_RESULT/")
pathToFeatureCounts          <- paste0(pathToThisProject,"FeatureCounts_RESULT/")

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r manage_renv, eval=FALSE}

# Initialize Renv
#renv::init()

```


```{r load_library, eval=FALSE}

# Load RNAseq library
source("load_scRNAseq_library.R")

```

```{r loading_RData, eval=FALSE}

# Load saved RData
load("cellOrigin_ATRT_scRNAseq_InnovRT003_noLowUMIclusters.RData")

#save.image(file="cellOrigin_ATRT_scRNAseq_InnovRT003_noLowUMIclusters.RData")

```

```{r load_plot_function}

# Load external files
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")
source("scRNAseq_functions.R")

```

```{r delete_all_objects}

# Delete all objects
#rm(list = ls())

```

## Import InnovRT003 data precessed in InnovRT003.Rmd script

```{r load_data}

scRNAseq_Seurat_InnovRT003 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_InnovRT003.rds")
                                      )

head(scRNAseq_Seurat_InnovRT003@meta.data$RNA_snn_res.0.8)

```


## Cell filtering

```{r cell_filtering}

# Remove cluster 3 and 4 where cells show low UMI counts
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- subset(scRNAseq_Seurat_InnovRT003,
                                                      RNA_snn_res.0.8 %in% c(1,2,5,6,7,8,9,10,11)
                                                      )

str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters)

```

# Data normalization

```{r data_normalization}

# Normalization
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- NormalizeData(object                = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            normalization.method  = "LogNormalize",
                                            scale.factor          = 10000
                                            )

```

# Assign cell-cycle scores

```{r assign_cellCycle_scores}

# Cell cycle gene markers (present in Seurat)
str(cc.genes)
length(cc.genes$s.genes)
length(cc.genes$g2m.genes)

cc.genes$g2m.genes
cc.genes$s.genes

# Compute cell cycle scoring  
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- CellCycleScoring(object       = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                s.features   = cc.genes$s.genes,
                                                                g2m.features = cc.genes$g2m.genes
                                                                )

# Check meta data
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data)

#pdf(file = paste0(pathToFigures,"plots_cellCyclPhase_scRNAseq_InnovRT003_noLowUMIclusters.pdf"), width = 16, height = 12)
plot_cellCyclePhase(scRNAseq_Seurat_InnovRT003_noLowUMIclusters)
#dev.off()

```


# Dimensionality reduction (PCA)

```{r feature_selection}

# Highly variable genes
nFeatures <- 2000
variableFeatures_method <- "vst"
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- FindVariableFeatures(object            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                   selection.method  = variableFeatures_method,
                                                   nfeatures         = nFeatures
                                                   )

# Mean-variance scatter plot
#pdf(file = paste0(pathToFigures,"variableFeaturePlot_scRNAseq_InnovRT003_noLowUMIclusters.pdf"), width = 12, height = 12)
VariableFeaturePlot(scRNAseq_Seurat_InnovRT003_noLowUMIclusters) +
    xlab("\nAverage Expression") +
    ylab("Standard Deviation\n") +
        theme_bw() +
        theme(text = element_text(size = 30),
              axis.title = element_text(size = 30, face = "bold"),
              legend.position = c(0.9,0.8)
              )
#dev.off()

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                         features = rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters)
                                                         )

# PCA
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- RunPCA(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters)

# PCs elbow plot
PCmin <- 12
#pdf(file = paste0(pathToFigures,"elbowPlot_scRNAseq_InnovRT003_noLowUMIclusters_PCA.pdf"), width = 12, height = 12)
ElbowPlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
          ndims = 50
          ) +
    geom_vline(xintercept = PCmin, lty = 2, color = "blue") +
    xlab("\nPC") +
    ylab("Standard Deviation\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Heatmap of each PC
#pdf(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_InnovRT003_noLowUMIclusters_PCA.pdf"), width = 15, height = 11)
DimHeatmap(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
           dims = 1:15,
           ncol = 5
           )
#dev.off()

# Plot PCA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA.pdf"), width = 12, height = 12)
DimPlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        pt.size = 2,
        reduction = "pca",
        cols = rep("black",length(levels(Idents(scRNAseq_Seurat_InnovRT003_noLowUMIclusters))))
        ) +
    xlab("\nPC_1") +
    ylab("PC_2\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

```

## Plot PCA and covariates

```{r PCA_and_covariates}

# Extract scaled matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                                    slot = "scale.data"
                                                                                    )
                                                  )
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix[,1:3])

# Get HVF
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes <- HVFInfo(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                               selection.method = "vst"
                                                               )
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes)
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes)

# Get most variable genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes[order(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes$variance.standardized, decreasing = TRUE),] %>% .[1:nFeatures,] -> mvGenes

# Subset matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix[as.vector(rownames(mvGenes)),]
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)

# PCA analysis
PC        <- prcomp(data.frame(t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)

pci       <- data.frame(PC$x,scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data)

# Plot eigen value
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_PCA_eigValPer.pdf"), width = 12, height = 12)
ggplot() +
    geom_bar(data = data.frame(PC = 1:50, eigValPer = eigValPer[1:50]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity"
             ) +
    geom_bar(data = data.frame(PC = 1:PCmin, eigValPer = eigValPer[1:PCmin]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity",
             fill = "blue"
             ) +
    xlab(label = "\nPC") +
    ylab(label = "% variability (eigen value)\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 30),
          axis.title = element_text(size = 30, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Plot PCA and nb of UMIs
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_nbUMIs.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "log10_nCount_RNA", legendTitle = "log10(nbUMIs)")
#dev.off()

# Plot PCA and nb of genes
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_nbGenes.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nFeature_RNA", legendTitle = "nbGenes")
#dev.off()

# Plot PCA and nb RNA per gene
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_nbRNA_per_gene.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "nCount_per_Feature", legendTitle = "UMIs/gene")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_percMito.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.mito", legendTitle = "% mito")
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_percRiboProt.pdf"), width = 12, height = 10)
plot_PCA_and_covariate(pci = pci, covariate = "percent.riboProt", legendTitle = "% riboProt")
#dev.off()

# Plot PCA and cell cycle phase
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_InnovRT003_noLowUMIclusters_PCA_cellCyclePhase.pdf"), width = 12, height = 12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = Phase),
               size    = 1#,
               #pch     = 21
               ) +
    #scale_fill_continuous(type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```


# Cell clustering

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- FindNeighbors(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                        dims = 1:PCmin
                                        )

clustering_resolutions <- seq(from = 0, to = 2, by = 0.1)

# Clustering by modularity optimization technique
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- FindClusters(object     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                           resolution = clustering_resolutions
                                           )      

# Save seurat object
saveRDS(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        file   = paste0(pathToReport,"scRNAseq_Seurat_InnovRT003_noLowUMIclusters.rds"
                        )
        )
    
```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_InnovRT003_noLowUMIclusters.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data,
         prefix          = "RNA_snn_res.",
         #node_colour     = "sc3_stability",
         #layout          = "sugiyama",
         use_core_edges  = TRUE,
         color           = "red"
         )
#dev.off()

choosenResolution <- "RNA_snn_res.0.6"

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_InnovRT003_noLowUMIclusters_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(choosenResolution)),
                             fill  = Phase
                             )
               ) +
    geom_text(mapping  = aes(x     = eval(as.name(choosenResolution)),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

```

## UMAP visualization

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_InnovRT003_noLowUMIclusters <- RunUMAP(object       = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                       dims         = 1:PCmin,
                                                       n.neighbors  = 30
                                                       )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = choosenResolution,
        label       = TRUE,
        label.size  = 15
        ) +
    ggtitle("IRT003") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

paperFig_fontsize <- 6

# Plot umap for figure
umap_scRNAseq_clust <- DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                               reduction   = "umap",
                               pt.size     = 0.25,
                               group.by    = choosenResolution,
                               label       = TRUE,
                               label.size  = paperFig_fontsize/3,
                               #cols        = cell_cluster_col[levels(factor(scRNAseq_Seurat_InnovRT003@meta.data$RNA_snn_res.0.8))],
                               label.color = "white",
                               label.box = TRUE
                                        #label.color = cluster_number_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))]
                                         ) +
    xlab("") +
    ylab("") +
    theme(legend.position   = "none",
          text              = element_text(size = paperFig_fontsize),
          axis.text         = element_text(size = paperFig_fontsize),
          axis.ticks        = element_line(size = 0.25),
          axis.line         = element_line(size = 0.25),
          panel.background  = element_rect(fill = "white")
          )
umap_scRNAseq_clust

```

# figureII answer To Reviewers LobonIglesias2022

```{r fig}

# Patchwork
figureII <- wrap_plots(umap_scRNAseq_clust,
                       umap_scRNAseq_clust,
                       widths = c(0.5,0.5)
                         ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Figure II\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )


# Save figure into pdf file
ggsave(plot   = figureII,
       file   = paste0(pathToFigures,"figureII_answerToReviewers_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297/3, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figureII_answerToReviewers_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbUMIs.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "nFeature_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = "log10(nbUMIs)") +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbGenes.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "log10_nCount_RNA",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "nCount_per_Feature",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_percMito.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "percent.mito",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of ribosomal proteine
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_percRiboprot.pdf"), width = 12, height = 12)
FeaturePlot(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "percent.riboProt",
            reduction = "umap",
            pt.size = 3
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_cellCyclePhase.pdf"), width = 12, height = 12)
DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        reduction   = "umap",
        pt.size     = 2,
        group.by    = "Phase",
        label       = FALSE,
        label.size  = 15
        ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

```

## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbUMIs.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, clustering = choosenResolution, covariate = "log10_nCount_RNA", ylab = "log10(UMI count)\n")
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbGenes.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, clustering = choosenResolution, covariate = "nFeature_RNA", ylab = "nbGenes")
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, clustering = choosenResolution, covariate = "nCount_per_Feature", ylab = "UMIs/gene")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_percMito.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, clustering = choosenResolution, covariate = "percent.mito", ylab = "% mito")
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_percRiboProt.pdf"), width = 12, height = 12)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, clustering = choosenResolution, covariate = "percent.riboProt", ylab = "% riboProt")
#dev.off()

```


# Cluster marker genes (Differential analysis)

## One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.25
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Set Idents to choosen resolution
Idents(scRNAseq_Seurat_InnovRT003_noLowUMIclusters) <- choosenResolution

levels(Idents(scRNAseq_Seurat_InnovRT003_noLowUMIclusters))

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers <- FindAllMarkers(object            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                     min.pct           = min_pct,
                                                     logfc.threshold   = min_logfc,
                                                     min.diff.pct      = min_diffPct,
                                                     only.pos          = TRUE,
                                                     min.cells.feature = min_cellsFeature
                                                     )

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10#,
              #fontface = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
          features = scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
          features = nTop_markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=20))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_InnovRT003_noLowUMIclusters_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
            file        = paste0(pathToReport,"scRNAseq_InnovRT003_noLowUMIclusters_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

### Heatmap using integrated INI254, INI255 and INI264 gene markers

```{r diffAnalysis}

# Import integrated gene markers
scRNAseq_integrated_cluster_geneMarkers <- read.table(file        = paste0(pathToReport,"scRNAseq_integrated_cluster_geneMarkers.txt"),
                                                      header = TRUE,
                                                      sep         = "\t"
                                                      )
head(scRNAseq_integrated_cluster_geneMarkers)

# Add cluster cell type
scRNAseq_integrated_cluster_geneMarkers <- mutate(.data = scRNAseq_integrated_cluster_geneMarkers,
                                                  cell_type = case_when(cluster %in% c(0,1) ~ "UD cells",
                                                                        cluster %in% c(3,5) ~ "NPL cells",
                                                                        cluster %in% c(2,4) ~ "Cycling Cells",
                                                                        cluster %in% c(6,7,8) ~ "Non-Neuronal",
                                                                        TRUE ~ ""
                                                                        ),
                                                  cell_sub_type = case_when(cluster == 0 ~ "Undifferentiated cells 0",
                                                                            cluster == 1 ~ "Undifferentiated cells 1",
                                                                            cluster == 2 ~ "Cycling Cells (G1S)",
                                                                            cluster == 3 ~ "Neuronal progenitor like 1 (NPL1)",
                                                                            cluster == 4 ~ "Cycling Cells (G2M)",
                                                                            cluster == 5 ~ "Neuronal progenitor like 2 (NPL2)",
                                                                            cluster == 6 ~ "Non-Neuronal cells 1",
                                                                            cluster == 7 ~ "Non-Neuronal cells 2",
                                                                            cluster == 8 ~ "Non-Neuronal cells 3",
                                                                            cluster == 9 ~ "Hypoxia and Inflammation Reactive cells (HIR)",
                                                                            TRUE ~ ""
                                                                            )
                                                  )

# Select top markers for each cluster
nTop <- 5
nTop_integrated_markers <- scRNAseq_integrated_cluster_geneMarkers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_integratedTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   =  scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
          features = nTop_integrated_markers$gene,
          assay    = "RNA"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=20))
#dev.off()

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                                        assay = "RNA",#"integrated",
                                                                                        slot = "data"#"scale.data"
                                                                                        )
                                                         )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix[,1:4])

# Subset matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix_integratedTopMarkers <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix[nTop_integrated_markers$gene,]

# Order cluster (for heatmap visualization)
metadata <- data.frame(cell_barcode = rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data),
                       IRT003_cluster = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data[,choosenResolution]
                       )

head(metadata)

target <- c("1","0","5","4","10","2","8","9","7","3","6")

metadata %>% arrange(factor(IRT003_cluster, levels = target)) -> metadata_ordered
head(metadata_ordered)

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix_integratedTopMarkers[,1:4])

# Subset matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix_integratedTopMarkers_ordered <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix_integratedTopMarkers[,metadata_ordered$cell_barcode]

# Create top annotation
topAnnot_InnovRT003_cluster <- HeatmapAnnotation(df = metadata_ordered["IRT003_cluster"],
                                                 col = list(IRT003_cluster = cluster_n11_col),
                                                 show_legend = TRUE
                                                 )

cellSubtype_integratedCluster_col <- c("Undifferentiated cells 0" = "skyblue",
                                       "Undifferentiated cells 1" = "skyblue4",
                                       "Cycling Cells (G1S)" = "violet",
                                       "Neuronal progenitor like 1 (NPL1)" = "springgreen4",
                                       "Cycling Cells (G2M)" = "violetred",
                                       "Neuronal progenitor like 2 (NPL2)" = "springgreen",
                                       "Non-Neuronal cells 1" = "sienna1",
                                       "Non-Neuronal cells 2" = "sienna3",
                                       "Non-Neuronal cells 3" = "sienna4",
                                       "Hypoxia and Inflammation Reactive cells (HIR)" = "blue4"
                                       )

# Create row annotation
rowAnnot_integratedMarkers <- HeatmapAnnotation(df = data.frame(integrated_cluster = factor(nTop_integrated_markers$cell_sub_type)),
                                                which = "row",
                                                col = list(integrated_cluster = cellSubtype_integratedCluster_col)
                                                )

# Complex Heatmap
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_integratedTopMarkers.pdf"), width = 16, height = 12)
Heatmap(matrix             = t(scale(t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_normCounts_matrix_integratedTopMarkers_ordered))),
        name               = "scaled norm. count",
        top_annotation     = topAnnot_InnovRT003_cluster,
        left_annotation    = rowAnnot_integratedMarkers,
        cluster_columns    = FALSE,
        show_column_names  = FALSE,
        show_row_names     = TRUE,
        cluster_rows       = FALSE,
        #column_split       = metadata_ordered$IRT003_cluster,
        row_split          = nTop_integrated_markers$cell_type,
        row_names_gp       = gpar(fontsize = 15)
        )
#dev.off()

```


# Expression of genes/gene sets of interest

## SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = "SMARCB1",
            pt.size  = 1,
            cols = c("gray","yellow","brown1","brown4")
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

#### ATRT MYC signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select MYC gene signatures
geneSet_ATRT_Torchia2016_MYC <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "MYC"
                                       )
geneSet_ATRT_Torchia2016_MYC$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = geneSet_ATRT_Torchia2016_MYC$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_MYC_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = geneSet_ATRT_Torchia2016_MYC$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

#### ATRT SHH specific TFs (Johann et al, 2016: Table S6)

```{r SHH_markers_expression_Torchia2016}

# Select SHH TFs
geneSet_ATRT_Johann2016_SHH_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "SHH"
                                          )
geneSet_ATRT_Johann2016_SHH_TFs$Gene

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 2,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

#### ATRT MYC specific TFs (Johann et al, 2016: Table S6)

```{r MYC_markers_expression_Torchia2016}

# Select MYC TFs
geneSet_ATRT_Johann2016_MYC_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "MYC"
                                          )
geneSet_ATRT_Johann2016_MYC_TFs$Gene

# ATRT MYC signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 1,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Johann2016_MYC_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = geneSet_ATRT_Johann2016_MYC_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 1
        )
#dev.off()

```

#### ATRT SHH markers found heterogeneously expressed in integrated INI254, INI255 and INI267 data 

```{r heterATRTSHHmarkers}

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 24, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5","INSM1"),
            pt.size  = 0.5,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

# ATRT MYC signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 24, height = 12)
VlnPlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
        features = c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5","INSM1"),
        slot     = "data",
        pt.size  = 0,
        ncol     = 4
        )
#dev.off()

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_min.pdf"), width = 26, height = 08)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = c("STMN4","DLL3","TTYH1"),
            pt.size  = 1,
            slot     = "data",
            ncol     = 3,
            cols     = c("gray","yellow","brown1","brown4")
            )
#dev.off()

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp_min_vertical.pdf"), width = 08, height = 16)
FeaturePlot(object   = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
            features = c("STMN4","DLL3","TTYH1"),
            pt.size  = 2,
            slot     = "data",
            ncol     = 1,
            cols     = c("gray","yellow","brown1","brown4")
            ) +
    theme(text       = element_text(size = 15),
          plot.title = element_text(size = 20, face = "bold")
          )
#dev.off()

```

# Cerebro analysis (for visualization)

## Copy seurat object

```{r copy_seurat_object}

stopifnot(require(cerebroApp))

scRNAseq_Seurat_InnovRT003_noLowUMIclusters
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro@meta.data)

```

## Clusters relationship trees

```{r clusters_relationship}

# Create vectors of clustering resolutions
groups <- paste0('RNA_snn_res.',seq(0,2,0.1))
groups

# Build phylogenetic tree of clusters
for (res in groups[-1])
{
    print(res)
    Idents(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro) <- res
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- BuildClusterTree(object           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                                                dims             = 1:PCmin,
                                                reorder          = FALSE,
                                                reorder.numeric  = FALSE
                                                )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro@misc$trees[[res]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro@tools$BuildClusterTree
}

```

## Compute percentage of mitochondrial and ribosomal transcripts

```{r cerebro_mito_ribo}

# % of mito and ribo 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- addPercentMtRibo(object             = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                                                                        organism           = 'hg',
                                                                        gene_nomenclature  = 'name'
                                                                        )

```

## Get most expressed genes

```{r cerebro_most_expressed_genes}

# Most expressed genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- getMostExpressedGenes(object  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                                                                             assay   = 'RNA',
                                                                             groups  = groups
                                                                             )

```

## Get marker genes

```{r cerebro_marker_genes}

# Mark genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- getMarkerGenes(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                                                                      assay     = 'RNA',
                                                                      organism  = 'hg',
                                                                      groups    = groups,
                                                                      name      = 'scRNAseq_InnovRT003_noLowUMIclusters',
                                                                      only_pos  = TRUE
                                                                      )

```

## Perform pathway enrichment on marker genes

```{r cerebro_pathway_enrichment_on_markers}

# Pathway analysis
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- getEnrichedPathways(object              = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                                                                           marker_genes_input  = 'scRNAseq_InnovRT003_noLowUMIclusters',
                                                                           adj_p_cutoff        = 0.01,
                                                                           max_terms           = 100
                                                                           )

```
 
## Gene set enrichment

```{r cerebro_gene_set_enrichment}

# Pathway analysis
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro <- performGeneSetEnrichmentAnalysis(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
 #                                                           assay     = "RNA",
  #                                                          GMT_file  = "xx/xx/msigdb.v5.2.symbols_mouse.gmt",
   #                                                         groups    = groups
    #                                                        )

```


## Export to cerebro format

```{r export_to_cerebro_format}

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro@meta.data)

# Export to cerebro format
exportFromSeurat(object             = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cerebro,
                 assay              = 'RNA',
                 slot               = 'data',
                 file               = paste0(pathToReport,'scRNAseq_Seurat_InnovRT003_noLowUMIclusters.crb'),
                 experiment_name    = 'scRNAseq_InnovRT003_noLowUMIclusters',
                 organism           = 'mm',
                 groups             = groups,
                 nUMI               = 'nCount_RNA',
                 nGene              = 'nFeature_RNA',
                 add_all_meta_data  = TRUE,
                 verbose            = FALSE
                 )

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToReport,'scRNAseq_Seurat_InnovRT003_noLowUMIclusters.crb'),
              zerkaloDir = 'mandrian_cellOrigin_ATRT'
              )

```


# CelliD analysis

## Compute MCA on InnovRT003_noLowUMIclusters data

```{r MCA}

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters

# Compute MCA
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- RunMCA(X     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                            nmcs  = 30,
                                            slot  = "data"#"scale.data"
                                            )

# Plot MCA
DimPlotMC(X        = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
          group.by = choosenResolution
          )

```

## Automatic cell type prediction using pre-established marker lists

### Import MSigDB C8 gene signatures

```{r sessionInfo}

# Extract C8 MSigDB gene set collection
msigdb_hsapiens_c8 <- msigdbr(species = "Homo sapiens",
                              category = "C8"
                              )
head(msigdb_hsapiens_c8)                                   

msigdb_hsapiens_c8_min <- msigdb_hsapiens_c8[,c("gene_symbol","gs_name")]

```

### Overlap with MSigDB C8 FAN_EMBRYONIC gene sets (Fan et al, 2018)

```{r FAN_EMBRYONIC}

# Select FAN_EMBRYONIC gene sets
msigdb_hsapiens_c8_FAN_EMBRYONIC <- filter(.data = msigdb_hsapiens_c8_min,
                                            grepl("FAN_EMBRYONIC_",gs_name)
                                            )
msigdb_hsapiens_c8_FAN_EMBRYONIC

# Convert data frame to list
msigdb_hsapiens_c8_FAN_EMBRYONIC_list <- as.list(unstack(msigdb_hsapiens_c8_FAN_EMBRYONIC))

# Run hypergeometric test
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC <- RunCellHGT(X          = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                                                 pathways   = msigdb_hsapiens_c8_FAN_EMBRYONIC_list,
                                                                                 reduction  = "mca",
                                                                                 dim        = 30
                                                                                 )

dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_FAN_EMBRYONIC <- rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC)[apply(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC, 2, which.max)]

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif <- ifelse(apply(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_FAN_EMBRYONIC, 2, max) > hgt_thrs,
                                                     yes = cellTypes_by_msigdbC8_FAN_EMBRYONIC,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID) == names(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- AddMetaData(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                 metadata  = cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif,
                                                 col.name  = "msigdb_C8_FAN_EMBRYONIC"
                                                 )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_cellID_msigdb_C8_FAN_EMBRYONIC.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_FAN_EMBRYONIC",
        pt.size    = 3,
        cols = c("red","red3","violet","cyan","brown","darkblue","black","palegoldenrod","green","violet","violet","blue","yellow","mediumblue","orange","snow3"),
        label      = FALSE
        ) #+
    #theme(legend.position = "none")
#dev.off()

```

### Overlap with MSigDB C8 DESCARTES_FETAL_CEREBELLUM gene sets (Cao et al, 2020)

```{r DESCARTES_FETAL_CEREBELLUM}

# Select DESCARTES_FETAL_CEREBELLUM gene sets
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM <- filter(.data = msigdb_hsapiens_c8_min,
                                                        grepl("DESCARTES_FETAL_CEREBELLUM_",gs_name)
                                                        )
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM

# Convert data frame to list
msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list <- as.list(unstack(msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM))

# Run hypergeometric test
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- RunCellHGT(X          = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                                             pathways   = msigdb_hsapiens_c8_DESCARTES_FETAL_CEREBELLUM_list,
                                                                             reduction  = "mca",
                                                                             dims        = 30
                                                                             )

dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM[,1:4])

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM)[apply(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, which.max)]

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM)

# Keep cell annotation only if the score are higher than a given threshold
hgt_thrs <- 2
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif <- ifelse(apply(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, max) > hgt_thrs,
                                                                  yes = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM,
                                                                  no = "unassigned"
                                                                  )
head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID) == names(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- AddMetaData(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                                  metadata  = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif,
                                                                  col.name  = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM"
                                                                  )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_cellID_msigdb_C8_DESCARTES_FETAL_CEREBELLUM.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM",
        pt.size    = 2,
        cols = c("red","blue","mediumblue","tan1","black","darkblue","green","grey"),
        label      = FALSE
        )
#dev.off()

```

## Comparing integrated data (INI254,INI255,INI267) to InnovRT003_noLowUMIclusters using Cell-ID

### Compute MCA on integrated dataset

```{r MCA_integrated}

# Import integrated Seurat object
scRNAseq_Seurat_integrated <- readRDS(file = paste0(pathToReport,"scRNAseq_Seurat_integrated.rds")
                                      )

scRNAseq_Seurat_integrated

# Compute MCA
scRNAseq_Seurat_integrated <- RunMCA(X     = scRNAseq_Seurat_integrated,
                                     nmcs  = 30,
                                     slot  = "data"#"scale.data"
                                     )

# Plot MCA
DimPlotMC(X        = scRNAseq_Seurat_integrated,
          group.by = "integrated_snn_res.0.2"
          )

```

### Extract gene signatures from integrated data

```{r MCA_integrated}

nbSignatures <- 200

# Extracting per-cluster gene signatures
integrated_cluster_signatures <- GetGroupGeneSet(X         = scRNAseq_Seurat_integrated,
                                                dims       = 1:30,
                                                n.features = nbSignatures,
                                                group.by   = "integrated_snn_res.0.2"
                                                )
str(integrated_cluster_signatures)
head(integrated_cluster_signatures)

# Extracting per-cell gene signatures
integrated_cell_signatures <- GetCellGeneSet(X           = scRNAseq_Seurat_integrated,
                                             dims        = 1:30,
                                             n.features  = nbSignatures
                                             )
str(integrated_cell_signatures)
head(integrated_cell_signatures)

```

### Match integrated (INI254,INI255,INI267) cluster to InnovRT003_noLowUMIclusters cells

```{r group_to_cell_matching}

# Group-to-cell matching
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters <- RunCellHGT(X         = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                                            pathways  = integrated_cluster_signatures,
                                                                            dims      = 30
                                                                            )

str(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters)
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters)))
enrichment_thrs <- 0.25

# Density distribution of enrichement score
melt(as.matrix(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing InnovRT003_noLowUMIclusters cell to integrated cluster according to max enrichment score
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction <- rownames(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters)[apply(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters, 2, which.max)]
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction)
levels(factor(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction))

# Assing InnovRT003_noLowUMIclusters cell to integrated cluster according to max enrichment score with a significant thresold
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign <- ifelse(test = apply(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters, 2, max) > enrichment_thrs,
                                                                       yes = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign)

# Convert into data frame
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df <- data.frame(integratedCluster = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign)

cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df <- rownames_to_column(.data = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df,
                                                                                      var = "InnovRT003_noLowUMIclusters_cell"
                                                                                      )

head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data) == cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df$InnovRT003_noLowUMIclusters_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- AddMetaData(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                 metadata  = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedClusters_prediction_sign_df$integratedCluster,
                                                 col.name  = "cellID_integratedCluster"
                                                 )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID_integratedCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = cellID_integratedCluster,
                             fill  = eval(as.name(choosenResolution))
                             )
               ) +
    labs(fill = "InnovRT003\nnoLowUMI\nclusters") +
    xlab("\nCAL-ATRT integrated\nmatching cluster") +
    geom_text(mapping  = aes(x     = cellID_integratedCluster,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_cellID_integratedCluster.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cellID_integratedCluster",
        label       = TRUE,
        label.size  = 15
        ) +
    labs(color = "CAL-ATRT\nintegrated\nmatching\ncluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

cell_cluster_col <- c("2"  = "violet",
                      "4"  = "violetred",
                      "9"  = "blue4",
                      "3"  = "springgreen4",
                      "5"  = "springgreen",
                      "6"  = "sienna1",
                      "7"  = "sienna3",
                      "8"  = "sienna4",
                      "0"  = "skyblue",
                      "1"  = "skyblue4"
                      )

paperFig_fontsize <- 6

# Plot umap for figure
umap_scRNAseq_cellID_integratedCluster <- DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                  reduction   = "umap",
                                                  pt.size     = 0.25,
                                                  group.by    = "cellID_integratedCluster",
                                                  label       = TRUE,
                                                  label.size  = paperFig_fontsize/2,
                                                  cols        = cell_cluster_col[levels(factor(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data$RNA_snn_res.0.6))],
                                                  label.color = "white",
                                                  label.box = TRUE
                                        #label.color = cluster_number_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))]
                                                  ) +
    xlab("") +
    ylab("") +
    ggtitle("IRT003") +
    theme(legend.position   = "none",
          text              = element_text(size = paperFig_fontsize*1.5),
          axis.text         = element_text(size = paperFig_fontsize*1.5),
          axis.ticks        = element_line(size = 0.25),
          axis.line         = element_line(size = 0.25),
          panel.background  = element_rect(fill = "white")
          )
umap_scRNAseq_cellID_integratedCluster

umap_scRNAseq_integratedCluster <- DimPlot(object      = scRNAseq_Seurat_integrated,
                                                  reduction   = "umap",
                                                  pt.size     = 0.25,
                                                  group.by    = "integrated_snn_res.0.2",
                                                  label       = TRUE,
                                                  label.size  = paperFig_fontsize/2,
                                                  cols        = cell_cluster_col[levels(factor(scRNAseq_Seurat_integrated@meta.data$integrated_snn_res.0.2))],
                                                  label.color = "white",
                                                  label.box = TRUE
                                        #label.color = cluster_number_col[levels(factor(scRNAseq_Seurat_integrated_cellAnnot@meta.data$integrated_snn_res.0.2))]
                                                  ) +
    xlab("") +
    ylab("") +
    ggtitle("integrated") +
    theme(legend.position   = "none",
          text              = element_text(size = paperFig_fontsize*1.5),
          axis.text         = element_text(size = paperFig_fontsize*1.5),
          axis.ticks        = element_line(size = 0.25),
          axis.line         = element_line(size = 0.25),
          panel.background  = element_rect(fill = "white")
          )
umap_scRNAseq_integratedCluster

```

### Figure 8bis patchwork (figure7_LobonIglesias2022)

```{r figure8bis_patchwork}

# UMAP and clusters
figureIII_A <- wrap_plots(umap_scRNAseq_integratedCluster)

# Heterogeneous expression of ATRT-SHH genes
figureIII_B <- wrap_plots(umap_scRNAseq_cellID_integratedCluster)

# Patchwork
figureIII_answerToReviewers_LobonIglesias2022 <- wrap_plots(figureIII_A,
                                                            figureIII_B,
                                                            widths = c(0.5,0.5)
                                                            ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(#title = "FigureIII (Response to Reviewers)\n",
                    theme = theme(plot.margin = unit(c(1,2,2,1),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Save figure into pdf file
ggsave(plot   = figureIII_answerToReviewers_LobonIglesias2022,
       file   = paste0(pathToFigures,"figureIII_answerToReviewers_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297*1/3, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig11_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )
```

### Match integrated (INI254,INI255,INI267) cell to InnovRT003_noLowUMIclusters cells

```{r cell_to_cell_matching}

# Group-to-cell matching
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells <- RunCellHGT(X         = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                                         pathways  = integrated_cell_signatures,
                                                                         dims      = 30
                                                                         )

str(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells)
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells)))
enrichment_thrs <- 0.1

# Density distribution of enrichement score
melt(as.matrix(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing InnovRT003_noLowUMIclusters cell to integrated cluster according to max enrichment score
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction <- rownames(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells)[apply(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells, 2, which.max)]
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction)

# Assing InnovRT003_noLowUMIclusters cell to integrated cluster according to max enrichment score with a significant thresold
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign <- ifelse(test = apply(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells, 2, max) > enrichment_thrs,
                                                                       yes = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign)

# Convert into data frame
cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df <- data.frame(integratedCell = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign)

cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df <- rownames_to_column(.data = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df,
                                                                                      var = "InnovRT003_noLowUMIclusters_cell"
                                                                                      )

head(cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data) == cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df$InnovRT003_noLowUMIclusters_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID <- AddMetaData(object    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
                                                 metadata  = cellID_InnovRT003_noLowUMIclusterscells_vs_integratedCells_prediction_sign_df$integratedCell,
                                                 col.name  = "cellID_integratedCell"
                                                 )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data)

# Extract integrated cell cluster
scRNAseq_Seurat_integrated_cellCluster <- data.frame(integratedCluster = scRNAseq_Seurat_integrated@meta.data[,"integrated_snn_res.0.2"],
                                                     row.names = rownames(scRNAseq_Seurat_integrated@meta.data)
                                                     )

scRNAseq_Seurat_integrated_cellCluster <- rownames_to_column(.data = scRNAseq_Seurat_integrated_cellCluster,
                                                             var = "integratedCell"
                                                             )

head(scRNAseq_Seurat_integrated_cellCluster)

# Merge integrated cell cluster to meta data
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data <- left_join(x      = rownames_to_column(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data),
                                                                          y      = scRNAseq_Seurat_integrated_cellCluster,
                                                                          by   = c("cellID_integratedCell" = "integratedCell")
                                                                          )

rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data) <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data$rowname
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data$rowname <- NULL

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID_integratedCell.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(choosenResolution)),
                             fill  = integratedCluster
                             )
               ) +
    labs(fill = "CAL-ATRT\nintegrated\nmatching\ncell cluster") +
    xlab("InnovRT003 noLowUMI clusters") +
    geom_text(mapping  = aes(x     = eval(as.name(choosenResolution)),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_cellID_integratedCell.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "integratedCluster",
        label       = TRUE,
        label.size  = 15
        ) +
    labs(color = "\nintegrated\nmatching\ncell cluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```


# Trajectory Inference Analysis

## ElpiGraph method

### Trajectory reconstruction for all clusters

#### Prepare matrix

```{r cellCycle}

# Scaled matrix of all genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix[,1:3])
dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix[,1:3])

# Scaled matrix of highly variable genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes[,1:3])
dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes[,1:3])

# Scaled matrix of DEGs
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs <- unique(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix[scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers$gene,])
dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs)

```

##### Principal tree

```{r cellCycle}

computeElasticPrincipalTree_func <- function(matrix)
{
    computeElasticPrincipalTree(X                       = t(matrix),
                                NumNodes                = 30,
                                Lambda                  = .03,
                                Mu                      = 0.1,
                                Do_PCA                  = TRUE,
                                CenterData              = TRUE,
                                drawAccuracyComplexity  = FALSE,
                                drawPCAView             = FALSE,
                                drawEnergy              = FALSE
                                )
}


# Construct a principal Tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)

# Check elpigraph object
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_all_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal Tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs)

# Check elpigraph object
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_all_DEGs_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

computeElasticPrincipalCurve_func <- function(matrix)
{
    computeElasticPrincipalCurve(X                      = t(matrix),
                                 NumNodes               = 30,
                                 Lambda                 = .03,
                                 Mu                     = 0.1,
                                 Do_PCA                 = TRUE,
                                 CenterData             = TRUE,
                                 drawAccuracyComplexity = FALSE,
                                 drawPCAView            = FALSE,
                                 drawEnergy             = FALSE
                                 )
}

# Construct a principal Curve
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)

# Check elpigraph object
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_all_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal Curve
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs)

# Check elpigraph object
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_all_DEGs_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 0, 1, 4 and 10

#### Extract cluster 0, 1, 4 and 10 seurat object and matrix

```{r extract_cluster01410_object}

# Select cluster 0, 2, 4, 6 and 10 marker genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

cluster_01410_DEG <- rownames(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
                                      cluster %in% c(0,1,4,10)))
head(cluster_01410_DEG)
length(cluster_01410_DEG)

# Subset Seurat object with cluster markers
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            RNA_snn_res.0.8 %in% c(0,1,4,10),
                                            features = cluster_01410_DEG
                                            )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410,
                                                                   assay = "RNA",#"integrated",
                                                                   slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_matrix)

length(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters))

# Subset Seurat object with all genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                     RNA_snn_res.0.8 %in% c(0,1,4,10)#,
                                        #features = cluster_024610_DEG
                                                                     )

# Highly variable features
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes <- FindVariableFeatures(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes,
                                                                                    selection.method = "vst",
                                                                                    nfeatures         = nFeatures
                                                                                    )

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes,
                                                                                            assay = "RNA",#"integrated",
                                                                                            slot = "scale.data"#"scale.data"
                                                                                            )
                                                                               )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix)

```

#### Extract cluster 0, 1, 2, 4 and 10 seurat object and matrix

```{r extract_cluster012410_object}

# Select cluster 0, 1, 2, 4, 6 and 10 marker genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

cluster_012410_DEG <- rownames(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
                                      cluster %in% c(0,1,4,10)))
head(cluster_012410_DEG)
length(cluster_012410_DEG)

# Subset Seurat object with cluster markers
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            RNA_snn_res.0.8 %in% c(0,1,4,10),
                                            features = cluster_012410_DEG
                                            )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410,
                                                                   assay = "RNA",#"integrated",
                                                                   slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_matrix)

length(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters))

# Subset Seurat object with all genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                     RNA_snn_res.0.8 %in% c(0,1,4,10)#,
                                        #features = cluster_024610_DEG
                                                                     )

# Highly variable features
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes <- FindVariableFeatures(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes,
                                                                                    selection.method = "vst",
                                                                                    nfeatures         = nFeatures
                                                                                    )

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes,
                                                                                            assay = "RNA",#"integrated",
                                                                                            slot = "scale.data"#"scale.data"
                                                                                            )
                                                                               )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix)

```


### Trajectory reconstruction for cluster 0, 2, 4, 6 and 10

#### Extract cluster 0, 2, 4, 6 and 10 seurat object and matrix

```{r extract_cluster0135_object}

# Select cluster 0, 2, 4, 6 and 10 marker genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

cluster_024610_DEG <- rownames(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
                                      cluster %in% c(0,2,4,6,10)))
head(cluster_024610_DEG)
length(cluster_024610_DEG)

# Subset Seurat object with cluster markers
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            RNA_snn_res.0.8 %in% c(0,2,4,6,10),
                                            features = cluster_024610_DEG
                                            )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610,
                                                                   assay = "RNA",#"integrated",
                                                                   slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix)

length(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters))

# Subset Seurat object with all genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                     RNA_snn_res.0.8 %in% c(0,2,4,6,10)#,
                                        #features = cluster_024610_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes <- FindVariableFeatures(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                                                                                    selection.method = "vst",
                                                                                    nfeatures         = nFeatures
                                                                                    )

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                                                                                            assay = "RNA",#"integrated",
                                                                                            slot = "scale.data"#"scale.data"
                                                                                            )
                                                                               )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610 <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_024610_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610_mvGenes <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_024610_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610_mvGenes[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610_mvGenes)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610_mvGenes[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_024610_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

#### Principal Curve

```{r 024610_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_024610_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal curve
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610_mvGenes <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_024610_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610_mvGenes[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610_mvGenes)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610_mvGenes[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_024610_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610@meta.data$RNA_snn_res.0.8
                )
#dev.off()

```

### Trajectory reconstruction for cluster 0,1,2,4,5,6 and 10

#### Extract cluster 0,1,2,4,5,6 and 10 seurat object and matrix

```{r extract_cluster04610_object}

# Select cluster 0,1,2,4,5,6 and 10 marker genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

cluster_01245610_DEG <- rownames(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
                                      cluster %in% c(0,1,2,4,5,6,10)))
head(cluster_01245610_DEG)
length(cluster_01245610_DEG)

# Subset Seurat object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            RNA_snn_res.0.8 %in% c(0,1,2,4,5,6,10),
                                            features = cluster_01245610_DEG
                                            )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                     RNA_snn_res.0.8 %in% c(0,1,2,4,5,6,10)#,
                                        #features = cluster_024610_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes <- FindVariableFeatures(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes,
                                                                   selection.method = "vst",
                                                                   nfeatures         = nFeatures
                                                                   )

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "scale.data"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_mvGenes_matrix)

```

### Trajectory reconstruction for cluster 0, 4, 6 and 10

#### Extract cluster 0, 4, 6 and 10 seurat object and matrix

```{r extract_cluster04610_object}

# Select cluster 0, 4, 6 and 10 marker genes
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers)

cluster_04610_DEG <- rownames(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters.markers,
                                      cluster %in% c(0,4,6,10)))
head(cluster_04610_DEG)
length(cluster_04610_DEG)

# Subset Seurat object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                            RNA_snn_res.0.8 %in% c(0,4,6,10),
                                            features = cluster_04610_DEG
                                            )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "scale.data"#"scale.data"
                                                                   )
                                                      )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix)

# Subset Seurat object with all genes
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                     RNA_snn_res.0.8 %in% c(0,4,6,10)#,
                                        #features = cluster_024610_DEG
                                                     )

# Highly variable features
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes <- FindVariableFeatures(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                                                                   selection.method = "vst",
                                                                   nfeatures         = nFeatures
                                                                   )

# Data scaling 
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes <- ScaleData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes)

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                                                                           assay = "RNA",#"integrated",
                                                                           slot = "scale.data"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix)

```

#### Principal Tree

```{r cellCycle}

# Construct a principal tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610 <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_04610_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610_mvGenes <- computeElasticPrincipalTree_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_04610_mvGenes_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610_mvGenes[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610_mvGenes)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610_mvGenes[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_tree_04610_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

```

#### Principal Curve

```{r 04610_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610 <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_04610_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data$RNA_snn_res.0.8
                )
#dev.off()

# Construct a principal tree
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610_mvGenes <- computeElasticPrincipalCurve_func(matrix = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix)

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_InnovRT003_noLowUMIclusters_04610_mvGenes_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix),
                TargetPG  = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610_mvGenes[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610_mvGenes)]],
                BootPG    = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610_mvGenes[1:(length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_ElPiGraph_curve_04610_mvGenes)-1)],
                                        #PointSize = 1.5,
                p.alpha = 1,
                GroupsLab = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data$RNA_snn_res.0.8
                  )
#dev.off()

```


### Monocle version 3 method

#### Monocle on all clusters (using mvGenes)

```{r monocle_method}

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes[,1:4])
dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes),
                                                                               row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_mvGenes)
                                                                               )
                                                    )


nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds <- learn_graph(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds,
                                              close_loop          = TRUE,
                                              learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                         prune_graph               = TRUE,
                                                                         rann.k                    = NULL
                                                                         )
                                              )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list[[i]],
                                                  color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list)[i])
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot
}

library(gridExtra)

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_plot_list, ncol = 3))
#dev.off()

```

#### Monocle on all clusters (using DEGs)

```{r monocle_method}

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs[,1:4])
dim(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs)

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs),
                                                                               row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_scaled_matrix_DEGs)
                                                                               )
                                                    )


nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds,
                                                     reduction_method  = "UMAP"
                                                     )
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds <- learn_graph(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds,
                                                   close_loop          = TRUE,
                                                   learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                              prune_graph               = TRUE,
                                                                              rann.k                    = NULL
                                                                              )
                                                   )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_branch_points = FALSE,
                                                       label_cell_groups = FALSE,
                                                       label_leaves = FALSE
                                                       ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_list)[i])
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_allClusters_DEGs_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_DEGs_cds_plot_list, ncol = 3))
#dev.off()

```


#### Monocle on cluster 0, 1, 4 and 10 (using DEG)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- learn_graph(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01410_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01410.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = TRUE,
           label_leaves                   = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

#### Monocle on cluster 0, 1, 4 and 10 (using mvGenes)

```{r monocle_method}

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix,
                                                                   cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_allGenes@meta.data),
                                                                   gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix),
                                                                                              row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01410_mvGenes_matrix)
                                                                                              )
                                                                   )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes,
                                                                num_dim       = nbPC,
                                                                scaling       = TRUE,
                                                                method        = "PCA",
                                                                norm_method   = "none",
                                                                pseudo_count  = 0
                                                                )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes,
                                                                  max_components    = 2,
                                                                  reduction_method  = "UMAP"
                                                                  )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes,
                                                                   reduction_method  = "UMAP"
                                                                   )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes <- learn_graph(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes,
                                                                 close_loop          = FALSE,
                                                                 learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                            prune_graph               = TRUE,
                                                                                            rann.k                    = NULL
                                                                                            )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01410_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01410_mvGenes_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01410_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           cell_size                      = 2,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```


#### Monocle on cluster 0, 1, 2, 4 and 10 (using mvGenes)

```{r monocle_method}

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix,
                                                                   cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_allGenes@meta.data),
                                                                   gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix),
                                                                                              row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012410_mvGenes_matrix)
                                                                                              )
                                                                   )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes,
                                                                num_dim       = nbPC,
                                                                scaling       = TRUE,
                                                                method        = "PCA",
                                                                norm_method   = "none",
                                                                pseudo_count  = 0
                                                                )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes,
                                                                  max_components    = 2,
                                                                  reduction_method  = "UMAP"
                                                                  )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes,
                                                                   reduction_method  = "UMAP"
                                                                   )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes <- learn_graph(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes,
                                                                 close_loop          = FALSE,
                                                                 learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                            prune_graph               = TRUE,
                                                                                            rann.k                    = NULL
                                                                                            )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list[[i]],
                                                       color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust012410_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust012410_mvGenes_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_012410_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           cell_size                      = 2,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points            = FALSE,
           label_roots                    = FALSE,
           label_leaves                   = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```


#### Monocle on cluster 0, 4, 6 and 10 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610 <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust04610_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 4, 6 and 10 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust04610_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 2, 4, 5, 6 and 10 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01245610_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01245610_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01245610_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust0157911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust0157911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911 <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust0157911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_0157911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust0157911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_0157911_mvGenes_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 2, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911 <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust27911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 2, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_27911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust27911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_27911_mvGenes_plot_list, ncol = 2))
#dev.off()

```


#### Monocle on cluster 0, 1, 2, 5, 7, 9 and 11 (using DEGs)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911 <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01257911_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_plot_list, ncol = 2))
#dev.off()

```

#### Monocle on cluster 0, 1, 2, 5, 7, 9 and 11 (using mvGenes)

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- new_cell_data_set(expression_data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_matrix)
                                                                                    )
                                                         )

# Process data
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- preprocess_cds(cds           = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
#scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- align_cds(cds = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes,
 #                                                alignment_group = "orig.ident"
  #                                               )
# Reduce dimension
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- reduce_dimension(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- cluster_cells(cds               = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes <- learn_graph(cds                 = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes,
                                                        close_loop          = FALSE,
                                                        learn_graph_control = list(minimal_branch_len        = 2.5,
                                                                                   prune_graph               = TRUE,
                                                                                   rann.k                    = NULL
                                                                                   )
                                                        )
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes
}

# Rename each cds object in the list
names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list))
{
# Plot
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list[[i]],
                                                        color_cells_by                 = "RNA_snn_res.0.8",
                                        #group_cells_by                               = "cell_type",
                                                        cell_size                      = 1,
                                                        trajectory_graph_color         = "black",
                                                        trajectory_graph_segment_size  = 2,
                                                        label_cell_groups              = FALSE,
                                                        label_branch_points            = FALSE,
                                                        label_leaves                   = FALSE
                                                        ) +
    ggtitle(names(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list)[i])
    scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot_list[[length(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot_list)+1]] <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_InnovRT003_noLowUMIclusters_clust01257911_mvGenes_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_plot_list, ncol = 2))
#dev.off()

```


### CytoTRACE method

CytoTRACE predicts the differentiation state of cells using gene Counts and expression.

#### CytoTRACE all clusters

```{r cytoTrace}

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                                      assay = "RNA",#"integrated",
                                                                      slot = "counts"#"scale.data"
                                                                      )
                                                         )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_rawCounts_matrix)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace <- CytoTRACE(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_rawCounts_matrix)
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_InnovRT003_noLowUMIclusters_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace$CytoTRACE)))

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes <- scRNAseq_Seurat_InnovRT003_noLowUMIclusters

# Append metadata with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes,
                                                         metadata = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace$CytoTRACE)
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes@meta.data)

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes,
                                                         metadata = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = RNA_snn_res.0.8
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_allGenes,
                              clustering  = choosenResolution,
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "median"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes <- data.frame(corr = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoTrace$cytoGenes)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes <- rownames_to_column(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes, var = "gene")
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes)-5):nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

#### CytoTRACE 024610

```{r cytoTrace}

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                                                                            assay = "RNA",#"integrated",
                                                                            slot = "counts"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_rawCounts_matrix)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace <- CytoTRACE(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_rawCounts_matrix)
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_InnovRT003_noLowUMIclusters_024610_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace$CytoTRACE)))

# Append metadata with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                                                         metadata = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace$CytoTRACE)
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes@meta.data)

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                                                         metadata = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_024610_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_024610_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = RNA_snn_res.0.8
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_024610_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_allGenes,
                              clustering  = choosenResolution,
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "median"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes <- data.frame(corr = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoTrace$cytoGenes)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes <- rownames_to_column(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes, var = "gene")
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_024610_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes)-5):nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

#### CytoTRACE 04610

```{r cytoTrace}

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                                                                            assay = "RNA",#"integrated",
                                                                            slot = "counts"#"scale.data"
                                                                           )
                                                              )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_rawCounts_matrix)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace <- CytoTRACE(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_rawCounts_matrix)
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_InnovRT003_noLowUMIclusters_04610_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw()  +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace$CytoTRACE)))

# Append metadata with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                                                         metadata = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace$CytoTRACE)
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes@meta.data)

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                                                         metadata = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_04610_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_04610_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = RNA_snn_res.0.8
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_04610_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_allGenes,
                              clustering  = choosenResolution,
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "median"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes <- data.frame(corr = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoTrace$cytoGenes)
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes <- rownames_to_column(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes, var = "gene")
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_04610_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes)-5):nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

#### CytoTRACE 01257911

```{r cytoTrace}

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes,
                                                                               assay = "RNA",#"integrated",
                                                                               slot = "counts"#"scale.data"
                                                                               )
                                                              )
nrow(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_rawCounts_matrix)
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace <- CytoTRACE(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_rawCounts_matrix)
str(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_InnovRT003_noLowUMIclusters_01257911_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace$CytoTRACE)))

# Append metadata with CytoTRACE
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes,
                                                         metadata = data.frame(CytoTRACE = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace$CytoTRACE)
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes@meta.data)

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes <- AddMetaData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes,
                                                         metadata = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                         )

colnames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_01257911_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_01257911_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = RNA_snn_res.0.8
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_InnovRT003_noLowUMIclusters_01257911_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_allGenes,
                              clustering = choosenResolution,
                              covariate = "CytoTRACE",
                              ylab = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder = "median"
                              )
#dev.off()

#plotCytoTRACE(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace)
#plotCytoGenes(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace)

# Genes correlated with CytoTRACE
cyToGenes_corr <- data.frame(corr = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cytoTrace$cytoGenes)
cyToGenes_corr <- rownames_to_column(cyToGenes_corr, var = "gene")
head(cyToGenes_corr)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_InnovRT003_noLowUMIclusters_01257911_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = cyToGenes_corr[c(1:5,(nrow(cyToGenes_corr)-5):nrow(cyToGenes_corr)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()
      
```


# RNA velocity

## Extract meta-data for scVelo 

### Extract all cluster meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters)),
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters, reduction = "umap"),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters@meta.data,
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_mvGenes_monocle3_UMAP_metaData.csv"
          )

```

#### Extract cluster  0,2,4,6,10 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610)),
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x     = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610, reduction = "umap"),
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_UMAP.csv",
          )

# Extract other mata data
write.csv(x     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610@meta.data,
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_metaData.csv"
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610, reduction = "umap"),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_UMAP.csv",
          )

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_024610 = cluster_024610_DEG),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_DEG.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_024610_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_024610_mvGenes_monocle3_UMAP_metaData.csv"
          )


```


#### Extract cluster  0,4,6,10 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610)),
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x     = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610, reduction = "umap"),
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_UMAP.csv",
          )

# Extract other mata data
write.csv(x     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610@meta.data,
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_metaData.csv"
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610, reduction = "umap"),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_UMAP.csv",
          )

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_04610 = cluster_04610_DEG),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_DEG.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_04610_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_04610_mvGenes_monocle3_UMAP_metaData.csv"
          )

```

#### Extract cluster  0,1,2,5,7,9,11 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911)

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911)),
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_cellIDs.csv",
          row.names  = FALSE
          )

# Extract UMAP embeddings
write.csv(x     = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911, reduction = "umap"),
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_UMAP.csv",
          )

# Extract other mata data
write.csv(x     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911@meta.data,
          file  = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_metaData.csv"
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911, reduction = "umap"),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_UMAP.csv",
          )

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_01257911 = cluster_01257911_DEG),
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_DEG.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_cds_01257911_mvGenes_list[["PCmin"]],
           color_cells_by                 = "RNA_snn_res.0.8",
           #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups = FALSE,
           label_branch_points = FALSE,
           label_leaves = FALSE
           )

scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]
head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_monocle3_UMAP_metaData)

write.csv(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_01257911_mvGenes_monocle3_UMAP_metaData.csv"
          )

```


scVelo analysis is performed in a python script (cellOrigin_ATRT_scRNAseq_integrated_INI254_INI255_INI267_scVelo.py), see you there



# SCENIC analysis

```{r scenic}

# Extract Seurat object for cluster 0,1,2,5,6,7,8,9,10,11,12
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011 <- subset(x = scRNAseq_Seurat_InnovRT003_noLowUMIclusters,
                                                  RNA_snn_res.0.8 %in% c(0,1,2,5,6,7,8,9,10,11)
                                                  )

# Extract cell IDs
write.csv(x          = (Cells(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011)),
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_cellIDs.csv",
          row.names  = FALSE
          )

# Export meta data
write.table(x      = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data,
            file   = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

SCENIC analysis were performed using python on the jupyter notebook cellOrigin_ATRT_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC.ipynb document. The conda-env environment needs to be activated prior running the analysis.


```{r import_SCENIC_AUCscore}

# Path to SCENIC result
pathToSCENICresult <- paste0(pathToThisProject,"svn/analyse/script/SCENIC_InnovRT003_noLowUMIclusters/")

```

### Gene Regulatory Network

#### Import GRN data from SCENIC analysis

```{r import_SCENIC_GRN}

# Import GNR table
scRNAseq_InnovRT003_noLowUMIclusters_scenic_GNR <- read.table(file = paste0(pathToSCENICresult,"scRNASeq_InnovRT003_noLowUMIclusters_scenic_GRN.tsv"),
                                             header = TRUE,
                                             sep = "\t",
                                             check.names = TRUE,
                                             skip = 2
                                             )
colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_GNR) <- c("TF","MotifID","AUC","Annotation","Context","MotifSimilarityQvalue","NES","OrthologousIdentity","RankAtMax","TargetGenes")
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_GNR)

```

### Cellular enrichment of regulons using AUC score

#### Import AUC score matrix

```{r import_SCENIC_AUCscore}

# Import SCENIC AUC score
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix.csv"),
                                                  check.names = FALSE)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix)
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix)

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix.csv"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

# Transpose AUC matrix
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t <- t(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix)

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t[,1:3])
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t)

# Density distribution
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_melt <- melt(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_melt)

AUCthres <- 0.01
ggplot(data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_melt) +
    geom_density(mapping = aes(x = value
                               ),
                 fill = "grey"
                 ) +
    geom_vline(xintercept=AUCthres, color = "red") +
    theme_bw()

# Filtering
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt <- expFilter(data      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t,
                                                          threshold = AUCthres,
                                                          p         = 0.05,
                                                          graph     = TRUE
                                                          )

dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt)

```

#### Heatmap of AUC score

```{r SCENIC_AUCscore_heatmap}

# Most variable genes
mvRegulons <- genes.selection(data      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt,
                              thres.num = 50
                              )

# Subset matrix
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt_mvRegulons <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt[mvRegulons,]

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data) == colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt_mvRegulons))

# Top annotation
topAnnot_PCmin <- HeatmapAnnotation(df                      = data.frame(Cluster = as.character(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data[,"RNA_snn_res.0.8"]), check.names = FALSE),
                                    col                     = list(Cluster = cluster_n12_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(2,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Define heatmap color
col_fun_AUCscore = colorRamp2(breaks = c(0,0.2,0.4,0.6), colors = c("cyan","wheat1","red","darkred"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt_mvRegulons,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_split                = 6,
             row_split                   = 14,
             col                         = col_fun_AUCscore,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "AUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_AUCscore.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

#### UMAP of cells based on AUC score

```{r AUCscore_umap}

# Compute UMAP
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap <- data.frame(umap(t(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_filt_mvRegulons)))
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap)

colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap) <- c("UMAP1","UMAP2")
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap)

# Bind meta data
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap_plus <- cbind(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap,
                                              cluster = as.character(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data$RNA_snn_res.0.8)
                                              )

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap_plus)

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_AUCscore.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_umap_plus,
       mapping = aes(x = UMAP1, y = UMAP2)
       ) +
    geom_point(mapping = aes(color = cluster
                             ),
               size = 2
               )+
    xlab("UMAP1") +
    ylab("UMAP2") +
    ggtitle(NULL) +
    theme_bw()+
    theme(text               = element_text(size = 20),
          legend.position    = "none"
          #legend.title       = element_text(face = "bold"),
          #legend.key.height  = unit(1.5,"cm")
          )
#dev.off()

```	

### Cellular enrichment of regulons using binarized AUC score

#### Import binarized AUC score matrix

```{r import_SCENIC_bin_AUCscore}

# Import SCENIC AUC score
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin.csv"),
                                                      check.names = FALSE)

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin)
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin)

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin.csv"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

# Transpose AUC_BIN matrix
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t <- t(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin)

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t[,1:3])
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t)

# Density distribution
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt <- melt(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt)

scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt <- mutate(.data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt,
                                                       value = as.character(value)
                                                       )

ggplot(data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt) +
    geom_bar(mapping = aes(x = value
                                 ),
                   stat = "count",
                   fill = "grey"
             ) +
    xlab("\nbinarized AUC score") +
    ylab("Count\n") +
    theme_bw()

```

#### Filtering

```{r AUC_bin_filtering}

# Remove all lines where counts are all equal to zero
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t[apply(X = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t,
                                                                                            MARGIN = 1,
                                                                                            FUN = function(x) !all(x==0)),
                                                                                      ]

dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t)
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ)

# Filtering
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt <- expFilter(data      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ,
                                                             threshold = 0.5,
                                                             p         = 0.05,
                                                             graph     = TRUE
                                                             )

dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt)

```

#### Heatmap of AUC_BIN score

```{r SCENIC_AUC_BINscore_heatmap}

# Most variable genes
mvRegulons_auc_bin <- genes.selection(data      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt,
                                      thres.num = 50
                                      )

# Subset matrix
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt[mvRegulons_auc_bin,]

# Check order
all(rownames(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data) == colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons))

# Define heatmap color
col_fun_AUCbinScore = colorRamp2(breaks = c(0,1), colors = c("white","black"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons,
             clustering_distance_rows    = "binary",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_row_dend               = FALSE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data[,"RNA_snn_res.0.8"])),
             column_split                = factor(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data[,"RNA_snn_res.0.8"]),
             #row_split                   = 2,
             col                         = col_fun_AUCbinScore,
             show_heatmap_legend         = FALSE
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_AUCbinScore.pdf"), width = 12, height = 12)
draw(h)
#dev.off()

```

#### Heatmap of the percentage of cells with activated regulon

```{r heatmap_perc_active_regulon}

# Melt matrix
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt <- melt(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons)
colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt) <- c("regulon","cellID","AUCbin")
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt)

# Merge matrix and meta data
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus <- merge(x     = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data["RNA_snn_res.0.8"],
                                                                              y     = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt,
                                                                              by.x  = 0,
                                                                              by.y  = "cellID",
                                                                              all   = FALSE
                                                                              )
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus)

# Create table of cluster, regulon and number of cells
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells <- group_by(.data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus,
                                                                                          RNA_snn_res.0.8,
                                                                                          regulon
                                                                                          ) %>%
    summarise(number =  n())
colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells) = c("cluster","regulon","nbCell")
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells)

# Create table of cluster, regulon and number of cells in which regulon is activated
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells <- filter(.data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus, AUCbin == 1) %>%
    group_by(RNA_snn_res.0.8,
             regulon
             ) %>%
    summarise(number =  n())
colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells) = c("cluster","regulon","nbCell_AUCbin1")
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells)

# Create a table of total number of cells and cells in which regulon is activated
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells <- merge(x = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells,
                                                                                                   y = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells,
                                                                                                   by = c("cluster","regulon"),
                                                                                                   all = FALSE
                                                                                                   )
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Compute percentage of cells with active regulon
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$activeRegulon <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell_AUCbin1*100/scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Transform table to matrix (for ComplexHeatmap function)
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix <- spread(data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells[,c("cluster","regulon","activeRegulon")],
                                                                                                 key = cluster,
                                                                                                 value = activeRegulon
                                                                                                 )
rownames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix) <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon <- NULL
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix)



# Top annotation
topAnnot_cluster <- HeatmapAnnotation(cluster                  = colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
                                      text                     = anno_text(x= colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix), rot = 0),
                                      col                      = list(cluster = cluster_n12_col),
                                      gap                      = unit(4,"mm"),
                                      show_annotation_name     = FALSE,
                                      annotation_name_gp       = gpar(fontsize = 20),
                                      annotation_legend_param  = list(fontsize   = 20,
                                                                      title_gp    = gpar(fontsize = 20, font = 2),
                                                                      labels_gp   = gpar(fontsize = 20),
                                                                      grid_width  = unit(12,"mm"),
                                                                      grid_height = unit(12,"mm"),
                                                                      ncol        = 1
                                                                      ),
                                      simple_anno_size        = unit(2,"cm"),
                                      na_col                  = "white",
                                      show_legend             = FALSE
                                      )

# Heatmap
h <- Heatmap(matrix                    = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix,
             cluster_columns           = FALSE,
             clustering_distance_rows  = "euclidean",
             clustering_method_rows    = "average",
             top_annotation            = topAnnot_cluster,
             show_row_dend             = FALSE,
             column_title              = "cluster\n",
             row_title                 = "Top 50 most variable regulons",
             column_split              = colnames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
             show_column_names         = FALSE,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(2,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n% of cells with active regulon",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter",
                                                direction = "horizontal"
                                                )
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_AUCbinScore_perc.pdf"), width = 12, height = 12)
draw(h,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Cell type specific regulators

#### Import SCENIC RSS score

```{r import_SCENIC_AUCscore}

# Import SCENIC RSS score
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix.csv"),
                                                  check.names = FALSE
                                                  )
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix[,1:4]
dim(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix)

# Create column of cluster
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix$cluster <- rownames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix)

# Melt table
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt <- melt(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt)

# Density distribution
ggplot(data = scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_bin_t_melt) +
    geom_density(mapping = aes(x = value),
                   fill = "grey"
             ) +
    xlab("\nRSS score") +
    ylab("density\n") +
    theme_bw()

# Transpose RSS matrix (to be copied to zerkalo and then explored by Maria)
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t <- t(data.matrix(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix))
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t)
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t <- data.frame(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t,
                                                      check.names = FALSE
                                                      )
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t$TF <- rownames(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t)

# Export (to be copied to zerkalo)
write.table(x = scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t,
            file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t.txt"),
            quote = FALSE,
            sep = "\t",
            row.names = FALSE
            )
            
# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_t.txt"),
              zerkaloDir = "mandrian_cellOrigin_ATRT"
              )

```

#### Plot top 5 RSS score

```{r import_SCENIC_AUCscore}

# Convert variable column (regulon) into factor
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$variable <- factor(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$variable)
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt)

levels(factor(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$cluster))

scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt[which(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$cluster == "9"),] %>%
    .[order(.$value, decreasing = TRUE),] %>%
    .[1:5,"variable"]

# Plot RSS per cluster
scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_plot_list = list()
all_top5 <- data.frame(regulon = character(0), cluster = character(0))
for (cluster in levels(factor(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt[which(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_matrix_melt$cluster == cluster),]
    # Pick top 5 regulons
    top5 <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    #all_top5 <- c(all_top5, as.vector(top5))
    all_top5 <- rbind(all_top5,
                      data.frame(regulon = top5, cluster = cluster)
                      )
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% top5),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% top5),
                        mapping = aes(label = variable),
                        color = "red",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_plot_list[[length(scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_plot_list)+1]] <- p
}

library(ggpubr)

#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_RSS.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_InnovRT003_noLowUMIclusters_scenic_rss_plot_list, ncol=5, nrow=2)
#dev.off()

all_top5

all_top5[duplicated(all_top5$regulon),]

```

#### Heatmap (AUC score) of top 5 RSS score regulons

```{r import_SCENIC_AUCscore}

head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t)

# Heatmap of all top5
scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_allTop5 <- scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t[as.vector(all_top5$regulon),]
head(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_allTop5[,1:4])

# Density of scaled UCS score
plot(density(t(scale(t(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_allTop5)))))

col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0), colors = c("cyan","white","red","darkred"))
#col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0,5), colors = c("blue","cyan","white","yellow","red"))
# Heatmap
h <- Heatmap(matrix                      = t(scale(t(scRNAseq_InnovRT003_noLowUMIclusters_scenic_auc_matrix_t_allTop5))),
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = FALSE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Cluster specific regulons",
             top_annotation              = topAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data[,"RNA_snn_res.0.8"])),
             column_split                = factor(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011@meta.data[,"RNA_snn_res.0.8"]),
             row_split                   = all_top5$cluster, #rep(c("0","1","3","4","5","6","7","8","9"), each = 5),
             col                         = col_fun_AUCscore_scaled,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "scaled\nAUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_InnovRT003_noLowUMIclusters_SCENIC_AUCscore_scaled_allTop5.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```


# Pseudo-bulk analysis

```{r PseudoBulk}

# Extract expression matrix
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_counts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011,
                                                                   assay = "RNA",#"integrated",
                                                                   slot = "counts"#"scale.data"
                                                                   )
                                                           )

head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_counts_matrix[,1:3])

# Create pseudo-bulk by summing the raw counts of all cells
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix <- data.frame(InnovRT003_noLowUMIclusters = rowSums(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_counts_matrix))
scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix <- rownames_to_column(.data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix,
                                                                                  var = "gene"
                                                                                  )


head(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix)

nrow(filter(scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix, InnovRT003_noLowUMIclusters != 0))

# Check the density distribution of pseudo counts
ggplot(data = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix
       ) +
    geom_density(mapping = aes(x = log2(InnovRT003_noLowUMIclusters + 1)),
                 color = "blue",
                 size = 1.5
                 ) +
    theme_bw()

# Extract cell IDs
write.csv(x          = scRNAseq_Seurat_InnovRT003_noLowUMIclusters_012567891011_pseudoCounts_matrix,
          file       = "scRNAseq_Seurat_InnovRT003_noLowUMIclusters_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )

```

# inferCNV analysis

## Create inferCNV object

### Extract raw count matrix

```{r extract_rawCount}

# Extract raw count matrix
scRNAseq_rawCount_InnovRT003 <- scRNAseq_Seurat_InnovRT003@assays$RNA@counts

head(scRNAseq_rawCount_InnovRT003[,1:4])
dim(scRNAseq_rawCount_InnovRT003)

```

### Create gene position file

```{r gene_order_file}

# Import gene position file
gene_position <- read.table(file = paste0(pathToData,"gencode_v19_gen_pos.complete.txt"),
                            header = FALSE,
                            sep = "\t"
                            )
colnames(gene_position) <- c("gene","chr","start","end")
head(gene_position)

# Keep only gene name
gene_position_plus <- mutate(gene_position,
                             gene = gsub("\\|.*","",gene)
                             )
head(gene_position_plus)

# Keep only genes present in matrix
gene_position_Plus <- filter(.data = gene_position_plus,
                             gene %in% rownames(scRNAseq_rawCount_InnovRT003)
                             )

head(gene_position_Plus)
dim(gene_position_Plus)

# Remove duplicated gene name
gene_position_PLUS <- gene_position_Plus[!duplicated(gene_position_Plus$gene),]

head(gene_position_PLUS)
dim(gene_position_PLUS)


# Export cell annotation
write.table(file       = "InnovRT003_gene_position.txt",
            x          = gene_position_PLUS,
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE,
            sep        = "\t"
            )

```


### Generate annotation file

```{r generate_annotation_file}

# Generate annotations table
InnovRT003_cell_annotation <- scRNAseq_Seurat_InnovRT003@meta.data["orig.ident"]

InnovRT003_cell_annotation$cell_ID <- rownames(InnovRT003_cell_annotation)
rownames(InnovRT003_cell_annotation) <- NULL

InnovRT003_cell_annotation$cell_type <- gsub("InnovRT003_","",InnovRT003_cell_annotation$orig.ident)

InnovRT003_cell_annotation$orig.ident <- NULL

head(InnovRT003_cell_annotation)

levels(factor(InnovRT003_cell_annotation$cell_type))

# Export cell annotation
write.table(file       = "InnovRT003_all_cell_annotation.txt",
            x          = InnovRT003_cell_annotation,
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE,
            sep        = "\t"
            )

```

### Create infercnv object

```{r create_infercnv_object}

# Create inferCNV object
InnovRT003_infercnvObject <- CreateInfercnvObject(raw_counts_matrix        = scRNAseq_rawCount_InnovRT003,
                                                  gene_order_file          = "InnovRT003_gene_position.txt",
                                                  annotations_file         = "InnovRT003_all_cell_annotation.txt",
                                                  ref_group_names          = NULL,
                                                  delim                    = "\t",
                                                  max_cells_per_group      = NULL,
                                                  min_max_counts_per_cell  = NULL,
                                                  chr_exclude              = c("chrX","chrY","chrM")
                                                  )


```

### Run inferCNV

```{r run_infercnv}

# Create inferCNV object
InnovRT003_infercnvObject_res <- infercnv::run(infercnv_obj       = InnovRT003_infercnvObject,
                                               cutoff             = 0.1,
                                               out_dir            = tempfile(), 
                                               cluster_by_groups  = TRUE, 
                                               denoise            = TRUE,
                                               HMM                = TRUE,
                                               num_threads        = 4,
                                               no_plot            = TRUE
                                               )

```

### Explore inferCNV result

```{r explore_infercnv}

# Processed expession matrix
head(InnovRT003_infercnvObject_res@expr.data[,1:4])

# List of normal cell indices
InnovRT003_infercnvObject_res@reference_grouped_cell_indices

# List of tumor cell indices
InnovRT003_infercnvObject_res@observation_grouped_cell_indices

```

### Plot CNV with plot_cnv

```{r plot_cnv}

# Plot CNV
plot_cnv(infercnv_obj        = InnovRT003_infercnvObject_res,
         out_dir             = "./inferCNV_InnovRT003",
         title               = "inferCNV",
         obs_title           = "Observations (Cells)",
         ref_title           = "References (Cells)",
         cluster_by_groups   = TRUE,
         cluster_references  = FALSE,
         plot_chr_scale      = FALSE,
         chr_lengths         = NULL,
         k_obs_groups        = 3,
         contig_cex          = 1,
         x.center            = mean(infercnv_obj@expr.data),
         x.range             = "auto",
         hclust_method       = "ward.D",
         custom_color_pal    = NULL,
         color_safe_pal      = FALSE,
         output_filename     = "InnovRT003_infercnv",
         output_format       = "png",
         png_res             = 300,
         dynamic_resize      = 0,
         ref_contig          = NULL,
         write_expr_matrix   = FALSE,
         useRaster           = FALSE
         )

```

### Plot CNV with ComplexHeatmap

#### Plot for all chr

```{r complexheatmap}

# Extract processed expression data
InnovRT003_infercnvObject_res_expr <- data.frame(InnovRT003_infercnvObject_res@expr.data, check.names = FALSE)

head(InnovRT003_infercnvObject_res_expr[,1:2])
dim(InnovRT003_infercnvObject_res_expr)

# Transpose matrix
InnovRT003_infercnvObject_res_expr_t <- data.frame(t(InnovRT003_infercnvObject_res_expr), check.names = FALSE)

head(InnovRT003_infercnvObject_res_expr_t[,1:10])
dim(InnovRT003_infercnvObject_res_expr_t)

# Check order for row split
all(InnovRT003_cell_annotation$cell_ID == rownames(InnovRT003_infercnvObject_res_expr_t))

# Subset genes in matrix
gene_position_PLUS_all <- filter(.data = gene_position_PLUS,
                                 gene %in% colnames(InnovRT003_infercnvObject_res_expr_t)
                                 )

# Order chromosome column
chrs <- data.frame(chr = c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22")
                   )

gene_position_PLUS_all_plus <- left_join(x = chrs,
                                         y = gene_position_PLUS_all,
                                         by = "chr"
                                         )

# Check order
all(gene_position_PLUS_all_plus$gene == colnames(InnovRT003_infercnvObject_res_expr_t))

InnovRT003_infercnvObject_res_expr_t_all <- InnovRT003_infercnvObject_res_expr_t[,as.vector(gene_position_PLUS_all_plus$gene)]

all(gene_position_PLUS_all_plus$gene == colnames(InnovRT003_infercnvObject_res_expr_t_all))

head(scRNAseq_Seurat_InnovRT003@meta.data)

# Import metadata from integrated analysis
#integrated_metadata <- read.table(file       = paste0(pathToReport,"scRNAseq_Seurat_integrated_metadata.csv"),
 #                                 sep        = ",",
  #                                header  = TRUE
   #                               )

#tail(integrated_metadata)

#integrated_metadata_InnovRT003 <- filter(.data = integrated_metadata,
 #                                    orig.ident == "InnovRT003"
  #                                   )
#head(integrated_metadata_InnovRT003)
#nrow(integrated_metadata_InnovRT003)

#tail(integrated_metadata_InnovRT003$item)

#integrated_metadata_InnovRT003$cell_ID <- gsub(pattern = "_3",
 #                                          replacement = "",
                                        #paste0("InnovRT003_tumor_",integrated_metadata_InnovRT003$item)
  #                                         integrated_metadata_InnovRT003$item
   #                                        )

#head(InnovRT003_cell_annotation)
#nrow(InnovRT003_cell_annotation)

head(scRNAseq_Seurat_InnovRT003@meta.data["RNA_snn_res.0.8"])

InnovRT003_cell_annotation_plus <- merge(x      = InnovRT003_cell_annotation,
                                         y      = scRNAseq_Seurat_InnovRT003@meta.data["RNA_snn_res.0.8"],#integrated_metadata_InnovRT003[,c("cell_ID","integrated_snn_res.0.2")],
                                         by.x = 'cell_ID',
                                         by.y = 0,
                                         all.x  = TRUE,
                                         all.y  = FALSE
                                         )


head(InnovRT003_cell_annotation_plus)
nrow(InnovRT003_cell_annotation_plus)

# Create cluster vector color
cell_cluster_col <- c("2"  = "violet",
                      "4"  = "violetred",
                      "9"  = "blue4",
                      "3"  = "springgreen4",
                      "5"  = "springgreen",
                      "6"  = "sienna1",
                      "7"  = "sienna3",
                      "8"  = "sienna4",
                      "0"  = "skyblue",
                      "1"  = "skyblue4",
                      "10" = "grey",
                      "11" = "red"
                      )

# Top annotation
row_annot_inferCNV <- HeatmapAnnotation(df = data.frame(cell_type = factor(InnovRT003_cell_annotation_plus[,c("RNA_snn_res.0.8")])),
                                        col = list(cell_type = cell_cluster_col),
                                        which = 'row'
                                        )

head(InnovRT003_infercnvObject_res_expr_t_all[,1:4])
nrow(InnovRT003_infercnvObject_res_expr_t_all)


# Complexheatmap
#pdf(file = paste0(pathToFigures,"inferCNV_scRNAseq_InnovRT003_tumor_Tcells_all.pdf"), width = 16, height = 10)
Heatmap(matrix               = InnovRT003_infercnvObject_res_expr_t_all,
        cluster_columns      = FALSE,
        show_row_dend        = FALSE,
        jitter               = TRUE,
        show_column_names    = FALSE,
        show_row_names       = FALSE,
        row_split            = InnovRT003_cell_annotation$cell_type,
        column_split         = gene_position_PLUS_all_plus$chr ,
        column_title_rot     = 90,
        column_order         = colnames(InnovRT003_infercnvObject_res_expr_t_all),
        row_gap              = unit(5, "mm"),
        show_heatmap_legend  = FALSE
        )
#dev.off()

```

#### Plot for all chr for down sampled cells

```{r complexheatmap}

head(InnovRT003_cell_annotation_plus)

levels(factor(InnovRT003_cell_annotation_plus$RNA_snn_res.0.8))

# Remove tumor cells without cluster (filtered tumor cells)
InnovRT003_cell_annotation_filtTumorCells <- filter(.data     = InnovRT003_cell_annotation_plus,
(cell_type == "tumor" & !is.na(RNA_snn_res.0.8)) |
cell_type != "tumor"
)

nrow(InnovRT003_cell_annotation_plus)
nrow(InnovRT003_cell_annotation_filtTumorCells)

# Down sample cells
InnovRT003_cell_annotation_n <- sample_n(InnovRT003_cell_annotation_filtTumorCells,
                                    1000
                                    )

# Top annotation
row_annot_inferCNV_n <- HeatmapAnnotation(df = data.frame(tumoral_cluster = as.character(InnovRT003_cell_annotation_n[,c("RNA_snn_res.0.8")])),
                                          col = list(tumoral_cluster = cell_cluster_col),
                                          which = 'row',
                                          show_annotation_name = FALSE
                                          )

# Subset matrix of down sampled cells
InnovRT003_infercnvObject_res_expr_t_n <- InnovRT003_infercnvObject_res_expr_t_all[as.vector(InnovRT003_cell_annotation_n$cell_ID),]

# Check order for row split
all(InnovRT003_cell_annotation_n$cell_ID == rownames(InnovRT003_infercnvObject_res_expr_t_n))

# Order chromosome column
chrs <- data.frame(chr = c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22")
                   )
gene_position_PLUS_all_plus <- left_join(x = chrs,
                                         y = gene_position_PLUS_all,
                                         by = "chr"
                                         )

gene_position_PLUS_all_plus$chr <- factor(gene_position_PLUS_all_plus$chr,
                                          levels = c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22")
                                          )

head(gene_position_PLUS_all_plus)

# Check order
all(gene_position_PLUS_all_plus$gene == colnames(InnovRT003_infercnvObject_res_expr_t_n))

InnovRT003_infercnvObject_res_expr_t_all_n <- InnovRT003_infercnvObject_res_expr_t_n[,as.vector(gene_position_PLUS_all_plus$gene)]

all(gene_position_PLUS_all_plus$gene == colnames(InnovRT003_infercnvObject_res_expr_t_all_n))

# Complexheatmap
#pdf(file = paste0(pathToFigures,"inferCNV_scRNAseq_InnovRT003_tumor_Tcells_all_n.pdf"), width = 16, height = 10)
Heatmap(matrix               = InnovRT003_infercnvObject_res_expr_t_all_n,
        left_annot            = row_annot_inferCNV_n,
        cluster_columns      = FALSE,
        show_row_dend        = FALSE,
        #jitter               = TRUE,
        show_column_names    = FALSE,
        show_row_names       = FALSE,
        row_split            = InnovRT003_cell_annotation_n$cell_type,
        column_split         = gene_position_PLUS_all_plus$chr,
        column_title_rot     = 90,
        #column_order         = factor(gene_position_PLUS_all_plus$chr),
        row_gap              = unit(5, "mm"),
        show_heatmap_legend  = FALSE
        )
#dev.off()

```


#### Plot for chr22 (to check SMARCB1 CNV)

```{r complexheatmap}

# Subset chr22 genes
gene_position_PLUS_chr22 <- filter(.data = gene_position_PLUS,
                                   chr   == "chr22"
                                   )

head(gene_position_PLUS_chr22)
dim(gene_position_PLUS_chr22)

# Subset chr22 genes in matrix
gene_position_PLUS_chr22 <- filter(.data = gene_position_PLUS_chr22,
                                   gene %in% colnames(InnovRT003_infercnvObject_res_expr_t)
                                   )


head(gene_position_PLUS_chr22)
dim(gene_position_PLUS_chr22)


# Subset matrix
InnovRT003_infercnvObject_res_expr_t_chr22 <- InnovRT003_infercnvObject_res_expr_t[,as.vector(gene_position_PLUS_chr22$gene)]

dim(InnovRT003_infercnvObject_res_expr_t_chr22)

# Check order for row split
all(InnovRT003_cell_annotation$cell_ID == rownames(InnovRT003_infercnvObject_res_expr_t_chr22))


# Complexheatmap
Heatmap(matrix           = InnovRT003_infercnvObject_res_expr_t_chr22,
        cluster_columns  = FALSE,
        show_row_dend    = FALSE,
        jitter           = TRUE,
        show_row_names   = FALSE,
        row_split        = InnovRT003_cell_annotation$cell_type
        )

```


# Session info

```{r sessionInfo}

# Session info
sessionInfo()

```
