---
title: "Rhabdoid tumors in mouse model"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    fig_width: 9
    fig_height: 8
    fig_caption: true
    df_print: paged
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r library_file_import, echo=FALSE}

# Paths
pathToHome                   <- "/bioinfo/users/mandrian/"
pathToRhabdoid_U830_projects <- paste0(pathToHome,"rhabdoid_U830_projects/")
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages/")
pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.6/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToMurine_RT              <- paste0(pathToANALYSIS_PROJECTS,"murine_RT/")
pathToMurine_RT_info         <- paste0(pathToMurine_RT,"svn/analyse/info/")
pathToReport                 <- paste0(pathToMurine_RT,"svn/analyse/report/") 
pathToData                       <- paste0(pathToMurine_RT,"data/") 
pathToFigures                <- paste0(pathToMurine_RT,"svn/analyse/script/figures/") 

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r load_library}

#.libPaths(pathToRpackages)

source("load_integration_library.R")

```

```{r load_saved_RData, eval=TRUE}

load("murine_RT_integration_murineRNAseq_humanRNAseq.RData")

#save.image("murine_RT_integration_murineRNAseq_humanRNAseq.RData")

```

```{r function_plot_files, eval=TRUE}

# Load external files
source("/bioinfo/users/mandrian/rhabdoid_U830_projects/programmes/plot_functions.R")
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")

```

# Import human ATRT RNAseq data

```{r import_human_ATRT_samples}

# Load RData
load("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/script/human_ATRT_primTum_Curie_RNAseq_list.RData")

str(human_ATRT_primTum_Curie_RNAseq_list, max.level=1)

# Extract sample plan
sample_ATRT_primTum_Curie_RNAseq <- human_ATRT_primTum_Curie_RNAseq_list$sampleTable
nrow(sample_ATRT_primTum_Curie_RNAseq)

# Extract rlog matrix
matrix_rld_ATRT_primTum_Curie_RNAseq <- human_ATRT_primTum_Curie_RNAseq_list$matrix_rld
dim(matrix_rld_ATRT_primTum_Curie_RNAseq)

# Import Patient ID vs paper ID table
patientID_vs_paperID <- read.table(file = paste0(pathToDataAnnotationFile,"Patient_ID_vs_paper_ID.csv"),
                                   header = TRUE
                                   )
patientID_vs_paperID

```

# Import Curie murine RNAseq data

```{r import_murine_RNAseq_samples}

# Load mouse Curie data
load("mouse_Curie_RNAseq_list.RData")

# Extract sample annotation
sample_murine_RNAseq <- mouse_Curie_RNAseq_list$sampleTable
head(sample_murine_RNAseq)
nrow(sample_murine_RNAseq)

# Extract matrix
matrix_rld_murine_RNAseq <- data.frame(mouse_Curie_RNAseq_list$matrixRld, check.names=FALSE)
head(matrix_rld_murine_RNAseq)
dim(matrix_rld_murine_RNAseq)

```

## Import murine assigned sample annotation

```{r import_murine_RNAseq_samples}

# Import R26 in situ assigned sample annotation
sample_R26_primaryTum <- read.table(file       = "sample_Curie_R26_primTum.csv",
                                    #quote     = FALSE,
                                    sep      = "\t",
                                    header = TRUE
                                    )
head(sample_R26_primaryTum)
nrow(sample_R26_primaryTum)
ncol(sample_R26_primaryTum)

# Import R26 allograft "assigned" sample annotation
sample_R26_allo <- read.table(file    = "sample_R26_allo.csv",
                              header  = TRUE,
                              sep     = ","
                              )
head(sample_R26_allo)
nrow(sample_R26_allo)
ncol(sample_R26_allo)

# Import Prss56 "assigned" sample annotation
sample_Prss56_plus <- read.table(file      = "sample_Prss56_plus.csv",
                                 header    = TRUE,
                                 sep       = ","
                                 )
head(sample_Prss56_plus)
nrow(sample_Prss56_plus)
ncol(sample_Prss56_plus)

# Import Nestin independent assigned sample annotation
sample_NestinInd_plus <- read.table(file      = "sample_NestinInd_plus.csv",
                                    header    = TRUE,
                                    sep       = ","
                                    )
# Remove silhouette and cluster columns 
sample_NestinInd_min <- dplyr::select(.data = sample_NestinInd_plus, -silhouette, -cluster)
head(sample_NestinInd_min)
nrow(sample_NestinInd_min)
ncol(sample_NestinInd_min)

# Import wild-type "assigned" sample annotation
sample_wildType_plus <- read.table(file      = "sample_wildType_plus.csv",
                                   header    = TRUE,
                                   sep       = ","
                                   )
head(sample_wildType_plus)
nrow(sample_wildType_plus)
ncol(sample_wildType_plus)

# Bind assigned samples
sample_murine_RNAseq_plus <- Reduce(f= rbind,
                                    x = list(sample_R26_primaryTum
                                            ,sample_R26_allo
                                            ,sample_Prss56_plus
                                            ,sample_NestinInd_min
                                            ,sample_wildType_plus
                                             )
                                    )
head(sample_murine_RNAseq_plus)
nrow(sample_murine_RNAseq_plus)
ncol(sample_murine_RNAseq_plus)

```

## Subset murine matrix (using assigned sample plan)

```{r biomart_correspondance_ID}

# Subset matrix
matrix_rld_murine_RNAseq <- matrix_rld_murine_RNAseq[,as.vector(sample_murine_RNAseq_plus$Sample_name)]

colnames(matrix_rld_murine_RNAseq) == as.vector(sample_murine_RNAseq_plus$Sample_name)

```

# Convert human gene names to murine gene names

```{r convert_humanGene_to_mouseGene}

# Load human and mouse ensembl datasets annotation
human_mart <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")

# P
humanGeneName_mouseGeneName <- getLDS(attributes = "hgnc_symbol",
                                      mart       = human_mart,
                                      attributesL= "mgi_symbol",
                                      martL      = musculus_mart,
                                      uniqueRows = TRUE
                                      )
head(humanGeneName_mouseGeneName)

# Convert human gene name into mouse gene name
matrix_rld_ATRT_primTum_Curie_RNAseq_plus <- merge(x     = matrix_rld_ATRT_primTum_Curie_RNAseq,
                                                   y     = humanGeneName_mouseGeneName,
                                                   by.x  = 0,
                                                   by.y  = "HGNC.symbol",
                                                   all   = FALSE
                                                   )
head(matrix_rld_ATRT_primTum_Curie_RNAseq_plus)

# Remove duplicated MGI symbol
matrix_rld_ATRT_primTum_Curie_RNAseq_Plus <- matrix_rld_ATRT_primTum_Curie_RNAseq_plus[!duplicated(matrix_rld_ATRT_primTum_Curie_RNAseq_plus$MGI.symbol),]

# Put MGI symbol as rownames
rownames(matrix_rld_ATRT_primTum_Curie_RNAseq_Plus) <- matrix_rld_ATRT_primTum_Curie_RNAseq_Plus$MGI.symbol

# Remove extra-columns
matrix_rld_ATRT_primTum_Curie_RNAseq_Plus$Row.names <- NULL
matrix_rld_ATRT_primTum_Curie_RNAseq_Plus$MGI.symbol <- NULL

head(matrix_rld_ATRT_primTum_Curie_RNAseq_Plus)

```

# Distribution of human and mouse data

```{r distribution}

# Melt human matrix
matrix_rld_ATRT_primTum_Curie_RNAseq_Plus_melt <- melt(matrix_rld_ATRT_primTum_Curie_RNAseq_Plus)
colnames(matrix_rld_ATRT_primTum_Curie_RNAseq_Plus_melt) <- c("Sample_name","rlogExp")
head(matrix_rld_ATRT_primTum_Curie_RNAseq_Plus_melt)

# Melt mouse matrix
matrix_rld_murine_RNAseq_melt <- melt(matrix_rld_murine_RNAseq)
colnames(matrix_rld_murine_RNAseq_melt) <- c("Sample_name","rlogExp")
head(matrix_rld_murine_RNAseq_melt)

# Plot densities
ggplot() +
    geom_density(data = matrix_rld_ATRT_primTum_Curie_RNAseq_Plus_melt,
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = matrix_rld_murine_RNAseq_melt,
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept=5, color = "red", lty = 3, lwd = 1) +
    theme_bw()

```

# Combine human and mouse data

```{r combine_matrix}

# Rename RNAseq_subgroup to group_name
sample_ATRT_primTum_Curie_RNAseq <- mutate(.data = sample_ATRT_primTum_Curie_RNAseq,
                                           group_name = RNAseq_subgroup
                                           )
sample_ATRT_primTum_Curie_RNAseq$RNAseq_subgroup <- NULL

# Bind sample annotations
sample_murine_human <- rbind(sample_ATRT_primTum_Curie_RNAseq,
                             sample_murine_RNAseq_plus
                             )

# Merge human and mouse matrix
matrix_human_mouse_rld <- merge(x = matrix_rld_ATRT_primTum_Curie_RNAseq_Plus,
                                y = matrix_rld_murine_RNAseq,
                                by = 0
                                )
rownames(matrix_human_mouse_rld) <- matrix_human_mouse_rld$Row.names
matrix_human_mouse_rld$Row.names <- NULL

head(matrix_human_mouse_rld)
ncol(matrix_human_mouse_rld)

```

# Distribution of combined matrix

```{r distribution}

# Melt matrix
matrix_human_mouse_rld_melt <- melt(matrix_human_mouse_rld)
colnames(matrix_human_mouse_rld_melt) <- c("Sample_name","rlogExp")
head(matrix_human_mouse_rld_melt)

# Add sample annotation to melted matrix
matrix_human_mouse_rld_melt_plus <- merge(x   = sample_murine_human,
                                          y   = matrix_human_mouse_rld_melt,
                                          by  = "Sample_name"
                                          )

# Plot densities
ggplot() +
    geom_density(data = matrix_human_mouse_rld_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_human_mouse_rld_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_human_mouse_rld_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept=5, color = "red", lty = 3, lwd = 1) +
    theme_bw()

```

# Filtering of combined matrix

```{r filtering}

# Melt matrix
matrix_human_mouse_rld_melt <- melt(matrix_human_mouse_rld)
colnames(matrix_human_mouse_rld_melt) <- c("Sample_name","rlogExp")
head(matrix_human_mouse_rld_melt)

# Plot densities
ggplot() +
    geom_density(data = matrix_human_mouse_rld_melt,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_vline(xintercept=5, color = "red", lty = 3, lwd = 1) +
    theme_bw()

# Filter matrix
matrix_human_mouse_rld_filt <- expFilter(data       = as.matrix(matrix_human_mouse_rld),
                                         threshold  = 5,
                                         p          = 0.01,
                                         graph      = FALSE
                                         )
nrow(matrix_human_mouse_rld_filt)

# Melt matrix
matrix_human_mouse_rld_filt_melt <- melt(matrix_human_mouse_rld_filt)
colnames(matrix_human_mouse_rld_filt_melt) <- c("symbol","Sample_name","rlogExp")
head(matrix_human_mouse_rld_filt_melt)

# Add sample annotation to melted matrix
matrix_human_mouse_rld_filt_melt_plus <- merge(x   = sample_murine_human,
                                               y   = matrix_human_mouse_rld_filt_melt,
                                               by  = "Sample_name"
                                               )

# Plot densities
ggplot() +
    geom_density(data = matrix_human_mouse_rld_filt_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(.data = matrix_human_mouse_rld_filt_melt_plus,
                               Organism == "Homo_sapiens"
                               ),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(.data = matrix_human_mouse_rld_filt_melt_plus,
                               Organism == "Mus_musculus"
                               ),
                 mapping = aes(x = rlogExp),
                 color = "red"
                 ) +
    geom_vline(xintercept=5, color = "red", lty = 3, lwd = 1) +
    theme_bw()

```

# Correlate human and murine data

## Sample-wise correlation

```{r correlation}

# Check order
colnames(matrix_human_mouse_rld_filt) == sample_murine_human$Sample_name

matrix_human_mouse_rld_filt <- matrix_human_mouse_rld_filt[,sample_murine_human$Sample_name]

colnames(matrix_human_mouse_rld_filt) == sample_murine_human$Sample_name

# Correlation matrix
matrix_human_mouse_rld_filt_cor <- cor(x       = matrix_human_mouse_rld_filt,
                                       method  = "pearson"
                                       )

head(matrix_human_mouse_rld_filt_cor)

# Scale matrix
matrix_human_mouse_rld_filt_cor <- scale(matrix_human_mouse_rld_filt_cor)

matrix_human_mouse_rld_filt_cor <- scale(t(matrix_human_mouse_rld_filt_cor))

# Check order
colnames(matrix_human_mouse_rld_filt_cor) == sample_murine_human$Sample_name
rownames(matrix_human_mouse_rld_filt_cor) == sample_murine_human$Sample_name

# Subset matrix
matrix_human_mouse_rld_filt_cor_min <- matrix_human_mouse_rld_filt_cor[filter(sample_murine_human, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_murine_human, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Row annotation (human)
row_annot_murine_human <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_murine_human[sample_murine_human$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"],
                                                            RNAseq_subgroup = sample_murine_human[sample_murine_human$Organism == "Homo_sapiens","group_name"]
                                                            ), 
                                            col = list(anatomic_location = sub_sub_localization_of_primary_tumor_col,
                                                       RNAseq_subgroup = group_name_col
                                                       ),
                                            gap = unit(2,"mm"),
                                            show_annotation_name = TRUE,
                                            annotation_name_side = "bottom",
                                            annotation_name_gp = gpar(fontsize = 20, fontface = "bold"),
                                            which = "row",
                                            show_legend = c(FALSE,FALSE)
                                            )

# Bottom annotation (mouse)
bottom_annot_murine_human <- HeatmapAnnotation(df = data.frame(Mouse_subgroup = filter(sample_murine_human, Organism == "Mus_musculus")$group_name), 
                                               col = list(Mouse_subgroup = group_name_col),
                                               gap = unit(2,"mm"),
                                               show_annotation_name = FALSE,
                                               show_legend = FALSE
                                               )

# Heatmap
h <- Heatmap(matrix               = matrix_human_mouse_rld_filt_cor_min,
             cluster_rows         = TRUE,
             cluster_columns      = FALSE,
             bottom_annotation    = bottom_annot_murine_human,
             show_row_names       = FALSE,
             show_column_names    = FALSE,
             row_order            = order(filter(sample_murine_human, Organism == "Homo_sapiens")$group_name),
             column_order         = order(filter(sample_murine_human, Organism == "Mus_musculus")$group_name),
             row_split            = filter(sample_murine_human, Organism == "Homo_sapiens")$group_name,
             column_split         = filter(sample_murine_human, Organism == "Mus_musculus")$group_name,
             show_heatmap_legend  = FALSE,
             cluster_row_slices   = FALSE,
             column_title_gp = gpar(fontsize = 15, fontface="bold"),
             row_title_gp = gpar(fontsize = 20, fontface="bold")
             )
#pdf(file = paste0(pathToFigures, "complexHeatmap_murineRT_humanATRT_RNAseq_corr.pdf"), width=18, height=8)
draw(h + row_annot_murine_human)
#dev.off()

# Legend of correlation score
max <- max(matrix_human_mouse_rld_filt_cor_min)
min <- min(matrix_human_mouse_rld_filt_cor_min)
mid <- (max+min)/2

col_fun = colorRamp2(breaks = c(min,mid,max), colors = c("blue","white","red"))

legend_score <- Legend(col_fun = col_fun,
                       title = "Correlation score \n(based on Pearson's coeff.)\n",
                       grid_height    = unit(1,"cm"),
                       legend_width   = unit(10,"cm"),
                       title_gp       = gpar(fontsize = 20),
                       title_position = "topcenter",
                       direction = "horizontal",
                       labels_gp = gpar(fontsize=15)
                       )
draw(legend_score)

# Legend anatomic location
legend_location <- Legend(labels = levels(factor(sample_murine_human[sample_murine_human$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"])),
                          legend_gp = gpar(fill = sub_sub_localization_of_primary_tumor_col[levels(factor(sample_murine_human[sample_murine_human$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"]))],
                                            size = 20
                                            ),
                          title = "anatomic location\n",
                          direction = "horizontal",
                          nrow = 3,
                          title_gp       = gpar(fontsize = 20),
                          title_position = "topcenter",
                          labels_gp = gpar(fontsize=15),
                          grid_height = unit(1,"cm"),
                          grid_width = unit(1,"cm"),
                          gap = unit(2,"cm")
                          )
draw(legend_location)

# Combine legends
legends <- packLegend(list = list(legend_score,legend_location),
                      direction="horizontal",
                      column_gap = unit(2,"cm")
                      )

#pdf(file = paste0(pathToFigures,"legend_heatmap_correlation_score_mouseBulkRNAseq_ATRTbulkRNAseq.pdf"), width = 20, height =10) 
draw(legends)
#dev.off()

```

# Correlation study between human ATRT (RNA-seq) and mouse RNA-seq data

## Human ATRT 

```{r human_ATRT_RNAseq}

# Sample plan
head(sample_ATRT_primTum_Curie_RNAseq)
nrow(sample_ATRT_primTum_Curie_RNAseq)

# Martix (rld)
head(matrix_rld_ATRT_primTum_Curie_RNAseq)

```

## Murine RT

### Import Curie and published (Vitte et al., 2017) mouse RT data

```{r murine_RT_RNAseq}

# Load murine data
load("mouse_Curie_Vitte2017_RNAseq_list.RData")

str(mouse_Curie_Vitte2017_RNAseq_list, max.level=1)

# Extract sample plan
sample_Curie_Vitte2017 <- mouse_Curie_Vitte2017_RNAseq_list$sampleTable

head(sample_Curie_Vitte2017)
nrow(sample_Curie_Vitte2017)

sample_Curie_Vitte2017_minus <- sample_Curie_Vitte2017
sample_Curie_Vitte2017_minus$group_name <- NULL

# Add subgroup assignment for R26 in situ
sample_Curie_Vitte2017_plus <- merge(x      = sample_Curie_Vitte2017_minus,
                                     y      = dplyr::select(.data = sample_murine_RNAseq_plus,
                                                            Sample_name,
                                                            group_name
                                                            ),
                                     by     = "Sample_name",
                                     all.x  = TRUE,
                                     all.y  = FALSE
                                     )

colnames(sample_Curie_Vitte2017_plus)

sample_Curie_Vitte2017_plus[,c("Background_genotype","Genotype","group_name")]

# Extract rlog matrix
matrix_rld_Curie_Vitte2017 <- mouse_Curie_Vitte2017_RNAseq_list$matrix_rld

head(matrix_rld_Curie_Vitte2017)
dim(matrix_rld_Curie_Vitte2017)

# Check order
colnames(matrix_rld_Curie_Vitte2017) == sample_Curie_Vitte2017_plus$Sample_name

matrix_rld_Curie_Vitte2017 <- matrix_rld_Curie_Vitte2017[,c(sample_Curie_Vitte2017_plus$Sample_name)]

colnames(matrix_rld_Curie_Vitte2017) == sample_Curie_Vitte2017_plus$Sample_name

```

### Convert mouse gene ID into human gene name

```{r murine_RT_RNAseq}

head(matrix_rld_Curie_Vitte2017[,1:4])

head(humanGeneName_mouseGeneName)

# Convert mouse gene name into human gene name
matrix_rld_Curie_Vitte2017_plus <- merge(x     = matrix_rld_Curie_Vitte2017,
                                        y     = humanGeneName_mouseGeneName,
                                        by.x  = 0,
                                        by.y  = "MGI.symbol",
                                        all   = FALSE
                                        )
head(matrix_rld_Curie_Vitte2017_plus)

# Remove duplicated symbol
matrix_rld_Curie_Vitte2017_plus <- matrix_rld_Curie_Vitte2017_plus[!duplicated(matrix_rld_Curie_Vitte2017_plus$HGNC.symbol),]

# Put human symbol as rownames
rownames(matrix_rld_Curie_Vitte2017_plus) <- matrix_rld_Curie_Vitte2017_plus$HGNC.symbol

head(matrix_rld_Curie_Vitte2017_plus)
dim(matrix_rld_Curie_Vitte2017_plus)

# Remove extra-columns
matrix_rld_Curie_Vitte2017_plus$Row.names <- NULL
matrix_rld_Curie_Vitte2017_plus$HGNC.symbol <- NULL

head(matrix_rld_Curie_Vitte2017_plus)
dim(matrix_rld_Curie_Vitte2017_plus)

```

# Combine human and mouse data

```{r combine_matrix}

head(sample_ATRT_primTum_Curie_RNAseq)
colnames(sample_ATRT_primTum_Curie_RNAseq)

#sample_ATRT_primTum_Curie_RNAseq_plus <- mutate(sample_ATRT_primTum_Curie_RNAseq,
 #                                               RNAseq_subgroup = group_name,
  #                                              .keep = "unused"
   #                                             )

#colnames(sample_ATRT_primTum_Curie_RNAseq_plus)

head(sample_Curie_Vitte2017_plus)
colnames(sample_Curie_Vitte2017_plus)

# Bind sample annotations
sample_humanATRT_murine <- rbind(sample_ATRT_primTum_Curie_RNAseq,
                                 sample_Curie_Vitte2017_plus
                                 )

head(sample_humanATRT_murine)

head(matrix_rawCounts_ATRT_primTum_Curie_RNAseq_WZ)
nrow(matrix_rawCounts_ATRT_primTum_Curie_RNAseq_WZ)

# Merge human and mouse matrix
matrix_rld_humanATRT_murine <- merge(x   = matrix_rld_ATRT_primTum_Curie_RNAseq,
                                     y   = matrix_rld_Curie_Vitte2017_plus,
                                     by  = 0
                                     )
head(matrix_rld_humanATRT_murine)

rownames(matrix_rld_humanATRT_murine) <- matrix_rld_humanATRT_murine$Row.names
matrix_rld_humanATRT_murine$Row.names <- NULL

head(matrix_rld_humanATRT_murine)
dim(matrix_rld_humanATRT_murine)

# Add DKFZ_web, methArray, etc subgroups
#sample_Curie_ATRT_primTum_PLUS2 <- read_delim(file = paste0("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/hu#man_ATRT/svn/analyse/report/sample_Curie_ATRT_primTum_PLUS2.csv"),
  #                                            delim = ";"
   #                                           )

sample_Curie_ATRT_primTum_groupedLoc <- read_tsv(file = paste0("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/sample_Curie_ATRT_primTum_groupedLoc.txt")
                                                 )
colnames(sample_Curie_ATRT_primTum_groupedLoc)


colnames(sample_humanATRT_murine)

sample_humanATRT_murine_plus <- merge(x = sample_humanATRT_murine,
                                      y = sample_Curie_ATRT_primTum_groupedLoc[,c("Sample_name","affy_subgroup","methArray_pearson","methArray_euclidean","DKFZ_web_subgroup","DKFZ_web_score","groupedLoc")],
                                      by = "Sample_name",
                                      all.x = TRUE,
                                      all.y = FALSE
                                      )

# Check order
colnames(matrix_rld_humanATRT_murine) == sample_humanATRT_murine_plus$Sample_name

matrix_rld_humanATRT_murine <- matrix_rld_humanATRT_murine[,sample_humanATRT_murine_plus$Sample_name]

colnames(matrix_rld_humanATRT_murine) == sample_humanATRT_murine_plus$Sample_name

```

# Distribution of combined matrix

```{r distribution}

# Melt matrix
matrix_rld_humanATRT_murine_melt <- melt(as.matrix(matrix_rld_humanATRT_murine))
colnames(matrix_rld_humanATRT_murine_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_humanATRT_murine_melt)

# Add sample annotation to melted matrix
matrix_rld_humanATRT_murine_melt_plus <- merge(x   = sample_humanATRT_murine,
                                               y   = matrix_rld_humanATRT_murine_melt,
                                               by  = "Sample_name"
                                               )

thrs <- 6

# Plot densities
ggplot() +
    geom_density(data = matrix_humanATRT_murine_melt_plus,
                 mapping = aes(x = rlogExp),
                 color = "red"
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()

```

# Filtering of combined matrix

```{r filtering}


expFilter <- function (data, threshold = 3.5, p = 0.01, graph = TRUE)
{
    if (graph) {
        par(font.lab = 2)
        hist(data, breaks = 200, main = "Data distribution",
             xlab = "Expression Level")
        abline(v = threshold, col = 2)
        mtext(threshold, at = threshold, side = 1, las = 1, col = 2,
              cex = 0.6)
    }
    vecL <- apply(data, 1, function(x) {
        length(which(x > threshold))
    })
    cat("Keep probes with at least", ceiling(ncol(data) * p),
        "sample(s) with an expression level higher than", threshold,
        "\n")
    data <- data[which(vecL >= ceiling(ncol(data) * p)), ]
}


# Filter matrix
matrix_rld_humanATRT_murine_filt <- expFilter(data       = as.matrix(matrix_rld_humanATRT_murine),
                                              threshold  = thrs,
                                              p          = 0.25,
                                              graph      = FALSE
                                              )
nrow(matrix_rld_humanATRT_murine_filt)

# Melt matrix
matrix_rld_humanATRT_murine_filt_melt <- melt(matrix_rld_humanATRT_murine_filt)
colnames(matrix_rld_humanATRT_murine_filt_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_humanATRT_murine_filt_melt)

# Add sample annotation to melted matrix
matrix_rld_humanATRT_murine_filt_melt_plus <- merge(x   = sample_humanATRT_murine_plus,
                                                    y   = matrix_rld_humanATRT_murine_filt_melt,
                                                    by  = "Sample_name"
                                                    )

# Plot densities
ggplot() +
    geom_density(data = matrix_rld_humanATRT_murine_filt_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_humanATRT_murine_filt_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_humanATRT_murine_filt_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()

```

### Sample-wise correlation

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_rld_humanATRT_murine) == sample_humanATRT_murine_plus$Sample_name

# Correlation matrix
matrix_rld_humanATRT_murine_cor <- cor(x       = matrix_rld_humanATRT_murine_filt,
                                       method  = "pearson"
                                       )
head(matrix_rld_humanATRT_murine_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_cor_min <- matrix_rld_humanATRT_murine_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Top annotation (mouse)
top_annot_humanATRT_murine <- HeatmapAnnotation(df                    = data.frame(Tumor_origin = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Tumor_origin,
                                                                                   RNAseq_subgroup = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$group_name,
                                                                                   Background_genotype = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Background_genotype
                                                                                   ), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col,
                                                                             RNAseq_subgroup = group_name_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Row annotation (human)
row_annot_humanATRT_murine <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_humanATRT_murine_plus[sample_humanATRT_murine_plus$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"],
                                                                RNAseq_subgroup = sample_humanATRT_murine_plus[sample_humanATRT_murine_plus$Organism == "Homo_sapiens","group_name"],
                                                                methArray_subgroup = sample_humanATRT_murine_plus[sample_humanATRT_murine_plus$Organism == "Homo_sapiens","DKFZ_web_subgroup"]
                                                                ), 
                                                col = list(anatomic_location = sub_sub_localization_of_primary_tumor_col,
                                                           RNAseq_subgroup = group_name_col,
                                                           methArray_subgroup = group_name_col
                                                           ),
                                                gap = unit(2,"mm"),
                                                show_annotation_name = TRUE,
                                                annotation_name_side = "bottom",
                                                annotation_name_gp = gpar(fontsize = 20, fontface = "bold"),
                                                which = "row",
                                                show_legend = c(FALSE,FALSE,FALSE)
                                                )

# Bottom annotation (mouse)
bottom_annot_humanATRT_murine <- HeatmapAnnotation(df                    = data.frame(Mouse_subgroup = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype), 
                                                   col                   = list(Mouse_subgroup = genotype_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = FALSE,
                                                   show_legend           = FALSE
                                                   )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_humanATRT_murine) == sample_humanATRT_murine_plus$Sample_name

# Most variable genes
mvGenes <- genes.selection(data       = matrix_rld_humanATRT_murine,
                           thres.num  = 1000,
                           probs      = 0.25
                           )

matrix_rld_humanATRT_murine_mvgenes <- matrix_rld_humanATRT_murine[as.vector(mvGenes),]

# Correlation matrix
matrix_rld_humanATRT_murine_mvgenes_cor <- cor(x       = matrix_rld_humanATRT_murine_mvgenes,
                                               method  = "pearson"
                                               )
head(matrix_rld_humanATRT_murine_mvgenes_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_mvgenes_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_mvgenes_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_mvgenes_cor_min <- matrix_rld_humanATRT_murine_mvgenes_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_mvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_humanATRT_murine) == sample_humanATRT_murine_plus$Sample_name

# Most variable genes
human_mvGenes<- genes.selection(data       = matrix_rld_ATRT_primTum_Curie_RNAseq,
                                thres.num  = 1500,
                                probs      = 0.25
                                )

matrix_rld_humanATRT_murine_humanMvGenes <- matrix_rld_humanATRT_murine[rownames(matrix_rld_humanATRT_murine) %in% as.vector(human_mvGenes),]
nrow(matrix_rld_humanATRT_murine_humanMvGenes)

# Correlation matrix
matrix_rld_humanATRT_murine_humanMvGenes_cor <- cor(x       = matrix_rld_humanATRT_murine_humanMvGenes,
                                                    method  = "pearson"
                                                    )
head(matrix_rld_humanATRT_murine_humanMvGenes_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_humanMvGenes_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_humanMvGenes_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_humanMvGenes_cor_min <- matrix_rld_humanATRT_murine_humanMvGenes_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_humanMvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Import subgroup location top DEG
groupedLoc_topGenes <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/groupedLoc_topGenes_df.txt",
                                  header = TRUE
                                  )
head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

matrix_rld_humanATRT_murine_groupedLocDEG <- matrix_rld_humanATRT_murine[rownames(matrix_rld_humanATRT_murine) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_rld_humanATRT_murine_groupedLocDEG)

# Correlation matrix
matrix_rld_humanATRT_murine_groupedLocDEG_cor <- cor(x       = matrix_rld_humanATRT_murine_groupedLocDEG,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_humanATRT_murine_groupedLocDEG_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_groupedLocDEG_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_groupedLocDEG_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_groupedLocDEG_cor_min <- matrix_rld_humanATRT_murine_groupedLocDEG_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_groupedLocDEG_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

#### Using or not immunologic signature genes

Immune genes are selected using the GO:BP gene sets collection. All gene sets having IMMUN term in their name are considered to immune gene sets. All genes belonging to these gene sets are here considered to immune genes and will be removed for the subsequent analysis.


##### Select immune genes

```{r correlation}

library("msigdbr")

msigdbr(species = "Homo sapiens") %>%
    dplyr::filter(gs_cat == "C5",
                  gs_subcat == "GO:BP",
                  grepl(pattern = "*IMMUN*",gs_name)
                  ) %>%
    dplyr::select(gene_symbol) -> immuneGenes

immuneGenes <- unique(as.vector(immuneGenes$gene_symbol))

head(immuneGenes)
length(immuneGenes)

```

##### Using immune genes

```{r correlation}

# Subset matrix
matrix_rld_humanATRT_murine_immuneGenes <- matrix_rld_humanATRT_murine[rownames(matrix_rld_humanATRT_murine) %in% as.vector(immuneGenes),]
nrow(matrix_rld_humanATRT_murine_immuneGenes)

# Correlation matrix
matrix_rld_humanATRT_murine_immuneGenes_cor <- cor(x       = matrix_rld_humanATRT_murine_immuneGenes,
                                                   method  = "pearson"
                                                   )
head(matrix_rld_humanATRT_murine_immuneGenes_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_immuneGenes_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_immuneGenes_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_immuneGenes_cor_min <- matrix_rld_humanATRT_murine_immuneGenes_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_immuneGenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_immuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

##### All genes but immune genes

```{r correlation}

# Subset matrix
matrix_rld_humanATRT_murine_noImmuneGenes <- matrix_rld_humanATRT_murine[!(rownames(matrix_rld_humanATRT_murine) %in% as.vector(immuneGenes)),]
nrow(matrix_rld_humanATRT_murine_noImmuneGenes)

# Correlation matrix
matrix_rld_humanATRT_murine_noImmuneGenes_cor <- cor(x       = matrix_rld_humanATRT_murine_noImmuneGenes,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_humanATRT_murine_noImmuneGenes_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_noImmuneGenes_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_noImmuneGenes_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_noImmuneGenes_cor_min <- matrix_rld_humanATRT_murine_noImmuneGenes_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_noImmuneGenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_noImmuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```

##### Most variable genes (after removing immune genes)

```{r correlation}

# Most variable genes
noImmuneGenes_mvGenes <- genes.selection(data       = matrix_rld_humanATRT_murine_noImmuneGenes,
                                         thres.num  = 1000,
                                         probs      = 0.25
                                         )

# Subset matrix
matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes <- matrix_rld_humanATRT_murine_noImmuneGenes[rownames(matrix_rld_humanATRT_murine_noImmuneGenes) %in% noImmuneGenes_mvGenes,]

# Correlation matrix
matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor <- cor(x       = matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes,
                                                             method  = "pearson"
                                                             )
head(matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor)

# Check order
colnames(matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor) == sample_humanATRT_murine_plus$Sample_name
rownames(matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor) == sample_humanATRT_murine_plus$Sample_name

# Subset matrix
matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor_min <- matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor[filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_noImmuneGenes_mvGenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_humanATRT_murine_noImmuneGenes_mvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name),
        column_order                 = order(filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_plus, Organism == "Homo_sapiens")$group_name,
        column_split                 = filter(sample_humanATRT_murine_plus, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine,
        top_annotation               = top_annot_humanATRT_murine,
        bottom_annotation            = bottom_annot_humanATRT_murine
        )
#dev.off()

```



# Correlation study between human ATRT (RNA-seq) with known anatomical localization and all mouse RNA-seq data

## Select human ATRT with localization data

```{r human_ATRT_RNAseq}

nrow(sample_Curie_ATRT_primTum_groupedLoc)

# Keep only human samples with anatomic location
sample_humanATRT_groupedLoc <- filter(.data = sample_Curie_ATRT_primTum_groupedLoc,
                                      !is.na(groupedLoc)
                                      )
nrow(sample_humanATRT_groupedLoc)

# Extract human ATRT rlog matrix
str(human_ATRT_primTum_Curie_RNAseq_list, max.level = 1)

matrix_rawCount_ATRT_primTum_Curie_RNAseq <- human_ATRT_primTum_Curie_RNAseq_list$matrix_rawCount

head(matrix_rawCount_ATRT_primTum_Curie_RNAseq)
dim(matrix_rawCount_ATRT_primTum_Curie_RNAseq)

# Put gene symbol as rownames
matrix_rawCount_ATRT_primTum_Curie_RNAseq <- column_to_rownames(.data = matrix_rawCount_ATRT_primTum_Curie_RNAseq,
                                                                var = "symbol"
                                                                )

# Subset human ATRT samples with anatomic location matrix of raw count
matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc <- matrix_rawCount_ATRT_primTum_Curie_RNAseq[,sample_humanATRT_groupedLoc$Sample_name]

dim(matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc)

# Remove lines with 0 counts in all samples
matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc_WZ <- matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc[apply(matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc, 1, function(x) !all(x==0)),]
dim(matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc_WZ)

# Create DESeq object for each sample (dds)
dds_humanATRT_groupedLoc <- DESeqDataSetFromMatrix(countData = matrix_rawCounts_ATRT_primTum_Curie_RNAseq_groupedLoc,
                                                   colData   = sample_Curie_ATRT_primTum_groupedLoc_noNA,
                                                   design    = ~ 1
                                                   )


# Stabilize variance using rlog
rld_humanATRT_groupedLoc <- rlog(object = dds_humanATRT_groupedLoc,
                                 blind  = TRUE
                                 )
matrix_rld_humanATRT_groupedLoc <- assay(rld_humanATRT_groupedLoc)
head(matrix_rld_humanATRT_groupedLoc)

```

## Combine human ATRT (rlog) with murine RT (rlog)

```{r murine}

# Human sample plan
head(sample_humanATRT_groupedLoc)
colnames(sample_humanATRT_groupedLoc)

sample_humanATRT_groupedLoc$affy_subgroup <- NULL

sample_humanATRT_groupedLoc <- relocate(.data = sample_humanATRT_groupedLoc,
                                        Patient_ID,
                                        .before = Storage_dir1
                                        )

# Mouse sample plan
head(sample_Curie_Vitte2017_plus)
colnames(sample_Curie_Vitte2017_plus)

# Add subgroup columns to murine sample plan
sample_Curie_Vitte2017_Plus <- sample_Curie_Vitte2017_plus

sample_Curie_Vitte2017_Plus$methArray_pearson <- NA
sample_Curie_Vitte2017_Plus$methArray_euclidean <- NA              
sample_Curie_Vitte2017_Plus$DKFZ_web_subgroup <- NA            
sample_Curie_Vitte2017_Plus$DKFZ_web_score <- NA              
sample_Curie_Vitte2017_Plus$groupedLoc <- NA

head(sample_Curie_Vitte2017_Plus)
colnames(sample_Curie_Vitte2017_Plus)

sample_Curie_Vitte2017_Plus <- relocate(.data = sample_Curie_Vitte2017_Plus,
                                        Sample_name,
                                        .before = Sample_ID
                                        )

colnames(sample_humanATRT_groupedLoc) == colnames(sample_Curie_Vitte2017_Plus)

# Bind sample annotations
sample_murine_humanATRT_groupedLoc <- rbind(sample_humanATRT_groupedLoc,
                                            sample_Curie_Vitte2017_Plus
                                            )

head(sample_murine_humanATRT_groupedLoc)

# Merge human and mouse matrix
head(matrix_rld_humanATRT_groupedLoc)
dim(matrix_rld_humanATRT_groupedLoc)


head(matrix_rld_Curie_Vitte2017_plus)
dim(matrix_rld_Curie_Vitte2017_plus)

matrix_rld_murine_humanATRT_groupedLoc <- merge(x   = matrix_rld_humanATRT_groupedLoc,
                                                y   = matrix_rld_Curie_Vitte2017_plus,
                                                by  = 0
                                                )
head(matrix_rld_murine_humanATRT_groupedLoc)

matrix_rld_murine_humanATRT_groupedLoc <- column_to_rownames(.data = matrix_rld_murine_humanATRT_groupedLoc,
                                                             var = "Row.names"
                                                             )

head(matrix_rld_murine_humanATRT_groupedLoc)
dim(matrix_rld_murine_humanATRT_groupedLoc)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc) == sample_murine_humanATRT_groupedLoc$Sample_name

```

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc) == sample_murine_humanATRT_groupedLoc$Sample_name

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc,
                                                  method  = "pearson"
                                                  )
head(matrix_rld_murine_humanATRT_groupedLoc_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Top annotation (mouse)
top_annot_murine_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Tumor_origin = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Tumor_origin,
                                                                                              RNAseq_subgroup = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$RNAseq_subgroup,
                                                                                              Background_genotype = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Background_genotype
                                                                                              ), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col,
                                                                             RNAseq_subgroup = group_name_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Convert sample plan into data frame (otherwise HeatmapAnnotation does not work well)
sample_murine_humanATRT_groupedLoc <- data.frame(sample_murine_humanATRT_groupedLoc)

# Row annotation (human)
row_annot_murine_humanATRT_groupedLoc <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_murine_humanATRT_groupedLoc[sample_murine_humanATRT_groupedLoc$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"]
                                                                           ,RNAseq_subgroup = sample_murine_humanATRT_groupedLoc[sample_murine_humanATRT_groupedLoc$Organism == "Homo_sapiens","RNAseq_subgroup"]
                                                                           ,methArray_subgroup = sample_murine_humanATRT_groupedLoc[sample_murine_humanATRT_groupedLoc$Organism == "Homo_sapiens","DKFZ_web_subgroup"]
                                                              ,
                                                               check.names = FALSE
                                                           ), 
                                                           col                            = list(anatomic_location = sub_sub_localization_of_primary_tumor_col
                                                               ,RNAseq_subgroup     = group_name_col
                                                              ,methArray_subgroup  = group_name_col
                                                           ),
                                                           gap                   = unit(2,"mm"),
                                                           show_annotation_name  = TRUE,
                                                           annotation_name_side  = "bottom",
                                                           annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                           which                 = "row",
                                                           show_legend           = c(FALSE,FALSE,FALSE)
                                                           )

# Bottom annotation (mouse)
bottom_annot_murine_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Mouse_subgroup = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype), 
                                                   col                   = list(Mouse_subgroup = genotype_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = FALSE,
                                                   show_legend           = FALSE
                                                   )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc) == sample_murine_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine_humanATRT_groupedLoc_mvGenes <- genes.selection(data       = matrix_rld_murine_humanATRT_groupedLoc,
                                                       thres.num  = 1000,
                                                       probs      = 0.25
                                                       )

matrix_rld_murine_humanATRT_groupedLoc_mvgenes <- matrix_rld_murine_humanATRT_groupedLoc[as.vector(murine_humanATRT_groupedLoc_mvGenes),]

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_mvgenes,
                                                          method  = "pearson"
                                                          )
head(matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_mvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc) == sample_murine_humanATRT_groupedLoc$Sample_name

# Most variable genes
humanATRT_groupedLoc_mvGenes<- genes.selection(data       = matrix_rld_humanATRT_groupedLoc,
                                               thres.num  = 1500,
                                               probs      = 0.25
                                               )

matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes <- matrix_rld_murine_humanATRT_groupedLoc[rownames(matrix_rld_murine_humanATRT_groupedLoc) %in% as.vector(humanATRT_groupedLoc_mvGenes),]
nrow(matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes)

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes,
                                                               method  = "pearson"
                                                               )
head(matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_humanMvgenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

##### Using immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_immuneGenes <- matrix_rld_murine_humanATRT_groupedLoc[rownames(matrix_rld_murine_humanATRT_groupedLoc) %in% as.vector(immuneGenes),]
nrow(matrix_rld_murine_humanATRT_groupedLoc_immuneGenes)

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_immuneGenes,
                                                              method  = "pearson"
                                                              )
head(matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_immuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_immuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

##### All genes but immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes <- matrix_rld_murine_humanATRT_groupedLoc[!(rownames(matrix_rld_murine_humanATRT_groupedLoc) %in% as.vector(immuneGenes)),]
nrow(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes)

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes,
                                                   method  = "pearson"
                                                   )
head(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_noImmuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

##### Most variable genes (after removing immune genes)

```{r correlation}

# Most variable genes
humanATRT_groupedLoc_noImmuneGenes_mvGenes <- genes.selection(data       = matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes,
                                                              thres.num  = 1000,
                                                              probs      = 0.25
                                                              )

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes <- matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes[rownames(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes) %in% humanATRT_groupedLoc_noImmuneGenes_mvGenes,]

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes,
                                                                        method  = "pearson"
                                                                        )
head(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Import subgroup location top DEG
groupedLoc_topGenes <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/groupedLoc_topGenes_df.txt",
                                  header = TRUE
                                  )
head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG <- matrix_rld_murine_humanATRT_groupedLoc[rownames(matrix_rld_murine_humanATRT_groupedLoc) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG)

# Correlation matrix
matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor <- cor(x       = matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG,
                                                                method  = "pearson"
                                                                )
head(matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor)

# Check order
colnames(matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor_min <- matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor[filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine_humanATRT_groupedLoc_groupedLocDEG_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine_humanATRT_groupedLoc_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine_humanATRT_groupedLoc
        )
#dev.off()

```


# Correlation study between human ATRT (RNA-seq) with known anatomical localization and R26 mouse RNA-seq data

## Select mouse R26 data

```{r human_ATRT_RNAseq}

# Select mouse R26 sample
sample_Curie_R26 <- filter(.data         = sample_Curie_Vitte2017_Plus,
                           Genotype      == "R26",
                           Tumor_origin  == "primary_tumor"
                           )

nrow(sample_Curie_R26)

group_by(sample_Curie_R26,
         Genotype,
         group_name
         ) %>%
    summarize(n = n())

nrow(sample_Curie_ATRT_primTum_groupedLoc)

# Extract mouse raw count matrix
matrix_rawCounts_Curie_Vitte2017 <- mouse_Curie_Vitte2017_RNAseq_list$matrix_rawCounts

head(matrix_rawCounts_Curie_Vitte2017)

# Convert mouse gene name into human gene name
matrix_rawCounts_Curie_Vitte2017_plus <- merge(x     = matrix_rawCounts_Curie_Vitte2017,
                                               y     = humanGeneName_mouseGeneName,
                                               by.x  = 0,
                                               by.y  = "MGI.symbol",
                                               all   = FALSE
                                               )

head(matrix_rawCounts_Curie_Vitte2017_plus)

# Remove mouse gene name (Row.names column) and put human gene name into rownames
matrix_rawCounts_Curie_Vitte2017_plus$Row.names <- NULL
matrix_rawCounts_Curie_Vitte2017_plus <- matrix_rawCounts_Curie_Vitte2017_plus[!duplicated(matrix_rawCounts_Curie_Vitte2017_plus$HGNC.symbol),]
rownames(matrix_rawCounts_Curie_Vitte2017_plus) <- matrix_rawCounts_Curie_Vitte2017_plus$HGNC.symbol
matrix_rawCounts_Curie_Vitte2017_plus$HGNC.symbol <- NULL

# Extract mouse R26 raw count matrix
matrix_rawCounts_Curie_R26 <- matrix_rawCounts_Curie_Vitte2017_plus[,sample_Curie_R26$Sample_name]
dim(matrix_rawCounts_Curie_R26)

# Remove lines with 0 counts in all samples
matrix_rawCounts_Curie_R26_WZ <- matrix_rawCounts_Curie_R26[apply(matrix_rawCounts_Curie_R26, 1, function(x) !all(x==0)),]
dim(matrix_rawCounts_Curie_R26_WZ)

# Create DESeq object for each sample (dds)
dds_rawCounts_Curie_R26 <- DESeqDataSetFromMatrix(countData = matrix_rawCounts_Curie_R26_WZ,
                                                  colData   = sample_Curie_R26,
                                                  design    = ~ 1
                                                  )

# Stabilize variance using rlog
rld_Curie_R26 <- rlog(object = dds_rawCounts_Curie_R26,
                      blind  = TRUE
                      )
matrix_rld_Curie_R26 <- assay(rld_Curie_R26)
head(matrix_rld_Curie_R26)

```

## Combine human ATRT (rlog) with murine R26 RT (rlog)

```{r murine}

# Human sample plan
head(sample_humanATRT_groupedLoc)
colnames(sample_humanATRT_groupedLoc)

# Mouse sample plan
head(sample_Curie_R26)
colnames(sample_Curie_R26)

# Bind sample annotations
sample_murine26_humanATRT_groupedLoc <- rbind(sample_humanATRT_groupedLoc,
                                            sample_Curie_R26
                                            )
head(sample_murine26_humanATRT_groupedLoc)

# Merge human and mouse matrix
head(matrix_rld_humanATRT_groupedLoc)
dim(matrix_rld_humanATRT_groupedLoc)


head(matrix_rld_Curie_R26)
dim(matrix_rld_Curie_R26)

matrix_rld_murine26_humanATRT_groupedLoc <- merge(x   = matrix_rld_humanATRT_groupedLoc,
                                                  y   = matrix_rld_Curie_R26,
                                                  by  = 0
                                                  )
head(matrix_rld_murine26_humanATRT_groupedLoc)

matrix_rld_murine26_humanATRT_groupedLoc <- column_to_rownames(.data = matrix_rld_murine26_humanATRT_groupedLoc,
                                                               var = "Row.names"
                                                               )

head(matrix_rld_murine26_humanATRT_groupedLoc)
dim(matrix_rld_murine26_humanATRT_groupedLoc)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Melt matrix
matrix_rld_murine26_humanATRT_groupedLoc_melt <- melt(as.matrix(matrix_rld_murine26_humanATRT_groupedLoc))
colnames(matrix_rld_murine26_humanATRT_groupedLoc_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26_humanATRT_groupedLoc_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_melt_plus <- merge(x   = sample_murine26_humanATRT_groupedLoc,
                                                            y   = matrix_rld_murine26_humanATRT_groupedLoc_melt,
                                                            by  = "Sample_name"
                                                            )

# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26_humanATRT_groupedLoc.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26_humanATRT_groupedLoc_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()
#dev.off()

```

# Filtering of combined matrix

```{r filtering}

# Filter matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt <- expFilter(data       = as.matrix(matrix_rld_murine26_humanATRT_groupedLoc),
                                                           threshold  = thrs,
                                                           p          = 0.25,
                                                           graph      = TRUE
                                                           )
nrow(matrix_rld_murine26_humanATRT_groupedLoc_filt)

# Melt matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_filt)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26_humanATRT_groupedLoc_filt_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_melt_plus <- merge(x   = sample_murine26_humanATRT_groupedLoc,
                                                                 y   = matrix_rld_murine26_humanATRT_groupedLoc_filt_melt,
                                                                 by  = "Sample_name"
                                                                 )

# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26_humanATRT_groupedLoc_filt.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26_humanATRT_groupedLoc_filt_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_filt_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_filt_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()
#dev.off()

```

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc,
                                                  method  = "pearson"
                                                  )
head(matrix_rld_murine26_humanATRT_groupedLoc_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Top annotation (mouse)
top_annot_murine26_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(RNAseq_subgroup = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$RNAseq_subgroup), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col,
                                                                             RNAseq_subgroup = group_name_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Convert sample plan into data frame (otherwise HeatmapAnnotation does not work well)
sample_murine26_humanATRT_groupedLoc <- data.frame(sample_murine26_humanATRT_groupedLoc)

# Row annotation (human)
row_annot_murine26_humanATRT_groupedLoc <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_murine26_humanATRT_groupedLoc[sample_murine26_humanATRT_groupedLoc$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"]
                                                                           ,RNAseq_subgroup = sample_murine26_humanATRT_groupedLoc[sample_murine26_humanATRT_groupedLoc$Organism == "Homo_sapiens","RNAseq_subgroup"]
                                                                           ,methArray_subgroup = sample_murine26_humanATRT_groupedLoc[sample_murine26_humanATRT_groupedLoc$Organism == "Homo_sapiens","DKFZ_web_subgroup"]
                                                              ,
                                                               check.names = FALSE
                                                           ), 
                                                           col                            = list(anatomic_location = sub_sub_localization_of_primary_tumor_col
                                                               ,RNAseq_subgroup     = group_name_col
                                                              ,methArray_subgroup  = group_name_col
                                                           ),
                                                           gap                   = unit(2,"mm"),
                                                           show_annotation_name  = TRUE,
                                                           annotation_name_side  = "bottom",
                                                           annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                           which                 = "row",
                                                           show_legend           = c(FALSE,FALSE,FALSE)
                                                           )

# Bottom annotation (mouse)
bottom_annot_murine26_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Mouse_subgroup = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype), 
                                                   col                   = list(Mouse_subgroup = genotype_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = FALSE,
                                                   show_legend           = FALSE
                                                   )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

```{r figure_for_paper}

data.frame(subgroup = sample_murine26_humanATRT_groupedLoc_Plus[sample_murine26_humanATRT_groupedLoc_Plus$Organism == "Homo_sapiens","subgroup"],
           check.names = FALSE
           )

# Row annotation (human)
row_annot_murine26_humanATRT_groupedLoc_min <- HeatmapAnnotation(df = data.frame(subgroup = sample_murine26_humanATRT_groupedLoc_Plus[sample_murine26_humanATRT_groupedLoc_Plus$Organism == "Homo_sapiens","subgroup"],
                                                                                 check.names = FALSE
                                                                                 ), 
                                                                 col                            = list(subgroup  = group_name_col),
                                                                 gap                   = unit(2,"mm"),
                                                                 show_annotation_name  = FALSE,
                                                                 #annotation_name_side  = "bottom",
                                                                 #annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                                 which                 = "row",
                                                                 show_legend           = FALSE,
                                                                 )


# Heatmap
heatmap_murine26_humanATRT_cor <- Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_cor_min)))),
                                          cluster_rows                 = TRUE,
                                          clustering_distance_rows     = "euclidean",
                                          clustering_distance_columns  = "euclidean",
                                          clustering_method_rows       = "ward",
                                          clustering_method_columns    = "ward",
                                          cluster_columns              = TRUE,
                                          show_column_dend             = FALSE,
                                          show_row_dend                = FALSE,
                                          show_row_names               = FALSE,
                                          show_column_names            = FALSE,
                                          row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc_Plus, Organism == "Homo_sapiens")$groupedLoc),
                                          column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc_Plus, Organism == "Mus_musculus")$subgroup),
                                          row_split                    = filter(sample_murine26_humanATRT_groupedLoc_Plus, Organism == "Homo_sapiens")$subgroup,
                                          column_split                 = filter(sample_murine26_humanATRT_groupedLoc_Plus, Organism == "Mus_musculus")$subgroup,
                                          show_heatmap_legend          = TRUE,
                                          cluster_row_slices           = FALSE,
                                          cluster_column_slices        = FALSE,
                                          column_title_gp              = gpar(fontsize = paperFig_fontsize*1.5, fontface="bold"),
                                          row_title_gp                 = gpar(fontsize = paperFig_fontsize*1.5, fontface="bold"),
                                          heatmap_legend_param = list(fontsize    = paperFig_fontsize,
                                                                      title                        = "correlation\nscore",
                                                                      title_gp    = gpar(fontsize = paperFig_fontsize*1.5, font = 2),
                                                                      labels_gp   = gpar(fontsize = paperFig_fontsize),
                                                                      grid_width  = unit(3,"mm"),
                                                                      grid_height = unit(9,"mm"),
                                                                      ncol        = 1,
                                                                      side = "bottom",
                                                                      direction = "vertical"
                                                                      )#,
                                          #right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
                                          #top_annotation               = top_annot_murine26_humanATRT_groupedLoc#,
                                          #bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
                                          )

heatmap_murine26_humanATRT_cor_grab <- grid.grabExpr(draw(heatmap_murine26_humanATRT_cor,
                                                          heatmap_legend_side = "left"
                                                          )
                                                     )

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine_humanATRT_groupedLoc_mvGenes <- genes.selection(data       = matrix_rld_murine26_humanATRT_groupedLoc,
                                                       thres.num  = 1000,
                                                       probs      = 0.25
                                                       )

matrix_rld_murine26_humanATRT_groupedLoc_mvgenes <- matrix_rld_murine26_humanATRT_groupedLoc[as.vector(murine_humanATRT_groupedLoc_mvGenes),]

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_mvgenes,
                                                          method  = "pearson"
                                                          )
head(matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_mvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Most variable genes
humanATRT_groupedLoc_mvGenes<- genes.selection(data       = matrix_rld_humanATRT_groupedLoc,
                                               thres.num  = 1500,
                                               probs      = 0.25
                                               )

matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% as.vector(humanATRT_groupedLoc_mvGenes),]
nrow(matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes,
                                                                 method  = "pearson"
                                                                 )
head(matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_humanMvgenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

##### Using immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% as.vector(immuneGenes),]
nrow(matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes,
                                                                method  = "pearson"
                                                                )
head(matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_immuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_immuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

##### All genes but immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes <- matrix_rld_murine26_humanATRT_groupedLoc[!(rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% as.vector(immuneGenes)),]
nrow(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes,
                                                   method  = "pearson"
                                                   )
head(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_noImmuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

##### Most variable genes (after removing immune genes)

```{r correlation}

# Most variable genes
humanATRT_groupedLoc_noImmuneGenes_mvGenes <- genes.selection(data       = matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes,
                                                              thres.num  = 1000,
                                                              probs      = 0.25
                                                              )

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes <- matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes[rownames(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes) %in% humanATRT_groupedLoc_noImmuneGenes_mvGenes,]

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes,
                                                             method  = "pearson"
                                                             )
head(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Import subgroup location top DEG
groupedLoc_topGenes <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/groupedLoc_topGenes_df.txt",
                                  header = TRUE
                                  )
head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_groupedLocDEG_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using ganglionic eminnence genes

```{r correlation}

# Ganglionic eminence TFs
ganglionicEminenceGenes <- c("ARC","ARX","ASCL1","DLX1","DLX2","DLX5","DLX6","EMX2","FOXG1","GSX1","GSX2","LHX2","LHX6","NKX2-1","NKX6-2","OLIG1","OLIG2","OTX2","SOX6","STAB1","VAX1")

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% ganglionicEminenceGenes,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes,
                                                                            method  = "pearson"
                                                                            )
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor)
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt)

# Add R26-SHH and R26-MYCto sample plan
sample_murine26_humanATRT_groupedLoc_plus <- mutate(.data    = sample_murine26_humanATRT_groupedLoc,
                                                  subgroup = case_when(is.na(groupedLoc) ~ RNAseq_subgroup,
                                                                       TRUE ~ groupedLoc
                                                                       ),
                                                  subgroup = case_when(subgroup == "non_neuronal" ~ "R26-MYC",
                                                                       subgroup == "neuronal" ~ "R26-SHH",
                                                                       TRUE ~ subgroup
                                                                       )
                                                  )
head(sample_murine26_humanATRT_groupedLoc_plus)
sample_murine26_humanATRT_groupedLoc_plus$subgroup


# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

plot_avgCorr_heatmap <- function(matrix,
                                 show_heatmap_legend  = TRUE,
                                 draw                 = TRUE,
                                 column_names_rot     = 0,
                                 column_title         = NULL,
                                 column_names_side    = "top",
                                 fontsize             = 30,
                                 cell_fun             = NULL
                                 )
{
    h <- Heatmap(matrix                       = matrix,
                 clustering_distance_rows     = "euclidean",
                 clustering_distance_columns  = "euclidean",
                 clustering_method_rows       = "ward",
                 clustering_method_columns    = "ward",
                 cluster_columns              = FALSE,
                 cluster_rows                 = FALSE,
                 show_row_names               = TRUE,
                 show_column_names            = TRUE,
                 column_title                 = column_title,
                 column_title_gp              = gpar(fontsize = fontsize, fontface = "bold"),
                 column_dend_side             = "bottom",
                 column_names_gp              = gpar(fontsize = fontsize, fontface = "bold"),
                 column_names_side            = column_names_side,
                 column_names_rot             = column_names_rot,
                 row_names_gp                 = gpar(fontsize = fontsize, fontface = "bold"),
                 show_heatmap_legend          = show_heatmap_legend,
                 cluster_row_slices           = FALSE,
                 cluster_column_slices        = FALSE,
                 cell_fun                     = cell_fun,
                 heatmap_legend_param = list(title = "\nAverage Pearson's Correlation\n",
                                             direction = "horizontal",
                                             labels_gp = gpar(size = 20),
                                             title_gp = gpar(fontsize = 25, fontface = "bold"),
                                             legend_width= unit(20,"cm"),
                                             legend_height= unit(10,"cm")
                                             )
                 )
    if (draw)
    {
        draw(h, heatmap_legend_side = "bottom")
    }
    else
    {
        return(h)
    }
}

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_ganglionicEminenceGenes.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceGenes_meanCorr_min)
#dev.off()

```

#### Using ganglionic eminnence TFs

```{r correlation}

# Ganglionic eminence TFs
ganglionicEminenceTFs <- c("SP8","SP9","SATB1","NKX2.1","NKX6.2","DLX","LHX6","LHX7","ASCL1","GSX2","GSX1","NR2F2","VAX1","VAX2","SHH","SOX6")

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% ganglionicEminenceTFs,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs,
                                                                            method  = "pearson"
                                                                            )
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor)
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_ganglionicEminenceTFs.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_ganglionicEminenceTFs_meanCorr_min)
#dev.off()

```


#### Using midbrain hindbrain boundary genes

```{r correlation}

# Ganglionic eminence TFs
midbrainHindbrainBoundaryGenes <- c("EN1","EN2","FGF8","GBX2","HES3","IRX1","IRX2","LMX1B","OTX2","PAX2","PAX3","WNT1","WNT3A")

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% midbrainHindbrainBoundaryGenes,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes,
                                                                                   method  = "pearson"
                                                                                   )
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_humanATRT_groupedLoc
        )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor)
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_midbrainHindbrainBoundaryGenes.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_midbrainHindbrainBoundaryGenes_meanCorr_min)
#dev.off()

```



#### Using TF genes that are activated in R26-neuronal single cell clusters

##### Import SCENIC RSS table

```{r import_SCENIC_RSS_table}

# Import SCENIC RSS results
scRNAseq_integrated_R26_scenic_rss_matrix <- read.csv(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/scRNAseq/SCENIC_integrated_R26/scRNAseq_integrated_R26_scenic_rss_matrix.csv",
                                                      check.names = FALSE
                                                       )
head(scRNAseq_integrated_R26_scenic_rss_matrix)
nrow(scRNAseq_integrated_R26_scenic_rss_matrix)

# Transpose matrix
scRNAseq_integrated_R26_scenic_rss_matrix_t <- t(scRNAseq_integrated_R26_scenic_rss_matrix)
head(scRNAseq_integrated_R26_scenic_rss_matrix_t)

```

##### Using TF genes that are activated in all R26-neuronal single cell clusters

```{r allClustersTFs}

# Create list of TFs that are specific for each cluster
top <- 5
topTF_vect <- vector()
for (i in 1:ncol(scRNAseq_integrated_R26_scenic_rss_matrix_t))
{
    topTF <- scRNAseq_integrated_R26_scenic_rss_matrix_t[order(scRNAseq_integrated_R26_scenic_rss_matrix_t[,i], decreasing = TRUE),] %>%
        .[1:top,] %>%
        rownames(.)
    topTF_vect <- c(topTF_vect,topTF)
    #topTF_list[[length(topTF_list)+1]] <- topTF
}
topTF_vect <- unique(topTF_vect)

# Remove "(+)" TF suffix
topTF_vect <- gsub("\\(\\+\\)", "", topTF_vect)

topTF_vect

# Convert mouse symbol to human symbol
topTF_vect_humanSymbol <- filter(humanGeneName_mouseGeneName,
                                 MGI.symbol %in% topTF_vect
                                 )$HGNC.symbol
topTF_vect_humanSymbol

matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% topTF_vect_humanSymbol,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Sample-wise correlation heatmap function
plot_sample_wise_corr_heatmap <- function(matrix,
                                          samplePlan,
                                          right_annotation,
                                          top_annotation,
                                          bottom_annotation
                                          )
{
    h <- Heatmap(matrix                       = t(scale(t(scale(matrix)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(samplePlan, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(samplePlan, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(samplePlan, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(samplePlan, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = right_annotation,
        top_annotation               = top_annotation,
        bottom_annotation            = bottom_annotation
        )
    draw(h)
}

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_topTFclusters_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()


```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor)
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt)

# Add R26-SHH and R26-MYCto sample plan
sample_murine26_humanATRT_groupedLoc_plus <- mutate(.data    = sample_murine26_humanATRT_groupedLoc,
                                                  subgroup = case_when(is.na(groupedLoc) ~ RNAseq_subgroup,
                                                                       TRUE ~ groupedLoc
                                                                       ),
                                                  subgroup = case_when(subgroup == "non_neuronal" ~ "R26-MYC",
                                                                       subgroup == "neuronal" ~ "R26-SHH",
                                                                       TRUE ~ subgroup
                                                                       )
                                                  )
head(sample_murine26_humanATRT_groupedLoc_plus)
sample_murine26_humanATRT_groupedLoc_plus$subgroup


# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_topTFclusters.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_topTFclusters_meanCorr_min)
#dev.off()

```

##### Using TF genes that are activated in R26-neuronal single cell cluster 1,2,3,4,5,6,7,9

```{r correlation}

# Choose clusters
clusters <- c(1,2,3,4,5,6,7,9)

# Create list of TFs that are specific for each cluster
top <- 5
topTF_clust12345679_vect <- vector()
for (i in clusters)
{
    topTF <- scRNAseq_integrated_R26_scenic_rss_matrix_t[order(scRNAseq_integrated_R26_scenic_rss_matrix_t[,i], decreasing = TRUE),] %>%
        .[1:top,clusters] %>%
        rownames(.)
    topTF_clust12345679_vect <- c(topTF_clust12345679_vect,topTF)
}
topTF_clust12345679_vect <- unique(topTF_clust12345679_vect)

# Remove "(+)" TF suffix
topTF_clust12345679_vect <- gsub("\\(\\+\\)", "", topTF_clust12345679_vect)
topTF_clust12345679_vect

# Convert mouse symbol to human symbol
topTF_clust12345679_vect_humanSymbol <- filter(humanGeneName_mouseGeneName,
                                               MGI.symbol %in% topTF_clust12345679_vect
                                               )$HGNC.symbol
topTF_clust12345679_vect_humanSymbol

matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679 <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% topTF_clust12345679_vect_humanSymbol,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_topTFclust12345679_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor)
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_topTFclust12345679.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust12345679_meanCorr_min)
#dev.off()

```

##### Using TF genes that are activated in R26-neuronal single cell cluster 6

```{r correlation}

head(scRNAseq_integrated_R26_scenic_rss_matrix_t[order(scRNAseq_integrated_R26_scenic_rss_matrix_t[,"6"], decreasing = TRUE),])

# Extract TF specific to cluster 6
topTF_clust6_vect <- scRNAseq_integrated_R26_scenic_rss_matrix_t[order(scRNAseq_integrated_R26_scenic_rss_matrix_t[,"6"], decreasing = TRUE),] %>% .[1:50,] %>% rownames(.)
head(topTF_clust6_vect)

# Remove "(+)" TF suffix
topTF_clust6_vect <- gsub("\\(\\+\\)", "", topTF_clust6_vect)
topTF_clust6_vect

# Convert mouse symbol to human symbol
topTF_clust6_vect_humanSymbol <- filter(humanGeneName_mouseGeneName,
                                        MGI.symbol %in% topTF_clust6_vect
                                        )$HGNC.symbol
topTF_clust6_vect_humanSymbol

matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6 <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% topTF_clust6_vect_humanSymbol,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_topTFclust6_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor)
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_topTFclust6.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_topTFclust6_meanCorr_min)
#dev.off()

```

```{r group_wise_correlation_per_cluster}

# Extract TF specific for every cluster
top <- 50
topTF_list <- list()
for (i in colnames(scRNAseq_integrated_R26_scenic_rss_matrix_t))
{
    # Order according to RSS
    topTF <- scRNAseq_integrated_R26_scenic_rss_matrix_t[order(scRNAseq_integrated_R26_scenic_rss_matrix_t[,i],
                                                               decreasing = TRUE),] %>%
        .[1:top,] %>%
        rownames(.)
    # Remove (+)
    topTF <- gsub("\\(\\+\\)", "", topTF)
    # Convert mouse symbol to human symbol
    topTF <- filter(humanGeneName_mouseGeneName,
                    MGI.symbol %in% topTF
                    )$HGNC.symbol
    # Append list
    topTF_list[[length(topTF_list)+1]] <- topTF
}
names(topTF_list) <- colnames(scRNAseq_integrated_R26_scenic_rss_matrix_t)
topTF_list

# Construct matrix of top TFs for each cluster
matrix_topTF_list <- list()
for (list in topTF_list)
{
    matrix_topTF <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% list,]
    matrix_topTF_list[[length(matrix_topTF_list)+1]] <- matrix_topTF
}
names(matrix_topTF_list) <- colnames(scRNAseq_integrated_R26_scenic_rss_matrix_t)

# Compute correlation for each top TFs matrix
matrix_topTF_corr_list <- list()
for (matrix in matrix_topTF_list)
{
    # Compute correlation
    matrix_topTF_corr <- cor(x      = matrix,
                             method = "pearson"
                             )
    matrix_topTF_corr_list[[length(matrix_topTF_corr_list)+1]] <- matrix_topTF_corr
}
names(matrix_topTF_corr_list) <- names(matrix_topTF_list)
head(matrix_topTF_corr_list[["0"]])


# Add subgroup info and summarize correlation by mean by subgroup
matrix_topTF_corr_mean_list <- list()
for (matrix in matrix_topTF_corr_list)
{
    # Melt correlation matrix
    matrix_corr_melt <- melt(matrix)
    colnames(matrix_corr_melt) <- c("Sample_name_1","Sample_name_2","correlation")
    # Add subgroup to melted matrix
    matrix_corr_melt_plus <- merge(x = matrix_corr_melt,
                                   y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                   by.x = "Sample_name_1",
                                   by.y = "Sample_name"
                                   )
    colnames(matrix_corr_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
    #
    matrix_corr_melt_Plus <- merge(x = matrix_corr_melt_plus,
                                   y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                   by.x = "Sample_name_2",
                                   by.y = "Sample_name"
                                   )
    colnames(matrix_corr_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")  
    # Summarize by mean
    matrix_corr_meanCorr <- group_by(.data = matrix_corr_melt_Plus,
                                          subgroup_1,
                                          subgroup_2
                                          ) %>%
    summarize(mean_corr = mean(correlation)
              )
    # Spread table
    matrix_corr_meanCorr <- spread(data = matrix_corr_meanCorr,
                                   key = subgroup_2,
                                   value = mean_corr
                                   ) %>%
    data.frame(check.names = FALSE)
    rownames(matrix_corr_meanCorr) <- matrix_corr_meanCorr$subgroup_1
    matrix_corr_meanCorr$subgroup_1 <- NULL
    # Subset matrix
    matrix_corr_meanCorr_min <- matrix_corr_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]
    # Rename rownames
    rownames(matrix_corr_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")
    # Append list
    matrix_topTF_corr_mean_list[[length(matrix_topTF_corr_mean_list)+1]] <- matrix_corr_meanCorr_min
}
names(matrix_topTF_corr_mean_list) <- names(matrix_topTF_corr_list)

# Plot average correlation
plot_topTF_corr_mean_list <- HeatmapList()
for (k in 1:length(matrix_topTF_corr_mean_list))
{
    p <- plot_avgCorr_heatmap(matrix_topTF_corr_mean_list[[k]],
                              draw                 = FALSE,
                              show_heatmap_legend  = FALSE,
                              column_names_rot     = 90,
                              column_names_side    = "bottom",
                              column_title         = names(matrix_topTF_corr_mean_list[k]),
                              fontsize             = 20,
                              cell_fun             = function(j,i,x,y,width,height,fill)
                              {
                                  grid.text(label = sprintf("%.2f",matrix_topTF_corr_mean_list[[k]][i,j]),
                                            x,
                                            y,
                                            gp = gpar(fontsize = 10)
                                            )
                              }
                              )
    plot_topTF_corr_mean_list <- plot_topTF_corr_mean_list + p
}
#names(plot_topTF_corr_mean_list) <- names(matrix_topTF_corr_mean_list)

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_topTF_clusters.pdf"), width=24, height=6)
draw(plot_topTF_corr_mean_list)
#dev.off()

```


#### Using TF genes that are activated in CAL-ATRT single cell clusters

##### Import SCENIC RSS table

```{r import_SCENIC_RSS_table}

# Import SCENIC RSS results
scRNAseq_integrated_CAL_scenic_rss_matrix <- read.csv(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/SCENIC_integrated/scRNAseq_integrated_scenic_rss_matrix.csv",
                                                      check.names = FALSE
                                                       )
head(scRNAseq_integrated_CAL_scenic_rss_matrix)
nrow(scRNAseq_integrated_CAL_scenic_rss_matrix)

# Transpose matrix
scRNAseq_integrated_CAL_scenic_rss_matrix_t <- t(scRNAseq_integrated_CAL_scenic_rss_matrix)
head(scRNAseq_integrated_CAL_scenic_rss_matrix_t)

```

##### Using TF genes that are activated in all CAL-ATRT single cell clusters

```{r allClustersTFs}

# Create list of TFs that are specific for each cluster
top <- 5
CALtopTF_vect <- vector()
for (i in 1:ncol(scRNAseq_integrated_CAL_scenic_rss_matrix_t))
{
    CALtopTF <- scRNAseq_integrated_CAL_scenic_rss_matrix_t[order(scRNAseq_integrated_CAL_scenic_rss_matrix_t[,i], decreasing = TRUE),] %>%
        .[1:top,] %>%
        rownames(.)
    CALtopTF_vect <- c(CALtopTF_vect,CALtopTF)
    #CALtopTF_list[[length(CALtopTF_list)+1]] <- CALtopTF
}
CALtopTF_vect <- unique(CALtopTF_vect)

# Remove "(+)" TF suffix
CALtopTF_vect <- gsub("\\(\\+\\)", "", CALtopTF_vect)
CALtopTF_vect

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% CALtopTF_vect,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_CALtopTFclusters_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor)
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_CALtopTFclusters.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclusters_meanCorr_min)
#dev.off()

```

##### Using TF genes that are activated in CAL-ATRT single cell cluster 0,1,3,5

```{r correlation}

# Choose clusters
clusters <- c(0,1,3,5)

# Create list of TFs that are specific for each cluster
top <- 5
CALtopTF_clust0135_vect <- vector()
for (i in as.character(clusters))
{
    CALtopTF <- scRNAseq_integrated_CAL_scenic_rss_matrix_t[order(scRNAseq_integrated_CAL_scenic_rss_matrix_t[,i], decreasing = TRUE),] %>%
        .[1:top,] %>%
        rownames(.)
    CALtopTF_clust0135_vect <- c(CALtopTF_clust0135_vect,CALtopTF)
}
CALtopTF_clust0135_vect <- unique(CALtopTF_clust0135_vect)

# Remove "(+)" TF suffix
CALtopTF_clust0135_vect <- gsub("\\(\\+\\)", "", CALtopTF_clust0135_vect)
CALtopTF_clust0135_vect

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135 <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% CALtopTF_clust0135_vect,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_CALtopTFclust0135_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor)
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt,
                                                                                  y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                                  by.x = "Sample_name_1",
                                                                                  by.y = "Sample_name"
                                                                                  )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_CALtopTFclust0135.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust0135_meanCorr_min)
#dev.off()

```

##### Using TF genes that are activated in CAL-ATRT single cell cluster 3

```{r correlation}

head(scRNAseq_integrated_CAL_scenic_rss_matrix_t[order(scRNAseq_integrated_CAL_scenic_rss_matrix_t[,"3"], decreasing = TRUE),])

# Extract TF specific to cluster 10
CALtopTF_clust3_vect <- scRNAseq_integrated_CAL_scenic_rss_matrix_t[order(scRNAseq_integrated_CAL_scenic_rss_matrix_t[,"3"], decreasing = TRUE),] %>% .[1:50,] %>% rownames(.)
head(CALtopTF_clust3_vect)

# Remove "(+)" TF suffix
CALtopTF_clust3_vect <- gsub("\\(\\+\\)", "", CALtopTF_clust3_vect)
CALtopTF_clust3_vect

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3 <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% CALtopTF_clust3_vect,]

nrow(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3)

# Correlation matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor <- cor(x       = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor)

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor[filter(sample_murine26_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_humanATRT_groupedLoc_CALtopTFclust3_corr.pdf"), width=16, height=12)
plot_sample_wise_corr_heatmap(matrix             = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_min,
                              samplePlan         = sample_murine26_humanATRT_groupedLoc,
                              right_annotation   = row_annot_murine26_humanATRT_groupedLoc,
                              top_annotation     = top_annot_murine26_humanATRT_groupedLoc,
                              bottom_annotation  = bottom_annot_murine26_humanATRT_groupedLoc
                              )
#dev.off()

```

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor)
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt)

# Add subgroup to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_1",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_plus)
#
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_Plus <- merge(x = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_plus,
                                                                              y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                                              by.x = "Sample_name_2",
                                                                              by.y = "Sample_name"
                                                                              )
colnames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_Plus)

matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr <- group_by(.data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_cor_melt_Plus,
                                                        subgroup_1,
                                                        subgroup_2
                                                        ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr)

# Spread table
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr <- spread(data = matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr,
                                                                          key = subgroup_2,
                                                                          value = mean_corr
                                                                          ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr) <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr$subgroup_1
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr$subgroup_1 <- NULL
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr_min <- matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]

# Rename rownames
rownames(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_CALtopTFclust3.pdf"), width=16, height=10)
plot_avgCorr_heatmap(matrix_rld_murine26_humanATRT_groupedLoc_CALtopTFclust3_meanCorr_min)
#dev.off()

```

```{r group_wise_correlation_per_cluster}

head(scRNAseq_integrated_CAL_scenic_rss_matrix_t)

# Extract TF specific for every cluster
top <- 50
CALtopTF_list <- list()
for (i in colnames(scRNAseq_integrated_CAL_scenic_rss_matrix_t))
{
    # Order according to RSS
    topTF <- scRNAseq_integrated_CAL_scenic_rss_matrix_t[order(scRNAseq_integrated_CAL_scenic_rss_matrix_t[,i],
                                                               decreasing = TRUE),] %>%
        .[1:top,] %>%
        rownames(.)
    # Remove (+)
    topTF <- gsub("\\(\\+\\)", "", topTF)
    # Append list
    CALtopTF_list[[length(CALtopTF_list)+1]] <- topTF
}
names(CALtopTF_list) <- colnames(scRNAseq_integrated_CAL_scenic_rss_matrix_t)
CALtopTF_list

# Construct matrix of top TFs for each cluster
matrix_CALtopTF_list <- list()
for (list in CALtopTF_list)
{
    matrix_CALtopTF <- matrix_rld_murine26_humanATRT_groupedLoc[rownames(matrix_rld_murine26_humanATRT_groupedLoc) %in% list,]
    matrix_CALtopTF_list[[length(matrix_CALtopTF_list)+1]] <- matrix_CALtopTF
}
names(matrix_CALtopTF_list) <- colnames(scRNAseq_integrated_CAL_scenic_rss_matrix_t)

# Compute correlation for each top TFs matrix
matrix_CALtopTF_corr_list <- list()
for (matrix in matrix_CALtopTF_list)
{
    # Compute correlation
    matrix_CALtopTF_corr <- cor(x      = matrix,
                             method = "pearson"
                             )
    matrix_CALtopTF_corr_list[[length(matrix_CALtopTF_corr_list)+1]] <- matrix_CALtopTF_corr
}
names(matrix_CALtopTF_corr_list) <- names(matrix_CALtopTF_list)
head(matrix_CALtopTF_corr_list[["0"]])

# Add subgroup info and summarize correlation by mean by subgroup
matrix_CALtopTF_corr_mean_list <- list()
for (matrix in matrix_CALtopTF_corr_list)
{
    # Melt correlation matrix
    matrix_corr_melt <- melt(matrix)
    colnames(matrix_corr_melt) <- c("Sample_name_1","Sample_name_2","correlation")
    # Add subgroup to melted matrix
    matrix_corr_melt_plus <- merge(x = matrix_corr_melt,
                                   y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                   by.x = "Sample_name_1",
                                   by.y = "Sample_name"
                                   )
    colnames(matrix_corr_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
    #
    matrix_corr_melt_Plus <- merge(x = matrix_corr_melt_plus,
                                   y = sample_murine26_humanATRT_groupedLoc_plus[,c("Sample_name","subgroup")],
                                   by.x = "Sample_name_2",
                                   by.y = "Sample_name"
                                   )
    colnames(matrix_corr_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")  
    # Summarize by mean
    matrix_corr_meanCorr <- group_by(.data = matrix_corr_melt_Plus,
                                          subgroup_1,
                                          subgroup_2
                                          ) %>%
    summarize(mean_corr = mean(correlation)
              )
    # Spread table
    matrix_corr_meanCorr <- spread(data = matrix_corr_meanCorr,
                                   key = subgroup_2,
                                   value = mean_corr
                                   ) %>%
    data.frame(check.names = FALSE)
    rownames(matrix_corr_meanCorr) <- matrix_corr_meanCorr$subgroup_1
    matrix_corr_meanCorr$subgroup_1 <- NULL
    # Subset matrix
    matrix_corr_meanCorr_min <- matrix_corr_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH")]
    # Rename rownames
    rownames(matrix_corr_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")
    # Append list
    matrix_CALtopTF_corr_mean_list[[length(matrix_CALtopTF_corr_mean_list)+1]] <- matrix_corr_meanCorr_min
}
names(matrix_CALtopTF_corr_mean_list) <- names(matrix_CALtopTF_corr_list)

# Plot average correlation
plot_CALtopTF_corr_mean_list <- HeatmapList()
for (i in 1:length(matrix_CALtopTF_corr_mean_list))
{
    p <- plot_avgCorr_heatmap(matrix_CALtopTF_corr_mean_list[[i]],
                              draw = FALSE,
                              show_heatmap_legend = FALSE,
                              column_names_rot = 90,
                              column_names_side = "bottom",
                              column_title = names(matrix_CALtopTF_corr_mean_list[i]),
                              fontsize = 20
                              )
    plot_CALtopTF_corr_mean_list <- plot_CALtopTF_corr_mean_list + p
}
#names(plot_CALtopTF_corr_mean_list) <- names(matrix_CALtopTF_corr_mean_list)

#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc_CALtopTF_clusters.pdf"), width=24, height=6)
draw(plot_CALtopTF_corr_mean_list)
#dev.off()

```




### Unsupervised analysis after organism effect correction

#### Organism effect correction

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt) == sample_murine26_humanATRT_groupedLoc$Sample_name

# PCA before data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_before_batchCor.pdf"), width=16, height=10)
pca_func(matrix          = matrix_rld_murine26_humanATRT_groupedLoc_filt,
         sampleTable     = sample_murine26_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         colors          = organism_col
         )
#dev.off()

# Data integration 
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor <- ComBat(dat = scale(matrix_rld_murine26_humanATRT_groupedLoc_filt),
                                                                 batch = sample_murine26_humanATRT_groupedLoc$Organism
                                                                 )

# Melt matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt <- melt(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt_plus <- merge(x   = sample_murine26_humanATRT_groupedLoc,
                                                                          y   = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt,
                                                                          by  = "Sample_name"
                                                                          )


# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26_humanATRT_groupedLoc_filt_batchCor.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    theme_bw()
#dev.off()

# PCA after data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_after_batchCor.pdf"), width=16, height=10)
pca_func(matrix          = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor,
         sampleTable     = sample_murine26_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         colors          = organism_col
         )
#dev.off()

```

#### Hierarchical clustering using most variable genes

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor <- genes.selection(data = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor,
                                                                  thres.num = 1000
                                                                  )

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[murine26_humanATRT_groupedLoc_mvGenes_batchCor,]

# Top annotation  
top_annot_groupedLoc_batchCor <- HeatmapAnnotation(df                    = sample_murine26_humanATRT_groupedLoc[,c("RNAseq_subgroup","groupedLoc","Organism")], 
                                                   col                   = list(RNAseq_subgroup=group_name_col, Organism = organism_col, groupedLoc = group_name_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = TRUE,
                                                   show_legend           = TRUE
                                                   )

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height           = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

```{r heatmap_for_figure}

# Check order
sample_murine26_humanATRT_groupedLoc_Plus$Sample_name == colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes)

colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes) <- sample_murine26_humanATRT_groupedLoc_Plus$Patient_ID

# Top annotation  
top_annot_groupedLoc_batchCor_fig1 <- HeatmapAnnotation(df                    = data.frame(subgroup = sample_murine26_humanATRT_groupedLoc_Plus$subgroup), 
                                                       col                   = list(subgroup=group_name_col
                                                                                    ),
                                                       gap                   = unit(2,"mm"),
                                                       show_annotation_name  = FALSE,
                                                       show_legend           = TRUE,
                                                       annotation_legend_param = list(fontsize    = 30,
                                                                                      nrow = 2,
                                                                                      title_position = "topcenter")
                                                       )

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_mvgenes_fig.pdf"), width=14, height=10)
h <- Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height           = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        show_row_dend                = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 15, fontface = "bold"),
        column_title_gp              = gpar(fontsize = 20, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 15, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = TRUE,
        column_split                 = 4,
        column_title                 = c("CAL-SHH\nR26-Shh","MCP/ICV-TYR\n","BG/IV-SHH\n","CNCS-MYC\nR26-Myc"),
        top_annotation               = top_annot_groupedLoc_batchCor_fig1,
        heatmap_legend_param = list(fontsize    = 15,
                                    title                        = "row scaled\nlog2(TPM + 1)",
                                    title_gp    = gpar(fontsize = 15, font = 2),
                                    labels_gp   = gpar(fontsize = 15),
                                    grid_width  = unit(3,"mm"),
                                    grid_height = unit(9,"mm"),
                                    ncol        = 1,
                                    side = "bottom",
                                    direction = "vertical"
                                    )
)
draw(h,
     annotation_legend_side = "bottom"
     )
#dev.off()

# Top annotation  
top_annot_groupedLoc_batchCor_fig2 <- HeatmapAnnotation(df                    = data.frame(subgroup = sample_murine26_humanATRT_groupedLoc_Plus$subgroup), 
                                                       col                   = list(subgroup=group_name_col
                                                                                    ),
                                                       gap                   = unit(2,"mm"),
                                                       show_annotation_name  = TRUE,
                                                       show_legend           = TRUE,
                                                       annotation_legend_param = list(nrow = 2,
                                                                                      title_position = "topcenter")
                                                       )

# Complex heatmap
heatmap_murine26_humanATRT <- Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes,
                                      cluster_rows                 = TRUE,
                                      cluster_columns              = TRUE,
                                      column_dend_height           = unit(10, "mm"),
                                      clustering_distance_rows     = "euclidean",
                                      clustering_distance_columns  = "pearson",
                                      clustering_method_rows       = "ward",
                                      clustering_method_columns    = "ward",
                                      show_row_names               = FALSE,
                                      show_column_names            = FALSE,
                                      show_row_dend                = FALSE,
                                      column_split                 = 4,
                                      column_dend_side             = "top",
                                      column_title                 = c("CAL-SHH\nR26-Shh","MCP/ICV-TYR\n","CNCS-MYC\nR26-Myc","BG/IV-SHH\n"),
                                      column_title_gp              = gpar(fontsize = paperFig_fontsize*1.5, fontface = "bold"),
                                      column_names_side            = "top",
                                      column_names_rot             = 0,
                                      row_names_gp                 = gpar(fontsize = paperFig_fontsize, fontface = "bold"),
                                      show_heatmap_legend          = TRUE,
                                      cluster_column_slices        = TRUE,
                                      top_annotation               = top_annot_groupedLoc_batchCor_fig2,
                                      heatmap_legend_param = list(fontsize    = paperFig_fontsize,
                                                                  title                        = "corrected\nrlog",
                                                                  title_gp    = gpar(fontsize = paperFig_fontsize*1.5, font = 2),
                                                                  labels_gp   = gpar(fontsize = paperFig_fontsize),
                                                                  grid_width  = unit(3,"mm"),
                                                                  grid_height = unit(9,"mm"),
                                                                  ncol        = 1,
                                                                  side = "right",
                                                                  direction = "vertical"
                                                                  )
        )

heatmap_murine26_humanATRT_grab <- grid.grabExpr(draw(heatmap_murine26_humanATRT,
                                                          annotation_legend_side = "bottom"
                                                          )
                                                     )


head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes)

# Change colnames to paper ATRT_ID 
data.frame(Patient_ID = colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes)
           ) %>%
    left_join(y = patientID_vs_paperID) %>%
    mutate(newColnames = case_when(is.na(LobonIglesias2023_ATRT_ID) ~ Patient_ID,
                                  TRUE ~ LobonIglesias2023_ATRT_ID
                                  )
           ) -> newColnames_df

matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes_plus <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes

colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes_plus) == newColnames_df$Patient_ID

colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes_plus) <- newColnames_df$newColnames

head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes_plus)

# Export data for source data
write.csv(x = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_mvgenes_plus,
          file = "LobonIglesias_sourceData/SuppFig5_A.csv",
          row.names = TRUE
          )

```


#### Hierarchical clustering using most variable genes across human samples

```{r organism_effect_correction}

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_humanMvGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% humanATRT_groupedLoc_mvGenes,]
nrow(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_humanMvGenes)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_humanMvGenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_humanMvGenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using immune genes

```{r organism_effect_correction}

length(immuneGenes)

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_immuneGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% immuneGenes,]
nrow(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_immuneGenes)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_immuneGenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_immuneGenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using most variable genes after removing immune genes

```{r organism_effect_correction}

head(immuneGenes)

nrow(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)

matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[!rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% immuneGenes,]

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes <- genes.selection(data = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes,
                                                                                thres.num = 1000
                                                                                )

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes[murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes,]

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_noImmuneGenes_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using top 5 TFs that are activated in every R26 single cell cluster

```{r organism_effect_correction}

head(topTF_vect_humanSymbol)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTF <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% topTF_vect_humanSymbol,]

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedR26TopTF.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTF,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 15, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using top 5 TFs that are activated in single cell R26-SHH cluster 1,2,3,4,5,6,7 and 9

```{r organism_effect_correction}

head(topTF_clust12345679_vect_humanSymbol)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust12345679 <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% topTF_clust12345679_vect_humanSymbol,]

dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust12345679)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedR26TopTFclust12345679.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust12345679,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```


#### Hierarchical clustering using top 50 TFs that are activated in R26-SHH single cell cluster 6

```{r organism_effect_correction}

head(topTF_clust6_vect_humanSymbol)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust6 <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% topTF_clust6_vect_humanSymbol,]

dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust6)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedR26TopTFclust6.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedR26TopTFclust6,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using top 5 TFs that are activated in every CAL-ATRT single cell cluster

```{r organism_effect_correction}

head(CALtopTF_vect)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTF <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% CALtopTF_vect,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTF)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedCALTopTF.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTF,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 15, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using top 5 TFs that are activated in single cell CAL-ATRT cluster 0,1,3 and 5

```{r organism_effect_correction}

head(CALtopTF_clust0135_vect)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust0135 <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% CALtopTF_clust0135_vect,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust0135)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedCALTopTFclust0135.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust0135,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```


#### Hierarchical clustering using top 50 TFs that are activated in CAL-ATRT single cell cluster 3

```{r organism_effect_correction}

head(CALtopTF_clust3_vect)

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust3 <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% CALtopTF_clust3_vect,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust3)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_integratedCALTopTFclust3.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_integratedCALTopTFclust3,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using ganglionic eminence genes

```{r organism_effect_correction}

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% ganglionicEminenceGenes,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceGenes)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_ganglionicEminenceGenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceGenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using ganglionic eminence TFs (see e-mail from Maria on Oct 6 2022)

```{r organism_effect_correction}

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceTFs <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% ganglionicEminenceTFs,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceTFs)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_ganglionicEminenceTFs.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_ganglionicEminenceTFs,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using midbrain/hindbrain boundary genes

```{r organism_effect_correction}

# Select matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_midbrainHindbrainBoundaryGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) %in% midbrainHindbrainBoundaryGenes,]
dim(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_midbrainHindbrainBoundaryGenes)

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_batchCor_midbrainHindbrainBoundaryGenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_midbrainHindbrainBoundaryGenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = TRUE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 20, fontface = "plain"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```


#### Hierarchical clustering using midbrain/hindbrain boundary and ganglionic eminence genes

```{r heatmap_with_MHB_GE_genes}

# Create a data frame for ganglionic eminence genes
ganglionicEminenceGenes_df <- data.frame(gene = ganglionicEminenceGenes,
                                         region = "ganglionic eminence"
                                         )
head(ganglionicEminenceGenes_df)

icicicicicicicici
# Create a data frame for midbrain/hindbrain boundary genes
midbrainHindbrainBoundaryGenes_df <- data.frame(gene = midbrainHindbrainBoundaryGenes,
                                                region = "midbrain/hindbrain boundary"
                                                )
head(midbrainHindbrainBoundaryGenes_df)

icicicicicicicici

# Bind data frames
ganglionicEminenceMidbrainHindbrainBoundaryGenes <- rbind(ganglionicEminenceGenes_df,
                                                          midbrainHindbrainBoundaryGenes_df
                                                          )

ganglionicEminenceMidbrainHindbrainBoundaryGenes[duplicated(ganglionicEminenceMidbrainHindbrainBoundaryGenes$gene),]

icicicicicicicici

# Keep genes in filtered matrix
ganglionicEminenceMidbrainHindbrainBoundaryGenes_min <- filter(.data = ganglionicEminenceMidbrainHindbrainBoundaryGenes,
                                                               gene %in% rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)
                                                               )

# Subset TPM matrix
head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_midbrainHindbrainBoundaryGenes)

icicicicicicicicicicicicici

matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$gene,sample_murine26_humanATRT_groupedLoc$Sample_name]


# Check order
sample_murine26_humanATRT_groupedLoc$Sample_name == colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB)

# Check order
ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$gene == rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB)


gene_category_color <- c("ganglionic eminence" = "brown",
                         "midbrain/hindbrain boundary" = "violet"
                         )

# Row annotation
row_annot_ganglionicEminenceMidbrainHindbrainBoundaryGenes <- rowAnnotation(df  = data.frame(`Gene category` = ganglionicEminenceMidbrainHindbrainBoundaryGenes_min[,"region"], check.names = FALSE),
                                                                            col = list(`Gene category` = gene_category_color),
                                                                            show_legend = FALSE,
                                                                            show_annotation_name    = FALSE
                                                                            )

colnames(sample_murine26_humanATRT_groupedLoc)

sample_murine26_humanATRT_groupedLoc_plus$subgroup


sample_murine26_humanATRT_groupedLoc_Plus <- mutate(.data = sample_murine26_humanATRT_groupedLoc_plus,
subgroup = case_when(subgroup == "MYC_MYC" ~ "CNCS-MYC",
subgroup == "MYC_SHH" ~ "BG/IV-SHH",
subgroup == "SHH_SHH" ~ "CAL-SHH",
subgroup == "TYR_TYR" ~ "MCP/ICV-TYR",
subgroup == "R26-MYC" ~ "R26-Myc",
subgroup == "R26-SHH" ~ "R26-Shh"
)
)

sample_murine26_humanATRT_groupedLoc_Plus$subgroup

sample_murine26_humanATRT_groupedLoc_Plus$Sample_name == sample_murine26_humanATRT_groupedLoc$Sample_name

# Hierarchical clustering and heatmap
h <- Heatmap(matrix = t(scale(t(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB))),
                                        #name                        = "scaled log2(TPM+1)",
             row_title                   = NULL,
                                        #column_title                = c("BG/IV-SHH","CAL-SHH"),
             cluster_columns             = TRUE,
             cluster_rows                = FALSE,
             clustering_distance_columns = "euclidean",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows  = "ward",
                                        #top_annotation    = top_annot_Curie_ATRT_primTum_MBAyrault_SHH_antVermis_min,
             left_annotation         = row_annot_ganglionicEminenceMidbrainHindbrainBoundaryGenes,
             show_row_dend           = FALSE,
             show_column_dend        = FALSE,
             show_column_names        = FALSE,
             row_names_gp            = gpar(fontface = 4, fontsize = 15),
             column_names_gp         = gpar(fontsize = 15),
             column_title_gp         = gpar(fontsize = 20,fontface="bold"),
        column_split            = sample_murine26_humanATRT_groupedLoc_Plus$subgroup,
        row_split            = ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$region,
                                        #row_split                      = interestedGenes_ATRT_SHH_vs_MB_SHH_antVermis$tumor_subtype,
             heatmap_legend_param = list(fontsize    = paperFig_fontsize,
                                         title                        = "scaled\nlog2(TPM + 1)",
                                         title_gp    = gpar(fontsize = 15, font = 2),
                                         labels_gp   = gpar(fontsize = 15),
                                         grid_width  = unit(3,"mm"),
                                         grid_height = unit(9,"mm"),
                                         ncol        = 1,
                                         side = "bottom",
                                         direction = "vertical"
                                         )
             )
#pdf(paste0(pathToFigures,"complexHeatmap_RNAseq_humanATRT_murine26_batchCor_ganglionicEminenceMidbrainHindbrainBoundaryGenes.pdf"), width=14, height=14)
draw(h
     #,heatmap_legend_side = "right"
     )
#dev.off()

```

```{r heatmap_with_MHB_GE_genes}

# Subset R26-Shh and ATRT SHH samples
sample_murine26_humanATRT_groupedLoc_min <- filter(.data    = sample_murine26_humanATRT_groupedLoc_Plus,
                                                   subgroup %in% c("R26-Shh","CAL-SHH","BG/IV-SHH")
                                                   )

icicicicicicicicicicici

# Subset  R26-Shh and ATRT SHH samples matrix
matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB[,sample_murine26_humanATRT_groupedLoc_min$Sample_name]

# Check order
sample_murine26_humanATRT_groupedLoc_min$Sample_name == colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min)

# Use Patient_ID as colnames
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min) <- sample_murine26_humanATRT_groupedLoc_min$Patient_ID

# Check order
ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$gene == rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min)

# Hierarchical clustering and heatmap
h <- Heatmap(matrix = t(scale(t(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min))),
                                        #name                        = "scaled log2(TPM+1)",
             row_title                   = c("ganglionic\neminence genes","midbrain/hindbrain\nboundary genes"),
                                        #column_title                = c("BG/IV-SHH","CAL-SHH"),
             cluster_columns             = TRUE,
             cluster_rows                = FALSE,
             clustering_distance_columns = "euclidean",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows  = "ward",
             left_annotation         = row_annot_ganglionicEminenceMidbrainHindbrainBoundaryGenes,
             show_row_dend           = FALSE,
             show_column_dend        = FALSE,
             show_column_names        = FALSE,
             row_names_gp            = gpar(fontface = 4, fontsize = 15),
             column_names_gp         = gpar(fontsize = 15),
             column_title_gp         = gpar(fontsize = 20,fontface="bold"),
             column_split            = sample_murine26_humanATRT_groupedLoc_min$subgroup,
             row_split            = ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$region,
             heatmap_legend_param = list(fontsize    = paperFig_fontsize,
                                         title                        = "row scaled\nlog2(TPM + 1)",
                                         title_gp    = gpar(fontsize = 15, font = 2),
                                         labels_gp   = gpar(fontsize = 15),
                                         grid_width  = unit(3,"mm"),
                                         grid_height = unit(9,"mm"),
                                         ncol        = 1,
                                         side = "bottom",
                                         direction = "vertical"
                                         )
             )
#pdf(paste0(pathToFigures,"complexHeatmap_RNAseq_humanATRT_murine26_batchCor_ganglionicEminenceMidbrainHindbrainBoundaryGenes_min.pdf"), width=14, height=14)
draw(h
                                        #,heatmap_legend_side = "right"
     )
#dev.off()

heatmap_MHB_GE_min <- Heatmap(matrix = t(scale(t(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min))),
                                        #name                        = "scaled log2(TPM+1)",
                              row_title                   = c("ganglionic\neminence genes","midbrain/hindbrain\nboundary genes"),
                                        #column_title                = c("BG/IV-SHH","CAL-SHH"),
                              cluster_columns             = TRUE,
                              cluster_rows                = FALSE,
                              clustering_distance_columns = "euclidean",
                              clustering_method_columns   = "ward",
                              clustering_distance_rows    = "euclidean",
                              clustering_method_rows  = "ward",
                              left_annotation         = row_annot_ganglionicEminenceMidbrainHindbrainBoundaryGenes,
                              show_row_dend           = FALSE,
                              show_column_dend        = FALSE,
                              show_column_names        = FALSE,
                              row_names_gp            = gpar(fontface = 4, fontsize = paperFig_fontsize*1.5),
                              column_names_gp         = gpar(fontsize = paperFig_fontsize*1.5),
                              column_title_gp         = gpar(fontsize = paperFig_fontsize*1.5,fontface="bold"),
                              column_split            = sample_murine26_humanATRT_groupedLoc_min$subgroup,
                              row_split            = ganglionicEminenceMidbrainHindbrainBoundaryGenes_min$region,
                              heatmap_legend_param = list(fontsize    = paperFig_fontsize,
                                                          title                        = "row scaled\nlog2(TPM + 1)",
                                                          title_gp    = gpar(fontsize = paperFig_fontsize*1.5, font = 2),
                                                          labels_gp   = gpar(fontsize = paperFig_fontsize),
                                                          grid_width  = unit(3,"mm"),
                                                          grid_height = unit(9,"mm"),
                                                          ncol        = 1,
                                                          side = "bottom",
                                                          direction = "vertical"
                                                          )
             )

heatmap_MHB_GE_min_grab <- grid.grabExpr(draw(heatmap_MHB_GE_min,
                                          heatmap_legend_side = "right"
                                          )
                                     )

head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min)

# Change colnames to paper ATRT_ID 
data.frame(Patient_ID = colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min)
           ) %>%
    left_join(y = patientID_vs_paperID) %>%
    mutate(newColnames = case_when(is.na(LobonIglesias2023_ATRT_ID) ~ Patient_ID,
                                  TRUE ~ LobonIglesias2023_ATRT_ID
                                  )
           ) -> newColnames_df

matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min_plus <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min

colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min_plus) == newColnames_df$Patient_ID

colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min_plus) <- newColnames_df$newColnames

head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min_plus)


# Export data for source data
write.csv(x = t(scale(t(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor_GE_MHB_min_plus))),
          file = "LobonIglesias_sourceData/SuppFig5_B.csv",
          row.names = TRUE
          )

```

#### Supplementary Fig. 4bis

```{r figure6_patchwork}

library(patchwork)

suppFig4bis_A <- wrap_elements(heatmap_murine26_humanATRT_grab) + labs(tag = 'A')

suppFig4bis_B <- wrap_elements(heatmap_MHB_GE_min_grab) + labs(tag = 'B')

# Patchwork
suppFig4bis <- wrap_plots(suppFig4bis_A,
                         suppFig4bis_B,
                         heights = c(0.5,0.5)
                         ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 15, face = 'bold')
          ) &
    plot_annotation(title = "Supplementary Fig. 5\n",
                    tag_levels = 'A',
                    theme = theme(plot.margin = unit(c(1,2,2,2),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Rename figure
SuppFig4bis_LobonIglesias2022 <- suppFig4bis

# Save figure into pdf file
ggsave(plot   = SuppFig4bis_LobonIglesias2022,
       file   = paste0(pathToFigures,"SuppFig5new_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig5new_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_human_ATRT"
              )

```



# PAM analysis (Nearest shrunken centroid) to identify specific human/mouse gene signatures

## Mouse SHH (neuronal) and human BG/IV (MYC_SHH) specific genes by PAM analysis

```{r pam_analysis}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Create a group combining human MYC_SHH and mouse SHH
sample_murine26_humanATRT_groupedLoc_pamr <- mutate(.data = sample_murine26_humanATRT_groupedLoc,
                                                    groupLocPamr = case_when(groupedLoc == "MYC_SHH" | RNAseq_subgroup == "neuronal" ~ "BGIV_neuronal",
                                                                             RNAseq_subgroup %in% c("MYC","non_neuronal") ~ "CNCS_nonNeuronal",
                                                                             TRUE ~ RNAseq_subgroup
                                                                             )
                                                    )
sample_murine26_humanATRT_groupedLoc_pamr[,c("Organism","RNAseq_subgroup","groupedLoc","groupLocPamr")]

head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[,1:4])

# Create pamr data object
x <- as.matrix(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)
colnames(x) <- NULL

y <- sample_murine26_humanATRT_groupedLoc_pamr$groupLocPamr

genenames <- rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)

geneid <- rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)

samplelabels <- as.vector(sample_murine26_humanATRT_groupedLoc_pamr$Sample_name)

pamr_murine26_humanATRT_groupedLoc <- list(x, y, genenames, geneid, samplelabels)

names(pamr_murine26_humanATRT_groupedLoc) <- c("x", "y", "genenames", "geneid", "samplelabels")

library(pamr)

# Train the classifier
pamr_murine26_humanATRT_groupedLoc.train <- pamr.train(data = pamr_murine26_humanATRT_groupedLoc)
pamr_murine26_humanATRT_groupedLoc.train

# Cross-validate the classifier
pamr_murine26_humanATRT_groupedLoc.results <- pamr.cv(fit = pamr_murine26_humanATRT_groupedLoc.train,
                                                      data = pamr_murine26_humanATRT_groupedLoc
                                                      )
pamr_murine26_humanATRT_groupedLoc.results

# Plot the cross-validated error curves
#pdf(paste0(pathToFigures,"pamr_plotcv_murine26_humanATRT_BGIVneuronal.pdf"), width=14, height=16)
pamr.plotcv(pamr_murine26_humanATRT_groupedLoc.results)
#dev.off()

thr <- 4.785
# Compute the confusion matrix for a particular model
pamr_murine26_humanATRT_groupedLoc_confusion <- pamr.confusion(fit       = pamr_murine26_humanATRT_groupedLoc.results,
                                                               threshold = thr,
                                                               extra     = FALSE
                                                               )
pamr_murine26_humanATRT_groupedLoc_confusion

plot_pamr_confusion <- function(parmConfusionTable)
{
melt(parmConfusionTable) %>%
    ggplot(mapping = aes(x = true, y = predicted, fill = value)
           ) +
    geom_tile() +
    scale_fill_gradient2(low = "white", mid = "grey90",high = "red") +
    theme_bw() +
    xlab("\nReference") +
    ylab("Prediction\n") +
    geom_text(mapping = aes(label = value),
              size = 10
              ) +
    theme(text = element_text(size = 30),
          axis.text = element_text(face = "bold"),
          axis.title = element_text(face = "bold"),
          axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
          legend.position = "none",
          axis.line = element_blank(),
          axis.ticks = element_blank(),
          panel.grid = element_blank(),
          panel.border = element_blank(),
          panel.background = element_blank()
          )
}

#pdf(paste0(pathToFigures,"pamr_confusion_murine26_humanATRT_BGIVneuronal.pdf"), width=12, height=12)
plot_pamr_confusion(pamr_murine26_humanATRT_groupedLoc_confusion)
#dev.off()

# Plot the cross-validated class probabilities by class
#pdf(paste0(pathToFigures,"pamr_plotcvprob_murine26_humanATRT_BGIVneuronal.pdf"), width=18, height=10)
pamr.plotcvprob(fit      = pamr_murine26_humanATRT_groupedLoc.results,
                data     = pamr_murine26_humanATRT_groupedLoc,
                threshold = thr
                )
#dev.off()

# Plot the class centroids
#pdf(paste0(pathToFigures,"pamr_plotcen_murine26_humanATRT_BGIVneuronal.pdf"), width=14, height=24)
pamr.plotcen(fit       = pamr_murine26_humanATRT_groupedLoc.train,
             data      = pamr_murine26_humanATRT_groupedLoc,
             threshold = thr
             )
#dev.off()

# Make a gene plot of the most significant genes
#pdf(paste0(pathToFigures,"pamr_geneplot_murine26_humanATRT_BGIVneuronal.pdf"), width=14, height=12)
pamr.geneplot(fit       = pamr_murine26_humanATRT_groupedLoc.train,
              data      = pamr_murine26_humanATRT_groupedLoc,
              threshold = thr
              )
#dev.off()

# Estimate false discovery rate and plot them
fdrObj_pamr_murine26_humanATRT_groupedLoc.train <- pamr.fdr(trained.obj = pamr_murine26_humanATRT_groupedLoc.train,
                                                            data        = pamr_murine26_humanATRT_groupedLoc,
                                                            nperms      = 100
                                                            )

#pdf(paste0(pathToFigures,"pamr_plotfdr_murine26_humanATRT_BGIVneuronal.pdf"), width=12, height=14)
pamr.plotfdr(fdrfit = fdrObj_pamr_murine26_humanATRT_groupedLoc.train,
             call.win.metafile = TRUE # TRUE: allow plot to be printed in R device, set to FALSE if you want to export the plot in pdf
             )
#dev.off()

# List the significant genes
pamr_murine26_humanATRT_groupedLoc_pamGeneList <- pamr.listgenes(fit        = pamr_murine26_humanATRT_groupedLoc.train,
                                                                 data       = pamr_murine26_humanATRT_groupedLoc,
                                                                 threshold  = thr,
                                                                 fitcv      = NULL,
                                                                 genenames  = FALSE
                                                                 )
pamr_murine26_humanATRT_groupedLoc_pamGeneList <- data.frame(pamr_murine26_humanATRT_groupedLoc_pamGeneList)

head(pamr_murine26_humanATRT_groupedLoc_pamGeneList)
nrow(pamr_murine26_humanATRT_groupedLoc_pamGeneList)

pamr_murine26_humanATRT_groupedLoc_pamGeneList_order <- pamr_murine26_humanATRT_groupedLoc_pamGeneList[order(pamr_murine26_humanATRT_groupedLoc_pamGeneList$BGIV_neuronal.score, decreasing = TRUE),]

pamr_murine26_humanATRT_groupedLoc_pamGeneList_order[1:20,]

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[pamr_murine26_humanATRT_groupedLoc_pamGeneList_order[1:16,"id"],]

head(matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes[,1:4])
nrow(matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes)

# Order matrix to have the same ordre as in sample annotation
matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes <- matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes[,sample_murine26_humanATRT_groupedLoc_pamr$Sample_name]

sample_murine26_humanATRT_groupedLoc_pamr$groupLocPamr

# Top annotation
top_annot_groupedLoc_batchCor_pamr <- HeatmapAnnotation(df                    = sample_murine26_humanATRT_groupedLoc_pamr[,c("groupLocPamr","Organism")], 
                                                        col                   = list(groupLocPamr = c(CNCS_nonNeuronal = "darkgreen",
                                                                                                      BGIV_neuronal = "orange",
                                                                                                      SHH = "blue",
                                                                                                      TYR = "red"
                                                                                                      ),
                                                                                     Organism = organism_col
                                                                                     ),
                                                        gap                   = unit(2,"mm"),
                                                        show_annotation_name  = TRUE,
                                                        show_legend           = TRUE,
                                                        annotation_name_gp = gpar(fontsize = 20),
                                                        annotation_legend_param        = list(title_gp = gpar(fontsize = 30),
                                                                                              labels_gp = gpar(fontsize = 20)
                                                                                              )
                                                        )

# Complex heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_pamrGenes_murine26_humanATRT_BGIVneuronal.pdf"), width=16, height=16)
h <- Heatmap(matrix                      = scaleRow_func(matrix_rld_murine26_humanATRT_groupedLoc_pamrGenes),
             cluster_columns = FALSE,
        #clustering_distance_columns = "pearson",
        #clustering_method_columns   = "ward",
        clustering_distance_rows    = "euclidean",
        show_row_names              = TRUE,
        #column_title                = paste0("Unsupervised clustering \n(distance: ",my_dist,", method: ",my_meth,")"),
        #row_title                   = paste0("pam genes (4612, threshold =",thr,") \n Unsupervised clustering (distance: ",my_dist,", method: ",my_meth,")"),
        top_annotation              = top_annot_groupedLoc_batchCor_pamr,
        show_heatmap_legend         = TRUE,
        show_row_dend               = FALSE,
        column_dend_height          = unit(40, "mm"),
        column_title_gp             = gpar(fontsize = 30),
        row_names_gp                = gpar(fontsize = 30),
        row_names_side                = "left",
        column_order                = order(sample_murine26_humanATRT_groupedLoc_pamr$groupLocPamr),
        column_split                = sample_murine26_humanATRT_groupedLoc_pamr$groupLocPamr,
        heatmap_legend_param        = list(color_bar = "continuous",
                                           direction = "horizontal",
                                           grid_width = unit(20,"cm"),
                                           grid_height = unit(1,"cm"),
                                           title  = "expression level\n (scaled log2Intensity) \n",
                                           title_gp = gpar(fontsize = 22),
                                           title_position = "topcenter"
                                           )
        )
draw(h, heatmap_legend_side = "bottom", annotation_legend_side = "right")
#dev.off()

```

## Mouse SHH (neuronal) and human CAL (SHH_SHH) specific genes by PAM analysis

```{r pam_analysis}

# Check order
colnames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor) == sample_murine26_humanATRT_groupedLoc$Sample_name

# Create a group combining human MYC_SHH and mouse SHH
sample_murine26_humanATRT_groupedLoc_pamr2 <- mutate(.data = sample_murine26_humanATRT_groupedLoc,
                                                     groupLocPamr2 = case_when(groupedLoc == "SHH_SHH" | RNAseq_subgroup == "neuronal" ~ "CAL_neuronal",
                                                                               groupedLoc == "MYC_SHH" ~ "BGIV",
                                                                               groupedLoc == "MYC_MYC" | RNAseq_subgroup == "non_neuronal"  ~ "CNCS_nonNeuronal",
                                                                               groupedLoc == "TYR_TYR" ~ "MCPICV"
                                                                             )
                                                    )

sample_murine26_humanATRT_groupedLoc_pamr2[,c("Organism","RNAseq_subgroup","groupedLoc","groupLocPamr2")]

head(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[,1:4])

# Create pamr data object
x <- as.matrix(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)
colnames(x) <- NULL

y <- sample_murine26_humanATRT_groupedLoc_pamr2$groupLocPamr2

genenames <- rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)

geneid <- rownames(matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor)

samplelabels <- as.vector(sample_murine26_humanATRT_groupedLoc_pamr2$Sample_name)

pamr2_murine26_humanATRT_groupedLoc <- list(x, y, genenames, geneid, samplelabels)

names(pamr2_murine26_humanATRT_groupedLoc) <- c("x", "y", "genenames", "geneid", "samplelabels")

# Train the classifier
pamr2_murine26_humanATRT_groupedLoc.train <- pamr.train(data = pamr2_murine26_humanATRT_groupedLoc)
pamr2_murine26_humanATRT_groupedLoc.train

# Cross-validate the classifier
pamr2_murine26_humanATRT_groupedLoc.results <- pamr.cv(fit = pamr2_murine26_humanATRT_groupedLoc.train,
                                                      data = pamr2_murine26_humanATRT_groupedLoc
                                                      )
pamr2_murine26_humanATRT_groupedLoc.results

# Plot the cross-validated error curves
#pdf(paste0(pathToFigures,"pamr_plotcv_murine26_humanATRT_CALneuronal.pdf"), width=14, height=16)
pamr.plotcv(pamr2_murine26_humanATRT_groupedLoc.results)
#dev.off()

thr <- 0.373
# Compute the confusion matrix for a particular model
pamr2_murine26_humanATRT_groupedLoc_confusion <- pamr.confusion(fit       = pamr2_murine26_humanATRT_groupedLoc.results,
                                                               threshold = thr,
                                                               extra     = FALSE
                                                               )
pamr2_murine26_humanATRT_groupedLoc_confusion

#pdf(paste0(pathToFigures,"pamr_confusion_murine26_humanATRT_CALneuronal.pdf"), width=12, height=12)
plot_pamr_confusion(pamr2_murine26_humanATRT_groupedLoc_confusion)
#dev.off()

# Plot the cross-validated class probabilities by class
#pdf(paste0(pathToFigures,"pamr_plotcvprob_murine26_humanATRT_CALneuronal.pdf"), width=18, height=10)
pamr.plotcvprob(fit      = pamr2_murine26_humanATRT_groupedLoc.results,
                data     = pamr2_murine26_humanATRT_groupedLoc,
                threshold = thr
                )
#dev.off()

# Plot the class centroids
#pdf(paste0(pathToFigures,"pamr_plotcen_murine26_humanATRT_CALneuronal.pdf"), width=14, height=24)
pamr.plotcen(fit       = pamr2_murine26_humanATRT_groupedLoc.train,
             data      = pamr2_murine26_humanATRT_groupedLoc,
             threshold = thr
             )
#dev.off()

# Make a gene plot of the most significant genes
#pdf(paste0(pathToFigures,"pamr_geneplot_murine26_humanATRT_CALneuronal.pdf"), width=14, height=12)
pamr.geneplot(fit       = pamr2_murine26_humanATRT_groupedLoc.train,
              data      = pamr2_murine26_humanATRT_groupedLoc,
              threshold = thr
              )
#dev.off()

# Estimate false discovery rate and plot them
fdrObj_pamr2_murine26_humanATRT_groupedLoc.train <- pamr.fdr(trained.obj = pamr2_murine26_humanATRT_groupedLoc.train,
                                                            data        = pamr2_murine26_humanATRT_groupedLoc,
                                                            nperms      = 100
                                                            )

#pdf(paste0(pathToFigures,"pamr2_plotfdr_murine26_humanATRT_CALneuronal.pdf"), width=12, height=14)
pamr.plotfdr(fdrfit = fdrObj_pamr2_murine26_humanATRT_groupedLoc.train,
             call.win.metafile = TRUE # TRUE: allow plot to be printed in R device, set to FALSE if you want to export the plot in pdf
             )
#dev.off()

# List the significant genes
pamr2_murine26_humanATRT_groupedLoc_pamGeneList <- pamr.listgenes(fit        = pamr2_murine26_humanATRT_groupedLoc.train,
                                                                 data       = pamr2_murine26_humanATRT_groupedLoc,
                                                                 threshold  = thr,
                                                                 fitcv      = NULL,
                                                                 genenames  = FALSE
                                                                 )
pamr2_murine26_humanATRT_groupedLoc_pamGeneList <- data.frame(pamr2_murine26_humanATRT_groupedLoc_pamGeneList)

head(pamr2_murine26_humanATRT_groupedLoc_pamGeneList)
nrow(pamr2_murine26_humanATRT_groupedLoc_pamGeneList)

pamr2_murine26_humanATRT_groupedLoc_pamGeneList_order <- pamr2_murine26_humanATRT_groupedLoc_pamGeneList[order(as.numeric(pamr2_murine26_humanATRT_groupedLoc_pamGeneList$CAL_neuronal.score), decreasing = TRUE),]

pamr2_murine26_humanATRT_groupedLoc_pamGeneList_order[1:20,]

# Subset matrix
matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes <- matrix_rld_murine26_humanATRT_groupedLoc_filt_batchCor[pamr2_murine26_humanATRT_groupedLoc_pamGeneList_order[1:40,"id"],]

head(matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes[,1:4])
nrow(matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes)

# Order matrix to have the same ordre as in sample annotation
matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes <- matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes[,sample_murine26_humanATRT_groupedLoc_pamr2$Sample_name]

sample_murine26_humanATRT_groupedLoc_pamr2$groupLocPamr2

# Top annotation
top_annot_groupedLoc_batchCor_pamr2 <- HeatmapAnnotation(df                    = sample_murine26_humanATRT_groupedLoc_pamr2[,c("groupLocPamr2","Organism")], 
                                                        col                   = list(groupLocPamr2 = c(CNCS_nonNeuronal = "darkgreen",
                                                                                                       CAL_neuronal = "blue",
                                                                                                       BGIV = "orange",
                                                                                                       MCPICV = "red"
                                                                                                       ),
                                                                                     Organism = organism_col
                                                                                     ),
                                                        gap                   = unit(2,"mm"),
                                                        show_annotation_name  = TRUE,
                                                        show_legend           = TRUE,
                                                        annotation_name_gp = gpar(fontsize = 20),
                                                        annotation_legend_param        = list(title_gp = gpar(fontsize = 30),
                                                                                              labels_gp = gpar(fontsize = 20)
                                                                                              )
                                                        )

# Complex heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_pamrGenes_murine26_humanATRT_CALneuronal.pdf"), width=16, height=16)
h <- Heatmap(matrix                      = scaleRow_func(matrix_rld_murine26_humanATRT_groupedLoc_pamr2Genes),
             cluster_columns = FALSE,
        #clustering_distance_columns = "pearson",
        #clustering_method_columns   = "ward",
        clustering_distance_rows    = "euclidean",
        show_row_names              = TRUE,
        #column_title                = paste0("Unsupervised clustering \n(distance: ",my_dist,", method: ",my_meth,")"),
        #row_title                   = paste0("pam genes (4612, threshold =",thr,") \n Unsupervised clustering (distance: ",my_dist,", method: ",my_meth,")"),
        top_annotation              = top_annot_groupedLoc_batchCor_pamr2,
        show_heatmap_legend         = TRUE,
        show_row_dend               = FALSE,
        column_dend_height          = unit(40, "mm"),
        column_title_gp             = gpar(fontsize = 30),
        row_names_gp                = gpar(fontsize = 20),
        row_names_side                = "left",
        column_order                = order(sample_murine26_humanATRT_groupedLoc_pamr2$groupLocPamr2),
        column_split                = sample_murine26_humanATRT_groupedLoc_pamr2$groupLocPamr2,
        heatmap_legend_param        = list(color_bar = "continuous",
                                           direction = "horizontal",
                                           grid_width = unit(20,"cm"),
                                           grid_height = unit(1,"cm"),
                                           title  = "expression level\n (scaled log2Intensity) \n",
                                           title_gp = gpar(fontsize = 22),
                                           title_position = "topcenter"
                                           )
        )
draw(h, heatmap_legend_side = "bottom", annotation_legend_side = "right")
#dev.off()

```

# Compare single cell R26 samples to mouse and human bulk RNA-seq

## Import pseudo bulk from single cell RNA-seq

```{r import_pseudobulk}

# Import pseudo bulk of each sample
scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix <- read.csv(file       = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/scRNAseq/scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix.csv"
         )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix <- read.csv(file       = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/scRNAseq/scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix.csv"
         )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix <- read.csv(file       = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/scRNAseq/scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix.csv"
         )

# TO REMOVE!
colnames(scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix) <- c("gene","R26-P87")
colnames(scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix) <- c("gene","R26-P27")
colnames(scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix) <- c("gene","R26-P26")

head(scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix)

scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix <- column_to_rownames(scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix, var = "gene")
scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix <- column_to_rownames(scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix, var = "gene")
scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix <- column_to_rownames(scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix, var = "gene")

# Bind pseudobulk matrices
scRNAseq_Seurat_integrated_R26_tumCells_pseudoCounts_matrix <- Reduce(f= "cbind",
                                                                      x = list(scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix,
                                                                               scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix,
                                                                               scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix
                                                                               )
                                                                      )
                                                                           

head(scRNAseq_Seurat_integrated_R26_tumCells_pseudoCounts_matrix)

```

## Append R26 pseudo-bulk 

```{r import_pseudobulk}

# Import data file annotation (file containing all the data wih their respective description)
sampleTable <- read_tsv(file = paste0(pathToDataAnnotationFile,"data_file_annotation.csv")
                        )
sampleTable

# Select R26 information (InnovRT003 is a scRNA-seq sample) 
sample_R26P87_R26P27_R26P26_dbl <- filter(.data                = sampleTable,
                                          Biological_category == "transcriptomics",
                                          Data_source         == "Curie",
                                          Data_type           == "scRNAseq",
                                          Organism            == "Mus_musculus",
                                          File_format         == "FASTQ",
                                          Patient_ID        %in%  c("R26P87","R26P27","R26P26")
                                          )

sample_R26P87_R26P27_R26P26_dbl

sample_R26P87_R26P27_R26P26 <- sample_R26P87_R26P27_R26P26_dbl[!duplicated(sample_R26P87_R26P27_R26P26_dbl$Patient_ID),]

# Add columns
sample_R26P87_R26P27_R26P26_plus <- mutate(sample_R26P87_R26P27_R26P26,
                                           RNAseq_subgroup = NA,                    
                                           methArray_pearson = NA,
                                           methArray_euclidean = NA,                
                                           DKFZ_web_subgroup = NA,       
                                           DKFZ_web_score = NA,                      
                                           groupedLoc = NA
                                           )

sample_R26P87_R26P27_R26P26_plus$group_name <- NULL

# Append ATRT sample plan
sample_Curie_R26_R26P87_R26P27_R26P26 <- rbind(sample_Curie_R26,sample_R26P87_R26P27_R26P26_plus)

# Data summary
group_by(.data = sample_Curie_R26_R26P87_R26P27_R26P26,
         Genotype,
         Data_type
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Extract R26 matrix with mouse gene name
matrix_rawCounts_Curie_R26_mouseSymbol <- matrix_rawCounts_Curie_Vitte2017[,sample_Curie_R26$Sample_name]
dim(matrix_rawCounts_Curie_R26_mouseSymbol)
head(matrix_rawCounts_Curie_R26_mouseSymbol[,1:4])

# Append ATRT matrix
matrix_Curie_R26_integratedR26tumCells_rawCounts <- merge(x     = matrix_rawCounts_Curie_R26_mouseSymbol,
                                                                    y     = scRNAseq_Seurat_integrated_R26_tumCells_pseudoCounts_matrix,
                                                                    by  = 0,
                                                                    all   = FALSE
                                                                    )

head(matrix_Curie_R26_integratedR26tumCells_rawCounts[,1:4])

matrix_Curie_R26_integratedR26tumCells_rawCounts <- column_to_rownames(matrix_Curie_R26_integratedR26tumCells_rawCounts,
                                                                                 var = "Row.names"
                                                                                 )

head(matrix_Curie_R26_integratedR26tumCells_rawCounts)
dim(matrix_Curie_R26_integratedR26tumCells_rawCounts)

# Check order
sample_Curie_R26_R26P87_R26P27_R26P26$Sample_name == colnames(matrix_Curie_R26_integratedR26tumCells_rawCounts)

```

## Data distribution before filtering

```{r data_distribution}

matrix_Curie_R26_integratedR26tumCells_rawCounts %>%
    melt(.) %>%
    merge(x = .,
          y = sample_Curie_R26_R26P87_R26P27_R26P26,
          by.x = "variable",
          by.y = "Sample_name"
          ) %>%
    #filter(Data_type != "scRNAseq") %>%
    ggplot(mapping = aes(x = log2(value+1),
                         group = Sample_ID,
                         color = Data_type
                         )
           ) +
    geom_density(size = 1) +
    xlab("log2(count + 1)") +
    ggtitle(label = "Density distribution of raw counts")+
    scale_color_manual(values = c("grey","red")) +
    theme_bw() +
    theme(text = element_text(size = 30)
          ) -> p
#pdf(paste0(pathToFigures,"density_murine_Curie_R26_integratedR26tumCells.pdf"), width=12, height=12)
p
#dev.off()

```

## Filtering out 0 lines

```{r remove_zeroLines}

# Remove all lines where counts are all equal to zero
matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ <- matrix_Curie_R26_integratedR26tumCells_rawCounts[apply(matrix_Curie_R26_integratedR26tumCells_rawCounts, 1, function(x) !all(x==0)),]

head(matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ)
nrow(matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ)
ncol(matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ)

dim(matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ)

```

## Exploratory analysis 

### Counts normalization with rlog transformation (DESeq2) 

```{r rlog}

# Create DESeq object for each sample (dds)
dds_Curie_R26_integratedR26tumCells <- DESeqDataSetFromMatrix(countData = matrix_Curie_R26_integratedR26tumCells_rawCounts_WZ,
                                                            colData   = sample_Curie_R26_R26P87_R26P27_R26P26,
                                                            design    = ~1
                                                            )

# Stabilize variance using rlog
rld_Curie_R26_integratedR26tumCells        <- rlog(dds_Curie_R26_integratedR26tumCells, blind=TRUE)
matrix_rld_Curie_R26_integratedR26tumCells <- assay(rld_Curie_R26_integratedR26tumCells)

head(matrix_rld_Curie_R26_integratedR26tumCells[,1:4])
nrow(matrix_rld_Curie_R26_integratedR26tumCells)
ncol(matrix_rld_Curie_R26_integratedR26tumCells)

```

## Data distribution of rlog data

```{r data_distribution}

rlog_thrs <- 7.5

matrix_rld_Curie_R26_integratedR26tumCells %>%
    melt(.) %>%
    merge(x = .,
          y = sample_Curie_R26_R26P87_R26P27_R26P26,
          by.x = "Var2",
          by.y = "Sample_name"
          ) %>%
    ggplot(mapping = aes(x = value,
                         group = Sample_ID,
                         color = Data_type
                         )
       ) +
    geom_density(size = 1) +
    xlab("rlog") +
    ggtitle(label = "Density distribution of norm. counts")+
    scale_color_manual(values = c("grey","red")) +
    geom_vline(xintercept = rlog_thrs, size = 1, color = "green", lty = 2) +
    theme_bw() +
    theme(text = element_text(size = 30)
          ) -> p
#pdf(paste0(pathToFigures,"density_murine_Curie_R26_integratedR26tumCells_rlog.pdf"), width=12, height=12)
p
#dev.off()

```

### Hierarchical clustering and heatmap

```{r heatmap}

# Filtering
matrix_rld_Curie_R26_integratedR26tumCells_filt <- expFilter(data      = matrix_rld_Curie_R26_integratedR26tumCells,
                                                           threshold = 7.5,
                                                           p         = 0.01,
                                                           graph     = TRUE
                                                           )

#matrix_rld_Curie_R26_integratedR26tumCells_filt <- matrix_rld_Curie_R26_integratedR26tumCells
nrow(matrix_rld_Curie_R26_integratedR26tumCells_filt)

# Plot density
plot(density(matrix_rld_Curie_R26_integratedR26tumCells_filt))

# Check matrix colnames and sample plan sample name order
sample_Curie_R26_R26P87_R26P27_R26P26$Sample_name == colnames(matrix_rld_Curie_R26_integratedR26tumCells_filt)

# Most variable genes
mvgenes_Curie_R26_integratedR26tumCells <- genes.selection(data      = matrix_rld_Curie_R26_integratedR26tumCells_filt,
                                                         thres.num = 5000
                                                         )

# Subset matrix
matrix_rld_Curie_R26_integratedR26tumCells_filt_mvgenes <- matrix_rld_Curie_R26_integratedR26tumCells_filt[rownames(matrix_rld_Curie_R26_integratedR26tumCells_filt) %in% mvgenes_Curie_R26_integratedR26tumCells,]

# Top annotation
top_annot_Curie_R26_integratedR26tumCells <- createTopAnnot_func(sampleTable = sample_Curie_R26_R26P87_R26P27_R26P26,
                                                                 columns     = c("RNAseq_subgroup","Data_type"),
                                                                 colorsList  = list(group_name_col,data_type_col),
                                                                 show_legend = TRUE,
                                        #height = 4,
                                                                 fontsize = 20
                                                                 )

# Hierarchical clustering and heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_Curie_R26_integratedR26tumCells_mvgenes_plus.pdf"), width=14, height=14)
drawHeatmap_func(matrix               = matrix_rld_Curie_R26_integratedR26tumCells_filt_mvgenes,
                 column_title         = "Primary ATRT samples",
                 row_title            = "5000 most variable genes",
                 top_annot            = top_annot_Curie_R26_integratedR26tumCells,
                 row_annot            = NULL,
                 bottom_annot         = NULL,
                 legend_title         = "rlog(count)",
                 show_column_names    = TRUE,
                 show_row_names       = FALSE,
                 add_column_dend_axis = TRUE,
                 columns_dist         = "pearson",
                 columns_meth         = "ward",
                 rows_dist            = "euclidean",
                 rows_meth            = "ward",
                 fontsize             = 12,
                 split                = NULL,
                 annotation_legend_side = "right"
                 )
#dev.off()

# Plot PCA (PC1 and PC2)
#pdf(paste0(pathToFigures,"pca_murine_Curie_R26_integratedR26tumCells_RNAseqSubgroup.pdf"), width=16, height=12)
pca_func(matrix                     = matrix_rld_Curie_R26_integratedR26tumCells_filt_mvgenes,
         sampleTable                = sample_Curie_R26_R26P87_R26P27_R26P26,
         highlightedVar             = "RNAseq_subgroup",
         highlightedVarName         = "RNA-seq\nsubgroup",
         colors                     = group_name_col,
         sampleLabel                = "Patient_ID",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition  = "scRNAseq",
         fontsize                   = 30
         )
#dev.off()

# Plot PCA (PC1 and PC2)
#pdf(paste0(pathToFigures,"pca_murine_Curie_R26_integratedR26tumCells_dataType.pdf"), width=16, height=12)
pca_func(matrix                     = matrix_rld_Curie_R26_integratedR26tumCells_filt_mvgenes,
         sampleTable                = sample_Curie_R26_R26P87_R26P27_R26P26,
         highlightedVar             = "Data_type",
         highlightedVarName         = "Data type",
         colors                     = data_type_col,
         sampleLabel                = "Patient_ID",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition  = "scRNAseq",
         fontsize                   = 30
         )
#dev.off()

```


# Correlation study between human ATRT (RNA-seq) with known anatomical localization and R26 mouse bulk- and single cell RNA-seq data

## Combine human ATRT (rlog) with murine bulk- and single cell R26 RT (rlog)

```{r murine}

# Human sample plan
head(sample_humanATRT_groupedLoc)
colnames(sample_humanATRT_groupedLoc)

# Mouse sample plan
head(sample_Curie_R26_R26P87_R26P27_R26P26)
colnames(sample_Curie_R26_R26P87_R26P27_R26P26)

# Check order
colnames(sample_Curie_R26_R26P87_R26P27_R26P26) == colnames(sample_humanATRT_groupedLoc)

# Bind sample annotations
sample_murine26_bulkScRNAseq_humanATRT_groupedLoc <- rbind(sample_humanATRT_groupedLoc,
                                                           sample_Curie_R26_R26P87_R26P27_R26P26
                                                           )
head(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc)

# Merge human and mouse matrix
head(matrix_rld_humanATRT_groupedLoc)
dim(matrix_rld_humanATRT_groupedLoc)


head(matrix_rld_Curie_R26_integratedR26tumCells)
dim(matrix_rld_Curie_R26_integratedR26tumCells)

head(humanGeneName_mouseGeneName)

matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol <- merge(x = humanGeneName_mouseGeneName,
                                                                y = matrix_rld_Curie_R26_integratedR26tumCells,
                                                                by.x = "MGI.symbol",
                                                                by.y = 0,
                                                                all = FALSE
                                                                )
matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol <- matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol[!duplicated(matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol$HGNC.symbol),]
rownames(matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol) <- NULL
head(matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol)

# Put gene symbol to rownames
matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol <- column_to_rownames(matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol,
                                                                             var = "HGNC.symbol"
                                                                             )
matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol$MGI.symbol <- NULL
head(matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol)

# Merge matrices
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc <- merge(x   = matrix_rld_humanATRT_groupedLoc,
                                                              y   = matrix_rld_Curie_R26_integratedR26tumCells_humanSymbol,
                                                              by  = 0
                                                              )

matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc <- column_to_rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc,
                                                                           var = "Row.names"
                                                                           )

head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

```

# Filtering of combined matrix

```{r filtering}

# Filter matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt <- expFilter(data       = as.matrix(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc),
                                                                       threshold  = thrs,
                                                                       p          = 0.25,
                                                                       graph      = TRUE
                                                                       )
nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt)

# Melt matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_melt <- melt(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt)
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_melt_plus <- merge(x   = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc,
                                                                        y   = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_melt,
                                                                        by  = "Sample_name"
                                                                        )


# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26bulkScRNaseq_humanATRT_groupedLoc_filt.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    #geom_vline(xintercept = thrs,
     #          color = "red",
      #         lty = 3,
       #        lwd = 1
        #       ) +
    theme_bw()
#dev.off()

```

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc,
                                                                method  = "pearson"
                                                                )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Top annotation (mouse)
top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(RNAseq_subgroup = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$RNAseq_subgroup), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col,
                                                                             RNAseq_subgroup = group_name_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Convert sample plan into data frame (otherwise HeatmapAnnotation does not work well)
sample_murine26_bulkScRNAseq_humanATRT_groupedLoc <- data.frame(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc)

# Row annotation (human)
row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc[sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"]
                                                                           ,RNAseq_subgroup = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc[sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Organism == "Homo_sapiens","RNAseq_subgroup"]
                                                                           ,methArray_subgroup = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc[sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Organism == "Homo_sapiens","DKFZ_web_subgroup"]
                                                              ,
                                                               check.names = FALSE
                                                           ), 
                                                           col                            = list(anatomic_location = sub_sub_localization_of_primary_tumor_col
                                                               ,RNAseq_subgroup     = group_name_col
                                                              ,methArray_subgroup  = group_name_col
                                                           ),
                                                           gap                   = unit(2,"mm"),
                                                           show_annotation_name  = TRUE,
                                                           annotation_name_side  = "bottom",
                                                           annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                           which                 = "row",
                                                           show_legend           = c(FALSE,FALSE,FALSE)
                                                           )

data.frame(Data_type= sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Data_type)

# Bottom annotation (mouse)
bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Data_type= sample_murine26_bulkScRNAseq_humanATRT_groupedLoc[sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Organism == "Mus_musculus", "Data_type"]), 
                                                                             col                   = list(Data_type = data_type_col),
                                                                             gap                   = unit(2,"mm"),
                                                                             show_annotation_name  = FALSE,
                                                                             show_legend           = FALSE
                                                                             )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine_humanATRT_groupedLoc_mvGenes <- genes.selection(data       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc,
                                                       thres.num  = 1000,
                                                       probs      = 0.25
                                                       )

matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc[as.vector(murine_humanATRT_groupedLoc_mvGenes),]

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes,
                                                                        method  = "pearson"
                                                                        )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_mvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Most variable genes
humanATRT_groupedLoc_mvGenes<- genes.selection(data       = matrix_rld_humanATRT_groupedLoc,
                                               thres.num  = 1500,
                                               probs      = 0.25
                                               )

matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) %in% as.vector(humanATRT_groupedLoc_mvGenes),]
nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes,
                                                                 method  = "pearson"
                                                                 )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_humanMvgenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

##### Using immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) %in% as.vector(immuneGenes),]
nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes,
                                                                            method  = "pearson"
                                                                            )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_immuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_immuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

##### All genes but immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc[!(rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) %in% as.vector(immuneGenes)),]
nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes,
                                                   method  = "pearson"
                                                   )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_noImmuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

##### Most variable genes (after removing immune genes)

```{r correlation}

# Most variable genes
humanATRT_groupedLoc_noImmuneGenes_mvGenes <- genes.selection(data       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes,
                                                              thres.num  = 1000,
                                                              probs      = 0.25
                                                              )

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes[rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes) %in% humanATRT_groupedLoc_noImmuneGenes_mvGenes,]

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes,
                                                             method  = "pearson"
                                                             )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Import subgroup location top DEG
groupedLoc_topGenes <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/groupedLoc_topGenes_df.txt",
                                  header = TRUE
                                  )
head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG)

# Correlation matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor <- cor(x       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor_min <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseq_humanATRT_groupedLoc_groupedLocDEG_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseq_humanATRT_groupedLoc
        )
#dev.off()

```

### Unsupervised analysis after organism effect correction

#### Organism effect correction

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# PCA before data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_bulkScRNAseq_before_batchCor.pdf"), width=14, height=12)
pca_func(matrix          = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt,
         sampleTable     = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         sampleLabel = "Sample_name",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition = "scRNAseq",
         colors          = organism_col
         )
#dev.off()

# Data integration 
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor <- ComBat(dat = scale(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt),
                                                                             batch = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Organism
                                                                             )

# Melt matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt <- melt(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor)
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt_plus <- merge(x   = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc,
                                                                                      y   = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt,
                                                                                      by  = "Sample_name"
                                                                                      )

# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    theme_bw()
#dev.off()

# PCA after data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_bulkScRNAseq_after_batchCor.pdf"), width=14, height=12)
pca_func(matrix          = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor,
         sampleTable     = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         sampleLabel = "Sample_name",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition = "scRNAseq",
         colors          = organism_col
         )
#dev.off()

```

#### Hierarchical clustering using most variable genes

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor) == sample_murine26_bulkScRNAseq_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor <- genes.selection(data = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor,
                                                                  thres.num = 1000
                                                                  )

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_mvgenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor[murine26_humanATRT_groupedLoc_mvGenes_batchCor,]

# Top annotation  
top_annot_murine26_bulkScRNaseq_humanATRT_groupedLoc_batchCor <- HeatmapAnnotation(df                    = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc[,c("RNAseq_subgroup","groupedLoc","Organism")], 
                                                                                   col                   = list(RNAseq_subgroup=group_name_col, Organism = organism_col, groupedLoc = group_name_col),
                                                                                   gap                   = unit(2,"mm"),
                                                                                   show_annotation_name  = TRUE,
                                                                                   show_legend           = TRUE
                                                                                   )

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_bulkScRNaseq_batchCor_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_murine26_bulkScRNaseq_humanATRT_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using most variable genes after removing immune genes

```{r organism_effect_correction}

head(immuneGenes)

nrow(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor)

matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor[!rownames(matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor) %in% immuneGenes,]

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes <- genes.selection(data = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes,
                                                                                thres.num = 1000
                                                                                )

# Subset matrix
matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes <- matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes[murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes,]

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_bulkScRNaseq_batchCor_noImmuneGenes_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26bulkScRNaseq_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_murine26_bulkScRNaseq_humanATRT_groupedLoc_batchCor
        )
#dev.off()

```


# Compare integrated single cell R26 clusters to mouse and human bulk RNA-seq

## Import pseudo bulk from single cell RNA-seq

```{r import_pseudobulk}

# Import pseudo bulk of each sample
scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix <- read.csv(file       = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/scRNAseq/scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_matrix.csv"
         )

scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix <- column_to_rownames(scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix, var = "gene")

head(scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix)

```

## Append R26 pseudo-bulk 

```{r import_pseudobulk}

# Create sample plan for integrated R26 scRNAseq cluster

data.frame(Sample_name = colnames(scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix))

sample_Curie_R26

# Append murine R26 sample plan
sample_Curie_R26_bulkScRNAseqClusters <- merge(x = sample_Curie_R26,
                                               y = data.frame(Sample_name = colnames(scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix)),
                                               by = "Sample_name",
                                               all = TRUE
                                               )


# Fill some columns
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Genotype),"Genotype"] <- "R26"
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Data_type),"Data_type"] <- "scRNAseq"
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Organism),"Organism"] <- "Mus_musculus"
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Localization_of_primary_tumor),"Localization_of_primary_tumor"] <- "IC"
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Data_source),"Data_source"] <- "Curie"
sample_Curie_R26_bulkScRNAseqClusters[is.na(sample_Curie_R26_bulkScRNAseqClusters$Tumor_type),"Tumor_type"] <- "RT"

sample_Curie_R26_bulkScRNAseqClusters

# Data summary
group_by(.data = sample_Curie_R26_bulkScRNAseqClusters,
         Genotype,
         Data_type
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Append ATRT matrix
head(matrix_rawCounts_Curie_R26_mouseSymbol[,1:4])

matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts <- merge(x     = matrix_rawCounts_Curie_R26_mouseSymbol,
                                                                  y     = scRNAseq_Seurat_integrated_R26_tumCells_clusters_pseudoCounts_matrix,
                                                                  by  = 0,
                                                                  all   = FALSE
                                                                  )

head(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts[,1:4])

matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts <- column_to_rownames(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts,
                                                                                 var = "Row.names"
                                                                                 )

head(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts)
dim(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts)

# Check order
sample_Curie_R26_bulkScRNAseqClusters$Sample_name == colnames(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts)

matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts <- matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts[,sample_Curie_R26_bulkScRNAseqClusters$Sample_name]

sample_Curie_R26_bulkScRNAseqClusters$Sample_name == colnames(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts)

```

## Data distribution before filtering

```{r data_distribution}

sample_Curie_R26_bulkScRNAseqClusters$Sample_name

matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts %>%
    melt(.) %>%
    merge(x = .,
          y = sample_Curie_R26_bulkScRNAseqClusters,
          by.x = "variable",
          by.y = "Sample_name"
          ) %>%
    ggplot(mapping = aes(x = log2(value+1),
                         group = variable,
                         color = Data_type
                         )
           ) +
    geom_density(size = 1) +
    xlab("log2(count + 1)") +
    ggtitle(label = "Density distribution of raw counts")+
    scale_color_manual(values = c("grey","red")) +
    theme_bw() +
    theme(text = element_text(size = 30)
          ) -> p
#pdf(paste0(pathToFigures,"density_murine_Curie_R26_integratedR26tumCellsClusters.pdf"), width=12, height=12)
p
#dev.off()

```

## Filtering out 0 lines

```{r remove_zeroLines}

# Remove all lines where counts are all equal to zero
matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ <- matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts[apply(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts, 1, function(x) !all(x==0)),]

head(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ)
nrow(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ)
ncol(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ)

dim(matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ)

```

## Exploratory analysis 

### Counts normalization with rlog transformation (DESeq2) 

```{r rlog}

# Create DESeq object for each sample (dds)
dds_Curie_R26_integratedR26tumCellsClusters <- DESeqDataSetFromMatrix(countData = matrix_Curie_R26_integratedR26tumCellsClusters_rawCounts_WZ,
                                                            colData   = sample_Curie_R26_bulkScRNAseqClusters,
                                                            design    = ~1
                                                            )

# Stabilize variance using rlog
rld_Curie_R26_integratedR26tumCellsClusters        <- rlog(dds_Curie_R26_integratedR26tumCellsClusters, blind=TRUE)
matrix_rld_Curie_R26_integratedR26tumCellsClusters <- assay(rld_Curie_R26_integratedR26tumCellsClusters)

head(matrix_rld_Curie_R26_integratedR26tumCellsClusters[,1:4])
nrow(matrix_rld_Curie_R26_integratedR26tumCellsClusters)
ncol(matrix_rld_Curie_R26_integratedR26tumCellsClusters)

```

## Data distribution of rlog data

```{r data_distribution}

rlog_thrs <- 7.5

matrix_rld_Curie_R26_integratedR26tumCellsClusters %>%
    melt(.) %>%
    merge(x = .,
          y = sample_Curie_R26_bulkScRNAseqClusters,
          by.x = "Var2",
          by.y = "Sample_name"
          ) %>%
     ggplot(mapping = aes(x = value,
                         group = Var2,
                         color = Data_type
                         )
       ) +
    geom_density(size = 1) +
    xlab("rlog") +
    ggtitle(label = "Density distribution of norm. counts")+
    scale_color_manual(values = c("grey","red")) +
    geom_vline(xintercept = rlog_thrs, size = 1, color = "green", lty = 2) +
    theme_bw() +
    theme(text = element_text(size = 30)
          ) -> p
#pdf(paste0(pathToFigures,"density_murine_Curie_R26_integratedR26tumCellsClusters_rlog.pdf"), width=12, height=12)
p
#dev.off()

```

### Hierarchical clustering and heatmap

```{r heatmap}

# Filtering
matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt <- expFilter(data      = matrix_rld_Curie_R26_integratedR26tumCellsClusters,
                                                                     threshold = 7.5,
                                                                     p         = 0.01,
                                                                     graph     = TRUE
                                                                     )

#matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt <- matrix_rld_Curie_R26_integratedR26tumCellsClusters
nrow(matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt)

# Plot density
plot(density(matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt))

# Check matrix colnames and sample plan sample name order
sample_Curie_R26_bulkScRNAseqClusters$Sample_name == colnames(matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt)

# Most variable genes
mvgenes_Curie_R26_integratedR26tumCellsClusters <- genes.selection(data      = matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt,
                                                         thres.num = 5000
                                                         )

# Subset matrix
matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt_mvgenes <- matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt[rownames(matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt) %in% mvgenes_Curie_R26_integratedR26tumCellsClusters,]

# Top annotation
top_annot_Curie_R26_integratedR26tumCellsClusters <- createTopAnnot_func(sampleTable = sample_Curie_R26_bulkScRNAseqClusters,
                                                                         columns     = c("RNAseq_subgroup","Data_type"),
                                                                         colorsList  = list(group_name_col,data_type_col),
                                                                         show_legend = TRUE,
                                        #height = 4,
                                                                         fontsize = 20
                                                                         )

# Hierarchical clustering and heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_Curie_R26_integratedR26tumCellsClusters_mvgenes_plus.pdf"), width=14, height=14)
drawHeatmap_func(matrix               = matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt_mvgenes,
                 column_title         = "Primary ATRT samples",
                 row_title            = "5000 most variable genes",
                 top_annot            = top_annot_Curie_R26_integratedR26tumCellsClusters,
                 row_annot            = NULL,
                 bottom_annot         = NULL,
                 legend_title         = "rlog(count)",
                 show_column_names    = TRUE,
                 show_row_names       = FALSE,
                 add_column_dend_axis = TRUE,
                 columns_dist         = "pearson",
                 columns_meth         = "ward",
                 rows_dist            = "euclidean",
                 rows_meth            = "ward",
                 fontsize             = 12,
                 split                = NULL,
                 annotation_legend_side = "right"
                 )
#dev.off()

# Plot PCA (PC1 and PC2)
#pdf(paste0(pathToFigures,"pca_murine_Curie_R26_integratedR26tumCellsClusters_RNAseqSubgroup.pdf"), width=16, height=12)
pca_func(matrix                     = matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt_mvgenes,
         sampleTable                = sample_Curie_R26_bulkScRNAseqClusters,
         highlightedVar             = "RNAseq_subgroup",
         highlightedVarName         = "RNA-seq\nsubgroup",
         colors                     = group_name_col,
         sampleLabel                = "Patient_ID",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition  = "scRNAseq",
         fontsize                   = 30
         )
#dev.off()

# Plot PCA (PC1 and PC2)
#pdf(paste0(pathToFigures,"pca_murine_Curie_R26_integratedR26tumCellsClusters_dataType.pdf"), width=16, height=12)
pca_func(matrix                     = matrix_rld_Curie_R26_integratedR26tumCellsClusters_filt_mvgenes,
         sampleTable                = sample_Curie_R26_bulkScRNAseqClusters,
         highlightedVar             = "Data_type",
         highlightedVarName         = "Data type",
         colors                     = data_type_col,
         sampleLabel                = "Patient_ID",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition  = "scRNAseq",
         fontsize                   = 30
         )
#dev.off()

```


# Correlation study between human ATRT (RNA-seq) with known anatomical localization and R26 mouse bulk- and single cell RNA-seq cell population (clusters) data

## Combine human ATRT (rlog) with murine bulk- and single cell cluster R26 RT (rlog)

```{r murine}

# Human sample plan
head(sample_humanATRT_groupedLoc)
colnames(sample_humanATRT_groupedLoc)

# Mouse sample plan
head(sample_Curie_R26_bulkScRNAseqClusters)

colnames(sample_Curie_R26_bulkScRNAseqClusters)

sample_Curie_R26_bulkScRNAseqClusters <- relocate(sample_Curie_R26_bulkScRNAseqClusters,
                                                  Sample_name,
                                                  .after = Treatment
                                                  )


# Check order
colnames(sample_Curie_R26_bulkScRNAseqClusters) == colnames(sample_humanATRT_groupedLoc)

# Bind sample annotations
sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc <- rbind(sample_humanATRT_groupedLoc,
                                                           sample_Curie_R26_bulkScRNAseqClusters
                                                           )
head(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc)

# Merge human and mouse matrix
head(matrix_rld_humanATRT_groupedLoc)
dim(matrix_rld_humanATRT_groupedLoc)


head(matrix_rld_Curie_R26_integratedR26tumCellsClusters)
dim(matrix_rld_Curie_R26_integratedR26tumCellsClusters)

head(humanGeneName_mouseGeneName)

matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol <- merge(x = humanGeneName_mouseGeneName,
                                                                y = matrix_rld_Curie_R26_integratedR26tumCellsClusters,
                                                                by.x = "MGI.symbol",
                                                                by.y = 0,
                                                                all = FALSE
                                                                )
matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol <- matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol[!duplicated(matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol$HGNC.symbol),]
rownames(matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol) <- NULL
head(matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol)

# Put gene symbol to rownames
matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol <- column_to_rownames(matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol,
                                                                             var = "HGNC.symbol"
                                                                             )
matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol$MGI.symbol <- NULL
head(matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol)

# Merge matrices
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc <- merge(x   = matrix_rld_humanATRT_groupedLoc,
                                                              y   = matrix_rld_Curie_R26_integratedR26tumCellsClusters_humanSymbol,
                                                              by  = 0
                                                              )

matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc <- column_to_rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc,
                                                                           var = "Row.names"
                                                                           )

head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

```

# Filtering of combined matrix

```{r filtering}

# Filter matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt <- expFilter(data       = as.matrix(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc),
                                                                       threshold  = thrs,
                                                                       p          = 0.25,
                                                                       graph      = TRUE
                                                                       )
nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt)

# Melt matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_melt <- melt(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt)
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_melt_plus <- merge(x   = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
                                                                        y   = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_melt,
                                                                        by  = "Sample_name"
                                                                        )

# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    theme_bw()
#dev.off()

```

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc,
                                                                method  = "pearson"
                                                                )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor[filter(sample_murine26_bulkScRNAseq_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                       ]

# Top annotation (mouse)
top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(RNAseq_subgroup = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$RNAseq_subgroup), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col,
                                                                             RNAseq_subgroup = group_name_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Convert sample plan into data frame (otherwise HeatmapAnnotation does not work well)
sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc <- data.frame(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc)

# Row annotation (human)
row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc[sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"]
                                                                           ,RNAseq_subgroup = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc[sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Organism == "Homo_sapiens","RNAseq_subgroup"]
                                                                           ,methArray_subgroup = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc[sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Organism == "Homo_sapiens","DKFZ_web_subgroup"]
                                                              ,
                                                               check.names = FALSE
                                                           ), 
                                                           col                            = list(anatomic_location = sub_sub_localization_of_primary_tumor_col
                                                               ,RNAseq_subgroup     = group_name_col
                                                              ,methArray_subgroup  = group_name_col
                                                           ),
                                                           gap                   = unit(2,"mm"),
                                                           show_annotation_name  = TRUE,
                                                           annotation_name_side  = "bottom",
                                                           annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                           which                 = "row",
                                                           show_legend           = c(FALSE,FALSE,FALSE)
                                                           )

data.frame(Data_type= sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Data_type)

# Bottom annotation (mouse)
bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Data_type= sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc[sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Organism == "Mus_musculus", "Data_type"]), 
                                                                             col                   = list(Data_type = data_type_col),
                                                                             gap                   = unit(2,"mm"),
                                                                             show_annotation_name  = FALSE,
                                                                             show_legend           = FALSE
                                                                             )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine_humanATRT_groupedLoc_mvGenes <- genes.selection(data       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc,
                                                       thres.num  = 1000,
                                                       probs      = 0.25
                                                       )

matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc[as.vector(murine_humanATRT_groupedLoc_mvGenes),]

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes,
                                                                        method  = "pearson"
                                                                        )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_mvgenes_corr.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Most variable genes
humanATRT_groupedLoc_mvGenes<- genes.selection(data       = matrix_rld_humanATRT_groupedLoc,
                                               thres.num  = 1500,
                                               probs      = 0.25
                                               )

matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) %in% as.vector(humanATRT_groupedLoc_mvGenes),]
nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes,
                                                                 method  = "pearson"
                                                                 )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_humanMvgenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

##### Using immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) %in% as.vector(immuneGenes),]
nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes,
                                                                            method  = "pearson"
                                                                            )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_immuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_immuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

##### All genes but immune genes

```{r correlation}

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc[!(rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) %in% as.vector(immuneGenes)),]
nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes)

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes,
                                                   method  = "pearson"
                                                   )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_noImmuneGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

##### Most variable genes (after removing immune genes)

```{r correlation}

# Most variable genes
humanATRT_groupedLoc_noImmuneGenes_mvGenes <- genes.selection(data       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes,
                                                              thres.num  = 1000,
                                                              probs      = 0.25
                                                              )

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes[rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes) %in% humanATRT_groupedLoc_noImmuneGenes_mvGenes,]

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes,
                                                             method  = "pearson"
                                                             )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_noImmuneGenes_mvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Import subgroup location top DEG
groupedLoc_topGenes <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/report/groupedLoc_topGenes_df.txt",
                                  header = TRUE
                                  )
head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc[rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG)

# Correlation matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor <- cor(x       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG,
                                                     method  = "pearson"
                                                     )
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor)

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name
rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor_min <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor[filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc_groupedLocDEG_corr.pdf"), width=16, height=12)
Heatmap(matrix                       = t(scale(t(scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine26_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc),
        column_order                 = order(filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Homo_sapiens")$groupedLoc,
        column_split                 = filter(sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        top_annotation               = top_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
        bottom_annotation            = bottom_annot_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc
        )
#dev.off()

```

### Unsupervised analysis after organism effect correction

#### Organism effect correction

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# PCA before data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_bulkScRNAseqClusters_before_batchCor.pdf"), width=14, height=12)
pca_func(matrix          = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt,
         sampleTable     = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         sampleLabel = "Sample_name",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition = "scRNAseq",
         colors          = organism_col
         )
#dev.off()

# Data integration 
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor <- ComBat(dat = scale(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt),
                                                                             batch = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Organism
                                                                             )

# Melt matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt <- melt(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor)
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt)

# Add sample annotation to melted matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt_plus <- merge(x   = sample_murine26_bulkScRNAseq_humanATRT_groupedLoc,
                                                                                      y   = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt,
                                                                                      by  = "Sample_name"
                                                                                      )

# Plot densities
#pdf(file = paste0(pathToFigures, "density_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor.pdf"), width=16, height=12)
ggplot() +
    geom_density(data = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    theme_bw()
#dev.off()

# PCA after data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine26_bulkScRNAseqClusters_after_batchCor.pdf"), width=14, height=12)
pca_func(matrix          = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor,
         sampleTable     = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc,
         highlightedVar  = "Organism",
         sampleLabel = "Sample_name",
         sampleLabelColumnCondition = "Data_type",
         sampleLabelValueCondition = "scRNAseq",
         colors          = organism_col
         )
#dev.off()

```

#### Hierarchical clustering using most variable genes

```{r organism_effect_correction}

# Check order
colnames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor) == sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc$Sample_name

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor <- genes.selection(data = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor,
                                                                  thres.num = 1000
                                                                  )

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_mvgenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor[murine26_humanATRT_groupedLoc_mvGenes_batchCor,]

# Top annotation  
top_annot_murine26_bulkScRNaseqClusters_humanATRT_groupedLoc_batchCor <- HeatmapAnnotation(df                    = sample_murine26_bulkScRNAseqClusters_humanATRT_groupedLoc[,c("RNAseq_subgroup","groupedLoc","Organism")], 
                                                                                   col                   = list(RNAseq_subgroup=group_name_col, Organism = organism_col, groupedLoc = group_name_col),
                                                                                   gap                   = unit(2,"mm"),
                                                                                   show_annotation_name  = TRUE,
                                                                                   show_legend           = TRUE
                                                                                   )

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_bulkScRNaseqClusters_batchCor_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_murine26_bulkScRNaseqClusters_humanATRT_groupedLoc_batchCor
        )
#dev.off()

```

#### Hierarchical clustering using most variable genes after removing immune genes

```{r organism_effect_correction}

head(immuneGenes)

nrow(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor)

matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor[!rownames(matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor) %in% immuneGenes,]

# Most variable genes
murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes <- genes.selection(data = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes,
                                                                                thres.num = 1000
                                                                                )

# Subset matrix
matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes <- matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes[murine26_humanATRT_groupedLoc_mvGenes_batchCor_noImmuneGenes,]

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine26_bulkScRNaseqClusters_batchCor_noImmuneGenes_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_rld_murine26bulkScRNaseqClusters_humanATRT_groupedLoc_filt_batchCor_noImmuneGenes_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_murine26_bulkScRNaseqClusters_humanATRT_groupedLoc_batchCor
        )
#dev.off()

```























































### Select murine R26 and P0 data

```{r murine_RT_RNAseq}

head(sample_Curie_Vitte2017)
nrow(sample_Curie_Vitte2017)

sample_Curie_Vitte2017$Genotype

# Extract murine R26 and P0 sample
sample_murine_R26primTum_P0 <- filter(.data = sample_Curie_Vitte2017,
                                      Genotype %in% c("R26","P0-CreC_Smarcb1FloxFlox"),
                                      Tumor_origin != "allograft_flanc"
                                      )

# Subset murine matrix
rawCounts_murine_R26primTum_P0 <- rawCounts_Curie_Vitte2017[,sample_murine_R26primTum_P0$Sample_name]

head(rawCounts_murine_R26primTum_P0)
dim(rawCounts_murine_R26primTum_P0)

# Check order
colnames(rawCounts_murine_R26primTum_P0) == sample_murine_R26primTum_P0$Sample_name

```

### Convert mouse gene ID into human gene name

```{r murine_RT_RNAseq}

head(rawCounts_murine_R26primTum_P0[,1:4])

head(humanGeneName_mouseGeneName)

# Convert mouse gene name into human gene name
rawCounts_murine_R26primTum_P0_plus <- merge(x     = rawCounts_murine_R26primTum_P0,
                                             y     = humanGeneName_mouseGeneName,
                                             by.x  = 0,
                                             by.y  = "MGI.symbol",
                                             all   = FALSE
                                             )

head(rawCounts_murine_R26primTum_P0_plus[,1:4])

# Remove duplicated symbol
rawCounts_murine_R26primTum_P0_plus <- rawCounts_murine_R26primTum_P0_plus[!duplicated(rawCounts_murine_R26primTum_P0_plus$HGNC.symbol),]

# Put human symbol as rownames
rownames(rawCounts_murine_R26primTum_P0_plus) <- rawCounts_murine_R26primTum_P0_plus$HGNC.symbol

head(rawCounts_murine_R26primTum_P0_plus[,1:4])
dim(rawCounts_murine_R26primTum_P0_plus)

# Remove extra-columns
rawCounts_murine_R26primTum_P0_plus$Row.names <- NULL
rawCounts_murine_R26primTum_P0_plus$HGNC.symbol <- NULL

head(rawCounts_murine_R26primTum_P0_plus)
nrow(rawCounts_murine_R26primTum_P0_plus)

# Remove lines with 0 counts in all samples
rawCounts_murine_R26primTum_P0_WZ <- rawCounts_murine_R26primTum_P0_plus[apply(rawCounts_murine_R26primTum_P0_plus, 1, function(x) !all(x==0)),]
nrow(rawCounts_murine_R26primTum_P0_WZ)

# Check order
colnames(rawCounts_murine_R26primTum_P0_WZ) == sample_murine_R26primTum_P0$Sample_name

# Create DESeq object for each sample (dds)
dds_murine_R26primTum_P0 <- DESeqDataSetFromMatrix(countData = rawCounts_murine_R26primTum_P0_WZ,
                                                   colData   = sample_murine_R26primTum_P0,
                                                   design    = ~ 1
                                                   )

# Stabilize variance using rlog
rld_murine_R26primTum_P0 <- rlog(object = dds_murine_R26primTum_P0,
                                 blind  = TRUE
                                 )
matrix_rld_murine_R26primTum_P0 <- assay(rld_murine_R26primTum_P0)

head(matrix_rld_murine_R26primTum_P0)


```

# Combine human and mouse data

```{r combine_matrix}

head(sample_Curie_ATRT_primTum_groupedLoc_noNA)
colnames(sample_Curie_ATRT_primTum_groupedLoc_noNA)

head(sample_murine_R26primTum_P0)
colnames(sample_murine_R26primTum_P0)

# Add columns to murine sample plan
sample_murine_R26primTum_P0_plus <- mutate(.data                = sample_murine_R26primTum_P0,
                                           affy_subgroup        = NA,
                                           RNAseq_subgroup      = group_name,
                                           methArray_pearson    = NA,
                                           methArray_euclidean  = NA,
                                           DKFZ_web_subgroup    = NA,
                                           DKFZ_web_score       = NA,
                                           groupedLoc           = NA,
                                           .keep                = "unused"
                                           )

colnames(sample_murine_R26primTum_P0_plus)

# Bind sample annotations
sample_humanATRT_murine_groupedLoc <- rbind(sample_Curie_ATRT_primTum_groupedLoc_noNA,
                                            sample_murine_R26primTum_P0_plus
                                            )
head(sample_humanATRT_murine_groupedLoc)

# Merge human and mouse matrix
matrix_humanATRT_murine_groupedLoc <- merge(x   = matrix_rld_humanATRT_groupedLoc,
                                            y   = matrix_rld_murine_R26primTum_P0,
                                            by  = 0
                                            )
head(matrix_humanATRT_murine_groupedLoc)

rownames(matrix_humanATRT_murine_groupedLoc) <- matrix_humanATRT_murine_groupedLoc$Row.names
matrix_humanATRT_murine_groupedLoc$Row.names <- NULL

head(matrix_humanATRT_murine_groupedLoc)
dim(matrix_humanATRT_murine_groupedLoc)

# Check order
colnames(matrix_humanATRT_murine_groupedLoc) == sample_humanATRT_murine_groupedLoc$Sample_name

```

# Distribution of combined matrix

```{r distribution}

# Melt matrix
matrix_humanATRT_murine_groupedLoc_melt <- melt(as.matrix(matrix_humanATRT_murine_groupedLoc))
colnames(matrix_humanATRT_murine_groupedLoc_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_humanATRT_murine_groupedLoc_melt)

# Add sample annotation to melted matrix
matrix_humanATRT_murine_groupedLoc_melt_plus <- merge(x   = sample_humanATRT_murine_groupedLoc,
                                                      y   = matrix_humanATRT_murine_groupedLoc_melt,
                                                      by  = "Sample_name"
                                                      )

thrs <- 6

# Plot densities
ggplot() +
    geom_density(data = matrix_humanATRT_murine_groupedLoc_melt_plus,
                 mapping = aes(x = rlogExp),
                 color = "red"
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_groupedLoc_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_groupedLoc_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()

```

# Filtering of combined matrix

```{r filtering}

# Filter matrix
matrix_humanATRT_murine_groupedLoc_filt <- expFilter(data       = as.matrix(matrix_humanATRT_murine_groupedLoc),
                                              threshold  = thrs,
                                              p          = 0.25,
                                              graph      = FALSE
                                              )
nrow(matrix_humanATRT_murine_groupedLoc_filt)

# Melt matrix
matrix_humanATRT_murine_groupedLoc_filt_melt <- melt(matrix_humanATRT_murine_groupedLoc_filt)
colnames(matrix_humanATRT_murine_groupedLoc_filt_melt) <- c("Gene","Sample_name","rlogExp")
head(matrix_humanATRT_murine_groupedLoc_filt_melt)

# Add sample annotation to melted matrix
matrix_humanATRT_murine_groupedLoc_filt_melt_plus <- merge(x   = sample_humanATRT_murine_groupedLoc,
                                                           y   = matrix_humanATRT_murine_groupedLoc_filt_melt,
                                                           by  = "Sample_name"
                                                           )

# Plot densities
ggplot() +
    geom_density(data = matrix_humanATRT_murine_groupedLoc_filt_melt_plus,
                 mapping = aes(x = rlogExp)
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_groupedLoc_filt_melt_plus, Organism == "Homo_sapiens"),
                 mapping = aes(x = rlogExp),
                 color = "blue"
                 ) +
    geom_density(data = filter(matrix_humanATRT_murine_groupedLoc_filt_melt_plus, Organism == "Mus_musculus"),
                 mapping = aes(x = rlogExp),
                 color = "darkgreen"
                 ) +
    geom_vline(xintercept = thrs,
               color = "red",
               lty = 3,
               lwd = 1
               ) +
    theme_bw()

```

### Sample-wise correlation

#### Using all filtered genes

```{r correlation}

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_filt) == sample_humanATRT_murine_groupedLoc$Sample_name

# Correlation matrix
matrix_humanATRT_murine_groupedLoc_filt_cor <- cor(x       = matrix_humanATRT_murine_groupedLoc_filt,
                                                   method  = "pearson"
                                                   )
head(matrix_humanATRT_murine_groupedLoc_filt_cor)

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_filt_cor) == sample_humanATRT_murine_groupedLoc$Sample_name
rownames(matrix_humanATRT_murine_groupedLoc_filt_cor) == sample_humanATRT_murine_groupedLoc$Sample_name

# Subset matrix
matrix_humanATRT_murine_groupedLoc_filt_cor_min <- matrix_humanATRT_murine_groupedLoc_filt_cor[filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                       filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                       ]


# Top annotation (mouse)
top_annot_humanATRT_murine_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Tumor_origin = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Tumor_origin,
                                                                                              Background_genotype = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Background_genotype
                                                                                              ), 
                                                col                   = list(Tumor_origin = tumor_origin_col,
                                                                             Background_genotype = background_genotype_col
                                                                             ),
                                                gap                   = unit(2,"mm"),
                                                show_annotation_name  = TRUE,
                                                show_legend           = FALSE,
                                                annotation_name_gp    = gpar(fontsize = 20, fontface = "bold"),
                                                na_col                = "white"
                                                )

# Row annotation (human)
row_annot_humanATRT_murine_groupedLoc <- HeatmapAnnotation(df = data.frame(anatomic_location = sample_humanATRT_murine_groupedLoc[sample_humanATRT_murine_groupedLoc$Organism == "Homo_sapiens","sub_sub_localization_of_primary_tumor"],
                                                                           methArray_subgroup = sample_humanATRT_murine_groupedLoc[sample_humanATRT_murine_groupedLoc$Organism == "Homo_sapiens","DKFZ_web_subgroup"],
                                                                                                                                           RNAseq_subgroup = sample_humanATRT_murine_groupedLoc[sample_humanATRT_murine_groupedLoc$Organism == "Homo_sapiens","RNAseq_subgroup"]
                                                                ), 
                                                col = list(anatomic_location = sub_sub_localization_of_primary_tumor_col,
                                                           RNAseq_subgroup = group_name_col,
                                                           methArray_subgroup = group_name_col
                                                           ),
                                                gap = unit(2,"mm"),
                                                show_annotation_name = TRUE,
                                                annotation_name_side = "bottom",
                                                annotation_name_gp = gpar(fontsize = 20, fontface = "bold"),
                                                which = "row",
                                                show_legend = c(FALSE,FALSE,FALSE)
                                                )

# Bottom annotation (mouse)
bottom_annot_humanATRT_murine_groupedLoc <- HeatmapAnnotation(df                    = data.frame(Mouse_subgroup = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype), 
                                                   col                   = list(Mouse_subgroup = genotype_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = FALSE,
                                                   show_legend           = FALSE
                                                   )

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_corr_groupedLoc.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_humanATRT_murine_groupedLoc_filt_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup),
        column_order                 = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup,
        column_split                 = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine_groupedLoc,
        #top_annotation               = top_annot_humanATRT_murine_groupedLoc,
        bottom_annotation            = bottom_annot_humanATRT_murine_groupedLoc
        )
#dev.off()

```

#### Using most variable genes

```{r correlation}

# Check order
colnames(matrix_humanATRT_murine_groupedLoc) == sample_humanATRT_murine_groupedLoc$Sample_name

# Most variable genes
mvGenes_groupedLoc <- genes.selection(data       = matrix_humanATRT_murine_groupedLoc,
                                      thres.num  = 1000,
                                      probs      = 0.25
                                      )

matrix_humanATRT_murine_groupedLoc_mvgenes <- matrix_humanATRT_murine_groupedLoc[as.vector(mvGenes_groupedLoc),]

# Correlation matrix
matrix_humanATRT_murine_groupedLoc_mvgenes_cor <- cor(x       = matrix_humanATRT_murine_groupedLoc_mvgenes,
                                                      method  = "pearson"
                                                      )
head(matrix_humanATRT_murine_groupedLoc_mvgenes_cor)

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_mvgenes_cor) == sample_humanATRT_murine_groupedLoc$Sample_name
rownames(matrix_humanATRT_murine_groupedLoc_mvgenes_cor) == sample_humanATRT_murine_groupedLoc$Sample_name

# Subset matrix
matrix_humanATRT_murine_groupedLoc_mvgenes_cor_min <- matrix_humanATRT_murine_groupedLoc_mvgenes_cor[filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_mvgenes_corr_groupedLoc.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup),
        column_order                 = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup,
        column_split                 = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine_groupedLoc,
        #top_annotation               = top_annot_humanATRT_murine_groupedLoc,
        bottom_annotation            = bottom_annot_humanATRT_murine_groupedLoc
        )
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Most variable genes
human_mvGenes_groupedLoc <- genes.selection(data       = matrix_rld_humanATRT_groupedLoc,
                                            thres.num  = 1500,
                                            probs      = 0.25
                                            )

matrix_humanATRT_murine_groupedLoc_humanMvGenes <- matrix_humanATRT_murine_groupedLoc[rownames(matrix_humanATRT_murine_groupedLoc) %in% as.vector(human_mvGenes),]
nrow(matrix_humanATRT_murine_groupedLoc_humanMvGenes)

# Correlation matrix
matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor <- cor(x       = matrix_humanATRT_murine_groupedLoc_humanMvGenes,
                                                           method  = "pearson"
                                                           )
head(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor)

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor) == sample_humanATRT_murine_groupedLoc$Sample_name
rownames(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor) == sample_humanATRT_murine_groupedLoc$Sample_name

# Subset matrix
matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_min <- matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor[filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                 filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                       ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_humanMvgenes_corr_groupedLoc.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        row_order                    = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup),
        column_order                 = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup,
        column_split                 = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine_groupedLoc,
        #top_annotation               = top_annot_humanATRT_murine_groupedLoc,
        bottom_annotation            = bottom_annot_humanATRT_murine_groupedLoc
        )
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

head(groupedLoc_topGenes)
nrow(groupedLoc_topGenes)

# Subset matrix
matrix_humanATRT_murine_groupedLocDEG <- matrix_humanATRT_murine_groupedLoc[rownames(matrix_humanATRT_murine_groupedLoc) %in% as.vector(groupedLoc_topGenes$DEG),]

nrow(matrix_humanATRT_murine_groupedLocDEG)

# Correlation matrix
matrix_humanATRT_murine_groupedLocDEG_cor <- cor(x       = matrix_humanATRT_murine_groupedLocDEG,
                                                 method  = "pearson"
                                                 )
head(matrix_humanATRT_murine_groupedLocDEG_cor)

# Check order
colnames(matrix_humanATRT_murine_groupedLocDEG_cor) == sample_humanATRT_murine_groupedLoc$Sample_name
rownames(matrix_humanATRT_murine_groupedLocDEG_cor) == sample_humanATRT_murine_groupedLoc$Sample_name

# Subset matrix
matrix_humanATRT_murine_groupedLocDEG_cor_min <- matrix_humanATRT_murine_groupedLocDEG_cor[filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$Sample_name,
                                                                                                   filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Sample_name
                                                                                                   ]

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_groupedLocDEG_corr_groupedLoc.pdf"), width=16, height=10)
Heatmap(matrix                       = t(scale(t(scale(matrix_humanATRT_murine_groupedLocDEG_cor_min)))), #t(scale(t(scale(matrix_humanATRT_murine_rld_filt_cor_min)))),
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = FALSE,
        show_column_names            = TRUE,
        row_order                    = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup),
        column_order                 = order(filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype),
        row_split                    = filter(sample_humanATRT_murine_groupedLoc, Organism == "Homo_sapiens")$RNAseq_subgroup,
        column_split                 = filter(sample_humanATRT_murine_groupedLoc, Organism == "Mus_musculus")$Genotype,
        show_heatmap_legend          = FALSE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        column_title_gp              = gpar(fontsize = 15, fontface="bold"),
        row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
        right_annotation             = row_annot_humanATRT_murine_groupedLoc,
        #top_annotation               = top_annot_humanATRT_murine_groupedLoc,
        bottom_annotation            = bottom_annot_humanATRT_murine_groupedLoc
        )
#dev.off()

```

### Group-wise correlation

#### Prepare sample plan

```{r correlation}

# Add R26-SHH and R26-MYC as well as P0 to sample plan
head(sample_humanATRT_murine_groupedLoc)

sample_humanATRT_murine_groupedLoc_plus <- mutate(.data    = sample_humanATRT_murine_groupedLoc,
                                                  subgroup = case_when(Sample_name %in% c("29050A","29048","38594A") ~ "R26-SHH",
                                                                       Genotype == "R26" & !(Sample_name %in% c("29050A","29048","38594A")) ~ "R26-MYC",
                                                                       Genotype == "P0-CreC_Smarcb1FloxFlox" ~ "P0",
                                                                       TRUE ~ groupedLoc
                                                                       )
                                                  )
head(sample_humanATRT_murine_groupedLoc_plus)
sample_humanATRT_murine_groupedLoc_plus$subgroup

```

#### Using all filtered genes

```{r group_wise_correlation_allGenes}

# Melt correlation matrix
head(matrix_humanATRT_murine_groupedLoc_filt_cor)
matrix_humanATRT_murine_groupedLoc_filt_cor_melt <- melt(matrix_humanATRT_murine_groupedLoc_filt_cor)
colnames(matrix_humanATRT_murine_groupedLoc_filt_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_humanATRT_murine_groupedLoc_filt_cor_melt)

# Add subgroup to melted matrix
matrix_humanATRT_murine_groupedLoc_filt_cor_melt_plus <- merge(x = matrix_humanATRT_murine_groupedLoc_filt_cor_melt,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_1",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_filt_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_humanATRT_murine_groupedLoc_filt_cor_melt_plus)
#
matrix_humanATRT_murine_groupedLoc_filt_cor_melt_Plus <- merge(x = matrix_humanATRT_murine_groupedLoc_filt_cor_melt_plus,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_2",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_filt_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_humanATRT_murine_groupedLoc_filt_cor_melt_Plus)


matrix_humanATRT_murine_groupedLoc_meanCorr <- group_by(.data = matrix_humanATRT_murine_groupedLoc_filt_cor_melt_Plus,
                                                         subgroup_1,
                                                         subgroup_2
                                                         ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_humanATRT_murine_groupedLoc_meanCorr)

# Spread table
matrix_humanATRT_murine_groupedLoc_meanCorr <- spread(data = matrix_humanATRT_murine_groupedLoc_meanCorr,
                                                      key = subgroup_2,
                                                      value = mean_corr
                                                      ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_humanATRT_murine_groupedLoc_meanCorr) <- matrix_humanATRT_murine_groupedLoc_meanCorr$subgroup_1
matrix_humanATRT_murine_groupedLoc_meanCorr$subgroup_1 <- NULL
matrix_humanATRT_murine_groupedLoc_meanCorr

# Subset matrix
matrix_humanATRT_murine_groupedLoc_meanCorr_min <- matrix_humanATRT_murine_groupedLoc_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH","P0")]

# Rename rownames
rownames(matrix_humanATRT_murine_groupedLoc_meanCorr_min) <- c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")

# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLoc.pdf"), width=16, height=10)
h <- Heatmap(matrix                       = matrix_humanATRT_murine_groupedLoc_meanCorr_min,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = TRUE,
        show_column_names            = TRUE,
        column_dend_side             = "bottom",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        heatmap_legend_param = list(title = "\nAverage Pearson's Correlation\n",
                                    direction = "horizontal",
                                    labels_gp = gpar(size = 20),
                                    title_gp = gpar(fontsize = 25, fontface = "bold"),
                                    legend_width= unit(20,"cm"),
                                    legend_height= unit(10,"cm")
                                    )
        )
draw(h, heatmap_legend_side = "bottom")
#dev.off()

paperFig_fontsize <- 6

# Heatmap (for figure)
humanATRT_murineRT_groupedLoc_meanCorr <- Heatmap(matrix                       = matrix_humanATRT_murine_groupedLoc_meanCorr_min,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = TRUE,
        show_column_names            = TRUE,
        show_column_dend            = TRUE,
        show_row_dend            = FALSE,
        column_dend_side             = "bottom",
        column_names_gp              = gpar(fontsize = paperFig_fontsize*1.5, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 45,
        row_names_gp                 = gpar(fontsize = paperFig_fontsize*1.5, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        heatmap_legend_param = list(title          = "\nAverage Pearson's Correlation",
                                    direction      = "horizontal",
                                    labels_gp      = gpar(fontsize = paperFig_fontsize),
                                    at             = c(0.45,0.50,0.55),
                                    title_gp       = gpar(fontsize = paperFig_fontsize*1.5, fontface = "bold"),
                                    legend_width   = unit(2.5,"cm"),
                                    legend_height  = unit(5,"cm")
                                    )
        )
humanATRT_murineRT_groupedLoc_meanCorr

# Save as RDS object (to be imported in figure 4)
saveRDS(object  = humanATRT_murineRT_groupedLoc_meanCorr,
        file    = "humanATRT_murineRT_groupedLoc_meanCorr.RDS"
        )      

```

#### Using most variable genes

```{r correlation}

# Melt correlation matrix
head(matrix_humanATRT_murine_groupedLoc_mvgenes_cor)
matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt <- melt(matrix_humanATRT_murine_groupedLoc_mvgenes_cor)
colnames(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt)

# Add subgroup to melted matrix
matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_plus <- merge(x = matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_1",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_plus)
#
matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_Plus <- merge(x = matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_plus,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_2",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_Plus)


matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes <- group_by(.data = matrix_humanATRT_murine_groupedLoc_mvgenes_cor_melt_Plus,
                                                                subgroup_1,
                                                                subgroup_2
                                                                ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes)

# Spread table
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes <- spread(data = matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes,
                                                              key = subgroup_2,
                                                              value = mean_corr
                                                              ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes) <- matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes$subgroup_1
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes$subgroup_1 <- NULL
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes

# Subset matrix
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes_min <- matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH","P0")]
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes_min


# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_mvgenes_avgCorr_groupedLoc.pdf"), width=16, height=10)
h <- Heatmap(matrix                       = matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes_min,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = TRUE,
        show_column_names            = TRUE,
        column_dend_side             = "bottom",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        heatmap_legend_param = list(title = "\nAverage Pearson's Correlation\n",
                                    direction = "horizontal",
                                    labels_gp = gpar(size = 20),
                                    title_gp = gpar(fontsize = 25, fontface = "bold"),
                                    legend_width= unit(20,"cm"),
                                    legend_height= unit(10,"cm")
                                    )
        )
draw(h, heatmap_legend_side = "bottom")
#dev.off()

```

#### Using human most variable genes

```{r correlation}

# Melt correlation matrix
head(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor)
matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt <- melt(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor)
colnames(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt)

# Add subgroup to melted matrix
matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_plus <- merge(x = matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_1",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_plus)
#
matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_Plus <- merge(x = matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_plus,
                                                               y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                               by.x = "Sample_name_2",
                                                               by.y = "Sample_name"
                                                               )
colnames(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_Plus)


matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes <- group_by(.data = matrix_humanATRT_murine_groupedLoc_humanMvGenes_cor_melt_Plus,
                                                                subgroup_1,
                                                                subgroup_2
                                                                ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes)

# Spread table 
matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes <- spread(data = matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes,
                                                              key = subgroup_2,
                                                              value = mean_corr
                                                              ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes) <- matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes$subgroup_1
matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes$subgroup_1 <- NULL
matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes

# Subset matrix
matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes_min <- matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH","P0")]
matrix_humanATRT_murine_groupedLoc_meanCorr_mvgenes_min


# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_humanMvGenes_avgCorr_groupedLoc.pdf"), width=16, height=10)
h <- Heatmap(matrix                       = matrix_humanATRT_murine_groupedLoc_meanCorr_humanMvGenes_min,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = TRUE,
        show_column_names            = TRUE,
        column_dend_side             = "bottom",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        heatmap_legend_param = list(title = "\nAverage Pearson's Correlation\n",
                                    direction = "horizontal",
                                    labels_gp = gpar(size = 20),
                                    title_gp = gpar(fontsize = 25, fontface = "bold"),
                                    legend_width= unit(20,"cm"),
                                    legend_height= unit(10,"cm")
                                    )
        )
draw(h, heatmap_legend_side = "bottom")
#dev.off()

```

#### Using subgroup location genes

```{r correlation}

# Melt correlation matrix
head(matrix_humanATRT_murine_groupedLocDEG_cor)
matrix_humanATRT_murine_groupedLocDEG_cor_melt <- melt(matrix_humanATRT_murine_groupedLocDEG_cor)
colnames(matrix_humanATRT_murine_groupedLocDEG_cor_melt) <- c("Sample_name_1","Sample_name_2","correlation")
head(matrix_humanATRT_murine_groupedLocDEG_cor_melt)

# Add subgroup to melted matrix
matrix_humanATRT_murine_groupedLocDEG_cor_melt_plus <- merge(x = matrix_humanATRT_murine_groupedLocDEG_cor_melt,
                                                             y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                             by.x = "Sample_name_1",
                                                             by.y = "Sample_name"
                                                             )
colnames(matrix_humanATRT_murine_groupedLocDEG_cor_melt_plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1")
head(matrix_humanATRT_murine_groupedLocDEG_cor_melt_plus)
#
matrix_humanATRT_murine_groupedLocDEG_cor_melt_Plus <- merge(x = matrix_humanATRT_murine_groupedLocDEG_cor_melt_plus,
                                                             y = sample_humanATRT_murine_groupedLoc_plus[,c("Sample_name","subgroup")],
                                                             by.x = "Sample_name_2",
                                                             by.y = "Sample_name"
                                                             )
colnames(matrix_humanATRT_murine_groupedLocDEG_cor_melt_Plus) <- c("Sample_name_1","Sample_name_2","correlation","subgroup_1","subgroup_2")
head(matrix_humanATRT_murine_groupedLocDEG_cor_melt_Plus)


matrix_humanATRT_murine_groupedLocDEG_meanCorr <- group_by(.data = matrix_humanATRT_murine_groupedLocDEG_cor_melt_Plus,
                                                                subgroup_1,
                                                                subgroup_2
                                                                ) %>%
    summarize(mean_corr = mean(correlation)
              )
head(matrix_humanATRT_murine_groupedLocDEG_meanCorr)

# Spread table 
matrix_humanATRT_murine_groupedLocDEG_meanCorr <- spread(data = matrix_humanATRT_murine_groupedLocDEG_meanCorr,
                                                         key = subgroup_2,
                                                         value = mean_corr
                                                         ) %>%
    data.frame(check.names = FALSE)
rownames(matrix_humanATRT_murine_groupedLocDEG_meanCorr) <- matrix_humanATRT_murine_groupedLocDEG_meanCorr$subgroup_1
matrix_humanATRT_murine_groupedLocDEG_meanCorr$subgroup_1 <- NULL
matrix_humanATRT_murine_groupedLocDEG_meanCorr

# Subset matrix
matrix_humanATRT_murine_groupedLocDEG_meanCorr_min <- matrix_humanATRT_murine_groupedLocDEG_meanCorr[c("MYC_MYC","MYC_SHH","SHH_SHH","TYR_TYR"),c("R26-MYC","R26-SHH","P0")]
matrix_humanATRT_murine_groupedLocDEG_meanCorr_min


# Heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_avgCorr_groupedLocDEG.pdf"), width=16, height=10)
h <- Heatmap(matrix                       = matrix_humanATRT_murine_groupedLocDEG_meanCorr_min,
        cluster_rows                 = TRUE,
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "euclidean",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        cluster_columns              = TRUE,
        show_row_names               = TRUE,
        show_column_names            = TRUE,
        column_dend_side             = "bottom",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = TRUE,
        cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        heatmap_legend_param = list(title = "\nAverage Pearson's Correlation\n",
                                    direction = "horizontal",
                                    labels_gp = gpar(size = 20),
                                    title_gp = gpar(fontsize = 25, fontface = "bold"),
                                    legend_width= unit(20,"cm"),
                                    legend_height= unit(10,"cm")
                                    )
        )
draw(h, heatmap_legend_side = "bottom")
#dev.off()

```

### Unsupervised analysis after organism effect correction

#### Organism effect correction

```{r organism_effect_correction}

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_filt) == sample_humanATRT_murine_groupedLoc_plus$Sample_name

# PCA before data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine_before_batchCor.pdf"), width=14, height=12)
pca_func(matrix = matrix_humanATRT_murine_groupedLoc_filt,
         sampleTable = sample_humanATRT_murine_groupedLoc_plus,
         highlightedVar = "Organism",
         colors = organism_col
         )
#dev.off()

# Data integration 
matrix_humanATRT_murine_groupedLoc_filt_batchCor <- ComBat(dat = scale(matrix_humanATRT_murine_groupedLoc_filt),
                                                           batch = sample_humanATRT_murine_groupedLoc_plus$Organism
                                                           )

# PCA after data integration
#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine_after_batchCor.pdf"), width=14, height=12)
pca_func(matrix = matrix_humanATRT_murine_groupedLoc_filt_batchCor,
         sampleTable = sample_humanATRT_murine_groupedLoc_plus,
         highlightedVar = "Organism",
         colors = organism_col
         )
#dev.off()

```

#### Hierarchical clustering

```{r organism_effect_correction}

# Check order
colnames(matrix_humanATRT_murine_groupedLoc_filt_batchCor) == sample_humanATRT_murine_groupedLoc_plus$Sample_name

# Most variable genes
mvGenes_batchCor <- genes.selection(data = matrix_humanATRT_murine_groupedLoc_filt_batchCor,
                                    thres.num = 1000
                                    )

# Subset matrix
matrix_humanATRT_murine_groupedLoc_filt_batchCor_mvgenes <- matrix_humanATRT_murine_groupedLoc_filt_batchCor[mvGenes_batchCor,]

# Top annotation  
top_annot_groupedLoc_batchCor <- HeatmapAnnotation(df                    = sample_humanATRT_murine_groupedLoc_plus[,c("subgroup","Organism")], 
                                                   col                   = list(subgroup = group_name_col, Organism = organism_col),
                                                   gap                   = unit(2,"mm"),
                                                   show_annotation_name  = TRUE,
                                                   show_legend           = TRUE
                                                   )

# Complex heatmap
#pdf(file = paste0(pathToFigures, "complexHeatmap_RNAseq_humanATRT_murine_batchCor_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                       = matrix_humanATRT_murine_groupedLoc_filt_batchCor_mvgenes,
        cluster_rows                 = TRUE,
        cluster_columns              = TRUE,
        column_dend_height          = unit(40, "mm"),
        clustering_distance_rows     = "euclidean",
        clustering_distance_columns  = "pearson",
        clustering_method_rows       = "ward",
        clustering_method_columns    = "ward",
        show_row_names               = FALSE,
        show_column_names            = FALSE,
        column_dend_side             = "top",
        column_names_gp              = gpar(fontsize = 30, fontface = "bold"),
        column_names_side            = "top",
        column_names_rot             = 0,
        row_names_gp                 = gpar(fontsize = 30, fontface = "bold"),
        show_heatmap_legend          = FALSE,
        #cluster_row_slices           = FALSE,
        cluster_column_slices        = FALSE,
        top_annotation               = top_annot_groupedLoc_batchCor
        )
#dev.off()

```

#### PCA

```{r pca_after_correction}

#pdf(file = paste0(pathToFigures, "pca_RNAseq_humanATRT_murine_batchCor_mvgenes.pdf"), width=14, height=12)
pca_func(matrix          = matrix_humanATRT_murine_groupedLoc_filt_batchCor_mvgenes,
         sampleTable     = sample_humanATRT_murine_groupedLoc_plus,
         highlightedVar  = "subgroup",
         colors          = group_name_col
         )
#dev.off()

```

#### t-SNE

```{r tsne_after_correction}

#pdf(file = paste0(pathToFigures, "tsne_RNAseq_humanATRT_murine_batchCor_mvgenes.pdf"), width=14, height=12)
tsne_func(matrix          = matrix_humanATRT_murine_groupedLoc_filt_batchCor_mvgenes,
          sampleTable     = sample_humanATRT_murine_groupedLoc_plus,
          highlightedVar  = "subgroup",
          colors          = group_name_col,
          perplexity      = 6,
          seed            = 6
          )
#dev.off()

```

#### UMAP

```{r umap_after_correction}

#pdf(file = paste0(pathToFigures, "umap_RNAseq_humanATRT_murine_batchCor_mvgenes.pdf"), width=14, height=12)
umap_func(matrix          = matrix_humanATRT_murine_groupedLoc_filt_batchCor_mvgenes,
          sampleTable     = sample_humanATRT_murine_groupedLoc_plus,
          highlightedVar  = "subgroup",
          colors          = group_name_col,
          n_neighbors     = 5,
          metric          = "pearson",
          seed            = 9
          )
#dev.off()

```
