---
title: "Rhabdoid tumors in mouse model"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    fig_width: 9
    fig_height: 8
    fig_caption: true
    df_print: paged
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r library_file_import, echo=FALSE}

# Paths
pathToHome                   <- "/bioinfo/users/mandrian/"
pathToRhabdoid_U830_projects <- paste0(pathToHome,"rhabdoid_U830_projects/")
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
#pathToRpackages              <- paste0(pathToProgrammes,"Rpackages/")
pathToRpackages              <- paste0(pathToProgrammes,"Rpackages_R.3.6/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToMurine_RT              <- paste0(pathToANALYSIS_PROJECTS,"murine_RT/")
pathToMurine_RT_info         <- paste0(pathToMurine_RT,"svn/analyse/info/") 
pathToData                   <- paste0(pathToMurine_RT,"data/") 
pathToFigures                <- paste0(pathToMurine_RT,"svn/analyse/script/figures/") 

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r load_library}

#.libPaths(pathToRpackages)

source("load_affy_library.R")

```

```{r load_saved_RData, eval=TRUE}

#load(".RData_affy")

load("murine_RT_affy.RData")

#save.image(file = "murine_RT_affy.RData")

```

```{r load_plot_function}

# Load external files
source("/bioinfo/users/mandrian/rhabdoid_U830_projects/programmes/plot_functions.R")
source("plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")

```

```{r other_functions}

# PVAC density plot
plot_pvac_filtering <- function(ft)
    {
        ggplot(data = as.data.frame(ft$pvac[ft$nullset]), aes(x = ft$pvac[ft$nullset])) +
            geom_density(fill = "gray75", colour = "white", show.legend = TRUE) +
                geom_density(data = as.data.frame(ft$pvac), aes(x = ft$pvac), colour = "blue", show.legend = TRUE) +
                                        #geom_density(data = as.data.frame(ft$pvac[ft$aset]), aes(x = ft$pvac[ft$aset]), colour = "green") +
                    geom_vline(xintercept = ft$cutoff, colour = "red", linetype = "longdash") +
                        xlab("PVAC score") +
                            theme(legend.position  = "right",
                                  legend.text      = element_text(face="bold", color="black", size=14),
                                  legend.title     = element_blank(),
                                  axis.text.x      = element_text(face="bold", color="black", size=12),
                                  axis.text.y      = element_text(face="bold", color="black", size=12),
                                  axis.title.x     = element_text(face="bold", color="black", size=16),
                                  axis.title.y     = element_text(face="bold", color="black", size=16),
                                  axis.line.x      = element_line(color="grey", size=0.5),
                                  axis.line.y      = element_line(color="grey", size=0.5),
                                  panel.background = element_blank()
                                  )
    }

```

This file contains the analysis of RNA expression from affymetrix data of malignant rhabdoid tumors (MRT) in mouse model.
The study is the continuation of the works published in [Han et al., 2016]. This paper shows, for the first time, the possibility to develop intracranial MRT in mouse by time-dependent inactivation of SMARCB1. Cells of origin has been also addressed. Using additional data and genotype, we continue to characterize the MRT in mouse model.

This analysis report is devided in 4 sections:

* Samples
* Data pre-processing
* Exploratory analysis
* Differential expression analysis
* Specific and functional analysis

# Samples

Expression data were generated using the affymetrix technology with the Moe30_2 array type. Details about samples are shown in the table below. Primary tumors, allograft and orthotopic grafts were used during the study.

```{r data_file_annotation}

# Import data file annotation (file containing all the data wih their respective description)
sampleTable <- read_tsv(file      = paste0(pathToDataAnnotationFile,
                                            "data_file_annotation.csv"
                                            ),
                         col_names = TRUE
                         )
sampleTable

```


```{r sample_data_selection, cache=FALSE}

# Select samples to be analyzed
sampleTable_select <- filter(.data                   = sampleTable,
                             Biological_category    == "transcriptomics",
                             Technical_plateform    == "Affymetrix",
                             Organism               == "Mus_musculus",
                             Storage_dir1           != "?",
                             include                == "YES"
                             )
nrow(sampleTable_select)

# Data selected for analysis
group_by(.data = sampleTable_select,
         Data_source,
         Tumor_type,
         Tumor_origin,
         Localization_of_primary_tumor,
         sub_localization_of_primary_tumor
         ) %>%
    summarize(number = n()) %>%
    print(n = nrow(.), width = Inf)

```


# Data Pre-Processing

This section includes raw data importation, data quality control, data normalization and data filtering.

## Raw data importation

Raw data (in .CEL format) are imported using the *ReadAffy()* function from *simpleaffy* Bioconductor package (Miller et al.,2017). Chip number matching and annotations are provided by the Bioconductor "mouse4302mmentrezg.db" and the BrainArray custom CDF package "mouse4302mentrezgcdf version 23.0.0" (http://brainarray.mbni.med.umich.edu/Brainarray/Database/CustomCDF/23.0.0/entrezg.asp).

```{r rawData_reading}

rawDataCEL <- ReadAffy(filenames  = as.vector(paste0(sampleTable_select$Storage_dir1,
                                                     sampleTable_select$File_name)
                                              ),
                       cdfname    = 'mouse4302mmentrezg'
                       )

```


## Raw data quality controls

Quality of raw data is assessed by ploting boxplot and density of raw intensity.

```{r quality_control}

# Boxplot of raw data
boxplot(rawDataCEL, las=2, frame=FALSE, xaxt='n')

# Density of raw data
hist(rawDataCEL)

```
Expression intensity boxplot and density do not show abberant samples. All samples seem to be correct according these plots.


## Data Normalization

Data normalization is performed according to the RMA algorithm (Irizarry et al, 2003). It follows 3 steps: background correcting, quantile normalization followed by a summarization step using the median polish algorithm. The rma function of simpleaffy package is used with the default setting.

```{r rma}

# rma normalization (please wait)
eset <- rma(object = rawDataCEL) # ERROR

# Convert ExpressionSet object into matrix
matrix_rma <- exprs(eset)
nrow(matrix_rma)
head(matrix_rma)

my_matrix_unfiltered <- matrix_rma
colnames(my_matrix_unfiltered) <- as.vector(sampleTable_select$Sample_name)

# Use sample names as colnames instead of file names
colnames(matrix_rma) <- as.vector(sampleTable_select$Sample_name)

# Boxplot after normalization 
boxplot(matrix_rma, las=2, frame=FALSE, outline=FALSE)

# Global distribution after normalization
plot(density(matrix_rma))

# Unfiltered matix
my_matrix_unfiltered <- matrix_rma

```

Boxplot of normalized intensity for each sample indicates that normalization was successfully done. 

## Data filtering

As we will perform differentially analysis later, we filtering our data to increase the power of the subsequent differential analysis by reducing the number of hypotheses to be tested and therefore the side effect of multiple testing correction.
We use the method described in Nucleic Acids Research by Lu et al, in 2011 which is based on the measure of concordance of probes and probset expression. In the algorithm, this concordance is obtained using the so called pvac score computed from the 1st component of the pca at probe level.
For more details about the algorithm, please refer to (Lu et al, 2011). 
Implementation of this algorithm is available in Bioconductor R package "pvac" that we use below (link to the manual page: https://www.bioconductor.org/packages/release/bioc/vignettes/pvac/inst/doc/pvac.pdf).

```{r pvac_filtering}

# pvac filtering (You may want to take a coffee!)
ft <- pvacFilter(abatch = rawDataCEL,
                 pct    = 0.99
                 )

# Plot PVAC score distributions and cutoff
plot_pvac_filtering(ft)

# Selec filtered in probeset
eset_rma_pvac <- eset[ft$aset,]

# Convert expressionSet object into matrix
matrix_rma_pvac <- exprs(eset_rma_pvac)
 
# Use sample names as colnames instead of file names
colnames(matrix_rma_pvac) <- as.vector(sampleTable_select$Sample_name)

# Boxplot after normalization 
boxplot(matrix_rma_pvac, las=2, frame=FALSE, outline=FALSE)

# Global distribution after normalization
plot(density(matrix_rma_pvac))

# Rename matrix
my_matrix <- matrix_rma_pvac
nrow(my_matrix)

```

After filtering, 11,278 transcripts are retained among the 18075 previously present in our unfiltered matrix.

## Annotation

In this step, we just add the official gene symbol to our matrix. We choose the gene symbol because we juge it more convenient for the interpretation and visualization of differential expression and functional analyses results.

```{r annotation_of_unfiltered_matrix}

# Annotation list present in the package 
ls("package:mouse4302mmentrezg.db")

# Remove control pprobes (AFFX*)
my_matrix_unfiltered <- my_matrix_unfiltered[- grep("AFFX*", rownames(my_matrix_unfiltered), perl = TRUE),]
nrow(my_matrix_unfiltered)

# Extract probes from our matrix
probes_unfiltered <- row.names(my_matrix_unfiltered)

# Extract gene symbol corresponding to our filtered matrix probes
symbol_unfiltered <- unlist(mget(probes_unfiltered, mouse4302mmentrezgSYMBOL))

ENSMUSG_unfiltered <- unlist(mget(probes_unfiltered, mouse4302mmentrezgENSEMBL))

# Convert symbol to data frame
symbol_unfiltered_frame <- as.data.frame(symbol_unfiltered)

ENSMUSG_unfiltered_frame <- as.data.frame(ENSMUSG_unfiltered)

head(ENSMUSG_unfiltered_frame)

# Merge symbol and ensembl gene ID (for data integration for RNAseq ID conversion)
ENSMUSG_ENTREZsymbol_unfiltered_frame <- merge(x = symbol_unfiltered_frame,
                                               y = ENSMUSG_unfiltered_frame,
                                               by = 0
                                               )
ENSMUSG_ENTREZsymbol_unfiltered_frame$Row.names <- NULL
head(ENSMUSG_ENTREZsymbol_unfiltered_frame)


# Merge symbol to our matrix
head(symbol_unfiltered_frame)
head(my_matrix_unfiltered[,1:4])

my_matrix_unfiltered_annotated <- merge(x     = symbol_unfiltered_frame,
                                        y     = my_matrix_unfiltered,
                                        by    = 0,
                                        all   = FALSE
                                        )
rownames(my_matrix_unfiltered_annotated) <- my_matrix_unfiltered_annotated[,1]
my_matrix_unfiltered_annotated           <- my_matrix_unfiltered_annotated[,-1]
nrow(my_matrix_unfiltered_annotated)

my_matrix_unfiltered_annotated$Row.names <- NULL




head(my_matrix_unfiltered_annotated)

```

```{r annotation_of_filtered_matrix}
# Remove control pprobes (AFFX*)
my_matrix <- my_matrix[- grep("AFFX*", rownames(my_matrix), perl = TRUE),]

# Extract probes from our matrix
probes <- row.names(my_matrix)

# Extract gene symbol corresponding to our filtered matrix probes
symbol <- unlist(mget(probes, mouse4302mmentrezgSYMBOL))

# Convert symbol to data frame
symbol_frame <- as.data.frame(symbol)

# Merge symbol to our matrix
head(symbol_frame)
head(my_matrix[,1:4])

my_matrix_annotated <- merge(x     = symbol_frame,
                             y     = my_matrix,
                             by    = 0,
                             all.y = TRUE
                             )
rownames(my_matrix_annotated) <- my_matrix_annotated[,1]
my_matrix_annotated           <- my_matrix_annotated[,-1]
nrow(my_matrix_annotated)

```


# Data Analysis

The main goal of the analysis is to answer the following questions:

* are mouse MRT form an independent mouse tumor entity? (mainly answered in (Han et al., 2016)
* cells of origin of  murine MRT (using the Nestin genotype)

As unsupervised analysis, we use PCA and hierarchical clustering with the 1000 most variable genes. All hierarchical clustering done in this study are performed using the pearson correlation coefficient as distance matrix and the Ward method. Most variable genes are obained using the genes.selection() function implemented in EMA package [Servant et al, 2011].

## Are murine MRT clustered with other neural tumours?

To answer this question, we use murine neuroblastoma and medulloblastoma samples as neural tumors and lymphoma as non neural tumors. Lymphoma murine tumor samples are obtained from late inactivation of SMARCB1 [Han et al, 2016].

### Medulloblastoma, neuroblastoma and lymphoma clustering 

In this part, as expected, we show that Lymphoma, medulloblastoma and neuroblastoma form a two class that we denote later as neural (medulloblastoma and neuroblastoma) and non-neural tumors (Lymphoma) [Han et al, 2016].

#### Samples

```{r subset_LMN_sample, cache=FALSE}

# Subset Lymphoma, medulloBlastoma and neuroBlastoma samples
samples_LMN <- filter(.data               = sampleTable_select,
                      Data_source        == "Curie",
                      Tumor_type         == "lymphoma" |
                          Tumor_type     == "MB" |
                              Tumor_type == "NB",
                      Tumor_origin       == "primary_tumor"
                      )

# Data summary
group_by(.data = samples_LMN
        ,Tumor_type
        ,Tumor_origin
        ,Localization_of_primary_tumor
        ,Genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)


# Subset matrix
my_matrix_LMN <- my_matrix_annotated[, as.vector(samples_LMN$Sample_name)]

```

#### most variable genes

```{r LMN_select_most_variable_genes}

# Most variable genes
mvgenes_LMN <- genes.selection(my_matrix_LMN, thres.num=1000)

# Subset matrix
my_matrix_LMN_mvgenes_LMN <- my_matrix_LMN[mvgenes_LMN,]

```

#### PCA

```{r pca_LMN}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMN_mvgenes.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMN_mvgenes_LMN,
         sampleTable                = samples_LMN,
         highlightedVar             = "Tumor_type",
         highlightedVarName         = "Tumor type",
         colors                     = tumor_type_col,
         sampleLabel                = NULL,
         sampleLabelColumnCondition = NULL,
         sampleLabelValueCondition  = NULL
         )
#dev.off()

```

## t-SNE

```{r tSNE_RT_SDNRT_HB_NB_EwS}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMN_mvgenes.pdf"), width=14, height=10)
tsne_func(matrix             = my_matrix_LMN_mvgenes_LMN,
          sampleTable        = samples_LMN,
          highlightedVar     = "Tumor_type",
          highlightedVarName = "Tumor type",
          colors             = tumor_type_col,
          seed               = 9,
          perplexity         = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_RT_SDNRT_HB_NB_EwS}

#pdf(paste0(pathToFigures,"umap_LMN_mvgenes.pdf"), width=16, height=12)
umap_func(matrix             = my_matrix_LMN_mvgenes_LMN,
          sampleTable        =  samples_LMN,
          highlightedVar     = "Tumor_type",
          highlightedVarName = "Tumor type",
          colors             = tumor_type_col
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_RT}

# Top annotation
top_annot_LMN <- createTopAnnot_func(sampleTable = samples_LMN,
                                     columns     = c("Tumor_type","group_name"),
                                     colorsList  = list(tumor_type_col, group_name_col)
                                     )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMN_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMN_mvgenes_LMN,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMN,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```


According to the PCA and hierachical clustering, neuronal (neuroblastoma and medulloblastoma) and non-neuronal (Lymphoma) murine tumors are clearly clustered separately.


### Are mice RT classified as Neural or Non-neural?

Based on the previous classification, we try to see if the murine RT samples form a specific entity or belong to either the neural or non-neural tumor. 

#### Samples

```{r subset_LMNR_sample}

# Subset Lymphoma, medulloBlastoma and neuroBlastoma samples
samples_LMNR <- filter(.data         = sampleTable_select,
                       Data_source  == "Curie",
                       Tumor_type   == "lymphoma" |
                       Tumor_type   == "MB" |
                       Tumor_type   == "NB" |
                       Tumor_type   == "RT",
                       Tumor_origin == "primary_tumor"
                       )

# Data summary
group_by(.data = samples_LMNR
        ,Tumor_type
        ,Tumor_origin
        ,Localization_of_primary_tumor
        ,Genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Subset matrix
my_matrix_LMNR <- my_matrix_annotated[, as.vector(samples_LMNR$Sample_name)]

```

### Most variable genes

```{r mvgenes_LMNR}

# Most variable genes
mvgenes_LMNR <- genes.selection(my_matrix_LMNR, thres.num=1000)

# Subset matrix
my_matrix_LMNR_mvgenes_LMNR <- my_matrix_LMNR[mvgenes_LMNR,]

```

#### PCA

```{r pca_LMNR}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNR_mvgenes.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNR_mvgenes_LMNR,
         sampleTable                = samples_LMNR,
         highlightedVar             = "Tumor_type",
         highlightedVarName         = "Tumor type",
         colors                     = tumor_type_col,
         sampleLabel                = NULL,
         sampleLabelColumnCondition = NULL,
         sampleLabelValueCondition  = NULL
         )
#dev.off()

```

## t-SNE

```{r tSNE_LMNR}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNR_mvgenes.pdf"), width=14, height=10)
tsne_func(matrix             = my_matrix_LMNR_mvgenes_LMNR,
          sampleTable        = samples_LMNR,
          highlightedVar     = "Tumor_type",
          highlightedVarName = "Tumor type",
          colors             = tumor_type_col,
          seed               = 9,
          perplexity         = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_RT_SDNRT_HB_NB_EwS}

#pdf(paste0(pathToFigures,"umap_LMNR_mvgenes.pdf"), width=16, height=12)
umap_func(matrix             = my_matrix_LMNR_mvgenes_LMNR,
          sampleTable        =  samples_LMNR,
          highlightedVar     = "Tumor_type",
          highlightedVarName = "Tumor type",
          colors             = tumor_type_col
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_RT}

# Top annotation
top_annot_LMNR <- createTopAnnot_func(sampleTable = samples_LMNR,
                                      columns     = "Tumor_type",
                                      colorsList  = list(tumor_type_col)
                                      )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNR_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNR_mvgenes_LMNR,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNR,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```


As already shown in [Han et al, 2017], the murine RT samples are clustered in two subgroups. The first subgroup clustered with neural tumors and the second subgroup clustered with the non-neural tumors. We will name these two subgroups as mIC1 and mIC2, respectively.


```{r Pgdb5_expression}

head(my_matrix_LMNR[,1:4])

# Add gene symbol column
my_matrix_LMNR_annotated <- my_matrix_annotated[, c("symbol",as.vector(samples_LMNR$Sample_name))]

# Extract Pdgb5 gene expression 
my_matrix_LMNR_Pgbd5 <- filter(.data = my_matrix_LMNR_annotated,
                               symbol == "Pgbd5"
                               )%>%
    #dplyr::select(-ENSGID) %>%
    melt(.) %>%
    dplyr::select(-symbol)
colnames(my_matrix_LMNR_Pgbd5) <- c("Sample_name","Pgbd5")
head(my_matrix_LMNR_Pgbd5)

# Merge coldata and matrix
sample_LMNR_Pgbd5 <- merge(x   = samples_LMNR,
                           y   = my_matrix_LMNR_Pgbd5,
                           by  = "Sample_name",
                           all = FALSE
                           )
head(sample_LMNR_Pgbd5)

# Check order
sample_LMNR_Pgbd5$Sample_name == samples_LMNR$Sample_name

sample_LMNR_Pgbd5 <- sample_LMNR_Pgbd5[match(samples_LMNR$Sample_name, sample_LMNR_Pgbd5$Sample_name),]

sample_LMNR_Pgbd5$Sample_name == samples_LMNR$Sample_name

# Bottom annotation
bottom_annot_LMNR_Pgbd5 <- HeatmapAnnotation(Pgdb5_expression = anno_barplot(x       = sample_LMNR_Pgbd5$Pgbd5,
                                                                             axis    = TRUE,
                                                                             border  = FALSE,
                                                                             outline = FALSE,
                                                                             gp      = gpar(fill = tumor_type_col[as.vector(as.factor(sample_LMNR_Pgbd5$Tumor_type))], col = "white")
                                                                             ),
                                             gap                     = my_gap,
                                             show_annotation_name    = TRUE,
                                             annotation_name_gp      = my_annotation_name_gp,
                                             annotation_legend_param = my_annotation_legend_param,
                                             na_col                  = my_na_col,
                                             height                  = unit(5,"cm")
                                             )

#pdf(paste0(pathToFigures,"complexHeatmap_LMNR_mvgenes_Pgbd5.pdf"), width=12, height=16)
Heatmap(matrix                      = my_matrix_LMNR_mvgenes_LMNR,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = "euclidean",
        show_row_names              = FALSE,
        column_title                = paste0("murine R26 primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes (IQR)",
        top_annotation              = top_annot_LMNR,
        show_row_dend               = FALSE,
        bottom_annotation           = bottom_annot_LMNR_Pgbd5,
        #bottom_annotation_height    = unit(5,"cm"),
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40,"mm")
        )
#dev.off()

```

```{r Smarcb1_expression}

# Plot density plot from unfiltered matrix to look at background level
my_matrix_unfiltered_annotated_melt <- melt(my_matrix_unfiltered_annotated)

background_level <- 5.9

pdf(paste0(pathToFigures,"density_murine_affy.pdf"), width=12, height=10)
ggplot(data = my_matrix_unfiltered_annotated_melt) +
    geom_density(mapping = aes(x = value), fill = "papayawhip") +
    geom_vline(xintercept = background_level, color = "red", lty=2) +
    scale_x_continuous(breaks = c(0,5,background_level,10)) +
    xlab("\nlog2(expression)") +
    ylab("density\n") +
    theme_bw() +
    theme(text = element_text(size=30),
          axis.text.x = element_text(color = c("black","red","black")),
          axis.ticks.x = element_line(color = c("black","red","black"), size = c(0.5,1,0.5))
          )
dev.off()

# Subset matrix
my_matrix_LMNR_annotated <- my_matrix_annotated[, c("symbol",as.vector(samples_LMNR$Sample_name))]

head(my_matrix_LMNR_annotated)

# Extract Pdgb5 gene expression 
my_matrix_LMNR_Smarcb1 <- filter(.data = my_matrix_LMNR_annotated,
                               symbol == "Smarcb1"
                               )%>%
    #dplyr::select(-ENSGID) %>%
    melt(.) %>%
    dplyr::select(-symbol)
colnames(my_matrix_LMNR_Smarcb1) <- c("Sample_name","Smarcb1")
head(my_matrix_LMNR_Smarcb1)

# Merge coldata and matrix
sample_LMNR_Smarcb1 <- merge(x   = samples_LMNR,
                           y   = my_matrix_LMNR_Smarcb1,
                           by  = "Sample_name",
                           all = FALSE
                           )
head(sample_LMNR_Smarcb1)

# Check order
sample_LMNR_Smarcb1$Sample_name == samples_LMNR$Sample_name

sample_LMNR_Smarcb1 <- sample_LMNR_Smarcb1[match(samples_LMNR$Sample_name, sample_LMNR_Smarcb1$Sample_name),]

sample_LMNR_Smarcb1$Sample_name == samples_LMNR$Sample_name

# Bottom annotation
bottom_annot_LMNR_Smarcb1 <- HeatmapAnnotation(Smarcb1 = anno_barplot(x       = sample_LMNR_Smarcb1$Smarcb1,
                                                                               axis    = TRUE,
                                                                               border  = FALSE,
                                                                               outline = FALSE,
                                                                               gp      = gpar(fill = tumor_type_col[as.vector(as.factor(sample_LMNR_Smarcb1$Tumor_type))], col = "white"),
                                                                               baseline = 5
                                                                               ),
                                               gap                     = my_gap,
                                               show_annotation_name    = TRUE,
                                               annotation_name_gp      = my_annotation_name_gp,
                                               annotation_legend_param = my_annotation_legend_param,
                                               na_col                  = my_na_col,
                                               height                  = unit(5,"cm")
                                               )

#pdf(paste0(pathToFigures,"complexHeatmap_LMNR_mvgenes_Smarcb1.pdf"), width=12, height=16)
Heatmap(matrix                      = my_matrix_LMNR_mvgenes_LMNR,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = "euclidean",
        show_row_names              = FALSE,
        column_title                = paste0("murine R26 primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes (IQR)",
        top_annotation              = top_annot_LMNR,
        show_row_dend               = FALSE,
        bottom_annotation           = bottom_annot_LMNR_Smarcb1,
        #bottom_annotation_height    = unit(5,"cm"),
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40,"mm")
        )
#dev.off()

```

### Expression of Pgdb5 gene

#### Prepare data

```{r expression_tpm}

head(my_matrix_LMNR_annotated)

# Subset matrix
matrix_LMNR_PGBD5 <- my_matrix_LMNR_annotated[which(my_matrix_LMNR_annotated$symbol == "Pgbd5"),]

# Remove rownames
rownames(matrix_LMNR_PGBD5) <- NULL

# Melt matrix
matrix_LMNR_PGBD5_melt <- melt(matrix_LMNR_PGBD5)

colnames(matrix_LMNR_PGBD5_melt) <- c("symbol","Sample_name","log2Intensity")

# Add sample annotation 

head(samples_LMNR[,1:4])

matrix_LMNR_PGBD5_melt_plus <- merge(x   = matrix_LMNR_PGBD5_melt,
                                     y   = samples_LMNR,
                                     by  = "Sample_name",
                                     all = FALSE
                                     )

head(matrix_LMNR_PGBD5_melt_plus)

```

#### Boxplot expression of Pgbd5 gene in MB, NB, lymphoma and RT murine primary tumors

```{r boxplot_PGBD5}

plot(density(as.matrix(my_matrix_unfiltered)))

bg_level <- 5.5


#pdf(paste0(pathToFigures,"boxplot_murine_affy_LMNR_PGDB5.pdf"), width=12, height=20)
ggplot(data = matrix_LMNR_PGBD5_melt_plus,
       mapping = aes(x=Tumor_type, y=log2Intensity, fill=Tumor_type)) +
    #facet_wrap(facets = ~ counts_normalisation, ncol = 2, scales = "free") +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(name = "Tumor type", values=tumor_type_col[levels(as.factor(matrix_LMNR_PGBD5_melt_plus$Tumor_type))]) +
    #geom_point(size=6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 3) +
    xlab("Tumor type") +
    ylab("log2Intensity") +
    #scale_color_manual(values=c("darkgreen","brown","black")) +
    ylim(2.5,10) +
    ggtitle(label="Pgdb5 expression", subtitle="murine R26 in situ (affy)") +
    geom_hline(yintercept = bg_level, color= "black", lwd=.5, lty = "dotted") +
    annotate(geom="rect", xmin = 0, xmax=5, ymin = 2.5, ymax=bg_level, alpha=.15) +
    annotate(geom="text", x = 1.5, y=bg_level+0.2, label="background level",col="grey45", size=12) +
    theme(
        text = element_text(size = 40), #my_fontsize),
        axis.text.x = element_text(angle = 90),
        #axis.text.x = element_blank(),
        panel.background = element_rect(fill = "white",
                                        colour = "grey",
                                        size = 0.5, linetype = "solid"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                        colour = "grey85"),
        panel.grid.minor = element_line(size = 0.25, linetype = 'dotted',
                                        colour = "grey85"),
        strip.background = element_rect(fill = "slategray1",
                                        colour = "grey85"),
        strip.text = element_text(face = "bold"),
        legend.key.height = unit(1.5,"cm")
    )
#dev.off()

```

#### Boxplot expression of Pgbd5 gene in RT murine primary tumors

```{r boxplot_PGBD5}

# Subset RT matrix
matrix_LMNR_PGBD5_melt_plus_RT <- filter(.data       = matrix_LMNR_PGBD5_melt_plus,
                                         Tumor_type == "RT"
                                         )

# Boxplot
p <- ggplot(data = matrix_LMNR_PGBD5_melt_plus_RT,
       mapping = aes(x=group_name, y=log2Intensity, fill=group_name)) +
    #facet_wrap(facets = ~ counts_normalisation, ncol = 2, scales = "free") +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(name = "RT subtype", values=group_name_col[levels(as.factor(matrix_LMNR_PGBD5_melt_plus_RT$group_name))]) +
    geom_point(size=6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 3) +
    xlab("RT subtype") +
    ylab("log2Intensity") +
    #scale_color_manual(values=c("darkgreen","brown","black")) +
    ylim(2.5,10) +
    ggtitle(label="Pgdb5 expression", subtitle="murine R26 RT (affy)") +
    geom_hline(yintercept = bg_level, color= "black", lwd=.5, lty = "dotted") +
    annotate(geom="rect", xmin = 0, xmax=3, ymin = 2.5, ymax=bg_level, alpha=.15) +
    annotate(geom="text", x = 0.7, y=bg_level-0.5, label="background level",col="grey45", size=12) +
    theme(
        text = element_text(size = 40), #my_fontsize),
        axis.text.x = element_text(angle = 90),
        #axis.text.x = element_blank(),
        panel.background = element_rect(fill = "white",
                                        colour = "grey",
                                        size = 0.5, linetype = "solid"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                        colour = "grey85"),
        panel.grid.minor = element_line(size = 0.25, linetype = 'dotted',
                                        colour = "grey85"),
        strip.background = element_rect(fill = "slategray1",
                                        colour = "grey85"),
        strip.text = element_text(face = "bold"),
        legend.key.height = unit(1.5,"cm")
    ) +
    stat_compare_means(comparisons = list(c("neuronal","non_neuronal")),
                       method = "t.test",
                       tip.length = 0.05,
                       label = "p.format",
                                        #hide.ns = TRUE,
                       label.y = 9
                       )
# Increase geom_signif text size
p$layers[[which_layers(p, "GeomSignif")]]$aes_params$textsize <- 15
#pdf(paste0(pathToFigures,"boxplot_murine_affy_R_PGDB5.pdf"), width=16, height=10)
p
#dev.off()

```

### Assigning group name to murine RT samples

This classification in 2 transcriptomics groups of ATRT in mice (as seen in human ATRT) suggest that the ATRT tumors may have 2 different cell origin.

```{r assign_samples_neural_non_neural}

# Cut dendogram into 2 groups
pl <- Heatmap(matrix                      = my_matrix_LMNR_mvgenes_LMNR,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth
              )
hc <- as.hclust(column_dend(pl))
group_num <- cutree(tree = hc,
                    k    = 2,
                    h    = NULL
                    )
group <- melt(group_num)
colnames(group) <- "group_num"
#group
group$group_name <- ifelse(group$group_num == "1","non_neuronal","neuronal")
#group$Sample_name <- rownames(group)
group

# Add assigned group_name
samples_LMNR_plus <- samples_LMNR
samples_LMNR_plus$group_num <- group$group_num
samples_LMNR_plus$group_name <- group$group_name

samples_LMNR_plus[,c("Sample_name","Tumor_type","group_name")]

```

#### PCA (after RT subgroup assignment)

```{r pca_LMNR}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNR_mvgenes_plus.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNR_mvgenes_LMNR,
         sampleTable                = samples_LMNR_plus,
         highlightedVar             = "group_name",
         highlightedVarName         = "subgroup",
         colors                     = group_name_col,
         sampleLabel                = "Tumor_type",
         sampleLabelColumnCondition = NULL,
         sampleLabelValueCondition  = NULL
         )
#dev.off()

```

#### t-SNE (after RT subgroup assignment)

```{r tSNE_LMNR}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNR_mvgenes_plus.pdf"), width=14, height=10)
tsne_func(matrix             = my_matrix_LMNR_mvgenes_LMNR,
          sampleTable        = samples_LMNR_plus,
          highlightedVar     = "group_name",
          highlightedVarName = "subgroup",
          colors             = group_name_col,
          sampleLabel        = "Tumor_type",
          seed               = 9,
          perplexity         = 7
          )
#dev.off()

```

#### UMAP analysis (after RT subgroup assignment)

```{r umap_LMNR}

#pdf(paste0(pathToFigures,"umap_LMNR_mvgenes_plus.pdf"), width=16, height=12)
umap_func(matrix             = my_matrix_LMNR_mvgenes_LMNR,
          sampleTable        =  samples_LMNR_plus,
          highlightedVar     = "group_name",
          highlightedVarName = "subgroup",
          colors             = group_name_col,
          sampleLabel        = "Tumor_type"
          )
#dev.off()

```


#### Hierarchical clustering (after RT subgroup assignment)

```{r complexHeatmap_U830_RT}

# Top annotation
top_annot_LMNR_plus <- createTopAnnot_func(sampleTable = samples_LMNR_plus,
                                           columns     = c("Tumor_type","group_name"),
                                           colorsList  = list(tumor_type_col, group_name_col)
                                           )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNR_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNR_mvgenes_LMNR,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNR_plus,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```

### Export matrix of LMNR samples to data integration

```{r export_LMNR_matrix}

# Subset normal sample
samples_H <- filter(.data               = sampleTable_select,
                    Tumor_type         == "normal"
                    )
samples_H
samples_LMNRH <- rbind(samples_LMNR,samples_H)
my_matrix_unfiltered_LMNRH <- my_matrix_unfiltered_annotated[, c("symbol_unfiltered",as.vector(samples_LMNRH$Sample_name))]
murine_LMNRH_affy_list     <- list(matrix      = my_matrix_unfiltered_LMNRH,
                                  sampleTable = samples_LMNRH,
                                  description = "Mouse affymetrix rma normalized matrix, array type : MOE_430_2, annotated using the CDF package, from murine_RT analysis project"
                                  )
save(murine_LMNRH_affy_list, file = "murine_LMNRH_affy_list.RData")

# Export Affy Curie sample plan
samples_LMNRH_Curie <- filter(.data = samples_LMNRH,
                              Data_source == "Curie"
                              )

write.table(x         = samples_LMNRH_Curie,
            file      = paste0(pathToReport,"sample_murine_affy_Curie.txt"),
            sep       = "\t",
            col.names = TRUE,
            quote     = FALSE
            )

samples_LMNRH_Curie$Tumor_origin

# Export Affy Curie matrix
my_matrix_unfiltered_LMNRH_Curie <- my_matrix_unfiltered_annotated[, c("symbol_unfiltered",as.vector(samples_LMNRH_Curie$Sample_name))]

head(my_matrix_unfiltered_LMNRH_Curie)

write.table(x         = my_matrix_unfiltered_LMNRH_Curie,
            file      = paste0(pathToReport,"matrix_murine_affy_Curie.csv"),
            sep       = ";",
            col.names = TRUE,
            quote     = FALSE,
            row.names = FALSE
            )

```

## Allograft ATRT classification

To facilitate tumor tracking, allograft models were established. Samples from these grafted tumors models are called "allograft_flanc" or "alloG".


### Samples

```{r subset_LMNRA_sample}

# Subset Lymphoma, medulloBlastoma, neuroBlastoma, primary RT and allograft RT tumor samples
samples_LMNRA <- filter(.data          = sampleTable_select,
                        Data_source   == "Curie",
                        Organism      == "Mus_musculus",
                        Tumor_type    == "lymphoma" |
                        Tumor_type    == "NB" |
                        Tumor_type    == "MB" |
                        (Tumor_origin == "allograft_flanc" &
                         Genotype     != "Nestin") |
                        (Tumor_type   == "RT" &
                         Tumor_origin == "primary_tumor")
                        )

# Add previously assigned group_name
samples_LMNRA <- merge(x = samples_LMNRA[,-49],
                       y = samples_LMNR_plus[,c("Sample_name","group_name")],
                       by = "Sample_name",
                       all=TRUE
                       )

samples_LMNRA[,c("Tumor_type","Tumor_origin","Genotype","Sample_name","group_name")]

# Create a matrix including only the selected samples (file names) 
my_matrix_LMNRA <- my_matrix_annotated[,as.vector(samples_LMNRA$Sample_name)]

```

```{r mvgenes_LMNRA}

# Selecting the most variable genes according to IQR values (using the function implemented in EMA package)
mvgenes_LMNRA <- genes.selection(my_matrix_LMNRA, thres.num = 1000)

# Create a matrix with the lymphoma, mMB and mNB, RT and RT allograft samples with the 1000 most variable genes between all samples
my_matrix_LMNRA_mvgenes_LMNRA <- my_matrix_LMNRA[mvgenes_LMNRA,]

```

#### PCA

```{r pca_LMNRA}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNRA_mvgenes_plus.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNRA_mvgenes_LMNRA,
         sampleTable                = samples_LMNRA,
         highlightedVar             = "group_name",
         highlightedVarName         = "subgroup",
         colors                     = group_name_col,
         sampleLabel                = "Sample_name",
         sampleLabelColumnCondition = "Tumor_origin",
         sampleLabelValueCondition  = "allograft_flanc"
         )
#dev.off()

```

#### t-SNE

```{r tSNE_LMNRA}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNRA_mvgenes_plus.pdf"), width=14, height=10)
tsne_func(matrix                     = my_matrix_LMNRA_mvgenes_LMNRA,
          sampleTable                = samples_LMNRA,
          highlightedVar             = "group_name",
          highlightedVarName         = "subgroup",
          colors                     = group_name_col,
          sampleLabel                = "Sample_name",
          sampleLabelColumnCondition = "Tumor_origin",
          sampleLabelValueCondition  = "allograft_flanc",
          seed                       = 9,
          perplexity                 = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_LMNRA}

#pdf(paste0(pathToFigures,"umap_LMNRA_mvgenes_plus.pdf"), width=16, height=12)
umap_func(matrix                     = my_matrix_LMNRA_mvgenes_LMNRA,
          sampleTable                =  samples_LMNRA,
          highlightedVar             = "group_name",
          highlightedVarName         = "subgroup",
          colors                     = group_name_col,
          sampleLabel                = "Sample_name",
          sampleLabelColumnCondition = "Tumor_origin",
          sampleLabelValueCondition  = "allograft_flanc",
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_RT}

# Top annotation
top_annot_LMNRA <- createTopAnnot_func(sampleTable = samples_LMNRA,
                                       columns     = c("Tumor_type","group_name"),
                                       colorsList  = list(tumor_type_col, group_name_col)
                                       )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNRA_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNRA_mvgenes_LMNRA,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNRA,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```

Intriguingly, all allograft tumors are clustered with non-neural primary tumors subgroup.
According to this result, two hypothesis are advanced : 

* cells have a high plasticity and the subcutaneous environment modify the transcriptomic profile?
* only neural subgroups of MRT can grow in subcutaneous conditions?

To partially test the first hypothesis, samples from allograft tumors were re-implanted in the CNS. Samples taken from this re-implantation in the CNS are called the "orthotopic grafts".


## Orthotopic graft analysis

### Samples

```{r subset_LMNRAO_sample, warning=FALSE, message=FALSE, include=TRUE, cache=FALSE}

# Subset samples
samples_LMNRAO <- filter(.data          = sampleTable_select,
                         Data_source   == "Curie",
                         Organism      == "Mus_musculus",
                         Tumor_type    == "lymphoma" |
                         Tumor_type    == "NB" |
                         Tumor_type    == "MB" |
                         Tumor_origin  == "orthotopic_graft" |
                         (Tumor_origin == "allograft_flanc" &
                          Genotype     != "Nestin") |
                         (Tumor_type   == "RT" &
                          Tumor_origin == "primary_tumor")
                         )

# Add previously assigned group_name
samples_LMNRAO <- merge(x = samples_LMNRAO[,-49],
                        y = samples_LMNRA[,c("Sample_name","group_name")],
                        by = "Sample_name",
                        all=TRUE
                        )

names(samples_LMNRAO)

samples_LMNRAO[,c("Tumor_type","Tumor_origin","Genotype","Sample_name","group_name")]

# Subset matrix
my_matrix_LMNRAO <- my_matrix_annotated[,as.vector(samples_LMNRAO_plus$Sample_name)]

```

```{r mvgenes_LMNRAO}

# most variable genes
mvgenes_LMNRAO <- genes.selection(my_matrix_LMNRAO, thres.num = 1000)

# Select matrix
my_matrix_LMNRAO_mvgenes_LMNRAO <- my_matrix_LMNRAO[mvgenes_LMNRAO,]

```

#### PCA

```{r pca_LMNRAO}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNRAO_mvgenes.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNRAO_mvgenes_LMNRAO,
         sampleTable                = samples_LMNRAO,
         highlightedVar             = "group_name",
         highlightedVarName         = "subgroup",
         colors                     = group_name_col,
         sampleLabel                = "Sample_name",
         sampleLabelColumnCondition = "Tumor_origin",
         sampleLabelValueCondition  = "orthotopic_graft"
         )
#dev.off()

```

#### t-SNE

```{r tSNE_RT_SDNRT_HB_NB_EwS}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNRAO_mvgenes.pdf"), width=14, height=10)
tsne_func(matrix                     = my_matrix_LMNRAO_mvgenes_LMNRAO,
          sampleTable                = samples_LMNRAO,
          highlightedVar             = "group_name",
          highlightedVarName         = "subgroup",
          colors                     = group_name_col,
          sampleLabel                = "Sample_name",
          sampleLabelColumnCondition = "Tumor_origin",
          sampleLabelValueCondition  = "orthotopic_graft",
          seed                       = 9,
          perplexity                 = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_RT_SDNRT_HB_NB_EwS}

#pdf(paste0(pathToFigures,"umap_LMNRAO_mvgenes_plus.pdf"), width=16, height=12)
umap_func(matrix                     = my_matrix_LMNRAO_mvgenes_LMNRAO,
          sampleTable                =  samples_LMNRAO,
          highlightedVar             = "group_name",
          highlightedVarName         = "subgroup",
          colors                     = group_name_col,
          sampleLabel                = "Sample_name",
          sampleLabelColumnCondition = "Tumor_origin",
          sampleLabelValueCondition  = "orthotopic_graft",
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_RT}

# Top annotation
top_annot_LMNRAO <- createTopAnnot_func(sampleTable = samples_LMNRAO,
                                        columns     = c("Tumor_type","group_name","Tumor_origin"),
                                        colorsList  = list(tumor_type_col, group_name_col, tumor_origin_col)
                                        )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAO_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNRAO_mvgenes_LMNRAO,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNRAO,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```

Intriguingly, all orthotopic graft tumors are clustered with mIC2 subgroup. This may suggest that only mIC2 primary tumors can grow in subcutaneous environment. 

```{r Pgdb5_expression}

head(my_matrix_LMNRAO[,1:4])

# Add gene symbol column
my_matrix_LMNRAO_annotated <- my_matrix_annotated[, c("symbol",as.vector(samples_LMNRAO$Sample_name))]

# Extract Pdgb5 gene expression 
my_matrix_LMNRAO_Pgbd5 <- filter(.data = my_matrix_LMNRAO_annotated,
                                 symbol == "Pgbd5"
                                 )%>%
    #dplyr::select(-ENSGID) %>%
    melt(.) %>%
    dplyr::select(-symbol)
colnames(my_matrix_LMNRAO_Pgbd5) <- c("Sample_name","Pgbd5")
head(my_matrix_LMNRAO_Pgbd5)

# Merge coldata and matrix
sample_LMNRAO_Pgbd5 <- merge(x   = samples_LMNRAO,
                             y   = my_matrix_LMNRAO_Pgbd5,
                             by  = "Sample_name",
                             all = FALSE
                             )
head(sample_LMNRAO_Pgbd5)

sample_LMNRAO_Pgbd5$Sample_name == samples_LMNRAO$Sample_name

# Bottom annotation
bottom_annot_LMNRAO_Pgbd5 <- HeatmapAnnotation(Pgdb5_expression = anno_barplot(x       = sample_LMNRAO_Pgbd5$Pgbd5,
                                                                                                      axis    = TRUE,
                                                                                                      border  = FALSE,
                                                                                                      outline = FALSE,
                                                                                                      gp      = gpar(fill = tumor_type_col[as.vector(as.factor(sample_LMNRAO_Pgbd5$Tumor_type))], col = "white")
                                                                                                      ),
                                                                      gap                     = my_gap,
                                                                      show_annotation_name    = TRUE,
                                                                      annotation_name_gp      = my_annotation_name_gp,
                                     annotation_legend_param = my_annotation_legend_param,
                                     na_col                  = my_na_col,
                                     height                  = unit(5,"cm")
                                     )

#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAO_mvgenes_Pgbd5.pdf"), width=12, height=16)
h <- Heatmap(matrix                      = my_matrix_LMNRAO_mvgenes_LMNRAO,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             clustering_distance_rows    = "euclidean",
             show_row_names              = FALSE,
             column_title                = paste0("murine R26 primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = "1000 most variable genes (IQR)",
             top_annotation              = top_annot_LMNRAO,
             show_row_dend               = FALSE,
             bottom_annotation           = bottom_annot_LMNRAO_Pgbd5,
             column_dend_height          = unit(40,"mm"),
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.5,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                title          = paste0("log2Intensity\n"),
                                                title_gp       = gpar(fontsize = my_fontsize),
                                                title_position = "topcenter"
                                                )
             )
draw(h, heatmap_legend_side = "left")
#dev.off()

```


## Nestin ATRT clustering

In order to identify cells of origin of ATRT tumors, mouse models with NestinCre_flox_Smarcb1_flox have been established. Nestin is a promoter that specifically activated in neural cells unlike ROSA26 which is an ubuquituous promoter (ROSA26 were used to control SMARCB1 inactivation in previous mouse model). SMARCB1-def cells from Nestin mouse model were graft subcutaneously. Samples were collected from developped tumors. For simplicity, we name these allograft tumor samples as "Nestin".


### Samples


```{r subset_LMNRAON_sample}

# Subset Lymphoma, medulloBlastoma, neuroBlastoma, primary RT, allograft RT, orthotopic graft RT and orthotopic graft RT tumor samples
samples_LMNRAON <- filter(.data          = sampleTable_select,
                          Data_source   == "Curie",
                          Organism      == "Mus_musculus",
                          Tumor_type    == "lymphoma" |
                          Tumor_type    == "NB" |
                          Tumor_type    == "MB" |
                          Tumor_origin  == "orthotopic_graft" |
                          Tumor_origin  == "allograft_flanc" |
                          (Tumor_type   == "RT" &
                           Tumor_origin == "primary_tumor")
                          )

# Add previously assigned group_name
samples_LMNRAON <- merge(x = samples_LMNRAON[,-49],
                         y = samples_LMNRAO[,c("Sample_name","group_name")],
                         by = "Sample_name",
                         all=TRUE
                         )

samples_LMNRAON[,c("Tumor_type","Tumor_origin","Genotype","Sample_name","group_name")]

# Subset matrix
my_matrix_LMNRAON <- my_matrix_annotated[,as.vector(samples_LMNRAON_plus$Sample_name)]

```

```{r mvgenes_LMNRAON}

# most variable genes
mvgenes_LMNRAON <- genes.selection(my_matrix_LMNRAON, thres.num = 1000)

# Select matrix
my_matrix_LMNRAON_mvgenes_LMNRAON <- my_matrix_LMNRAON[mvgenes_LMNRAON,]

```

#### PCA

```{r pca_LMNRAON}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNRAON_mvgenes.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNRAON_mvgenes_LMNRAON,
         sampleTable                = samples_LMNRAON,
         highlightedVar             = "Genotype",
         highlightedVarName         = "Genotype",
         colors                     = genotype_col,
         sampleLabel                = "group_name"
         )
#dev.off()

```

#### t-SNE

```{r tSNE_LMNRAON}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNRAON_mvgenes.pdf"), width=14, height=10)
tsne_func(matrix                     = my_matrix_LMNRAON_mvgenes_LMNRAON,
          sampleTable                = samples_LMNRAON,
          highlightedVar             = "Genotype",
          highlightedVarName         = "Genotype",
          colors                     = genotype_col,
          sampleLabel                = "group_name",
          seed                       = 9,
          perplexity                 = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_LMNRAON}

#pdf(paste0(pathToFigures,"umap_LMNRAON_mvgenes_plus.pdf"), width=16, height=12)
umap_func(matrix                     = my_matrix_LMNRAON_mvgenes_LMNRAON,
          sampleTable                =  samples_LMNRAON,
          highlightedVar             = "Genotype",
          highlightedVarName         = "Genotype",
          colors                     = genotype_col,
          sampleLabel                = "group_name"
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_LMNRAON}

# Top annotation
top_annot_LMNRAON <- createTopAnnot_func(sampleTable = samples_LMNRAON,
                                        columns     = c("Tumor_type","group_name","Tumor_origin","Genotype"),
                                        colorsList  = list(tumor_type_col, group_name_col, tumor_origin_col,genotype_col)
                                        )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAON_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNRAON_mvgenes_LMNRAON,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNRAON,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```

```{r Pgdb5_expression}

head(my_matrix_LMNRAON[,1:4])

# Add gene symbol column
my_matrix_LMNRAON_annotated <- my_matrix_annotated[, c("symbol",as.vector(samples_LMNRAON$Sample_name))]

# Extract Pdgb5 gene expression 
my_matrix_LMNRAON_Pgbd5 <- filter(.data = my_matrix_LMNRAON_annotated,
                                 symbol == "Pgbd5"
                                 )%>%
    #dplyr::select(-ENSGID) %>%
    melt(.) %>%
    dplyr::select(-symbol)
colnames(my_matrix_LMNRAON_Pgbd5) <- c("Sample_name","Pgbd5")
head(my_matrix_LMNRAON_Pgbd5)

# Merge coldata and matrix
sample_LMNRAON_Pgbd5 <- merge(x   = samples_LMNRAON,
                             y   = my_matrix_LMNRAON_Pgbd5,
                             by  = "Sample_name",
                             all = FALSE
                             )
head(sample_LMNRAON_Pgbd5)

sample_LMNRAON_Pgbd5$Sample_name == samples_LMNRAON$Sample_name

# Bottom annotation
bottom_annot_LMNRAON_Pgbd5 <- HeatmapAnnotation(Pgdb5_expression = anno_barplot(x       = sample_LMNRAON_Pgbd5$Pgbd5,
                                                                                                      axis    = TRUE,
                                                                                                      border  = FALSE,
                                                                                                      outline = FALSE,
                                                                                                      gp      = gpar(fill = genotype_col[as.vector(as.factor(sample_LMNRAON_Pgbd5$Genotype))], col = "white")
                                                                                                      ),
                                                                      gap                     = my_gap,
                                                                      show_annotation_name    = TRUE,
                                                                      annotation_name_gp      = my_annotation_name_gp,
                                     annotation_legend_param = my_annotation_legend_param,
                                     na_col                  = my_na_col,
                                     height                  = unit(5,"cm")
                                     )

#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAON_mvgenes_Pgbd5.pdf"), width=12, height=16)
Heatmap(matrix                      = my_matrix_LMNRAON_mvgenes_LMNRAON,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = "euclidean",
        show_row_names              = FALSE,
        column_title                = paste0("murine R26 primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes (IQR)",
        top_annotation              = top_annot_LMNRAON,
        show_row_dend               = FALSE,
        bottom_annotation           = bottom_annot_LMNRAON_Pgbd5,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40,"mm")
        )
#dev.off()

```

## Add normal samples to the clustering

### Samples


```{r subset_LMNRAONN_sample}

# Subset Lymphoma, medulloBlastoma, neuroBlastoma, primary RT, allograft RT, orthotopic graft RT and orthotopic graft RT tumor and normal samples
samples_LMNRAONN <- filter(.data          = sampleTable_select,
                           Data_source   == "Curie",
                           Organism      == "Mus_musculus",
                           Tumor_type    == "lymphoma" |
                           Tumor_type    == "NB" |
                           Tumor_type    == "MB" |
                           Tumor_origin  == "orthotopic_graft" |
                           Tumor_origin  == "allograft_flanc" |
                           (Tumor_type   == "RT" &
                            Tumor_origin == "primary_tumor") |
                           Tumor_origin  == "normal_tissue"
                           )

# Add previously assigned group_name
samples_LMNRAONN <- merge(x = samples_LMNRAONN[,-49],
                          y = samples_LMNRAON[,c("Sample_name","group_name")],
                          by = "Sample_name",
                          all=TRUE
                          )

nrow(samples_LMNRAONN)

samples_LMNRAONN[c("Sample_name","Genotype","Tumor_origin","Notes")]

# Summarize
group_by(.data = samples_LMNRAONN,
         Tumor_type,
         Tumor_origin,
         Genotype,
         Notes
         #group_name
         ) %>%
    summarise(number = n()) %>%
    print(n = nrow(.), width = Inf)

# Subset matrix
#my_matrix_LMNRAONN <- my_matrix_annotated[,c("symbol",as.vector(samples_LMNRAONN$Sample_name))]
my_matrix_LMNRAONN <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_LMNRAONN$Sample_name))]

head(my_matrix_unfiltered_annotated)

# Check order
c("symbol_unfiltered",as.vector(samples_LMNRAONN$Sample_name)) == colnames(my_matrix_LMNRAONN)

# Export sample plan
write.table(x         = samples_LMNRAONN,
            file      = paste0(pathToReport,"samples_murine_LR_LMNRAONN.csv"),
            quote     = FALSE,
            row.names = FALSE,
            col.names = TRUE,
            sep       = ";"
            )

# Export matrix
write.table(x         = my_matrix_LMNRAONN,
            file      = paste0(pathToReport,"matrix_murine_RT_LMNRAONN_unfiltered.csv"),
            quote     = FALSE,
            row.names = FALSE,
            col.names = TRUE,
            sep       = ";"
            )

```

```{r mvgenes_LMNRAON}

# most variable genes
mvgenes_LMNRAONN <- genes.selection(my_matrix_LMNRAONN, thres.num = 1000)

# Select matrix
my_matrix_LMNRAONN_mvgenes_LMNRAONN <- my_matrix_LMNRAONN[mvgenes_LMNRAONN,]

```

#### PCA

```{r pca_LMNRAONN}

# Plot PCA
#pdf(paste0(pathToFigures,"pca_LMNRAONN_mvgenes.pdf"), width=14, height=10)
pca_func(matrix                     = my_matrix_LMNRAONN_mvgenes_LMNRAONN,
         sampleTable                = samples_LMNRAONN,
         highlightedVar             = "Genotype",
         highlightedVarName         = "Genotype",
         colors                     = genotype_col,
         sampleLabel                = "Tumor_origin"
         )
#dev.off()

```

#### t-SNE

```{r tSNE_LMNRAONN}

# Plot tSNE 
#pdf(paste0(pathToFigures,"tsne_LMNRAONN_mvgenes.pdf"), width=14, height=10)
tsne_func(matrix                     = my_matrix_LMNRAONN_mvgenes_LMNRAONN,
          sampleTable                = samples_LMNRAONN,
          highlightedVar             = "Genotype",
          highlightedVarName         = "Genotype",
          colors                     = genotype_col,
          sampleLabel                = "Tumor_origin",
          seed                       = 9,
          perplexity                 = 7
          )
#dev.off()

```

#### UMAP analysis

```{r umap_LMNRAONN}

#pdf(paste0(pathToFigures,"umap_LMNRAONN_mvgenes_plus.pdf"), width=16, height=12)
umap_func(matrix                     = my_matrix_LMNRAONN_mvgenes_LMNRAONN,
          sampleTable                =  samples_LMNRAONN,
          highlightedVar             = "Genotype",
          highlightedVarName         = "Genotype",
          colors                     = genotype_col,
          sampleLabel                = "Tumor_origin"
          )
#dev.off()

```


#### Hierarchical clustering

```{r complexHeatmap_U830_LMNRAONN}

# Top annotation
top_annot_LMNRAONN <- createTopAnnot_func(sampleTable = samples_LMNRAONN,
                                        columns     = c("Tumor_type","group_name","Tumor_origin","Genotype"),
                                        colorsList  = list(tumor_type_col, group_name_col, tumor_origin_col,genotype_col)
                                        )

# Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAONN_mvgenes.pdf"), width=12, height=12)
drawHeatmap_func(matrix       = my_matrix_LMNRAONN_mvgenes_LMNRAONN,
                 column_title = "Tumor samples",
                 row_title    = "top variable genes based on IQR values",
                 top_annot    = top_annot_LMNRAONN,
                 legend_title = "log2Intensity"
                 )
#dev.off()

```


# Differential Analysis

Differential expression analysis is performed with moderate Welch t-test implemented in limma package (Ritchie et al., 2015). P-values are corrected for multi-test problem using BH method.

## Which genes are differentially expressed between mIC1 and mIC2 subgroups (primary tumor)?

In this section, we try to find out which genes are differentially expressed between neural and non-neural primary MRT tumors subgroups.

### Differential analysis with limma package

```{r diff_analysis}

# Select neural and non-neural primary tumor
samples_R <- filter(.data          = samples_LMNRAON,
                    Tumor_type     == "RT",
                    Tumor_origin   == "primary_tumor"
                    )
samples_R[,c("Tumor_type","Tumor_origin","Genotype","Sample_name","group_name","sub_localization_of_primary_tumor")]

# Subset RT primary tumors matrix
my_matrix_R <- my_matrix_annotated[,as.vector(samples_R$Sample_name)]

# Create a design matrix
my_class  <- factor(samples_R$group_name)
my_design <- model.matrix(~ 0 + my_class)
my_design
my_fit    <- lmFit(my_matrix_R, my_design)

# Create contrast matrix for pair-wise comparisons
contrast_matrix <- makeContrasts(my_classneural - my_classnon_neural,
                                 levels = my_design
                                 )
fit2 <- contrasts.fit(my_fit, contrast_matrix)
fit2 <- eBayes(fit2)

# Differentially expressed genes
neural_nonNeural <- topTable(fit           = fit2,
                             coef          = 1,
                             number        = Inf,
                             adjust.method = "BH",
                             sort.by       = "logFC"
                             )

# Annotate differentially expressed genes
neural_nonNeural_annotated <- merge(x   = symbol_frame,
                                    y   = neural_nonNeural,
                                    by  = 0,
                                    all = FALSE
                                    )

# Plot P.value histogram
hist(neural_nonNeural_annotated$P.Value)

head(neural_nonNeural_annotated)

# Datatable of toptable
datatable(neural_nonNeural_annotated[,c(1,2,3,7)], filter='top')

# Volcano plot
#pdf(paste0(pathToFigures,"volcano_affy_neural_nonNeural.pdf"),width=12, height=12)
plot_volcano_lima(lima_result        = neural_nonNeural_annotated,
                  title              = "murine R26 neuronal vs non-neuronal",
                  genes              = "",
                  adjPVal_cutoff     = 0.05,
                  logFC_cutoff       = 1,
                  adjPVal_cutoff_lab = 0.001,
                  logFC_cutoff_lab   = 4,
                  geneSup            = ""
                  )
#dev.off()

# Select DEG
res_diffAnalysis_neural_nonNeural_adjPval005_logFC2UP <- filter(.data     = neural_nonNeural_annotated,
                                                                  logFC     > 2,
                                                                  adj.P.Val < 0.05
                                                                  )
nrow(res_diffAnalysis_neural_nonNeural_adjPval005_logFC2UP)

#
res_diffAnalysis_neural_nonNeural_adjPval005_logFC2DN <- filter(.data     = neural_nonNeural_annotated,
                                                                  logFC     < -2,
                                                                  adj.P.Val < 0.05
                                                                  )
nrow(res_diffAnalysis_neural_nonNeural_adjPval005_logFC2DN)

# Export DEG
write.table(x         = res_diffAnalysis_neural_nonNeural_adjPval005_logFC2UP$symbol,
            file      = paste0(pathToReport,"MSigDB_overlap/DEG_neural_nonNeural_adjPval005_logFC2UP.text"),
            quote     = FALSE,
            sep       = "\n",
            row.names = FALSE,
            col.names = FALSE
            )
#
write.table(x         = res_diffAnalysis_neural_nonNeural_adjPval005_logFC2DN$symbol,
            file      = paste0(pathToReport,"MSigDB_overlap/DEG_neural_nonNeural_adjPval005_logFC2DN.text"),
            quote     = FALSE,
            sep       = "\n",
            row.names = FALSE,
            col.names = FALSE
            )

```

### Overalp with GO gene sets

```{r plot_GO_overlap_res}

# Plot overalp result with GO BP gene sets

#pdf(paste0(pathToFigures,"MSigDB_GOBP_neural_nonNeural_adjPval005_logFC2UP.pdf"), width=22, height=8)
plot_top_MSigDB_GO_enriched_geneSets_func(MSigDB_GO_excel_result = paste0(pathToReport,"MSigDB_overlap/MSigDB_GOBP_neural_nonNeural_adjPval005_logFC2UP.xls"),
                                          top                    = 10,
                                          xlim                   = 100,
                                          sep                    = "\t",
                                          quote                  = NULL
                                          )
#dev.off()

#pdf(paste0(pathToFigures,"MSigDB_GOBP_neural_nonNeural_adjPval005_logFC2DN.pdf"), width=22, height=8)
plot_top_MSigDB_GO_enriched_geneSets_func(MSigDB_GO_excel_result = paste0(pathToReport,"MSigDB_overlap/MSigDB_GOBP_neural_nonNeural_adjPval005_logFC2DN.xls"),
                                          top                    = 10,
                                          xlim                   = 100,
                                          sep                    = "\t",
                                          quote                  = NULL
                                          )
#dev.off()

# Plot overalp result with all GO gene sets

#pdf(paste0(pathToFigures,"MSigDB_GO_neural_nonNeural_adjPval005_logFC2UP.pdf"), width=22, height=12)
plot_top_MSigDB_GO_enriched_geneSets_func(MSigDB_GO_excel_result = paste0(pathToReport,"MSigDB_overlap/MSigDB_GO_neural_nonNeural_adjPval005_logFC2UP.xls"),
                                          top                    = 20,
                                          xlim                   = 100,
                                          sep                    = "\t",
                                          quote                  = NULL
                                          )
#dev.off()

#pdf(paste0(pathToFigures,"MSigDB_GO_neural_nonNeural_adjPval005_logFC2DN.pdf"), width=22, height=12)
plot_top_MSigDB_GO_enriched_geneSets_func(MSigDB_GO_excel_result = paste0(pathToReport,"MSigDB_overlap/MSigDB_GO_neural_nonNeural_adjPval005_logFC2DN.xls"),
                                          top                    = 20,
                                          xlim                   = 100,
                                          sep                    = "\t",
                                          quote                  = NULL,
                                          selected_geneSet       = c("GO_IMMUNE_SYSTEM_PROCESS","GO_REGULATION_OF_IMMUNE_SYSTEM_PROCESS","GO_RESPONSE_TO_CYTOKINE")
                                          )
#dev.off()

```

# GSEA analysis

## GSEA neuronal vs non-neuronal

### Create GSEA files

```{r gsea_neuronal_vs_non-neuronal}

head(my_matrix_R_annotated)

# Convert gene symbol to rownames
my_matrix_R_annotated_plus <- my_matrix_R_annotated
my_matrix_R_annotated_plus <- my_matrix_R_annotated_plus[!is.na(my_matrix_R_annotated_plus$symbol),]
rownames(my_matrix_R_annotated_plus) <- my_matrix_R_annotated_plus$symbol
my_matrix_R_annotated_plus$Row.names <- NULL
my_matrix_R_annotated_plus$symbol <- NULL
head(my_matrix_R_annotated_plus)

# Check order
samples_R$Sample_name == colnames(my_matrix_R_annotated_plus)

samples_R$group_name

# Create phenotype or class file according to the .cls format
createGSEAclsFormat_func(class         = samples_R$group_name,
                         classFileName = paste0(pathToReport,"/GSEA_analysis/samples_R.cls")
                         )

# Create expression file according to the gct format                     
createGSEAgctFormat_func(matrix      = my_matrix_R_annotated_plus,
                         gctFileName = paste0(pathToReport,"/GSEA_analysis/matrix_R.gct")
                         )

```

### Plot GSEA results of analysis using GO BP

```{r gsea_DKFZ_U830_ATRT}

pathToGSEAanalysis <- "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/report/GSEA_analysis/"

# Plot GSEA result for SMARCA4_def
#pdf(paste0(pathToFigures,"barplot_GSEA_GO_neuronal_versus_non_neuronal_enriched.pdf"), width=20, height=10)
barplotGSEA_func(GSEAresult       = paste0(pathToGSEAanalysis,"GSEA_GO_neuronal_versus_non_neuronal.Gsea.1561981973478/gsea_report_for_neuronal_1561981973478.xls"),
                 top              = 20
                 )
#dev.off()

#pdf(paste0(pathToFigures,"barplot_GSEA_GO_neuronal_versus_non_neuronal_depleted.pdf"), width=20, height=10)
barplotGSEA_func(GSEAresult       = paste0(pathToGSEAanalysis,"GSEA_GO_neuronal_versus_non_neuronal.Gsea.1561981973478/gsea_report_for_non_neuronal_1561981973478.xls"),
                 top              = 20
                 )
#dev.off()

```

icicicicicici


## Laury Project

### Import Laury Project gene set

```{r import_geneSet}

# Importing geneSet
geneSet <- read.table(paste0(pathToRhabdoid_U830_projects,
                             "geneSet.csv"),
                      header = TRUE,
                      sep    = ";"
                      )
nrow(geneSet)

# Subset KEGG Laury genes
geneSet_KEGGLaury <- filter(.data     = geneSet,
                        Interest  == "KEGG_Laury_genes"
                        )
geneSet_KEGGLaury$Gene

# Subset Laury genes
geneSet_Laury <- filter(.data     = geneSet,
                        referee  == "L_Poulain"
                        )
geneSet_Laury$Gene

# Subset KEGG Laury genes
geneSet_KEGGLaury_TCA <- filter(.data     = geneSet_KEGGLaury,
                                Group  == "TCA_cycle"
                                )
geneSet_KEGGLaury_TCA$Gene

# Subset KEGG Laury genes
geneSet_KEGGLaury_oneCarbone <- filter(.data     = geneSet_KEGGLaury,
                                       Group  == "one_carbone_pool_by_folate"
                                       )
geneSet_KEGGLaury_oneCarbone$Gene

# Subset KEGG Laury genes
geneSet_KEGGLaury_GST <- filter(.data     = geneSet_KEGGLaury,
                                       Group  == "glycine_serine_and_threonine_metabolism"
                                       )
geneSet_KEGGLaury_GST$Gene

# Subset KEGG Laury genes
geneSet_KEGGLaury_gluthatione <- filter(.data     = geneSet_KEGGLaury,
                                        Group  == "glutathione_metabolism"
                                        )
geneSet_KEGGLaury_gluthatione$Gene

# Subset Laury SOG genes
geneSet_Laury_SOG <- filter(.data     = geneSet_Laury,
                                        Group  == "serine_glycine_one_carbon"
                                        )
geneSet_Laury_SOG$Gene

```

#### Convert human gene symbol to mouse gene symbol

```{r gene_symbol_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             filters    = "hgnc_symbol",
                             values     = my_matrix_human$symbol_unfiltered,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )
head(symbols_conversion)

# Convert human gene symbol to orthologous mouse gene symbol

Laury_genes <- merge(x     = geneSet_Laury,
                     y     = symbols_conversion,
                     by.x  = "Gene",
                     by.y  = "HGNC.symbol",
                     all.y = FALSE
                     )$MGI.symbol

#
KEGGLaury_genes_TCA <- merge(x     = geneSet_KEGGLaury_TCA,
                             y     = symbols_conversion,
                             by.x  = "Gene", 
                             by.y  = "HGNC.symbol",
                             all.y = FALSE 
                             )$MGI.symbol
#
KEGGLaury_genes_oneCarbone <- merge(x     = geneSet_KEGGLaury_oneCarbone,
                                    y     = symbols_conversion,
                                    by.x  = "Gene", 
                                    by.y  = "HGNC.symbol",
                                    all.y = FALSE 
                                    )$MGI.symbol
#
KEGGLaury_genes_GST <- merge(x     = geneSet_KEGGLaury_GST,
                             y     = symbols_conversion,
                             by.x  = "Gene", 
                             by.y  = "HGNC.symbol",
                             all.y = FALSE 
                             )$MGI.symbol
#
KEGGLaury_genes_gluthatione <- merge(x     = geneSet_KEGGLaury_gluthatione,
                                     y     = symbols_conversion,
                                     by.x  = "Gene", 
                                     by.y  = "HGNC.symbol",
                                     all.y = FALSE 
                                     )$MGI.symbol
#
Laury_genes_SOG <- merge(x     = geneSet_Laury_SOG,
                                     y     = symbols_conversion,
                                     by.x  = "Gene", 
                                     by.y  = "HGNC.symbol",
                                     all.y = FALSE 
                         )$MGI.symbol

```

### Volcano plot showing Laury gene set

```{r volcano_laury_gene_set}

names(data_neural_nonNeural)

# Volcano plot
#pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_lauryGenes.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Welch t-test, limma package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% Laury_genes,], aes(x=logFC, y=-log(adj.P.Val)), col="blue", size=2.5) +
    geom_text_repel(data=filter(data_neural_nonNeural, -log(adj.P.Val) > 50, logFC > 5 | logFC < -5), aes(label=symbol), size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% Laury_genes,], aes(label=symbol), size=6, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
#dev.off()

```

### Volcano plot showing KEGG Laury gene set

```{r volcano_KEGGlaury_gene_set}

# Volcano plot
#pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_KEGGlauryGenes.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_TCA,], aes(x=logFC, y=-log(adj.P.Val)), col="blue", size=2.5) +
    #geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_TCA,], aes(label=symbol), size=4, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_oneCarbone,], aes(x=logFC, y=-log(adj.P.Val)), col="yellow", size=2.5) +
    #geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_oneCarbone,], aes(label=symbol), size=4, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_GST,], aes(x=logFC, y=-log(adj.P.Val)), col="orange", size=2.5) +
    #geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_GST,], aes(label=symbol), size=4, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_gluthatione,], aes(x=logFC, y=-log(adj.P.Val)), col="black", size=2.5) +
    #geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_gluthatione,], aes(label=symbol), size=4, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    #geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
#dev.off()

```

```{r volcano_KEGGlaury_TCA_gene_set}

# Volcano plot
pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_KEGGlauryGenes_TCA.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_TCA,], aes(x=logFC, y=-log(adj.P.Val)), col="blue", size=2.5) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_TCA,], aes(label=symbol), size=6, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
dev.off()

```

```{r volcano_KEGGlaury_oneCarbone_gene_set}

# Volcano plot
pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_KEGGlauryGenes_oneCarbone.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_oneCarbone,], aes(x=logFC, y=-log(adj.P.Val)), col="yellow", size=2.5) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_oneCarbone,], aes(label=symbol), size=6, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
dev.off()

```

```{r volcano_KEGGlaury_GST_gene_set}

# Volcano plot
pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_KEGGlauryGenes_GST.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_GST,], aes(x=logFC, y=-log(adj.P.Val)), col="orange", size=2.5) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_GST,], aes(label=symbol), size=6, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
dev.off()

```

```{r volcano_KEGGlaury_GST_gene_set}

# Volcano plot
pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_KEGGlauryGenes_gluthatione.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_gluthatione,], aes(x=logFC, y=-log(adj.P.Val)), col="black", size=2.5) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% KEGGLaury_genes_gluthatione,], aes(label=symbol), size=6, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
dev.off()

```

```{r volcano_laury_SOG_gene_set}

# Volcano plot
pdf(paste0(pathToFigures,"volcano_affy_mIC_mEIC_LauryGenes_SOG.pdf"),width=12, height=12)
ggplot(data = data_neural_nonNeural,
       aes(x=logFC, y=-log(adj.P.Val))
       ) +
    #ylim(0,100) + 
    geom_point(aes(col=crit), pch=21) +
    ggtitle("neural vs non neural (Wald test, DESeq2 package)") +
    scale_color_manual(values=c("forestgreen","grey","brown","grey45")) +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% Laury_genes_SOG,], aes(x=logFC, y=-log(adj.P.Val)), col="black", size=2.5) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol %in% Laury_genes_SOG,], aes(label=symbol), size=6, fontface="bold") +
    #
    geom_point(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(x=logFC, y=-log(adj.P.Val)), col="red", size=4) +
    geom_text_repel(data = data_neural_nonNeural[data_neural_nonNeural$symbol == "Myc",], aes(label=symbol), size=4, fontface="bold") +
    geom_hline(yintercept = -log(0.05), color= "red")+
    geom_vline(xintercept = 1, color= "blue")+
    geom_vline(xintercept = -1, color= "blue")+
    theme_bw() +
    theme(legend.position = "none")
dev.off()

```


### Gene ontology analysis

```{r gene_ontology}

# Export neural overexpressed genes for Gene Ontology analysis
neural_vs_nonNeural_DEG_UP <- filter(.data = neural_nonNeural_annotated,
                                     adj.P.Val < 0.001,
                                     logFC > 3
                                     )["symbol"]
write.table(x         = neural_vs_nonNeural_DEG_UP,
            file      = paste0(pathToReport,"neural_vs_nonNeural_DEG_UP.txt"),
            quote     = FALSE,
            row.names = FALSE,
            col.names = FALSE
            )

# Export non-neural overexpressed genes for Gene Ontology analysis
neural_vs_nonNeural_DEG_DOWN <- filter(.data = neural_nonNeural_annotated,
                                       adj.P.Val < 0.001,
                                       logFC < 3
                                       )["symbol"]
write.table(x         = neural_vs_nonNeural_DEG_UP,
            file      = paste0(pathToReport,"neural_vs_nonNeural_DEG_UP.txt"),
            quote     = FALSE,
            row.names = FALSE,
            col.names = FALSE
            )

```

### Heatmap on differentially expressed genes (neural vs nonNeural)

```{r heatmap_on_DEG}

# filter differentially expressed genes
logFC_threshold     <- 4
adj.P.Val_threshold <- 0.001

neural_nonNeural_annotated_filtered <- filter(.data      = neural_nonNeural_annotated,
                                              logFC     >= logFC_threshold |
                                              logFC     <= -logFC_threshold,
                                              adj.P.Val <= adj.P.Val_threshold
                                       )
nrow(neural_nonNeural_annotated_filtered)

neural_nonNeural_DEG <- neural_nonNeural_annotated_filtered$symbol

# Annotate matrix
my_matrix_R_annotated <- merge(x   = symbol_frame,
                               y   = my_matrix_R,
                               by  = 0,
                               all = FALSE
                               )

# Subset matrix
my_matrix_R_DEG <- filter(.data = my_matrix_R_annotated,
                                 symbol %in% neural_nonNeural_DEG
                                 )
#head(my_matrix_R_DEG)

my_matrix_R_DEG_plus           <- my_matrix_R_DEG
rownames(my_matrix_R_DEG_plus) <- my_matrix_R_DEG_plus$symbol
my_matrix_R_DEG_plus           <- my_matrix_R_DEG_plus[,-c(1,2)]
head(my_matrix_R_DEG_plus)

# Top annotation
my_top_annot_DEG <- HeatmapAnnotation(df                      = data.frame(samples_R[,c("Tumor_origin","group_name")]),
                                          col                     = list(Tumor_origin = tumor_origin_col,
                                                                         group_name = group_name_col
                                                                         ),
                                          gap                     = my_gap,
                                          show_annotation_name    = TRUE,
                                          annotation_name_gp      = my_annotation_name_gp,
                                          annotation_legend_param = my_annotation_legend_param,
                                          na_col                  = my_na_col
                                        )

#pdf(paste0(pathToFigures,"complexHeatmap_mIC1_mIC2_DEG.pdf"), width=8, height=18)
Heatmap(matrix                      = my_matrix_R_DEG_plus,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = TRUE,
        cluster_columns             = TRUE,
        column_title                = "Hierarchical clustering of murine primary tumors",
        row_title                   = "neural and non-neural DEG genes",
        top_annotation              =  my_top_annot_DEG,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

### GO enrichment analysis with topGO

```{r GO_analysis, eval=FALSE}

#TO DO

# Data preparation
library(ALL)
data(geneList)
head(geneList)

?geneList

```

# GSEA analysis

## GSEA between neural and non-neural

```{r gsea_neural_vs_nonNeural}

head(my_matrix_R)
samples_R$group_name

# Add gene symbol to matrix
head(symbol_unfiltered_frame)

my_matrix_R_annot <- merge(x = symbol_unfiltered_frame,
                           y = my_matrix_R,
                           by = 0,
                           all = FALSE
                           )
my_matrix_R_annot <- my_matrix_R_annot[!duplicated(my_matrix_R_annot$symbol_unfiltered),]
my_matrix_R_annot <- my_matrix_R_annot[!is.na(my_matrix_R_annot$symbol_unfiltered),]
rownames(my_matrix_R_annot) <- my_matrix_R_annot$symbol_unfiltered
my_matrix_R_annot$Row.names <- NULL
colnames(my_matrix_R_annot)[1] <- "symbol"
head(my_matrix_R_annot)

# Save R26 in situ data in RDS file
murine_RT_R26_affy_list <- list(sampleTable = samples_R,
                                matrix = my_matrix_R_annot
                                )

saveRDS(object = murine_RT_R26_affy_list,
        file = 'murine_RT_R26_affy_list.RDS'
        )


# Create phenotype or class file according to the .cls format
gsea_mICs_class      <- as.vector(samples_R$group_name)
gsea_mICs_class
gsea_mICs_class_file <- file("gsea_mICs_affy_class.cls")
writeLines(text = paste(as.character(length(gsea_mICs_class)),
                        as.character(length(unique(gsea_mICs_class))),
                        "1",
                        sep="\t"),
           con  = gsea_mICs_class_file
           )
close(gsea_mICs_class_file)
cat(c("#", unique(gsea_mICs_class), "\n"),
    sep    = "\t",
    file   = "gsea_mICs_affy_class.cls",
    append = TRUE
    )
cat(gsea_mICs_class,
    sep    = "\t",
    file   = "gsea_mICs_affy_class.cls",
    append = TRUE
    )

# Create expression file according to the gct format                     
my_matrix_R_gct_file <- file("my_matrix_R_affy.gct")
writeLines(text = "#1.2",
           con  = my_matrix_R_gct_file
           )
close(my_matrix_R_gct_file)
cat(c(nrow(my_matrix_R_annot), ncol(my_matrix_R_annot)-1, "\n"),
    sep    = "\t",
    file   = "my_matrix_R_affy.gct",
    append = TRUE
    )
head(my_matrix_R_annot)
my_matrix_R_gct             <- my_matrix_R_annot
my_matrix_R_gct$Description <- rownames(my_matrix_R_gct)
my_matrix_R_gct$NAME        <- my_matrix_R_gct$symbol
my_matrix_R_gct             <- cbind(my_matrix_R_gct[c("NAME", "Description")],
                                     my_matrix_R_gct[,4:ncol(my_matrix_R_gct)-2]
                                     )
head(my_matrix_R_gct)
write.table(x         = my_matrix_R_gct,
            file      = "my_matrix_R_affy.gct",
            quote     = FALSE,
            sep       = "\t",
            row.names = FALSE,
            append    = TRUE
            )
write.table(x         = my_matrix_R_gct,
            file      = "my_matrix_R_affy.txt",
            quote     = FALSE,
            sep       = "\t",
            row.names = FALSE,
            append    = FALSE
            )

# TO DO: include here the gsea script (bash or R)

```

# Analysis of normal brain samples (3 from Curie and 4 from Cole et al., 2010)

### Samples


```{r subset_normal_sample}

# Subset samples
samples_normal <- filter(.data          = sampleTable_select,
                         #Data_source   == "literature",
                         Tumor_type    == "normal"
                         )

# Subset matrix
my_matrix_normal <- my_matrix_annotated[,as.vector(samples_normal$Sample_name)]

```

```{r mvgenes_normal}

# most variable genes
mvgenes_normal <- genes.selection(my_matrix_normal, thres.num = 1000)

# Select matrix
my_matrix_normal_mvgenes_normal <- my_matrix_normal[mvgenes_normal,]

```

### PCA analysis

```{r pca_normal}

# PCA
p <- plot_pca(matrix      = my_matrix_normal_mvgenes_normal,
              category    = samples_normal$Tumor_type,
              addEllipses = FALSE,
              axes        = c(1,2)
         )
p

```

### Heatmap

```{r complexHeatmap_normal}

# Top annotation
my_top_annot_normal <- HeatmapAnnotation(df                      = data.frame(samples_normal[,c("Data_source","Tumor_type")]),
                                         col                     = list(Data_source = data_source_col,
                                                                        Tumor_type = tumor_type_col
                                                                        ),
                                         gap                     = my_gap,
                                         show_annotation_name    = TRUE,
                                         annotation_name_gp      = my_annotation_name_gp,
                                         annotation_legend_param = my_annotation_legend_param,
                                         na_col                  = my_na_col
                                         )


# Complex heatmap
my_compHeat_normal <- Heatmap_func(matrix         = my_matrix_normal_mvgenes_normal,
                                   column_title   = paste0("murine tumors clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
                                   row_title      = "1000 most variable genes",
                                   top_annotation = my_top_annot_normal
                                   )
#pdf(paste0(pathToFigures,"complexHeatmap_LMNRAON_mvgenes.pdf"), width=12, height=12)
my_compHeat_normal
#dev.off()

```

# Analysis of Ng samples (5 ATRT and 5 MB)

### Samples

```{r subset_normal_sample}

# Subset samples
samples_Ng2015 <- filter(.data              = sampleTable_select,
                         Data_source       == "literature",
                         Author_Researcher == "Ng_et_al_2015"
                         )

samples_Ng2015

# Subset matrix
my_matrix_Ng2015 <- my_matrix_annotated[,as.vector(samples_Ng2015$Sample_name)]

```

```{r mvgenes_normal}

# most variable genes
mvgenes_Ng2015 <- genes.selection(my_matrix_Ng2015, thres.num = 1000)

# Select matrix
my_matrix_Ng2015_mvgenes_Ng2015 <- my_matrix_Ng2015[mvgenes_Ng2015,]

```

### PCA analysis

```{r pca_Ng2015}

# PCA
p <- plot_pca(matrix      = my_matrix_Ng2015_mvgenes_Ng2015,
              category    = samples_Ng2015$Tumor_type,
              addEllipses = FALSE,
              axes        = c(1,2)
         )
p

```

### Heatmap

```{r complexHeatmap_normal}

# Top annotation
my_top_annot_Ng2015 <- HeatmapAnnotation(df                      = data.frame(samples_Ng2015[,c("Tumor_type","Author_Researcher")]),
                                         col                     = list(Tumor_type = tumor_type_col,
                                                                        Author_Researcher= author_researcher_col
                                                                        ),
                                         gap                     = my_gap,
                                         show_annotation_name    = TRUE,
                                         annotation_name_gp      = my_annotation_name_gp,
                                         annotation_legend_param = my_annotation_legend_param,
                                         na_col                  = my_na_col
                                         )

# Complex heatmap
my_compHeat_Ng2015 <- Heatmap_func(matrix         = my_matrix_Ng2015_mvgenes_Ng2015,
                                   column_title   = paste0("murine tumors clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
                                   row_title      = "1000 most variable genes",
                                   top_annotation = my_top_annot_Ng2015
                                   )
#pdf(paste0(pathToFigures,"complexHeatmap_murine_ATRT_MB_Ng2015_mvgenes.pdf"), width=12, height=16)
my_compHeat_Ng2015
#dev.off()

```

```{r Pgdb5_expression}

head(my_matrix_Ng2015[,1:4])

# Add gene symbol column
my_matrix_Ng2015_annotated <- my_matrix_annotated[,c("symbol",as.vector(samples_Ng2015$Sample_name))]

# Extract Pdgb5 gene expression 
my_matrix_Ng2015_Pgbd5 <- filter(.data = my_matrix_Ng2015_annotated,
                               symbol == "Pgbd5"
                               )%>%
    #dplyr::select(-ENSGID) %>%
    melt(.) %>%
    dplyr::select(-symbol)
colnames(my_matrix_Ng2015_Pgbd5) <- c("Sample_name","Pgbd5")
head(my_matrix_Ng2015_Pgbd5)

# Merge coldata and matrix
samples_Ng2015_Pgbd5 <- merge(x   = samples_Ng2015,
                           y   = my_matrix_Ng2015_Pgbd5,
                           by  = "Sample_name",
                           all = FALSE
                           )
head(samples_Ng2015_Pgbd5)

samples_Ng2015_Pgbd5$Sample_name == samples_Ng2015$Sample_name

# Bottom annotation
bottom_Ng2015_Pgbd5 <- HeatmapAnnotation(Pgdb5_expression = anno_barplot(x       = samples_Ng2015_Pgbd5$Pgbd5,
                                                                                                      axis    = TRUE,
                                                                                                      border  = FALSE,
                                                                                                      outline = FALSE,
                                                                                                      gp      = gpar(fill = tumor_type_col[as.vector(as.factor(samples_Ng2015_Pgbd5$Tumor_type))], col = "white")
                                                                                                      ),
                                                                      gap                     = my_gap,
                                                                      show_annotation_name    = TRUE,
                                                                      annotation_name_gp      = my_annotation_name_gp,
                                     annotation_legend_param = my_annotation_legend_param,
                                     na_col                  = my_na_col
                                     )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_ATRT_MB_Ng2015_mvgenes_Pgbd5.pdf"), width=12, height=16)
Heatmap(matrix                      = my_matrix_Ng2015_mvgenes_Ng2015,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = "euclidean",
        show_row_names              = FALSE,
        column_title                = paste0("murine primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes (IQR)",
        top_annotation              = my_top_annot_Ng2015,
        show_row_dend               = FALSE,
        bottom_annotation           = bottom_Ng2015_Pgbd5,
        bottom_annotation_height    = unit(5,"cm"),
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40,"mm")
        )
#dev.off()

# Student test
t.test(x=sample_LMNR_Pgbd5[which(sample_LMNR_Pgbd5$group_name == "neural" & sample_LMNR_Pgbd5$Tumor_type == "RT"),"Pgbd5"],
       y=sample_LMNR_Pgbd5[which(sample_LMNR_Pgbd5$group_name == "non_neural"& sample_LMNR_Pgbd5$Tumor_type == "RT" & sample_LMNR_Pgbd5$Localization_of_primary_tumor == "IC"
                                 ),
                           "Pgbd5"])$p.value

```

# Specific Analysis

## Import interested gene sets

```{r importGeneSet}

# Importing geneSet
geneSet <- read.table(file   = paste0(pathToRhabdoid_U830_projects, "geneSet.csv"),
                      header = TRUE,
                      sep    = ";"
                      )

# Subset human_ATRT geneSet
geneSet_ATRT <- filter(.data     = geneSet,
                       Interest == "ATRT",
                       referee  == "M_Andrianteranagna",
                       source   == ""
                       )
nrow(geneSet_ATRT)

```

### Convert human gene symbol to mouse gene symbol

```{r create_geneSymbol_conversion}

# Create a data frame of human and murine gene symbols correspondance
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             #attributes = c("external_gene_name"),
                             filters    = "hgnc_symbol",
                             #filters    = "external_gene_name",
                             #values     = hs_cytokines_genes_selected,
                             values     = geneSet_ATRT$Gene,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )

colnames(symbols_conversion) <- c("Gene","murine_symbol")

head(symbols_conversion)

names(geneSet_ATRT)

# Convert human symbol of ATRT gene set into murine gene symbol 
geneSet_ATRT <- merge(x   = geneSet_ATRT,
                      y   = symbols_conversion,
                      by  = "Gene",
                      all = FALSE
                      )
names(geneSet_ATRT)

```

## Murine primary tumor samples

### Heatmap of murine primary tumor samples on 1000 most variable genes

```{r pca_hc_heatmap_mIC}

samples_R[,c("Sample_name","Tumor_origin","Tumor_type","group_name","Localization_of_primary_tumor","sub_localization_of_primary_tumor")]

# most variable genes across mICs samples
mvgenes_R <- genes.selection(my_matrix_R, thres.num = 1000)

# Select matrix
my_matrix_R_mvgenes_R <- my_matrix_R[mvgenes_R,]

group_name_col

data.frame(groupe_name = samples_R[,"group_name"])

# Top annotation
my_top_annot_R <- HeatmapAnnotation(df                      = data.frame(subgroup = samples_R[,"group_name"]),
                                    col                     = list(subgroup = group_name_col
                                                                   ),
                                    gap                     = my_gap,
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = my_annotation_name_gp,
                                    annotation_legend_param = my_annotation_legend_param,
                                    na_col                  = my_na_col
                                    )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_R26_primTum_mvgenes.pdf"), width=10, height=12)
Heatmap(matrix                      = my_matrix_R_mvgenes_R,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Unsupervised hierarchical clustering of murine primary tumors",
        row_title                   = "1000 top most variable genes",
        top_annotation              =  my_top_annot_R,
        show_heatmap_legend         = TRUE,
        show_row_dend               = FALSE,
        heatmap_legend_param = list(title = "log2intensity")
        #column_dend_height          = my_column_dend_height,
        #column_names_gp             = my_column_names_gp,
        #row_title_gp                = my_row_title_gp,
        #row_names_gp                = my_row_names_gp
        )
#dev.off()

```

### Assigning mIC and mIC_EC murine RT samples

```{r assign_samples}

# Cut dendogram into 2 groups
pl <- Heatmap(matrix                      = my_matrix_R_mvgenes_R,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth
              )
hc <- as.hclust(column_dend(pl))
group_num <- cutree(tree = hc,
                    k    = 2,
                    h    = NULL
                    )
group <- melt(group_num)
colnames(group) <- "group_num"
group$group_name <- ifelse(group$group_num == "1","neural","nonNeural")
group$Sample_name <- rownames(group)
group

#samples_R$group_name <- ifelse(samples_R$group_name == "neural", "mIC", "mIC_EC")
#samples_R[,c("Sample_name","group_name")]

sampleTable_without_RTprimaryTumors <- filter(.data          = sampleTable_select,
                                              Tumor_type    == "healthy" |
                                              Tumor_type    == "lymphoma" |
                                              Tumor_type    == "medulloBlastoma" |
                                              Tumor_type    == "neuroBlastoma" |
                                              (Tumor_type   == "RT" &
                                               Tumor_origin != "primary_tumor"
                                              )
                                              )
    
sampleTable_murine <- rbind(sampleTable_without_RTprimaryTumors,
                            samples_R
                            )

sampleTable_murine[,c("Genotype","Sample_name","group_name")]

```

### Construct an ExpressionSet object

For data integration using the AGDEX method.

```{r expressionSet_construction, eval=FALSE}

my_matrix_murine <- my_matrix_unfiltered_annotated[,-1]
names(my_matrix_murine)
ncol(my_matrix_murine)

# assay data
assayData_murine <- as.matrix(my_matrix_murine)

# select sample with Genotype == "Nestin" and create a groupe_name_tmp
sampleTable_murine_Nes <- filter(.data = sampleTable_murine,
                                 Genotype == "Nestin"
                                 )
sampleTable_murine_Nes[,c("Genotype","group_name")]
sampleTable_murine_Nes$group_name_tmp <- sampleTable_murine_Nes$Genotype
sampleTable_murine_Nes[,c("Sample_name","Genotype","group_name","group_name_tmp")]

# select sample with Genotype != "Nestin" and create a groupe_name_tmp
sampleTable_murine_withoutNes <- sampleTable_murine[which(sampleTable_murine$Genotype != "Nestin" | is.na(sampleTable_murine$Genotype)),]
sampleTable_murine_withoutNes[,c("Genotype","group_name")]
sampleTable_murine_withoutNes$group_name_tmp <- sampleTable_murine_withoutNes$group_name
sampleTable_murine_withoutNes[,c("Genotype","group_name","group_name_tmp")]

# Create a data frame with goup_name includes the "Nestin"
sampleTable_murine_pData <- rbind(sampleTable_murine_Nes,
                                  sampleTable_murine_withoutNes
                                  )
sampleTable_murine_pData[,c("Genotype","group_name","group_name_tmp")]

sampleTable_murine_pData$group_name <- sampleTable_murine_pData$group_name_tmp
sampleTable_murine_pData$group_name_tmp <- NULL
sampleTable_murine_pData[,c("Sample_name","Genotype","group_name")]

# Reorde table Sample_name according to matrix colnames order
sampleTable_murine_pData_reordered <- sampleTable_murine_pData[match(colnames(my_matrix_murine),
                                                         sampleTable_murine_pData$Sample_name),]

phenotypicData_murine <- sampleTable_murine_pData_reordered
rownames(phenotypicData_murine) <- phenotypicData_murine[,"Sample_name"]

# import metaData
metaData <- read.table(file      = paste0(pathToRhabdoid_U830_projects, "metaData.csv"),
                       header    = TRUE,
                       sep       = ";",
                       fill      = TRUE,
                       row.names = 1
                       )

# check if the metaData and the phenotypicData have the same row/colnames
colnames(phenotypicData_murine) == rownames(metaData)

# build pheno data
phenoData_murine <- new("AnnotatedDataFrame",
                        data        = phenotypicData_murine,
                        varMetadata = metaData
                        )

# built ExpressionSet object
expressionSet_murine <- ExpressionSet(assayData  = assayData_murine,
                                      phenoData  = phenoData_murine,
                                      annotation = "mouse4302mmentrezg"
                                      )

# save ExpressionSet object
save(expressionSet_murine,
     file               = "expressionSet_murine.RData",
     version            = NULL
     )

```


### Clustering of murine primary tumor samples using the human ATRT signature genes

```{r mICs_and_ATRT_geneSet}

# Subset matrix
my_matrix_unfiltered_R <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_R$Sample_name))]

# Select ATRT signature genes
my_matrix_R_ATRTsign_plus <- merge(x = geneSet_ATRT,
                              y = my_matrix_unfiltered_R,
                              by.x = "murine_symbol",
                              by.y = "symbol_unfiltered",
                              all = FALSE
                              )
row.names(my_matrix_R_ATRTsign_plus) <- my_matrix_R_ATRTsign_plus$murine_symbol
my_matrix_R_ATRTsign_plus <- my_matrix_R_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_R_ATRTsign_plus      <- my_matrix_R_ATRTsign_plus[order(my_matrix_R_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_R_ATRTsign           <- my_matrix_R_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_R_ATRTsign           <- my_matrix_R_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_R_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot       <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_R_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "mICs samples clustering",
              row_title                   = "ATRT genes",
              #top_annotation              = my_top_annot,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin samples (all passages)

### Nestin samples and (mIC1 versus mIC2) differentially expressed genes

```{r nestin_samples_heatmap_on_mIC_diffGenes}

# Subset samples
samples_Nestin <- filter(.data           = samples_LMNRAON,
                         #Tumor_type     == "RT",
                         #Tumor_sub_type == "ATRT",
                         Tumor_origin   == "allograft_flanc",
                         Genotype       == "Nestin"
                         )

samples_Nestin

# Subset matrix
my_matrix_Nestin_annotated <- my_matrix_annotated[,c("symbol",as.vector(samples_Nestin$Sample_name))]
head(my_matrix_Nestin_annotated)

# Subset matrix
my_matrix_Nestin_DEG <- filter(.data = my_matrix_Nestin_annotated,
                               symbol %in% neural_nonNeural_DEG
                               )
#head(my_matrix_Nestin_DEG)

my_matrix_Nestin_DEG_plus           <- my_matrix_Nestin_DEG
rownames(my_matrix_Nestin_DEG_plus) <- my_matrix_Nestin_DEG_plus$symbol
my_matrix_Nestin_DEG_plus           <- my_matrix_Nestin_DEG_plus[,-1]
#head(my_matrix_Nestin_DEG_plus)

my_top_annot_Nestin <- HeatmapAnnotation(df  = samples_Nestin[,c("Tumor_origin","Genotype")],
                                         col = list(Genotype     = genotype_col,
                                                    Tumor_origin = tumor_origin_col
                                                    ),
                                         gap = my_gap,
                                         show_annotation_name = TRUE,
                                         annotation_name_gp = my_annotation_name_gp,
                                         annotation_legend_param = my_annotation_legend_param,
                                         na_col = my_na_col
                                         )

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_DEG_mIC1_mIC2.pdf"), width=6, height=18)
#dev.new(width=6, height=18)
Heatmap(matrix                      = my_matrix_Nestin_DEG_plus,
        show_row_names              = TRUE,
        cluster_columns             = TRUE,
        column_title                = "Unsupervised hierarchical clustering of murine Nestin samples",
        row_names_gp                = gpar(fontsize = 8),
        row_title                   = "neural vs nonNeural DEG genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Clustering of Nestin samples using the human ATRT signature genes

```{r NestinAll_and_ATRT_geneSet}

# Subset matrix
my_matrix_unfiltered_Nestin <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_Nestin$Sample_name))]

# Select ATRT signature genes
my_matrix_Nestin_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                        y    = my_matrix_unfiltered_Nestin,
                                        by.x = "murine_symbol",
                                        by.y = "symbol_unfiltered",
                                        all  = FALSE
                                        )
row.names(my_matrix_Nestin_ATRTsign_plus) <- my_matrix_Nestin_ATRTsign_plus$murine_symbol
my_matrix_Nestin_ATRTsign_plus <- my_matrix_Nestin_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_Nestin_ATRTsign_plus <- my_matrix_Nestin_ATRTsign_plus[order(my_matrix_Nestin_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_Nestin_ATRTsign           <- my_matrix_Nestin_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_Nestin_ATRTsign           <- my_matrix_Nestin_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_Nestin_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_Nestin_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Unsupervised hierarchical clustering of murine primary tumor samples",
              row_title                   = "ATRT signature genes",
              #top_annotation              = my_top_annot_Nestin,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```


## Nestin (all passages), mIC1 and mIC2 samples

### Samples

```{r Nestin_mICs_samples}

herehere

# Subset samples
#samples_R_Nestin <- rbind.fill(samples_R,
 #                              samples_Nestin
  #                             )

samples_R_Nestin <- rbind(samples_R,
                          samples_Nestin
                          )

samples_R_Nestin

# Subset matrix
my_matrix_R_Nestin_annotated <- my_matrix_annotated[,c("symbol",as.vector(samples_R_Nestin$Sample_name))]
#head(my_matrix_R_Nestin_annotated)

```

### Heatmap of Nestin and mICs samples on mIC1 vs mIC2 DEG

```{r Nestin_mICs_heatmap}

# Subset matrix
my_matrix_R_Nestin_DEG <- filter(.data     = my_matrix_R_Nestin_annotated,
                                 symbol %in% neural_nonNeural_DEG
                                 )
#head(my_matrix_R_Nestin_DEG)

my_matrix_R_Nestin_DEG_plus           <- my_matrix_R_Nestin_DEG
rownames(my_matrix_R_Nestin_DEG_plus) <- my_matrix_R_Nestin_DEG_plus$symbol
my_matrix_R_Nestin_DEG_plus           <- my_matrix_R_Nestin_DEG_plus[,-1]
#head(my_matrix_R_Nestin_DEG_plus)

# Top annotation
my_top_annot_R_Nestin <- HeatmapAnnotation(df  = samples_R_Nestin[,c("Genotype","group_name")],
                                           col = list(Genotype   = genotype_col,
                                                      group_name = group_name_col
                                                      ),
                                           gap = my_gap,
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_DEG_mIC1_mIC2.pdf"), width=6, height=18)
#dev.new(width=6, height=18)
Heatmap(matrix                      = my_matrix_R_Nestin_DEG_plus,
        show_row_names              = TRUE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and allograft Nestin samples",
        row_names_gp                = gpar(fontsize = 8),
        row_title                   = "neural vs non-neural DEG genes",
        top_annotation              =  my_top_annot_R_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Heatmap of Nestin samples (all passages) and mICs samples on 100 and 1000 most variables genes

```{r pca_hc_heatmap_Nestin_mICs}

# Top annotation
my_top_annot_R_Nestin2 <- HeatmapAnnotation(df  = samples_R_Nestin[,c("Tumor_origin","Genotype","group_name")],
                                  col = list(Genotype     = genotype_col,
                                             Tumor_origin = tumor_origin_col,
                                             group_name   = group_name_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                  )

# most variable genes across Nestin and mICs samples
mvgenes100_R_Nestin  <- genes.selection(my_matrix_R_Nestin_annotated[,-1], thres.num = 100)
mvgenes1000_R_Nestin <- genes.selection(my_matrix_R_Nestin_annotated[,-1], thres.num = 1000)

# matrix with mvgenes
my_matrix_R_Nestin_mvgenes100_R_Nestin  <- my_matrix_R_Nestin_annotated[mvgenes100_R_Nestin,-1]
my_matrix_R_Nestin_mvgenes1000_R_Nestin <- my_matrix_R_Nestin_annotated[mvgenes1000_R_Nestin,-1]

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_mIC_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_R_Nestin_mvgenes100_R_Nestin,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and allograft samples",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_R_Nestin2,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_mIC_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_R_Nestin_mvgenes1000_R_Nestin,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and allograft samples",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_R_Nestin2,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Clustering of Nestin (all passages) and mICs samples using the human ATRT signature genes

```{r NestinAll_mICs_and_ATRT_geneSet}

# Subset matrix
my_matrix_unfiltered_R_Nestin <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_R_Nestin$Sample_name))]

# Select ATRT signature genes
my_matrix_R_Nestin_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                          y    = my_matrix_unfiltered_R_Nestin,
                                          by.x = "murine_symbol",
                                          by.y = "symbol_unfiltered",
                                          all  = FALSE
                                          )
row.names(my_matrix_R_Nestin_ATRTsign_plus) <- my_matrix_R_Nestin_ATRTsign_plus$murine_symbol
my_matrix_R_Nestin_ATRTsign_plus <- my_matrix_R_Nestin_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_R_Nestin_ATRTsign_plus      <- my_matrix_R_Nestin_ATRTsign_plus[order(my_matrix_R_Nestin_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_R_Nestin_ATRTsign           <- my_matrix_R_Nestin_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_R_Nestin_ATRTsign           <- my_matrix_R_Nestin_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_R_Nestin_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_R_Nestin_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Murine primary tumors and Nestin samples (all passages)",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_R_Nestin2,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin (passage 0 only) and mICs samples

### Samples

```{r samples_NestinP0_mICs}

# Subset sample
samples_R_NestinP0 <- filter(.data            = samples_R_Nestin,
                             graft_passage   == "P0" |
                             is.na(graft_passage)
                             )
samples_R_NestinP0

```

### Heatmap of Nestin samples (P0 only) and mICs samples on 100 and 1000 most variables genes

```{r hc_heatmap_NestinP0_mICs}

# Subset matrix
my_matrix_R_NestinP0 <- my_matrix_annotated[,as.vector(samples_R_NestinP0$Sample_name)]

# most variable genes across Nestin and mICs samples
mvgenes100_R_NestinP0  <- genes.selection(my_matrix_R_NestinP0, thres.num = 100)
mvgenes1000_R_NestinP0 <- genes.selection(my_matrix_R_NestinP0, thres.num = 1000)

# matrix with mvgenes
my_matrix_R_NestinP0_mvgenes100_R_NestinP0  <- my_matrix_R_NestinP0[mvgenes100_R_NestinP0,]
my_matrix_R_NestinP0_mvgenes1000_R_NestinP0 <- my_matrix_R_NestinP0[mvgenes1000_R_NestinP0,]

# Top annotation
my_top_annot_R_NestinP0 <- HeatmapAnnotation(df  = samples_R_NestinP0[,c("Tumor_origin","Genotype","group_name")],
                                  col = list(Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col,
                                             group_name   = group_name_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                 )

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_P0_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_R_NestinP0_mvgenes100_R_NestinP0,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and Nestin P0 samples",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_R_NestinP0,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_P0_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_R_NestinP0_mvgenes1000_R_NestinP0,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and Nestin P0 samples",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_R_NestinP0,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Clustering of Nestin samples using the human ATRT signature genes

```{r nestin_and_ATRT_geneSet}

# Subset matrix
my_matrix_unfiltered_R_NestinP0 <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_R_NestinP0$Sample_name))]

# Select ATRT signature genes
my_matrix_R_NestinP0_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                            y    = my_matrix_unfiltered_R_NestinP0,
                                            by.x = "murine_symbol",
                                            by.y = "symbol_unfiltered",
                                            all  = FALSE
                                            )
row.names(my_matrix_R_NestinP0_ATRTsign_plus) <- my_matrix_R_NestinP0_ATRTsign_plus$murine_symbol
my_matrix_R_NestinP0_ATRTsign_plus <- my_matrix_R_NestinP0_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_R_NestinP0_ATRTsign_plus      <- my_matrix_R_NestinP0_ATRTsign_plus[order(my_matrix_R_NestinP0_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_R_NestinP0_ATRTsign           <- my_matrix_R_NestinP0_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_R_NestinP0_ATRTsign           <- my_matrix_R_NestinP0_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_R_NestinP0_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_R_NestinP0_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Nestin samples (P0 only) clustering",
              row_title                   = "ATRT genes",
              top_annotation              = my_top_annot_R_NestinP0,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin (independent tumor) and mICs samples

### Samples

```{r samples_NestinIndep_mICs}

# Subset sample
samples_R_NestinInd <- filter(.data           = samples_R_Nestin,
                              Tumor_origin   == "primary_tumor" |
                              graft_passage  == "P0" |
                              graft_passage  == "" |
                              (graft_passage == "P1" & Sample_name == "43971-A-NBA4P1")
                              )
samples_R_NestinInd

```

### Heatmap of Nestin (independent tumours only) and mICs samples on 100 and 1000 most variables genes

```{r heatmap_NestinIndep_mICs}

# Subset mICs and Nestin independent samples matrix
my_matrix_R_NestinInd <- my_matrix_annotated[,as.vector(samples_R_NestinInd$Sample_name)]

# most variable genes
mvgenes100_R_NestinInd  <- genes.selection(my_matrix_R_NestinInd, thres.num = 100)
mvgenes1000_R_NestinInd <- genes.selection(my_matrix_R_NestinInd, thres.num = 1000)

# subset matrix mvgenes
my_matrix_R_NestinInd_mvgenes100_R_NestinInd  <- my_matrix_R_NestinInd[mvgenes100_R_NestinInd,]
my_matrix_R_NestinInd_mvgenes1000_R_NestinInd <- my_matrix_R_NestinInd[mvgenes1000_R_NestinInd,]

# Top annotation    
my_top_annot_R_NestinInd <- HeatmapAnnotation(df  = samples_R_NestinInd[,c("group_name","Tumor_origin","Genotype")],
                                  col = list(group_name   = group_name_col,
                                             Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col
                                             )
                                  )

#pdf(paste0(pathToFigures,"complexHeatmap_R_NestinInd_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_R_NestinInd_mvgenes100_R_NestinInd,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and Nestin independent tumor samples clustering",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_R_NestinInd,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_R_NestinInd_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_R_NestinInd_mvgenes1000_R_NestinInd,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Murine primary tumors and Nestin independent tumorsamples",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_R_NestinInd,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Clustering of Nestin samples (independent tumors only) and mICs samples with selected ATRT signature genes

```{r nestinInd_mICs_and_ATRT_geneSet}

# Subset matrix
my_matrix_unfiltered_R_NestinInd <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_R_NestinInd$Sample_name))]

# Select ATRT signature genes
my_matrix_R_NestinInd_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                          y    = my_matrix_unfiltered_R_NestinInd,
                                          by.x = "murine_symbol",
                                          by.y = "symbol_unfiltered",
                                          all  = FALSE
                                          )
row.names(my_matrix_R_NestinInd_ATRTsign_plus) <- my_matrix_R_NestinInd_ATRTsign_plus$murine_symbol
my_matrix_R_NestinInd_ATRTsign_plus <- my_matrix_R_NestinInd_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_R_NestinInd_ATRTsign_plus      <- my_matrix_R_NestinInd_ATRTsign_plus[order(my_matrix_R_NestinInd_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_R_NestinInd_ATRTsign           <- my_matrix_R_NestinInd_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_R_NestinInd_ATRTsign           <- my_matrix_R_NestinInd_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_R_NestinInd_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_R_NestinInd_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Murine primary tumors and Nestin passage 0 (P0)",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_R_NestinInd,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin samples (all passages) only

### Samples

```{r samples_Nestin}

# Subset sample
samples_Nestin <- filter(.data     = samples_R_Nestin,
                         Genotype == "Nestin" 
                         )
samples_Nestin

```

### Heatmap of Nestin samples (all passages) on 100 and 1000 most variables genes

```{r hc_heatmap_Nestin}

# Subset matrix
my_matrix_Nestin <- my_matrix_annotated[,as.vector(samples_Nestin$Sample_name)]

# Most variable genes
mvgenes100_Nestin  <- genes.selection(my_matrix_Nestin, thres.num = 100)
mvgenes1000_Nestin <- genes.selection(my_matrix_Nestin, thres.num = 1000)

# Subset matrix with mvgenes
my_matrix_Nestin_mvgenes100_Nestin <- my_matrix_Nestin[mvgenes100_Nestin,]
my_matrix_Nestin_mvgenes1000_Nestin <- my_matrix_Nestin[mvgenes1000_Nestin,]

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_Nestin_mvgenes100_Nestin,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin all samples clustering",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_Nestin_mvgenes1000_Nestin,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin all samples clustering",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

## Clustering of Nestin (all passages) with selected ATRT signature genes

```{r nestin_ATRTgenes}

# Subset matrix
my_matrix_unfiltered_Nestin <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_Nestin$Sample_name))]

# Select ATRT signature genes
my_matrix_Nestin_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                        y    = my_matrix_unfiltered_Nestin,
                                        by.x = "murine_symbol",
                                        by.y = "symbol_unfiltered",
                                        all  = FALSE
                                        )
row.names(my_matrix_Nestin_ATRTsign_plus) <- my_matrix_Nestin_ATRTsign_plus$murine_symbol
my_matrix_Nestin_ATRTsign_plus <- my_matrix_Nestin_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_Nestin_ATRTsign_plus      <- my_matrix_Nestin_ATRTsign_plus[order(my_matrix_Nestin_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_Nestin_ATRTsign           <- my_matrix_Nestin_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_Nestin_ATRTsign           <- my_matrix_Nestin_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_Nestin_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

# Top annotation
my_top_annot_Nestin <- HeatmapAnnotation(df  = samples_Nestin[,c("Tumor_origin","Genotype")],
                                  col = list(Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col,
                                             group_name   = group_name_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                 )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_Nestin_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Nestin samples (P0 only) clustering",
              row_title                   = "ATRT genes",
              top_annotation              = my_top_annot_Nestin,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin Passage 0 only

### Samples

```{r samples_NestinP0}

# Subset sample
samples_NestinP0 <- filter(.data          = samples_R_Nestin,
                           Genotype      == "Nestin",
                           graft_passage == "P0"
                           )
samples_NestinP0

```

### Heatmap of Nestin samples (P0 only) on 100 and 1000 most variables genes

```{r heatmap_NestinP0}

# Subset matrix
my_matrix_NestinP0 <- my_matrix_annotated[,as.vector(samples_NestinP0$Sample_name)]

# Most variable genes
mvgenes100_NestinP0  <- genes.selection(my_matrix_NestinP0, thres.num = 100)
mvgenes1000_NestinP0 <- genes.selection(my_matrix_NestinP0, thres.num = 1000)

# Subset matrix with mvgenes
my_matrix_NestinP0_mvgenes100  <- my_matrix_NestinP0[mvgenes100_Nestin,]
my_matrix_NestinP0_mvgenes1000 <- my_matrix_NestinP0[mvgenes1000_Nestin,]

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_P0_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_NestinP0_mvgenes100,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin passage 0 samples clustering",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_Nestin_P0_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_NestinP0_mvgenes1000,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin passage 0 samples clustering",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

## Clustering of Nestin samples (P0 only) with selected ATRT signature genes

```{r nestinP0_ATRTgenes}

# Subset matrix
my_matrix_unfiltered_NestinP0 <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_NestinP0$Sample_name))]

# Select ATRT signature genes
my_matrix_NestinP0_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                          y    = my_matrix_unfiltered_NestinP0,
                                          by.x = "murine_symbol",
                                          by.y = "symbol_unfiltered",
                                          all  = FALSE
                                          )
row.names(my_matrix_NestinP0_ATRTsign_plus) <- my_matrix_NestinP0_ATRTsign_plus$murine_symbol
my_matrix_NestinP0_ATRTsign_plus <- my_matrix_NestinP0_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_NestinP0_ATRTsign_plus      <- my_matrix_NestinP0_ATRTsign_plus[order(my_matrix_NestinP0_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_NestinP0_ATRTsign           <- my_matrix_NestinP0_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_NestinP0_ATRTsign           <- my_matrix_NestinP0_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_NestinP0_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

# Top annotation
my_top_annot_NestinP0 <- HeatmapAnnotation(df  = samples_NestinP0[,c("Tumor_origin","Genotype")],
                                  col = list(Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param
                                 )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_NestinP0_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Nestin samples (P0 only) clustering",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_NestinP0,
              show_heatmap_legend         = FALSE,
              column_dend_height          = unit(40, "mm")
              )
hm + row_annot
#dev.off()

```

## Nestin (Independant passage) only

### Samples

```{r samples_NestinInd}

# Subset sample
samples_NestinInd <- filter(.data           = samples_R_Nestin,
                            Genotype       == "Nestin",
                            graft_passage  == "P0" |
                            (graft_passage == "P1" & Sample_name == "43971-A-NBA4P1")
                            )

samples_NestinInd

```

### Heatmap of Nestin samples (Independent passage only) on 100 and 1000 most variables genes

```{r hc_heatmap_NestinInd}

# Subset matrix
my_matrix_NestinInd <- my_matrix_annotated[,as.vector(samples_NestinInd$Sample_name)]

# Most variable genes
mvgenes100_NestinInd  <- genes.selection(my_matrix_NestinInd, thres.num = 100)
mvgenes1000_NestinInd <- genes.selection(my_matrix_NestinInd, thres.num = 1000)

# Subset matrix with mvgenes
my_matrix_NestinInd_mvgenes100  <- my_matrix_NestinInd[mvgenes100_NestinInd,]
my_matrix_NestinInd_mvgenes1000 <- my_matrix_NestinInd[mvgenes1000_NestinInd,]

#pdf(paste0(pathToFigures,"complexHeatmap_NestinInd_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_NestinInd_mvgenes100,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin independant tumors samples clustering",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_NestinInd_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_NestinInd_mvgenes100,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin independant tumors samples clustering",
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_Nestin,
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40, "mm")
        )
#dev.off()

```

### Clustering of Nestin independant samples with selected ATRT signature genes

```{r nestinInd_ATRTgenes}

# Subset matrix
my_matrix_unfiltered_NestinInd <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_NestinInd$Sample_name))]

# Select ATRT signature genes
my_matrix_NestinInd_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                          y    = my_matrix_unfiltered_NestinInd,
                                          by.x = "murine_symbol",
                                          by.y = "symbol_unfiltered",
                                          all  = FALSE
                                          )
row.names(my_matrix_NestinInd_ATRTsign_plus) <- my_matrix_NestinInd_ATRTsign_plus$murine_symbol
my_matrix_NestinInd_ATRTsign_plus <- my_matrix_NestinInd_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_NestinInd_ATRTsign_plus      <- my_matrix_NestinInd_ATRTsign_plus[order(my_matrix_NestinInd_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_NestinInd_ATRTsign           <- my_matrix_NestinInd_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_NestinInd_ATRTsign           <- my_matrix_NestinInd_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_NestinInd_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

# Top annotation
my_top_annot_NestinInd <- HeatmapAnnotation(df  = samples_NestinInd[,c("Tumor_origin","Genotype")],
                                  col = list(Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                 )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_NestinInd_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Nestin samples (Independent tumors only) clustering",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_NestinInd,
              show_heatmap_legend         = FALSE,
              column_dend_height          = my_column_dend_height,
              column_names_gp             = my_column_names_gp,
              row_title_gp                = my_row_title_gp,
              row_names_gp                = my_row_names_gp
              )
hm + row_annot
#dev.off()

```

## Nestin (all passages) with mIC1, mIC2 and allograft (Passage 0)

### Samples

```{r samples_R_Nestin_allograft_P0}

# Subset allograft P0 samples
samples_AllograftP0 <- filter(.data          = samples_LMNRA,
                              Tumor_origin  == "allograft_flanc",
                              graft_passage == "P0"
)
#samples_AllograftP0$value      <- NA
samples_AllograftP0$group_name <- NA

# Subset sample
samples_R_Nestin_allograftP0 <- rbind.fill(samples_R_Nestin,
                                           samples_AllograftP0
                                           )
samples_R_Nestin_allograftP0

```

### Heatmap of Nestin samples (all passages), primary tumors samples and allograft samples (P0 only) on 100 and 1000 most variables genes

```{r hc_heatmap_R_Nestin_allograft_P0}

# Subset matrix
my_matrix_R_Nestin_allograftP0 <- my_matrix_annotated[,as.vector(samples_R_Nestin_allograftP0$Sample_name)]

# Most variable genes
mvgenes100_R_Nestin_allograftP0  <- genes.selection(my_matrix_R_Nestin_allograftP0, thres.num = 100)
mvgenes1000_R_Nestin_allograftP0 <- genes.selection(my_matrix_R_Nestin_allograftP0, thres.num = 1000)

# Subset matrix with mvgenes
my_matrix_R_Nestin_allograftP0_mvgenes100  <- my_matrix_R_Nestin_allograftP0[mvgenes100_R_Nestin_allograftP0,]
my_matrix_R_Nestin_allograftP0_mvgenes1000 <- my_matrix_R_Nestin_allograftP0[mvgenes1000_R_Nestin_allograftP0,]

# Top annotation    
my_top_annot_R_Nestin_allograftP0 <- HeatmapAnnotation(df  = samples_R_Nestin_allograftP0[,c("Tumor_origin","Genotype","group_name")],
                                                       col = list(group_name   = group_name_col,
                                                                  Tumor_origin = tumor_origin_col,
                                                                  Genotype     = genotype_col
                                                                  ),
                                                       gap = my_gap,
                                                       show_annotation_name = TRUE,
                                                       annotation_name_gp = my_annotation_name_gp,
                                                       annotation_legend_param = my_annotation_legend_param,
                                                       na_col = my_na_col
                                                       )

pdf(paste0(pathToFigures,"complexHeatmap_Nestin_R_allograftP0_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_R_Nestin_allograftP0_mvgenes100,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin (all), primary tumors and allograft (P0) samples clustering",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_R_Nestin_allograftP0,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_Nestin_R_allograftP0_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_R_Nestin_allograftP0_mvgenes1000,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin (all), primary tumors and allograft (P0) samples clustering",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_R_Nestin_allograftP0,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
dev.off()

```

### Clustering of Nestin, primary tumors and allograft samples with selected ATRT signature genes

```{r nestin_R_allograftP0_ATRTgenes}

# Subset matrix
my_matrix_unfiltered_R_Nestin_allograftP0 <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_R_Nestin_allograftP0$Sample_name))]

# Select ATRT signature genes
my_matrix_R_Nestin_allograftP0_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                                      y    = my_matrix_unfiltered_R_Nestin_allograftP0,
                                                      by.x = "murine_symbol",
                                                      by.y = "symbol_unfiltered",
                                                      all  = FALSE
                                                      )
row.names(my_matrix_R_Nestin_allograftP0_ATRTsign_plus) <- my_matrix_R_Nestin_allograftP0_ATRTsign_plus$murine_symbol
my_matrix_R_Nestin_allograftP0_ATRTsign_plus <- my_matrix_R_Nestin_allograftP0_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_R_Nestin_allograftP0_ATRTsign_plus      <- my_matrix_R_Nestin_allograftP0_ATRTsign_plus[order(my_matrix_R_Nestin_allograftP0_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_R_Nestin_allograftP0_ATRTsign           <- my_matrix_R_Nestin_allograftP0_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_R_Nestin_allograftP0_ATRTsign           <- my_matrix_R_Nestin_allograftP0_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_R_Nestin_allograftP0_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_R_Nestin_allograftP0_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Murine primary tumors and Nestin samples (Independent tumors only)",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_R_Nestin_allograftP0,
              show_heatmap_legend         = FALSE,
              column_dend_height          = my_column_dend_height,
              column_names_gp             = my_column_names_gp,
              row_title_gp                = my_row_title_gp,
              row_names_gp                = my_row_names_gp
              )
hm + row_annot
dev.off()

```

## Nestin (all passages) with mIC1, mIC2, medulloblastoma and neroblastoma

### Samples

```{r samples_mICs_MN}

# Subset medulloBlastoma and neuroBlastoma samples
samples_MN <- filter(.data        = samples_LMN,
                     Tumor_type == "MB" |
                     Tumor_type == "NB"
                    )

#samples_MN$value     <- NA
samples_MN$group_name <- "neural"

samples_MN

# Subset sample
samples_Nestin_MN <- rbind.fill(samples_R_Nestin,
                                samples_MN
                                )
samples_Nestin_MN

# Subset medulloBlastoma and neuroBlastoma samples
samples_L <- filter(.data        = samples_LMN,
                    Data_source == "Curie",
                    Tumor_type == "lymphoma"
                    )

samples_L

```
### Heatmap of Nestin samples (all passages), primary tumors samples and allograft samples (P0 only) on 100 and 1000 most variables genes

```{r heatmap_mICs_MN}

# Subset matrix
my_matrix_Nestin_MN <- my_matrix_annotated[,as.vector(samples_Nestin_MN$Sample_name)]

# Most variable genes across Nestin and mICs samples
mvgenes100_Nestin_MN  <- genes.selection(my_matrix_Nestin_MN, thres.num = 100)
mvgenes1000_Nestin_MN <- genes.selection(my_matrix_Nestin_MN, thres.num = 1000)

# matrix with mvgenes
my_matrix_Nestin_MN_mvgenes100 <- my_matrix_Nestin_MN[mvgenes100_Nestin_MN,]
my_matrix_Nestin_MN_mvgenes1000 <- my_matrix_Nestin_MN[mvgenes1000_Nestin_MN,]

# Top annotation    
my_top_annot_Nestin_MN <- HeatmapAnnotation(df  = samples_Nestin_MN[,c("Tumor_type","Tumor_origin","Genotype")],
                                  col = list(Tumor_type   = tumor_type_col,
                                             Tumor_origin = tumor_origin_col,
                                             Genotype     = genotype_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                  )

pdf(paste0(pathToFigures,"complexHeatmap_Nestin_R_allograftP0_mvgenes100.pdf"), width=6, height=18)
Heatmap(matrix                      = my_matrix_Nestin_MN_mvgenes100,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin (all), RT, neuroblastoma, medulloblastoma primary tumors and allograft (P0) samples",
        row_title                   = "100 most variable genes",
        top_annotation              =  my_top_annot_Nestin_MN,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_Nestin_R_allograftP0_mvgenes1000.pdf"), width=8, height=12)
Heatmap(matrix                      = my_matrix_Nestin_MN_mvgenes1000,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = "Nestin (all), RT, neuroblastoma, medulloblastoma primary tumors and allograft (P0) samples",
        row_title                   = "1000 most variable genes",
        top_annotation              =  my_top_annot_Nestin_MN,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
dev.off()

```

### Clustering of Nestin, medulloblastoma and neuroblastoma samples with selected ATRT signature genes

```{r nestin_MN_ATRTgenes}

# Subset matrix
my_matrix_unfiltered_Nestin_MN <- my_matrix_unfiltered_annotated[,c("symbol_unfiltered",as.vector(samples_Nestin_MN$Sample_name))]

# Select ATRT signature genes
my_matrix_Nestin_MN_ATRTsign_plus <- merge(x    = geneSet_ATRT,
                                           y    = my_matrix_unfiltered_Nestin_MN,
                                           by.x = "murine_symbol",
                                           by.y = "symbol_unfiltered",
                                           all  = FALSE
                                           )
row.names(my_matrix_Nestin_MN_ATRTsign_plus) <- my_matrix_Nestin_MN_ATRTsign_plus$murine_symbol
my_matrix_Nestin_MN_ATRTsign_plus <- my_matrix_Nestin_MN_ATRTsign_plus[,-1]

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
my_matrix_Nestin_MN_ATRTsign_plus      <- my_matrix_Nestin_MN_ATRTsign_plus[order(my_matrix_Nestin_MN_ATRTsign_plus$Group, decreasing = TRUE),]
my_matrix_Nestin_MN_ATRTsign           <- my_matrix_Nestin_MN_ATRTsign_plus[,-c(1,2,3,4)]
my_matrix_Nestin_MN_ATRTsign           <- my_matrix_Nestin_MN_ATRTsign[,-1]


# Row annotation
my_df           <- as.data.frame(my_matrix_Nestin_MN_ATRTsign_plus[,c("Group")])
colnames(my_df) <- "Group"
row_annot <- rowAnnotation(df = my_df,
                           col = list(Group = c("TYR"="blue", "SHH"="green", "MYC"="red")
                                      )
                           )

#pdf(paste0(pathToFigures,"complexHeatmap_mICs_ATRTgenes.pdf"), width=8, height=12)
hm <- Heatmap(matrix                      = my_matrix_Nestin_MN_ATRTsign,
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              cluster_columns             = TRUE,
              column_title                = "Nestin samples (Independent tumors only) clustering",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_Nestin_MN,
              show_heatmap_legend         = FALSE,
              column_dend_height          = my_column_dend_height,
              column_names_gp             = my_column_names_gp,
              row_title_gp                = my_row_title_gp,
              row_names_gp                = my_row_names_gp
              )
hm + row_annot
#dev.off()

```


# Analysis of ATRT and MB affymetrix data from Ng et al, Cancer Res. 2015

## Extract samples plan from GEO 

```{r GEOquery_GSE68627_pData, eval=FALSE}

# Load series matrix file
GSE68627_matrix <- getGEO(filename = paste0(pathToData,
                                            "Ng2015_affy/GSE68627_series_matrix.txt"
                                            )
                          )

# Extract tumor subgroup and subtype
GSE68627_pData <- pData(GSE68627_matrix)[c("characteristics_ch1",  # Tumor_type
                                           "title",                  # Sample_name
                                           "geo_accession"          # Sample_ID
                                           )]
head(GSE68627_pData)

# Rename colnames
colnames(GSE68627_pData) <- c("Tumor_type","Sample_name","Sample_ID")

# Remove "subgroup:" and "subtype:"
GSE68627_pData$Tumor_type   <- gsub(".+: ","",GSE68627_pData$Tumor_type)

head(GSE68627_pData)
nrow(GSE68627_pData)

# Export annotation (to fill data annotation file)
write.table(x         = GSE68627_pData,
            file      = paste0(pathToData,"Ng2015_affy/GSE68627_pData.csv"),
            quote     = FALSE,
            col.names = TRUE,
            row.names = FALSE,
            sep       = ';'
            )

```



















# TODO: DATA INTEGRATION: TO MOVE TO OTHER FILE

# Cross-species data integration

## murine and human RT affymetrix data integration

### Integration by Quantile normalization

#### All murine samples matrix (affymetrix)

```{r murine_affy}

# Samples
sampleTable_murine <- sampleTable_select
# Add murine subgroup name to sample table murine
sampleTable_murine <- merge(x     = sampleTable_murine,
                            y     = samples_R[, c("Sample_name","group_name")],
                            by    = "Sample_name",
                            all.x = TRUE
                            )

sampleTable_murine$group_name   <- sampleTable_murine$group_name.y
sampleTable_murine$group_name.x <- NULL
sampleTable_murine$group_name.y <- NULL
sampleTable_murine[, c("Genotype","Sample_name","group_name")]

# Murine matrix
my_matrix_murine <- my_matrix_unfiltered_annotated
head(my_matrix_murine[,1:4])
nrow(my_matrix_murine[,1:4])

# Remove duplicated and NA murine gene symbol
my_matrix_murine <- my_matrix_murine[!duplicated(my_matrix_murine[,1]),]
my_matrix_murine <- my_matrix_murine[!is.na(my_matrix_murine[,1]),]
head(my_matrix_murine[,1:4])
nrow(my_matrix_murine)

# Convert gene symbol to rownames
rownames(my_matrix_murine) <- my_matrix_murine[,1]

```

#### Murine RT samples matrix

```{r murine_RT_samples}

# Selecting murine RT RNAseq data 
sampleTable_murine_RT <- filter(.data                = sampleTable_murine,
                                Biological_category == "transcriptomics",
                                Tumor_type          == "RT",
                                include             == "YES"
                                )
nrow(sampleTable_murine_RT)

sampleTable_murine_RT[, c("Tumor_origin","Genotype","Sample_name","group_name")]

# Subset matrix
my_matrix_murine_RT <- my_matrix_murine[,as.vector(sampleTable_murine_RT$Sample_name)]
head(my_matrix_murine_RT[,1:4])

```

#### All murine independent tumors samples (New analysis)

```{r murine_affy_ind}

samples_murine_affy <- rbind(samples_R,
                             samples_normal,
                             samples_Ng2015,
                             samples_NestinInd,
                             samples_AllograftP0,
                             samples_MN,
                             samples_L
                             )
  
# Murine matrix
my_matrix_murine <- my_matrix_unfiltered_annotated
head(my_matrix_murine[,1:4])
nrow(my_matrix_murine[,1:4])

my_matrix_murine_filtered <- my_matrix_annotated
head(my_matrix_murine_filtered[,1:4])
nrow(my_matrix_murine_filtered[,1:4])

# Remove duplicated and NA murine gene symbol
my_matrix_murine <- my_matrix_murine[!duplicated(my_matrix_murine[,1]),]
my_matrix_murine <- my_matrix_murine[!is.na(my_matrix_murine[,1]),]
head(my_matrix_murine[,1:4])
nrow(my_matrix_murine)

my_matrix_murine_filtered <- my_matrix_murine_filtered[!duplicated(my_matrix_murine_filtered[,1]),]
my_matrix_murine_filtered <- my_matrix_murine_filtered[!is.na(my_matrix_murine_filtered[,1]),]
head(my_matrix_murine_filtered[,1:4])
nrow(my_matrix_murine_filtered)

# Convert gene symbol to rownames
rownames(my_matrix_murine) <- my_matrix_murine[,1]

rownames(my_matrix_murine_filtered) <- my_matrix_murine_filtered[,1]

# Subset matrix
my_matrix_murine_affy <- my_matrix_murine_filtered[,as.vector(samples_murine_affy$Sample_name)]
head(my_matrix_murine_affy[,1:4])

# Most variable genes
mvgenes_murine_affy <- genes.selection(my_matrix_murine_affy, thres.num=1000)

my_matrix_murine_affy_mvgenes <- my_matrix_murine_affy[mvgenes_murine_affy,]

stopifnot(require(Rtsne))

# Plot density
plot(density(as.matrix(my_matrix_murine_affy)))

```

### Scaling step of murine affy (New)

```{r scaling_murine}

# Scale
my_matrix_murine_affy <- scale(x      = my_matrix_murine_affy,
                                      center = TRUE,
                                      scale  = TRUE
                                      )
# Plot density of scaled matrix
plot(density(as.matrix(my_matrix_murine_affy)))

```

### t-SNE analysis for murine affy independant tumor samples (New)

```{r tSNE_murine_affy}

set.seed(seed = 9)

# Comput t-SNE
tsne_model_murine_affy <- Rtsne(X                = t(my_matrix_murine_affy_mvgenes),
                                dims             = 2,
                                perplexity       = 10,
                                theta            = 0.0,
                                pca              = TRUE,
                                check_duplicates = FALSE,
                                is_distance      = FALSE
                                )

# Get t-SNE matrix
tsne_model_murine_affy <- as.data.frame(tsne_model_murine_affy$Y)

# Bind sample plan
tsne_model_murine_affy_plus <- cbind(samples_murine_affy, tsne_model_murine_affy)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_all_murine_independent_tumors_affy.pdf"), width=18, height=12)
ggplot(data = tsne_model_murine_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = Tumor_type), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(genotype_col[c("Nestin","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn","wild_type")],"white")) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

```

### Heatmap of murine affy 

```{r complexHeatmap_murine_affy}

# Top annotation    
my_top_annot_murine_affy <- HeatmapAnnotation(df  = samples_murine_affy[,c("Tumor_type","Tumor_origin","Author_Researcher","Genotype","group_name")],
                                              col = list(Tumor_type        = tumor_type_col,
                                                         Tumor_origin      = tumor_origin_col,
                                                         Author_Researcher = author_researcher_col,
                                                         Genotype          = genotype_col,
                                                         group_name        = group_name_col
                                                         ),
                                              gap = my_gap,
                                              show_annotation_name = TRUE,
                                              annotation_name_gp = my_annotation_name_gp,
                                              annotation_legend_param = my_annotation_legend_param,
                                              na_col = my_na_col
                                              )

#pdf(paste0(pathToFigures,"complexHeatmap_all_murine_independent_tumors_affy.pdf"), width=12, height=16)
Heatmap(matrix                      = my_matrix_murine_affy_mvgenes,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        cluster_columns             = TRUE,
        column_title                = paste0("all murine independent tumors (affy) clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_murine_affy,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```


#### Import human affymetrix data

```{r import_human_affy}

# Import human RT rlog transformed matrix (see human_ATRT_affy.Rmd script)
load(file = paste0(pathToANALYSIS_PROJECTS,"/human_ATRT/svn/analyse/script/my_matrix_human.RData"))
head(my_matrix_human[,1:4])

# Import human sample table with DKFZ group_name (see human_ATRT_affy.Rmd script)
load(file = paste0(pathToANALYSIS_PROJECTS,"/human_ATRT/svn/analyse/script/sampleTable_human.RData"))
sampleTable_human[,c("Tumor_type","Tumor_sub_type","Sample_name","group_name")]

```

#### Import human affymetrix data (New)

```{r import_human_affy}

# Load human RT (Curie + DKFZ) and normal (Somel et al) affymetrix data
load(file = paste0(pathToANALYSIS_PROJECTS,"/human_ATRT/svn/analyse/script/human_affy_list.RData"))
sample_human_affy <- human_affy_list$sampleTable
matrix_human_affy <- human_affy_list$matrix

# Check correspondance of matrix colnames and sample name in sample table
sample_human_affy$Sample_name == colnames(matrix_human_affy)

```

#### Convert human gene symbol to mouse gene symbol

```{r gene_symbol_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             filters    = "hgnc_symbol",
                             values     = my_matrix_human$symbol_unfiltered,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )
head(symbols_conversion)

# Convert human gene symbol to orthologous mouse gene symbol
my_matrix_human_mouseSymbol <- merge(x    = symbols_conversion,
                                     y    = my_matrix_human,
                                     by.x = "HGNC.symbol",
                                     by.y = "symbol_unfiltered",
                                     all  = FALSE
                                     ) %>% dplyr::select(-HGNC.symbol, -Row.names)
head(my_matrix_human_mouseSymbol[,1:4])
nrow(my_matrix_human_mouseSymbol)

# Remove duplicated murine genes
my_matrix_human_mouseSymbol <- my_matrix_human_mouseSymbol[!duplicated(my_matrix_human_mouseSymbol[,1]),]
head(my_matrix_human_mouseSymbol[,1:4])
nrow(my_matrix_human_mouseSymbol)

# Put murine gene symbol to rownames
rownames(my_matrix_human_mouseSymbol) <- my_matrix_human_mouseSymbol[,1]
my_matrix_human_mouseSymbol <- my_matrix_human_mouseSymbol[,-1]
head(my_matrix_human_mouseSymbol[,1:4])

```

#### Convert human gene symbol to mouse gene symbol (New)

```{r gene_symbol_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             filters    = "hgnc_symbol",
                             values     = matrix_human_affy$symbol_unfiltered,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )
head(symbols_conversion)

head(matrix_human_affy[,1:4])

# Convert human gene symbol to orthologous mouse gene symbol
matrix_human_affy_mouseSymbol <- merge(x    = symbols_conversion,
                                       y    = matrix_human_affy,
                                       by.x = "HGNC.symbol",
                                       by.y = 0,
                                       all  = FALSE
                                       ) %>% dplyr::select(-HGNC.symbol)
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

# Remove duplicated murine genes
matrix_human_affy_mouseSymbol <- matrix_human_affy_mouseSymbol[!duplicated(matrix_human_affy_mouseSymbol[,1]),]
head(matrix_human_affy_mouseSymbol[,1:4])

# Put murine gene symbol to rownames
rownames(matrix_human_affy_mouseSymbol) <- matrix_human_affy_mouseSymbol[,1]
matrix_human_affy_mouseSymbol <- matrix_human_affy_mouseSymbol[,-1]
head(matrix_human_affy_mouseSymbol[,1:4])

# Plot density of human matrix
plot(density(as.matrix(matrix_human_affy_mouseSymbol)))

```

### Scaling step of human affy (New)

```{r scaling_human}

# Scale human matrix
matrix_human_affy_mouseSymbol <- scale(x      = matrix_human_affy_mouseSymbol,
                                       center = TRUE,
                                       scale  = TRUE
                                       )

# Plot density of human scaled matrix
plot(density(as.matrix(matrix_human_affy_mouseSymbol)))

```

#### Human RT samples

```{r human_RT_samples}

# Selecting human MRT RNAseq data 
sampleTable_human_RT <- filter(.data                = sampleTable_human,
                               Biological_category == "transcriptomics",
                               Technical_plateform == "Affymetrix",
                               Organism            == "Homo_sapiens",
                               Tumor_type          == "RT",
                               include             == "YES"
                               )
nrow(sampleTable_human_RT)

sampleTable_human_RT[, c("Tumor_type","Tumor_sub_type","Localization_of_primary_tumor","Tumor_origin","Sample_name","group_name")]

# Subset matrix
my_matrix_human_RT <- my_matrix_human_mouseSymbol[,as.vector(sampleTable_human_RT$Sample_name)]

names(my_matrix_human_mouseSymbol)

sampleTable_human_RT$Sample_name

head(my_matrix_human_RT[,1:4])

```

#### Human RT hierarchical clustering and heatmap

```{r human_clustering_heatmap}

# Most variable genes
mvgenes_human_RT <- genes.selection(my_matrix_human_RT, thres.num=1000)

# Subset matrix
my_matrix_human_RT_mvgene <- my_matrix_human_RT[mvgenes_human_RT,]

# Top annotation
my_top_annot_human_RT <- HeatmapAnnotation(df = sampleTable_human_RT[,c("Tumor_sub_type","group_name","Localization_of_primary_tumor")], 
                                           col = list(Tumor_sub_type = tumor_sub_type_col,
                                                      group_name = group_name_col,
                                                      Localization_of_primary_tumor = localization_of_primary_tumor_col
                                                      ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )


# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_humanRT_affy_mvgenes.pdf"), width=12, height=12)
Heatmap(matrix                      = my_matrix_human_RT_mvgene,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Unsupervised clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_human_RT,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

#### Human RT hierarchical clustering and heatmap before integration (New)

```{r human_clustering_heatmap}

# Most variable genes
mvgenes_human_affy_mouseSymbol <- genes.selection(matrix_human_affy_mouseSymbol,
                                                  thres.num = 1000
                                                  )

# Remove paralog genes
paralogGenes <- c(mvgenes_human_affy_mouseSymbol[c(grep(pattern = "Gm*", x = mvgenes_human_affy_mouseSymbol),
                                 grep(pattern = "Prame*", x = mvgenes_human_affy_mouseSymbol),
                                 grep(pattern = "Oog*", x = mvgenes_human_affy_mouseSymbol)
                                 )
                               ], "C87414","B020004J07Rik","C130073F10Rik","D5Ertd577e","BC061212","A430089|19Rik","AA792892","B020004J07Rik","E330014E10Rik","C879977","C87414"
)

paralogGenes

mvgenes_human_affy_mouseSymbol <- setdiff(x = mvgenes_human_affy_mouseSymbol,
                                                 y = paralogGenes
                                                 )

length(mvgenes_human_affy_mouseSymbol)

# Subset matrix
matrix_human_affy_mouseSymbol_mvgenes <- matrix_human_affy_mouseSymbol[rownames(matrix_human_affy_mouseSymbol) %in% mvgenes_human_affy_mouseSymbol,]

sample_human_affy$Sample_name == colnames(matrix_human_affy_mouseSymbol_mvgenes)

# Top annotation
my_top_annot_human_affy_mouseSymbol <- HeatmapAnnotation(df = data.frame(sample_human_affy[,c("Tumor_sub_type","group_name","Localization_of_primary_tumor","Data_source")]), 
                                           col = list(Tumor_sub_type = tumor_sub_type_col,
                                                      group_name = group_name_col,
                                                      Localization_of_primary_tumor = localization_of_primary_tumor_col,
                                                      Data_source = data_source_col
                                                      ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_human_affy_RT_normal.pdf"), width=20, height=16)
Heatmap(matrix                      = matrix_human_affy_mouseSymbol_mvgenes,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        show_column_names           = FALSE,  
        column_title                = paste0("Human affy data (RT + normal brain) \n Unsupervised clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "100 most variable genes",
        top_annotation              = my_top_annot_human_affy_mouseSymbol,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = gpar(fontsize = 10)#my_row_names_gp
        )
#dev.off()

```

### t-SNE analysis for human affy before integration with murine affy

```{r tSNE_human_affy}

set.seed(seed = 9)

# Comput t-SNE
tsne_model_human_affy <- Rtsne(X                = t(my_matrix_human_RT_mvgene),
                               dims             = 2,
                               perplexity       = 10,
                               theta            = 0.0,
                               pca              = TRUE,
                               check_duplicates = FALSE,
                               is_distance      = FALSE
                               )

# Get t-SNE matrix
tsne_model_human_affy <- as.data.frame(tsne_model_human_affy$Y)

# Bind sample plan
tsne_model_human_affy_plus <- cbind(sampleTable_human_RT, tsne_model_human_affy)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_human_affy.pdf"), width=12, height=12)
ggplot(data = tsne_model_human_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(color = Data_source, fill = group_name), size = 6, pch=21) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(group_name_col[c("MYC","SHH","TYR")],"white")) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40))
#dev.off()

```

### t-SNE analysis for human affy before integration with murine affy (New)

```{r tSNE_human_affy}

set.seed(seed = 9)

# Comput t-SNE
tsne_model_human_affy <- Rtsne(X                = t(matrix_human_affy_mouseSymbol_scaled_mvgenes),
                               dims             = 2,
                               perplexity       = 10,
                               theta            = 0.0,
                               pca              = TRUE,
                               check_duplicates = FALSE,
                               is_distance      = FALSE
                               )

# Get t-SNE matrix
tsne_model_human_affy <- as.data.frame(tsne_model_human_affy$Y)

# Bind sample plan
tsne_model_human_affy_plus <- cbind(sample_human_affy, tsne_model_human_affy)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_human_affy_RT_normal.pdf"), width=18, height=12)
ggplot(data = tsne_model_human_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(shape = Data_source, fill = group_name), size = 6) +
    scale_shape_manual(values = c(23,25,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(group_name_col[c("MYC","SHH","TYR")],"white")) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40))
#dev.off()

```


#### Merge human and murine matrix

```{r merge_human_murine}

my_matrix_human_murine <- merge(x   = my_matrix_murine_RT,
                                y   = my_matrix_human_RT,
                                by  = 0,
                                all = FALSE
                                )
rownames(my_matrix_human_murine) <- my_matrix_human_murine[,1]
my_matrix_human_murine           <- my_matrix_human_murine[,-1]
head(my_matrix_human_murine[,1:4])
nrow(my_matrix_human_murine)

```

#### Merge human and murine matrix and sample table(New)

```{r merge_human_murine}

# Bind sample table

colnames(samples_murine_affy)
ncol(samples_murine_affy)

colnames(sample_human_affy)
ncol(sample_human_affy)

colnames(samples_murine_affy) == colnames(sample_human_affy)

sample_human_murine_affy <- rbind(samples_murine_affy,
                                  sample_human_affy
                                  )

nrow(sample_human_murine_affy)


# Bind matrix

head(my_matrix_murine_affy_scaled[,1:4])
head(matrix_human_affy_mouseSymbol_scaled[,1:4])

matrix_human_murine_affy           <- merge(x   = my_matrix_murine_affy,
                                            y   = matrix_human_affy_mouseSymbol,
                                            by  = 0,
                                            all = FALSE
                                            )
rownames(matrix_human_murine_affy) <- matrix_human_murine_affy[,1]
matrix_human_murine_affy           <- matrix_human_murine_affy[,-1]

head(matrix_human_murine_affy[,1:4])
nrow(matrix_human_murine_affy)

plot(density(as.matrix(matrix_human_murine_affy)))

```

## Quantile normalization step

```{r quantile_normalization_step}

quantile_normalisation <- function(df){
    df_rank <- apply(df,2,rank,ties.method="min")
    df_sorted <- data.frame(apply(df, 2, sort))
    df_mean <- apply(df_sorted, 1, mean)

    index_to_mean <- function(my_index, my_mean){
        return(my_mean[my_index])
    }

    df_final <- apply(df_rank, 2, index_to_mean, my_mean=df_mean)
    rownames(df_final) <- rownames(df)
    return(df_final)
}

matrix_human_murine_affy <- quantile_normalisation(matrix_human_murine_affy)

```

#### Bind murine and human RT samples table

```{r bind_murine_human_sample_table}

sampleTable_murine_human_RT <- rbind(sampleTable_murine_RT,
                                     sampleTable_human_RT
                                     )

sampleTable_murine_human_RT[, c("Date","Tumor_type","Tumor_sub_type","Localization_of_primary_tumor","Tumor_origin","Sample_name","group_name")]

```

#### Most variable genes in murine and human combined matrix

```{r mvgenes_murine_human}

# Most variable genes
mvgenes_human_murine <- genes.selection(my_matrix_human_murine, thres.num=1000)

# Subset matrix
my_matrix_human_murine_mvgenes <- my_matrix_human_murine[mvgenes_human_murine,]

```

#### Most variable genes in murine and human combined matrix (New)

```{r mvgenes_murine_human}

# Most variable genes
mvgenes_murine_human_affy <- genes.selection(matrix_human_murine_affy, thres.num=1000)

# Subset matrix
matrix_human_murine_affy_mvgenes <- matrix_human_murine_affy[mvgenes_murine_human_affy,]

```

#### Check if there is batch effect or not between human and murine affymetrix data

```{r pca_human_murine_batch}

# PCA
p <- plot_pca(matrix      = my_matrix_human_murine_mvgenes,
              category    = sampleTable_murine_human_RT$Organism,
              addEllipses = FALSE,
              axes        = c(1,2)
              )
#pdf(paste0(pathToFigures,"pca_human_murine_affy_Organism_before_batchCorr.pdf"), width=12, height=12)
p
#dev.off()

```

#### Remove Organism effect using PCA (New)

```{r pca_human_murine_batch}

PC        <- prcomp(data.frame(t(matrix_human_murine_affy_mvgenes)))
#PC        <- prcomp(data.frame(t(matrix_human_murine_affy)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_human_murine_affy
                  )

# PCA organism PC1 and PC2
#pdf(paste0(pathToFigures,"pca_human_murine_affy_Organism_pc1pc2.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Organism), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# PCA organism PC1 and PC3
#pdf(paste0(pathToFigures,"pca_human_murine_affy_Organism_pc1pc3.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC3)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Organism), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC3 (",eigValPer[3]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# PCA organism PC1 and PC3
#pdf(paste0(pathToFigures,"pca_human_murine_affy_groupName_pc1pc3.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC3)
       ) +
    geom_point(aes(shape = Tumor_type, fill = group_name), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(group_name_col[c("MYC","neural","non_neural","normal","SHH","TYR")],"white")) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC3 (",eigValPer[3]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# PCA genotype PC1 and PC3
#pdf(paste0(pathToFigures,"pca_human_murine_affy_Genotype_pc1pc3.pdf"), width=18, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC3)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Genotype), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(genotype_col[c("Nestin","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn","wild_type")],"white")) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC3 (",eigValPer[3]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")
          ) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# Compute genes contribution to PC1 and PC3 (PC1-PC2+PC3)
PCAvar <- get_pca_var(PC)

PCAvar_contrib <- data.frame(PCAvar$contrib)

PCAvar_contrib$PC1PC3 <- PCAvar_contrib[,1] - PCAvar_contrib[,2] + PCAvar_contrib[,3]

PCAvar_contrib <- PCAvar_contrib[order(PCAvar_contrib$PC1PC3, decreasing = TRUE),]

mcGenes <- rownames(PCAvar_contrib[1:500,])

# Subset matrix
matrix_human_murine_affy_mcGenes <- matrix_human_murine_affy[rownames(matrix_human_murine_affy) %in% mcGenes,]

colnames(matrix_human_murine_affy_mcGenes) == sample_human_murine_affy$Sample_name

# Top annotation
my_top_annot_human_murine_affy <- HeatmapAnnotation(df = data.frame(sample_human_murine_affy[,c("Organism","Tumor_type","group_name","Genotype","Data_source","Author_Researcher")]), 
                                                    col = list(Organism    = organism_col,
                                                               Tumor_type  = tumor_type_col,
                                                               group_name  = group_name_col,
                                                               Genotype    = genotype_col,
                                                               Data_source = data_source_col,
                                                               Author_Researcher = author_researcher_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

#pdf(paste0(pathToFigures,"complexHeatmap_human_murine_affy_batchCorr.pdf"), width=18, height=12)
h <- Heatmap(matrix                      = matrix_human_murine_affy_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine and human affy clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_human_murine_affy,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Comput t-SNE
#set.seed(seed = 9)
tsne_model_human_murine_affy_pcaBatchCorr <- Rtsne(X                = t(matrix_human_murine_affy_mcGenes),
                                                   dims             = 2,
                                                   perplexity       = 10,
                                                   theta            = 0.0,
                                                   pca              = TRUE,
                                                   check_duplicates = FALSE,
                                                   is_distance      = FALSE
                                                   )

# Get t-SNE matrix
tsne_model_human_murine_affy_pcaBatchCorr <- as.data.frame(tsne_model_human_murine_affy_pcaBatchCorr$Y)

# Bind sample plan
tsne_model_human_murine_affy_pcaBatchCorr_plus <- cbind(sample_human_murine_affy, tsne_model_human_murine_affy_pcaBatchCorr)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_human_affy_RT_normal.pdf"), width=18, height=12)
ggplot(data = tsne_model_murine_human_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(shape = Tumor_type, fill = Organism), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()


```






### t-SNE analysis for combined human and murine affy before batch effect correction (New)

```{r tSNE_murine_human_affy_before_batchCorr}

set.seed(seed = 9)

# Comput t-SNE
tsne_model_murine_human_affy <- Rtsne(X                = t(matrix_human_murine_affy_mvgenes),
                                      dims             = 2,
                                      perplexity       = 10,
                                      theta            = 0.0,
                                      pca              = TRUE,
                                      check_duplicates = FALSE,
                                      is_distance      = FALSE
                                      )

# Get t-SNE matrix
tsne_model_murine_human_affy <- as.data.frame(tsne_model_murine_human_affy$Y)

# Bind sample plan
tsne_model_murine_human_affy_plus <- cbind(sample_human_murine_affy, tsne_model_murine_human_affy)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_human_affy_RT_normal.pdf"), width=18, height=12)
ggplot(data = tsne_model_murine_human_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(shape = Tumor_type, fill = Organism), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

```

#### Remove organism batch effect

#### Remove batch effect using surrogate variable

```{r remove_batch_effect}

sampleTable_murine_human_RT$group_name

# Convert mIC into SHH and mI/EC into MYC 
sampleTable_murine_human_RT$group_name_tmp <- ifelse(sampleTable_murine_human_RT$group_name == "mIC", "SHH",
                                              ifelse(sampleTable_murine_human_RT$group_name == "mIC_EC", "MYC",
                                              ifelse(sampleTable_murine_human_RT$group_name == "TYR", "TYR",
                                              ifelse(sampleTable_murine_human_RT$group_name == "MYC", "MYC",
                                              ifelse(sampleTable_murine_human_RT$group_name == "SHH", "SHH",NA
                                                     )
                                              )
                                              )
                                              )
                                              )

sampleTable_murine_human_RT[,c("Sample_name","group_name","group_name_tmp")]

sampleTable_murine_human_RT_noNA <- sampleTable_murine_human_RT[!is.na(sampleTable_murine_human_RT$group_name_tmp),]

nrow(sampleTable_murine_human_RT_noNA)

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(group_name_tmp),
                              data   = sampleTable_murine_human_RT_noNA
                              )

my_null_model <- model.matrix(object = ~1,
                              data   = sampleTable_murine_human_RT_noNA
                              )

my_matrix_human_murine_noNA <- my_matrix_human_murine[,as.vector(sampleTable_murine_human_RT_noNA$Sample_name)]

# Estimate the number of latent factor (surrogate variables)
nb_sv <- num.sv(dat    = as.matrix(my_matrix_human_murine_noNA),
                mod    = my_full_model,
                method = "leek"
                )
nb_sv

my_svobj <- sva(dat = as.matrix(my_matrix_human_murine_noNA),
             mod = my_full_model,
             mod0 = my_null_model,
             n.sv=nb_sv
             )

str(my_svobj)

my_svobj$sv

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

my_matrix_human_murine_noNA$pprob.gam <- my_svobj$pprob.gam

head(my_matrix_human_murine_noNA)

# Filter genes with prior probability less than 0.05
pprob.gam_thr <- 0.5
my_matrix_human_murine_noNA_pprobFilt <- my_matrix_human_murine_noNA[which(my_matrix_human_murine_noNA$pprob.gam < pprob.gam_thr),]

my_matrix_human_murine_noNA_pprobFilt$pprob.gam <- NULL

head(my_matrix_human_murine_noNA_pprobFilt)

nrow(my_matrix_human_murine_noNA_pprobFilt)

my_matrix_human_murine_pprobFilt <- my_matrix_human_murine[rownames(my_matrix_human_murine_noNA_pprobFilt),]

nrow(my_matrix_human_murine_pprobFilt)

```

## Remove organism batch effect using normal samples and sva (New)

```{r remove_organism_batch_effect_affy}

# Subset normal samples
sample_human_murine_affy_normal <- filter(.data = sample_human_murine_affy,
                                           Tumor_type == "normal"
                                           )

sample_human_murine_affy_normal[,c("Organism","Tumor_type")]

# Subset normal matrix
matrix_human_murine_affy_normal <- matrix_human_murine_affy[,as.vector(sample_human_murine_affy_normal$Sample_name)]
head(matrix_human_murine_affy_normal)

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(Organism),
                              data   = droplevels(sample_human_murine_affy_normal)
                              )
my_null_model <- model.matrix(object = ~1,
                              data   = droplevels(sample_human_murine_affy_normal)
                              )

# Estimate the number of latent factor (surrogate varibles)
nb_sv <- num.sv(dat    = as.matrix(matrix_human_murine_affy_normal),
                mod    = my_full_model,
                method = "leek"
                )
nb_sv

my_svobj <- sva(dat  = as.matrix(matrix_human_murine_affy_normal),
                mod  = my_full_model,
                mod0 = my_null_model,
                n.sv = nb_sv
                )

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

# add pprob.gam to total matrix
matrix_human_murine_affy_plus <- matrix_human_murine_affy

class(matrix_human_murine_affy_plus)

head(my_svobj)

#matrix_human_murine_affy_plus$pprob.gam <- my_svobj$pprob.gam
matrix_human_murine_affy_plus$pprob.b <- my_svobj$pprob.b

head(matrix_human_murine_affy_plus)

#pprob.gam_thr <- 0.8
pprob.b_thr <- 0.5

#matrix_human_murine_affy_plus_filt <- matrix_human_murine_affy_plus[which(matrix_human_murine_affy_plus$pprob.gam > pprob.gam_thr),] 
matrix_human_murine_affy_plus_filt <- matrix_human_murine_affy_plus[which(matrix_human_murine_affy_plus$pprob.b < pprob.b_thr),] 
#matrix_human_murine_affy_plus_filt$pprob.gam <- NULL
matrix_human_murine_affy_plus_filt$pprob.b <- NULL

head(matrix_human_murine_affy_plus_filt)
nrow(matrix_human_murine_affy_plus_filt)

# Most variable genes
mvgenes_murine_human_affy_plus_filt <- genes.selection(matrix_human_murine_affy_plus_filt, thres.num=100)

# Subset matrix
matrix_human_murine_affy_plus_filt_mvgenes <- matrix_human_murine_affy_plus_filt[mvgenes_murine_human_affy_plus_filt,]

# Top annotation
my_top_annot_human_murine_affy <- HeatmapAnnotation(df = data.frame(sample_human_murine_affy[,c("Organism","Tumor_type","Data_source","group_name")]), 
                                                    col = list(Organism    = organism_col,
                                                               Tumor_type  = tumor_type_col,
                                                               Data_source = data_source_col,
                                                               group_name  = group_name_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_affy_batchCorr.pdf"), width=18, height=12)
h <- Heatmap(matrix                      = matrix_human_murine_affy_plus_filt,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine and human affy clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_human_murine_affy,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_human_murine_affy_plus_filt_mvgenes),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_human_murine_affy_batchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

# t-SNE
set.seed(seed = 9)

# Comput t-SNE
tsne_model_human_murine_affy <- Rtsne(X                = t(matrix_human_murine_affy_plus_filt),
                                      dims             = 2,
                                      perplexity       = 10,
                                      theta            = 0.0,
                                       pca              = TRUE,
                                      check_duplicates = FALSE,
                                      is_distance      = FALSE
                                      )

# Get t-SNE matrix
tsne_model_human_murine_affy <- as.data.frame(tsne_model_human_murine_affy$Y)

# Bind sample plan
tsne_model_human_murine_affy_plus <- cbind(sample_human_murine_affy, tsne_model_human_murine_affy)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_batchCorr.pdf"), width=18, height=12)
ggplot(data = tsne_model_human_murine_affy_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Organism, shape = Tumor_type), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")])+
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

```









#### Check if there is batch effect correction is successfull

```{r pca_human_murine_after_batchCorr}

# PCA
p <- plot_pca(matrix      = my_matrix_human_murine_pprobFilt,
              category    = sampleTable_murine_human_RT$Organism,
              addEllipses = FALSE,
              axes        = c(1,2)
         )
#pdf(paste0(pathToFigures,"pca_human_murine_affy_Organism_after_batchCorr.pdf"), width=12, height=12)
p
#dev.off()

```

#### Murine and human RT hierarchical clustering and heatmap

```{r murine_human_clustering_heatmap}

# Top annotation
my_top_annot_human_murine_sva <- HeatmapAnnotation(df = sampleTable_murine_human_RT[,c("Organism","Tumor_sub_type","Tumor_origin","group_name")], 
                                  col = list(Organism       = organism_col,
                                             Tumor_sub_type = tumor_sub_type_col,
                                             Tumor_origin   = tumor_origin_col,
                                             group_name       = group_name_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param,
                                  na_col = my_na_col
                                  )


# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_human_affy_sva.pdf"), width=16, height=10)
Heatmap(matrix                      = my_matrix_human_murine_pprobFilt,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_human_murine_sva,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

#### murine RT and human ATRT

##### Samples

```{r murine_human_ATRT}

# Subset sample
sampleTable_murine_human_ATRT <- filter(.data           = sampleTable_murine_human_RT,
                                        Tumor_sub_type == "ATRT" |
                                        Organism       == "Mus_musculus"
                                        )
sampleTable_murine_human_ATRT

# Subset matrix
my_matrix_murine_human_ATRT_sva <- my_matrix_human_murine_pprobFilt[, as.vector(sampleTable_murine_human_ATRT$Sample_name)]

```

##### hierarchical clustering and heatmap or murine (all samples) and human ATRT

```{r Curie_murine_human_RT_heatmap}

# Top annotation
my_top_annot_sva_ATRT <- HeatmapAnnotation(df = sampleTable_murine_human_ATRT[,c("Organism","Tumor_origin","group_name")], 
                                  col = list(Organism       = organism_col,
                                             Tumor_origin   = tumor_origin_col,
                                             group_name = group_name_col
                                             ),
                                  gap = my_gap,
                                  show_annotation_name = TRUE,
                                  annotation_name_gp = my_annotation_name_gp,
                                  annotation_legend_param = my_annotation_legend_param
                                  )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_humanATRT_affy.pdf"), width=16, height=12)
Heatmap(matrix                      = my_matrix_murine_human_ATRT_sva,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_sva_ATRT,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

##### hierarchical clustering and heatmap or murine (primary tumors) and human ATRT

```{r Curie_murine_human_RT_heatmap}

sampleTable_murinePrimaryTumors_human_ATRT <- filter(.data = sampleTable_murine_human_ATRT,
                                                         Tumor_origin == "primary_tumor"
                                                         )

# Subset matrix

my_matrix_sva_murinePrimaryTumors_human_ATRT <- my_matrix_murine_human_ATRT_sva[,as.vector(sampleTable_murinePrimaryTumors_human_ATRT$Sample_name)]

# Top annotation
my_top_annot_murinePrimaryTumors_human_ATRT <- HeatmapAnnotation(df = sampleTable_murinePrimaryTumors_human_ATRT[,c("Organism","group_name")], 
                                                                 col = list(Organism       = organism_col,
                                                                            group_name = group_name_col
                                                                            ),
                                                                 gap = my_gap,
                                                                 show_annotation_name = TRUE,
                                                                 annotation_name_gp = my_annotation_name_gp,
                                                                 annotation_legend_param = my_annotation_legend_param
                                                                 )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murinePrimaryTumors_humanATRT_affy.pdf"), width=16, height=12)
Heatmap(matrix                      = my_matrix_sva_murinePrimaryTumors_human_ATRT,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_murinePrimaryTumors_human_ATRT,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

#### murine RT and Curie human ATRT

##### Samples

```{r Curie_murine_human_ATRT_Curie}

# Subset sample
sampleTable_murine_human_ATRT_Curie <- filter(.data           = sampleTable_murine_human_RT,
                                              Data_source    == "InsermU830",
                                              Tumor_sub_type == "ATRT" |
                                              Organism       == "Mus_musculus"
                                              )
sampleTable_murine_human_ATRT_Curie[,c("Sample_name","Genotype","Tumor_sub_type","group_name")]

# Subset matrix
my_matrix_murine_human_ATRT_Curie_sva <- my_matrix_human_murine_pprobFilt[, as.vector(sampleTable_murine_human_ATRT_Curie$Sample_name)]

```

##### hierarchical clustering and heatmap

```{r Curie_murine_human_RT_heatmap}

# Top annotation
my_top_annot_sva_ATRT_Curie <- HeatmapAnnotation(df = sampleTable_murine_human_ATRT_Curie[,c("Organism","Tumor_origin","group_name")], 
                                                 col = list(Organism       = organism_col,
                                                            Tumor_origin   = tumor_origin_col,
                                                            group_name = group_name_col
                                                            ),
                                                 gap = my_gap,
                                                 show_annotation_name = TRUE,
                                                 annotation_name_gp = my_annotation_name_gp,
                                                 annotation_legend_param = my_annotation_legend_param,
                                                 na_col = my_na_col
                                                 )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_human_affy_Curie.pdf"), width=12, height=12)
Heatmap(matrix                      = my_matrix_murine_human_ATRT_Curie_sva,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_sva_ATRT_Curie,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

#### murine rhabdoid primary tumors and Curie human ATRT

##### Samples

```{r Curie_murine_human_ATRT_Curie}

# Subset sample
sampleTable_murinePT_human_ATRT_Curie <- filter(.data           = sampleTable_murine_human_RT,
                                                Data_source    == "InsermU830",
                                                Tumor_origin == "primary_tumor",
                                                Tumor_sub_type == "ATRT" |
                                                Organism       == "Mus_musculus"
                                                )
sampleTable_murinePT_human_ATRT_Curie[,c("Sample_name","Genotype","Tumor_sub_type","group_name")]

# Subset matrix
my_matrix_murinePT_human_ATRT_Curie_sva <- my_matrix_human_murine_pprobFilt[, as.vector(sampleTable_murinePT_human_ATRT_Curie$Sample_name)]

```

##### hierarchical clustering and heatmap

```{r Curie_murine_human_RT_heatmap}

# Top annotation
my_top_annot__murinePT_human_ATRT_Curie <- HeatmapAnnotation(df = sampleTable_murinePT_human_ATRT_Curie[,c("Organism","group_name")], 
                                                 col = list(Organism       = organism_col,
                                                            group_name = group_name_col
                                                            ),
                                                 gap = my_gap,
                                                 show_annotation_name = TRUE,
                                                 annotation_name_gp = my_annotation_name_gp,
                                                 annotation_legend_param = my_annotation_legend_param
                                                 )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murinePT_human_affy_Curie.pdf"), width=12, height=12)
Heatmap(matrix                      = my_matrix_murinePT_human_ATRT_Curie_sva,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot__murinePT_human_ATRT_Curie,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

#### murine RT (with Nestin independant samples) and Curie human ATRT

##### Samples

```{r Curie_murine_human_ATRT_Curie}

# Subset sample
sampleTable_murineNesInd_human_ATRT_Curie <- filter(.data        = sampleTable_murine_human_RT,
                                                    Data_source == "InsermU830",
                                                    Tumor_sub_type == "ATRT" |
                                                    #Organism == "Mus_musculus",
                                                    (Genotype == "R26" |
                                                     (Genotype == "Nestin" & graft_passage == "P0")
                                                    )
                                                    )
sampleTable_murineNesInd_human_ATRT_Curie[,c("Sample_name","Genotype","Tumor_sub_type","group_name")]

# Subset matrix
my_matrix_murineNesInd_human_ATRT_Curie_sva <- my_matrix_human_murine_pprobFilt[, as.vector(sampleTable_murineNesInd_human_ATRT_Curie$Sample_name)]

```

##### hierarchical clustering and heatmap

```{r Curie_murineNesInd_human_ATRT_heatmap}

# Top annotation
my_top_annot_sva_ATRT_Curie_NesInd <- HeatmapAnnotation(df = sampleTable_murineNesInd_human_ATRT_Curie[,c("Organism","Tumor_origin","group_name")], 
                                                 col = list(Organism       = organism_col,
                                                            Tumor_origin   = tumor_origin_col,
                                                            group_name = group_name_col
                                                            ),
                                                 gap = my_gap,
                                                 show_annotation_name = TRUE,
                                                 annotation_name_gp = my_annotation_name_gp,
                                                 annotation_legend_param = my_annotation_legend_param
                                                 )

# Plot heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_human_affy_ATRTCurie_NesInd.pdf"), width=16, height=12)
Heatmap(matrix                      = my_matrix_murineNesInd_human_ATRT_Curie_sva,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("Human Rhabdoid Tumors \n Affymetrix data (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_sva_ATRT_Curie_NesInd,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = my_column_names_gp,
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```

### Integration using AGDEX

#### Import ExpressionSet objects

```{r import_murine_expressionSet_object}

# Import murine expressionSet object
load("expressionSet_murine.Rdata")

# Preview expression data
head(exprs(expressionSet_murine))
nrow(exprs(expressionSet_murine))

# Preview phenotype data
head(pData(expressionSet_murine))

# See the number in each group
table(pData(expressionSet_murine)$group_name)

```

```{r import_human_expressionSet_object}

# Import human expressionSet object
load("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/script/expressionSet_human.Rdata")

# Preview expression data
head(exprs(expressionSet_human))
nrow(exprs(expressionSet_human))

# Preview phenotype data
head(pData(expressionSet_human))

# See the number in each group
table(pData(expressionSet_human)$group_name)

```

#### Create *dex.set* object

```{r create_dexSet_object}

# Create murine dex.set object
dexSet_murine_mIC_healthy <- make.dex.set.object(Eset.data       = expressionSet_murine,
                                                 comp.var        = "group_name",
                                                 comp.def        = "mIC-healthy",
                                                 gset.collection = NULL
                                                 )
dexSet_murine_mICEC_healthy <- make.dex.set.object(Eset.data       = expressionSet_murine,
                                     comp.var        = "group_name",
                                     comp.def        = "mIC_EC-healthy",
                                     gset.collection = NULL
                                     )
dexSet_murine_Nestin_healthy <- make.dex.set.object(Eset.data       = expressionSet_murine,
                                     comp.var        = "group_name",
                                     comp.def        = "Nestin-healthy",
                                     gset.collection = NULL
                                     )

# Create human dex.set object
dexSet_human_ATRT_healthy <- make.dex.set.object(Eset.data       = expressionSet_human,
                                                 comp.var        = "Tumor_sub_type",
                                                 comp.def        = "ATRT-healthy",
                                                 gset.collection = NULL
                                                 )
dexSet_human_TYR_healthy <- make.dex.set.object(Eset.data       = expressionSet_human,
                                                comp.var        = "group_name",
                                                comp.def        = "TYR-healthy",
                                                gset.collection = NULL
                                                )
dexSet_human_MYC_healthy <- make.dex.set.object(Eset.data       = expressionSet_human,
                                                comp.var        = "group_name",
                                                comp.def        = "MYC-healthy",
                                                gset.collection = NULL
                                                )
dexSet_human_SHH_healthy <- make.dex.set.object(Eset.data       = expressionSet_human,
                                                comp.var        = "group_name",
                                                comp.def        = "SHH-healthy",
                                                gset.collection = NULL
                                                )

```

#### Create map.data object

```{r create_mapData_object}

# Create murine mapData object

human_mart <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse_mart <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

affy_probesID_conversion <- getLDS(attributes = c("affy_hg_u133_plus_2"),
                                 filters    = "affy_hg_u133_plus_2",
                                 values     = rownames(exprs(expressionSet_human)),
                                 mart       = human_mart,
                                 attributesL= "affy_mouse430_2",
                                 martL      = mouse_mart,
                                 uniqueRows = TRUE
                                 )
head(affy_probesID_conversion)
nrow(affy_probesID_conversion)

entrezID_conversion <- getLDS(attributes = c("entrezgene"),
                              filters    = "entrezgene",
                              values     = gsub("_at","",rownames(exprs(expressionSet_murine))),
                              mart       = mouse_mart,
                              attributesL= "entrezgene",
                              martL      = human_mart,
                              uniqueRows = TRUE
                              )

colnames(entrezID_conversion) <- c("murine_entrezgene","human_entrezgene")

head(entrezID_conversion)
nrow(entrezID_conversion)

entrezID_conversion_at <- sapply(entrezID_conversion, paste0, "_at")

entrezID_conversion_at <- as.data.frame(entrezID_conversion_at)

head(entrezID_conversion_at)
nrow(entrezID_conversion_at)

map_murine_human_probe <- list(probe.map = entrezID_conversion_at,
                               murine_probe=1,
                               human_probe=2
                               )

str(map_murine_human_probe)

head(map_murine_human_probe$probe.map)
nrow(map_murine_human_probe$probe.map)

```

#### Perform AGDEX analysis

```{r AGDEX_analysis}

# AGDEX analysis of mIC vs human RT
agdex_murine_mIC_human_ATRT <- agdex(dex.setA = dexSet_murine_mIC_healthy,
                                     dex.setB = dexSet_human_ATRT_healthy,
                                     map.data = map_murine_human_probe,
                                     min.nperms = 5,
                                     max.nperms = 10
                                     )
agdex_murine_mIC_human_TYR <- agdex(dex.setA = dexSet_murine_mIC_healthy,
                                    dex.setB = dexSet_human_TYR_healthy,
                                    map.data = map_murine_human_probe,
                                    min.nperms = 5,
                                    max.nperms = 10
                                    )
agdex_murine_mIC_human_MYC <- agdex(dex.setA = dexSet_murine_mIC_healthy,
                                    dex.setB = dexSet_human_MYC_healthy,
                                    map.data = map_murine_human_probe,
                                    min.nperms = 5,
                                    max.nperms = 10
                                    )
agdex_murine_mIC_human_SHH <- agdex(dex.setA = dexSet_murine_mIC_healthy,
                                    dex.setB = dexSet_human_SHH_healthy,
                                    map.data = map_murine_human_probe,
                                    min.nperms = 5,
                                    max.nperms = 10
                                    )

# AGDEX analysis of mIC_EC vs human RT
agdex_murine_mICEC_human_ATRT <- agdex(dex.setA = dexSet_murine_mICEC_healthy,
                                       dex.setB = dexSet_human_ATRT_healthy,
                                       map.data = map_murine_human_probe,
                                       min.nperms = 5,
                                       max.nperms = 10
                                       )
agdex_murine_mICEC_human_TYR <- agdex(dex.setA = dexSet_murine_mICEC_healthy,
                                      dex.setB = dexSet_human_TYR_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )
agdex_murine_mICEC_human_MYC <- agdex(dex.setA = dexSet_murine_mICEC_healthy,
                                      dex.setB = dexSet_human_MYC_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )
agdex_murine_mICEC_human_SHH <- agdex(dex.setA = dexSet_murine_mICEC_healthy,
                                      dex.setB = dexSet_human_SHH_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )

# AGDEX analysis of Nestin vs human RT
agdex_murine_Nestin_human_ATRT <- agdex(dex.setA = dexSet_murine_Nestin_healthy,
                                       dex.setB = dexSet_human_ATRT_healthy,
                                       map.data = map_murine_human_probe,
                                       min.nperms = 5,
                                       max.nperms = 10
                                       )
agdex_murine_Nestin_human_TYR <- agdex(dex.setA = dexSet_murine_Nestin_healthy,
                                      dex.setB = dexSet_human_TYR_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )
agdex_murine_Nestin_human_MYC <- agdex(dex.setA = dexSet_murine_Nestin_healthy,
                                      dex.setB = dexSet_human_MYC_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )
agdex_murine_Nestin_human_SHH <- agdex(dex.setA = dexSet_murine_Nestin_healthy,
                                      dex.setB = dexSet_human_SHH_healthy,
                                      map.data = map_murine_human_probe,
                                      min.nperms = 5,
                                      max.nperms = 10
                                      )

```

#### Explore AGDEX result

```{r AGDEX_result}

# Group-labels
agdex_murine_mIC_human_ATRT$dex.asgnA
agdex_murine_mIC_human_ATRT$dex.asgnB

agdex_murine_mIC_human_TYR$dex.asgnA
agdex_murine_mIC_human_TYR$dex.asgnB

agdex_murine_mIC_human_MYC$dex.asgnA
agdex_murine_mIC_human_MYC$dex.asgnB

agdex_murine_mIC_human_SHH$dex.asgnA
agdex_murine_mIC_human_SHH$dex.asgnB

agdex_murine_mICEC_human_ATRT$dex.asgnA
agdex_murine_mICEC_human_ATRT$dex.asgnB

agdex_murine_mICEC_human_TYR$dex.asgnA
agdex_murine_mICEC_human_TYR$dex.asgnB

agdex_murine_mICEC_human_MYC$dex.asgnA
agdex_murine_mICEC_human_MYC$dex.asgnB

agdex_murine_mICEC_human_SHH$dex.asgnA
agdex_murine_mICEC_human_SHH$dex.asgnB

agdex_murine_Nestin_human_ATRT$dex.asgnA
agdex_murine_Nestin_human_ATRT$dex.asgnB

agdex_murine_Nestin_human_TYR$dex.asgnA
agdex_murine_Nestin_human_TYR$dex.asgnB

agdex_murine_Nestin_human_MYC$dex.asgnA
agdex_murine_Nestin_human_MYC$dex.asgnB

agdex_murine_Nestin_human_SHH$dex.asgnA
agdex_murine_Nestin_human_SHH$dex.asgnB

# Differential expression results
head(agdex_murine_mIC_human_ATRT$dex.resA)
head(agdex_murine_mIC_human_ATRT$dex.resB)

head(agdex_murine_mIC_human_TYR$dex.resA)
head(agdex_murine_mIC_human_TYR$dex.resB)

head(agdex_murine_mIC_human_MYC$dex.resA)
head(agdex_murine_mIC_human_MYC$dex.resB)

head(agdex_murine_mIC_human_SHH$dex.resA)
head(agdex_murine_mIC_human_SHH$dex.resB)

head(agdex_murine_mICEC_human_ATRT$dex.resA)
head(agdex_murine_mICEC_human_ATRT$dex.resB)

head(agdex_murine_mICEC_human_TYR$dex.resA)
head(agdex_murine_mICEC_human_TYR$dex.resB)

chead(agdex_murine_mICEC_human_MYC$dex.resA)
head(agdex_murine_mICEC_human_MYC$dex.resB)

head(agdex_murine_mICEC_human_SHH$dex.resA)
head(agdex_murine_mICEC_human_SHH$dex.resB)

head(agdex_murine_Nestin_human_ATRT$dex.resA)
head(agdex_murine_Nestin_human_ATRT$dex.resB)

head(agdex_murine_Nestin_human_TYR$dex.resA)
head(agdex_murine_Nestin_human_TYR$dex.resB)

head(agdex_murine_Nestin_human_MYC$dex.resA)
head(agdex_murine_Nestin_human_MYC$dex.resB)

head(agdex_murine_Nestin_human_SHH$dex.resA)
head(agdex_murine_Nestin_human_SHH$dex.resB)

# Meta-analysis results
head(agdex_murine_mIC_human_ATRT$meta.dex.res)
head(agdex_murine_mIC_human_TYR$meta.dex.res)
head(agdex_murine_mIC_human_MYC$meta.dex.res)
head(agdex_murine_mIC_human_SHH$meta.dex.res)

head(agdex_murine_mICEC_human_ATRT$meta.dex.res)
head(agdex_murine_mICEC_human_TYR$meta.dex.res)
head(agdex_murine_mICEC_human_MYC$meta.dex.res)
head(agdex_murine_mICEC_human_SHH$meta.dex.res)

head(agdex_murine_Nestin_human_ATRT$meta.dex.res)
head(agdex_murine_Nestin_human_TYR$meta.dex.res)
head(agdex_murine_Nestin_human_MYC$meta.dex.res)
head(agdex_murine_Nestin_human_SHH$meta.dex.res)

# Scatterplot of the difference-of-means statistics for probe-set pairs
agdex.scatterplot(agdex_murine_mIC_human_ATRT)
agdex.scatterplot(agdex_murine_mIC_human_TYR)
agdex.scatterplot(agdex_murine_mIC_human_MYC)
agdex.scatterplot(agdex_murine_mIC_human_SHH)

agdex.scatterplot(agdex_murine_mICEC_human_ATRT)
agdex.scatterplot(agdex_murine_mICEC_human_TYR)
agdex.scatterplot(agdex_murine_mICEC_human_MYC)
agdex.scatterplot(agdex_murine_mICEC_human_SHH)

agdex.scatterplot(agdex_murine_Nestin_human_ATRT)
agdex.scatterplot(agdex_murine_Nestin_human_TYR)
agdex.scatterplot(agdex_murine_Nestin_human_MYC)
agdex.scatterplot(agdex_murine_Nestin_human_SHH)

# Genome wide analysis result
agdex_murine_mIC_human_ATRT$gwide.agdex.result
agdex_murine_mIC_human_TYR$gwide.agdex.result
agdex_murine_mIC_human_MYC$gwide.agdex.result
agdex_murine_mIC_human_SHH$gwide.agdex.result

agdex_murine_mICEC_human_ATRT$gwide.agdex.result
agdex_murine_mICEC_human_TYR$gwide.agdex.result
agdex_murine_mICEC_human_MYC$gwide.agdex.result
agdex_murine_mICEC_human_SHH$gwide.agdex.result

agdex_murine_Nestin_human_ATRT$gwide.agdex.result
agdex_murine_Nestin_human_TYR$gwide.agdex.result
agdex_murine_Nestin_human_MYC$gwide.agdex.result
agdex_murine_Nestin_human_SHH$gwide.agdex.result

```

#### Create cos matrix

```{r cos_matrix}

# Retrieve cos value
cos_mIC_ATRT <- agdex_murine_mIC_human_ATRT$gwide.agdex.result$stat.value[1]
cos_mIC_TYR <- agdex_murine_mIC_human_TYR$gwide.agdex.result$stat.value[1]
cos_mIC_MYC <- agdex_murine_mIC_human_MYC$gwide.agdex.result$stat.value[1]
cos_mIC_SHH <- agdex_murine_mIC_human_SHH$gwide.agdex.result$stat.value[1]

cos_mICEC_ATRT <- agdex_murine_mICEC_human_ATRT$gwide.agdex.result$stat.value[1]
cos_mICEC_TYR <- agdex_murine_mICEC_human_TYR$gwide.agdex.result$stat.value[1]
cos_mICEC_MYC <- agdex_murine_mICEC_human_MYC$gwide.agdex.result$stat.value[1]
cos_mICEC_SHH <- agdex_murine_mICEC_human_SHH$gwide.agdex.result$stat.value[1]

cos_Nestin_ATRT <- agdex_murine_mIC_human_ATRT$gwide.agdex.result$stat.value[1]
cos_Nestin_TYR <- agdex_murine_Nestin_human_TYR$gwide.agdex.result$stat.value[1]
cos_Nestin_MYC <- agdex_murine_Nestin_human_MYC$gwide.agdex.result$stat.value[1]
cos_Nestin_SHH <- agdex_murine_Nestin_human_SHH$gwide.agdex.result$stat.value[1]

# Create a matrix of cos value
cos_vector <- c(cos_mIC_ATRT,cos_mIC_TYR,cos_mIC_MYC,cos_mIC_SHH,
                cos_mICEC_ATRT,cos_mICEC_TYR,cos_mICEC_MYC,cos_mICEC_SHH,
                cos_Nestin_ATRT,cos_Nestin_TYR,cos_Nestin_MYC,cos_Nestin_SHH
                )

cos_matrix <- matrix(data = cos_vector,
                     nrow = 4,
                     ncol = 3,
                     byrow = FALSE
                     )
colnames(cos_matrix) <- c("mIC","mI/EC","Nestin")
rownames(cos_matrix) <- c("ATRT","TYR","MYC","SHH")
cos_matrix

# Plot heatmap of cos matrix
my_colorRamp <- colorRamp2(breaks = c(0.25,0.75),
                           colors = c("white","blue"),
                           transparency = 0,
                           space = "LAB"
                           )

#pdf(paste0(pathToFigures,"AGDEX_cos_murine_human_affy.pdf"), width=12, height=12)
Heatmap(matrix                      = cos_matrix[,c("mIC","mI/EC")],
        col                         = my_colorRamp,
        cluster_columns             = FALSE,
        cluster_rows                = FALSE,
        show_row_names              = TRUE,
        show_column_names           = TRUE,
        show_heatmap_legend         = TRUE,
        column_title                = "Murine MRT",
        row_title                   = "Human MRT",
        column_names_side           = "top",
        column_names_gp             = gpar(fontsize = 30),
        column_title_gp             = gpar(fontsize = 30),
        row_title_gp                = gpar(fontsize = 30),
        row_names_side              = "left",
        row_names_gp                = gpar(fontsize = 30),
        heatmap_legend_param        = list(title = "cos value",
                                           color_bar = "continuous",
                                           title_gp = gpar(fontsize = 30)
                                           )
        )
#dev.off()

cos_matrix_Nestin <- as.data.frame(cos_matrix[,c("Nestin")])

colnames(cos_matrix_Nestin) <- "Nestin"

#pdf(paste0(pathToFigures,"AGDEX_cos_murineNestin_human_affy.pdf"), width=12, height=12)
Heatmap(matrix                      = cos_matrix_Nestin,#cos_matrix[,c("Nestin")],
        col                         = my_colorRamp,
        cluster_columns             = FALSE,
        cluster_rows                = FALSE,
        show_row_names              = TRUE,
        show_column_names           = TRUE,
        show_heatmap_legend         = TRUE,
        column_title                = "Murine MRT",
        row_title                   = "Human MRT",
        column_names_side           = "top",
        column_names_gp             = gpar(fontsize = 30),
        column_title_gp             = gpar(fontsize = 30),
        row_title_gp                = gpar(fontsize = 30),
        row_names_side              = "left",
        row_names_gp                = gpar(fontsize = 30),
        heatmap_legend_param        = list(title = "cos value",
                                           color_bar = "continuous",
                                           title_gp = gpar(fontsize = 30)
                                           )
        )
#dev.off()

```

## Cross-platform data integration 

### Murine affy and RNAseq data integration

#### Affy samples

```{r murine_affy_sample}

head(my_matrix_R)
nrow(my_matrix_R)
samples_R$group_name
plot(density(as.matrix(my_matrix_R)))

# Rename affy matrix and sample table
samples_R_affy <- samples_R
my_matrix_R_affy <- my_matrix_R

# Scaling normalization
my_matrix_R_affy_scaled <- scale(x = my_matrix_R_affy,
                                 center = TRUE,
                                 scale = TRUE
                                 )

# Plot density after scaling
plot(density(as.matrix(my_matrix_R_affy_scaled)))

```

#### RNAseq samples

```{r murine_RNAseq_sample}

# Load murine RNAseq TPM primary tumor data
load("murine_R_RNAseq_tpm_list.RData")
samples_R_RNAseq <- murine_R_RNAseq_tpm_list$sampleTable
my_matrix_R_RNAseq <- murine_R_RNAseq_tpm_list$matrix

nrow(my_matrix_R_RNAseq)
plot(density(my_matrix_R_RNAseq), xlim=c(0,100))

# Log2p1 transform primary tumor TPM data
my_matrix_R_RNAseq_log2p1 <- log2(my_matrix_R_RNAseq+1)
plot(density(my_matrix_R_RNAseq_log2p1))

# Load murine RNAseq TPM Nestin data
load("murine_N_RNAseq_tpm_list.RData")
samples_N_RNAseq <- murine_N_RNAseq_tpm_list$sampleTable
my_matrix_N_RNAseq <- murine_N_RNAseq_tpm_list$matrix

nrow(my_matrix_N_RNAseq)
plot(density(my_matrix_N_RNAseq))

# Log2p1 transform Nestin TPM data
my_matrix_N_RNAseq_log2p1 <- log2(my_matrix_N_RNAseq+1)
plot(density(my_matrix_N_RNAseq_log2p1))

```


```{r murine_N_RNAseq_filtering}

# Filter log2p1 RNAseq TPM data based on value of sum of log2p1 in all samples 
my_matrix_N_RNAseq_log2p1_sum <- data.frame(my_matrix_N_RNAseq_log2p1)
my_matrix_N_RNAseq_log2p1_sum$sum <- rowSums(my_matrix_N_RNAseq_log2p1_sum, na.rm=FALSE)

# Plot sum density
#pdf(paste0(pathToFigures,"density_murine_RNAseq_sumTPM.pdf"), width=12, height=12)
plot(density(my_matrix_N_RNAseq_log2p1_sum$sum))
abline(v=30, col="red")
#dev.off()

# Filter out rows with sum less than 100
sum_thr <- 30
my_matrix_N_RNAseq_log2p1_filtered <- my_matrix_N_RNAseq_log2p1_sum[my_matrix_N_RNAseq_log2p1_sum$sum >= sum_thr,]
nrow(my_matrix_N_RNAseq_log2p1_filtered)

# Plot sum density after filtering
plot(density(my_matrix_N_RNAseq_log2p1_filtered$sum))

# Remove sum column
my_matrix_N_RNAseq_log2p1_filtered$sum <- NULL

# Plot sum density after filtering
plot(density(as.matrix(head(my_matrix_N_RNAseq_log2p1_filtered))))

# Scaling normalization
my_matrix_N_RNAseq_log2p1_filtered_scaled <- scale(x = my_matrix_N_RNAseq_log2p1_filtered,
                                                   center = TRUE,
                                                   scale = TRUE
                                                   )

# Plot density after scaling
plot(density(as.matrix(my_matrix_N_RNAseq_log2p1_filtered_scaled)))

```

#### Merge RNAseq and affy scaled data

```{r merge_murine_affy_RNAseq_data}

head(my_matrix_R_affy_scaled)
nrow(my_matrix_R_affy_scaled)

# Add gene symbol to affy matrix
head(symbol_unfiltered_frame)

my_matrix_R_affy_scaled_annot <- merge(x=symbol_unfiltered_frame,
                                       y = my_matrix_R_affy_scaled,
                                       by = 0,
                                       all = FALSE
                                       )
rownames(my_matrix_R_affy_scaled_annot) <- my_matrix_R_affy_scaled_annot$symbol_unfiltered
head(my_matrix_R_affy_scaled_annot)
nrow(my_matrix_R_affy_scaled_annot)

# Remove duplicated and NA symbol
my_matrix_R_affy_scaled_annot <- my_matrix_R_affy_scaled_annot[!duplicated(my_matrix_R_affy_scaled_annot$symbol_unfiltered),]
my_matrix_R_affy_scaled_annot <- my_matrix_R_affy_scaled_annot[!is.na(my_matrix_R_affy_scaled_annot$symbol_unfiltered),]
nrow(my_matrix_R_affy_scaled_annot)

rownames(my_matrix_R_affy_scaled_annot) <- my_matrix_R_affy_scaled_annot$symbol_unfiltered
my_matrix_R_affy_scaled_annot$Row.names <- NULL
my_matrix_R_affy_scaled_annot$symbol_unfiltered <- NULL
head(my_matrix_R_affy_scaled_annot)

head(my_matrix_R_RNAseq_log2p1_filtered_scaled)


# Remove X before R matrix colnames (TODO: remove from RNAseq script!)
colnames(my_matrix_R_RNAseq_log2p1_filtered_scaled) <- gsub("X","",colnames(my_matrix_R_RNAseq_log2p1_filtered_scaled))

# Merge RNAseq and affy primary tumor matrix (FOR SVA ANALYSIS)
my_matrix_murine_RNAseq_affy <- merge(x = my_matrix_R_affy_scaled_annot,
                                      y = my_matrix_R_RNAseq_log2p1_filtered_scaled,
                                      by = 0,
                                      all = FALSE
                                      )                                      

rownames(my_matrix_murine_RNAseq_affy) <- my_matrix_murine_RNAseq_affy$Row.names
my_matrix_murine_RNAseq_affy$Row.names <- NULL
head(my_matrix_murine_RNAseq_affy)

# Remove X before N matrix colnames (TODO: remove from RNAseq script!)
colnames(my_matrix_N_RNAseq_log2p1_filtered_scaled) <- gsub("X","",colnames(my_matrix_N_RNAseq_log2p1_filtered_scaled))

# Rename 43940.bisA sample to its original name (43940-bisA)
colnames(my_matrix_N_RNAseq_log2p1_filtered_scaled)[ncol(my_matrix_N_RNAseq_log2p1_filtered_scaled)] <- "43940-bisA"

# Merge RNAseq and affy primary tumor and Nestin RNAseq matrices
my_matrix_murine_R_N_RNAseq_R_affy <- merge(x = my_matrix_murine_RNAseq_affy,
                                            y = my_matrix_N_RNAseq_log2p1_filtered_scaled,
                                            by = 0,
                                            all = FALSE
                                            )                                      

rownames(my_matrix_murine_R_N_RNAseq_R_affy) <- my_matrix_murine_R_N_RNAseq_R_affy$Row.names
my_matrix_murine_R_N_RNAseq_R_affy$Row.names <- NULL
head(my_matrix_murine_R_N_RNAseq_R_affy)

```

#### Merge sample table

```{r merge_R_sample_table}

# Merge RNAseq and affy sample table
samples_R_RNAseq_affy <- rbind(samples_R_RNAseq,
                               samples_R_affy
                               )
samples_R_RNAseq_affy[,c("Data_type","Sample_name","group_name")]

# Reorder to follow matrix colnames order
samples_R_RNAseq_affy <- samples_R_RNAseq_affy[match(colnames(my_matrix_murine_RNAseq_affy), samples_R_RNAseq_affy$Sample_name),]

samples_R_RNAseq_affy[,c("Data_type","Sample_name","group_name")]

```

```{r merge_RN_sample_table}

# Merge RNAseq and affy sample table
samples_R_N_RNAseq_R_affy <- rbind(samples_R_RNAseq_affy,
                                   samples_N_RNAseq
                                   )
samples_R_N_RNAseq_R_affy[,c("Data_type","Sample_name","Genotype","group_name")]

nrow(samples_R_N_RNAseq_R_affy)
ncol(my_matrix_murine_R_N_RNAseq_R_affy)

head(my_matrix_murine_R_N_RNAseq_R_affy)

# Reorder to follow matrix colnames order
samples_R_N_RNAseq_R_affy <- samples_R_N_RNAseq_R_affy[match(colnames(my_matrix_murine_R_N_RNAseq_R_affy), samples_R_N_RNAseq_R_affy$Sample_name),]

samples_R_N_RNAseq_R_affy[,c("Data_type","Sample_name","Genotype","group_name")]

```

#### Check platform batch effect

```{r check_batch_effect}

# Most variable genes
murine_RNAseq_affy_mvgenes <- genes.selection(data = my_matrix_murine_RNAseq_affy,
                                              thres.num = 1000
                                              )

# Subset matrix
my_matrix_murine_RNAseq_affy_mvgenes <- my_matrix_murine_RNAseq_affy[murine_RNAseq_affy_mvgenes,]

# PCA before batch correction
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq.pdf"), width=12, height=12)
plot_pca(matrix      = my_matrix_murine_RNAseq_affy_mvgenes,
         category    = as.vector(samples_R_RNAseq_affy$Data_type),
         plotInertia = FALSE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )
#dev.off()

```

#### Remove batch effect using surrogate variable

```{r remove_batch_effect}

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(group_name),
                              data = samples_R_RNAseq_affy
                              )

my_null_model <- model.matrix(object = ~1,
                              data = samples_R_RNAseq_affy
                              )

# Estimate the number of latent factor (surrogate varibles)
nb_sv <- num.sv(dat = as.matrix(my_matrix_murine_RNAseq_affy),
                mod = my_full_model,
                method = "leek"
                )
nb_sv

my_svobj <- sva(dat = as.matrix(my_matrix_murine_RNAseq_affy),
             mod = my_full_model,
             mod0 = my_null_model,
             n.sv=nb_sv
             )

str(my_svobj)

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

my_matrix_murine_RNAseq_affy_tmp <- my_matrix_murine_RNAseq_affy
my_matrix_murine_RNAseq_affy_tmp$pprob.gam <- my_svobj$pprob.gam

head(my_matrix_murine_RNAseq_affy_tmp)

pprob.gam_thr <- 0.5
my_matrix_murine_RNAseq_affy_pprobFilt <- my_matrix_murine_RNAseq_affy_tmp[which(my_matrix_murine_RNAseq_affy_tmp$pprob.gam < pprob.gam_thr),]
my_matrix_murine_RNAseq_affy_pprobFilt$pprob.gam <- NULL

head(my_matrix_murine_RNAseq_affy_pprobFilt)
nrow(my_matrix_murine_RNAseq_affy_pprobFilt)

# PCA after batch correction
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_batchCorr.pdf"), width=12, height=12)
plot_pca(matrix      = my_matrix_murine_RNAseq_affy_pprobFilt,
         category    = as.vector(samples_R_RNAseq_affy$Data_type),
         plotInertia = FALSE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )
#dev.off()

```

#### Complex heatmap human ATRT and murine

```{r complesHeatmap_RNAseq_affy_ATRT}

# Most variable genes
murine_RNAseq_affy_pprobFilt_mvgenes <- genes.selection(data = my_matrix_murine_RNAseq_affy_pprobFilt,
                                                        thres.num = 1000
                                                        )

# Subset matrix
my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes <- my_matrix_murine_RNAseq_affy_pprobFilt[murine_RNAseq_affy_pprobFilt_mvgenes,]

samples_R_RNAseq_affy$Sample_name

colnames(my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes)

# Top Annotation
my_top_annot_murine_affy_RNAseq <- HeatmapAnnotation(df                      = data.frame(samples_R_RNAseq_affy[,c("Tamoxifen_date","Localization_of_primary_tumor")]),
                                                     col                     = list(Localization_of_primary_tumor = localization_of_primary_tumor_col,
                                                                                    Tamoxifen_date = tamoxifen_date_col
                                                     ),
                                                     gap                     = my_gap,
                                                     show_annotation_name    = TRUE,
                                                     annotation_name_gp      = my_annotation_name_gp,
                                                     annotation_legend_param = my_annotation_legend_param,
                                                     na_col = my_na_col,
                                                     height = unit(2,"cm")
                                                     )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq.pdf"), width=16, height=18)
#dev.new(height=16,width=8)
Heatmap(matrix                      = my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine affymetrix and RNAseq primary tumors \n unsupervised clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        column_title_gp             = gpar(fontsize = 20),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_murine_affy_RNAseq,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 20),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = gpar(fontsize = 20)
        )
#dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes)))

h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes)),
             cluster_columns    = colDend,
             top_annotation     = my_top_annot_murine_affy_RNAseq_bis,
             column_dend_height = my_column_dend_height,
             show_row_names     = TRUE,
             column_names_gp    = gpar(fontsize = 14)
             ) 

draw(h, annotation_legend_side = "bottom")

h <- Heatmap(matrix                 = my_matrix_murine_RNAseq_affy_pprobFilt_mvgenes,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = my_meth,
             rect_gp                     = gpar(type = "none"),
             cluster_rows                = FALSE,
             show_row_names              = FALSE,
             column_title                = "Unsupervised clustering of murine tumors",
             column_title_gp             = gpar(fontsize = 20),
                                        #row_title                   = "1000 most variable genes",
             top_annotation              = my_top_annot_murine_affy_RNAseq,
             show_heatmap_legend         = FALSE,
             column_names_side           = "top",
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = gpar(fontsize = 20)
             )
# Add lengend on top annotation
for(legend in c("Data_type","group_name","Localization_of_primary_tumor","sub_localization_of_primary_tumor","Tamoxifen_date"))
{
    decorate_annotation(legend, {
    # Annotation names on the right
        grid.text(legend, unit(1, "npc") + unit(10, "mm"), 0.5, default.units = "npc", just = "left")
    })
}

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_withoutHeatmap.pdf"), width=18, height=8)
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

## Overall integration: Murine, human, affy, RNAseq data integration

### Z-scale method

```{r cross_species_and_cross_technics_data_integration_by_Zscaling}

# Load RNAseq (murine + human) batch corrected and scaled data
load("my_matrix_human_murine_RNAseq_batchCorr_filt_scaled.RData")

head(my_matrix_human_murine_RNAseq_batchCorr_filt_scaled)

plot(density(my_matrix_human_murine_RNAseq_batchCorr_filt_scaled))

# Scale affymetrix (murine + human) batch corrected data
head(my_matrix_human_murine_batchCorr)

my_matrix_human_murine_affy_batchCorr <- my_matrix_human_murine_batchCorr

plot(density(as.matrix(my_matrix_human_murine_affy_batchCorr)))

my_matrix_human_murine_affy_batchCorr_scaled <- scale(x = my_matrix_human_murine_affy_batchCorr,
                                                      center = TRUE,
                                                      scale = TRUE
                                                      )

plot(density(as.matrix(my_matrix_human_murine_affy_batchCorr_scaled)))

# Merge RNAseq and affymetrix scaled data
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled <- merge(x = my_matrix_human_murine_RNAseq_batchCorr_filt_scaled,
                                                             y = my_matrix_human_murine_affy_batchCorr_scaled,
                                                             by = 0,
                                                             all = FALSE
                                                             )

head(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled[,1:6])

rownames(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled) <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled$Row.names
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled[,-1]
nrow(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled)

head(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled[,1:6])

plot(density(as.matrix(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled)))

# Sample table

samples_affy_RNAseq <- filter(.data = sampleTable,
                              Sample_name %in% colnames(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled),
                              Biological_category == "transcriptomics",
                              include == "YES",
                              Data_type == "RNA_microarray" |
                              Data_type == "RNA_seq"
                              )

nrow(samples_affy_RNAseq)

ncol(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled)

# Reorder sample name in sample table
samples_affy_RNAseq <- samples_affy_RNAseq[match(colnames(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled), samples_affy_RNAseq$Sample_name),]

head(samples_affy_RNAseq$Sample_name)

is.element(el = samples_affy_RNAseq$Sample_name,
           set = colnames(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled)
           )

head(my_matrix_human_murine_affy_RNAseq_batchCorr_scaled[,1:4])

# Most variable genes
mvgenes_affy_RNAseq <- genes.selection(data = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled,
                                       thres.num = 1000
                                       )

# Subset matrix
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_mvgenes <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled[mvgenes_affy_RNAseq,]

data_type_col <- c(RNA_microarray = "deeppink",
                      RNA_seq = "green"
                      )

# PCA before batch correction
plot_pca(matrix      = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_mvgenes,
         category    = as.vector(samples_affy_RNAseq$Data_type),
         plotInertia = TRUE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )

# Remove data type batch effect
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr <- ComBat(dat   = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled,
                                           batch = samples_affy_RNAseq$Organism
                                           )
samples_affy_RNAseq$Data_type

# Most variable genes
mvgenes_affy_RNAseq_batchCorr <- genes.selection(data = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr,
                                       thres.num = 1000
                                       )

# Subset matrix
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_mvgenes <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr[mvgenes_affy_RNAseq_batchCorr,]

data_type_col <- c(RNA_microarray = "deeppink",
                      RNA_seq = "green"
                      )

# PCA after batch effect correction
plot_pca(matrix      = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_mvgenes,
         category    = as.vector(samples_affy_RNAseq$Data_type),
         plotInertia = TRUE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )

# Top Annotation
my_top_annot_affy_RNAseq <- HeatmapAnnotation(df                      = samples_affy_RNAseq[,c("Tumor_type","Tumor_sub_type","Organism","Data_type")],
                                              col                     = list(Tumor_type = tumor_type_col,
                                                                             Data_type = data_type_col,
                                                                             Organism = organism_col,
                                                                             Tumor_sub_type = tumor_sub_type_col
                                                                             ),
                                              gap                     = my_gap,
                                              show_annotation_name    = TRUE,
                                              annotation_name_gp      = my_annotation_name_gp,
                                              annotation_legend_param = my_annotation_legend_param
                                              )

#pdf(paste0(pathToFigures,"complexHeatmap_affy_RNAseq_mvgenes.pdf"), width=12, height=12)
Heatmap(matrix                      = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_mvgenes,
        clustering_distance_columns = "spearman",
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine tumors clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_affy_RNAseq,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 6),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = gpar(fontsize = 12)
        )
#dev.off()

```

#### Complex heatmap human ATRT and murine

```{r complesHeatmap_RNAseq_affy_ATRT}

# Subset samples
samples_affy_RNAseq_ATRT <- filter(.data            = samples_affy_RNAseq,
                                   (Organism       == "Homo_sapiens" &
                                    Tumor_sub_type == "ATRT") |
                                   Organism        == "Mus_musculus"
                                   )

# Subset matrix
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_ATRT <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr[,as.vector(samples_affy_RNAseq_ATRT$Sample_name)]

# Most variable genes
mvgenes_affy_RNAseq_ATRT <- genes.selection(data = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_ATRT,
                                       thres.num = 100
                                       )

# Subset matrix
my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_ATRT_mvgenes <- my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_ATRT[mvgenes_affy_RNAseq_ATRT,]

data_type_col <- c(RNA_microarray = "darkslateblue",
                   RNA_seq = "green"
                   )

# Top Annotation
my_top_annot_affy_RNAseq_ATRT <- HeatmapAnnotation(df                      = samples_affy_RNAseq_ATRT[,c("Data_type","Organism","Tumor_sub_type")],
                                              col                     = list(#Tumor_type = tumor_type_col,
                                                                             Data_type = data_type_col,
                                                                             Organism = organism_col,
                                                                             Tumor_sub_type = tumor_sub_type_col
                                                                             ),
                                              gap                     = my_gap,
                                              show_annotation_name    = TRUE,
                                              annotation_name_gp      = my_annotation_name_gp,
                                              annotation_legend_param = my_annotation_legend_param
                                              )

#pdf(paste0(pathToFigures,"complexHeatmap_affy_RNAseq_murine_human_ATRT.pdf"), width=18, height=12)
Heatmap(matrix                      = my_matrix_human_murine_affy_RNAseq_batchCorr_scaled_batchCorr_ATRT_mvgenes,
        clustering_distance_columns = "pearson",
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = my_dist,
        clustering_method_rows      = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine and human tumors clustering \n from RNAseq and affymetrix data \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_affy_RNAseq_ATRT,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 6),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = gpar(fontsize = 14)
        )
#dev.off()

```


### SVA method


#### Prepare matrices

```{r cross_species_and_cross_technics_data_integration_by_Zscaling}

# Murine sva and scaled matrix
head(my_matrix_murine_RNAseq_affy)
head(samples_R_RNAseq_affy)

ncol(my_matrix_murine_RNAseq_affy)

samples_murine_RNAseq_affy <- samples_R_RNAseq_affy

# Human sva and scaled matrix
load(paste0(pathToANALYSIS_PROJECTS,"/human_ATRT/svn/analyse/script/human_RNAseq_affy_list.RData"))

my_matrix_human_RNAseq_affy <- human_RNAseq_affy_list$matrix
head(my_matrix_human_RNAseq_affy)

head(symbols_conversion)

# Add murine gene symbol to human matrix
my_matrix_human_RNAseq_affy_mouseSymbol <- merge(x = symbols_conversion,
                                                 y = my_matrix_human_RNAseq_affy,
                                                 by.x = "HGNC.symbol",
                                                 by.y = 0,
                                                 all = FALSE
                                                 )
rownames(my_matrix_human_RNAseq_affy_mouseSymbol) <- my_matrix_human_RNAseq_affy_mouseSymbol$MGI.symbol
my_matrix_human_RNAseq_affy_mouseSymbol$MGI.symbol <- NULL
my_matrix_human_RNAseq_affy_mouseSymbol$HGNC.symbol <- NULL
head(my_matrix_human_RNAseq_affy_mouseSymbol[,1:6])

samples_human_RNAseq_affy <- human_RNAseq_affy_list$sampleTable

samples_human_RNAseq_affy[,c("Data_type","Sample_name","group_name")]

```

#### Merge human and murine matrix

```{r merge_human_murine_RNAseq_affy}

nrow(my_matrix_murine_RNAseq_affy)

nrow(my_matrix_human_RNAseq_affy_mouseSymbol)

# Merge human and murine primary tumors matrices
my_matrix_murine_human_RNAseq_affy <- merge(x   = my_matrix_murine_RNAseq_affy,
                                            y   = my_matrix_human_RNAseq_affy_mouseSymbol,
                                            by  = 0,
                                            all = FALSE
                                            )
rownames(my_matrix_murine_human_RNAseq_affy) <- my_matrix_murine_human_RNAseq_affy$Row.names
my_matrix_murine_human_RNAseq_affy$Row.names <- NULL
head(my_matrix_murine_human_RNAseq_affy[,1:6])
nrow(my_matrix_murine_human_RNAseq_affy)


# Merge human primary tumors and murine primary tumors + Nestin matrices

my_matrix_murineRN_humanR_RNAseq_affy <- merge(x   = my_matrix_murine_R_N_RNAseq_R_affy,
                                               y   = my_matrix_human_RNAseq_affy_mouseSymbol,
                                               by  = 0,
                                               all = FALSE
                                               )
rownames(my_matrix_murineRN_humanR_RNAseq_affy) <- my_matrix_murineRN_humanR_RNAseq_affy$Row.names
my_matrix_murineRN_humanR_RNAseq_affy$Row.names <- NULL
head(my_matrix_murineRN_humanR_RNAseq_affy[,1:6])
nrow(my_matrix_murineRN_humanR_RNAseq_affy)

```

#### Bind sample tables

```{r bind_human_murine_RNAseq_affy_sampleTable}

head(samples_murine_RNAseq_affy)

head(samples_human_RNAseq_affy)

samples_murine_human_RNAseq_affy <- rbind(samples_murine_RNAseq_affy,samples_human_RNAseq_affy)

samples_murine_human_RNAseq_affy$Sample_name

colnames(my_matrix_murine_human_RNAseq_affy)

# Reorder sample name according to matrix colnames order
samples_murine_human_RNAseq_affy <- samples_murine_human_RNAseq_affy[match(colnames(my_matrix_murine_human_RNAseq_affy),samples_murine_human_RNAseq_affy$Sample_name),]

samples_murine_human_RNAseq_affy$Sample_name

samples_murine_human_RNAseq_affy$group_name

```

```{r bind_humanR_murineRN_RNAseq_affy_sampleTable}

head(samples_R_N_RNAseq_R_affy)

head(samples_murine_human_RNAseq_affy)

samples_murineRN_humanR_RNAseq_affy <- rbind(samples_R_N_RNAseq_R_affy,samples_human_RNAseq_affy)

samples_murineRN_humanR_RNAseq_affy$Sample_name
nrow(samples_murineRN_humanR_RNAseq_affy)

colnames(my_matrix_murineRN_humanR_RNAseq_affy)
ncol(my_matrix_murineRN_humanR_RNAseq_affy)

# Reorder sample name according to matrix colnames order
samples_murineRN_humanR_RNAseq_affy <- samples_murineRN_humanR_RNAseq_affy[match(colnames(my_matrix_murineRN_humanR_RNAseq_affy),samples_murineRN_humanR_RNAseq_affy$Sample_name),]

samples_murineRN_humanR_RNAseq_affy$Sample_name

samples_murineRN_humanR_RNAseq_affy$group_name

```

#### Hierarchical clustering

```{r complexHeatmap_human_murine_RNAseq_affy}

samples_murine_human_RNAseq_affy_plus <- samples_murine_human_RNAseq_affy

samples_murine_human_RNAseq_affy_plus$group_name_tmp <- ifelse(samples_murine_human_RNAseq_affy_plus$group_name == "mIC", "SHH",
                                                        ifelse(samples_murine_human_RNAseq_affy_plus$group_name == "mIC_EC", "MYC",
                                                        ifelse(samples_murine_human_RNAseq_affy_plus$group_name == "TYR", "TYR",
                                                        ifelse(samples_murine_human_RNAseq_affy_plus$group_name == "SHH", "SHH", "MYC")
                                                        )
                                                        )
                                                        )

samples_murine_human_RNAseq_affy_plus[,c("Sample_name","group_name","group_name_tmp")]

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(group_name_tmp),
                              data = samples_murine_human_RNAseq_affy_plus
                              )

my_null_model <- model.matrix(object = ~1,
                              data = samples_murine_human_RNAseq_affy_plus
                              )

# Estimate the number of latent factor (surrogate varibles)
nb_sv <- num.sv(dat = as.matrix(my_matrix_murine_human_RNAseq_affy),
                mod = my_full_model,
                method = "leek"
                )

nb_sv

my_svobj <- sva(dat = as.matrix(my_matrix_murine_human_RNAseq_affy),
             mod = my_full_model,
             mod0 = my_null_model,
             n.sv = 1
             )

str(my_svobj)

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

my_matrix_murine_human_RNAseq_affy_tmp <- my_matrix_murine_human_RNAseq_affy
my_matrix_murine_human_RNAseq_affy_tmp$pprob.gam <- my_svobj$pprob.gam

head(my_matrix_murine_human_RNAseq_affy_tmp)

pprob.gam_thr <- 0.5
my_matrix_murine_human_RNAseq_affy_pprobFilt <- my_matrix_murine_human_RNAseq_affy_tmp[which(my_matrix_murine_human_RNAseq_affy_tmp$pprob.gam < pprob.gam_thr),]
my_matrix_murine_human_RNAseq_affy_pprobFilt$pprob.gam <- NULL

head(my_matrix_murine_human_RNAseq_affy_pprobFilt)

nrow(my_matrix_murine_human_RNAseq_affy_pprobFilt)

# PCA after batch correction
plot_pca(matrix      = my_matrix_murine_human_RNAseq_affy_pprobFilt,
         category    = as.vector(samples_murine_human_RNAseq_affy_plus$Organism),
         plotInertia = TRUE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )


# Top annotation
my_top_annot_human_murine_RNAseq_affy = HeatmapAnnotation(df                      = samples_murine_human_RNAseq_affy[,c("Organism","Data_type","group_name")],
                                                          col                     = list(Organism = organism_col,
                                                                                         Data_type = data_type_col,
                                                                                         group_name = group_name_col
                                                                                         ),
                                                          gap                     = my_gap,
                                                          show_annotation_name    = TRUE,
                                                          annotation_name_gp      = my_annotation_name_gp,
                                                          annotation_legend_param = my_annotation_legend_param
                                                          )

#pdf(paste0(pathToFigures,"/complexHeatmap_human_murine_RNAseq_affy.pdf"), width=16, height=10)
Heatmap(matrix                      = my_matrix_murine_human_RNAseq_affy_pprobFilt,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine tumors clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "133 genes after sva analysis",
        top_annotation              = my_top_annot_human_murine_RNAseq_affy,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```



#### Hierarchical clustering with Nestin RNAseq

```{r complexHeatmap_humanR_murineRN_RNAseq_affy}

my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt <- my_matrix_murineRN_humanR_RNAseq_affy[rownames(my_matrix_murine_human_RNAseq_affy_pprobFilt),]

nrow(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt)

my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt <- (my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt[complete.cases(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt),])

nrow(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt)

plot(density(as.matrix(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt)))

colnames(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt)

# Remove human RNAseq sample
samples_murineRN_humanR_RNAseq_affy_ <- filter(.data = samples_murineRN_humanR_RNAseq_affy,
                                               Organism == "Mus_musculus" |
                                               (Organism == "Homo_sapiens" &
                                                Data_type == "RNA_microarray"
                                               )
                                               )

nrow(samples_murineRN_humanR_RNAseq_affy_)

# Subset matrix after removing human RNAseq sample
my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt_ <- my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt[,as.vector(samples_murineRN_humanR_RNAseq_affy_$Sample_name)]

ncol(my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt_)

# PCA after batch correction
plot_pca(matrix      = my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt_,
         category    = as.vector(samples_murineRN_humanR_RNAseq_affy_$Genotype),
         plotInertia = TRUE,
         addEllipses = FALSE,
         axes        = c(1,2)
         )


samples_murineRN_humanR_RNAseq_affy_plus <- samples_murineRN_humanR_RNAseq_affy_

samples_murineRN_humanR_RNAseq_affy_plus$Genotype <- ifelse(samples_murineRN_humanR_RNAseq_affy_plus$Genotype == "R26", "R26",
                                                     ifelse(samples_murineRN_humanR_RNAseq_affy_plus$Genotype == "Nestin", "Nestin",
                                                            NA))

samples_murineRN_humanR_RNAseq_affy_plus$Genotype

# Top annotation
my_top_annot_humanR_murineRN_RNAseq_affy = HeatmapAnnotation(df                      = samples_murineRN_humanR_RNAseq_affy_plus[,c("Organism","Data_type","Genotype","group_name")],
                                                          col                     = list(Organism = organism_col,
                                                                                         Data_type = data_type_col,
                                                                                         group_name = group_name_col,
                                                                                         Genotype = genotype_col
                                                                                         ),
                                                          gap                     = my_gap,
                                                          show_annotation_name    = TRUE,
                                                          annotation_name_gp      = my_annotation_name_gp,
                                                          annotation_legend_param = my_annotation_legend_param
                                                          )

#pdf(paste0(pathToFigures,"/complexHeatmap_human_murine_RNAseq_affy_plusNestin.pdf"), width=16, height=12)
Heatmap(matrix                      = my_matrix_murineRN_humanR_RNAseq_affy_pprobFilt_,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine tumors clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "133 genes after sva analysis",
        top_annotation              = my_top_annot_humanR_murineRN_RNAseq_affy,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
#dev.off()

```
































# Immunology

## mICs Cytokine genes expression on mIC primary tumors

Here we want to see the expression level of the murine homologous of the following cytokine genes : CCL2, CCL4, CCL5, CXCL6, CXCL10, CXCL12, CSF1, CSF2, CSF3 in mouse mIC and mE/IC. (cf mail Amaury on 24/02/2017))

### Convert human symbol to mouse symbol

```{r cytokine_genes_murine_orthologous, eval=FALSE}

# Import selected genes 
hs_cytokines_genes_selected <- c("CCL2","CCL4","CCL5","CXCL6","CXCL10","CXCL12","CSF1","CSF2","CSF3")
    
# Convert human cytokines genes symbol to murine annotation using biomaRt package

human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             #attributes = c("external_gene_name"),
                             filters    = "hgnc_symbol",
                             #filters    = "external_gene_name",
                             values     = hs_cytokines_genes_selected,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )
symbols_conversion

# Add Ccl2 symbol that could not be converted by biomart
mm_cytokines_genes_selected <- c("Ccl2",as.vector(symbols_conversion$MGI.symbol))

mm_cytokines_genes_selected

```

### Subset samples and matrix

```{r select_murine_black6_primary_tumor, eval=FALSE}

# Select mIC1 and mIC2 primary tumor
samples_ATRT_primaryTumor_black6 <- filter(.data               = samples_LMNR,
                                           Tumor_type          == "RT",
                                           Tumor_sub_type      == "ATRT",
                                           Tumor_origin        == "primary_tumor",
                                           Background_genotype == "black6"
                                           )

# Subset matrix
my_matrix_ATRT_primaryTumor_black6 <- my_matrix_annotated[,c("symbol",as.vector(samples_ATRT_primaryTumor_black6$Sample_name))]

my_matrix_cytokines <- my_matrix_ATRT_primaryTumor_black6[which(my_matrix_ATRT_primaryTumor_black6$symbol %in% mm_cytokines_genes_selected),]

my_matrix_cytokines

# Plot boxplot
my_matrix_cytokines_m <- melt(my_matrix_cytokines)

head(my_matrix_cytokines_m)

my_matrix_cytokines_m$group_name <- rep(as.vector(samples_ATRT_primaryTumor_black6$group_name), each=nrow(my_matrix_cytokines))

colnames(my_matrix_cytokines_m) <- c("gene", "sample", "expression", "subgroup")

#pdf(file="Cytokines_genes_in_mouse_ATRT.pdf", width=12, height=8)
p <- ggplot(data = my_matrix_cytokines_m,
       aes(x=subgroup, y=expression, fill=subgroup)) +
    geom_boxplot() + facet_grid(.~gene) +
    scale_x_discrete(limits=c("mIC1","mIC2")) +
    scale_fill_manual(values = c("red", "green")) +
    theme(legend.position = "none",
          axis.text.x     = element_text(face="bold", color="black", size=12, angle=90),
          axis.text.y     = element_text(face="bold", color="black", size=18),
          axis.title.x    = element_text(face="bold", color="black", size=18),
          axis.title.y    = element_text(face="bold", color="black", size=18),
          axis.line.x     = element_line(color="black", size=1),
          axis.line.y     = element_line(color="black", size=1),
          strip.text.x    = element_text(size=12, colour="black", angle = 0),
          panel.background = element_blank()
          )
ggplotly(p)
#dev.off()

```

### Correlation study

```{r correlation_study, eval=FALSE}

head(my_matrix_ATRT_primaryTumor_black6)

# Remove duplicated symbol
my_matrix_annotated_uniqSym <- my_matrix_annotated[!duplicated(my_matrix_ATRT_primaryTumor_black6[,1]),]

# Remove NA

my_matrix_annotated_uniqSym <- my_matrix_annotated_uniqSym[complete.cases(my_matrix_annotated_uniqSym),]
                                                   
#names(my_matrix_annotated_uniqSym)

# Change row names to gene symbol
row.names(my_matrix_annotated_uniqSym) <- my_matrix_annotated_uniqSym[,1]

# Select U830 ATRT samples

my_matrix_annotated_uniqSymq_black6_ATRT <- my_matrix_annotated_uniqSym[,as.vector(samples_ATRT_primaryTumor_black6_nameSubGroup$Sample_name)]

geneCor <- cor(t(my_matrix_annotated_uniqSymq_black6_ATRT))

mm_cytokines_genes_selected

macrophagic_genes <- c("Ccr1","Cd14","Itgam","Csf1r","Mrc1","Arg1","Cd163")

#is.element(mm_cytokines_genes_selected, rownames(geneCor))

#is.element(macrophagic_genes, rownames(geneCor))

geneCor_interest <- geneCor[which(rownames(geneCor) %in% mm_cytokines_genes_selected),macrophagic_genes]

nrow(geneCor_interest)

# Look at the correlation scores of the two genes
geneCor_interest

geneCor_interest_m <- melt(geneCor_interest)

head(geneCor_interest_m)

#pdf(file="Cytokines_macrophagic_genes_correlation_on_black6_ATRT.pdf", width=12, height=8)
p <- ggplot(data = geneCor_interest_m,
       aes(x=reorder(Var1, value, FUN=mean), y=Var2, fill=value)) +
    geom_tile() +
    #geom_text(aes(Var2, Var1, label=round(value, digits=2)), color = "black", size = 3) +
    scale_fill_gradient2(low = "grey", high = "red", mid = "blue", midpoint = 0, limit = c(min(geneCor_interest_m$value),1), space = "Lab", name="Pearson\nCorrelation") +
    ggtitle("Correlation between cytokines and macrophagic genes \n in murine ATRT primary tumor (black6)") +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          panel.grid.major = element_blank(),
          panel.border = element_blank(),
          panel.background = element_blank(),
          axis.ticks = element_blank(),
          axis.text.x = element_text(face="bold", color="black", size=18, angle=90),
          axis.text.y = element_text(face="bold", color="black", size=18)
          )
ggplotly(p)
#dev.off()

```

## TLR3 pathway associated genes in mICs of black6 Rosa26

### Extract TLR3 gene set from the file sended by

```{r import_TLR3_associated_genes, eval=FALSE} 

TLR3_genes <- read.csv(file = paste0(info.path,"Gene list for TLR3 pathway.csv"),
                       header = TRUE,
                       sep = ";"
                       )

TLR3_genes

TLR3_genes_vect <- as.vector(TLR3_genes[,2])

# Remove empty element in vector
TLR3_genes_vect <- TLR3_genes_vect[TLR3_genes_vect != ""]

#TLR3_genes_vect

# Convert human gene symbol into mouse gene symbol

TLR3_genes_human_mouse <- getLDS(attributes = c("hgnc_symbol"),
                                 filters    = "hgnc_symbol",
                                 values     = TLR3_genes_vect,
                                 mart       = human,
                                 attributesL= "mgi_symbol",
                                 martL      = mouse,
                                 uniqueRows = TRUE
                                 )

TLR3_genes_human_mouse

TLR3_genes_mouse <- as.vector(TLR3_genes_human_mouse[,2])
                              
TLR3_genes_mouse

```

### Subset black6 Rosa26 primary RT tumor samples

```{r select_black6_RT_samples, eval=FALSE}

# Subset black6 RT primary tumor
samples_black6_RT_primaryTumor <- filter(.data                = samples_R_nameSubGroup,
                                         Tumor_type          == "RT",
                                         Tumor_origin        == "primary_tumor" |
                                         (Tumor_origin       == "allograft_flanc" &
                                          Genotype           == "black6"),
                                         Background_genotype == "black6",
                                         include             == "YES"
                                         )

samples_black6_RT_primaryTumor[, c("Date",
                                   "Tumor_type",
                                   "Tumor_origin",
                                   "Background_genotype",
                                   "Genotype",
                                   "Sample_name",
                                   "File_name",
                                   "RT_group_name"
                                   )]

nrow(samples_black6_RT_primaryTumor)

```

### Subset matrix 
include only primary RT (black6) samples and TLR3 genes

```{r matrix_RT_black6_rosa26, eval=FALSE}

# Subset RT black6 matrix and change rownames to gene symbol
my_matrix_annotated_TLR3_genes <- my_matrix_annotated[which(my_matrix_annotated$symbol %in% TLR3_genes_mouse), c("symbol",as.vector(samples_black6_RT_primaryTumor$Sample_name))]
rownames(my_matrix_annotated_TLR3_genes) <- my_matrix_annotated_TLR3_genes[,1]
my_matrix_annotated_TLR3_genes <- my_matrix_annotated_TLR3_genes[,-1]

my_matrix_annotated_TLR3_genes

```

### Unsupervised clustering and heatmap

```{r clust_heatmap_TLR3_genes_RT, eval=FALSE}

my_matrix_annotated_TLR3_genes_hc <- clustering(data   = my_matrix_annotated_TLR3_genes,
                                                metric = "pearson",
                                                method = "ward"
                                                )

my_matrix_annotated_TLR3_genes_ghc <- clustering(data   = t(my_matrix_annotated_TLR3_genes),
                                                 metric = "pearson",
                                                 method = "ward"
                                                 )

clustering.plot(tree         = my_matrix_annotated_TLR3_genes_hc,
                tree.sup     = my_matrix_annotated_TLR3_genes_ghc,
                data         = my_matrix_annotated_TLR3_genes,
                lab          = samples_black6_RT_primaryTumor$RT_group_name,
                trim.heatmap = 0.99
                )

```



## Cytokine genes expression on mICs black6 primary tumors and syngenic

Here we want to see the expression level of the murine homologous of the following cytokine genes : CCL2, CCL4, CCL5, CXCL6, CXCL10, CXCL12, CSF1, CSF2, CSF3 in mouse mIC and mE/IC (*Amaury, cf mail on 24/02/2017*)

### Convert human symbol to mouse symbol

```{r cytokine_genes_murine_orthologous_idem, eval=FALSE}

# Import selected genes 
hs_cytokines_genes_selected <- c("CCL2","CCL4","CCL5","CXCL6","CXCL10","CXCL12","CSF1","CSF2","CSF3")
    
# Convert human cytokines genes symbol to murine annotation using biomaRt package
library(biomaRt)

human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

symbols_conversion <- getLDS(attributes = c("hgnc_symbol"),
                             #attributes = c("external_gene_name"),
                             filters    = "hgnc_symbol",
                             #filters    = "external_gene_name",
                             values     = hs_cytokines_genes_selected,
                             mart       = human,
                             attributesL= "mgi_symbol",
                             martL      = mouse,
                             uniqueRows = TRUE
                             )

#datatable(symbols_conversion, filter="top")

mm_cytokines_genes_selected <- c("Ccl2",as.vector(symbols_conversion$MGI.symbol))

mm_cytokines_genes_selected

```

### Subset black6 Rosa26 primary RT tumor and syngenic

```{r select_black6_RT_samples_and_syngenic, eval=FALSE}

# Subset black6 RT primary tumor
samples_black6_RT_primaryTumor_syngenic <- filter(.data                = sampleTable,
                                                 #.data                = samples_R_nameSubGroup,
                                                  Tumor_type          == "RT",
                                                  Tumor_origin        == "primary_tumor" |
                                                  (Tumor_origin       == "allograft_flanc" &
                                                   Genotype           == "black6"),
                                                  Background_genotype == "black6",
                                                  include             == "YES"
                                                  )

# add group name for primary tumor
samples_black6_RT_primaryTumor_syngenic_RTgroupeName <- merge(x = samples_black6_RT_primaryTumor_syngenic,
                                                              y = samples_R_nameSubGroup[,c("Sample_name","RT_group_name")],
                                                              all.x = TRUE,
                                                              all.y = FALSE,
                                                              by = "Sample_name"
                                                              )

samples_black6_RT_primaryTumor_syngenic_RTgroupeName[, c("Date",
                                                         "Tumor_type",
                                                         "Tumor_origin",
                                                         "Background_genotype",
                                                         "Genotype",
                                                         "Sample_name",
                                                         "File_name",
                                                         "RT_group_name"
                                                         )]

nrow(samples_black6_RT_primaryTumor_syngenic_RTgroupeName)

```

### subset corresponding matrix

```{r subset_mIC1_mIC2_tumor_black6_primary_tumor_syngenic_cytokine_genes, eval=FALSE}

# Subset matrix
my_matrix_ATRT_primaryTumor_black6_syngenic <- my_matrix_annotated[,c("symbol",as.vector(samples_black6_RT_primaryTumor_syngenic_RTgroupeName$Sample_name))]

my_matrix_cytokines <- my_matrix_ATRT_primaryTumor_black6_syngenic[which(my_matrix_ATRT_primaryTumor_black6_syngenic$symbol %in% mm_cytokines_genes_selected),]

my_matrix_cytokines

# Plot boxplot
my_matrix_cytokines_m <- melt(my_matrix_cytokines)

my_matrix_cytokines_m

my_matrix_cytokines_m$group_name <- rep(as.vector(samples_black6_RT_primaryTumor_syngenic_RTgroupeName$RT_group_name), each=nrow(my_matrix_cytokines))

colnames(my_matrix_cytokines_m) <- c("gene", "sample", "expression", "subgroup")

#pdf(file="Cytokines_genes_in_mouse_ATRT.pdf", width=12, height=8)
p <- ggplot(data = my_matrix_cytokines_m,
       aes(x    = subgroup, 
           y    = expression, 
           fill = subgroup)) +
    geom_boxplot() + facet_grid(.~gene) +
    scale_x_discrete(limits = c("mIC1","mIC2")) +
    scale_fill_manual(values = c("red", "green")) +
    theme(legend.position = "none",
          axis.text.x     = element_text(face="bold", color="black", size=12, angle=90),
          axis.text.y     = element_text(face="bold", color="black", size=18),
          axis.title.x    = element_text(face="bold", color="black", size=18),
          axis.title.y    = element_text(face="bold", color="black", size=18),
          axis.line.x     = element_line(color="black", size=1),
          axis.line.y     = element_line(color="black", size=1),
          strip.text.x    = element_text(size=12, colour="black", angle = 0),
          panel.background = element_blank()
          )
ggplotly(p)
#dev.off()

```

### Correlation study between cytokine and macrophagic genes

```{r correlation_study_black6_primaryTumor_syng, eval=FALSE}

head(my_matrix_ATRT_primaryTumor_black6_syngenic)

# Remove duplicated symbol
my_matrix_annotated_uniqSym <- my_matrix_unfiltered_annotated[!duplicated(my_matrix_ATRT_primaryTumor_black6_syngenic[,1]),]

# Remove NA

my_matrix_annotated_uniqSym <- my_matrix_annotated_uniqSym[complete.cases(my_matrix_annotated_uniqSym),]
                                                   
names(my_matrix_annotated_uniqSym)

# Change row names to gene symbol
row.names(my_matrix_annotated_uniqSym) <- my_matrix_annotated_uniqSym[,1]

# Select U830 ATRT samples

my_matrix_annotated_uniqSymq_black6_ATRT <- my_matrix_annotated_uniqSym[,as.vector(samples_ATRT_primaryTumor_black6_nameSubGroup$Sample_name)]

geneCor <- cor(t(my_matrix_annotated_uniqSymq_black6_ATRT))

mm_cytokines_genes_selected

macrophagic_genes <- c("Ccr1","Cd14","Itgam","Csf1r","Mrc1","Arg1","Cd163")

is.element(mm_cytokines_genes_selected, rownames(geneCor))

is.element(macrophagic_genes, rownames(geneCor))

geneCor_interest <- geneCor[which(rownames(geneCor) %in% mm_cytokines_genes_selected),macrophagic_genes]

nrow(geneCor_interest)

# Look at the correlation scores of the two genes
geneCor_interest

geneCor_interest_m <- melt(geneCor_interest)

head(geneCor_interest_m)

#pdf(file="Cytokines_macrophagic_genes_correlation_on_black6_ATRT.pdf", width=12, height=8)
p <- ggplot(data = geneCor_interest_m,
       aes(x=reorder(Var1, value, FUN=mean), y=Var2, fill=value)) +
    geom_tile() +
    #geom_text(aes(Var2, Var1, label=round(value, digits=2)), color = "black", size = 3) +
    scale_fill_gradient2(low = "grey", high = "red", mid = "blue", midpoint = 0, limit = c(min(geneCor_interest_m$value),1), space = "Lab", name="Pearson\nCorrelation") +
    ggtitle("Correlation between cytokines and macrophagic genes \n in murine ATRT primary tumor (black6)") +
    theme(axis.title.x     = element_blank(),
          axis.title.y     = element_blank(),
          panel.grid.major = element_blank(),
          panel.border     = element_blank(),
          panel.background = element_blank(),
          axis.ticks       = element_blank(),
          axis.text.x      = element_text(face="bold", color="black", size=18, angle=90),
          axis.text.y      = element_text(face="bold", color="black", size=18)
          )
ggplotly(p)
#dev.off()

```

## TLR3 pathway associated genes and murine RT primary tumor and syngenic graft

### Extract TLR3 gene set from the file sended by Amaury

```{r import_TLR3_associated_genes_idem, eval=FALSE} 

TLR3_genes <- read.csv(file   = paste0(info.path,"Gene list for TLR3 pathway.csv"),
                       header = TRUE,
                       sep    = ";"
                       )

TLR3_genes

TLR3_genes_vect <- as.vector(TLR3_genes[,2])

# Remove empty element in vector
TLR3_genes_vect <- TLR3_genes_vect[TLR3_genes_vect != ""]

TLR3_genes_vect

# Convert human gene symbol into mouse gene symbol
TLR3_genes_human_mouse <- getLDS(attributes = c("hgnc_symbol"),
                                 filters    = "hgnc_symbol",
                                 values     = TLR3_genes_vect,
                                 mart       = human,
                                 attributesL= "mgi_symbol",
                                 martL      = mouse,
                                 uniqueRows = TRUE
                                 )

TLR3_genes_human_mouse

TLR3_genes_mouse <- as.vector(TLR3_genes_human_mouse[,2])
                              
TLR3_genes_mouse

```

### Subset matrix 

including only the primary RT (black6) samplesand TLR3 genes

```{r matrix_RT_black6_syng, eval=FALSE}

# Subset RT black6 matrix and change rownames to gene symbol
my_matrix_unfiltered_annotated_TLR3_genes <- my_matrix_unfiltered_annotated[which(my_matrix_unfiltered_annotated$symbol %in% TLR3_genes_mouse), c("symbol_unfiltered",as.vector(samples_black6_RT_primaryTumor_syngenic_RTgroupeName$Sample_name))]
rownames(my_matrix_unfiltered_annotated_TLR3_genes) <- my_matrix_unfiltered_annotated_TLR3_genes[,1]
my_matrix_unfiltered_annotated_TLR3_genes <- my_matrix_unfiltered_annotated_TLR3_genes[,-1]
my_matrix_unfiltered_annotated_TLR3_genes

```

### Unsupervised clustering and heatmap

```{r clust_heatmap_TLR3_genes_RT_black6_syng, eval=FALSE}

my_matrix_unfiltered_annotated_TLR3_genes_hc <- clustering(data   = my_matrix_unfiltered_annotated_TLR3_genes,
                                                metric = "pearson",
                                                method = "ward"
                                                )

my_matrix_unfiltered_annotated_TLR3_genes_ghc <- clustering(data   = t(my_matrix_unfiltered_annotated_TLR3_genes),
                                                 metric = "pearson",
                                                 method = "ward"
                                                 )

clustering.plot(tree         = my_matrix_unfiltered_annotated_TLR3_genes_hc,
                tree.sup     = my_matrix_unfiltered_annotated_TLR3_genes_ghc,
                data         = my_matrix_unfiltered_annotated_TLR3_genes,
                lab          = samples_black6_RT_primaryTumor_syngenic_RTgroupeName$RT_group_name,
                trim.heatmap = 0.99
                )

```


xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
#!!! NEW DATA INTEGRATION (New)

## Cross-plateform data integration (New)

### Murine affy and RNAseq integration (New)

#### Affy samples (New)

```{r murine_affy_independent_tumors}

head(my_matrix_murine_affy)
nrow(my_matrix_murine_affy)

samples_murine_affy$group_name

plot(density(as.matrix(my_matrix_murine_affy)))

```

#### Scaling murine affy matrix

```{r scaling_murine_affy}

# Scaling normalization
matrix_murine_affy <- scale(x      = my_matrix_murine_affy,
                            center = TRUE,
                            scale  = TRUE
                            )

# Plot density after scaling
plot(density(as.matrix(matrix_murine_affy)))

```

#### murine RNAseq samples (New)

```{r murine_RNAseq_sample}

# Load murine RNAseq rlog primary tumor data
load("murine_RNAseq_list.RData")

sample_murine_RNAseq <- murine_RNAseq_list$sampleTable
matrix_murine_RNAseq <- murine_RNAseq_list$matrix

sample_murine_RNAseq$Genotype

head(matrix_murine_RNAseq)

nrow(matrix_murine_RNAseq)
plot(density(matrix_murine_RNAseq))

names(sample_murine_RNAseq)

```

#### murine RNAseq filtering (New)

```{r murine_RNAseq_filtering}

# Filter RNAseq using sum of rlog 
matrix_murine_RNAseq_plus <- data.frame(matrix_murine_RNAseq, check.names = FALSE)
matrix_murine_RNAseq_plus$sum <- rowSums(matrix_murine_RNAseq_plus, na.rm=FALSE)

sum_thr <- 300

# Plot sum density
#pdf(paste0(pathToFigures,"density_murine_RNAseq_sumTPM.pdf"), width=12, height=12)
plot(density(matrix_murine_RNAseq_plus$sum))
abline(v=sum_thr, col="red")
#dev.off()

# Filter out rows with sum less than 100
matrix_murine_RNAseq_filtered <- matrix_murine_RNAseq_plus[matrix_murine_RNAseq_plus$sum >= sum_thr,]
nrow(matrix_murine_RNAseq_filtered)

# Plot sum density after filtering
plot(density(matrix_murine_RNAseq_filtered$sum))

# Remove sum column
matrix_murine_RNAseq_filtered$sum <- NULL

plot(density(as.matrix(matrix_murine_RNAseq_filtered)))

```

#### Scaling RNAseq matrix (New)

```{r scaling_RNAseq_matrix}

# Scaling normalization
matrix_murine_RNAseq_filtered <- scale(x      = matrix_murine_RNAseq_filtered,
                                       center = TRUE,
                                       scale  = TRUE
                                       )

# Plot density after scaling
plot(density(as.matrix(matrix_murine_RNAseq_filtered)))

```

#### Convert ensemble id to gene symbol

```{r convert_ENSMUS_to_symbol}

# Convert ENSMUS_transcript to ENSMUS_gene by removing the ".X" at the end of the ID

head(ENSMUSG_ENTREZsymbol_unfiltered_frame)

head(matrix_murine_RNAseq_filtered[,1:4])

matrix_murine_RNAseq_filtered_plus <- data.frame(matrix_murine_RNAseq_filtered, check.names = FALSE)

matrix_murine_RNAseq_filtered_plus$ENSMUS_trans <- rownames(data.frame(matrix_murine_RNAseq_filtered_plus))

matrix_murine_RNAseq_filtered_plus$ENSMUSG_gene <- gsub(pattern     = "\\.+.+",
                                                        replacement = "",
                                                        x           = matrix_murine_RNAseq_filtered_plus$ENSMUS_trans,
                                                        perl        = FALSE
                                                        )


p <- ncol(matrix_murine_RNAseq_filtered_plus)-4

head(matrix_murine_RNAseq_filtered_plus[,p:ncol(matrix_murine_RNAseq_filtered_plus)])

# Convert ENSMUS ID to symbol (using the affy annotation package)

matrix_murine_RNAseq_filtered_symbol <- merge(x    = ENSMUSG_ENTREZsymbol_unfiltered_frame,
                                              y    = matrix_murine_RNAseq_filtered_plus,
                                              by.x = "ENSMUSG_unfiltered",
                                              by.y = "ENSMUSG_gene",
                                              all = FALSE
)
                                                     
matrix_murine_RNAseq_filtered_symbol$ENSMUS_trans <- NULL

matrix_murine_RNAseq_filtered_symbol$ENSMUSG_unfiltered <- NULL

head(matrix_murine_RNAseq_filtered_symbol)
nrow(matrix_murine_RNAseq_filtered_symbol)

```

### Merge murine affy and RNAseq samples (New)

```{r merge_murine_affy_RNAseq}

head(matrix_murine_affy)
head(matrix_murine_RNAseq_filtered_symbol)

matrix_murine_affy_RNAseq <- merge(x    = matrix_murine_affy,
                                   y    = matrix_murine_RNAseq_filtered_symbol,
                                   by.x = 0,
                                   by.y = "symbol_unfiltered",
                                   all  = FALSE
                                   )

head(matrix_murine_affy_RNAseq)
nrow(matrix_murine_affy_RNAseq)

colnames(matrix_murine_affy_RNAseq)

# Convert symbol to rownames
rownames(matrix_murine_affy_RNAseq) <- matrix_murine_affy_RNAseq$Row.names
matrix_murine_affy_RNAseq$Row.names <- NULL

head(matrix_murine_affy_RNAseq)
ncol(matrix_murine_affy_RNAseq)

# Combine RNAseq and affymetrix samples

samples_murine_affy$Sample_name

sample_murine_RNAseq$Sample_name

sample_murine_affy_RNAseq <- rbind(samples_murine_affy, sample_murine_RNAseq)

matrix_murine_affy_RNAseq <- matrix_murine_affy_RNAseq[,as.vector(sample_murine_affy_RNAseq$Sample_name)]

sample_murine_affy_RNAseq$Sample_name == colnames(matrix_murine_affy_RNAseq)

colnames(matrix_murine_affy_RNAseq)

# Check density
plot(density(as.matrix(matrix_murine_affy_RNAseq)))

```

### Check if there is batch effect between "affy" and "RNAseq" scaled data

```{r pca_batch}

mvgenes_murine_affy_RNAseq <- genes.selection(data      = matrix_murine_affy_RNAseq,
                                              thres.num = 1000
                                              )

matrix_murine_affy_RNAseq_mvgenes <- matrix_murine_affy_RNAseq[rownames(matrix_murine_affy_RNAseq) %in% mvgenes_murine_affy_RNAseq,]

# PCA
p <- plot_pca(matrix      = matrix_murine_affy_RNAseq_mvgenes,
              category    = sample_murine_affy_RNAseq$Data_type,
              addEllipses = FALSE,
              axes        = c(1,2)
         )
pdf(paste0(pathToFigures,"pca_murineRT_RNAseq_Sequencing_depth.pdf"), width=12, height=12)
p
dev.off()

# Top annotation
my_top_annot_murine_affy_RNAseq <- HeatmapAnnotation(df                      = data.frame(sample_murine_affy_RNAseq[,c("Data_type","Author_Researcher","Tumor_type","Tumor_origin","Genotype")]),
                                                    col                     = list(Data_type  = data_type_col,
                                                                                   Author_Researcher = author_researcher_col,
                                                                                   Tumor_type = tumor_type_col,
                                                                                   Tumor_origin = tumor_origin_col,
                                                                                   Genotype   = genotype_col
                                                                                   ),
                                                    gap                     = my_gap,
                                                    show_annotation_name    = TRUE,
                                                    annotation_name_gp      = my_annotation_name_gp,
                                                    annotation_legend_param = my_annotation_legend_param
                                                    )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_affy_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             column_title                = paste0("murine affy and RNAseq clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = "1000 most variable genes",
             top_annotation              = my_top_annot_murine_affy_RNAseq,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
dev.off()

```

## Quantile normalization of combined murine affy and RNAseq (New)

```{r quantile_normalization_step}

matrix_murine_affy_RNAseq_quantNorm <- quantile_normalisation(matrix_murine_affy_RNAseq)

```

### Remove batch effect using normal samples and sva (New)

```{r remove_batch_effect}

# Subset normal samples
sample_murine_affy_RNAseq_normal <- filter(.data       = sample_murine_affy_RNAseq,
                                           Tumor_type == "normal"
                                           )

sample_murine_affy_RNAseq_normal[,c("Data_type","Tumor_type")]

# Subset normal matrix
matrix_murine_affy_RNAseq_normal <- matrix_murine_affy_RNAseq[,as.vector(sample_murine_affy_RNAseq_normal$Sample_name)]
head(matrix_murine_affy_RNAseq_normal)

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(Data_type),
                              data   = droplevels(sample_murine_affy_RNAseq_normal)
                              )
my_null_model <- model.matrix(object = ~1,
                              data   = droplevels(sample_murine_affy_RNAseq_normal)
                              )

# Estimate the number of latent factor (surrogate varibles)
nb_sv <- num.sv(dat    = as.matrix(matrix_murine_affy_RNAseq_normal),
                mod    = my_full_model,
                method = "leek"
                )
nb_sv

my_svobj <- sva(dat  = as.matrix(matrix_murine_affy_RNAseq_normal),
                mod  = my_full_model,
                mod0 = my_null_model,
                n.sv = 1
                )

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

# add pprob.gam to total matrix
matrix_murine_affy_RNAseq_plus <- matrix_murine_affy_RNAseq

matrix_murine_affy_RNAseq_plus$pprob.gam <- my_svobj$pprob.gam

head(matrix_murine_affy_RNAseq_plus)

pprob.gam_thr <- 0.9

matrix_murine_affy_RNAseq_plus_filt <- matrix_murine_affy_RNAseq_plus[which(matrix_murine_affy_RNAseq_plus$pprob.gam > pprob.gam_thr),] 
matrix_murine_affy_RNAseq_plus_filt$pprob.gam <- NULL

head(matrix_murine_affy_RNAseq_plus_filt)
nrow(matrix_murine_affy_RNAseq_plus_filt)

pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_affy_svaBatchCorr.pdf"), width=18, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_plus_filt,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_murine_affy_RNAseq_plus_filt),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_murine_affy_RNAseq_batchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

# t-SNE
set.seed(seed = 9)

# Comput t-SNE
tsne_model_murine_affy_RNAseq <- Rtsne(X                = t(matrix_murine_affy_RNAseq_plus_filt),
                                       dims             = 2,
                                       perplexity       = 10,
                                       theta            = 0.0,
                                       pca              = TRUE,
                                       check_duplicates = FALSE,
                                       is_distance      = FALSE
                                       )

# Get t-SNE matrix
tsne_model_murine_affy_RNAseq <- as.data.frame(tsne_model_murine_affy_RNAseq$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_plus <- cbind(sample_murine_affy_RNAseq, tsne_model_murine_affy_RNAseq)

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_svaBatchCorr_genotype.pdf"), width=20, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = Tumor_type), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    scale_fill_manual(values = c(genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn","wild_type")],"white")) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_svaBatchCorr_dataType.pdf"), width=18, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Data_type, shape = Tumor_type), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    scale_fill_manual(values = data_type_col[c("RNA_microarray","RNA_seq")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

```

#### Remove data type effect using PCA (New)

```{r pca_human_murine_batch}

PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_murine_affy_RNAseq
                  )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_dataType_pc1pc2.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Data_type), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = data_type_col[c("RNA_microarray","RNA_seq")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# PCA organism PC1 and PC3
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_dataType_pc2pc5.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC2, y = PC5)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Data_type), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = data_type_col[c("RNA_microarray","RNA_seq")]) +
    xlab(paste0("PC2 (",eigValPer[2]," %)")) +
    ylab(paste0("PC5 (",eigValPer[5]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# PCA organism PC1 and PC3
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_groupName_pc2pc5.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC2, y = PC5)
       ) +
    geom_point(aes(shape = Tumor_type, fill = group_name), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(group_name_col[c("neural","non_neural","normal")],"white")) +
    xlab(paste0("PC2 (",eigValPer[2]," %)")) +
    ylab(paste0("PC5 (",eigValPer[5]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# PCA organism PC2 and PC5
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_Genotype_pc2pc5.pdf"), width=18, height=12)
ggplot(data = pci,
       mapping = aes(x = PC2, y = PC5)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Genotype), size = 6) +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn","wild_type")],"white")) +
    xlab(paste0("PC2 (",eigValPer[2]," %)")) +
    ylab(paste0("PC5 (",eigValPer[5]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# Compute genes contribution to PC1 and PC3 (PC1-PC2+PC3)
PCAvar <- get_pca_var(PC)

PCAvar_contrib <- data.frame(PCAvar$contrib)

PCAvar_contrib$PC2PC5 <- PCAvar_contrib[,2] + PCAvar_contrib[,5] - PCAvar_contrib[,1] - PCAvar_contrib[,3] - PCAvar_contrib[,4]

PCAvar_contrib <- PCAvar_contrib[order(PCAvar_contrib$PC2PC5, decreasing = TRUE),]

mcGenes <- rownames(PCAvar_contrib[1:100,])

# Subset matrix
matrix_murine_affy_RNAseq_mcGenes <- matrix_murine_affy_RNAseq[rownames(matrix_murine_affy_RNAseq) %in% mcGenes,]

colnames(matrix_murine_affy_RNAseq_mcGenes) == sample_murine_affy_RNAseq$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq <- HeatmapAnnotation(df = data.frame(sample_murine_affy_RNAseq[,c("Data_type","Tumor_type","Tumor_origin","group_name","Genotype","Data_source","Author_Researcher")]), 
                                                    col = list(Data_type   = data_type_col,
                                                               Tumor_type  = tumor_type_col,
                                                               Tumor_origin= tumor_origin_col,
                                                               group_name  = group_name_col,
                                                               Genotype    = genotype_col,
                                                               Data_source = data_source_col,
                                                               Author_Researcher = author_researcher_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_pcaBatchCorr.pdf"), width=24, height=10)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

#### Remove some samples (RRN)

```{r R26_RT_primarytum_and_NestinRNAseq}


# Sample
sample_R26RTprimaryTum_NestinRNAseq <- filter(.data         = sample_murine_affy_RNAseq,
(Genotype     == "R26" & Tumor_origin == "primary_tumor") |
(Genotype == "Nestin" & Data_type == "RNA_seq"),
Tumor_type   == "RT"
)

sample_R26RTprimaryTum_NestinRNAseq[,c("Sample_name","Tumor_origin","Data_type","Tumor_type","Genotype")]

# Subset matrix
matrix_murine_affy_RNAseq_RRN_mcGenes <- matrix_murine_affy_RNAseq_mcGenes[,as.vector(sample_R26RTprimaryTum_NestinRNAseq$Sample_name)]

colnames(matrix_murine_affy_RNAseq_RRN_mcGenes) == sample_R26RTprimaryTum_NestinRNAseq$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq_RRN <- HeatmapAnnotation(df = data.frame(sample_R26RTprimaryTum_NestinRNAseq[,c("Data_type","Tumor_origin","Genotype")]), 
                                                    col = list(Data_type   = data_type_col,
                                                               #Tumor_type  = tumor_type_col,
                                                               Tumor_origin= tumor_origin_col,
                                                               #group_name  = group_name_col,
                                                               Genotype    = genotype_col#,
                                                               #Data_source = data_source_col,
                                                               #Author_Researcher = author_researcher_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col,
                                           height = unit(3,"cm")
                                           )


# Complex heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RRN_pcaBatchCorr.pdf"), width=20, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_RRN_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_RRN,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             column_title_gp             = gpar(fontsize = 20)
             )
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Plot dendogram only
h <- Heatmap(matrix                 = matrix_murine_affy_RNAseq_RRN_mcGenes,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = my_meth,
             rect_gp                     = gpar(type = "none"),
             cluster_rows                = FALSE,
             show_row_names              = FALSE,
             column_title                = "Unsupervised clustering of murine tumors",
             column_title_gp             = gpar(fontsize = 20),
                                        #row_title                   = "1000 most variable genes",
             top_annotation              = my_top_annot_murine_affy_RNAseq_RRN,
             show_heatmap_legend         = FALSE,
             column_names_side           = "top",
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = gpar(fontsize = 20)
             )
# Add lengend on top annotation
for(legend in c("Data_type","Tumor_origin","Genotype"))
{
    decorate_annotation(legend, {
    # Annotation names on the right
        grid.text(legend, unit(1, "npc") + unit(10, "mm"), 0.5, default.units = "npc", just = "left")
    })
}

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RRN_pcaBatchCorr_dendOnly.pdf"), width=18, height=8)
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Top annotation (to plot only Genotype)
my_top_annot_murine_affy_RNAseq_RRN <- HeatmapAnnotation(df = data.frame(sample_R26RTprimaryTum_NestinRNAseq["Genotype"]), 
                                                    col = list(#Data_type   = data_type_col,
                                                               #Tumor_type  = tumor_type_col,
                                                               #Tumor_origin= tumor_origin_col,
                                                               #group_name  = group_name_col,
                                                               Genotype    = genotype_col#,
                                                               #Data_source = data_source_col,
                                                               #Author_Researcher = author_researcher_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col,
                                           height = unit(1.5,"cm")
                                           )


# Complex heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RRN_pcaBatchCorr_genotype.pdf"), width=20, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_RRN_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_RRN,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             column_title_gp             = gpar(fontsize = 20)
             )
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Plot dendogram only
h <- Heatmap(matrix                 = matrix_murine_affy_RNAseq_RRN_mcGenes,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = my_meth,
             rect_gp                     = gpar(type = "none"),
             cluster_rows                = FALSE,
             show_row_names              = FALSE,
             column_title                = "Unsupervised clustering of murine tumors",
             column_title_gp             = gpar(fontsize = 20),
                                        #row_title                   = "1000 most variable genes",
             top_annotation              = my_top_annot_murine_affy_RNAseq_RRN,
             show_heatmap_legend         = FALSE,
             column_names_side           = "top",
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = gpar(fontsize = 20)
             )
# Add lengend on top annotation
for(legend in c("Genotype"))
{
    decorate_annotation(legend, {
    # Annotation names on the right
        grid.text(legend, unit(1, "npc") + unit(10, "mm"), 0.5, default.units = "npc", just = "left")
    })
}

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RRN_pcaBatchCorr_dendOnly_genotype.pdf"), width=18, height=8)
draw(h, annotation_legend_side = "bottom")
#dev.off()

# PCA analysis
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_RRN_mcGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_R26RTprimaryTum_NestinRNAseq
                  )

# PCA organism PC1 and PC2
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_RRN_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(shape = Tumor_origin, fill = Genotype), size = 10) +
    scale_shape_manual(values = c(23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = genotype_col[c("Nestin","R26")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")
          ) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# Comput t-SNE
tsne_model_murine_affy_RNAseq_RRN <- Rtsne(X                = t(matrix_murine_affy_RNAseq_RRN_mcGenes),
                                           dims             = 2,
                                           perplexity       = 5,
                                           theta            = 0.0,
                                           pca              = TRUE,
                                           check_duplicates = FALSE,
                                           is_distance      = FALSE
                                           )

# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_RRN<- as.data.frame(tsne_model_murine_affy_RNAseq_RRN$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_RRN_plus <- cbind(sample_R26RTprimaryTum_NestinRNAseq, tsne_model_murine_affy_RNAseq_RRN)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_RRN_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_RRN_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = Tumor_origin), size = 10) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(23,21)) +
    scale_fill_manual(values = genotype_col[c("Nestin","R26")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")
      ) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_murine_affy_RNAseq_RRN_mcGenes),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_murine_affy_RNAseq_RRN_pcaBatchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

```

#### Remove some samples (RNP0Pr)

```{r R26RTprimarytum_NestinRNAseq_ATRT_P0_Prss56}


# Sample
sample_RNP0Pr <- filter(.data         = sample_murine_affy_RNAseq,
(Genotype     == "R26" & Tumor_origin == "primary_tumor") |
(Genotype %in% c("Nestin","Prss56") & Data_type == "RNA_seq") | Author_Researcher %in% c("Vitte_et_al","Ng_et_al_2015"),
Tumor_type   != "normal",
Tumor_type != "MB",
Tumor_type != "lymphoma"
)

sample_RNP0Pr[,c("Sample_name","Tumor_origin","Data_type","Author_Researcher","Genotype")]

# Subset matrix
matrix_murine_affy_RNAseq_RNP0Pr_mcGenes <- matrix_murine_affy_RNAseq_mcGenes[,as.vector(sample_RNP0Pr$Sample_name)]

colnames(matrix_murine_affy_RNAseq_RNP0Pr_mcGenes) == sample_RNP0Pr$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq_RNP0Pr <- HeatmapAnnotation(df = data.frame(sample_RNP0Pr[,c("Data_type","Tumor_type","Tumor_origin","Genotype","Data_source","Author_Researcher")]), 
                                                    col = list(Data_type   = data_type_col,
                                                               Tumor_type  = tumor_type_col,
                                                               Tumor_origin= tumor_origin_col,
                                                               #group_name  = group_name_col,
                                                               Genotype    = genotype_col,
                                                               Data_source = data_source_col,
                                                               Author_Researcher = author_researcher_col
                                                               ),
                                           gap = unit(2,"mm"),
                                           show_annotation_name = TRUE,
                                           annotation_name_gp = my_annotation_name_gp,
                                           annotation_legend_param = my_annotation_legend_param,
                                           na_col = my_na_col
                                           )

# Complex heatmap
pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RNP0Pr_pcaBatchCorr.pdf"), width=24, height=10)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_RNP0Pr_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_RNP0Pr,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
dev.off()

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_RNP0Pr_mcGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_RNP0Pr
                  )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_RNP0Pr_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(shape = Tumor_origin, fill = Genotype), size = 6) +
    scale_shape_manual(values = c(23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# Comput t-SNE
tsne_model_murine_affy_RNAseq_RNP0Pr <- Rtsne(X                = t(matrix_murine_affy_RNAseq_RNP0Pr_mcGenes),
                                              dims             = 2,
                                              perplexity       = 10,
                                              theta            = 0.0,
                                              pca              = TRUE,
                                              check_duplicates = FALSE,
                                              is_distance      = FALSE
                                              )

# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_RNP0Pr <- as.data.frame(tsne_model_murine_affy_RNAseq_RNP0Pr$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_RNP0Pr_plus <- cbind(sample_RNP0Pr, tsne_model_murine_affy_RNAseq_RNP0Pr)

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_RNP0Pr_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_RNP0Pr_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = Tumor_origin), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(23,21)) +
    scale_fill_manual(values = genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
        guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

```

#### Remove some samples (RNP0PrMN)

```{r R26RTprimarytum_NestinRNAseq_ATRT_P0_Prss56_RNP0PrMN}


# Sample
sample_RNP0PrMN <- filter(.data         = sample_murine_affy_RNAseq,
(Genotype     == "R26" & Tumor_origin == "primary_tumor") |
(Genotype %in% c("Nestin","Prss56") & Data_type == "RNA_seq") | Author_Researcher %in% c("Vitte_et_al","Ng_et_al_2015") | Tumor_type %in% c("MB","NB"),
Tumor_type   != "normal",
Tumor_type != "lymphoma"
)

sample_RNP0PrMN[,c("Sample_name","Tumor_origin","Data_type","Author_Researcher","Genotype")]

# Subset matrix
matrix_murine_affy_RNAseq_RNP0PrMN_mcGenes <- matrix_murine_affy_RNAseq_mcGenes[,as.vector(sample_RNP0PrMN$Sample_name)]

colnames(matrix_murine_affy_RNAseq_RNP0PrMN_mcGenes) == sample_RNP0PrMN$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq_RNP0PrMN <- HeatmapAnnotation(df = data.frame(sample_RNP0PrMN[,c("Data_type","Tumor_type","Tumor_origin","Genotype","Data_source","Author_Researcher")]), 
                                                              col = list(Data_type   = data_type_col,
                                                                         Tumor_type  = tumor_type_col,
                                                                         Tumor_origin= tumor_origin_col,
                                        #group_name  = group_name_col,
                                                                         Genotype    = genotype_col,
                                                                         Data_source = data_source_col,
                                                                         Author_Researcher = author_researcher_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = TRUE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Complex heatmap
pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_RNP0PrMN_pcaBatchCorr.pdf"), width=24, height=10)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_RNP0PrMN_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_RNP0PrMN,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
dev.off()

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_RNP0PrMN_mcGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_RNP0PrMN
                  )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_RNP0PrMN_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(shape = Tumor_type, fill = Genotype), size = 10) +
    scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = c(genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn")],"white")) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# Comput t-SNE
tsne_model_murine_affy_RNAseq_RNP0PrMN <- Rtsne(X                = t(matrix_murine_affy_RNAseq_RNP0PrMN_mcGenes),
                                                dims             = 10,
                                                perplexity       = 7,
                                                theta            = 0.0,
                                                pca              = TRUE,
                                                check_duplicates = FALSE,
                                                is_distance      = FALSE
                                                )

# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_RNP0PrMN <- as.data.frame(tsne_model_murine_affy_RNAseq_RNP0PrMN$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_RNP0PrMN_plus <- cbind(sample_RNP0PrMN, tsne_model_murine_affy_RNAseq_RNP0PrMN)

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_RNP0PrMN_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_RNP0PrMN_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = Tumor_origin), size = 10) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(23,21)) +
    scale_fill_manual(values = genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn")],"white") +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

```


#### Remove some samples (published)

```{r published_samples}

# Sample
sample_murine_published <- filter(.data         = sample_murine_affy_RNAseq,
                                  published     %in% c("YES","TRUE"),
                                  Tumor_type == "RT",
                                  Tumor_origin == "primary_tumor"
                                  )
nrow(sample_murine_published)

sample_murine_published[,c("Sample_name","Tumor_origin","published","Author_Researcher","Genotype")]

# Subset matrix
matrix_murine_affy_RNAseq_published_mcGenes <- matrix_murine_affy_RNAseq_mcGenes[,as.vector(sample_murine_published$Sample_name)]

colnames(matrix_murine_affy_RNAseq_published_mcGenes) == sample_murine_published$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq_published <- HeatmapAnnotation(df = data.frame(sample_murine_published[,c("Genotype","Author_Researcher")]), 
                                                              col = list(#Data_type   = data_type_col,
                                                                         #Tumor_type  = tumor_type_col,
                                                                         #Tumor_origin= tumor_origin_col,
                                        #group_name  = group_name_col,
                                                                         Genotype    = genotype_col,
                                                                         #Data_source = data_source_col,
                                                                         Author_Researcher = author_researcher_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = TRUE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Complex heatmap
pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=16)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
#draw(h, annotation_legend_side = "bottom")
draw(h)
dev.off()

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_published_mcGenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                  sample_murine_published
                  )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC3)
       ) +
    geom_point(aes(fill = Genotype),shape =23, size = 10) +
    scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = genotype_col[c("P0-CreC_Smarcb1FloxFlox","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC3 (",eigValPer[3]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 23)))
dev.off()

# Compute t-SNE
tsne_model_murine_affy_RNAseq_published <- Rtsne(X                = t(matrix_murine_affy_RNAseq_published_mcGenes),
                                                 dims             = 2,
                                                 perplexity       = 7,
                                                 theta            = 0.0,
                                                 pca              = TRUE,
                                                 check_duplicates = FALSE,
                                                 is_distance      = FALSE
                                                 )


# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_published <- as.data.frame(tsne_model_murine_affy_RNAseq_published$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_published_plus <- cbind(sample_murine_published, tsne_model_murine_affy_RNAseq_published)

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_published_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype), shape =23, size = 10) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(23,21)) +
    scale_fill_manual(values = genotype_col[c("P0-CreC_Smarcb1FloxFlox","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 23)))
dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_murine_affy_RNAseq_published_mcGenes),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_murine_affy_RNAseq_published_pcaBatchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

```

### Published data and ATRT signature genes

#### Convert ensemble id to gene symbol

```{r heatmap_published_ATRTsign}

# Matrix murine unfiltered
head(matrix_murine_RNAseq[,1:4])

matrix_murine_RNAseq_plus <- data.frame(matrix_murine_RNAseq, check.names=FALSE)

rownames(matrix_murine_RNAseq_plus)

# Convert ENSMUS_transcript to ENSMUS_gene by removing the ".X" at the end of the ID

head(ENSMUSG_ENTREZsymbol_unfiltered_frame)

matrix_murine_RNAseq_plus$ENSMUS_trans <- rownames(matrix_murine_RNAseq_plus)

matrix_murine_RNAseq_plus$ENSMUSG_gene <- gsub(pattern     = "\\.+.+",
                                               replacement = "",
                                               x           = matrix_murine_RNAseq_plus$ENSMUS_trans,
                                               perl        = FALSE
                                               )

head(matrix_murine_RNAseq_plus[,1:4])

p <- ncol(matrix_murine_RNAseq_plus)-4

head(matrix_murine_RNAseq_plus[,p:ncol(matrix_murine_RNAseq_plus)])

# Convert ENSMUS ID to symbol (using the affy annotation package)

matrix_murine_RNAseq_symbol <- merge(x    = ENSMUSG_ENTREZsymbol_unfiltered_frame,
                                     y    = matrix_murine_RNAseq_plus,
                                     by.x = "ENSMUSG_unfiltered",
                                     by.y = "ENSMUSG_gene",
                                     all  = FALSE
                                     )
                                                     
matrix_murine_RNAseq_symbol$ENSMUS_trans <- NULL
matrix_murine_RNAseq_symbol$ENSMUSG_unfiltered <- NULL

rownames(matrix_murine_RNAseq_symbol) <- matrix_murine_RNAseq_symbol$symbol_unfiltered

matrix_murine_RNAseq_symbol$symbol_unfiltered <- NULL

head(matrix_murine_RNAseq_symbol[,1:4])

nrow(matrix_murine_RNAseq_symbol)

plot(density(as.matrix(matrix_murine_RNAseq_symbol)))

matrix_murine_RNAseq_symbol_scaled <- scale(x = matrix_murine_RNAseq_symbol,
                                            center = TRUE,
                                            scale = TRUE
                                            )

plot(density(as.matrix(matrix_murine_RNAseq_symbol_scaled)))

# Merge murine RNAseq unfiltered and murine affy matrix

head(matrix_murine_affy[,1:4])

nrow(matrix_murine_affy)

plot(density(as.matrix(matrix_murine_affy)))

matrix_murine_affy_RNAseq_unfiltered <- merge(x = matrix_murine_RNAseq_symbol_scaled,
                                              y = matrix_murine_affy,
                                              by = 0,
                                              all = FALSE
                                              )

rownames(matrix_murine_affy_RNAseq_unfiltered) <- matrix_murine_affy_RNAseq_unfiltered$Row.names

matrix_murine_affy_RNAseq_unfiltered$Row.names <- NULL

head(matrix_murine_affy_RNAseq_unfiltered[,1:4])

plot(density(as.matrix(matrix_murine_affy_RNAseq_unfiltered)))

# Quantile normalization of combined matrix

matrix_murine_affy_RNAseq_unfiltered <- quantile_normalisation(matrix_murine_affy_RNAseq_unfiltered)

# Subset matrix
matrix_murine_affy_RNAseq_published <- matrix_murine_affy_RNAseq_unfiltered[,as.vector(sample_murine_published$Sample_name)]

head(matrix_murine_affy_RNAseq_published[,1:4])

names(geneSet_ATRT)

# Select ATRT signature genes
matrix_murine_affy_RNAseq_published_ATRTsign <- merge(x    = geneSet_ATRT,
                                                      y    = matrix_murine_affy_RNAseq_published,
                                                      by.x = "murine_symbol",
                                                      by.y = 0,
                                                      all  = FALSE
                                                      )
row.names(matrix_murine_affy_RNAseq_published_ATRTsign) <- matrix_murine_affy_RNAseq_published_ATRTsign$murine_symbol

# Reorder matrix according to ATRT subgroup (for complexHeatmap)
matrix_murine_affy_RNAseq_published_ATRTsign <- matrix_murine_affy_RNAseq_published_ATRTsign[order(matrix_murine_affy_RNAseq_published_ATRTsign$Group, decreasing = TRUE),]
#my_matrix_Nestin_MN_ATRTsign           <- my_matrix_Nestin_MN_ATRTsign_plus[,-c(1,2,3,4)]
#my_matrix_Nestin_MN_ATRTsign           <- my_matrix_Nestin_MN_ATRTsign[,-1]


# Row annotation
my_row_annot_published <- rowAnnotation(df = data.frame(Group = matrix_murine_affy_RNAseq_published_ATRTsign[,c("Group")]),
                                        col = list(Group = c("TYR"="red", "SHH"="green", "MYC"="blue")
                                                   )
                                        )

pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_ATRTsign.pdf"), width=12, height=16)
hm <- Heatmap(matrix                      = t(scale(t(matrix_murine_affy_RNAseq_published_ATRTsign[,-c(1:8)]))),
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              clustering_distance_rows    = "euclidean",
              cluster_rows                = TRUE,
              show_row_names              = TRUE,
              show_column_names           = FALSE,
              cluster_columns             = TRUE,
              column_title                = "Published murine RT \n unsupervised clustering (distance: pearson, method:ward)",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_murine_affy_RNAseq_published,
              show_heatmap_legend         = FALSE,
              column_dend_height          = my_column_dend_height,
              column_names_gp             = my_column_names_gp,
              row_title_gp                = my_row_title_gp,
              row_names_gp                = my_row_names_gp
              )
hm + my_row_annot_published
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_ATRTsign_ordered.pdf"), width=12, height=16)
hm <- Heatmap(matrix                      = t(scale(t(matrix_murine_affy_RNAseq_published_ATRTsign[,-c(1:8)]))),
              clustering_distance_columns = my_dist,
              clustering_method_columns   = my_meth,
              clustering_distance_rows    = "euclidean",
              cluster_rows                = FALSE,
              show_row_names              = TRUE,
              show_column_names           = FALSE,
              cluster_columns             = TRUE,
              column_title                = "Published murine RT \n unsupervised clustering (distance: pearson, method:ward)",
              row_title                   = "ATRT signature genes",
              top_annotation              = my_top_annot_murine_affy_RNAseq_published,
              show_heatmap_legend         = FALSE,
              column_dend_height          = my_column_dend_height,
              column_names_gp             = my_column_names_gp,
              row_title_gp                = my_row_title_gp,
              row_names_gp                = my_row_names_gp
              )
hm + my_row_annot_published
dev.off()

```

### Combine murine published data and human affy (New)

```{r combine_murine_published_and_human_affy}

# Combine murine matrix with human matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol[,1:4])
ncol(matrix_human_affy_mouseSymbol)

plot(density(as.matrix(matrix_human_affy_mouseSymbol)))

matrix_human_affy_mouseSymbol_scaled <- scale(x = matrix_human_affy_mouseSymbol,
                                              center = TRUE,
                                              scale = TRUE
                                              )

head(matrix_human_affy_mouseSymbol_scaled[,1:4])
nrow(matrix_human_affy_mouseSymbol_scaled[,1:4])
ncol(matrix_human_affy_mouseSymbol_scaled)

plot(density(as.matrix(matrix_human_affy_mouseSymbol_scaled)))


head(matrix_murine_affy_RNAseq_published[,1:4])
nrow(matrix_murine_affy_RNAseq_published[,1:4])
ncol(matrix_murine_affy_RNAseq_published)

plot(density(as.matrix(matrix_murine_affy_RNAseq_published)))

# Merge matrix 
matrix_murine_affy_RNAseq_published_human_affy <- merge(x = matrix_murine_affy_RNAseq_published,
                                                        y = matrix_human_affy_mouseSymbol_scaled,
                                                        by = 0,
                                                        all = FALSE
                                                        )
rownames(matrix_murine_affy_RNAseq_published_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_published_human_affy$Row.names <- NULL

head(matrix_murine_affy_RNAseq_published_human_affy[,1:4])
ncol(matrix_murine_affy_RNAseq_published_human_affy)
nrow(matrix_murine_affy_RNAseq_published_human_affy)

```

### Combine murine published data and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Merge samples
sample_human_affy_ATRT <- filter(.data = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

sample_murine_published_human_affy_ATRT <- rbind(sample_murine_published,sample_human_affy_ATRT)

nrow(sample_murine_published_human_affy_ATRT)

# Subset matrix
matrix_murine_affy_RNAseq_published_human_affy_ATRT <- matrix_murine_affy_RNAseq_published_human_affy[,as.vector(sample_murine_published_human_affy_ATRT$Sample_name)]

sample_murine_published_human_affy_ATRT$Sample_name == colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

# Quantile normalization
matrix_murine_affy_RNAseq_published_human_affy_ATRT <- quantile_normalisation(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

head(matrix_murine_affy_RNAseq_published_human_affy_ATRT[,1:4])
ncol(matrix_murine_affy_RNAseq_published_human_affy_ATRT)
nrow(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_affy_RNAseq_published_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_affy_RNAseq_published_human_affy_ATRT <- genes.selection(data = matrix_murine_affy_RNAseq_published_human_affy_ATRT,
                                                                   thres.num = 5000
                                                                   )

# Subset matrix of most variable genes
matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes <- matrix_murine_affy_RNAseq_published_human_affy[rownames(matrix_murine_affy_RNAseq_published_human_affy_ATRT) %in% mvgenes_murine_affy_RNAseq_published_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_published_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_PC1PC2.pdf"), width=14, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(fill = Organism),shape =23, size = 10) +
    scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 23)))
dev.off()

# PCA organism PC and PC3
pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_PC2PC4.pdf"), width=14, height=12)
ggplot(data = pci,
       mapping = aes(x = PC2, y = PC4)
       ) +
    geom_point(aes(fill = Organism),shape =23, size = 10) +
    scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab(paste0("PC2 (",eigValPer[2]," %)")) +
    ylab(paste0("PC4 (",eigValPer[4]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 23)))
dev.off()

# Compute genes contribution to PC2 and PC4 (-PC1+PC2-PC3+PC4)
PCAvar <- get_pca_var(PC)
PCAvar_contrib <- data.frame(PCAvar$contrib)
PCAvar_contrib$PC2PC4 <- PCAvar_contrib[,2] + PCAvar_contrib[,4] - PCAvar_contrib[,1] - PCAvar_contrib[,3]

head(PCAvar_contrib)

PCAvar_contrib <- PCAvar_contrib[order(PCAvar_contrib$PC2PC4, decreasing = TRUE),]

mcGenes <- rownames(PCAvar_contrib[1:500,])

# Subset matrix
matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes <- matrix_murine_affy_RNAseq_published_human_affy_ATRT[rownames(matrix_murine_affy_RNAseq_published_human_affy_ATRT) %in% mcGenes,]

colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes) == sample_murine_published_human_affy_ATRT$Sample_name

# Top annotation
my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(sample_murine_published_human_affy_ATRT[,c("Organism","group_name","Genotype","Author_Researcher")]), 
                                                                               col = list(#Data_type   = data_type_col,
                                                               #Tumor_type  = tumor_type_col,
                                                               #Tumor_origin= tumor_origin_col,
                                                        Organism = organism_col,
                                                        group_name  = group_name_col,
                                                        Genotype    = genotype_col,
                                                        #Data_source = data_source_col,
                                                        Author_Researcher = author_researcher_col
                                                    ),
                                                    gap = unit(2,"mm"),
                                                    show_annotation_name = TRUE,
                                                    annotation_name_gp = my_annotation_name_gp,
                                                    annotation_legend_param = my_annotation_legend_param,
                                                    na_col = my_na_col
                                                    )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr.pdf"), width=24, height=10)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h)
#draw(h, annotation_legend_side = "bottom")
#dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

# Compute t-SNE
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT <- Rtsne(X                = t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes),
                                                                 dims             = 3,
                                                                 perplexity       = 30,
                                                                 theta            = 0.0,
                                                                 pca              = TRUE,
                                                                 check_duplicates = FALSE,
                                                                 is_distance      = FALSE
                                                                 )


# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT <- as.data.frame(tsne_model_murine_affy_RNAseq_published_human_affy_ATRT$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT_plus <- cbind(sample_murine_published_human_affy_ATRT, tsne_model_murine_affy_RNAseq_published_human_affy_ATRT)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_published_human_affy_ATRT_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = group_name), size = 10) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(0,21,22,1,2,24)) +
    scale_fill_manual(values = genotype_col[c("P0-CreC_Smarcb1FloxFlox","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()


names(sample_murine_published_human_affy_ATRT)

# Plot for publication
sample_murine_published_human_affy_ATRT_plus <- mutate(.data = sample_murine_published_human_affy_ATRT,
                                                       `rhabdoid tumors` = ifelse(Organism == "Homo_sapiens",
                                                                         paste0("human ",group_name),
                                                                  ifelse(is.na(group_name),
                                                                         paste0("mouse ",Genotype),
                                                                         paste0("mouse ",Genotype," ",group_name)
                                                                         )
                                                                  )
                                                       )

sample_murine_published_human_affy_ATRT_plus[c("Organism","Genotype","group_name","rhabdoid tumors")]

# Top annotation
my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_pub <- HeatmapAnnotation(df = data.frame(sample_murine_published_human_affy_ATRT_plus["rhabdoid tumors"],check.names=FALSE), 
                                                                                   col = list(`rhabdoid tumors` = c("mouse R26 neural" = "brown",#"olivedrab1",
                                                                                                     "mouse R26 non_neural" = "cyan4",
                                                                                                     "mouse Snf5F/F/p53L/L/GFAP-Cre" = "violet",
                                                                                                     "mouse P0-CreC_Smarcb1FloxFlox" = "yellow",
                                                                                                     "human MYC" = "blue",
                                                                                                     "human TYR" = "red",
                                                                                                     "human SHH" = "green",
                                                                                                     "mouse R26" = "darkorange2"
                                                                                                     )
                                                                                              ),
                                                                                   gap = unit(2,"mm"),
                                                                                   show_annotation_name = FALSE,
                                                                                   annotation_name_gp = my_annotation_name_gp,
                                                                                   annotation_legend_param = my_annotation_legend_param,
                                                                                   na_col = my_na_col
                                                                                   )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr_pub.pdf"), width=24, height=10)
Heatmap(matrix                      = t(scale(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes))),
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        show_column_names              = TRUE,
             show_row_dend               = FALSE,
             #rect_gp                     = gpar(type = "none"),
             cluster_row                 = TRUE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_pub,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
#draw(h, annotation_legend_side = "bottom")
#dev.off()

```

```{r Pgdb5_expression}

head(matrix_murine_affy_RNAseq_published_human_affy_ATRT[,1:4])

matrix_murine_affy_RNAseq_published_human_affy_ATRT_plus <- data.frame(matrix_murine_affy_RNAseq_published_human_affy_ATRT, check.names = FALSE)

head(matrix_murine_affy_RNAseq_published_human_affy_ATRT_plus[,1:4])

# Extract Pdgb5 gene expression 
matrix_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5 <- matrix_murine_affy_RNAseq_published_human_affy_ATRT[which(rownames(matrix_murine_affy_RNAseq_published_human_affy_ATRT) == "Pgbd5"),]

matrix_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5_melt <- melt(matrix_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5)

colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5_melt) <- "Pgbd5"

# Merge coldata and matrix
samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5 <- merge(x   = sample_murine_published_human_affy_ATRT,
                                                                    y   = matrix_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5_melt,
                                                                    by.x  = "Sample_name",
                                                                    by.y = 0,
                                                                    all = FALSE
                                                                    )
head(samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5)

samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5$Sample_name == sample_murine_published_human_affy_ATRT$Sample_name

samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5 <- samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5[match(sample_murine_published_human_affy_ATRT$Sample_name, samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5$Sample_name),]

samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5$Sample_name == sample_murine_published_human_affy_ATRT$Sample_name

# Bottom annotation
bottom_annot_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5 <- HeatmapAnnotation(Pgdb5_expression = anno_barplot(x       = samples_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5$Pgbd5,
                                                                                                      axis    = TRUE,
                                                                                                      border  = FALSE,
                                                                                                      outline = FALSE,
                                                                                                      gp      = gpar(fill = "black", col = "white")
                                                                                                      ),
                                                                      gap                     = my_gap,
                                                                      show_annotation_name    = TRUE,
                                                                      annotation_name_gp      = my_annotation_name_gp,
                                     annotation_legend_param = my_annotation_legend_param,
                                     na_col                  = my_na_col
                                     )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr_Pgdb5.pdf"), width=12, height=16)
Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        clustering_distance_rows    = "euclidean",
        show_row_names              = FALSE,
        column_title                = paste0("murine primary tumors affymetrix data\n Hierarchical clustering (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes (IQR)",
        top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT,
        show_row_dend               = FALSE,
        bottom_annotation           = bottom_annot_murine_affy_RNAseq_published_human_affy_ATRT_Pgbd5,
        bottom_annotation_height    = unit(5,"cm"),
        show_heatmap_legend         = FALSE,
        column_dend_height          = unit(40,"mm")
        )
#dev.off()

```

### POSTER CODE HERE

### Combine murine published data and human ATRT affy (DKFZ)

### Combine murine published data and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Merge samples
sample_human_affy_ATRT <- filter(.data = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

# Data summary
group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Add subgroup/genotype column to murine sample info
sample_murine_published_plus <- sample_murine_published
sample_murine_published_plus$subgroup_genotype <- ifelse(sample_murine_published_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                                  ifelse(sample_murine_published_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP","mouse R26")
                                                  )
group_by(.data = sample_murine_published_plus
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_published_human_affy_ATRT <- rbind(sample_murine_published_plus,
                                                 sample_human_affy_ATRT_plus
                                                 )

group_by(.data = sample_murine_published_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

nrow(sample_murine_published_human_affy_ATRT)

# Subset matrix
matrix_murine_affy_RNAseq_published_human_affy_ATRT <- matrix_murine_affy_RNAseq_published_human_affy[,as.vector(sample_murine_published_human_affy_ATRT$Sample_name)]

sample_murine_published_human_affy_ATRT$Sample_name == colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

# Quantile normalization
matrix_murine_affy_RNAseq_published_human_affy_ATRT <- quantile_normalisation(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

head(matrix_murine_affy_RNAseq_published_human_affy_ATRT[,1:4])
ncol(matrix_murine_affy_RNAseq_published_human_affy_ATRT)
nrow(matrix_murine_affy_RNAseq_published_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_affy_RNAseq_published_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_affy_RNAseq_published_human_affy_ATRT <- genes.selection(data = matrix_murine_affy_RNAseq_published_human_affy_ATRT,
                                                                   thres.num = 5000
                                                                   )

# Subset matrix of most variable genes
matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes <- matrix_murine_affy_RNAseq_published_human_affy_ATRT[rownames(matrix_murine_affy_RNAseq_published_human_affy_ATRT) %in% mvgenes_murine_affy_RNAseq_published_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_published_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_PC1PC2_poster.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("MYC","SHH","TYR","GFAP","P0","R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "none",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
#dev.off()

library(cowplot) # To extract legend from ggplot object

my_legend <- get_legend(p)
#pdf(paste0(pathToFigures,"legend_poster.pdf"), width=14, height=10)
ggdraw(my_legend)
#dev.off()

# PCA organism PC and PC3
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_PC2PC4.pdf"), width=14, height=12)
ggplot(data = pci,
       mapping = aes(x = PC2, y = PC4)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("MYC","SHH","TYR","GFAP","P0","R26")])) +
    xlab(paste0("PC2 (",eigValPer[2]," %)")) +
    ylab(paste0("PC4 (",eigValPer[4]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm")
          ) #+
#dev.off()


# Top annotation
my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster <- HeatmapAnnotation(df = data.frame(RT = sample_murine_published_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("before organism effect correction \n"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
#dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

```{r human_samples}

xxxxxxxxxxxxxxxxxxxxxx

# Subset matrix
matrix_human_affy <- matrix_murine_affy_RNAseq_published_human_affy[,as.vector(sample_human_affy_ATRT_plus$Sample_name)]

# Most variable genes
mvgenes_human_affy <- genes.selection(data = matrix_human_affy,
                                      thres.num = 1000
                                      )

# Subset matrix of most variable genes
matrix_human_affy_mvgenes <- matrix_human_affy[rownames(matrix_human_affy) %in% mvgenes_human_affy,]

# Top annotation
top_annot_human_affy <- HeatmapAnnotation(df = data.frame(human_ATRT = sample_human_affy_ATRT_plus[,"group_name"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 human_ATRT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_human_affy_mvgenes)), method = "ward")
pdf(paste0(pathToFigures,"complexHeatmap_human_affy_ATRT.pdf"), width=12, height=16)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_human_affy_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             #top_annotation     = top_annot_human_affy,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "Human ATRT clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
dev.off()

```

#### POSTER: Remove batch using sva::ComBat

```{r remove_batch_effect_sva}

matrix_murine_affy_RNAseq_published_human_affy_ATRT_ComBat <- sva::ComBat(dat = matrix_murine_affy_RNAseq_published_human_affy_ATRT_mvgenes,
                                                                          batch = sample_murine_published_human_affy_ATRT$Organism
                                                                          )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_published_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_combatBatchCorr.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("MYC","SHH","TYR","GFAP","P0","R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "none",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
#dev.off()

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("after organism effect correction \n"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
#dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_affy_RNAseq_published_human_affy_ATRT_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```


```{r remove_batch_pca}

# Compute genes contribution to PC2 and PC4 (-PC1+PC2-PC3+PC4)
PCAvar <- get_pca_var(PC)
PCAvar_contrib <- data.frame(PCAvar$contrib)
PCAvar_contrib$PC2PC4 <- PCAvar_contrib[,2] + PCAvar_contrib[,4] - PCAvar_contrib[,1] - PCAvar_contrib[,3]

head(PCAvar_contrib)

PCAvar_contrib <- PCAvar_contrib[order(PCAvar_contrib$PC2PC4, decreasing = TRUE),]

mcGenes <- rownames(PCAvar_contrib[1:500,])

# Subset matrix
matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes <- matrix_murine_affy_RNAseq_published_human_affy_ATRT[rownames(matrix_murine_affy_RNAseq_published_human_affy_ATRT) %in% mcGenes,]

colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes) == sample_murine_published_human_affy_ATRT$Sample_name

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr.pdf"), width=24, height=10)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine and human RT samples \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_poster,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h)
#draw(h, annotation_legend_side = "bottom")
#dev.off()

# Consensus clustering
ConsensusClusterPlus(d          = as.matrix(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes),
                     maxK       = 6,
                     reps       = 1000,
                     pItem      = 0.8,
                     pFeature   = 0.8,
                     clusterAlg = "hc",
                     title      = "ConsClust_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr",
                     distance   = "pearson",
                     plot       = "pdf"
                     )

# Compute t-SNE
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT <- Rtsne(X                = t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes),
                                                                 dims             = 3,
                                                                 perplexity       = 30,
                                                                 theta            = 0.0,
                                                                 pca              = TRUE,
                                                                 check_duplicates = FALSE,
                                                                 is_distance      = FALSE
                                                                 )


# Get t-SNE matrix
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT <- as.data.frame(tsne_model_murine_affy_RNAseq_published_human_affy_ATRT$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_published_human_affy_ATRT_plus <- cbind(sample_murine_published_human_affy_ATRT, tsne_model_murine_affy_RNAseq_published_human_affy_ATRT)

# Plot t-SNE result without clustering
#pdf(paste0(pathToFigures,"tsne_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_published_human_affy_ATRT_plus,
       mapping = aes(x=V1, y=V2)) +
    geom_point(aes(fill = Genotype, shape = group_name), size = 10) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_shape_manual(values = c(0,21,22,1,2,24)) +
    scale_fill_manual(values = genotype_col[c("P0-CreC_Smarcb1FloxFlox","R26","Snf5F/F/p53L/L/GFAP-Cre")]) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
#dev.off()


names(sample_murine_published_human_affy_ATRT)

# Plot for publication

sample_murine_published_human_affy_ATRT_plus <- mutate(.data = sample_murine_published_human_affy_ATRT,
                                                       `rhabdoid tumors` = ifelse(Organism == "Homo_sapiens",
                                                                         paste0("human ",group_name),
                                                                  ifelse(is.na(group_name),
                                                                         paste0("mouse ",Genotype),
                                                                         paste0("mouse ",Genotype," ",group_name)
                                                                         )
                                                                  )
                                                       )

sample_murine_published_human_affy_ATRT_plus[c("Organism","Genotype","group_name","rhabdoid tumors")]

# Top annotation
my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_pub <- HeatmapAnnotation(df = data.frame(sample_murine_published_human_affy_ATRT_plus["rhabdoid tumors"],check.names=FALSE), 
                                                                                   col = list(`rhabdoid tumors` = c("mouse R26 neural" = "brown",#"olivedrab1",
                                                                                                     "mouse R26 non_neural" = "cyan4",
                                                                                                     "mouse Snf5F/F/p53L/L/GFAP-Cre" = "violet",
                                                                                                     "mouse P0-CreC_Smarcb1FloxFlox" = "yellow",
                                                                                                     "human MYC" = "blue",
                                                                                                     "human TYR" = "red",
                                                                                                     "human SHH" = "green"
                                                                                                     )
                                                                                              ),
                                                                                   gap = unit(2,"mm"),
                                                                                   show_annotation_name = FALSE,
                                                                                   annotation_name_gp = my_annotation_name_gp,
                                                                                   annotation_legend_param = my_annotation_legend_param,
                                                                                   na_col = my_na_col
                                                                                   )

#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_human_affy_ATRT_pcaBatchCorr_pub.pdf"), width=24, height=10)
Heatmap(matrix                      = t(scale(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT_mcGenes))),
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        show_column_names              = TRUE,
             show_row_dend               = FALSE,
             #rect_gp                     = gpar(type = "none"),
             cluster_row                 = TRUE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT_pub,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
#draw(h, annotation_legend_side = "bottom")
#dev.off()

```



### Remove organism effect using linear regression (New)

```{r linReg_removeBatch}

# Subset matrix
head(matrix_murine_affy_RNAseq_published_human_affy_ATRT[,1:4])

sample_murine_humanATRT_affy_published$Sample_name == colnames(matrix_murine_affy_RNAseq_published_human_affy_ATRT)


test <- lm(t(matrix_murine_affy_RNAseq_published_human_affy_ATRT) ~ sample_murine_humanATRT_affy_published$Organism + sample_murine_humanATRT_affy_published$Data_type + sample_murine_humanATRT_affy_published$Organism*sample_murine_humanATRT_affy_published$Data_type)

matrix_ln_res <- t(test$residual)

plot(density(matrix_ln_res))

# PCA
PC        <- prcomp(data.frame(t(matrix_ln_res)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_humanATRT_affy_published
                        )

# PCA organism PC1 and PC2
#pdf(paste0(pathToFigures,"pca_murine_affy_RNAseq_published_human_affy_ATRT_PC1PC2.pdf"), width=14, height=12)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(fill = Organism),shape =23, size = 10) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = organism_col[c("Homo_sapiens","Mus_musculus")]) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
      legend.key.height = unit(1.5,"cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 23)))
#dev.off()

# Most variable genes
mvgenes_murine_affy_RNAseq_published_human_affy_ATRT_ln <- genes.selection(data = matrix_ln_res,
                                                                   thres.num = 1000
                                                                   )

# Subset matrix
matrix_murine_affy_RNAseq_published_human_affy_ATRT_ln_mvgenes <- matrix_ln_res[rownames(matrix_ln_res) %in% mvgenes_murine_affy_RNAseq_published_human_affy_ATRT_ln,] 

# Complex heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_published_pcaBatchCorr.pdf"), width=16, height=16)
h <- Heatmap(matrix                      = matrix_murine_affy_RNAseq_published_human_affy_ATRT_ln_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("murine affy and RNAseq data clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
             row_title                   = NULL,
             top_annotation              = my_top_annot_murine_affy_RNAseq_published_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 20),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp
             )
draw(h, annotation_legend_side = "bottom")
#draw(h)
#dev.off()

```


### Nestin murine RNAseq and human ATRT affy sample-to-sample correlation(New)


```{r murine_published_and_human_ATRT_affy}

# Sample Nestin RNAseq
sample_murine_RNAseq_Nestin <- filter(.data = sample_murine_RNAseq,
                                      Genotype == "Nestin"
                                      )

sample_murine_RNAseq_Nestin$Sample_name

sample_murine_RNAseq_Nestin$group_name <- c("Nestin_slow","Nestin_fast","Nestin_fast","Nestin_slow","Nestin_fast","Nestin_slow","Nestin_slow",NA,NA,NA,NA,NA,NA)

sample_murine_RNAseq_Nestin_ordered <- sample_murine_RNAseq_Nestin[order(sample_murine_RNAseq_Nestin$group_name),]

sample_murine_RNAseq_Nestin_ordered_plus <- filter(sample_murine_RNAseq_Nestin_ordered, !is.na(group_name))

# Subset Nestin RNAseq matrix
matrix_murine_RNAseq_Nestin <- matrix_murine_RNAseq_filtered_symbol[,c("symbol_unfiltered",
                                                                       sample_murine_RNAseq_Nestin_ordered_plus$Sample_name
                                                                       )
                                                                    ]

# Convert symbol to rownames
rownames(matrix_murine_RNAseq_Nestin) <- matrix_murine_RNAseq_Nestin$symbol_unfiltered
matrix_murine_RNAseq_Nestin$symbol_unfiltered <- NULL
head(matrix_murine_RNAseq_Nestin[,1:4])
nrow(matrix_murine_RNAseq_Nestin)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_affy_ATRT_ordered <- sample_human_affy_ATRT[order(sample_human_affy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_affy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineRNAseqNestin_humanAffyATRT <- rbind(sample_murine_RNAseq_Nestin_ordered_plus,
                                                 sample_human_affy_ATRT_ordered
                                                 )
nrow(sample_murineRNAseqNestin_humanAffyATRT)

# Merge matrix
matrix_murineRNAseqNestin_humanAffyATRT <- merge(x   = matrix_murine_RNAseq_Nestin,
                                                 y   = matrix_human_affy_ATRT_scaled,
                                                 by  = 0,
                                                 all = FALSE
                                                 )
rownames(matrix_murineRNAseqNestin_humanAffyATRT) <- matrix_murineRNAseqNestin_humanAffyATRT$Row.names
matrix_murineRNAseqNestin_humanAffyATRT$Row.names <- NULL

head(matrix_murineRNAseqNestin_humanAffyATRT[,1:4])
nrow(matrix_murineRNAseqNestin_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqNestin_humanAffyATRT$Sample_name == colnames(matrix_murineRNAseqNestin_humanAffyATRT)

# Quantile normalization
matrix_murineRNAseqNestin_humanAffyATRT <- quantile_normalisation(matrix_murineRNAseqNestin_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqNestin_humanAffyATRT <- cor(x = matrix_murineRNAseqNestin_humanAffyATRT,
                                                   method = "pearson"
                                                   )

# Bottom annotation
bottom_annot_murineRNAseqNestin_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqNestin_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqNestin_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqNestin_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqNestin_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqNestin_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqNestin_humanAffyATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqNestin_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqNestin_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqNestin_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqNestin_humanAffyATRT[-c(1,2,3,4,5,6,7),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )
# Bottom annotation
bottom_annot_murineRNAseqNestin_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqNestin_humanAffyATRT[1:3,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

bottom_annot_murineRNAseqNestin_humanAffyATRT_2 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqNestin_humanAffyATRT[4:7,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqNestin_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqNestin_humanAffyATRT[-c(1,2,3,4,5,6,7),1:3],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqNestin_humanAffyATRT_1,
             show_row_names    =FALSE,
             show_column_names =FALSE,
             column_title      = "Nestin fast",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqNestin_humanAffyATRT[-c(1,2,3,4,5,6,7),c("group_name")],
             gap               = unit(3, "mm")
             )
h2 <- Heatmap(matrix           = sampleDist_murineRNAseqNestin_humanAffyATRT[-c(1,2,3,4,5,6,7),4:7],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqNestin_humanAffyATRT_2,
             show_row_names    = FALSE,
             show_column_names = FALSE,
             column_title      = "Nestin slow",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             )
draw(h1 + h2 + row_annot_murineRNAseqNestin_humanAffyATRT_plus,
     show_heatmap_legend    = FALSE,
     show_annotation_legend = FALSE
     )
#dev.off()

```


### R26 primary murine tumour RNAseq and human ATRT affy sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_affy}

# Sample murine primary tumor RNAseq
sample_murine_RNAseq_R26_primTum <- filter(.data = sample_murine_RNAseq,
                                           Tumor_origin == "primary_tumor",
                                           Genotype == "R26"
                                           )

sample_murine_RNAseq_R26_primTum$Sample_name

sample_murine_RNAseq_R26_primTum$group_name <- c("neural","neural","non_neural","non_neural","non_neural","non_neural","non_neural","non_neural","neural","non_neural","non_neural")

sample_murine_RNAseq_R26_primTum_ordered <- sample_murine_RNAseq_R26_primTum[order(sample_murine_RNAseq_R26_primTum$group_name),]

# Subset murine R26 primary tumor RNAseq matrix
matrix_murine_RNAseq_R26_primTum <- matrix_murine_RNAseq_filtered_symbol[,c("symbol_unfiltered",
                                                                       sample_murine_RNAseq_R26_primTum_ordered$Sample_name
                                                                       )
                                                                    ]

# Convert symbol to rownames
rownames(matrix_murine_RNAseq_R26_primTum) <- matrix_murine_RNAseq_R26_primTum$symbol_unfiltered
matrix_murine_RNAseq_R26_primTum$symbol_unfiltered <- NULL
head(matrix_murine_RNAseq_R26_primTum[,1:4])
nrow(matrix_murine_RNAseq_R26_primTum)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_affy_ATRT_ordered <- sample_human_affy_ATRT[order(sample_human_affy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_affy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineRNAseqR26primTum_humanAffyATRT <- rbind(sample_murine_RNAseq_R26_primTum_ordered,
                                                 sample_human_affy_ATRT_ordered
                                                 )
nrow(sample_murineRNAseqR26primTum_humanAffyATRT)

# Merge matrix
matrix_murineRNAseqR26primTum_humanAffyATRT <- merge(x   = matrix_murine_RNAseq_R26_primTum,
                                                     y   = matrix_human_affy_ATRT_scaled,
                                                     by  = 0,
                                                 all = FALSE
                                                 )
rownames(matrix_murineRNAseqR26primTum_humanAffyATRT) <- matrix_murineRNAseqR26primTum_humanAffyATRT$Row.names
matrix_murineRNAseqR26primTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineRNAseqR26primTum_humanAffyATRT[,1:4])
nrow(matrix_murineRNAseqR26primTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqR26primTum_humanAffyATRT$Sample_name == colnames(matrix_murineRNAseqR26primTum_humanAffyATRT)

# Quantile normalization
matrix_murineRNAseqR26primTum_humanAffyATRT <- quantile_normalisation(matrix_murineRNAseqR26primTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqR26primTum_humanAffyATRT <- cor(x = matrix_murineRNAseqR26primTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

# Bottom annotation
bottom_annot_murineRNAseqR26primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqR26primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqR26primTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqR26primTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqR26primTum_humanAffyATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqR26primTum_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqR26primTum_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqR26primTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Bottom annotation
bottom_annot_murineRNAseqR26primTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[1:3,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

bottom_annot_murineRNAseqR26primTum_humanAffyATRT_2 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[4:11,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

bottom_annot_murineRNAseqR26primTum_humanAffyATRT_poster <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqR26primTum_humanAffyATRT[1:11,c("Genotype")]), 
                                                              col = list(Genotype = genotype_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqR26primTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),1:3],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqR26primTum_humanAffyATRT_1,
             show_row_names    =FALSE,
             show_column_names =FALSE,
             column_title      = "neuronal",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),c("group_name")],
             gap               = unit(3, "mm")
             )
h2 <- Heatmap(matrix           = sampleDist_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),4:11],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqR26primTum_humanAffyATRT_2,
             show_row_names    = FALSE,
             show_column_names = FALSE,
             column_title      = "non-neuronal",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             )
draw(h1 + h2 + row_annot_murineRNAseqR26primTum_humanAffyATRT_plus,
     show_heatmap_legend    = FALSE,
     show_annotation_legend = FALSE
     )
#dev.off()

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqR26primTum_humanAffyATRT_poster.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),1:11],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqR26primTum_humanAffyATRT_poster,
             show_row_names    =FALSE,
             show_column_names =FALSE,
             column_title      = "R26",
             column_title_gp   = gpar(fontsize = 80, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 80, fontface = "bold"),
             split             = sample_murineRNAseqR26primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11),c("group_name")],
             gap               = unit(3, "mm"),
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                #grid_width     = unit(100,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n Pearson's coeff.\n",
                                                title_gp       = gpar(fontsize = 60),
                                                title_position = "topcenter",
                                                legend_direction = "horizontal"
                                                )
             )
draw(h1 + row_annot_murineRNAseqR26primTum_humanAffyATRT_plus,
     show_heatmap_legend    = TRUE,
     show_annotation_legend = FALSE,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Prss56 primary murine tumour RNAseq and human ATRT affy sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_affy}

# Sample murine primary tumor RNAseq
sample_murine_RNAseq_Prss56_primTum <- filter(.data         = sample_murine_RNAseq,
                                              Tumor_origin == "primary_tumor",
                                              Genotype     == "Prss56"
                                              )

sample_murine_RNAseq_Prss56_primTum$Sample_name

sample_murine_RNAseq_Prss56_primTum$group_name <- sample_murine_RNAseq_Prss56_primTum$Genotype

# Subset murine Prss56 primary tumor RNAseq matrix
matrix_murine_RNAseq_Prss56_primTum <- matrix_murine_RNAseq_filtered_symbol[,c("symbol_unfiltered",
                                                                       sample_murine_RNAseq_Prss56_primTum$Sample_name
                                                                       )
                                                                    ]

# Convert symbol to rownames
rownames(matrix_murine_RNAseq_Prss56_primTum) <- matrix_murine_RNAseq_Prss56_primTum$symbol_unfiltered
matrix_murine_RNAseq_Prss56_primTum$symbol_unfiltered <- NULL
head(matrix_murine_RNAseq_Prss56_primTum[,1:4])
nrow(matrix_murine_RNAseq_Prss56_primTum)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_affy_ATRT_ordered <- sample_human_affy_ATRT[order(sample_human_affy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_affy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineRNAseqPrss56primTum_humanAffyATRT <- rbind(sample_murine_RNAseq_Prss56_primTum,
                                                        sample_human_affy_ATRT_ordered
                                                        )
nrow(sample_murineRNAseqPrss56primTum_humanAffyATRT)

levels(factor(sample_murineRNAseqPrss56primTum_humanAffyATRT$sub_sub_localization_of_primary_tumor))

# Merge matrix
matrix_murineRNAseqPrss56primTum_humanAffyATRT <- merge(x   = matrix_murine_RNAseq_Prss56_primTum,
                                                        y   = matrix_human_affy_ATRT_scaled,
                                                        by  = 0,
                                                        all = FALSE
                                                        )
rownames(matrix_murineRNAseqPrss56primTum_humanAffyATRT) <- matrix_murineRNAseqPrss56primTum_humanAffyATRT$Row.names
matrix_murineRNAseqPrss56primTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineRNAseqPrss56primTum_humanAffyATRT[,1:4])
nrow(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqPrss56primTum_humanAffyATRT$Sample_name == colnames(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Quantile normalization
matrix_murineRNAseqPrss56primTum_humanAffyATRT <- quantile_normalisation(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqPrss56primTum_humanAffyATRT <- cor(x = matrix_murineRNAseqPrss56primTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

# Bottom annotation
bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqPrss56primTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqPrss56primTum_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqPrss56primTum_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Bottom annotation
bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),1:9],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             #bottom_annotation = bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT_1,
             show_row_names    = FALSE,
             show_column_names = TRUE,
             column_title      = "Prss56",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")],
             gap               = unit(3, "mm")
             )
draw(h1 + row_annot_murineRNAseqPrss56primTum_humanAffyATRT_plus,
     show_heatmap_legend    = FALSE,
     show_annotation_legend = FALSE
     )
#dev.off()
 
```

### Prss56 primary murine tumour RNAseq and human ATRT affy Curie sample-to-sample correlation (New - anatomic location)

```{r murine_published_and_human_ATRT_affy}

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_affy_ATRT_Curie <- filter(.data = sample_human_affy_ATRT,
                                       Data_source == "Curie"
                                       )

sample_human_affy_ATRT_Curie_ordered <- sample_human_affy_ATRT_Curie[order(sample_human_affy_ATRT_Curie$sub_sub_localization_of_primary_tumor),]

levels(factor(sample_human_affy_ATRT_Curie_ordered$sub_sub_localization_of_primary_tumor))

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_affy_ATRT_Curie_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineRNAseqPrss56primTum_humanAffyATRT <- rbind(sample_murine_RNAseq_Prss56_primTum,
                                                        sample_human_affy_ATRT_Curie_ordered
                                                        )
nrow(sample_murineRNAseqPrss56primTum_humanAffyATRT)

levels(factor(sample_murineRNAseqPrss56primTum_humanAffyATRT$sub_sub_localization_of_primary_tumor))

# Merge matrix
matrix_murineRNAseqPrss56primTum_humanAffyATRT <- merge(x   = matrix_murine_RNAseq_Prss56_primTum,
                                                        y   = matrix_human_affy_ATRT_scaled,
                                                        by  = 0,
                                                        all = FALSE
                                                        )
rownames(matrix_murineRNAseqPrss56primTum_humanAffyATRT) <- matrix_murineRNAseqPrss56primTum_humanAffyATRT$Row.names
matrix_murineRNAseqPrss56primTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineRNAseqPrss56primTum_humanAffyATRT[,1:4])
nrow(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqPrss56primTum_humanAffyATRT$Sample_name == colnames(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Quantile normalization
matrix_murineRNAseqPrss56primTum_humanAffyATRT <- quantile_normalisation(matrix_murineRNAseqPrss56primTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqPrss56primTum_humanAffyATRT <- cor(x = matrix_murineRNAseqPrss56primTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

# Bottom annotation
bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("sub_sub_localization_of_primary_tumor")]), 
                                                              col = list(sub_sub_localization_of_primary_tumor = sub_sub_localization_of_primary_tumor_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("sub_sub_localization_of_primary_tumor")]), 
                                                                col = list(sub_sub_localization_of_primary_tumor = sub_sub_localization_of_primary_tumor_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE,
width = unit(20,"mm")
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqPrss56primTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqPrss56primTum_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqPrss56primTum_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),c("sub_sub_localization_of_primary_tumor")]), 
                                                                col = list(sub_sub_localization_of_primary_tumor = sub_sub_localization_of_primary_tumor_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                #annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = TRUE,
                                                                width = unit(20,"mm")
                                                                )

# Bottom annotation
bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanAffyATRT[,c("sub_sub_localization_of_primary_tumor")]), 
                                                              col = list(sub_sub_localization_of_primary_tumor = sub_sub_localization_of_primary_tumor_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),1:9],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             #bottom_annotation = bottom_annot_murineRNAseqPrss56primTum_humanAffyATRT_1,
             show_row_names    = FALSE,
             show_column_names = TRUE,
             column_title      = "Prss56",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqPrss56primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")],
             gap               = unit(3, "mm")
             )
draw(h1 + row_annot_murineRNAseqPrss56primTum_humanAffyATRT_plus,
     show_heatmap_legend    = TRUE,
     show_annotation_legend = TRUE
     )
#dev.off()
 
```

### Prss56 primary murine tumour RNAseq and human ATRT (primary tumors) RNAseq sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_RNAseq}

# Sample murine primary tumor RNAseq
sample_murine_RNAseq_Prss56_primTum <- filter(.data         = sample_murine_RNAseq,
                                              Tumor_origin == "primary_tumor",
                                              Genotype     == "Prss56"
                                              )

sample_murine_RNAseq_Prss56_primTum$Sample_name

sample_murine_RNAseq_Prss56_primTum$group_name <- sample_murine_RNAseq_Prss56_primTum$Genotype

# Subset murine Prss56 primary tumor RNAseq matrix
matrix_murine_RNAseq_Prss56_primTum <- matrix_murine_RNAseq_filtered_symbol[,c("symbol_unfiltered",
                                                                       sample_murine_RNAseq_Prss56_primTum$Sample_name
                                                                       )
                                                                    ]

# Convert symbol to rownames
rownames(matrix_murine_RNAseq_Prss56_primTum) <- matrix_murine_RNAseq_Prss56_primTum$symbol_unfiltered
matrix_murine_RNAseq_Prss56_primTum$symbol_unfiltered <- NULL
head(matrix_murine_RNAseq_Prss56_primTum[,1:4])
nrow(matrix_murine_RNAseq_Prss56_primTum)

# Load human ATRT RNAseq data
load(file = "~/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/script/human_ATRT_primTum_Curie_RNAseq_list.RData")

str(human_ATRT_primTum_Curie_RNAseq_list)

matrix_human_ATRT_RNAseq <- human_ATRT_primTum_Curie_RNAseq_list$matrix
sample_human_ATRT_RNAseq <- mutate(human_ATRT_primTum_Curie_RNAseq_list$sampleTable,
                                   group_name = RNAseq_subgroup
                                   ) %>%
    dplyr::select(-RNAseq_subgroup)

# Convert human gene symbol to orthologous mouse gene symbol
matrix_human_ATRT_RNAseq_mouseSymbol <- merge(x    = symbols_conversion,
                                              y    = matrix_human_ATRT_RNAseq,
                                              by.x = "HGNC.symbol",
                                              by.y = 0,
                                              all  = FALSE
                                              ) %>% dplyr::select(-HGNC.symbol)
head(matrix_human_ATRT_RNAseq_mouseSymbol[,1:4])
nrow(matrix_human_ATRT_RNAseq_mouseSymbol)

# Convert gene symbol to rownames
matrix_human_ATRT_RNAseq_mouseSymbol <- matrix_human_ATRT_RNAseq_mouseSymbol[!duplicated(matrix_human_ATRT_RNAseq_mouseSymbol$MGI.symbol),]
rownames(matrix_human_ATRT_RNAseq_mouseSymbol) <- matrix_human_ATRT_RNAseq_mouseSymbol$MGI.symbol

# Order human sample plan and matrix according to tumor subgroup
sample_human_ATRT_RNAseq_ordered <- sample_human_ATRT_RNAseq[order(sample_human_ATRT_RNAseq$group_name),]
matrix_human_ATRT_RNAseq <- matrix_human_ATRT_RNAseq_mouseSymbol[,sample_human_ATRT_RNAseq_ordered$Sample_name]

head(matrix_human_ATRT_RNAseq[,1:4])
nrow(matrix_human_ATRT_RNAseq)

# Scale human affy ATRT matrix
matrix_human_ATRT_RNAseq_scaled <- scale(x = matrix_human_ATRT_RNAseq,
                                         center = TRUE,
                                         scale = TRUE
                                         )
plot(density(matrix_human_ATRT_RNAseq_scaled))

# Merge samples
sample_murineRNAseqPrss56primTum_humanRNAseqATRT <- rbind(sample_murine_RNAseq_Prss56_primTum,
                                                          sample_human_ATRT_RNAseq_ordered
                                                          )
nrow(sample_murineRNAseqPrss56primTum_humanRNAseqATRT)

levels(factor(sample_murineRNAseqPrss56primTum_humanRNAseqATRT$sub_sub_localization_of_primary_tumor))

# Merge matrix
matrix_murineRNAseqPrss56primTum_humanRNAseqATRT <- merge(x   = matrix_murine_RNAseq_Prss56_primTum,
                                                          y   = matrix_human_ATRT_RNAseq_scaled,
                                                          by  = 0,
                                                          all = FALSE
                                                          )
rownames(matrix_murineRNAseqPrss56primTum_humanRNAseqATRT) <- matrix_murineRNAseqPrss56primTum_humanRNAseqATRT$Row.names
matrix_murineRNAseqPrss56primTum_humanRNAseqATRT$Row.names <- NULL

head(matrix_murineRNAseqPrss56primTum_humanRNAseqATRT[,1:4])
nrow(matrix_murineRNAseqPrss56primTum_humanRNAseqATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqPrss56primTum_humanRNAseqATRT$Sample_name == colnames(matrix_murineRNAseqPrss56primTum_humanRNAseqATRT)

# Quantile normalization
matrix_murineRNAseqPrss56primTum_humanRNAseqATRT <- quantile_normalisation(matrix_murineRNAseqPrss56primTum_humanRNAseqATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqPrss56primTum_humanRNAseqATRT <- cor(x = matrix_murineRNAseqPrss56primTum_humanRNAseqATRT,
                                                            method = "pearson"
                                                            )

# Bottom annotation
bottom_annot_murineRNAseqPrss56primTum_humanRNAseqATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanRNAseqATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanRNAseqATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanRNAseqATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqPrss56primTum_humanRNAseqATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqPrss56primTum_humanRNAseqATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqPrss56primTum_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),1:9],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             show_row_names    = FALSE,
             show_column_names = TRUE,
             column_title      = "Prss56",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")],
             gap               = unit(3, "mm")
             )
draw(h1 + row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT_plus,
     show_heatmap_legend    = FALSE,
     show_annotation_legend = FALSE
     )
#dev.off()

icicicicicicicicicicicicicicicici

# Row annotation
row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT_subsubLoc <- HeatmapAnnotation(df = mutate(data.frame(sample_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),c("sub_sub_localization_of_primary_tumor")]), anatomic_location = sub_sub_localization_of_primary_tumor) %>% dplyr::select(-sub_sub_localization_of_primary_tumor),
                                                                col = list(anatomic_location = sub_sub_localization_of_primary_tumor_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = TRUE,
                                                                width = unit(20,"mm")
                                                                )

# Sample-to-sample heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqPrss56primTum_humanAffyATRT_subsubLoc.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),1:9],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             show_row_names    = FALSE,
             show_column_names = TRUE,
             column_title      = "Prss56",
             column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 30, fontface = "bold"),
             split             = sample_murineRNAseqPrss56primTum_humanRNAseqATRT[-c(1,2,3,4,5,6,7,8,9),c("group_name")],
             gap               = unit(3, "mm"),
             show_heatmap_legend    = FALSE
             )
draw(h1 + row_annot_murineRNAseqPrss56primTum_humanRNAseqATRT_subsubLoc,
     #show_heatmap_legend    = FALSE,
     show_annotation_legend = TRUE
     )
#dev.off()

```

icicicicicicicici


### P0 primary murine tumour RNAseq and human ATRT affy sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_affy}

# Sample murine primary tumor RNAseq
sample_murine_RNAseq_P0_primTum <- filter(.data = sample_murine_RNAseq,
                                      Tumor_origin == "primary_tumor",
                                      Genotype == "P0-CreC_Smarcb1FloxFlox"
                                      )

sample_murine_RNAseq_P0_primTum$group_name <- sample_murine_RNAseq_P0_primTum$Genotype

# Subset murine P0 primary tumor RNAseq matrix
matrix_murine_RNAseq_P0_primTum <- matrix_murine_RNAseq_filtered_symbol[,c("symbol_unfiltered",
                                                                       sample_murine_RNAseq_P0_primTum$Sample_name
                                                                       )
                                                                    ]

# Convert symbol to rownames
rownames(matrix_murine_RNAseq_P0_primTum) <- matrix_murine_RNAseq_P0_primTum$symbol_unfiltered
matrix_murine_RNAseq_P0_primTum$symbol_unfiltered <- NULL
head(matrix_murine_RNAseq_P0_primTum[,1:4])
nrow(matrix_murine_RNAseq_P0_primTum)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_DKFZaffy_ATRT_ordered <- sample_human_DKFZaffy_ATRT[order(sample_human_DKFZaffy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_DKFZaffy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineRNAseqP0primTum_humanAffyATRT <- rbind(sample_murine_RNAseq_P0_primTum,
                                                        sample_human_DKFZaffy_ATRT_ordered
                                                        )
nrow(sample_murineRNAseqP0primTum_humanAffyATRT)

# Merge matrix
matrix_murineRNAseqP0primTum_humanAffyATRT <- merge(x   = matrix_murine_RNAseq_P0_primTum,
                                                        y   = matrix_human_affy_ATRT_scaled,
                                                        by  = 0,
                                                        all = FALSE
                                                        )
rownames(matrix_murineRNAseqP0primTum_humanAffyATRT) <- matrix_murineRNAseqP0primTum_humanAffyATRT$Row.names
matrix_murineRNAseqP0primTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineRNAseqP0primTum_humanAffyATRT[,1:4])
nrow(matrix_murineRNAseqP0primTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineRNAseqP0primTum_humanAffyATRT$Sample_name == colnames(matrix_murineRNAseqP0primTum_humanAffyATRT)

# Quantile normalization
matrix_murineRNAseqP0primTum_humanAffyATRT <- quantile_normalisation(matrix_murineRNAseqP0primTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineRNAseqP0primTum_humanAffyATRT <- cor(x = matrix_murineRNAseqP0primTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

# Bottom annotation
bottom_annot_murineRNAseqP0primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqP0primTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineRNAseqP0primTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqP0primTum_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqP0primTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineRNAseqP0primTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqP0primTum_humanAffyATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqP0primTum_humanAffyATRT$group_name)
             )
h + row_annot_murineRNAseqP0primTum_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineRNAseqP0primTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqP0primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11,12),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Bottom annotation
bottom_annot_murineRNAseqP0primTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(sample_murineRNAseqP0primTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqP0primTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineRNAseqP0primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11,12),1:12],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineRNAseqP0primTum_humanAffyATRT_1,
             show_row_names    = FALSE,
             show_column_names = FALSE,
             column_title      = "P0",
             column_title_gp   = gpar(fontsize = 80, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 80, fontface = "bold"),
             split             = sample_murineRNAseqP0primTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8,9,10,11,12),c("group_name")],
             gap               = unit(3, "mm"),
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                #grid_width     = unit(100,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n Pearson's coeff.\n",
                                                title_gp       = gpar(fontsize = 60),
                                                title_position = "topcenter",
                                                legend_direction = "horizontal",
                                                labels_gp = gpar(fontsize = 30)
                                                )
             )
draw(h1 + row_annot_murineRNAseqP0primTum_humanAffyATRT_plus,
     show_heatmap_legend    = TRUE,
     show_annotation_legend = FALSE,
     heatmap_legend_side = "bottom"
     )
dev.off()
 
```

### Gfap primary murine tumour affy and human ATRT affy sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_affy}

# Sample murine primary tumor affy
samples_murine_affy_Gfap_primTum <- filter(.data = samples_murine_affy,
                                           Tumor_origin == "primary_tumor",
                                           Genotype == "Snf5F/F/p53L/L/GFAP-Cre"
                                           )

samples_murine_affy_Gfap_primTum$Sample_name

samples_murine_affy_Gfap_primTum$group_name <- samples_murine_affy_Gfap_primTum$Genotype

# Subset murine Gfap primary tumor affy matrix
matrix_murine_affy_Gfap_primTum <- matrix_murine_affy[,as.vector(samples_murine_affy_Gfap_primTum$Sample_name)]

head(matrix_murine_affy_Gfap_primTum[,1:4])
nrow(matrix_murine_affy_Gfap_primTum)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

xxxxxxxxxxxxxxxxxxxxxxxxxx
sample_human_DKFZaffy_ATRT_ordered <- sample_human_DKFZaffy_ATRT[order(sample_human_DKFZaffy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_DKFZaffy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineAffyGfapPrimTum_humanAffyATRT <- rbind(samples_murine_affy_Gfap_primTum,
                                                    sample_human_DKFZaffy_ATRT_ordered
                                                    )
nrow(sample_murineAffyGfapPrimTum_humanAffyATRT)

# Merge matrix
matrix_murineAffyGfapPrimTum_humanAffyATRT <- merge(x   = matrix_murine_affy_Gfap_primTum,
                                                    y   = matrix_human_affy_ATRT_scaled,
                                                    by  = 0,
                                                    all = FALSE
                                                    )
rownames(matrix_murineAffyGfapPrimTum_humanAffyATRT) <- matrix_murineAffyGfapPrimTum_humanAffyATRT$Row.names
matrix_murineAffyGfapPrimTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineAffyGfapPrimTum_humanAffyATRT[,1:4])
nrow(matrix_murineAffyGfapPrimTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineAffyGfapPrimTum_humanAffyATRT$Sample_name == colnames(matrix_murineAffyGfapPrimTum_humanAffyATRT)

# Quantile normalization
matrix_murineAffyGfapPrimTum_humanAffyATRT <- quantile_normalisation(matrix_murineAffyGfapPrimTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineAffyGfapPrimTum_humanAffyATRT <- cor(x = matrix_murineAffyGfapPrimTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

data.frame(sample_murineAffyGfapPrimTum_humanAffyATRT[,c("group_name")])

# Bottom annotation
bottom_annot_murineAffyGfapPrimTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyGfapPrimTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineAffyGfapPrimTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyGfapPrimTum_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
pdf(paste0(pathToFigures,"complexHeatmap_corr_murineAffyGfapPrimTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineAffyGfapPrimTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineAffyGfapPrimTum_humanAffyATRT,
             show_row_names    =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineAffyGfapPrimTum_humanAffyATRT$group_name)
             )
h + row_annot_murineAffyGfapPrimTum_humanAffyATRT
dev.off()

# Row annotation
row_annot_murineAffyGfapPrimTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyGfapPrimTum_humanAffyATRT[-c(1,2,3,4,5),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Bottom annotation
bottom_annot_murineAffyGfapPrimTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyGfapPrimTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
pdf(paste0(pathToFigures,"complexHeatmap_corr_murineAffyGfapPrimTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineAffyGfapPrimTum_humanAffyATRT[-c(1,2,3,4,5),1:5],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineAffyGfapPrimTum_humanAffyATRT_1,
             show_row_names    = FALSE,
             show_column_names = FALSE,
             column_title      = "Gfap",
             column_title_gp   = gpar(fontsize = 80, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 80, fontface = "bold"),
             split             = sample_murineAffyGfapPrimTum_humanAffyATRT[-c(1,2,3,4,5),c("group_name")],
             gap               = unit(3, "mm"),
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                #grid_width     = unit(100,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n Pearson's coeff.\n",
                                                title_gp       = gpar(fontsize = 60),
                                                title_position = "topcenter",
                                                legend_direction = "horizontal",
                                                labels_gp = gpar(fontsize = 30)
                                                )
             )
draw(h1 + row_annot_murineAffyGfapPrimTum_humanAffyATRT_plus,
     show_heatmap_legend    = TRUE,
     show_annotation_legend = FALSE,
     heatmap_legend_side = "bottom"
     )
dev.off()
 
```

### R26 primary murine tumour affy and human ATRT affy sample-to-sample correlation(New)

```{r murine_published_and_human_ATRT_affy}

# Sample murine primary tumor affy
samples_murine_affy_R26_primTum <- filter(.data = samples_murine_affy,
                                          Tumor_origin == "primary_tumor",
                                          Genotype == "R26",
                                          Tumor_type == "RT"
                                          )

samples_murine_affy_R26_primTum$Sample_name

samples_murine_affy_R26_primTum <- samples_murine_affy_R26_primTum[order(samples_murine_affy_R26_primTum$group_name),]

samples_murine_affy_R26_primTum$group_name <- samples_murine_affy_R26_primTum$Genotype

# Subset murine R26 primary tumor affy matrix
matrix_murine_affy_R26_primTum <- matrix_murine_affy[,as.vector(samples_murine_affy_R26_primTum$Sample_name)]

head(matrix_murine_affy_R26_primTum[,1:4])
nrow(matrix_murine_affy_R26_primTum)

# Subset human affy ATRT matrix
head(matrix_human_affy_mouseSymbol[,1:4])
nrow(matrix_human_affy_mouseSymbol)

sample_human_DKFZaffy_ATRT_ordered <- sample_human_DKFZaffy_ATRT[order(sample_human_DKFZaffy_ATRT$group_name),]

matrix_human_affy_ATRT <- matrix_human_affy_mouseSymbol[,sample_human_DKFZaffy_ATRT_ordered$Sample_name]

head(matrix_human_affy_ATRT[,1:4])
nrow(matrix_human_affy_ATRT)

# Scale human affy ATRT matrix
matrix_human_affy_ATRT_scaled <- scale(x = matrix_human_affy_ATRT,
                                       center = TRUE,
                                       scale = TRUE
                                       )
plot(density(matrix_human_affy_ATRT_scaled))

# Merge samples
sample_murineAffyR26PrimTum_humanAffyATRT <- rbind(samples_murine_affy_R26_primTum,
                                                    sample_human_DKFZaffy_ATRT_ordered
                                                    )
nrow(sample_murineAffyR26PrimTum_humanAffyATRT)

# Merge matrix
matrix_murineAffyR26PrimTum_humanAffyATRT <- merge(x   = matrix_murine_affy_R26_primTum,
                                                    y   = matrix_human_affy_ATRT_scaled,
                                                    by  = 0,
                                                    all = FALSE
                                                    )
rownames(matrix_murineAffyR26PrimTum_humanAffyATRT) <- matrix_murineAffyR26PrimTum_humanAffyATRT$Row.names
matrix_murineAffyR26PrimTum_humanAffyATRT$Row.names <- NULL

head(matrix_murineAffyR26PrimTum_humanAffyATRT[,1:4])
nrow(matrix_murineAffyR26PrimTum_humanAffyATRT)

# Check if sample annotation and matrix match
sample_murineAffyR26PrimTum_humanAffyATRT$Sample_name == colnames(matrix_murineAffyR26PrimTum_humanAffyATRT)

# Quantile normalization
matrix_murineAffyR26PrimTum_humanAffyATRT <- quantile_normalisation(matrix_murineAffyR26PrimTum_humanAffyATRT)

# Sample-to-sample correlation matrix
sampleDist_murineAffyR26PrimTum_humanAffyATRT <- cor(x = matrix_murineAffyR26PrimTum_humanAffyATRT,
                                                   method = "pearson"
                                                   )

data.frame(sample_murineAffyR26PrimTum_humanAffyATRT[,c("group_name")])

# Bottom annotation
bottom_annot_murineAffyR26PrimTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyR26PrimTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murineAffyR26PrimTum_humanAffyATRT <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyR26PrimTum_humanAffyATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineAffyR26PrimTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murineAffyR26PrimTum_humanAffyATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murineAffyR26PrimTum_humanAffyATRT,
             show_row_names    =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineAffyR26PrimTum_humanAffyATRT$group_name)
             )
h + row_annot_murineAffyR26PrimTum_humanAffyATRT
#dev.off()

# Row annotation
row_annot_murineAffyR26PrimTum_humanAffyATRT_plus <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyR26PrimTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Bottom annotation
bottom_annot_murineAffyR26PrimTum_humanAffyATRT_1 <- HeatmapAnnotation(df = data.frame(group_name = sample_murineAffyR26PrimTum_humanAffyATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Sample-to-sample heatmap
pdf(paste0(pathToFigures,"complexHeatmap_corr_murineAffyR26PrimTum_humanAffyATRT_2.pdf"), width=16, height=16)
h1 <- Heatmap(matrix           = sampleDist_murineAffyR26PrimTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8),1:8],
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murineAffyR26PrimTum_humanAffyATRT_1,
             show_row_names    = FALSE,
             show_column_names = FALSE,
             column_title      = "R26",
             column_title_gp   = gpar(fontsize = 80, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 80, fontface = "bold"),
             split             = sample_murineAffyR26PrimTum_humanAffyATRT[-c(1,2,3,4,5,6,7,8),c("group_name")],
             gap               = unit(3, "mm"),
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                #grid_width     = unit(100,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n Pearson's coeff.\n",
                                                title_gp       = gpar(fontsize = 60),
                                                title_position = "topcenter",
                                                legend_direction = "horizontal",
labels_gp = gpar(fontsize = 30)
                                                ),
             )
draw(h1 + row_annot_murineAffyR26PrimTum_humanAffyATRT_plus,
     show_heatmap_legend    = TRUE,
     show_annotation_legend = FALSE,
     heatmap_legend_side    = "bottom"
     )
dev.off()
 
```


### murine RT Curie RNAseq and human ATRT affy sample-to-sample correlation(New)

icicicicici

```{r murine_Curie_RNAseq_and_human_ATRT_affy}

# Load Curie RNAseq rld matrix and sample plan (cf murine_RT_RNAseq.Rmd script)
load("murine_RNAseq_R26_Nestin_Prss56.RData")

# Extract rld matrix
matrix_RNAseq_R26_Nestin_Prss56 <- murine_RNAseq_R26_Nestin_Prss56$matrix
head(matrix_RNAseq_R26_Nestin_Prss56[,1:4])

# Extract murine sample plan
sample_RNAseq_R26_Nestin_Prss56 <- murine_RNAseq_R26_Nestin_Prss56$sampleTable
sample_RNAseq_R26_Nestin_Prss56$group_name

# Convert murine ENSGID into gene symbol
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

ENSMGI_symbol <- getLDS(attributes = c("ensembl_gene_id"),
                             #filters    = "mgi_symbol",
                             #values     = my_matrix_human$symbol_unfiltered,
                        mart       = mouse,
                        attributesL= "mgi_symbol",
                        martL      = mouse,
                        uniqueRows = TRUE
                        )

colnames(ENSMGI_symbol) <- c("ENSMGI","symbol")

head(ENSMGI_symbol)

library(gsubfn)

matrix_RNAseq_R26_Nestin_Prss56_plus <- matrix_RNAseq_R26_Nestin_Prss56

matrix_RNAseq_R26_Nestin_Prss56_plus <- data.frame(matrix_RNAseq_R26_Nestin_Prss56_plus, check.names=FALSE)

matrix_RNAseq_R26_Nestin_Prss56_plus$ENSMGI <- rownames(matrix_RNAseq_R26_Nestin_Prss56_plus)

matrix_RNAseq_R26_Nestin_Prss56_plus$ENSMGI <- strapplyc(X        = matrix_RNAseq_R26_Nestin_Prss56_plus$ENSMGI,
                                                         pattern  = "(.*)\\..+",
                                                         simplify = TRUE
                                                         )

head(matrix_RNAseq_R26_Nestin_Prss56_plus$ENSMGI)

# Add gene symbol
matrix_RNAseq_R26_Nestin_Prss56_Plus <- merge(x   = ENSMGI_symbol,
                                              y   = matrix_RNAseq_R26_Nestin_Prss56_plus,
                                              by  = "ENSMGI",
                                              all = FALSE
                                              )

head(matrix_RNAseq_R26_Nestin_Prss56_Plus[,1:4])

# Remove duplicated rownames
matrix_RNAseq_R26_Nestin_Prss56_Plus <- matrix_RNAseq_R26_Nestin_Prss56_Plus[!duplicated(matrix_RNAseq_R26_Nestin_Prss56_Plus$symbol),]

# Convert symbol to rownames
rownames(matrix_RNAseq_R26_Nestin_Prss56_Plus) <- matrix_RNAseq_R26_Nestin_Prss56_Plus$symbol
matrix_RNAseq_R26_Nestin_Prss56_Plus$ENSMGI <- NULL
matrix_RNAseq_R26_Nestin_Prss56_Plus$symbol <- NULL
head(matrix_RNAseq_R26_Nestin_Prss56_Plus[,1:4])

# Order murine sample plan according to group name
sample_RNAseq_R26_Nestin_Prss56_ord <- sample_RNAseq_R26_Nestin_Prss56[order(sample_RNAseq_R26_Nestin_Prss56$group_name),]

sample_RNAseq_R26_Nestin_Prss56_ord$group_name

# Order murine matrix to follow sample plan order
sample_RNAseq_R26_Nestin_Prss56_ord$Sample_name == colnames(matrix_RNAseq_R26_Nestin_Prss56)

matrix_RNAseq_R26_Nestin_Prss56_PLUS <- matrix_RNAseq_R26_Nestin_Prss56_Plus[,sample_RNAseq_R26_Nestin_Prss56_ord$Sample_name]

sample_RNAseq_R26_Nestin_Prss56_ord$Sample_name == colnames(matrix_RNAseq_R26_Nestin_Prss56_PLUS)

# Plot density of murine matrix
plot(density(as.matrix(matrix_RNAseq_R26_Nestin_Prss56_PLUS)))

# Load human affy data (DKFZ + Curie)
pathToHumanATRT_script <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/script/"

load(file = paste0(pathToHumanATRT_script,"human_ATRT_DKFZ_Curie_affy_list.RData"))

# Extract human affy matrix
matrix_human_ATRT_affy <- human_ATRT_DKFZ_Curie_affy_list$matrix
head(matrix_human_ATRT_affy[,1:4])

# Extract human affy sample plan
sample_human_ATRT_affy <- human_ATRT_DKFZ_Curie_affy_list$sampleTable

# Remove some columns of human affy sample plan
colnames(sample_human_ATRT_affy)

sample_human_ATRT_affy$DKFZ_subgroup <- NULL
sample_human_ATRT_affy$DKFZ_450K_subroup <- NULL
sample_human_ATRT_affy$DKFZ_affy_subgroup <- NULL
sample_human_ATRT_affy$DKFZ_WGBS_subgroup <- NULL
sample_human_ATRT_affy$DKFZ_H3K27Ac_subgroup <- NULL

ncol(sample_human_ATRT_affy)

# Order human sample plan according to group name
sample_human_ATRT_affy$group_name
sample_human_ATRT_affy_ord <- sample_human_ATRT_affy[order(sample_human_ATRT_affy$group_name),]
sample_human_ATRT_affy_ord$group_name

# Convert human gene symbol to murine gene symbol
head(symbols_conversion)

matrix_human_ATRT_affy_plus <- merge(x = symbols_conversion,
                                     y = matrix_human_ATRT_affy,
                                     by.x = "HGNC.symbol",
                                     by.y = "symbol",
                                     all = FALSE
                                     )
matrix_human_ATRT_affy_plus$HGNC.symbol <- NULL
head(matrix_human_ATRT_affy_plus[,1:4])

# Plot density of human matrix
plot(density(as.matrix(matrix_human_ATRT_affy_plus)))

# Remove duplicated symbol
matrix_human_ATRT_affy_plus <- matrix_human_ATRT_affy_plus[!duplicated(matrix_human_ATRT_affy_plus$MGI.symbol),]

# Put gene symbol as rownames
rownames(matrix_human_ATRT_affy_plus) <- matrix_human_ATRT_affy_plus$MGI.symbol
matrix_human_ATRT_affy_plus$MGI.symbol <- NULL
head(matrix_human_ATRT_affy_plus[,1:4])

# Reorder human matrix to follow sample plan order
matrix_human_ATRT_affy_Plus <- matrix_human_ATRT_affy_plus[,sample_human_ATRT_affy_ord$Sample_name]

# Merge murine and human sample plan
sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- rbind(sample_RNAseq_R26_Nestin_Prss56_ord,
                                                                sample_human_ATRT_affy_ord
                                                                )
nrow(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)

# Merge murine and human matrix
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- merge(x   = matrix_RNAseq_R26_Nestin_Prss56_PLUS,
                                                                y   = matrix_human_ATRT_affy_Plus,
                                                                by  = 0,
                                                                all = FALSE
                                                                )
rownames(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT) <- matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Row.names
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Row.names <- NULL
head(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[,1:4])

# Plot density of combined human and murine matrix
plot(density(as.matrix(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)))

# Check if sample annotation and matrix match
sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Sample_name == colnames(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)

# Quantile normalization
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn <- quantile_normalisation(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)

# Sample-to-sample correlation matrix
sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- cor(x = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
                                                                  method = "pearson"
                                                                  )

sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn <- cor(x = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn,
                                                                  method = "pearson"
                                                                  )

# Check order
colnames(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT) == sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Sample_name

colnames(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn) == sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Sample_name

# Density plot

plot(density(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),1:30]))

plot(density(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30]))

plot(density(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30])))

plot(density(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30])/4))

plot(density(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30])/max(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn))))

plot(density(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30])/abs(min(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn)))))

plot(density(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30])/(max(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn))-min(scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn)))))

# Bottom annotation
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col
                                                              )

# Row annotation
row_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[,c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE
                                                                )

# Sample-to-sample Heatmap
#pdf(paste0(pathToFigures,"complexHeatmap_corr_murineRNAseqR26primTum_humanAffyATRT.pdf"), width=16, height=12)
h <- Heatmap(matrix          = sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             cluster_rows    = FALSE,
             cluster_columns = FALSE,
             bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             show_row_names =FALSE,
             show_column_names =FALSE#,
             #split = splitVect[order(match(splitVect,splitOrder))]#,
             #row_order = list(sample_murineRNAseqR26primTum_humanAffyATRT$group_name)
             )
h + row_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT
#dev.off()

# Row annotation
row_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")]), 
                                                                col = list(group_name = group_name_col
                                                                           ),
                                                                gap = unit(2,"mm"),
                                                                show_annotation_name = FALSE,
                                                                annotation_name_gp = my_annotation_name_gp,
                                                                annotation_legend_param = my_annotation_legend_param,
                                                                na_col = my_na_col,
                                                                which = "row",
                                                                show_legend = FALSE,
                                                                width = unit(20,"mm")
                                                                )

# Bottom annotation
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[1:30,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )

bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_NestinFast <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[1:7,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )
#
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_NestinSlow <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[8:10,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )
#
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Prss56 <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[11:19,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )
#
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Neuronal <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[20:22,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )
#
bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Non_Neuronal <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[23:30,c("group_name")]), 
                                                              col = list(group_name = group_name_col
                                                                         ),
                                                              gap = unit(2,"mm"),
                                                              show_annotation_name = FALSE,
                                                              annotation_name_gp = my_annotation_name_gp,
                                                              annotation_legend_param = my_annotation_legend_param,
                                                              na_col = my_na_col,
                                                              height = unit(20,"mm")
                                                              )

# Sample-to-sample heatmap
h <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:30]),
             cluster_rows      = FALSE,
             cluster_columns   = FALSE,
             bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus,
             show_row_names    =FALSE,
             show_column_names =FALSE,
             #column_title      = "R26",
             column_title_gp   = gpar(fontsize = 80, fontface = "bold"),
             row_title_gp      = gpar(fontsize = 80, fontface = "bold"),
             split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
             gap               = unit(3, "mm"),
             #column_order = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[1:30,c("group_name")],
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                #grid_width     = unit(100,"cm"),
                                                grid_height    = unit(1,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n Pearson's coeff.\n",
                                                title_gp       = gpar(fontsize = 60),
                                                title_position = "topcenter",
                                                legend_direction = "horizontal"
                                                )
              )

h_NestinFast <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),1:7])/4,
                        cluster_rows      = FALSE,
                        cluster_columns   = FALSE,
                        bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_NestinFast,
                        show_row_names    = FALSE,
                        show_column_names = FALSE,
show_heatmap_legend    = TRUE,
                        column_title      = "Nestin\nfast",
                        column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
                        row_title_gp      = gpar(fontsize = 40, fontface = "bold"),
                        split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
                        gap               = unit(3, "mm"),
             #column_order = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[1:30,c("group_name")],
                        heatmap_legend_param        = list(color_bar      = "continuous",
                                        #grid_width     = unit(100,"cm"),
                                                           grid_height    = unit(1,"cm"),
                                                           legend_width   = unit(20,"cm"),
                                                           title          = "\n Correlation score\n(Based on Pearson's coefficient)\n",
                                                           title_gp       = gpar(fontsize = 30),
                                                           title_position = "topcenter",
                                                           legend_direction = "horizontal",
labels_gp = gpar(fontsize=20)
                                                           )
                        )
#
h_NestinSlow <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),8:10])/4,
                        cluster_rows      = FALSE,
                        cluster_columns   = FALSE,
                        bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_NestinSlow,
                        show_row_names    =FALSE,
                        show_column_names =FALSE,
show_heatmap_legend    = FALSE,
                        column_title      = "Nestin\nslow",
                        column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
                        row_title_gp      = gpar(fontsize = 40, fontface = "bold"),
                        split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
                        gap               = unit(3, "mm")
                        )
#
h_Prss56 <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),11:19])/4,
                        cluster_rows      = FALSE,
                        cluster_columns   = FALSE,
                        bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Prss56,
                        show_row_names    =FALSE,
                        show_column_names =FALSE,
show_heatmap_legend    = FALSE,
                        column_title      = "Prss56\n",
                        column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
                        row_title_gp      = gpar(fontsize = 40, fontface = "bold"),
                        split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
                        gap               = unit(3, "mm")
                        )
#
h_Neuronal <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),20:22])/4,
                        cluster_rows      = FALSE,
                        cluster_columns   = FALSE,
                        bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Neuronal,
                        show_row_names    =FALSE,
                        show_column_names =FALSE,
show_heatmap_legend    = FALSE,
                        column_title      = "R26\nneuronal",
                        column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
                        row_title_gp      = gpar(fontsize = 40, fontface = "bold"),
                        split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
                        gap               = unit(3, "mm")
                        )
#
h_Non_Neuronal <- Heatmap(matrix           = scale(sampleDist_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[-c(1,2:30),23:30])/4,
                        cluster_rows      = FALSE,
                        cluster_columns   = FALSE,
                        bottom_annotation = bottom_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus_Non_Neuronal,
                        show_row_names    =FALSE,
                        show_column_names =FALSE,
show_heatmap_legend    = FALSE,
                        column_title      = "R26\nnon-neuronal",
                        column_title_gp   = gpar(fontsize = 30, fontface = "bold"),
                        row_title_gp      = gpar(fontsize = 40, fontface = "bold"),
                        split             = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[-c(1,2:30),c("group_name")],
                        gap               = unit(3, "mm")
)
pdf(paste0(pathToFigures,"complexHeatmap_corr_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT.pdf"), width=18, height=16)
draw(h_Neuronal + h_NestinSlow + h_Prss56 + h_NestinFast + h_Non_Neuronal + row_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_plus,
     show_annotation_legend = FALSE,
     heatmap_legend_side = "bottom"
    )
dev.off()

```










### Remove batch effect using all samples and sva (New)

```{r remove_batch_effect_sva}

# Create null and full models
my_full_model <- model.matrix(object = ~as.factor(Data_type),
                              data   = droplevels(sample_murine_affy_RNAseq)
                              )
my_null_model <- model.matrix(object = ~1,
                              data   = droplevels(sample_murine_affy_RNAseq)
                              )

# Estimate the number of latent factor (surrogate varibles)
nb_sv <- num.sv(dat    = as.matrix(matrix_murine_affy_RNAseq),
                mod    = my_full_model,
                method = "leek"
                )

nb_sv

my_svobj <- sva(dat  = as.matrix(matrix_murine_affy_RNAseq),
                mod  = my_full_model,
                mod0 = my_null_model,
                n.sv = 1
                )

str(my_svobj)

plot(density(my_svobj$pprob.gam))

plot(density(my_svobj$pprob.b))

# add pprob.gam to total matrix
matrix_murine_affy_RNAseq_plus <- matrix_murine_affy_RNAseq

matrix_murine_affy_RNAseq_plus$pprob.gam <- my_svobj$pprob.gam

head(matrix_murine_affy_RNAseq_plus)

pprob.gam_thr <- 0.95

matrix_murine_affy_RNAseq_plus_filt <- matrix_murine_affy_RNAseq_plus[which(matrix_murine_affy_RNAseq_plus$pprob.gam > pprob.gam_thr),] 
matrix_murine_affy_RNAseq_plus_filt$pprob.gam <- NULL

head(matrix_murine_affy_RNAseq_plus_filt)
nrow(matrix_murine_affy_RNAseq_plus_filt)

# ComplexHeatmap
pdf(paste0(pathToFigures,"complexHeatmap_murine_affy_RNAseq_batchCorr.pdf"), width=16, height=12)
Heatmap(matrix                      = matrix_murine_affy_RNAseq_plus_filt,
        clustering_distance_columns = my_dist,
        clustering_method_columns   = my_meth,
        show_row_names              = FALSE,
        column_title                = paste0("murine affy and RNAseq clustering \n (distance: ",my_dist,", method: ",my_meth,")"),
        row_title                   = "1000 most variable genes",
        top_annotation              = my_top_annot_murine_affy_RNAseq,
        show_heatmap_legend         = FALSE,
        column_dend_height          = my_column_dend_height,
        column_names_gp             = gpar(fontsize = 10),
        row_title_gp                = my_row_title_gp,
        row_names_gp                = my_row_names_gp
        )
dev.off()

# t-SNE
set.seed(seed = 9)

# Comput t-SNE
tsne_model_murine_affy_RNAseq <- Rtsne(X                = t(matrix_murine_affy_RNAseq_plus_filt),
                                       dims             = 2,
                                       perplexity       = 10,
                                       theta            = 0.0,
                                       pca              = TRUE,
                                       check_duplicates = FALSE,
                                       is_distance      = FALSE
                                       )

# Get t-SNE matrix
tsne_model_murine_affy_RNAseq <- as.data.frame(tsne_model_murine_affy_RNAseq$Y)

# Bind sample plan
tsne_model_murine_affy_RNAseq_plus <- cbind(sample_murine_affy_RNAseq, tsne_model_murine_affy_RNAseq)

# Plot t-SNE result without clustering
pdf(paste0(pathToFigures,"tsne_murine_affy_batchCorr.pdf"), width=12, height=12)
ggplot(data = tsne_model_murine_affy_RNAseq_plus,
       mapping = aes(x=V1, y=V2)) +
    #geom_point(aes(color = Data_source, fill = Tumor_type), size = 6, pch=21) +
    geom_point(aes(fill = Genotype, shape = Tumor_type), size = 6) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_fill_manual(values = genotype_col[c("Nestin","P0-CreC_Smarcb1FloxFlox","Prss56","Ptc+/-p53+/-","Ptch1+/-","R26","Snf5F/F/p53L/L/GFAP-Cre","Th-Mycn","wild_type")],"white") +
    scale_shape_manual(values = c(25,24,23,22,21)) +
    xlab("") +
    ylab("") +
    ggtitle("t-SNE") +
    theme_bw()+
    theme(text = element_text(size = 40)) +
    guides(fill = guide_legend(override.aes = list(shape = 22)))
dev.off()

# PCA
p <- plot_pca(matrix      = matrix_murine_affy_RNAseq_plus_filt,
              category    = sample_murine_affy_RNAseq$Genotype,
              addEllipses = FALSE,
              axes        = c(1,2)
         )
#pdf(paste0(pathToFigures,"pca_murineRT_RNAseq_Sequencing_depth.pdf"), width=12, height=12)
p
#dev.off()

```

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### Combine all murine RT data (affy + RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Select human ATRT samples
sample_human_affy_ATRT <- filter(.data = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Select murine RT samples
sample_murine_RT <- filter(.data      = sample_murine_affy_RNAseq,
                           Tumor_type == "RT"
                           )

# Add subgroup/genotype column to murine sample info
sample_murine_RT_plus <- sample_murine_RT
sample_murine_RT_plus$subgroup_genotype <- ifelse(sample_murine_RT_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                           ifelse(sample_murine_RT_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP",
                                           ifelse(sample_murine_RT_plus$Genotype == "R26","mouse R26",
                                           ifelse(sample_murine_RT_plus$Genotype == "Nestin","mouse Nestin",
                                           ifelse(sample_murine_RT_plus$Genotype == "Prss56","mouse Prss56","mouse NA")))))

                                                           
group_by(.data = sample_murine_RT_plus
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_RT_human_affy_ATRT <- rbind(sample_murine_RT_plus,
                                          sample_human_affy_ATRT_plus
                                          )

group_by(.data = sample_murine_RT_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

nrow(sample_murine_RT_human_affy_ATRT)

colnames(matrix_murine_affy_RNAseq_published_human_affy)

# Merge murine RT and human ATRT matrix matrix 
matrix_murine_affy_RNAseq_human_affy <- merge(x   = matrix_murine_affy_RNAseq_unfiltered,
                                              y   = matrix_human_affy_mouseSymbol_scaled,
                                              by  = 0,
                                              all = FALSE
                                              )
rownames(matrix_murine_affy_RNAseq_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_human_affy$Row.names <- NULL

# Subset matrix
matrix_murine_RT_human_affy_ATRT <- matrix_murine_affy_RNAseq_human_affy[,as.vector(sample_murine_RT_human_affy_ATRT$Sample_name)]

sample_murine_RT_human_affy_ATRT$Sample_name == colnames(matrix_murine_RT_human_affy_ATRT)

# Quantile normalization
matrix_murine_RT_human_affy_ATRT <- quantile_normalisation(matrix_murine_RT_human_affy_ATRT)

head(matrix_murine_RT_human_affy_ATRT[,1:4])
ncol(matrix_murine_RT_human_affy_ATRT)
nrow(matrix_murine_RT_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_RT_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_RT_human_affy_ATRT <- genes.selection(data      = matrix_murine_RT_human_affy_ATRT,
                                                     thres.num = 5000
                                                     )

# Subset matrix of most variable genes
matrix_murine_RT_human_affy_ATRT_mvgenes <-matrix_murine_RT_human_affy_ATRT[rownames(matrix_murine_RT_human_affy_ATRT) %in% mvgenes_murine_RT_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup genotype
#pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data    = pci,
            mapping = aes(x = PC1, y = PC2)
            ) +
    geom_point(aes(color= subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse GFAP","mouse Nestin","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          #legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
#dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data    = pci,
            mapping = aes(x = PC1, y = PC2)
            ) +
    geom_point(aes(color= Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          #legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on data type
pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data    = pci,
            mapping = aes(x = PC1, y = PC2)
            ) +
    geom_point(aes(color= Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          #legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()


# Top annotation
top_annot_murine_RT_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(RT = sample_murine_RT_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Remove batch using sva::ComBat
matrix_murine_RT_human_affy_ATRT_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_RT_human_affy_ATRT_mvgenes,
                                                               batch = sample_murine_RT_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_human_affy_ATRT_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup genotype
pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse GFAP","mouse Nestin","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on Organism
pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on data type
pdf(paste0(pathToFigures,"pca_murine_RT_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_human_affy_ATRT_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_human_affy_ATRT_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_human_affy_ATRT_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_human_affy_ATRT,,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### Combine murine RT in situ data (affy + RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Select human ATRT samples
sample_human_affy_ATRT <- filter(.data = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Select murine RT samples
sample_murine_RT_primTum <- filter(.data        = sample_murine_affy_RNAseq,
                                   Tumor_type   == "RT",
                                   Tumor_origin == "primary_tumor"
                                   )

# Add subgroup/genotype column to murine sample info
sample_murine_RT_primTum_plus <- sample_murine_RT_primTum
sample_murine_RT_primTum_plus$subgroup_genotype <- ifelse(sample_murine_RT_primTum_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                           ifelse(sample_murine_RT_primTum_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP",
                                           ifelse(sample_murine_RT_primTum_plus$Genotype == "R26","mouse R26",
                                           ifelse(sample_murine_RT_primTum_plus$Genotype == "Nestin","mouse Nestin",
                                           ifelse(sample_murine_RT_primTum_plus$Genotype == "Prss56","mouse Prss56","mouse NA")))))

                                                           
group_by(.data = sample_murine_RT_primTum_plus
        ,Genotype
        ,Tumor_origin
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_RT_primTum_human_affy_ATRT <- rbind(sample_murine_RT_primTum_plus,
                                                  sample_human_affy_ATRT_plus
                                                  )

group_by(.data = sample_murine_RT_primTum_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Merge murine RT and human ATRT matrix matrix 
matrix_murine_affy_RNAseq_human_affy <- merge(x   = matrix_murine_affy_RNAseq_unfiltered,
                                              y   = matrix_human_affy_mouseSymbol_scaled,
                                              by  = 0,
                                              all = FALSE
                                              )
rownames(matrix_murine_affy_RNAseq_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_human_affy$Row.names <- NULL

# Subset matrix
matrix_murine_RT_primTum_human_affy_ATRT <- matrix_murine_affy_RNAseq_human_affy[,as.vector(sample_murine_RT_primTum_human_affy_ATRT$Sample_name)]

sample_murine_RT_primTum_human_affy_ATRT$Sample_name == colnames(matrix_murine_RT_primTum_human_affy_ATRT)

# Quantile normalization
matrix_murine_RT_primTum_human_affy_ATRT <- quantile_normalisation(matrix_murine_RT_primTum_human_affy_ATRT)

head(matrix_murine_RT_primTum_human_affy_ATRT[,1:4])
ncol(matrix_murine_RT_primTum_human_affy_ATRT)
nrow(matrix_murine_RT_primTum_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_RT_primTum_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_RT_primTum_human_affy_ATRT <- genes.selection(data      = matrix_murine_RT_primTum_human_affy_ATRT,
                                                     thres.num = 5000
                                                     )

# Subset matrix of most variable genes
matrix_murine_RT_primTum_human_affy_ATRT_mvgenes <-matrix_murine_RT_primTum_human_affy_ATRT[rownames(matrix_murine_RT_primTum_human_affy_ATRT) %in% mvgenes_murine_RT_primTum_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_primTum_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# Top annotation
top_annot_murine_RT_primTum_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(RT = sample_murine_RT_primTum_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_primTum_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_primTum_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_primTum_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Remove batch using sva::ComBat
matrix_murine_RT_primTum_human_affy_ATRT_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_RT_primTum_human_affy_ATRT_mvgenes,
                                                               batch = sample_murine_RT_primTum_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_primTum_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup Genotype
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_primTum_human_affy_ATRT_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_primTum_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_primTum_human_affy_ATRT_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_primTum_human_affy_ATRT,,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### Combine murine RT data (RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Select human ATRT samples
sample_human_affy_ATRT <- filter(.data          = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Select murine RT samples
sample_murine_RT_RNAseq <- filter(.data        = sample_murine_affy_RNAseq,
                                  Tumor_type   == "RT",
                                  Data_type    == "RNA_seq"
                                  )

# Add subgroup/genotype column to murine sample info
sample_murine_RT_RNAseq_plus <- sample_murine_RT_RNAseq
sample_murine_RT_RNAseq_plus$subgroup_genotype <- ifelse(sample_murine_RT_RNAseq_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                           ifelse(sample_murine_RT_RNAseq_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP",
                                           ifelse(sample_murine_RT_RNAseq_plus$Genotype == "R26","mouse R26",
                                           ifelse(sample_murine_RT_RNAseq_plus$Genotype == "Nestin","mouse Nestin",
                                           ifelse(sample_murine_RT_RNAseq_plus$Genotype == "Prss56","mouse Prss56","mouse NA")))))

                                                           
group_by(.data = sample_murine_RT_RNAseq_plus
        ,Genotype
        ,Tumor_origin
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_RT_RNAseq_human_affy_ATRT <- rbind(sample_murine_RT_RNAseq_plus,
                                                  sample_human_affy_ATRT_plus
                                                  )

group_by(.data = sample_murine_RT_RNAseq_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Merge murine RT and human ATRT matrix matrix 
matrix_murine_affy_RNAseq_human_affy <- merge(x   = matrix_murine_affy_RNAseq_unfiltered,
                                              y   = matrix_human_affy_mouseSymbol_scaled,
                                              by  = 0,
                                              all = FALSE
                                              )
rownames(matrix_murine_affy_RNAseq_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_human_affy$Row.names <- NULL

# Subset matrix
matrix_murine_RT_RNAseq_human_affy_ATRT <- matrix_murine_affy_RNAseq_human_affy[,as.vector(sample_murine_RT_RNAseq_human_affy_ATRT$Sample_name)]

sample_murine_RT_RNAseq_human_affy_ATRT$Sample_name == colnames(matrix_murine_RT_RNAseq_human_affy_ATRT)

# Quantile normalization
matrix_murine_RT_RNAseq_human_affy_ATRT <- quantile_normalisation(matrix_murine_RT_RNAseq_human_affy_ATRT)

head(matrix_murine_RT_RNAseq_human_affy_ATRT[,1:4])
ncol(matrix_murine_RT_RNAseq_human_affy_ATRT)
nrow(matrix_murine_RT_RNAseq_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_RT_RNAseq_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_RT_RNAseq_human_affy_ATRT <- genes.selection(data      = matrix_murine_RT_RNAseq_human_affy_ATRT,
                                                     thres.num = 5000
                                                     )

# Subset matrix of most variable genes
matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes <-matrix_murine_RT_RNAseq_human_affy_ATRT[rownames(matrix_murine_RT_RNAseq_human_affy_ATRT) %in% mvgenes_murine_RT_RNAseq_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_RNAseq_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse Nestin","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# Top annotation
top_annot_murine_RT_RNAseq_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(RT = sample_murine_RT_RNAseq_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_RNAseq_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_RNAseq_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_RNAseq_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_RNAseq_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Remove batch using sva::ComBat
matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes,
                                                               batch = sample_murine_RT_RNAseq_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_RNAseq_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup Genotype
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse Nestin","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_RNAseq_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_RNAseq_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_RNAseq_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_RNAseq_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_RNAseq_human_affy_ATRT_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_RNAseq_human_affy_ATRT,,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### Combine murine RT in situ data (RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Select human ATRT samples
sample_human_affy_ATRT <- filter(.data          = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Select murine RT samples
sample_murine_RT_primTum_RNAseq <- filter(.data        = sample_murine_affy_RNAseq,
                                  Tumor_type   == "RT",
                                  Tumor_origin == "primary_tumor",
                                  Data_type    == "RNA_seq"
                                  )

# Add subgroup/genotype column to murine sample info
sample_murine_RT_primTum_RNAseq_plus <- sample_murine_RT_primTum_RNAseq
sample_murine_RT_primTum_RNAseq_plus$subgroup_genotype <- ifelse(sample_murine_RT_primTum_RNAseq_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                           ifelse(sample_murine_RT_primTum_RNAseq_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP",
                                           ifelse(sample_murine_RT_primTum_RNAseq_plus$Genotype == "R26","mouse R26",
                                           ifelse(sample_murine_RT_primTum_RNAseq_plus$Genotype == "Nestin","mouse Nestin",
                                           ifelse(sample_murine_RT_primTum_RNAseq_plus$Genotype == "Prss56","mouse Prss56","mouse NA")))))

                                                           
group_by(.data = sample_murine_RT_primTum_RNAseq_plus
        ,Genotype
        ,Tumor_origin
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_RT_primTum_RNAseq_human_affy_ATRT <- rbind(sample_murine_RT_primTum_RNAseq_plus,
                                                  sample_human_affy_ATRT_plus
                                                  )

group_by(.data = sample_murine_RT_primTum_RNAseq_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Merge murine RT and human ATRT matrix matrix 
matrix_murine_affy_RNAseq_human_affy <- merge(x   = matrix_murine_affy_RNAseq_unfiltered,
                                              y   = matrix_human_affy_mouseSymbol_scaled,
                                              by  = 0,
                                              all = FALSE
                                              )
rownames(matrix_murine_affy_RNAseq_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_human_affy$Row.names <- NULL

# Subset matrix
matrix_murine_RT_primTum_RNAseq_human_affy_ATRT <- matrix_murine_affy_RNAseq_human_affy[,as.vector(sample_murine_RT_primTum_RNAseq_human_affy_ATRT$Sample_name)]

sample_murine_RT_primTum_RNAseq_human_affy_ATRT$Sample_name == colnames(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT)

# Quantile normalization
matrix_murine_RT_primTum_RNAseq_human_affy_ATRT <- quantile_normalisation(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT)

head(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT[,1:4])
ncol(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT)
nrow(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_RT_primTum_RNAseq_human_affy_ATRT <- genes.selection(data      = matrix_murine_RT_primTum_RNAseq_human_affy_ATRT,
                                                     thres.num = 5000
                                                     )

# Subset matrix of most variable genes
matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes <-matrix_murine_RT_primTum_RNAseq_human_affy_ATRT[rownames(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT) %in% mvgenes_murine_RT_primTum_RNAseq_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_primTum_RNAseq_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# Top annotation
top_annot_murine_RT_primTum_RNAseq_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(RT = sample_murine_RT_primTum_RNAseq_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_RNAseq_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_primTum_RNAseq_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_RNAseq_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_primTum_RNAseq_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Remove batch using sva::ComBat
matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes,
                                                               batch = sample_murine_RT_primTum_RNAseq_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RT_primTum_RNAseq_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup Genotype
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse Nestin","mouse GFAP","mouse P0","mouse Prss56","mouse R26")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RT_primTum_RNAseq_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_RNAseq_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RT_primTum_RNAseq_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RT_primTum_RNAseq_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RT_primTum_RNAseq_human_affy_ATRT_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RT_primTum_RNAseq_human_affy_ATRT,,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

### Combine murine RT Prss56 data (RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Select human ATRT samples
sample_human_affy_ATRT <- filter(.data          = sample_human_affy,
                                 Tumor_sub_type == "ATRT"
                                 )

# Add subgroup/genotype column to human sample info
sample_human_affy_ATRT_plus <- sample_human_affy_ATRT
sample_human_affy_ATRT_plus$subgroup_genotype <- paste0("human ",sample_human_affy_ATRT_plus$group_name)

group_by(.data = sample_human_affy_ATRT_plus
        ,Data_source
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Select murine RT samples
sample_murine_Prss56 <- filter(.data        = sample_murine_affy_RNAseq,
                               Genotype    == "Prss56"
                               )

# Add subgroup/genotype column to murine sample info
sample_murine_Prss56_plus <- sample_murine_Prss56
sample_murine_Prss56_plus$subgroup_genotype <- ifelse(sample_murine_Prss56_plus$Genotype == "P0-CreC_Smarcb1FloxFlox","mouse P0",
                                           ifelse(sample_murine_Prss56_plus$Genotype == "Snf5F/F/p53L/L/GFAP-Cre","mouse GFAP",
                                           ifelse(sample_murine_Prss56_plus$Genotype == "R26","mouse R26",
                                           ifelse(sample_murine_Prss56_plus$Genotype == "Nestin","mouse Nestin",
                                           ifelse(sample_murine_Prss56_plus$Genotype == "Prss56","mouse Prss56","mouse NA")))))

                                                           
group_by(.data = sample_murine_Prss56_plus
        ,Genotype
        ,Tumor_origin
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Bind human and murine samples information
sample_murine_Prss56_human_affy_ATRT <- rbind(sample_murine_Prss56_plus,
                                                  sample_human_affy_ATRT_plus
                                                  )

group_by(.data = sample_murine_Prss56_human_affy_ATRT
        ,Genotype
        ,group_name
        ,subgroup_genotype
         ) %>%
    summarise(number =  n()) %>%
    print(n = nrow(.), width = Inf)

# Merge murine RT and human ATRT matrix matrix 
matrix_murine_affy_RNAseq_human_affy <- merge(x   = matrix_murine_affy_RNAseq_unfiltered,
                                              y   = matrix_human_affy_mouseSymbol_scaled,
                                              by  = 0,
                                              all = FALSE
                                              )
rownames(matrix_murine_affy_RNAseq_human_affy) <- matrix_murine_affy_RNAseq_published_human_affy$Row.names
matrix_murine_affy_RNAseq_human_affy$Row.names <- NULL

# Subset matrix
matrix_murine_Prss56_human_affy_ATRT <- matrix_murine_affy_RNAseq_human_affy[,as.vector(sample_murine_Prss56_human_affy_ATRT$Sample_name)]

sample_murine_Prss56_human_affy_ATRT$Sample_name == colnames(matrix_murine_Prss56_human_affy_ATRT)

# Quantile normalization
matrix_murine_Prss56_human_affy_ATRT <- quantile_normalisation(matrix_murine_Prss56_human_affy_ATRT)

head(matrix_murine_Prss56_human_affy_ATRT[,1:4])
ncol(matrix_murine_Prss56_human_affy_ATRT)
nrow(matrix_murine_Prss56_human_affy_ATRT)

plot(density(as.matrix(matrix_murine_Prss56_human_affy_ATRT)))

# Most variable genes
mvgenes_murine_Prss56_human_affy_ATRT <- genes.selection(data      = matrix_murine_Prss56_human_affy_ATRT,
                                                     thres.num = 5000
                                                     )

# Subset matrix of most variable genes
matrix_murine_Prss56_human_affy_ATRT_mvgenes <-matrix_murine_Prss56_human_affy_ATRT[rownames(matrix_murine_Prss56_human_affy_ATRT) %in% mvgenes_murine_Prss56_human_affy_ATRT,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_Prss56_human_affy_ATRT_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_Prss56_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse Prss56")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# Top annotation
top_annot_murine_Prss56_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(RT = sample_murine_Prss56_human_affy_ATRT[,"subgroup_genotype"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 RT = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = FALSE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_Prss56_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_Prss56_human_affy_ATRT_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_Prss56_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_Prss56_human_affy_ATRT_mvgenes)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_Prss56_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_Prss56_human_affy_ATRT_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_Prss56_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

# Remove batch using sva::ComBat
matrix_murine_Prss56_human_affy_ATRT_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_Prss56_human_affy_ATRT_mvgenes,
                                                               batch = sample_murine_Prss56_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_Prss56_human_affy_ATRT_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_Prss56_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup Genotype
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=subgroup_genotype), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[c("human MYC","human SHH","human TYR","mouse Prss56")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[c("Homo_sapiens","Mus_musculus")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_Prss56_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[c("RNA_microarray","RNA_seq")])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_Prss56_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_Prss56_human_affy_ATRT_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_Prss56_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_Prss56_human_affy_ATRT_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_Prss56_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_Prss56_human_affy_ATRT_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_Prss56_human_affy_ATRT,,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```

icicicicicic

### Combine murine RT Curie in situ data (RNAseq) and human ATRT affy (Curie & DKFZ)


```{r murine_published_and_human_ATRT_affy}

# Combined human and murine sample
sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT

# Combined human and murine matrix
head(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[,1:4])

plot(density(as.matrix(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)))

# Check order
sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Sample_name == colnames(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)

# Quantile normalization
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn <- quantile_normalisation(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT)

matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn <- matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT

head(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[,1:4])
ncol(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn)
nrow(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn)

plot(density(as.matrix(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn)))

# Most variable genes
mvgenes_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn <- genes.selection(data      = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn,
                                                                              thres.num = 5000
                                                                              )

# Subset matrix of most variable genes
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes <- matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn[rownames(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn) %in% mvgenes_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn,]

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT
                        )

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=group_name), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[levels(factor(pci$group_name))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[levels(factor(pci$Organism))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_PC1PC2_beforeBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[levels(factor(pci$Data_type))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# Top annotation
top_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT <- HeatmapAnnotation(df = data.frame(sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT[,"group_name"]), 
                                                                                      col = list(#Organism = organism_col,
                                                                                                 group_name = group_name_col),
                                                                                      gap = unit(2,"mm"),
                                                                                      show_annotation_name = TRUE,
                                                                                      annotation_name_gp = my_annotation_name_gp,
                                                                                      annotation_legend_param = my_annotation_legend_param,
                                                                                      na_col = my_na_col,
                                                                                      show_legend = TRUE
                                                                                      )

pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes)), method = "ward")
pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_beforeBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
dev.off()

# Remove batch using sva::ComBat
matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes_ComBat <- sva::ComBat(dat = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes,
                                                               batch = sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT$Organism
                                                               )

# PCA
PC        <- prcomp(data.frame(t(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes_ComBat)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)
pci       <- data.frame(PC$x,
                        sample_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT
                        )

# PCA organism PC1 and PC2 on subgroup Genotype
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_combatBatchCorr_subgroupGenotype.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=group_name), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "RT",values = as.vector(group_name_col[levels(factor(pci$group_name))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_combatBatchCorr_organism.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Organism), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Species",values = as.vector(organism_col[levels(factor(pci$Organism))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

# PCA organism PC1 and PC2 on organism
pdf(paste0(pathToFigures,"pca_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_combatBatchCorr_dataType.pdf"), width=12, height=10)
p <- ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(aes(color=Data_type), size = 5) +
    #scale_shape_manual(values = c(24,23,21)) +
    #geom_text_repel(mapping = aes(label = Sample_name), size = 5) +
    scale_color_manual(name = "Data type",values = as.vector(data_type_col[levels(factor(pci$Data_type))])) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text = element_text(size = 40),
          legend.key.height = unit(1.5,"cm"),
          legend.position = "bottom",
          legend.direction = "horizontal",
          legend.title = element_blank(),
          panel.background = element_rect(fill = "white"),
          panel.grid.major = element_line(colour = "white"),
          panel.grid.minor = element_line(colour = "white")
          )
#guides(fill = guide_legend(override.aes = list(shape = 23)))
#guide_legend(nrow = 1)
p
dev.off()

pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix                      = matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes_ComBat,
             clustering_distance_columns = my_dist,
             clustering_method_columns   = my_meth,
             show_row_names              = FALSE,
             show_column_names           = FALSE,
             show_row_dend               = FALSE,
             column_title                = paste0("hierarchical clustering \n"),
             row_title                   = NULL,
             top_annotation              = top_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             show_heatmap_legend         = FALSE,
             column_dend_height          = my_column_dend_height,
             column_names_gp             = gpar(fontsize = 10),
             column_title_gp             = gpar(fontsize = 40),
             row_title_gp                = my_row_title_gp,
             row_names_gp                = my_row_names_gp,
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             )
draw(h)
dev.off()

# Plot dendogram only
colDend <- hclust(pearson.dist(t(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes_ComBat)), method = "ward")
#pdf(paste0(pathToFigures,"complexHeatmap_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_combatBatchCorr.pdf"), width=16, height=12)
h <- Heatmap(matrix        = matrix(nrow= 0, ncol= ncol(matrix_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT_qn_mvgenes_ComBat)),
             cluster_columns    = colDend,
             clustering_method_columns   = my_meth,
             top_annotation     = top_annot_murine_RNAseq_R26_Nestin_Prss56_human_affy_ATRT,
             column_dend_height = my_column_dend_height,
             show_row_names     = FALSE,
             show_column_names     = TRUE,
             column_title = "hierarchical clustering\n",
             column_title_gp = gpar(fontsize = 40),
             column_names_gp    = gpar(fontsize = 14),
             rect_gp                     = gpar(type = "none"),
             top_annotation_height       = unit(2,"cm")
             ) 
draw(h, annotation_legend_side = "bottom")
#dev.off()

```


```{r sessionInfo}

sessionInfo()

```
