title: "Human ATRT RNAseq data"
author: "Mamy Andrianteranagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
    fig_caption: yes
    fig_height: 8
    fig_width: 9
    keep_md: yes
    toc: yes
    toc_depth: 2
    toc_float: yes
    number_sections: true
    code_folding: hide
  pdf_document:
    toc: yes
    toc_depth: '2'
  word_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, include=TRUE, cache=FALSE, error=TRUE)

```

```{r paths}

thisProject                  <- "murine_RT/"

# Paths
pathToRhabdoid_U830_projects <- "/bioinfo/users/mandrian/rhabdoid_U830_projects/"
pathToProgrammes             <- paste0(pathToRhabdoid_U830_projects,"programmes/")
pathToANALYSIS_PROJECTS      <- paste0(pathToRhabdoid_U830_projects,"ANALYSIS_PROJECTS/")
pathToThisProject            <- paste0(pathToANALYSIS_PROJECTS,thisProject)
pathToData                   <- paste0(pathToThisProject,"data/")
pathToFigures                <- paste0(pathToThisProject,"svn/analyse/script/figures/")
pathToReport                 <- paste0(pathToThisProject,"svn/analyse/report/")
pathToInfo                   <- paste0(pathToThisProject,"svn/analyse/info/")

pathToDataAnnotationFile     <- paste0(pathToRhabdoid_U830_projects,"annotation_file/")

```

```{r manage_renv, eval=FALSE}

# Initialize Renv
#renv::init()

```


```{r load_library, eval=FALSE}

# Load RNAseq library
source("load_scRNAseq_library.R")

#remove.packages("Seurat")

```

```{r reticulate, eval=FALSE}

library(reticulate)

use_python("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/conda_env_cytoTRACE/bin/python3.6")

#use_condaenv("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/conda_env_cytoTRACE")

#detach("package:CytoTRACE", unload=TRUE)

library(CytoTRACE)

```

```{r loading_RData, eval=FALSE}

# Load saved RData
load("murine_RT_scRNAseq_integrated_R26P87_R26P27_R26P26_tumCells.RData")

#save.image(file="murine_RT_scRNAseq_integrated_R26P87_R26P27_R26P26_tumCells.RData")

```

```{r load_plot_function}

# Load external files
source("../plot_R_util.R")
source("/data/users/mandrian/rhabdoid_U830_projects/annotation_file/colors.R")

```

```{r delete_all_objects}

# Delete all objects
#rm(list = ls())

```

# Load normalized Seurat object

## R26P87

```{r load_R26P87_data}

scRNAseq_Seurat_R26P87 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_R26P87.RDS")                                  )

```

## R26P27

```{r load_R26P27_data}

scRNAseq_Seurat_R26P27 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_R26P27.RDS")
                                  )

str(scRNAseq_Seurat_R26P27)

```

## R26P26

```{r load_R26P26_data}

scRNAseq_Seurat_R26P26 <- readRDS(file   = paste0(pathToReport,"scRNAseq_Seurat_R26P26.RDS")
                                  )

```


## Clustering with only tumoral cells

Immune and non-tumoral cells are identified by Zhi-Yan as cluster 5,6,7,9,10 (see e-mail from Zhi-Yan on jan 10 2020) and corresponding cells will be removed for the subsequent analysis

### Subset cells

```{r clustering}

# Import rds file
scRNAseq_Seurat_integrated_R26 <- readRDS(file = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26.rds"
                                                        )
                                          )

# Select non-tumoral cell IDs: cluster 5,6,7,9,10 (for SCENIC analysis)
nontumCells_IDs <- rownames(scRNAseq_Seurat_integrated_R26@meta.data[which(scRNAseq_Seurat_integrated_R26@meta.data$integrated_snn_res.0.3 %in% c(5,6,7,9,10)),])

head(nontumCells_IDs)
length(nontumCells_IDs)
#
nontumCells_IDs_R26P87 <- gsub(pattern = "_1",
                               replacement = "",
                               x = nontumCells_IDs[grepl("_1",nontumCells_IDs)]
                               )
head(nontumCells_IDs_R26P87)
length(nontumCells_IDs_R26P87)
#
nontumCells_IDs_R26P27 <- gsub(pattern = "_2",
                               replacement = "",
                               x = nontumCells_IDs[grepl("_2",nontumCells_IDs)]
                               )
head(nontumCells_IDs_R26P27)
length(nontumCells_IDs_R26P27)
#
nontumCells_IDs_R26P26 <- gsub(pattern = "_3",
                               replacement = "",
                               x = nontumCells_IDs[grepl("_3",nontumCells_IDs)]
                               )
head(nontumCells_IDs_R26P26)
length(nontumCells_IDs_R26P26)

# Export non tumoral cell IDs
write.table(x          = nontumCells_IDs_R26P87,
            file       = paste0(pathToReport,"scRNAseq_integrated_R26P87_nonTumCell_IDs.txt"),
            sep        = "\n",
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE
            )
#
write.table(x          = nontumCells_IDs_R26P27,
            file       = paste0(pathToReport,"scRNAseq_integrated_R26P27_nonTumCell_IDs.txt"),
            sep        = "\n",
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE
            )
#
write.table(x          = nontumCells_IDs_R26P26,
            file       = paste0(pathToReport,"scRNAseq_integrated_R26P26_nonTumCell_IDs.txt"),
            sep        = "\n",
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE
            )

# Select cell IDs of cluster 0,1,2,3,4,8 and 11
tumCells_IDs <- rownames(scRNAseq_Seurat_integrated_R26@meta.data[which(scRNAseq_Seurat_integrated_R26@meta.data$integrated_snn_res.0.3 %in% c(0,1,2,3,4,8,11)),])


head(tumCells_IDs)
length(tumCells_IDs)

# Export tumor cell IDs
write.table(x          = tumCells_IDs,
            file       = paste0(pathToReport,"scRNAseq_integrated_R26_tumCell_IDs.txt"),
            sep        = "\n",
            row.names  = FALSE,
            col.names  = FALSE,
            quote      = FALSE
            )
           
# Check the length of tumor cell IDs for each sampe
length(tumCells_IDs[grepl("_1",tumCells_IDs)])
length(tumCells_IDs[grepl("_2",tumCells_IDs)])
length(tumCells_IDs[grepl("_3",tumCells_IDs)])


# Extract tumor cells for each sample
scRNAseq_Seurat_R26P87_tumCells <- subset(scRNAseq_Seurat_R26P87,
                                          cells = gsub(pattern = "_1",
                                                       replacement = "",
                                                       x = tumCells_IDs[grepl("_1",tumCells_IDs)]
                                                       )
                                          )
#
scRNAseq_Seurat_R26P27_tumCells <- subset(scRNAseq_Seurat_R26P27,
                                          cells = gsub(pattern = "_2",
                                                       replacement = "",
                                                       x = tumCells_IDs[grepl("_2",tumCells_IDs)]
                                                       )
                                          )
#
scRNAseq_Seurat_R26P26_tumCells <- subset(scRNAseq_Seurat_R26P26,
                                          cells = gsub(pattern = "_3",
                                                       replacement = "",
                                                       x = tumCells_IDs[grepl("_3",tumCells_IDs)]
                                                       )
                                          )

# Save as loom object
#as.loom(x        = scRNAseq_Seurat_R26P87_tumCells,
 #       filename = "scRNAseq_Seurat_R26P87_tumCells.loom"
  #      )

```

### Integrate seurat object

```{r integrated_seurat_object}

# Create named list of seurat object
seurat_object_list_tumCells <- list(R26P87 = scRNAseq_Seurat_R26P87_tumCells,
                                    R26P27 = scRNAseq_Seurat_R26P27_tumCells,
                                    R26P26 = scRNAseq_Seurat_R26P26_tumCells
                                    )

# Find anchors
max_dim <- 30
scRNAseq_Seurat_anchors_tumCells <- FindIntegrationAnchors(object.list           = seurat_object_list_tumCells,
                                                           assay                 = NULL,
                                                           reference             = NULL,
                                                           anchor.features       = 2000,
                                                           scale                 = TRUE,
                                                           normalization.method  = "LogNormalize",
                                                           sct.clip.range        = NULL,
                                                           reduction             = "cca",
                                                           l2.norm               = TRUE,
                                                           dims                  = 1:max_dim,
                                                           k.anchor              = 5,
                                                           k.filter              = 200,
                                                           k.score               = 30,
                                                           max.features          = 200,
                                                           nn.method             = "rann",
                                                           eps                   = 0,
                                                           verbose               = TRUE
                                                           )

# Integrated seurat objects
scRNAseq_Seurat_integrated_R26_tumCells <- IntegrateData(anchorset              = scRNAseq_Seurat_anchors_tumCells,
                                                         new.assay.name         = "integrated",
                                                         normalization.method   = "LogNormalize",
                                                         features               = NULL,
                                                         features.to.integrate  = NULL,
                                                         dims                   = 1:max_dim,
                                                         k.weight               = 100,
                                                         weight.reduction       = NULL,
                                                         sd.weight              = 1,
                                                         sample.tree            = NULL,
                                                         preserve.order         = FALSE,
                                                         do.cpp                 = TRUE,
                                                         eps                    = 0,
                                                         verbose                = TRUE
                                                         )

```


# Linear dimensional reduction

```{r feature_selection}

# Data scaling 
scRNAseq_Seurat_integrated_R26_tumCells <- ScaleData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                     features = rownames(scRNAseq_Seurat_integrated_R26_tumCells)
                                                     )

# PCA
scRNAseq_Seurat_integrated_R26_tumCells <- RunPCA(object = scRNAseq_Seurat_integrated_R26_tumCells)

# PCs elbow plot
PCmin <- 13
#pdf(file = paste0(pathToFigures,"elbowPlot_scRNAseq_integrated_R26_tumCells_PCA.pdf"), width = 12, height = 12)
ElbowPlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
          ndims = 50
          ) +
    geom_vline(xintercept = PCmin, lty = 2, color = "blue") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Heatmap of each PC
#pdf(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_integrated_R26_tumCells_PCA.pdf"), width = 15, height = 12)
#png(file = paste0(pathToFigures,"dimHeatmap_scRNAseq_integrated_R26_tumCells_PCA.png"), width = 1000, height = 800)
DimHeatmap(object = scRNAseq_Seurat_integrated_R26_tumCells,
           dims   = 1:15,
           ncol   = 5,
           raster = FALSE,
           assay  = "integrated"
           )
#dev.off()

# Plot PCA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA.pdf"), width = 12, height = 12)
DimPlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
        reduction = "pca",
        group.by = "orig.ident"
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(text = element_text(size = 20),
          axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "right"
          )
#dev.off()

```

# Linear dimensional reduction (without integration, to check integration quality)

```{r linear_dimensional_reduction}

scRNAseq_Seurat_integrated_R26_tumCells_RNA <- scRNAseq_Seurat_integrated_R26_tumCells

DefaultAssay(scRNAseq_Seurat_integrated_R26_tumCells_RNA) <- "RNA"

# Highly variable genes
nFeatures <- 2000
variableFeatures_method <- "vst"
scRNAseq_Seurat_integrated_R26_tumCells_RNA <- FindVariableFeatures(object            = scRNAseq_Seurat_integrated_R26_tumCells_RNA,
                                                                    selection.method  = variableFeatures_method,
                                                                    nfeatures         = nFeatures
                                                                    )

# Data scaling 
scRNAseq_Seurat_integrated_R26_tumCells_RNA <- ScaleData(object = scRNAseq_Seurat_integrated_R26_tumCells_RNA,
                                                         features = rownames(scRNAseq_Seurat_integrated_R26_tumCells)
                                                         )


# Extract matrix
scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_RNA,
                                                                             slot = "scale.data"
                                                                             )
                                                                )

head(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA[,1:3])
dim(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA)

# PCA analysis
PC_RNA <- prcomp(data.frame(t(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA)))

eigVal_RNA <- get_eigenvalue(PC_RNA)
eigValPer_RNA <- round(eigVal_RNA$variance.percent, digits = 2)
pci_RNA <- data.frame(PC_RNA$x,scRNAseq_Seurat_integrated_R26_tumCells_RNA@meta.data)

# Plot PCA and sample
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_merge_R26_tumCells_PCA.pdf"), width = 12, height = 10)
ggplot(data = pci_RNA,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = orig.ident),
               size    = 1
               ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    xlab(paste0("PC1 (",eigValPer_RNA[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer_RNA[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```


## Plot PCA and covariates

```{r PCA_and_covariates}

# Extract matrix
scRNAseq_Seurat_integrated_R26_tumCells_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                slot = "scale.data"
                                                                )
                                                   )

head(scRNAseq_Seurat_integrated_R26_tumCells_matrix[,1:3])
dim(scRNAseq_Seurat_integrated_R26_tumCells_matrix)

# PCA analysis
PC        <- prcomp(data.frame(t(scRNAseq_Seurat_integrated_R26_tumCells_matrix)))
eigVal    <- get_eigenvalue(PC)
eigValPer <- round(eigVal$variance.percent, digits = 2)

pci       <- data.frame(PC$x,scRNAseq_Seurat_integrated_R26_tumCells@meta.data)

# Plot eigen value
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_R26_tumCells_PCA_eigValPer.pdf"), width = 10, height = 10)
ggplot() +
    geom_bar(data = data.frame(PC = 1:50, eigValPer = eigValPer[1:50]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity"
             ) +
    geom_bar(data = data.frame(PC = 1:PCmin, eigValPer = eigValPer[1:PCmin]),
             mapping = aes(x = PC, y = eigValPer),
             stat = "identity",
             fill = "blue"
             ) +
    xlab(label = "\nPC") +
    ylab(label = "% variability (eigen value)\n") +
    theme_bw() +
    theme(axis.text = element_text(size = 20),
          axis.title = element_text(size = 20, face = "bold"),
          legend.position = "none"
          )
#dev.off()

# Plot PCA and sample
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_sample.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = orig.ident),
               size    = 1
               ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb of UMIs
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_nbUMIs.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = log10(nCount_RNA)),
               size    = 1
               ) +
    scale_colour_continuous(name = "log10(nb of UMIs)",
                            type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb of genes
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_nbGenes.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = log10(nFeature_RNA)),
               size    = 1
               ) +
    scale_colour_continuous(name = "log10(nb of genes)",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and nb RNA per gene
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_nbRNA_per_gene.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(colour = nCount_per_Feature),
               size    = 1
               ) +
    scale_colour_continuous(name = "UMIs/gene",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and % mito RNA
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_percMito.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = percent.mito),
               size    = 1
               ) +
    scale_colour_continuous(name = "% mito",
                            type = "viridis"
                            ) +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 20),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

# Plot PCA and cell cycle phase
#pdf(file = paste0(pathToFigures,"pca_scRNAseq_integrated_R26_tumCells_PCA_cellCyclePhase.pdf"), width = 12, height = 10)
ggplot(data = pci,
       mapping = aes(x = PC1, y = PC2)
       ) +
    geom_point(mapping = aes(color = Phase),
               size    = 1#,
               #pch     = 21
               ) +
    #scale_fill_continuous(type = "viridis") +
    xlab(paste0("PC1 (",eigValPer[1]," %)")) +
    ylab(paste0("PC2 (",eigValPer[2]," %)")) +
    ggtitle("PCA") +
    theme_bw()+
    theme(text              = element_text(size = 30),
          legend.title = element_text(face = "bold"),
          legend.key.height = unit(1.5,"cm")
          )
#dev.off()

```

### Clustering

```{r linear_dim_red}

# Construct KNN graph
scRNAseq_Seurat_integrated_R26_tumCells <- FindNeighbors(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                         dims   = 1:PCmin
                                                         )

# Clustering by modularity optimization technique
clustering_resolutions <- seq(from  = 0, to = 2, by = 0.1)
scRNAseq_Seurat_integrated_R26_tumCells <- FindClusters(object     = scRNAseq_Seurat_integrated_R26_tumCells,
                                                        resolution = clustering_resolutions
                                                        )

```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_integrated_R26_tumCells.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_integrated_R26_tumCells@meta.data,
         prefix          = "integrated_snn_res.",
         #node_colour     = "sc3_stability",
         layout          = "sugiyama",
         use_core_edges  = TRUE
         )
#dev.off()

choosenResolution <- 0.4

# Set default ident
Idents(scRNAseq_Seurat_integrated_R26_tumCells) <- scRNAseq_Seurat_integrated_R26_tumCells@meta.data[["integrated_snn_res.0.4"]]

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_tumCells_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells@meta.data
       ) +
    stat_count(mapping = aes(x     = integrated_snn_res.0.4,
                             fill  = orig.ident
                             )
               ) +
    geom_text(mapping  = aes(x     = integrated_snn_res.0.4,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              face     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
scale_fill_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_integrated_R26_tumCells@meta.data)

write.table(x      = scRNAseq_Seurat_integrated_R26_tumCells@meta.data,
            file   = "scRNAseq_Seurat_integrated_R26_tumCells_PCmin_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_integrated_R26_tumCells <- RunUMAP(object       = scRNAseq_Seurat_integrated_R26_tumCells,
                                                   dims         = 1:PCmin,
                                                   n.neighbors  = 30
                                                   )

# Plot UMAP and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

# Plot UMAP and sample
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_sample.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "orig.ident"
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# Save seurat object
saveRDS(object = scRNAseq_Seurat_integrated_R26_tumCells,
        file   = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26_tumCells.RDS"
                        )
        )

# Copy saved Seurat object to zerkalo
copyToZerkalo(file = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26_tumCells.RDS"),
              zerkaloDir = "mandrian_murine_RT"
              )

# Save as singleCellExperiment object
scRNAseq_integrated_R26_sce <- as.SingleCellExperiment(x = scRNAseq_Seurat_integrated_R26_tumCells)

saveRDS(object = scRNAseq_integrated_R26_sce,
        file   = paste0("scRNAseq_integrated_R26_sce.rds"
                        )
        )

```

## Export matrix and metadata for data deposition (Lobon-Iglesias et al., 2022)

```{r umap_and_covariates}


```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_nbUMIs.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "nFeature_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_nbGenes.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "nCount_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_nbRNA_per_gene.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "nCount_per_Feature",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_percMito.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "percent.mito",
            reduction = "umap"
            )  +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_cellCyclePhase.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "Phase",
        label       = FALSE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

```

## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_clust_nbUMIs.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = log10(nCount_RNA),
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = log10(nCount_RNA)
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "log10(UMI count)\n")+
    ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_clust_nbGenes.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = log10(nFeature_RNA),
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = log10(nFeature_RNA)
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "log10(nbGenes)\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nCount_per_Feature,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nCount_per_Feature
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "UMIs per gene\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_clust_percMito.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = percent.mito,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = percent.mito
                               ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "% mito genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.25
min_diffPct      <- -Inf
min_cellsFeature <- 3
pval_thresh      <- 0.01 

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_R26_tumCells.markers <- FindAllMarkers(object            = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                  assay             = "RNA",
                                                                  min.pct           = min_pct,
                                                                  logfc.threshold   = min_logfc,
                                                                  min.diff.pct      = min_diffPct,
                                                                  only.pos          = TRUE,
                                                                  min.cells.feature = min_cellsFeature,
                                                                  return.thresh     = pval_thresh
                                                                  )

head(scRNAseq_Seurat_integrated_R26_tumCells.markers)

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_R26_tumCells.markers2 <- FindAllMarkers(object            = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                  assay             = "RNA",
                                                                  min.pct           = min_pct,
                                                                  logfc.threshold   = min_logfc,
                                                                  min.diff.pct      = min_diffPct,
                                                                  only.pos          = FALSE,
                                                                  min.cells.feature = min_cellsFeature,
                                                                  return.thresh     = pval_thresh
                                                                  )
head(scRNAseq_Seurat_integrated_R26_tumCells.markers2)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_R26_tumCells_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10,
              face = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_integrated_R26_tumCells.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_integrated_R26_tumCells.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_tumCells,
          features = scRNAseq_Seurat_integrated_R26_tumCells.markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_tumCells,
          features = nTop_markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=15))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_R26_tumCells_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_integrated_R26_tumCells.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_integrated_R26_tumCells.markers,
            file        = paste0(pathToReport,"scRNAseq_integrated_R26_tumCells_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

# Copy table to zerkalo
copyToZerkalo(file = paste0(pathToReport,"scRNAseq_integrated_R26_tumCells_cluster_geneMarkers.txt"),
              zerkaloDir = "mandrian_murine_RT"
              )

```


```{r heatmapOfHumanCALmarkers}

# Import human CAL-ATRT markers
scRNAseq_CAL_integrated_markers <- read.table(file        = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_integrated_cluster_geneMarkers.txt",
                                                                sep         = "\t"
                                                                )

head(scRNAseq_CAL_integrated_markers)
nrow(scRNAseq_CAL_integrated_markers)

# Add mouse orthologous genes
head(symbols_conversion)
scRNAseq_CAL_integrated_markers_plus <- merge(x     = scRNAseq_CAL_integrated_markers,
                                              y     = symbols_conversion,
                                              by.x  = "gene",
                                              by.y  = "HGNC.symbol",
                                              all   = FALSE
                                              )

head(scRNAseq_CAL_integrated_markers_plus)

# Heatmap of all CAL-ATRT markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_CALclusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_tumCells,
          features = scRNAseq_CAL_integrated_markers_plus$MGI.symbol,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Select CAL top markers for each cluster
CAL_nTop_markers <- scRNAseq_CAL_integrated_markers_plus %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

head(CAL_nTop_markers)

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_clusterCALTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_tumCells,
          features = CAL_nTop_markers$MGI.symbol,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=15))
#dev.off()

head(scRNAseq_Seurat_integrated_R26_tumCells_matrix[,1:4])
nrow(scRNAseq_Seurat_integrated_R26_tumCells_matrix[,1:4])

# Extract matrix
scRNAseq_Seurat_integrated_R26_tumCells_matrix_allGenes <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                slot = "data"
                                                                )
                                                                )

dim(scRNAseq_Seurat_integrated_R26_tumCells_matrix_allGenes)
head(scRNAseq_Seurat_integrated_R26_tumCells_matrix_allGenes[,1:4])

length(scRNAseq_CAL_integrated_markers_plus$MGI.symbol)

scRNAseq_Seurat_integrated_R26_tumCells_matrix_CALmarkers <- scRNAseq_Seurat_integrated_R26_tumCells_matrix_allGenes[rownames(scRNAseq_Seurat_integrated_R26_tumCells_matrix_allGenes) %in% scRNAseq_CAL_integrated_markers_plus$MGI.symbol,]

dim(scRNAseq_Seurat_integrated_R26_tumCells_matrix_CALmarkers)

Heatmap(matrix = scRNAseq_Seurat_integrated_R26_tumCells_matrix_CALmarkers,
        show_row_dend = FALSE,
        show_column_dend = FALSE,
        show_row_names = FALSE,
        show_column_names = FALSE
        )

```

## Expression of interested genes

### SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "Smarcb1",
            pt.size  = 1
            ) +
    ggtitle(label = "Smarcb1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_tumCells_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
        features = "Smarcb1",
        pt.size = 0
        ) +
    ggtitle(label = "Smarcb1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

### Interested gene expression

```{r interested_expression}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_tumCells) <- "RNA"


# Expression of genes of interested in the UMAP embedding
pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_interestedGenes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = c("Rest","Id1","Id2","Dcx","Nes","Stmn2"),
            pt.size  = 1,
            slot = "data",
            ncol = 3
            )# +
    #ggtitle(label = "Smarcb1") +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
dev.off()

# Violin plot of interested genes 
pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_tumCells_interestedGenes.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
        features = c("Rest","Id1","Id2","Dcx","Nes","Stmn2"),
        pt.size = 0,
        slot = "data",
        ncol = 3
        ) +
    #ggtitle(label = "Smarcb1") +
    xlab("\nCluster")# +
    #theme_bw() +
    #theme(legend.position = "none",
          #text = element_text(size = 20),
          #plot.title = element_text(hjust = 0.5, face = "bold")
          #)
dev.off()

```

## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes  = c("hgnc_symbol"),
                             mart        = human,
                             attributesL = "mgi_symbol",
                             martL       = mouse,
                             uniqueRows  = TRUE
                             )

head(symbols_conversion)

```

### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}


# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

##### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# Change default assay to RNA (some genes cannot find in integrated)
DefaultAssay(scRNAseq_Seurat_integrated_R26_tumCells) <- "integrated"#"RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

# Select SHH genes with heterogene expression
geneSet_ATRT_Torchia2016_SHH_genes_heteroExp <- c("ASCL1","STMN4","DLL3","TTYH1","FABP7","HES6","HES5")#"HES1"

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Torchia2016_SHH_genes_heteroExp.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 3
            )
#dev.off()

```

##### ATRT SHH specific TFs (Johann et al, 2016: Table S6)

```{r SHH_markers_expression_Torchia2016}

# Select SHH TFs
geneSet_ATRT_Johann2016_SHH_TFs <- filter(.data = geneSet_ATRT,
                                          source == "Johann et al., 2016",
                                          Group == "SHH"
                                          )
geneSet_ATRT_Johann2016_SHH_TFs$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_tumCells) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
        features = geneSet_ATRT_Johann2016_SHH_TFs$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

# Select SHH genes with heterogene expression
geneSet_ATRT_Johann2016_SHH_TFs_heteroExp <- "INSM1"

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_ATRT_Johann2016_SHH_TFs_heteroExp.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = geneSet_ATRT_Johann2016_SHH_TFs_heteroExp,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 1
            )
#dev.off()

```

##### ATRT SHH signature genes from Torchia et al., 2016 and Johann et al., 2016 to show intratumoral heterogeneity of ATRT-SHH expression profile

```{r SHH_markers_expression_heterogeneity}

geneSet_ATRT_SHH_genes_heteroExp <- c("Ascl1","Stmn4","Dll3","Ttyh1","Fabp7","Hes6","Hes5","Insm1")

# Change default assay to RNA (some genes cannot find in integrated)
DefaultAssay(scRNAseq_Seurat_integrated_R26_tumCells) <- "RNA"

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_murine_RT_scRNAseq_integrated_ATRT_Torchia2016_Johann2016_SHH_genes_heteroExp.pdf"), width = 22, height = 8)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            #features = c(geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,geneSet_ATRT_Johann2016_SHH_TFs_heteroExp),
            features = geneSet_ATRT_SHH_genes_heteroExp,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 4
            )
#dev.off()

umap_Torchia2016_Johann2016_SHH_genes <- FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
                                        #features = c(geneSet_ATRT_Torchia2016_SHH_genes_heteroExp,geneSet_ATRT_Johann2016_SHH_TFs_heteroExp),
                                                     features = geneSet_ATRT_SHH_genes_heteroExp,
                                                     pt.size  = 0.1,
                                                     slot     = "data",
                                                     cols     = c("gray","yellow","brown1","brown4"),
                                                     ncol     = 4
                                                     ) +
    plot_layout(guides = "collect") &
    theme(text = element_text(size = paperFig_fontsize),
          axis.text = element_text(size = paperFig_fontsize),
          plot.title = element_text(size = paperFig_fontsize*1.5, face = "bold")
          )
umap_Torchia2016_Johann2016_SHH_genes

# Export table for source data
write.csv(x = cbind(scRNAseq_Seurat_integrated_R26_tumCells@reductions$umap@cell.embeddings,
                    t(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells@assays$RNA@data[geneSet_ATRT_SHH_genes_heteroExp,]
                                )
                      )
                    ),
          file = "../LobonIglesias_sourceData/SuppFig6_A.csv",
          row.names = TRUE
          )

```

# Cerebro analysis (for visualization purpose)

## Copy seurat object

```{r export_to_cerebro_format}

scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- scRNAseq_Seurat_integrated_R26_tumCells
scRNAseq_Seurat_integrated_R26_tumCells_cerebro

head(scRNAseq_Seurat_integrated_R26_tumCells_cerebro@meta.data)

```

## Clusters relationship trees

```{r export_to_cerebro_format}

# Create vectors of clustering resolutions
groups <- paste0('integrated_snn_res.',clustering_resolutions)
groups <- c(groups,'seurat_clusters','orig.ident')

# Build phylogenetic tree for clusters
for (res in groups[-1])
{
    print(res)
    Idents(scRNAseq_Seurat_integrated_R26_tumCells_cerebro) <- res
    scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- BuildClusterTree(object           = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                               dims             = 1:PCmin,
                                                               reorder          = FALSE,
                                                               reorder.numeric  = FALSE
                                                               )
    scRNAseq_Seurat_integrated_R26_tumCells_cerebro@misc$trees[[res]] <- scRNAseq_Seurat_integrated_R26_tumCells_cerebro@tools$BuildClusterTree
}

```

## Compute percentage of mitochondrial and ribosomal transcripts

```{r cerebro_mito_ribo}

# % of mito and ribo 
scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- addPercentMtRibo(object             = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                                    organism           = 'mm',
                                                                    gene_nomenclature  = 'name'
                                                                    )

```

## Get most expressed genes

```{r cerebro_most_expressed_genes}

# Most expressed genes
scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- getMostExpressedGenes(object  = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                                         assay   = 'RNA',
                                                                         groups  = groups
                                                                         )

```

## Get marker genes

```{r cerebro_marker_genes}

min_pct <- 0.1
min_logfc <- 0.25
pval_thresh <- 0.05

# Mark genes
scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- getMarkerGenes(object        = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                                  assay         = 'RNA',
                                                                  organism      = 'mm',
                                                                  groups        = groups,
                                                                  min_pct       = min_pct,
                                                                  thresh_logFC  = min_logfc,
                                                                  thresh_p_val  = pval_thresh,
                                                                  test          = "wilcox",
                                                                  name          = 'R26_integrated_tumCells',
                                                                  only_pos      = TRUE
                                                                  )

```

## Perform pathway enrichment on marker genes

```{r cerebro_most_expressed_genes}

# Pathway analysis
scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- getEnrichedPathways(object             = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                                       marker_genes_input = 'R26_integrated_tumCells',
                                                                       adj_p_cutoff       = 0.01,
                                                                       max_terms          = 100
                                                                       )

```

## Gene set enrichment

```{r cerebro_gene_set_enrichment}


# Pathway analysis
scRNAseq_Seurat_integrated_R26_tumCells_cerebro <- performGeneSetEnrichmentAnalysis(object  = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                                                      assay                = "RNA",
                                                      GMT_file             = paste0(pathToData,"msigdb.v5.2.symbols_mouse.gmt"),
                                                      groups               = groups
                                                      )

```


## Export to cerebro format
  
```{r export_to_cerebro_format}

colnames(scRNAseq_Seurat_integrated_R26_tumCells_cerebro@meta.data)

# Export to cerebro format
exportFromSeurat(object             = scRNAseq_Seurat_integrated_R26_tumCells_cerebro,
                 assay              = 'RNA',
                 slot               = 'data',
                 file               = paste0(pathToReport,'scRNAseq_Seurat_integrated_R26_tumCells.crb'),
                 experiment_name    = 'scRNAseq_R26P87',
                 organism           = 'mm',
                 groups             = groups,
                 nUMI               = 'nCount_RNA',
                 nGene              = 'nFeature_RNA',
                 add_all_meta_data  = TRUE,
                 verbose            = FALSE
                 )

# Copy to zerkalo
copyToZerkalo(file       = paste0(pathToReport,'scRNAseq_Seurat_integrated_R26_tumCells.crb'),
              zerkaloDir = "mandrian_murine_RT"
              )

```



## Expression of interested genes

### SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
            features = "Smarcb1",
            pt.size  = 1
            ) +
    ggtitle(label = "Smarcb1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_tumCells,
        features = "Smarcb1",
        pt.size = 0
        ) +
    ggtitle(label = "Smarcb1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```


# Pseudo-bulk analysis

## Pseudo-bulk for each Patient

```{r PseudoBulk}

head(scRNAseq_Seurat_integrated_R26_tumCells@meta.data$orig.ident)

# Subset Seurat object for each patient
scRNAseq_Seurat_integrated_R26_tumCells_R26P87 <- subset(scRNAseq_Seurat_integrated_R26_tumCells,
                                                         orig.ident == "R26P87"
                                                         )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P27 <- subset(scRNAseq_Seurat_integrated_R26_tumCells,
                                                         orig.ident == "R26P27"
                                                         )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P26 <- subset(scRNAseq_Seurat_integrated_R26_tumCells,
                                                         orig.ident == "R26P26"
                                                         )

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_R26P87_counts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_R26P87,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "counts"#"scale.data"
                                                                                       )
                                                                          )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P27_counts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_R26P27,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "counts"#"scale.data"
                                                                                       )
                                                                          )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P26_counts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_R26P26,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "counts"#"scale.data"
                                                                                       )
                                                                          )


# Create pseudo-bulk by summing the raw counts of all cells
scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix <- data.frame(InnovRT003 = rowSums(scRNAseq_Seurat_integrated_R26_tumCells_R26P87_counts_matrix))
scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix <- rownames_to_column(.data = scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix,
                                                                                  var = "gene"
                                                                                  )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix <- data.frame(InnovRT003 = rowSums(scRNAseq_Seurat_integrated_R26_tumCells_R26P27_counts_matrix))
scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix <- rownames_to_column(.data = scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix,
                                                                                  var = "gene"
                                                                                  )
#
scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix <- data.frame(InnovRT003 = rowSums(scRNAseq_Seurat_integrated_R26_tumCells_R26P26_counts_matrix))
scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix <- rownames_to_column(.data = scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix,
                                                                                  var = "gene"
                                                                                  )


# Check the density distribution of pseudo counts
ggplot() +
    geom_density(data     = scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix,
                 mapping  = aes(x = log2(InnovRT003 + 1)),
                 color    = "blue",
                 size     = 1.5
                 ) +
    geom_density(data     = scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix,
                 mapping  = aes(x = log2(InnovRT003 + 1)),
                 color    = "red",
                 size     = 1.5
                 ) +
    geom_density(data     = scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix,
                 mapping  = aes(x = log2(InnovRT003 + 1)),
                 color    = "darkgreen",
                 size     = 1.5
                 ) +
    theme_bw()

# Extract cell IDs
write.csv(x          = scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix,
          file       = "scRNAseq_Seurat_integrated_R26_tumCells_R26P87_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )
#
write.csv(x          = scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix,
          file       = "scRNAseq_Seurat_integrated_R26_tumCells_R26P27_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )
#
write.csv(x          = scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix,
          file       = "scRNAseq_Seurat_integrated_R26_tumCells_R26P26_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )

```

## Pseudo-bulk for each cluster

```{r PseudoBulk}

levels(scRNAseq_Seurat_integrated_R26_tumCells@meta.data$integrated_snn_res.0.4)


# Subset Seurat object for each patient
scRNAseq_Seurat_integrated_R26_tumCells_cluster_list <- list()
for (cluster in levels(scRNAseq_Seurat_integrated_R26_tumCells@meta.data$integrated_snn_res.0.4))
{
    seuratObject <- subset(scRNAseq_Seurat_integrated_R26_tumCells,
                           integrated_snn_res.0.4 == cluster
                           )
    scRNAseq_Seurat_integrated_R26_tumCells_cluster_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_cluster_list)+1]] <- seuratObject
}
names(scRNAseq_Seurat_integrated_R26_tumCells_cluster_list) <- levels(scRNAseq_Seurat_integrated_R26_tumCells@meta.data$integrated_snn_res.0.4)

names(scRNAseq_Seurat_integrated_R26_tumCells_cluster_list[1])

# Extract expression matrix and create pseudo counts for each cluster and store them in a list
scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_cluster_list))
{
   pseudoCount <- as.matrix(GetAssayData(object  = scRNAseq_Seurat_integrated_R26_tumCells_cluster_list[[i]],
                                         assay   = "RNA",#"integrated",
                                         slot    = "counts"#"scale.data"
                                         )
                            )
   pseudoCount_matrix <- data.frame(rowSums(pseudoCount))
   colnames(pseudoCount_matrix) <- paste0("R26_cluster_",names(scRNAseq_Seurat_integrated_R26_tumCells_cluster_list[i]))
   scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_list)+1]] <- pseudoCount_matrix
}

# Convert list into data frame
scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df <- do.call(cbind.data.frame,scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_list)

head(melt(scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df))

# Check the density distribution of pseudo counts
melt(scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df) %>%
    ggplot() +
    geom_density(mapping  = aes(x = log2(value + 1), color = variable),
                 size     = 1.5
                 ) +
    scale_color_manual(values = c("red","blue","green","yellow","gray","black","darkgreen","orange","darkblue","cyan","violet","brown","wheat3")) +
    theme_bw() +
    theme(text = element_text(size = 20))

# Convert rownames into column
scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df <- rownames_to_column(.data = scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df,
                                                                                      var = "gene"
                                                                                      )

head(scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df)

# Extract cell IDs
write.csv(x          = scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_df,
          file       = "scRNAseq_Seurat_integrated_R26_tumCells_cluster_pseudoCounts_matrix.csv",
          row.names  = FALSE
          )

```



































## ATRT SSH signature genes expression

### Import signature genes

```{r import_geneSet}

# Import genSet
geneSet <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/geneSet.csv",
                      header  = TRUE,
                      sep     = ";"
                      )
head(geneSet)

# Gene markers per cell type
group_by(.data = geneSet,
         #,Cell.type
         ,Group
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

icicicicicicicici
 
### ATRT signature genes

```{r ATRT_signature_genes}

# Select ATRT genes
geneSet_ATRT <- filter(.data    = geneSet,
                       Interest == "ATRT"
                       )
geneSet_ATRT

# Gene markers per cell type
group_by(.data = geneSet_ATRT,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### ATRT signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}


# Select ATRT  Torchia et al. genes
geneSet_ATRT_Torchia2016 <- filter(.data  = geneSet_ATRT,
                                   source == "Torchia et al., 2016"
                                   )
geneSet_ATRT_Torchia2016

# Gene markers per cell type
group_by(.data = geneSet_ATRT_Torchia2016,
         #,Cell.type
        ,Group
        ,source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

##### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

# Select restricted SHH gene signatures
geneSet_ATRT_Torchia2016_SHH_genes_restricted <- c("ASCL1","STMN4","DLL3","TTYH1") #HES6

# ATRT SHH restricted signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_ATRT_Torchia2016_SHH_genes_restricted.pdf"), width = 14, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = geneSet_ATRT_Torchia2016_SHH_genes_restricted,
            pt.size  = 0.25,
            slot     = "data",
            cols     = c("gray","yellow","brown1","brown4"),
            ncol     = 2
            )
#dev.off()

```

## Cerebellum marker genes expression

### Import and explore cell types database

```{r importCellTypeMarkers}

# Import cannonical and published cell type markers
cellType_markers <- read.table(file    = "/data/users/mandrian/rhabdoid_U830_projects/annotation_file/cellType_markers.csv",
                               header  = TRUE,
                               sep     = "\t",
                               check.names = FALSE
                               )
head(cellType_markers)

# Gene markers per cell type
group_by(.data = filter(cellType_markers,
                        Group == "cerebellum"
                        )
         #,Cell.type
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

### Cerebellum marke genes expression from Wizeman et al., 2019

#### Select Wizeman cerebellar cell types marker

```{r importCellTypeMarkers}

cellType_markers_Wizeman2019 <- filter(.data  = cellType_markers,
                                       Group  == "cerebellum",
                                       Source == "Wizeman et al., 2019"
                                       )

# Gene markers per cell type
group_by(.data = cellType_markers_Wizeman2019
         ,`Cell type`
         ,Source
         ) %>%
    summarise(number = n()) %>%
    print(n=nrow(.), width = Inf)

```

#### Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Convert human murine symbol to human symbol
head(cellType_markers_Wizeman2019)

cellType_markers_Wizeman2019_plus <- merge(x      = cellType_markers_Wizeman2019,
                                           y      = symbols_conversion,
                                           by.x   = "Gene",
                                           by.y   = "MGI.symbol",
                                           all.x  = TRUE,
                                           all.y  = FALSE
                                           )
head(cellType_markers_Wizeman2019_plus)

```

#### Expression of cerebellar marker genes (Wezman et al., 2019)

##### Roof Plate

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RoofPlate <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Roof Plate"
                                                  )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 14)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```


##### Rhombic Lip

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_RhombicLip <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                  `Cell type` == "Rhombic Lip"
                                                  )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### Ventricular Zone

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_VentricularZone <- filter(.data       = cellType_markers_Wizeman2019_plus,
                                                       `Cell type` == "Ventricular Zone"
                                                       )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Bergmann Glia

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_BergamannGlia <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                     `Cell type` == "Bergmann Glia"
                                                     )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### GABAergic

```{r cerebellar_marker_genes_expression_Wezman2019}

# Select Bergmann Glia 
cellType_markers_Wizeman2019_GABAergic <- filter(.data = cellType_markers_Wizeman2019_plus,
                                                 `Cell type` == "GABAergic"
                                                 )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26,
        features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

## Cluster marker genes identified by Maria (see e-mail from Maria on March 3, 2021)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes <- c("CNPY1","EPHB1","RUNX1T1","EN2","IRX1","IRX2","OTX2","HES3","FGF8","FGF19","FZD10","WNT3A")

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_clusterMarkersMaria_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26,
            features = clusterMarkerGenes,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","brown1","brown4"),
            combine  = TRUE
            ) #+
    #theme(axis.line = element_blank(),
     #     axis.text.x = element_blank(),
      #    axis.text.y = element_blank(),
       #   axis.ticks = element_blank(),
        #  axis.title.x = element_blank(),
         # axis.title.y = element_blank()
          #)
#dev.off()

scRNAseq_Seurat_integrated_R26_cellAnnot

```

## Stacked violin plot for marker genes identified by Maria for each cell type (see e-mail from Maria on March 31, 2021 in the Clustering_synthesis_5.xlsx file)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes2 <- c("PCNA","TOP2A","BIRC5","NUSAP1","NSG1","STMN2","LHX9","GAP43","DLL3","TAGLN3","ASCL1","NEUROG1","DLX5","CD82","EDNRB","ZFP36","SPARCL1","GJA1","CYR61")
clusterMarkerGenes3 <- c("FOS","SOCS3","SOX4","SOX11", #stemness
                         "NRG1","CD82","NKAIN4", #astro like ,"GJA11","EDNRB1","EDNRB2"
                         "DLL3","PCP4","SYT1","ELAVL4","DLX6",#neuronal ,"ELAVL31","ELAV41"
                         "DNER","JAG1",#active notch pathway
                         "MCM3","MCM3","MCM3","MCM3","CDC20","CCNB1","DLGAP5"#cell cylce
                         )

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_cellAnnot) <- "RNA"

# Set default ident
Idents(scRNAseq_Seurat_integrated_R26_cellAnnot) <- scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data[["cell_type"]]

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"stackedVlnPlot_scRNAseq_integrated_R26_clusterMarkersMaria_genes.pdf"), width = 14, height = 14)
StackedVlnPlot(obj = scRNAseq_Seurat_integrated_R26_cellAnnot,
               features = c(clusterMarkerGenes2,clusterMarkerGenes3),
               #plot.margin = unit(c(0,0,0,0), "cm"),
               group.by = "cell_type"
               )
#dev.off()

```

## UMAP showing marker genes identified by Maria for each cell type (see e-mail from Maria on March 31, 2021 in the Clustering_synthesis_5.xlsx file)

```{r cluster_marker_genes_identified_by_Maria}

# Marker genes 
clusterMarkerGenes2_min <- c("PCNA","TOP2A","STMN2","DLL3","SPARCL1","ZFP36","CYR61","EDNRB")

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_cellAnnot) <- "RNA"

# Cell type marker gene gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_clusterMarkerGenes2_min.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_cellAnnot,
            features = clusterMarkerGenes2_min,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 4,
            cols     = c("gray","brown1","brown4"),
            combine  = TRUE
            ) #+
    #theme(axis.line = element_blank(),
     #     axis.text.x = element_blank(),
      #    axis.text.y = element_blank(),
       #   axis.ticks = element_blank(),
        #  axis.title.x = element_blank(),
         # axis.title.y = element_blank()
          #)
#dev.off()

```


## SCENIC Analysis (PCmin)

### Cellular enrichment of regulons using AUC score

#### Import AUC score matrix

```{r import_SCENIC_AUCscore}

# Path to SCENIC result
pathToSCENICresult <- paste0(pathToThisProject,"svn/analyse/script/scRNAseq/SCENIC_integrated_R26/")

# Import SCENIC AUC score
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_scenic_auc_matrix.csv"),
                                                       check.names = FALSE
                                                       )

head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)

# Remove "-1" suffix of cell IDs (rownames)
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix) <- gsub(pattern = "-1_R26P87",
                                                                     replacement = "_1",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
                                                                     )
#
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix) <- gsub(pattern = "-1_R26P27",
                                                                     replacement = "_2",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
                                                                     )
#
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix) <- gsub(pattern = "-1_R26P26",
                                                                     replacement = "_3",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
                                                                     )


head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
tail(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)

# Transpose AUC matrix
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t <- t(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix)
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t)
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t)

# Density distribution
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_melt <- melt(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t)
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_melt)

AUCthres <- 0.01
ggplot(data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_melt) +
    geom_density(mapping = aes(x = value
                               ),
                 fill = "grey"
                 ) +
    geom_vline(xintercept=AUCthres, color = "red") +
    theme_bw()

# Filtering
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt <- expFilter(data      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t,
                                                                       threshold = AUCthres,
                                                                       p         = 0.05,
                                                                       graph     = TRUE
                                                                       )

dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt)

```

#### Heatmap of AUC score

```{r SCENIC_AUCscore_heatmap}

# Most variable genes
mvRegulons <- genes.selection(data      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt,
                              thres.num = 50
                              )

# Subset matrix
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt_mvRegulons <- scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt[mvRegulons,]

head(scRNAseq_Seurat_integrated_R26_tumCells@meta.data)
tail(scRNAseq_Seurat_integrated_R26_tumCells@meta.data)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells@meta.data) == colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt_mvRegulons))

# Top annotation
topAnnot_PCmin <- HeatmapAnnotation(df                      = data.frame(Cluster = as.character(scRNAseq_Seurat_integrated_R26_tumCells@meta.data[,"integrated_snn_res.0.4"]), check.names = FALSE),
                                    col                     = list(Cluster = cluster_n13_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(2,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,0.2,0.4,0.6), colors = c("cyan","wheat1","red","darkred"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt_mvRegulons,
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_split                = 4,
             row_split                   = 4,
             col                         = col_fun,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "AUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_SCENIC_AUCscore.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

#### UMAP of cells based on AUC score

```{r AUCscore_umap}

# Compute UMAP
scRNAseq_integrated_R26_tumCells_scenic_auc_umap <- data.frame(umap(t(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_t_filt_mvRegulons)))
head(scRNAseq_integrated_R26_tumCells_scenic_auc_umap)

colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_umap) <- c("UMAP1","UMAP2")
head(scRNAseq_integrated_R26_tumCells_scenic_auc_umap)

# Bind meta data
scRNAseq_integrated_R26_tumCells_scenic_auc_umap_plus <- cbind(scRNAseq_integrated_R26_tumCells_scenic_auc_umap,
                                                       cluster = as.character(scRNAseq_Seurat_integrated_R26_tumCells@meta.data$RNA_snn_res.0.4)
                                                       )

head(scRNAseq_integrated_R26_tumCells_scenic_auc_umap_plus)

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_SCENIC_AUCscore.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_integrated_R26_tumCells_scenic_auc_umap_plus,
       mapping = aes(x = UMAP1, y = UMAP2)
       ) +
    geom_point(mapping = aes(color = cluster
                             ),
               size = 2
               )+
    scale_color_manual(values = cluster_n14_col[levels(factor(scRNAseq_integrated_R26_tumCells_scenic_auc_umap_plus$cluster))]) +
    xlab("UMAP1") +
    ylab("UMAP2") +
    ggtitle(NULL) +
    theme_bw()+
    theme(text               = element_text(size = 20),
          legend.position    = "none"
          #legend.title       = element_text(face = "bold"),
          #legend.key.height  = unit(1.5,"cm")
          )
#dev.off()

```	

### Cellular enrichment of regulons using binarized AUC score

#### Import binarized AUC score matrix

```{r import_SCENIC_bin_AUCscore}

# Import SCENIC AUC score
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_scenic_matrix_bin.csv"),
                                                           check.names = FALSE
                                                           )

head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)

# Remove "-1" suffix of cell IDs (rownames)
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin) <- gsub(pattern = "-1_R26P87",
                                                                     replacement = "_1",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)
                                                                     )
#
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin) <- gsub(pattern = "-1_R26P27",
                                                                     replacement = "_2",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)
                                                                     )
#
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin) <- gsub(pattern = "-1_R26P26",
                                                                     replacement = "_3",
                                                                     rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)
                                                                     )

# Transpose AUC_BIN matrix
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t <- t(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin)
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t)
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t)

# Density distribution
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_melt <- melt(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t)
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_melt)

scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_melt <- mutate(.data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_melt,
                                                       value = as.character(value)
                                                       )

ggplot(data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_melt) +
    geom_bar(mapping = aes(x = value
                                 ),
                   stat = "count",
                   fill = "grey"
             ) +
    xlab("\nbinarized AUC score") +
    ylab("Count\n") +
    theme_bw()

```

#### Filtering

```{r AUC_bin_filtering}

# Remove all lines where counts are all equal to zero
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ <- scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t[apply(X = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t,
                                                                                            MARGIN = 1,
                                                                                            FUN = function(x) !all(x==0)),
                                                                                      ]
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t)
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ)

# Filtering
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt <- expFilter(data      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ,
                                                                              threshold = 0.5,
                                                                              p         = 0.05,
                                                                              graph     = TRUE
                                                                              )
dim(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt)

```

#### Heatmap of AUC_BIN score

```{r SCENIC_AUC_BINscore_heatmap}

# Most variable genes
mvRegulons_auc_bin <- genes.selection(data      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt,
                                      thres.num = 50
                                      )

# Subset matrix
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons <- scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt[mvRegulons_auc_bin,]

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells@meta.data) == colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons))

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,1), colors = c("white","black"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons,
             clustering_distance_rows    = "binary",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_row_dend               = FALSE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PCmin,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated_R26_tumCells@meta.data[,"integrated_snn_res.0.4"])),
             column_split                = factor(scRNAseq_Seurat_integrated_R26_tumCells@meta.data[,"integrated_snn_res.0.4"]),
             #row_split                   = 2,
             col                         = col_fun,
             show_heatmap_legend         = FALSE
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_SCENIC_AUCbinScore.pdf"), width = 12, height = 12)
draw(h)
#dev.off()

```

#### Heatmap of the percentage of cells with activated regulon

```{r heatmap_perc_active_regulon}

# Melt matrix
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt <- melt(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons)
colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt) <- c("regulon","cellID","AUCbin")
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt)

# Merge matrix and meta data
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus <- merge(x     = scRNAseq_Seurat_integrated_R26_tumCells@meta.data["integrated_snn_res.0.4"],
                                                                                       y     = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt,
                                                                                       by.x  = 0,
                                                                                       by.y  = "cellID",
                                                                                       all   = FALSE
                                                                                       )
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus)

# Create table of cluster, regulon and number of cells
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells <- group_by(.data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus,
                                                                                          integrated_snn_res.0.4,
                                                                                          regulon
                                                                                          ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells) = c("cluster","regulon","nbCell")
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells)

# Create table of cluster, regulon and number of cells in which regulon is activated
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells <- filter(.data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus, AUCbin == 1) %>%
    group_by(RNA_snn_res.0.4,
             regulon
             ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells) = c("cluster","regulon","nbCell_AUCbin1")
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells)

# Create a table of total number of cells and cells in which regulon is activated
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells <- merge(x = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells,
                                                                                                   y = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells,
                                                                                                   by = c("cluster","regulon"),
                                                                                                   all = FALSE
                                                                                                   )
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Compute percentage of cells with active regulon
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$activeRegulon <- scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell_AUCbin1*100/scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell

head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Transform table to matrix (for ComplexHeatmap function)
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix <- spread(data = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells[,c("cluster","regulon","activeRegulon")],
                                                                                                 key = cluster,
                                                                                                 value = activeRegulon
                                                                                                 )
rownames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix) <- scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon
scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon <- NULL
head(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix)

# Top annotation
topAnnot_cluster <- HeatmapAnnotation(cluster                  = colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
                                      text                     = anno_text(x= colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix), rot = 0),
                                      col                      = list(cluster = cluster_n13_col),
                                      gap                      = unit(4,"mm"),
                                      show_annotation_name     = FALSE,
                                      annotation_name_gp       = gpar(fontsize = 20),
                                      annotation_legend_param  = list(fontsize   = 20,
                                                                      title_gp    = gpar(fontsize = 20, font = 2),
                                                                      labels_gp   = gpar(fontsize = 20),
                                                                      grid_width  = unit(12,"mm"),
                                                                      grid_height = unit(12,"mm"),
                                                                      ncol        = 1
                                                                      ),
                                      simple_anno_size        = unit(2,"cm"),
                                      na_col                  = "white",
                                      show_legend             = FALSE
                                      )

# Heatmap
h <- Heatmap(matrix                    = scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix,
             cluster_columns           = FALSE,
             clustering_distance_rows  = "euclidean",
             clustering_method_rows    = "average",
             top_annotation            = topAnnot_cluster,
             show_row_dend             = FALSE,
             column_title              = "cluster\n",
             row_title                 = "Top 50 most variable regulons",
             column_split              = colnames(scRNAseq_integrated_R26_tumCells_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
             show_column_names         = FALSE,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(2,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n% of cells with active regulon",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter",
                                                direction = "horizontal"
                                                )
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_tumCells_SCENIC_AUCbinScore_perc.pdf"), width = 12, height = 12)
draw(h,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Cell type specific regulators

#### Import SCENIC RSS score

```{r import_SCENIC_AUCscore}

# Import SCENIC RSS score
scRNAseq_integrated_R26_scenic_rss_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_scenic_rss_matrix.csv"),
                                                      check.names = FALSE
                                                      )

scRNAseq_integrated_R26_scenic_rss_matrix[,1:4]
dim(scRNAseq_integrated_R26_scenic_rss_matrix)

# Create column of cluster
scRNAseq_integrated_R26_scenic_rss_matrix$cluster <- rownames(scRNAseq_integrated_R26_scenic_rss_matrix)
head(scRNAseq_integrated_R26_scenic_rss_matrix)

# Density distribution
scRNAseq_integrated_R26_scenic_rss_matrix_melt <- melt(scRNAseq_integrated_R26_scenic_rss_matrix)
head(scRNAseq_integrated_R26_scenic_rss_matrix_melt)

ggplot(data = scRNAseq_integrated_R26_scenic_auc_matrix_bin_t_melt) +
    geom_density(mapping = aes(x = value),
                   fill = "grey"
             ) +
    xlab("\nRSS score") +
    ylab("density\n") +
    theme_bw()

# Transpose RSS matrix (to be copied to zerkalo and then explored by Maria)
scRNAseq_integrated_R26_scenic_rss_matrix_t <- t(data.matrix(scRNAseq_integrated_R26_scenic_rss_matrix))
head(scRNAseq_integrated_R26_scenic_rss_matrix_t)
scRNAseq_integrated_R26_scenic_rss_matrix_t <- data.frame(scRNAseq_integrated_R26_scenic_rss_matrix_t,
                                                      check.names = FALSE
                                                      )

scRNAseq_integrated_R26_scenic_rss_matrix_t$TF <- rownames(scRNAseq_integrated_R26_scenic_rss_matrix_t)

head(scRNAseq_integrated_R26_scenic_rss_matrix_t)

# Export (to be copied to zerkalo)
write.table(x = scRNAseq_integrated_R26_scenic_rss_matrix_t,
            file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_scenic_rss_matrix_t.csv"),
            quote = FALSE,
            sep = ";",
            dec =",",
            row.names = FALSE
            )

# Copy to zerkalo
copyToZerkalo(file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_scenic_rss_matrix_t.csv"),
              zerkaloDir = "mandrian_murine_RT"
              )

```

#### Plot RSS score

```{r import_SCENIC_AUCscore}

# Convert variable column (regulon) into factor
scRNAseq_integrated_R26_scenic_rss_matrix_melt$variable <- factor(scRNAseq_integrated_R26_scenic_rss_matrix_melt$variable)
head(scRNAseq_integrated_R26_scenic_rss_matrix_melt)

levels(factor(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster))

scRNAseq_integrated_R26_scenic_rss_matrix_melt[which(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster == "9"),] %>%
    .[order(.$value, decreasing = TRUE),] %>%
    .[1:5,"variable"]

# Plot RSS per cluster
scRNAseq_integrated_R26_scenic_rss_plot_list = list()
all_top5 <- data.frame(regulon = character(0), cluster = character(0))
for (cluster in levels(factor(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_integrated_R26_scenic_rss_matrix_melt[which(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster == cluster),]
    # Pick top 5 regulons
    top5 <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    #all_top5 <- c(all_top5, as.vector(top5))
    all_top5 <- rbind(all_top5,
                      data.frame(regulon = top5, cluster = cluster)
                      )
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% top5),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% top5),
                        mapping = aes(label = variable),
                        color = "red",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_integrated_R26_scenic_rss_plot_list[[length(scRNAseq_integrated_R26_scenic_rss_plot_list)+1]] <- p
}

library(ggpubr)

#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_SCENIC_RSS.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_integrated_R26_scenic_rss_plot_list, ncol=7, nrow=2)
#dev.off()

all_top5

all_top5[duplicated(all_top5$regulon),]


```

#### Heatmap (AUC score) of top 5 RSS score regulons

```{r import_SCENIC_AUCscore}

# Heatmap of all top5
scRNAseq_integrated_R26_scenic_auc_matrix_t_allTop5 <- scRNAseq_integrated_R26_scenic_auc_matrix_t[as.vector(all_top5$regulon),]
head(scRNAseq_integrated_R26_scenic_auc_matrix_t_allTop5[,1:4])

# Density of scaled UCS score
plot(density(t(scale(t(scRNAseq_integrated_R26_scenic_auc_matrix_t_allTop5)))))

col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0), colors = c("cyan","white","red","darkred"))
#col_fun_AUCscore_scaled = colorRamp2(breaks = c(-1,0.25,0.50,1.0,5), colors = c("blue","cyan","white","yellow","red"))
# Heatmap
h <- Heatmap(matrix                      = t(scale(t(scRNAseq_integrated_R26_scenic_auc_matrix_t_allTop5))),
             clustering_distance_columns = "pearson",
             clustering_method_columns   = "ward",
             clustering_distance_rows    = "euclidean",
             clustering_method_rows      = "ward",
             cluster_rows                = FALSE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Cluster specific regulons",
             top_annotation              = topAnnot_PCmin,
             bottom_annotation           = bottomAnnot_PCmin,
             show_heatmap_legend         = TRUE,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated_R26@meta.data[,"integrated_snn_res.0.3"])),
             column_split                = factor(scRNAseq_Seurat_integrated_R26@meta.data[,"integrated_snn_res.0.3"]),
             row_split                   = all_top5$cluster, #rep(c("0","1","3","4","5","6","7","8","9"), each = 5),
             col                         = col_fun_AUCscore_scaled,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(0.75,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(4,"cm"),
                                                title          = "scaled\nAUC\n score\n",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter"
                                                )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_SCENIC_AUCscore_scaled_allTop5.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

## CHETAH

### Import pre-sampled all developmentall mousebrain reference data from cellOrigin_ATRT project

```{r reference_data}

# Import reference metadata
mousebrain_dev_all_metadata_sampleN <- readRDS(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/mousebrain_dev_all_metadata_sampleN.RDS")

# Import reference matrix
mousebrain_dev_all_sampleN_matrix_T <- readRDS(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/mousebrain_dev_all_sampleN_matrix_T.RDS")

# Create sce object for mousebrain data
mousebrain_dev_all.sce <- SingleCellExperiment(assays = list(counts = mousebrain_dev_all_sampleN_matrix_T),
                                               colData = data.frame(mousebrain_dev_all_metadata_sampleN)
                                               )

mousebrain_dev_all.sce

```

#### Prepare input data

```{r input_data}

# Extract cell matrix 
chetah_input_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                              slot = "data"
                                              )
                                 )
head(chetah_input_matrix[,1:3])

# Extract UMAP coordinates 
chetah_input_umap <- scRNAseq_Seurat_integrated_R26_tumCells@reductions$umap@cell.embeddings
head(chetah_input_umap)

# Create singleCellExperiment object
chetah_input_se <- SingleCellExperiment(assays = list(counts = chetah_input_matrix),
                                        reducedDims = SimpleList(umap = chetah_input_umap)
                                        )

```

### Running CHETAH

```{r running_chetah}

# Run CHETAH using class
chetah_scRNAseq_integrated_vs_mousebrainDev_res <- CHETAHclassifier(input      = chetah_input_se,
                                                                    ref_cells  = mousebrain_dev_all.sce,
                                                                    ref_ct     = "Class"
                                                                    )

#pdf(file = paste0(pathToFigures,"plotCHETAH_scRNAseq_integrated_R26_tumCells_vs_mousbrainDev_class.pdf"), width = 20, height =10) 
PlotCHETAH(input = chetah_scRNAseq_integrated_vs_mousebrainDev_res)
#dev.off()

# Run CHETAH using Tissue
chetah_scRNAseq_integrated_vs_mousebrainDev_res_tissue <- CHETAHclassifier(input     = chetah_input_se,
                                                                           ref_cells = mousebrain_dev_all.sce,
                                                                           ref_ct = "Tissue"
                                                                           )

#pdf(file = paste0(pathToFigures,"plotCHETAH_scRNAseq_integrated_R26_tumCells_vs_mousbrainDev_tissue.pdf"), width = 20, height =10) 
PlotCHETAH(input = chetah_scRNAseq_integrated_vs_mousebrainDev_res_tissue)
#dev.off()

# Run CHETAH using Age
chetah_scRNAseq_integrated_vs_mousebrainDev_res_age <- CHETAHclassifier(input     = chetah_input_se,
                                                                        ref_cells = mousebrain_dev_all.sce,
                                                                        ref_ct    = "Age"
                                                                        )

#pdf(file = paste0(pathToFigures,"plotCHETAH_scRNAseq_integrated_R26_tumCells_vs_mousbrainDev_age.pdf"), width = 20, height =10) 
PlotCHETAH(input = chetah_scRNAseq_integrated_vs_mousebrainDev_res_age)
#dev.off()

```

## Pijuan-Sala et al. 2019 data

```{r import_Pijuan_dataset}

# Import Pijuan mouse single cell dataset
#BiocManager::install("MouseGastrulationData")

library("MouseGastrulationData")

head(AtlasSampleMetadata)
nrow(AtlasSampleMetadata)

```

# Trajectory inference analysis

## Prepare matrix

```{r cellCycle}

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                assay = "integrated",
                                                                slot = "scale.data"
                                                                )
                                                         )
dim(scRNAseq_Seurat_integrated_R26_tumCells_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_matrix[,1:4])

# ------------------------

# Select cluster marker genes
head(scRNAseq_Seurat_integrated_R26_tumCells.markers)

cluster01234679_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(0,1,2,3,4,6,7,9)
                                       )
                                )
head(cluster01234679_DEG)
length(cluster01234679_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_01234679 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,1,2,3,4,6,7,9),
                                                    features = cluster01234679_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_01234679)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_01234679,
                                                                assay = "integrated",
                                                                slot = "scale.data"
                                                                )
                                                         )
dim(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster01279_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(0,1,2,7,9)
                                       )
                                )
head(cluster01279_DEG)
length(cluster01279_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_01279 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,1,2,7,9),
                                                    features = cluster01279_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_01279)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_01279,
                                                                assay = "integrated",
                                                                slot = "scale.data"
                                                                )
                                                         )
dim(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster123469_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(1,2,3,4,6,9)
                                       )
                                )
head(cluster01279_DEG)
length(cluster01279_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_123469 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(1,2,3,4,6,9),
                                                    features = cluster123469_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_123469)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_123469,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster1234679_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(1,2,3,4,6,7,9)
                                       )
                                )
head(cluster1234679_DEG)
length(cluster1234679_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_1234679 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(1,2,3,4,6,7,9),
                                                    features = cluster1234679_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_1234679)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_1234679,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster129_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(1,2,9)
                                       )
                                )
head(cluster129_DEG)
length(cluster129_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_129 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(1,2,9),
                                                    features = cluster129_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_129)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_129_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_129,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster431_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(4,3,1)
                                       )
                                )
head(cluster431_DEG)
length(cluster431_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_431 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(4,3,1),
                                                    features = cluster431_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_431)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_431_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_431,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster012679_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(0,1,2,6,7,9)
                                       )
                                )
head(cluster012679_DEG)
length(cluster012679_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_012679 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,1,2,6,7,9),
                                                    features = cluster012679_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_012679)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_012679,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster01346_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCellsy.markers,
                                       cluster %in% c(0,1,3,4,6)
                                       )
                                )
head(cluster01346_DEG)
length(cluster01346_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_01346 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,1,3,4,6),
                                                    features = cluster01346_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_01346)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_01346,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster079_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(0,7,9)
                                       )
                                )
head(cluster079_DEG)
length(cluster079_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_079 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,7,9),
                                                    features = cluster079_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_079)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_079_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_079,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster12349_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(1,2,3,4,9)
                                       )
                                )
head(cluster12349_DEG)
length(cluster12349_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_12349 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(1,2,3,4,9),
                                                    features = cluster12349_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_12349)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_12349,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix[,1:4])

# ------------------------

# Select cluster marker genes
cluster0346_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_tumCells.markers,
                                       cluster %in% c(0,3,4,6)
                                       )
                                )
head(cluster0346_DEG)
length(cluster0346_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_0346 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                    integrated_snn_res.0.4 %in% c(0,3,4,6),
                                                    features = cluster0346_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_0346)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_0346,
                                                                       assay = "integrated",
                                                                       slot = "scale.data"
                                                                       )
                                                          )
dim(scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix[,1:4])

```


## Monocle version 3 method

### Monocle3 on all clusters

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_matrix)
                                                                               )
                                                    )

#nbPCs <- c(12,seq(10,100,10))
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_cds_plot
}

library(gridExtra)

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 0,1,2,3,4,6,7,9

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_01234679@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_01234679_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 0,1,2,7,9

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_01279@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01279_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_01279_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 123469

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_123469@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_123469_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_123469_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 1234679

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_1234679@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_1234679_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_1234679_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_plot_list, ncol = 3))
#dev.off()

```


### Monocle3 on cluster 129

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_129_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_129@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_129_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_129_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_129_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_129_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_129_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_129_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_129_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_129_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_129_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_129_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_129_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 431

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_431_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_431@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_431_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_431_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_431_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_431_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_431_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_431_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_431_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_431_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_431_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_431_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_431_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 012679

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_012679@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_012679_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_012679_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 01346

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_01346@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01346_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_01346_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 079

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_079_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_079@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_079_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_079_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_079_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_079_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_079_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_079_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_079_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_079_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_079_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_079_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_079_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 12349

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_12349@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_12349_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_12349_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_plot_list, ncol = 3))
#dev.off()

```

### Monocle3 on cluster 0346

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_tumCells_0346@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_tumCells_0346_matrix)
                                                                               )
                                                    )

# Run Monocle3
nbPCs <- c(10,PCmin,30,50,70,90)
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds
                                              )
    scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_0346_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list[[i]],
                                                  color_cells_by                 = "integrated_snn_res.0.4",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list)[i])
scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_tumCells_0346_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_plot_list, ncol = 3))
#dev.off()

```



## ElPiGraph

### ElPiGraph on all clusters

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_matrix),
                                                                             NumNodes    = 30,
                                                                             Lambda      = .03,
                                                                             Mu          = 0.1,
                                                                             Do_PCA      = TRUE,
                                                                             CenterData  = TRUE,
                                                                             drawAccuracyComplexity = FALSE,
                                                                             drawPCAView = FALSE,
                                                                             drawEnergy  = FALSE
                                                                             )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 0,1,2,3,4,6,7,9

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01234679_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01234679@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01234679_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01234679_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01234679_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01234679@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 0,1,2,7,9

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01279_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01279@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01279_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01279_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01279_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01279@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 1,2,3,4,6,9

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_123469_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_123469@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_123469_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_123469_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_123469_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_123469@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 1,2,9

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_129_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_129@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_129_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_129_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_129_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_129@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 4,3,1

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_431_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_431@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_431_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_431_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_431_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_431@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 012679

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_012679_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_012679@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_012679_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_012679_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_012679_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_012679@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 01346

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01346_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01346@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_01346_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_01346_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_01346_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_01346@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 079

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_079_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_079@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_079_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_079_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_079_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_079@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

### ElPiGraph on cluster 12349

##### Principal tree

```{r elpigraph_tree}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix),
                                                                                      NumNodes    = 30,
                                                                                      Lambda      = .03,
                                                                                      Mu          = 0.1,
                                                                                      Do_PCA      = TRUE,
                                                                                      CenterData  = TRUE,
                                                                                      drawAccuracyComplexity = FALSE,
                                                                                      drawPCAView = FALSE,
                                                                                      drawEnergy  = FALSE
                                                                                      )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_12349_allClusters_tree.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_12349@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix),
                                                                           NumNodes               = 30,
                                                                           Lambda                 = .03,
                                                                           Mu                     = 0.1,
                                                                           Do_PCA                 = TRUE,
                                                                           CenterData             = TRUE,
                                                                           drawAccuracyComplexity = FALSE,
                                                                           drawPCAView            = FALSE,
                                                                           drawEnergy             = FALSE
                                                                           )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_tumCells_12349_allClusters_curve.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_tumCells_12349_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_tumCells_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_tumCells_12349_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_tumCells_12349@meta.data$integrated_snn_res.0.4
                )
#dev.off()

```



### Monocle version 2 method

#### Monocle2 on all clusters

```{r monocle_method}

library(monocle)

# Create monocle2 cds object
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_R26_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_R26@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_R26_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )


# Create Monocle2 cds object
#scRNAseq_Seurat_integrated_R26_cds_mncl2 <- as.CellDataSet(x = scRNAseq_Seurat_integrated_R26_cellAnnot,
 #                                                      assay = "RNA"
  #                                                     )

# Choose genes that define a cell's progress
HVF <- VariableFeatures(object = scRNAseq_Seurat_integrated_R26,
                        assay = "integrated"
                        )

scRNAseq_Seurat_integrated_R26_cds_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                                                          ordering_genes = HVF
                                                          )

# Dimensionality reduction
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- reduceDimension(cds                      = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                                                            max_components           = 2,
                                                            norm_method              = "none",
                                                            relative_expr            = TRUE,
                                                            scaling                  = TRUE,
                                                            residualModelFormulaStr  = "~ orig.ident",
                                                            reduction_method         = "DDRTree"
                                                            )

# Order cells along trajectory
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- orderCells(scRNAseq_Seurat_integrated_R26_cds_mncl2)

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_allClusters.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                     color_by = "integrated_snn_res.0.4"
                     )
#dev.off()

```

### CytoTRACE method

CytoTRACE predicts the differentiation state of cells using gene Counts and expression.

#### CytoTRACE all clusters

```{r cytoTrace}

# Extract raw count expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                                   assay  = "RNA",#"integrated",
                                                                                   slot   = "counts"#"scale.data"
                                                                                   )
                                                                      )

head(scRNAseq_Seurat_integrated_R26_tumCells_rawCounts_matrix[,1:4])
nrow(scRNAseq_Seurat_integrated_R26_tumCells_rawCounts_matrix)

# Run CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace <- CytoTRACE(mat            = scRNAseq_Seurat_integrated_R26_tumCells_rawCounts_matrix,
                                                               batch          = scRNAseq_Seurat_integrated_R26_tumCells@meta.data$orig.ident,
                                                               enableFast     = TRUE,
                                                               ncores         = 1,
                                                               subsamplesize  = 1000
                                                               )
str(scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace)

# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_R26_tumCells_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace$CytoTRACE)))

scRNAseq_Seurat_integrated_R26_tumCells_allGenes <- scRNAseq_Seurat_integrated_R26_tumCells

# Append metadata with CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_R26_tumCells_allGenes,
                                                                metadata = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace$CytoTRACE)
                                                                )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_clust_CytoTRACE.pdf"), width = 16, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_tumCells_allGenes,
            pt.size = 1,
            features = "CytoTRACE",
            reduction = "umap",
            cols     = c("red","yellow","grey","cyan","blue"),
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

source("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/scRNAseq_functions.R")

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_integrated_R26_tumCells_allGenes,
                              clustering  = paste0("integrated_snn_res.",choosenResolution),
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "mean"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes <- data.frame(corr = scRNAseq_Seurat_integrated_R26_tumCells_cytoTrace$cytoGenes)
scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes <- rownames_to_column(scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes, var = "gene")
head(scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_R26_tumCells_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes)-5):nrow(scRNAseq_Seurat_integrated_R26_tumCells_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```

#### CytoTRACE 1234679

```{r cytoTrace}

# Subset cluster 0,1,3 and 5 with all clusters
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                                    integrated_snn_res.0.4 %in% c(1,2,3,4,6,7,9)#,
                                        #features = cluster1234679_DEG
                                                                    )
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes


# Extract expression matrix
scRNAseq_Seurat_integrated_R26_tumCells_1234679_rawCounts_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes,
                                                                                            assay = "RNA",#"integrated_R26_tumCells",
                                                                                            slot = "counts"#"scale.data"
                                                                                            )
                                                                               )
dim(scRNAseq_Seurat_integrated_R26_tumCells_1234679_rawCounts_matrix)
head(scRNAseq_Seurat_integrated_R26_tumCells_1234679_rawCounts_matrix[,1:4])

# Run CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace <- CytoTRACE(mat            = scRNAseq_Seurat_integrated_R26_tumCells_1234679_rawCounts_matrix,
                                                                        batch          = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data$orig.ident,
                                                                        enableFast     = TRUE,
                                                                        ncores         = 1,
                                                                        subsamplesize  = 1000
                                                                        )
str(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace)


# Distribution of CytoTRACE score
#pdf(file = paste0(pathToFigures,"density_scRNAseq_integrated_R26_tumCells_1234679_clust_CytoTRACE.pdf"), width = 14, height = 10)
ggplot(data = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace$CytoTRACE)
       ) +
    geom_density(mapping = aes(CytoTRACE)
                 ) +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data) == rownames(data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace$CytoTRACE)))


# Append metadata with CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes,
                                                                  metadata = data.frame(CytoTRACE = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace$CytoTRACE)
                                                                  )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data) == rownames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData))

# Append metadata with monocle3 UMAP
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes <- AddMetaData(object = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes,
                                                                  metadata = scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData[,c("data_dim_1","data_dim_2")]
                                                                  )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data)

# Plot UMAP embedding with CytoTRACE score
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_1234679_clust_CytoTRACE.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = CytoTRACE
                             ),
               size = 2
               ) +
    scale_color_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    theme_bw() +
    labs(colour = "Score") +
    theme(text            = element_text(size = 40),
          legend.key.height = unit(1.5,'cm')
          )
#dev.off()

# Plot UMAP embedding and cluster
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_1234679_clust_CytoTRACE_clust.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes@meta.data
       ) +
    geom_point(mapping = aes(x = data_dim_1,
                             y = data_dim_2,
                             color = integrated_snn_res.0.4
                             ),
               size = 2
               ) +
    labs(colour = "Cluster") +
    theme_bw() +
    theme(text            = element_text(size = 40)
          )
#dev.off()

# Boxplot of CytoTRACE score for each cluster 
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_tumCells_1234679_clust_CytoTRACE.pdf"), width = 16, height = 10)
boxplot_cluster_and_covariate(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes,
                              clustering  = paste0("integrated_snn_res.",choosenResolution),
                              covariate   = "CytoTRACE",
                              ylab        = "CytoTRACE\n(0.0: more diff. - 1.0: less diff.)",
                              reorder     = "mean"
                              )
#dev.off()

# Genes correlated with CytoTRACE
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes <- data.frame(corr = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoTrace$cytoGenes)
scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes <- rownames_to_column(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes, var = "gene")
head(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes)

#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_R26_tumCells_1234679_clust_CytoGenes.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes[c(1:5,(nrow(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes)-5):nrow(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cytoGenes)),]
       ) +
    geom_bar(mapping = aes(x = reorder(gene, corr), y = corr, fill = corr),
             stat = "identity"
             ) +
    scale_fill_gradientn(colours = c("red","yellow","grey","cyan","blue")) +
    coord_flip() +
    xlab("Gene\n") +
    ylab("\ncorrelation with CytoTRACE") +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```


## RNA velocity

### Extract meta-data for scVelo 

#### Extract all cluster meta data

```{r extract_metadata}

scRNAseq_Seurat_integrated_R26_tumCells

Cells(scRNAseq_Seurat_integrated_R26_tumCells)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_metaData.csv"
          )

```

#### Extract cluster  012679

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_012679)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_012679)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_012679_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_012679, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_012679_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_012679@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_012679_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_012679_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_012679_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_012679)

# Extract cluster 012679 DEG
write.csv(x = data.frame(DEG_0135=scRNAseq_Seurat_integrated_R26_tumCells_012679@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_012679_DEG.csv"
          )

```

#### Extract cluster  01346

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_01346)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_01346)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01346_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_01346, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01346_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_01346@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01346_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01346_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01346_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_01346)

# Extract cluster 01346 DEG
write.csv(x = data.frame(DEG_01346=scRNAseq_Seurat_integrated_R26_tumCells_01346@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01346_DEG.csv"
          )

```

#### Extract cluster  01279

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_01279)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_01279)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01279_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_01279, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01279_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_01279@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01279_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01279_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01279_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_01279)

# Extract cluster 01279 DEG
write.csv(x = data.frame(DEG_0135=scRNAseq_Seurat_integrated_R26_tumCells_01279@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01279_DEG.csv"
          )

```

#### Extract cluster  123469

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_123469)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_123469)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_123469_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_123469, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_123469_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_123469@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_123469_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_123469_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_123469_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_123469)

# Extract cluster 123469 DEG
write.csv(x = data.frame(DEG_0135=scRNAseq_Seurat_integrated_R26_tumCells_123469@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_123469_DEG.csv"
          )

```

#### Extract cluster  129

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_129)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_129)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_129, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_129@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_129_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_129)

# Extract cluster 129 DEG
write.csv(x = data.frame(DEG_129=scRNAseq_Seurat_integrated_R26_tumCells_129@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_DEG.csv"
          )

DEG_129 <- filter(.data = scRNAseq_Seurat_integrated_R26_tumCells.markers2,
                  cluster %in% c(1,2,9)
                  )$gene

write.csv(x = data.frame(DEG_129),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_129_DEG.csv"
          )


```

#### Extract cluster  431

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_431)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_431)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_431_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_431, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_431_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_431@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_431_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_431_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_431_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_431)

# Extract cluster 431 DEG
write.csv(x = data.frame(DEG_431=scRNAseq_Seurat_integrated_R26_tumCells_431@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_431_DEG.csv"
          )

```

#### Extract cluster  079

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_079)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_079)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_079, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_079@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_079_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_079)

# Extract cluster 079 DEG
write.csv(x = data.frame(DEG_079=scRNAseq_Seurat_integrated_R26_tumCells_079@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_DEG.csv"
          )

DEG_079 <- filter(.data = scRNAseq_Seurat_integrated_R26_tumCells.markers2,
                  cluster %in% c(0,7,9)
                  )$gene

write.csv(x = data.frame(DEG_079),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_079_DEG.csv"
          )

```

#### Extract cluster  01234679

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_01234679)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_01234679)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01234679_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_01234679, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01234679_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_01234679@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01234679_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
scRNAseq_Seurat_integrated_R26_tumCells_01234679_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_01234679_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01234679_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_01234679)

# Extract cluster 01234679 DEG
write.csv(x = data.frame(DEG_01234679=scRNAseq_Seurat_integrated_R26_tumCells_01234679@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_01234679_DEG.csv"
          )

```

#### Extract cluster  1234679

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_1234679)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_1234679)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_1234679_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_1234679, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_1234679_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_1234679@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_1234679_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_1234679)

# Extract cluster 1234679 DEG
write.csv(x = data.frame(DEG_1234679=scRNAseq_Seurat_integrated_R26_tumCells_1234679@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_1234679_DEG.csv"
          )

```

icicicicicicicicici

```{r dataSource}

colnames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData)

# Export for data source (RNA velocity raw data)
scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData %>%
    dplyr::select(data_dim_1, data_dim_2,sample_name,orig.ident,integrated_snn_res.0.4
                  ) %>%
    mutate(cell_barcode = sample_name,
           cell_cluster = integrated_snn_res.0.4,
           .keep = "unused"
           ) -> RNA_velocity_dataSource

tail(RNA_velocity_dataSource)
nrow(RNA_velocity_dataSource)

icicicicici

# Extract cluster 1,2,3,4,6,7,9 markers
DEG_1234679 <- filter(.data = scRNAseq_Seurat_integrated_R26_tumCells.markers2,
                      cluster %in% c(1,2,3,4,6,7,9)
                      )$gene

# Subset matrix of raw counts and transpose 

head(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA[,1:4])


scRNAseq_Seurat_merge_matrix_count_DEG0146_t <- t(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA[rownames(scRNAseq_Seurat_integrated_R26_tumCells_matrix_RNA) %in% DEG_1234679,])

head(scRNAseq_Seurat_merge_matrix_count_DEG0146_t)

scRNAseq_Seurat_merge_matrix_count_DEG0146_t <- rownames_to_column(data.frame(scRNAseq_Seurat_merge_matrix_count_DEG0146_t, check.names = FALSE), var = "cell_barcode")

install.packages("gsubfn")

library(gsubfn)

head(RNA_velocity_dataSource)

# Add raw gene count
RNA_velocity_dataSource_plus <- left_join(x = RNA_velocity_dataSource,
                                     y = scRNAseq_Seurat_merge_matrix_count_DEG0146_t,
                                     by = "cell_barcode"
                                     )

head(RNA_velocity_dataSource_plus)

# Export table for source data
write.csv(RNA_velocity_dataSource_plus,
          file = "../LobonIglesias_sourceData/Figure5_E.csv",
          row.names = FALSE
          )
```

#### Extract cluster  12349

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_12349)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_12349)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_12349_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_12349, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_12349_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_12349@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_12349_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_12349_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_12349_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_12349)

# Extract cluster 12349 DEG
write.csv(x = data.frame(DEG_12349=scRNAseq_Seurat_integrated_R26_tumCells_12349@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_12349_DEG.csv"
          )

```

#### Extract cluster  0346

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_tumCells_0346)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_tumCells_0346)),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_0346_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_tumCells_0346, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_0346_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_0346@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_0346_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_0346_cds_list[["PCmin"]],
                color_cells_by                 = "integrated_snn_res.0.4",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

# Export Monocle3 UMAP metadata
write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_0346_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_tumCells_0346)

# Extract cluster 0346 DEG
write.csv(x = data.frame(DEG_0346=scRNAseq_Seurat_integrated_R26_tumCells_0346@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_tumCells_0346_DEG.csv"
          )

```




scVelo analysis is performed in another script (python), see you there















icicicicicicicici

## Deconvolution of scRNAseq using mouse cell clusters from Vladoiu et al, 2019

###  Average expression matrix

```{r average_expression_matrix}

#Extract scaled integrated
matrix_scRNAseq_Seurat_integrated_R26 <- AverageExpression(object     = scRNAseq_Seurat_integrated_R26,
                                                       assays     = "RNA"
                                                       )$RNA

head(matrix_scRNAseq_Seurat_integrated_R26)
nrow(matrix_scRNAseq_Seurat_integrated_R26)

# Export matrix
write.table(x      = matrix_scRNAseq_Seurat_integrated_R26,
            file   = paste0(pathToReport,"matrix_scRNAseq_Seurat_integrated_R26.txt"),
            sep    = "\t",
            quote  = FALSE
            )

```

### Prepare mixture file

CIBERSORT need tab-delimited format (see CIBERSORT tutorial for more details).

```{r mixture_file}

# Create column "Gene" from rownames
matrix_scRNAseq_Seurat_integrated_R26_mixMat <- add_column(.data    = matrix_scRNAseq_Seurat_integrated_R26,
                                                       Gene     = rownames(matrix_scRNAseq_Seurat_integrated_R26),
                                                       .before  = 1
                                                       )
rownames(matrix_scRNAseq_Seurat_integrated_R26_mixMat) <- NULL
head(matrix_scRNAseq_Seurat_integrated_R26_mixMat)

# Export mixture file for CIBERSORT
write.table(x          = as.matrix(matrix_scRNAseq_Seurat_integrated_R26_mixMat),
            file       = paste0(pathToReport,"CIBERSORT/matrix_scRNAseq_Seurat_integrated_R26_mixMat.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```

### Import mouse signature matrix

```{r import_mouse_signature_matrix}

# Import mouse signature matrix
mouseSignature_valdoiu2019 <- read.table(file         = paste0(pathToData,"mouseSignatures_vladoiu2019.csv"),
                                          header       = TRUE,
                                          sep          = ";",
                                          dec          = ",",
                                          check.names  = FALSE
                                          )
head(mouseSignature_valdoiu2019)
ncol(mouseSignature_valdoiu2019)
nrow(mouseSignature_valdoiu2019)

```

### Prepare signature matrix

```{r signature_matrix}

# Import cell_type num, name and lineage
vladoiu_cellTypes <- read.table(file    = paste0(pathToData,"vladoiu2019_cellTypes.csv"),
                                header  = TRUE,
                                sep     = ";"
                                )
head(vladoiu_cellTypes)
nrow(vladoiu_cellTypes)

# Select cluster in mouse signature matrix
vladoiu_cellTypes_cibersort <- filter(.data = vladoiu_cellTypes,
                                      cluster_num %in% colnames(mouseSignature_valdoiu2019)
                                      )
head(vladoiu_cellTypes_cibersort)
nrow(vladoiu_cellTypes_cibersort)

# Check order
colnames(mouseSignature_valdoiu2019) == c("Gene", vladoiu_cellTypes_cibersort$cluster_num)

# Put cluster name as colnames
mouseSignature_valdoiu2019_cibersort <- mouseSignature_valdoiu2019
colnames(mouseSignature_valdoiu2019_cibersort) <- c("Gene", as.vector(vladoiu_cellTypes_cibersort$cluster_name))

head(mouseSignature_valdoiu2019_cibersort)
nrow(mouseSignature_valdoiu2019_cibersort)

# Export signature matrix
write.table(x          = mouseSignature_valdoiu2019_cibersort,
            file       = paste0(pathToReport,"CIBERSORT/mouseSignature_valdoiu2019_cibersort.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```


### Run CIBERSORT Fraction

For the moment CIBERSORT fraction is run from the website platform.
The result (fraction) is plotted as heatmap below.

```{r geneSubset}

# Import CIBERSORT Fraction results
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26 <- read.table(file         = paste0(pathToReport,"CIBERSORT/CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26.txt"),
                                                              sep          = "\t",
                                                              header       = TRUE,
                                                              check.names  = FALSE
                                                              )
head(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26)

# Put Mixture column (Sample name) as rownames and remove P-value, Correlation and RMSE columns
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min <- CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26
rownames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min)   <- CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min$Mixture
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min$Mixture     <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min$`P-value`   <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min$Correlation <- NULL
CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min$RMSE        <- NULL
head(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min)

# Density
melt(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min) %>%
    ggplot(mapping = aes(x = value)
           ) +
    geom_density() +
    geom_density(mapping = aes(x = -log10(value)),
                 color = "blue"
                 )

rownames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min)

# Check column order
vladoiu_cellTypes_cibersort$cluster_name == colnames(CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min)

# Define heatmap color
col_fun_mouseVladoiu2019 = colorRamp2(breaks = c(0,0.1,0.6), colors = c("lightskyblue","navajowhite","red"))

# Plot heatmap
h <- Heatmap(matrix                       = CIBERSORTx_Job32_scRNAseq_Seurat_integrated_R26_min,
             cluster_rows                 = TRUE,
             cluster_columns              = TRUE,
             show_row_names               = TRUE,
             show_column_names            = TRUE,
             show_row_dend                = TRUE,
             show_column_dend             = FALSE,
             clustering_distance_rows     = "euclidean",
             clustering_method_rows       = "ward",
             clustering_distance_columns  = "euclidean",
             clustering_method_columns    = "ward",
             #row_order                   = order(sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup),
             #row_split                    = sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup,
             column_split                 = vladoiu_cellTypes_cibersort$cluster_lineage,
             cluster_row_slices           = TRUE,
             cluster_column_slices        = FALSE,
             column_names_gp              = gpar(fontsize = 20),
             column_names_max_height      = unit(15,"cm"),
             row_names_gp                 = gpar(fontsize = 20),
             show_heatmap_legend          = FALSE,
             row_split                    = 3,
             column_title_gp              = gpar(fontsize = 20, fontface="bold"),
             row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
             col                          = col_fun_mouseVladoiu2019,
             row_title                    = c("A","B","C")
             )
#pdf(file = paste0(pathToFigures,"heatmap_CIBERSORT_fraction_scRNAseq_integrated_R26.pdf"), width = 20, height =10) 
draw(h)
#dev.off()

# Legend of relative abundance
legend_fraction <- Legend(col_fun         = col_fun_mouseVladoiu2019,
                          title           = "relative abundance\n",
                          grid_height     = unit(1,"cm"),
                          legend_width    = unit(10,"cm"),
                          title_gp        = gpar(fontsize = 20, fontface = "bold"),
                          title_position  = "topcenter",
                          direction       = "horizontal",
                          labels_gp       = gpar(fontsize=15)
                          )

# Combine legends
legends <- packLegend(list        = list(legend_fraction),
                      direction   ="horizontal",
                      column_gap  = unit(2,"cm")
                      )

#pdf(file = paste0(pathToFigures,"legend_heatmap_CIBERSORT_fraction_scRNAseq_integrated_R26.pdf"), width = 20, height =10) 
draw(legends)
#dev.off()

```


## Too-many-cells

### Extract matrix of cells

```{r extract_cell_matrix}

# Extract integrated assay table
scRNAseq_Seurat_integrated_R26_matrix <- scRNAseq_Seurat_integrated_R26_cellAnnot@assays$integrated@data
head(scRNAseq_Seurat_integrated_R26_matrix[,1:3])

# Export table
write.table(x      = scRNAseq_Seurat_integrated_R26_matrix,
            file   = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26_matrix.csv"),
            sep    = ",",
            quote  = FALSE
            )

```

### Extract meta data

```{r extract_metadata}

# Extract meta data
scRNAseq_Seurat_integrated_R26_metadata <- scRNAseq_Seurat_integrated_R26@meta.data
scRNAseq_Seurat_integrated_R26_metadata <- rownames_to_column(.data = scRNAseq_Seurat_integrated_R26_metadata,
                                                          var   = "item"
                                                          )
head(scRNAseq_Seurat_integrated_R26_metadata[,1:5])

# Export table
write.table(x          = scRNAseq_Seurat_integrated_R26_metadata,
            file       = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26_metadata.csv"),
            sep        = ",",
            quote      = FALSE,
            row.names  = FALSE
            )

```

### Launch too-many-cells

```{r launch_too_many_cells}

pathToReport

system2(command = "/bioinfo/local/build/singularity/singularity-3.5.2/bin/singularity",
        args = "run /bioinfo/local/Singularity/too-many-cells/too-many-cells_0.2.2.0.simg make-tree --matrix-path /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_Seurat_integrated_R26_matrix.csv --normalization NoneNorm  --labels-file /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_Seurat_integrated_R26_metadata.csv --output /bioinfo/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/tooManyCells_INI254"
        )

```

## Trajectory reconstruction

### ElpiGraph method

#### TO REMOVE: elpiGraph tutorial

```{r cellCycle}

##################
# sincellTE tuto #
##################

library(reticulate)

sk <- import("sklearn.manifold")

# MLLE projection
MLLE.Model <- sk$LocallyLinearEmbedding(n_neighbors=as.integer(ceiling(nrow(scRNAseq_Seurat_integrated_R26_matrix)*.1)),
                                        n_components = as.integer(3),
                                        method = 'modified',
                                        eigen_solver = 'dense',
                                        # n_jobs = as.integer(4),
                                        random_state = as.integer(10),
                                        neighbors_algorithm = 'kd_tree')

MLLE.Fit <- MLLE.Model$fit(ExpMat)
MLLE.Emb <- MLLE.Fit$embedding_







# Get igraph network object (to use in getPseudotime function below)
scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_iGraph <- ConstructGraph(PrintGraph = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph[[11]])

scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_iGraph

# Get the substructure of interest (here the whole nodes)
scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_circle <- GetSubGraph(Net        = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_iGraph,
                                                               Structure  = "circle",
                                                               Circular   = TRUE,
                                                               KeepEnds   = TRUE,
                                                               Nodes      = 50
                                                               )

scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_circle

# Project point onto graph
scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_projection <- project_point_onto_graph(X              = scRNAseq_Seurat_integrated_R26_ICA_matrix,
                                                                                NodePositions  = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph[[11]]$NodePositions,
                                                                                Edges          = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph[[11]]$Edges$Edges
                                                                                )

scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_projection

# Compute pseudotime
scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_pseudotime <- getPseudotime(ProjStruct = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_projection,
                                                                     NodeSeq = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_circle$Circle_1
                                                                     )

scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_pseudotime

# Plot pseudotime distribution
ggplot(data = data.frame(Pt = scRNAseq_Seurat_integrated_R26_ICA_ElPiGraph_pseudotime$Pt)
       ) +
    geom_histogram(mapping = aes(x = Pt),
                   fill = "grey",
                   color = "black"
                   ) +
    theme_bw()

```

#### Trajectory reconstruction for all clusters

##### Prepare matrix

```{r cellCycle}

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_cellAnnot_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_cellAnnot,
                                                                                  assay = "integrated",
                                                                                  slot = "scale.data"
                                                                                  )
                                                                     )
dim(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix)
head(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix[,1:4])

```

##### Principal tree

```{r cellCycle}

# Construct a principal Tree
scRNAseq_Seurat_integrated_R26_ElPiGraph_tree <- computeElasticPrincipalTree(X           = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                                                                         NumNodes    = 30,
                                                                         Lambda      = .03,
                                                                         Mu          = 0.1,
                                                                         Do_PCA      = TRUE,
                                                                         CenterData  = TRUE,
                                                                         drawAccuracyComplexity = FALSE,
                                                                         drawPCAView = FALSE,
                                                                         drawEnergy  = FALSE
                                                                         )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cellCycle}

# Construct a principal Curve
scRNAseq_Seurat_integrated_R26_ElPiGraph_curve <- computeElasticPrincipalCurve(X                      = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                                                                               NumNodes               = 30,
                                                                               Lambda                 = .03,
                                                                               Mu                     = 0.1,
                                                                               Do_PCA                 = TRUE,
                                                                               CenterData             = TRUE,
                                                                               drawAccuracyComplexity = FALSE,
                                                                               drawPCAView            = FALSE,
                                                                               drawEnergy             = FALSE
                                                                               )

# Check elpigraph object
str(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_all_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data$orig.ident
                )
#dev.off()

```

#### Trajectory reconstruction for cluster 0, 1, 3 and 5

##### Extract cluster 0, 1, 3 and 5 seurat object and matrix

```{r extract_cluster0135_object}

# Select cluster 0, 1, 3 and 5 marker genes
head(scRNAseq_Seurat_integrated_R26_choosenResolution.markers)

cluster0135_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_choosenResolution.markers,
       cluster %in% c(0,1,3,5)))
#cluster0135_DEG <- c(cluster0135_DEG,"TTYH1","SMARCB1")
head(cluster0135_DEG)
length(cluster0135_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_cellAnnot_0135 <- subset(x = scRNAseq_Seurat_integrated_R26_cellAnnot,
                                                    integrated_snn_res.0.3 %in% c(0,1,3,5),
                                                    features = cluster0135_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_cellAnnot_0135)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_cellAnnot_0135,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "sparseMatrix"
   )
nrow(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)

```

##### Principal Tree

```{r cellCycle}

# See below for matrix subsetting
dim(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)

# Construct a principal tree
scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135 <- computeElasticPrincipalTree(X                       = t(as.matrix(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)),
                                                                              NumNodes                = 30,
                                                                              Lambda                  = .03,
                                                                              Mu                      = .01,
                                                                              Do_PCA                  = TRUE,
                                                                              CenterData              = TRUE,
                                                                              drawAccuracyComplexity  = FALSE,
                                                                              drawPCAView             = FALSE,
                                                                              drawEnergy              = FALSE
                                                                              )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cluster0135_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135 <- computeElasticPrincipalCurve(X                       = t(as.matrix(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)),
                                                                                NumNodes                = 30,
                                                                                Lambda                  = .001,
                                                                                Mu                      = .1,
                                                                                Do_PCA                  = TRUE,
                                                                                CenterData              = TRUE,
                                                                                drawAccuracyComplexity  = FALSE,
                                                                                drawPCAView             = FALSE,
                                                                                drawEnergy              = FALSE
                                                                                )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_0135_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_0135)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data$orig.ident
                )
#dev.off()

```

#### Trajectory reconstruction for cluster 0, 1, 3, 5, 6 and 7

##### Extract cluster 0, 1, 3, 5, 6 and 7 seurat object and matrix

```{r extract_cluster013567_object}

# Select cluster marker genes
head(scRNAseq_Seurat_integrated_R26_choosenResolution.markers)

cluster013567_DEG <- rownames(filter(scRNAseq_Seurat_integrated_R26_choosenResolution.markers,
       cluster %in% c(0,1,3,5,6,7)))
#cluster013567_DEG <- c(cluster013567_DEG,"TTYH1","SMARCB1")
head(cluster013567_DEG)
length(cluster013567_DEG)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_cellAnnot_013567 <- subset(x = scRNAseq_Seurat_integrated_R26_cellAnnot,
                                                    integrated_snn_res.0.3 %in% c(0,1,3,5,6,7),
                                                    features = cluster013567_DEG
                                                    )
nrow(scRNAseq_Seurat_integrated_R26_cellAnnot_013567)

# Extract expression matrix
scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_cellAnnot_013567,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "sparseMatrix"
   )
nrow(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)

```

##### Principal Tree

```{r cellCycle}

# See below for matrix subsetting
dim(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)

# Construct a principal tree
scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567 <- computeElasticPrincipalTree(X                       = t(as.matrix(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)),
                                                                              NumNodes                = 30,
                                                                              Lambda                  = .03,
                                                                              Mu                      = .01,
                                                                              Do_PCA                  = TRUE,
                                                                              CenterData              = TRUE,
                                                                              drawAccuracyComplexity  = FALSE,
                                                                              drawPCAView             = FALSE,
                                                                              drawEnergy              = FALSE
                                                                              )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_tree_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_tree_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_tree_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_tree_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_tree_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_tree)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$orig.ident
                )
#dev.off()

```

##### Principal Curve

```{r cluster013567_elpigraph_curve}

# Construct a principal curve
scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567 <- computeElasticPrincipalCurve(X                       = t(as.matrix(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)),
                                                                                NumNodes                = 30,
                                                                                Lambda                  = .001,
                                                                                Mu                      = .1,
                                                                                Do_PCA                  = TRUE,
                                                                                CenterData              = TRUE,
                                                                                drawAccuracyComplexity  = FALSE,
                                                                                drawPCAView             = FALSE,
                                                                                drawEnergy              = FALSE
                                                                                )

# Plot principal graph with cell type
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_curve_cellType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$cell_type
                )
#dev.off()

# Plot principal graph with cell subtype
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_curve_cellSubType.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$cell_subtype
                )
#dev.off()

# Plot principal graph with cluster
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_curve_cluster.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$integrated_snn_res.0.3
                )
#dev.off()

# Plot principal graph with Patient ID
#pdf(file = paste0(pathToFigures,"elpiGraph_scRNAseq_integrated_R26_013567_curve_patientID.pdf"), width = 16, height = 12)
PlotPG_modified(X         = t(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                TargetPG  = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[[length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)]],
                BootPG    = scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567[1:(length(scRNAseq_Seurat_integrated_R26_ElPiGraph_curve_013567)-1)],
                                        #PointSize = 1.5,
                p.alpha = 0.5,
                GroupsLab = scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data$orig.ident
                )
#dev.off()

```


### Monocle version 3 method

#### Monocle on all clusters

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_cds <- new_cell_data_set(expression_data = as.matrix(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix, Class = "sparseMatrix"),
                                                    cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data),
                                                    gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                                                                               row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix)
                                                                               )
                                                    )


#nbPCs <- c(12,seq(10,100,10))
nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_R26_cds_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_cds <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_cds,
                                                 num_dim       = nbPC,
                                                 method        = "PCA",
                                                 norm_method   = "none",
                                                 pseudo_count  = 0
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_cds <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_cds,
                                                   max_components    = 2,
                                                   reduction_method  = "UMAP"
                                                   )
# Map pseudotime
scRNAseq_Seurat_integrated_R26_cds <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_cds,
                                                reduction_method  = "UMAP"
                                                )
scRNAseq_Seurat_integrated_R26_cds <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_cds
                                              )
    scRNAseq_Seurat_integrated_R26_cds_list[[length(scRNAseq_Seurat_integrated_R26_cds_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_cds_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_cds_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_cds_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_cds_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_list[[i]],
                                                  color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                  cell_size                      = 1,
                                                  trajectory_graph_color         = "black",
                                                  trajectory_graph_segment_size  = 2,
                                                  label_branch_points = FALSE,
                                                  label_cell_groups = FALSE,
                                                  label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_cds_list)[i])
scRNAseq_Seurat_integrated_R26_cds_plot_list[[length(scRNAseq_Seurat_integrated_R26_cds_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds_plot
}

library(gridExtra)

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_allClusters_list.pdf"), width = 20, height = 10)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_cds_plot_list, ncol = 3))
#dev.off()

```

#### Monocle on cluster 0, 1, 3 and 5

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_cds_0135 <- new_cell_data_set(expression_data = scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)
                                                                                    )
                                                         )

nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_R26_cds_0135_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_cds_0135 <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_cds_0135,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
scRNAseq_Seurat_integrated_R26_cds_0135 <- align_cds(cds = scRNAseq_Seurat_integrated_R26_cds_0135,
                                                 alignment_group = "orig.ident"
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_cds_0135 <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_cds_0135,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_integrated_R26_cds_0135 <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_cds_0135,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_integrated_R26_cds_0135 <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_cds_0135)
    scRNAseq_Seurat_integrated_R26_cds_0135_list[[length(scRNAseq_Seurat_integrated_R26_cds_0135_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds_0135
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_cds_0135_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_cds_0135_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_cds_0135_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_cds_0135_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[[i]],
                                                       color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_cds_0135_list)[i])
    scRNAseq_Seurat_integrated_R26_cds_0135_plot_list[[length(scRNAseq_Seurat_integrated_R26_cds_0135_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds_0135_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_cds_0135_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

# TO REMOVE: For meeting with Kornelius team to avoid giving more info
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_cellType_noName.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40),
          legend.position = "none"
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_cluster.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
           color_cells_by                 = "integrated_snn_res.0.3",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_patientID.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
           color_cells_by                 = "orig.ident",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_cellSubType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

# a helper function to identify the root principal points:
get_earliest_principal_node <- function(cds, cell_type="Neuronal progenitor"){
    cell_ids <- which(colData(cds)[, "cell_type"] == cell_type)

    closest_vertex <-
        cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
    closest_vertex <- as.matrix(closest_vertex[colnames(cds), ])
    root_pr_nodes <-
        igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
        (which.max(table(closest_vertex[cell_ids,]))))]

    root_pr_nodes
}

# Order cells
scRNAseq_Seurat_integrated_R26_cds_0135_pseudotime <- order_cells(cds = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
                                                              reduction_method = "UMAP",
                                                              root_pr_nodes = get_earliest_principal_node(scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]])
                                                              )

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_pseudotime.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_pseudotime,
           color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

lineage_genes <- c("STMN4","DLL3","HES6")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust0135_lineageGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_pseudotime,
           genes = lineage_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()


```

#### Monocle on cluster 0, 1, 3, 5, 6 and 7

```{r monocle_method}

# Create cds object
scRNAseq_Seurat_integrated_R26_cds_013567 <- new_cell_data_set(expression_data = scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix,
                                                         cell_metadata = data.frame(scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data),
                                                         gene_metadata = data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                                                                                    row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)
                                                                                    )
                                                         )

nbPCs <- c(10,12,30,50,70,90)
scRNAseq_Seurat_integrated_R26_cds_013567_list <- list()
for (nbPC in nbPCs)
{
# Pre-process pseudotime
scRNAseq_Seurat_integrated_R26_cds_013567 <- preprocess_cds(cds           = scRNAseq_Seurat_integrated_R26_cds_013567,
                                                      num_dim       = nbPC,
                                                      scaling       = TRUE,
                                                      method        = "PCA",
                                                      norm_method   = "none",
                                                      pseudo_count  = 0
                                                      )
# Correct batch effect
scRNAseq_Seurat_integrated_R26_cds_013567 <- align_cds(cds = scRNAseq_Seurat_integrated_R26_cds_013567,
                                                 alignment_group = "orig.ident"
                                                 )
# Reduce dimension
scRNAseq_Seurat_integrated_R26_cds_013567 <- reduce_dimension(cds               = scRNAseq_Seurat_integrated_R26_cds_013567,
                                                        max_components    = 2,
                                                        reduction_method  = "UMAP"
                                                        )
# Map pseudotime
    scRNAseq_Seurat_integrated_R26_cds_013567 <- cluster_cells(cds               = scRNAseq_Seurat_integrated_R26_cds_013567,
                                                         reduction_method  = "UMAP"
                                                         )
    scRNAseq_Seurat_integrated_R26_cds_013567 <- learn_graph(cds = scRNAseq_Seurat_integrated_R26_cds_013567)
    scRNAseq_Seurat_integrated_R26_cds_013567_list[[length(scRNAseq_Seurat_integrated_R26_cds_013567_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds_013567
}

# Rename each cds object in the list
names(scRNAseq_Seurat_integrated_R26_cds_013567_list) <- c("PC10","PCmin","PC30","PC50","PC70","PC90")

# Create plot list
scRNAseq_Seurat_integrated_R26_cds_013567_plot_list <- list()
for (i in 1:length(scRNAseq_Seurat_integrated_R26_cds_013567_list))
{
# Plot
scRNAseq_Seurat_integrated_R26_cds_013567_plot <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_list[[i]],
                                                       color_cells_by                 = "cell_type",
                                        #group_cells_by                 = "cell_type",
                                                       cell_size                      = 1,
                                                       trajectory_graph_color         = "black",
                                                       trajectory_graph_segment_size  = 2,
                                                       label_cell_groups = FALSE,
                                                       label_branch_points = FALSE,
                                                       label_leaves = FALSE
                                                  ) +
    ggtitle(names(scRNAseq_Seurat_integrated_R26_cds_013567_list)[i])
    scRNAseq_Seurat_integrated_R26_cds_013567_plot_list[[length(scRNAseq_Seurat_integrated_R26_cds_013567_plot_list)+1]] <- scRNAseq_Seurat_integrated_R26_cds_013567_plot
}

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_list.pdf"), width = 18, height = 12)
do.call("grid.arrange", c(scRNAseq_Seurat_integrated_R26_cds_013567_plot_list, ncol = 2))
#dev.off()

# Plot trajectory
#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_cellType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]],
           color_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_cluster.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]],
           color_cells_by                 = "integrated_snn_res.0.3",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_patientID.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]],
           color_cells_by                 = "orig.ident",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_cellSubType.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]],
           color_cells_by                 = "cell_subtype",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = TRUE,
           label_leaves = TRUE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

# a helper function to identify the root principal points:
get_earliest_principal_node <- function(cds, cell_type="Neuronal progenitor"){
    cell_ids <- which(colData(cds)[, "cell_type"] == cell_type)
    closest_vertex <-
        cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
    closest_vertex <- as.matrix(closest_vertex[colnames(cds), ])
    root_pr_nodes <-
        igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
        (which.max(table(closest_vertex[cell_ids,]))))]
    root_pr_nodes
}

# Order cells
scRNAseq_Seurat_integrated_R26_cds_013567_pseudotime <- order_cells(cds = scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]],
                                                              reduction_method = "UMAP",
                                                              root_pr_nodes = get_earliest_principal_node(scRNAseq_Seurat_integrated_R26_cds_013567_list[["PCmin"]])
                                                              )

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_pseudotime.pdf"), width = 20, height = 10)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_pseudotime,
           color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           trajectory_graph_color         = "black",
           trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

lineage_genes <- c("STMN4","DLL3","HES6")

#pdf(file = paste0(pathToFigures,"monocle_scRNAseq_integrated_R26_clust013567_lineageGenes.pdf"), width = 24, height = 8)
plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_013567_pseudotime,
           genes = lineage_genes,
           show_trajectory_graph = FALSE,
           #color_cells_by                 = "pseudotime",
                                        #group_cells_by                 = "cell_type",
           cell_size                      = 1,
           #trajectory_graph_color         = "black",
           #trajectory_graph_segment_size  = 2,
           label_cell_groups              = FALSE,
           label_branch_points = FALSE,
           label_roots = FALSE,
           label_leaves = FALSE
           ) +
    theme(text = element_text(size = 40)
          )
#dev.off()

```

icicicicicicicicicicici

### Monocle version 2 method

#### Monocle2 on all clusters

```{r monocle_method}

library(monocle)

# Create monocle2 cds object
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_R26_cellAnnot_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )


# Create Monocle2 cds object
#scRNAseq_Seurat_integrated_R26_cds_mncl2 <- as.CellDataSet(x = scRNAseq_Seurat_integrated_R26_cellAnnot,
 #                                                      assay = "RNA"
  #                                                     )

# Choose genes that define a cell's progress
HVF <- VariableFeatures(object = scRNAseq_Seurat_integrated_R26_cellAnnot,
                        assay = "integrated"
                        )

scRNAseq_Seurat_integrated_R26_cds_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                                                          ordering_genes = HVF
                                                          )

# Dimensionality reduction
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                                                        max_components = 2,
                                                        norm_method = "none",
                                                        relative_expr = TRUE,
                                                        scaling = TRUE,
                                                        residualModelFormulaStr = "~ orig.ident",
                                                        reduction_method = "DDRTree"
                                                        )

# Order cells along trajectory
scRNAseq_Seurat_integrated_R26_cds_mncl2 <- orderCells(scRNAseq_Seurat_integrated_R26_cds_mncl2)

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_allClusters_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_allClusters_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                     color_by = "integrated_snn_res.0.3"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_allClusters_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_allClusters_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

```

#### Monocle2 on cluster 0,1,3 and 5

```{r monocle_method}

# Create monocle2 cds object
scRNAseq_Seurat_integrated_R26_cds_0135_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_0135_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )



# Choose genes that define a cell's progress
head(cluster0135_DEG)
length(cluster0135_DEG)

#scRNAseq_Seurat_integrated_R26_cds_0135_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
 #                                                              ordering_genes = cluster0135_DEG
  #                                                             )

# Dimensionality reduction
scRNAseq_Seurat_integrated_R26_cds_0135_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
                                                             max_components = 2,
                                                             norm_method = "none",
                                                             relative_expr = TRUE,
                                                             scaling = TRUE,
                                                             residualModelFormulaStr = "~ orig.ident",
                                                             reduction_method = "DDRTree"
                                                             )

# Order cells along trajectory
scRNAseq_Seurat_integrated_R26_cds_0135_mncl2 <- orderCells(scRNAseq_Seurat_integrated_R26_cds_0135_mncl2)

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust0135_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust0135_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
                     color_by = "integrated_snn_res.0.3"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust0135_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust0135_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_0135_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

```

#### Monocle2 on cluster 0,1,3,5,6,7

```{r monocle_method}

# Create monocle2 cds object
scRNAseq_Seurat_integrated_R26_cds_013567_mncl2 <- newCellDataSet(cellData = scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix,
                                                       phenoData = new("AnnotatedDataFrame",scRNAseq_Seurat_integrated_R26_cellAnnot_013567@meta.data),
                                                       featureData = new("AnnotatedDataFrame",data.frame(gene_short_name = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix),
                                                                                                         row.names = row.names(scRNAseq_Seurat_integrated_R26_cellAnnot_013567_matrix)
                                                                                                         )
                                                                         ),
                                                       expressionFamily = uninormal()
                                                       )



# Choose genes that define a cell's progress
head(cluster013567_DEG)
length(cluster013567_DEG)

#scRNAseq_Seurat_integrated_R26_cds_013567_mncl2 <- setOrderingFilter(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
 #                                                              ordering_genes = cluster013567_DEG
  #                                                             )

# Dimensionality reduction
scRNAseq_Seurat_integrated_R26_cds_013567_mncl2 <- reduceDimension(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
                                                             max_components = 2,
                                                             norm_method = "none",
                                                             relative_expr = TRUE,
                                                             scaling = TRUE,
                                                             residualModelFormulaStr = "~ orig.ident",
                                                             reduction_method = "DDRTree"
                                                             )

# Order cells along trajectory
scRNAseq_Seurat_integrated_R26_cds_013567_mncl2 <- orderCells(scRNAseq_Seurat_integrated_R26_cds_013567_mncl2)

# Plot cell trajectory
#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust013567_cellType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
                     color_by = "cell_type"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust013567_cluster.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
                     color_by = "integrated_snn_res.0.3"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust013567_patient.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
                     color_by = "orig.ident"
                     )
#dev.off()

#pdf(file = paste0(pathToFigures,"monocle2_scRNAseq_integrated_R26_clust013567_cellSubType.pdf"), width = 16, height = 12)
plot_cell_trajectory(cds = scRNAseq_Seurat_integrated_R26_cds_013567_mncl2,
                     color_by = "cell_subtype"
                     )
#dev.off()

```

icicicicicicicicicicicicici

## RNA velocity

### Extract meta-data for scVelo 

#### Extract all cluster meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_cellAnnot)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_cellAnnot)),
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_cellAnnot, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_cellAnnot@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_metaData.csv"
          )

```

#### Extract cluster  0,1,3,5 meta data

```{r extract_metadata}

Cells(scRNAseq_Seurat_integrated_R26_cellAnnot_0135)

# Extract cell IDs
write.csv(x = (Cells(scRNAseq_Seurat_integrated_R26_cellAnnot_0135)),
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_0135_cellIDs.csv",
          row.names = FALSE
          )

# Extract UMAP embeddings
write.csv(x = Embeddings(scRNAseq_Seurat_integrated_R26_cellAnnot_0135, reduction = "umap"),
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_0135_UMAP.csv",
          )

# Extract other mata data
write.csv(x = scRNAseq_Seurat_integrated_R26_cellAnnot_0135@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_0135_metaData.csv"
          )

# Extract Monocle3 UMAP coordinates and meta.data
p <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_cds_0135_list[["PCmin"]],
                color_cells_by                 = "cell_subtype",
                cell_size                      = 1,
                trajectory_graph_color         = "black",
                trajectory_graph_segment_size  = 2,
                label_cell_groups              = FALSE,
                label_branch_points = FALSE,
                label_roots = TRUE,
                label_leaves = TRUE
                )
pdata <- ggplot_build(p)[["plot"]][["data"]]

head(pdata)

write.csv(x = pdata,
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_0135_monocle3_UMAP_metaData.csv"
          )

str(scRNAseq_Seurat_integrated_R26_cellAnnot_0135)

# Extract cluster 0,1,3,5 DEG
write.csv(x = data.frame(DEG_0135=scRNAseq_Seurat_integrated_R26_cellAnnot_0135@assays$RNA@counts@Dimnames[[1]]),
          file = "scRNAseq_Seurat_integrated_R26_cellAnnot_0135_DEG.csv"
          )

```


scVelo analysis is performed in another script (python), see you there










## Clustering (Using PCmin=50)

```{r clustering}

# Construct KNN graph
scRNAseq_Seurat_integrated_R26_PC50 <- FindNeighbors(object = scRNAseq_Seurat_integrated_R26,
                                                 dims   = 1:50
                                                 )

# Clustering by modularity optimization technique
scRNAseq_Seurat_integrated_R26_PC50 <- FindClusters(object     = scRNAseq_Seurat_integrated_R26_PC50,
                                                resolution = seq(from  = 0,
                                                                 to    = 2,
                                                                 by    = 0.1
                                                                 )
                                                )      

```

## Choose the best resolution of clustering

```{r choose_resolution}

# Choose resolution
#pdf(file = paste0(pathToFigures,"clustree_scRNAseq_integrated_R26_PC50.pdf"), width = 10, height = 12)
clustree(x               = scRNAseq_Seurat_integrated_R26_PC50@meta.data,
         prefix          = "integrated_snn_res.",
         #node_colour     = "sc3_stability",
         #layout          = "sugiyama",
         use_core_edges  = TRUE
         )
#dev.off()

choosenResolution <- 0.3

# Clustering by modularity optimization technique
#scRNAseq_Seurat_integrated_R26_PC50_choosenResolution <- FindClusters(object     = scRNAseq_Seurat_integrated_R26_PC50,
 #                                                                 resolution = choosenResolution
  #                                                                )      

# Set default ident
Idents(scRNAseq_Seurat_integrated_R26_PC50) <- scRNAseq_Seurat_integrated_R26_PC50@meta.data[["integrated_snn_res.0.3"]]

# Number of cell per cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_PC50_numberOfcellsPerCluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_PC50@meta.data
       ) +
    stat_count(mapping = aes(x     = integrated_snn_res.0.3,
                             fill  = orig.ident
                             )
               ) +
    geom_text(mapping  = aes(x     = integrated_snn_res.0.3,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              face     = "bold"
              ) +
    xlab("\nCluster") +
    ylab("# of cells\n") +
scale_fill_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = c(0.85,0.85)
              )
#dev.off()

# Export meta data (for SCENIC analysis)
head(scRNAseq_Seurat_integrated_R26_PC50@meta.data)

write.table(x      = scRNAseq_Seurat_integrated_R26_PC50@meta.data,
            file   = "scRNAseq_Seurat_integrated_R26_PC50_metaData.txt",
            sep    = ",",
            quote  = FALSE
            )

```

## UMAP

```{r UMAP}

# Run UMAP
scRNAseq_Seurat_integrated_R26_PC50 <- RunUMAP(object       = scRNAseq_Seurat_integrated_R26_PC50,
                                           dims         = 1:50,
                                           n.neighbors  = 30
                                           )

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = paste0("integrated_snn_res.",choosenResolution),
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_sample.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "orig.ident",
        label       = FALSE
        ) +
    scale_colour_manual(name = "Sample", values = c("red","green","blue")) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# Save seurat object
saveRDS(object = scRNAseq_Seurat_integrated_R26_PC50,
        file   = paste0(pathToReport,"scRNAseq_Seurat_integrated_R26_PC50.rds"
                        )
        )

```

## UMAP and covariates

```{r umap_and_covariates}

# UMAP and nb of UMIs
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_nbUMIs.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_PC50,
            features = "nFeature_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_nbGenes.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_PC50,
            features = "nCount_RNA",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_nbRNA_per_gene.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_PC50,
            features = "nCount_per_Feature",
            reduction = "umap"
            ) +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.9,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_percMito.pdf"), width = 12, height = 10)
FeaturePlot(object = scRNAseq_Seurat_integrated_R26_PC50,
            features = "percent.mito",
            reduction = "umap"
            )  +
    ggtitle(label = NULL) +
    theme_bw() +
    theme(legend.position = c(0.1,0.9),
          text = element_text(size = 20)
          )
#dev.off()

# UMAP and cell cycle phase
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_clust_cellCyclePhase.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_PC50,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "Phase",
        label       = TRUE,
        label.size  = 15
        ) +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20)
          )
#dev.off()

```

## Boxplot of covariates per cluster

```{r boxplot_and_covariates}

# Boxplot and nb of UMIs
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_PC50_clust_nbUMIs.pdf"), width = 12, height = 12)
 ggplot(data = scRNAseq_Seurat_integrated_R26_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = log10(nCount_RNA),
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = log10(nCount_RNA)
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "log10(UMI count)\n")+
    ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_PC50_clust_nbGenes.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nFeature_RNA,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nFeature_RNA
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "# genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and nb of UMIs per gene
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_PC50_clust_nbRNA_per_gene.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = nCount_per_Feature,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = nCount_per_Feature
                              ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "UMIs per gene\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

# Boxplot and percentage of mitochondrial genes
#pdf(file = paste0(pathToFigures,"boxplot_scRNAseq_integrated_R26_PC50_clust_percMito.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_Seurat_integrated_R26_PC50@meta.data) +
    geom_jitter(mapping = aes(x      = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                              y      = percent.mito,
                              color  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                              ),
                size    = 0.5
                ) +
    stat_summary(mapping = aes(x = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                               y = percent.mito
                               ),
                 fun     = median,
                 geom    = "crossbar",
                 width   = 0.75,
                 color   = "blue"
                 ) +
    xlab(label = "\nCluster") +
    ylab(label = "% mito genes\n")+
    #ylim(2,5) +
    theme_bw() +
    theme(legend.position = "none",
          text            = element_text(size = 40)
          )
#dev.off()

```


## Differential analysis

### One versus all (finding cluster biomarkers)

```{r diffAnalysis}

# Set thresholds
min_pct          <- 0.25
min_logfc        <- 0.50
min_diffPct      <- -Inf
min_cellsFeature <- 3

# Find markers for each cluster (versus all others)
scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers <- FindAllMarkers(object            = scRNAseq_Seurat_integrated_R26_PC50,
                                                                            assay             = "RNA",
                                                                            min.pct           = min_pct,
                                                                            logfc.threshold   = min_logfc,
                                                                            min.diff.pct      = min_diffPct,
                                                                            only.pos          = TRUE,
                                                                            min.cells.feature = min_cellsFeature
                                                                            )

head(scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers)

# Plot the number of markers detected for each cluster
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_integrated_R26_PC50_numberOfMarkers.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers,
       mapping = aes(x = cluster,
                     fill = cluster
                     )
       ) +
    stat_count() +
    geom_text(mapping = aes(label = stat(count)),
              stat = "count",
              vjust = -0.25,
              size = 10
              ) +
    xlab("\nCluster") +
    ylab("# of markers\n") +
    theme_bw() +
        theme(text = element_text(size = 40),
              legend.position = "none"
              )
#dev.off()

# Look at the top n markers for each cluster
nTop <- 5
scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Select top markers for each cluster
nTop_markers <- scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers %>% group_by(cluster) %>% top_n(n = nTop, wt = avg_logFC)

# Heatmap of all markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_clusterMarkers.pdf"), width = 16, height = 10)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_PC50_choosenResolution,
          features = scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_blank())
#dev.off()

# Heatmap of top markers
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_clusterTopMarkers.pdf"), width = 16, height = 12)
DoHeatmap(object   = scRNAseq_Seurat_integrated_R26_PC50_choosenResolution,
          features = nTop_markers$gene,
          assay    = "integrated",
          slot     = "scale.data"
          ) +
    NoLegend() +
    theme(axis.text.y = element_text(size=15))
#dev.off()

# Plot top markers for each cluster
#pdf(file = paste0(pathToFigures,"volcanoPlot_scRNAseq_integrated_R26_PC50_clusterTopMarkers.pdf"), width = 16, height = 10)
nTop = 20
scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers %>%
    group_by(cluster) %>%
    top_n(n = 20, wt = avg_logFC) %>%
    ggplot() +
    geom_point(mapping = aes(x = avg_logFC,
                             y = -log10(p_val_adj),
                             size = pct.1 - pct.2 
                             ),
               pch = 21,
               fill = "blue",
               alpha = 0.5
               ) +
    facet_wrap(facets = ~ cluster, nrow = 2) +
    geom_text_repel(mapping = aes(x = avg_logFC,
                                  y = -log10(p_val_adj),
                                  label = gene
                                  ),
                    size = 3
                    ) +
    xlab("\navg_logFC") +
    ylab("-log10(p_val_adj)\n") +
    theme_bw() +
    theme(text = element_text(size = 20),
          strip.background = element_rect(fill = "green"),
          strip.text = element_text(face = "bold")
          )
#dev.off()

# Export marker table
write.table(x           = scRNAseq_Seurat_integrated_R26_PC50_choosenResolution.markers,
            file        = paste0(pathToReport,"scRNAseq_integrated_R26_PC50_cluster_geneMarkers.txt"),
            sep         = "\t",
            quote       = FALSE
            )

```

## SMARCB1 expression

```{r SMARCB1_expression}

# SMARCB1 expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_SMARCB1.pdf"), width = 12, height = 12)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = "SMARCB1",
            pt.size  = 1
            ) +
    ggtitle(label = "SMARCB1") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

# SMARCB1 expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_SMARCB1.pdf"), width = 12, height = 12)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = "SMARCB1",
        pt.size = 0
        ) +
    ggtitle(label = "SMARCB1") +
    xlab("\nCluster") +
    theme_bw() +
    theme(legend.position = "none",
          text = element_text(size = 20),
          plot.title = element_text(hjust = 0.5, face = "bold")
          )
#dev.off()

```

## Expression of ATRT bulk signature genes

#### ATRT SHH signature from Torchia et al, 2016

```{r SHH_markers_expression_Torchia2016}

# Select SHH gene signatures
geneSet_ATRT_Torchia2016_SHH <- filter(.data = geneSet_ATRT_Torchia2016,
                                       Group == "SHH"
                                       )
geneSet_ATRT_Torchia2016_SHH$Gene

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = geneSet_ATRT_Torchia2016_SHH$Gene,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_ATRT_Torchia2016_SHH_genes.pdf"), width = 12, height = 20)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = geneSet_ATRT_Torchia2016_SHH$Gene,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

#### Expression of cerebellar marker genes (Wezman et al., 2019)

##### Roof Plate

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 15)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_RoofPlate_Wezman2019_genes.pdf"), width = 10, height = 14)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = cellType_markers_Wizeman2019_RoofPlate$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```


##### Rhombic Lip

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 2
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_RhombicLip_Wezman2019_genes.pdf"), width = 12, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = cellType_markers_Wizeman2019_RhombicLip$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 2
        )
#dev.off()

```

##### Ventricular Zone

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_VentricularZone_Wezman2019_genes.pdf"), width = 16, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = cellType_markers_Wizeman2019_VentricularZone$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### Bergmann Glia

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_BergmanGlia_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = cellType_markers_Wizeman2019_BergamannGlia$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

##### GABAergic

```{r cerebellar_marker_genes_expression_Wezman2019}

# Change default assay to RNA (some genes cannot find in integrated
DefaultAssay(scRNAseq_Seurat_integrated_R26_PC50) <- "RNA"

# ATRT SHH signature gene expression in UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
FeaturePlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
            features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
            pt.size  = 0.25,
            slot     = "data",
            ncol     = 3
            )
#dev.off()

# ATRT SHH signature gene expression in violin plot
#pdf(file = paste0(pathToFigures,"violinPlot_scRNAseq_integrated_R26_PC50_GABAergic_Wezman2019_genes.pdf"), width = 15, height = 10)
VlnPlot(object   = scRNAseq_Seurat_integrated_R26_PC50,
        features = cellType_markers_Wizeman2019_GABAergic$HGNC.symbol,
        slot     = "data",
        pt.size  = 0,
        ncol     = 3
        )
#dev.off()

```

## SCENIC Analysis (PC50)

### Cellular enrichment of regulons using AUC score

#### Heatmap of AUC score

```{r SCENIC_AUCscore_heatmap}

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26@meta.data) == colnames(scRNAseq_integrated_R26_scenic_auc_matrix_t_filt_mvRegulons))

# Top annotation
topAnnot_PC50 <- HeatmapAnnotation(df                      = data.frame(Cluster = as.character(scRNAseq_Seurat_integrated_R26_PC50@meta.data[,"integrated_snn_res.0.3"]), check.names = FALSE),
                                   col                     = list(Cluster = cluster_n12_col),
                                   gap                     = unit(2,"mm"),
                                   show_annotation_name    = TRUE,
                                   annotation_name_gp      = gpar(fontsize    = 20),
                                   annotation_legend_param = list(fontsize    = 20,
                                                                  title_gp    = gpar(fontsize = 20, font = 2),
                                                                  labels_gp   = gpar(fontsize = 20),
                                                                  grid_width  = unit(12,"mm"),
                                                                  grid_height = unit(12,"mm"),
                                                                  ncol        = 1
                                                                  ),
                                   simple_anno_size        = unit(2,"cm"),
                                   na_col                  = "white",
                                   show_legend             = TRUE
                                   )

# Bottom annotation
bottomAnnot_PC50 <- HeatmapAnnotation(df                      = data.frame(Patient_ID = as.character(scRNAseq_Seurat_integrated_R26_PC50@meta.data[,"orig.ident"]), check.names = FALSE),
                                    col                     = list(Patient_ID = patient_id_col),
                                    gap                     = unit(2,"mm"),
                                    show_annotation_name    = TRUE,
                                    annotation_name_gp      = gpar(fontsize    = 20),
                                    annotation_legend_param = list(fontsize    = 20,
                                                                   title_gp    = gpar(fontsize = 20, font = 2),
                                                                   labels_gp   = gpar(fontsize = 20),
                                                                   grid_width  = unit(12,"mm"),
                                                                   grid_height = unit(12,"mm"),
                                                                   ncol        = 1
                                                                   ),
                                    simple_anno_size        = unit(1,"cm"),
                                    na_col                  = "white",
                                    show_legend             = TRUE
                                    )

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,0.2,0.4,0.6), colors = c("cyan","wheat1","red","darkred"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_R26_scenic_auc_matrix_t_filt_mvRegulons,
             clustering_distance_columns = "pearson",
        clustering_method_columns   = "ward",
        clustering_distance_rows    = "euclidean",
        clustering_method_rows      = "ward",
        cluster_rows                = TRUE,
        show_row_names              = TRUE,
        show_column_names           = FALSE,
        column_title                = "cells",
        row_title                   = "Top 50 most variable regulons",
        top_annotation              = topAnnot_PC50,
        bottom_annotation           = bottomAnnot_PC50,
        show_heatmap_legend         = TRUE,
        column_dend_height          = unit(40, "mm"),
        column_split                = 6,
        row_split                   = 14,
        col                         = col_fun,
heatmap_legend_param        = list(color_bar      = "continuous",
                                   grid_width     = unit(0.75,"cm"),
                                   grid_height    = unit(2,"cm"),
                                   legend_width   = unit(4,"cm"),
                                   title          = "AUC\n score\n",
                                   title_gp       = gpar(fontsize = 20),
                                   legend_gp      = gpar(fontsize = 20),
                                   title_position = "topcenter"
                                   )
)
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_SCENIC_AUCscore.pdf"), width = 12, height = 12)
draw(h, heatmap_legend_side = "left")
#dev.off()

```

#### UMAP of cells based on AUC score

```{r AUCscore_umap}

# Bind meta data
scRNAseq_integrated_R26_PC50_scenic_auc_umap_plus <- cbind(scRNAseq_integrated_R26_scenic_auc_umap,
                                              cluster = as.character(scRNAseq_Seurat_integrated_R26_PC50@meta.data$integrated_snn_res.0.3)
                                              )
head(scRNAseq_integrated_R26_PC50_scenic_auc_umap_plus)

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_PC50_SCENIC_AUCscore.pdf"), width = 12, height = 12)
ggplot(data = scRNAseq_integrated_R26_PC50_scenic_auc_umap_plus,
       mapping = aes(x = UMAP1, y = UMAP2)
       ) +
    geom_point(mapping = aes(color = cluster
                             ),
               size = 2
               )+
    xlab("UMAP1") +
    ylab("UMAP2") +
    ggtitle(NULL) +
    theme_bw()+
    theme(text               = element_text(size = 20),
          legend.position    = "none"
          #legend.title       = element_text(face = "bold"),
          #legend.key.height  = unit(1.5,"cm")
          )
#dev.off()

```	

### Cellular enrichment of regulons using binarized AUC score

#### Heatmap of AUC_BIN score

```{r SCENIC_AUC_BINscore_heatmap}

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_PC50@meta.data) == colnames(scRNAseq_integrated_R26_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons))

# Define heatmap color
col_fun = colorRamp2(breaks = c(0,1), colors = c("white","black"))

# Heatmap
h <- Heatmap(matrix                      = scRNAseq_integrated_R26_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons,
             clustering_distance_rows    = "binary",
             clustering_method_rows      = "ward",
             cluster_rows                = TRUE,
             cluster_columns             = FALSE,
             show_row_names              = TRUE,
             show_row_dend               = FALSE,
             show_column_names           = FALSE,
             column_title                = "cells",
             row_title                   = "Top 50 most variable regulons",
             top_annotation              = topAnnot_PC50,
             bottom_annotation           = bottomAnnot_PC50,
             column_dend_height          = unit(40, "mm"),
             column_order                = order(factor(scRNAseq_Seurat_integrated_R26_PC50@meta.data[,"integrated_snn_res.0.3"])),
             column_split                = factor(scRNAseq_Seurat_integrated_R26_PC50@meta.data[,"integrated_snn_res.0.3"]),
             #row_split                   = 2,
             col                         = col_fun,
             show_heatmap_legend         = FALSE
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_SCENIC_AUCbinScore.pdf"), width = 12, height = 12)
draw(h)
#dev.off()

```

#### Heatmap of the percentage of cells with activated regulon

```{r heatmap_perc_active_regulon}

# Merge matrix and meta data
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus <- merge(x     = scRNAseq_Seurat_integrated_R26_PC50@meta.data["integrated_snn_res.0.3"],
                                                                                   y     = scRNAseq_integrated_R26_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt,
                                                                                   by.x  = 0,
                                                                                   by.y  = "cellID",
                                                                                   all   = FALSE
                                                                                   )
head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus)

# Create table of cluster, regulon and number of cells
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells <- group_by(.data = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus,
                                                                                                   integrated_snn_res.0.3,
                                                                                                   regulon
                                                                                                   ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells) = c("cluster","regulon","nbCell")
head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells)

# Create table of cluster, regulon and number of cells in which regulon is activated
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells <- filter(.data = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus, AUCbin == 1) %>%
    group_by(integrated_snn_res.0.3,
             regulon
             ) %>%
    summarise(number =  n())
colnames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells) = c("cluster","regulon","nbCell_AUCbin1")
head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells)

# Create a table of total number of cells and cells in which regulon is activated
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells <- merge(x = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells,
                                                                                                   y = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_allCells,
                                                                                                   by = c("cluster","regulon"),
                                                                                                   all = FALSE
                                                                                                   )
head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Compute percentage of cells with active regulon
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$activeRegulon <- scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell_AUCbin1*100/scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells$nbCell

head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells)

# Transform table to matrix (for ComplexHeatmap function)
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix <- spread(data = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_melt_plus_activeCells_allCells[,c("cluster","regulon","activeRegulon")],
                                                                                                 key = cluster,
                                                                                                 value = activeRegulon
                                                                                                 )
rownames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix) <- scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon
scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix$regulon <- NULL
head(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix)



# Top annotation
topAnnot_cluster_PC50 <- HeatmapAnnotation(cluster                  = colnames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
                                           text                     = anno_text(x= colnames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix), rot = 0),
                                           col                      = list(cluster = cluster_n12_col),
                                           gap                      = unit(4,"mm"),
                                           show_annotation_name     = FALSE,
                                           annotation_name_gp       = gpar(fontsize = 20),
                                           annotation_legend_param  = list(fontsize   = 20,
                                                                           title_gp    = gpar(fontsize = 20, font = 2),
                                                                           labels_gp   = gpar(fontsize = 20),
                                                                           grid_width  = unit(12,"mm"),
                                                                           grid_height = unit(12,"mm"),
                                                                           ncol        = 1
                                                                           ),
                                           simple_anno_size        = unit(2,"cm"),
                                           na_col                  = "white",
                                           show_legend             = FALSE
                                           )

# Heatmap
h <- Heatmap(matrix                    = scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix,
             cluster_columns           = FALSE,
             clustering_distance_rows  = "euclidean",
             clustering_method_rows    = "average",
             top_annotation            = topAnnot_cluster_PC50,
             show_row_dend             = FALSE,
             column_title              = "cluster\n",
             row_title                 = "Top 50 most variable regulons",
             column_split              = colnames(scRNAseq_integrated_R26_PC50_scenic_auc_matrix_bin_t_WZ_filt_mvRegulons_activeCells_allCells_matrix),
             show_column_names         = FALSE,
             heatmap_legend_param        = list(color_bar      = "continuous",
                                                grid_width     = unit(2,"cm"),
                                                grid_height    = unit(2,"cm"),
                                                legend_width   = unit(20,"cm"),
                                                title          = "\n% of cells with active regulon",
                                                title_gp       = gpar(fontsize = 20),
                                                legend_gp      = gpar(fontsize = 20),
                                                title_position = "topcenter",
                                                direction = "horizontal"
                                                )
             )
#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_SCENIC_AUCbinScore_perc.pdf"), width = 12, height = 12)
draw(h,
     heatmap_legend_side = "bottom"
     )
#dev.off()

```

### Cell type specific regulators

#### Import SCENIC RSS score

```{r import_SCENIC_AUCscore}

# Import SCENIC RSS score
scRNAseq_integrated_R26_PC50_scenic_rss_matrix <- read.csv(file = paste0(pathToSCENICresult,"scRNAseq_integrated_R26_PC50_scenic_rss_matrix.csv"))
head(scRNAseq_integrated_R26_PC50_scenic_rss_matrix)
dim(scRNAseq_integrated_R26_PC50_scenic_rss_matrix)

# Create column of cluster
scRNAseq_integrated_R26_PC50_scenic_rss_matrix$cluster <- rownames(scRNAseq_integrated_R26_PC50_scenic_rss_matrix)
head(scRNAseq_integrated_R26_PC50_scenic_rss_matrix)

# Density distribution
scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt <- melt(scRNAseq_integrated_R26_PC50_scenic_rss_matrix)
head(scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt)

ggplot(data = scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt) +
    geom_density(mapping = aes(x = value),
                   fill = "grey"
             ) +
    xlab("\nRSS score") +
    ylab("density\n") +
    theme_bw()

# Convert variable column (regulon) into factor
scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt$variable <- factor(scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt$variable)
head(scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt)

# Plot RSS per cluster
scRNAseq_integrated_R26_PC50_scenic_rss_plot_list = list()
for (cluster in levels(factor(scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt$cluster)))
{
    # Select cluster 
    f <- scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt[which(scRNAseq_integrated_R26_PC50_scenic_rss_matrix_melt$cluster == cluster),]
    # Pick top 5 regulons
    top5 <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    # Plot RSS for each regulon
    p <- ggplot(data = f,
                mapping = aes(x = reorder(variable,-value),
                              y = value
                              )
                ) +
        ggtitle(cluster) +
        xlab("regulon") +
        ylab("RSS score") +
        geom_point(color = "blue") +
        geom_point(data = filter(f, variable %in% top5),
                   color = "red",
                   size = 2
                   ) + 
        geom_text_repel(data = filter(f, variable %in% top5),
                        mapping = aes(label = variable),
                        color = "red",
                        size = 5
                        ) + 
        theme_bw() +
        theme(text = element_text(size = 20),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              plot.title = element_text(hjust = 0.5, face = "bold", size = 30)
              )
    # Append list of plot
    scRNAseq_integrated_R26_PC50_scenic_rss_plot_list[[length(scRNAseq_integrated_R26_PC50_scenic_rss_plot_list)+1]] <- p
}

#pdf(file = paste0(pathToFigures,"heatmap_scRNAseq_integrated_R26_PC50_SCENIC_RSS.pdf"), width = 16, height = 12)
ggarrange(plotlist = scRNAseq_integrated_R26_PC50_scenic_rss_plot_list)
#dev.off()

```

## Deconvolution of scRNAseq using mouse cell clusters from Vladoiu et al, 2019 (PC50)

###  Average expression matrix

```{r average_expression_matrix}

#Extract scaled integrated
matrix_scRNAseq_Seurat_integrated_R26_PC50 <- AverageExpression(object     = scRNAseq_Seurat_integrated_R26_PC50,
                                                       assays     = "RNA"
                                                       )$RNA

head(matrix_scRNAseq_Seurat_integrated_R26_PC50)
nrow(matrix_scRNAseq_Seurat_integrated_R26_PC50)

# Export matrix
write.table(x      = matrix_scRNAseq_Seurat_integrated_R26_PC50,
            file   = paste0(pathToReport,"matrix_scRNAseq_Seurat_integrated_R26_PC50.txt"),
            sep    = "\t",
            quote  = FALSE
            )

```

### Prepare mixture file

CIBERSORT need tab-delimited format (see CIBERSORT tutorial for more details).

```{r mixture_file}

# Create column "Gene" from rownames
matrix_scRNAseq_Seurat_integrated_R26_PC50_mixMat <- add_column(.data    = matrix_scRNAseq_Seurat_integrated_R26_PC50,
                                                       Gene     = rownames(matrix_scRNAseq_Seurat_integrated_R26_PC50),
                                                       .before  = 1
                                                       )
rownames(matrix_scRNAseq_Seurat_integrated_R26_PC50_mixMat) <- NULL
head(matrix_scRNAseq_Seurat_integrated_R26_PC50_mixMat)

# Export mixture file for CIBERSORT
write.table(x          = as.matrix(matrix_scRNAseq_Seurat_integrated_R26_PC50_mixMat),
            file       = paste0(pathToReport,"CIBERSORT/matrix_scRNAseq_Seurat_integrated_R26_PC50_mixMat.txt"),
            quote      = FALSE,
            sep        = "\t",
            row.names  = FALSE,
            col.names  = TRUE
            )

```


### Run CIBERSORT Fraction

For the moment CIBERSORT fraction is run from the website platform.
The result (fraction) is plotted as heatmap below.

```{r geneSubset}

# Import CIBERSORT Fraction results
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50 <- read.table(file         = paste0(pathToReport,"CIBERSORT/CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50.txt"),
                                                              sep          = "\t",
                                                              header       = TRUE,
                                                              check.names  = FALSE
                                                              )
head(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50)

# Put Mixture column (Sample name) as rownames and remove P-value, Correlation and RMSE columns
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min <- CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50
rownames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min)   <- CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min$Mixture
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min$Mixture     <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min$`P-value`   <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min$Correlation <- NULL
CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min$RMSE        <- NULL
head(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min)

# Density
melt(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min) %>%
    ggplot(mapping = aes(x = value)
           ) +
    geom_density() +
    geom_density(mapping = aes(x = -log10(value)),
                 color = "blue"
                 )

rownames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min)

# Check column order
vladoiu_cellTypes_cibersort$cluster_name == colnames(CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min)

# Define heatmap color
col_fun_mouseVladoiu2019 = colorRamp2(breaks = c(0,0.1,0.6), colors = c("lightskyblue","navajowhite","red"))

# Plot heatmap
h <- Heatmap(matrix                       = CIBERSORTx_Job33_scRNAseq_Seurat_integrated_R26_PC50_min,
             cluster_rows                 = TRUE,
             cluster_columns              = TRUE,
             show_row_names               = TRUE,
             show_column_names            = TRUE,
             show_row_dend                = TRUE,
             show_column_dend             = FALSE,
             clustering_distance_rows     = "euclidean",
             clustering_method_rows       = "ward",
             clustering_distance_columns  = "euclidean",
             clustering_method_columns    = "ward",
             #row_order                   = order(sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup),
             #row_split                    = sample_Curie_ATRT_primTum_Plus$RNAseq_subgroup,
             column_split                 = vladoiu_cellTypes_cibersort$cluster_lineage,
             cluster_row_slices           = TRUE,
             cluster_column_slices        = FALSE,
             column_names_gp              = gpar(fontsize = 20),
             column_names_max_height      = unit(15,"cm"),
             row_names_gp                 = gpar(fontsize = 20),
             show_heatmap_legend          = FALSE,
             row_split                    = 3,
             column_title_gp              = gpar(fontsize = 20, fontface="bold"),
             row_title_gp                 = gpar(fontsize = 20, fontface="bold"),
             col                          = col_fun_mouseVladoiu2019,
             row_title                    = c("A","B","C")
             )
#pdf(file = paste0(pathToFigures,"heatmap_CIBERSORT_fraction_scRNAseq_integrated_R26_PC50.pdf"), width = 20, height =10) 
draw(h)
#dev.off()

# Legend of relative abundance
legend_fraction <- Legend(col_fun         = col_fun_mouseVladoiu2019,
                          title           = "relative abundance\n",
                          grid_height     = unit(1,"cm"),
                          legend_width    = unit(10,"cm"),
                          title_gp        = gpar(fontsize = 20, fontface = "bold"),
                          title_position  = "topcenter",
                          direction       = "horizontal",
                          labels_gp       = gpar(fontsize=15)
                          )

# Combine legends
legends <- packLegend(list        = list(legend_fraction),
                      direction   ="horizontal",
                      column_gap  = unit(2,"cm")
                      )

#pdf(file = paste0(pathToFigures,"legend_heatmap_CIBERSORT_fraction_scRNAseq_integrated_R26_PC50.pdf"), width = 20, height =10) 
draw(legends)
#dev.off()

```


## CHETAH

### Prepare input data

```{r input_data}

str(scRNAseq_Seurat_integrated_R26_PC50, levels=3)

# Extract cell matrix 
scRNAseq_Seurat_integrated_R26_PC50_chetah_input_matrix <- as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_PC50,
                                                                              slot = "data"
                                                                              )
                                                                 )
head(scRNAseq_Seurat_integrated_R26_PC50_chetah_input_matrix[,1:3])

# Extract UMAP coordinates 
scRNAseq_Seurat_integrated_R26_PC50_chetah_input_umap <- scRNAseq_Seurat_integrated_R26_PC50@reductions$umap@cell.embeddings
head(scRNAseq_Seurat_integrated_R26_PC50_chetah_input_umap)

# Create singleCellExperiment object
scRNAseq_Seurat_integrated_R26_PC50_chetah_input_se <- SingleCellExperiment(assays = list(counts = scRNAseq_Seurat_integrated_R26_PC50_chetah_input_matrix),
                                                                        reducedDims = SimpleList(umap = scRNAseq_Seurat_integrated_R26_PC50_chetah_input_umap)
                                                                        )

```

### Prepare reference data

```{r reference_data}

# Load reference data from CHETAH
load(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/data/CHETAH_TME_reference.Rdata")

reference

head(colData(reference))

```

### Running CHETAH

```{r running_chetah}

# Run CHETAH
scRNAseq_Seurat_integrated_R26_PC50_chetah_result <- CHETAHclassifier(input      = scRNAseq_Seurat_integrated_R26_PC50_chetah_input_se,
                                                                  ref_cells  = reference
                                                                  )

scRNAseq_Seurat_integrated_R26_PC50_chetah_result

#pdf(file = paste0(pathToFigures,"plotCHETAH_scRNAseq_integrated_R26_PC50.pdf"), width = 20, height =10) 
PlotCHETAH(input = scRNAseq_Seurat_integrated_R26_PC50_chetah_result)
#dev.off()

```


### Figure 10 patchwork

#### Boxplot expression of ganglionic eminence gene in mouse and human bulk RNAseq

```{r GEgene_boxplot}

###############################################################
# Expression of Ganglionic Eminence gene in mouse bulk-RNAseq #
###############################################################

# Import mouse primary RT TPM table
murine_RT_R26_primaryTum_melt_plus <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/script/murine_RT_R26_primaryTum_melt_plus.txt",
                                                 header = TRUE
                                                 )
head(murine_RT_R26_primaryTum_melt_plus)


# Ganglionic Eminence genes
GE_genes <- c("Dlx1","Dlx2","Sox6","Vax1","Gsx1","Gsx2")

# Extract TPM table of GE genes
murine_RT_R26_primaryTum_melt_plus_GEgenes <- filter(.data = murine_RT_R26_primaryTum_melt_plus,
                                                     symbol %in% GE_genes
                                                     )
head(murine_RT_R26_primaryTum_melt_plus_GEgenes)

paperFig_fontsize = 6

# Boxplot of GE murie genes
boxplot_murine_RT_R26_primTum_GEgenes <- ggplot(data = murine_RT_R26_primaryTum_melt_plus_GEgenes
                                                ) +
    geom_boxplot(mapping = aes(x = group_name,
                               y = log2(TPM+1),
                               fill = group_name
                               )
                 ) +
    facet_wrap(facets = ~symbol,
               nrow = 3
               ) +
    xlab("") +
    ylab("") + #label = expression(paste(log[2],"TPM+1"))) +
    scale_fill_manual(values = group_name_col[levels(factor(murine_RT_R26_primaryTum_melt_plus_GEgenes$group_name))]
                      ) +
    scale_x_discrete(label = c("R26-SHH","R26-MYC")) +
    geom_hline(yintercept = 1.5, lty = 2, color = "red") +
    theme_bw() +
    theme(legend.position = "none",
          axis.text.x = element_text(size = paperFig_fontsize, colour = "black", angle = 45, hjust = 1, vjust = 0.85, face = "bold")
          )

boxplot_murine_RT_R26_primTum_GEgenes

###############################################################
# Expression of Ganglionic Eminence gene in human bulk-RNAseq #
###############################################################

# Import human primary RT TPM table
human_ATRT_primaryTum_melt_plus <- read.table(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/human_ATRT/svn/analyse/script/human_ATRT_primaryTum_melt_plus.txt",
                                                 header = TRUE
                                                 )
head(human_ATRT_primaryTum_melt_plus)


# Ganglionic Eminence genes
GE_genes_human <- c("DLX1","DLX2","SOX6","VAX1","GSX1","GSX2")

# Extract TPM table of GE genes
human_ATRT_primaryTum_melt_plus_GEgenes <- filter(.data = human_ATRT_primaryTum_melt_plus,
                                                     Gene %in% GE_genes_human
                                                     )
head(human_ATRT_primaryTum_melt_plus_GEgenes)

paperFig_fontsize = 6

# Boxplot of GE murie genes
boxplot_human_ATRT_primTum_GEgenes <- ggplot(data = human_ATRT_primaryTum_melt_plus_GEgenes
                                                ) +
    geom_boxplot(mapping = aes(x = groupedLoc,
                               y = log2(TPM+1),
                               fill = groupedLoc
                               )
                 ) +
    facet_wrap(facets = ~Gene,
               nrow = 3
               ) +
    xlab("") +
    ylab(label = expression(paste(log[2],"TPM+1"))) +
    scale_fill_manual(values = group_name_col[levels(factor(human_ATRT_primaryTum_melt_plus_GEgenes$groupedLoc))]
                      ) +
    scale_x_discrete(label = c("MYC/MYC","SHH/MYC","SHH/SHH","TYR/TYR")) +
    geom_hline(yintercept = 1.5, lty = 2, color = "red") +
    theme_bw() +
    theme(legend.position = "none",
          axis.text.x = element_text(size = paperFig_fontsize, colour = "black", angle = 45, hjust = 1, vjust = 0.85, face = "bold")
          )

dev.new()

boxplot_human_ATRT_primTum_GEgenes

```

#### UMAP of cell type annotation

```{r umap_figure10}

#################################
# UMAP and cell type annotation #
#################################

# Create table of cluster cell type
cellType_table <- data.frame(cluster = c(0,1,2,3,4,5,6,7,8,9,10,11,12),
                             cellType = c("unspecified",#0
                                          "intermediate neural progenitor like 1",#1
                                          "neuron restricted progenitor like 1",#2
                                          "NSC like 1",#3
                                          "NSC like 2",#4
                                          "cycling cells (G1S)",#5
                                          "intermediate neural progenitor like 2",#6
                                          "neuron restricted progenitor like 2",#7
                                          "cycling cells (G2M)",#8
                                          "committed neural progenitor like",#9
                                          "immune reactive",#10
                                          "astrocyte like",#11
                                          "mature cell type"#12
                                          ),
                             cellType2 = c("unspecified",#0
                                          "INP like 1",#1
                                          "NRP like 1",#2
                                          "NSC like 1",#3
                                          "NSC like 2",#4
                                          "G1S",#5
                                          "INP like 2",#6
                                          "NRP like 2",#7
                                          "G2M",#8
                                          "CNP like",#9
                                          "immune reactive",#10
                                          "astrocyte like",#11
                                          "mature cell type"#12
                                          ),
                             cellType3 = c("unspecified",#0
                                          "intermediate neural progenitor (INP) like 1",#1
                                          "neuron restricted progenitor (NRP) like 1",#2
                                          "NSC like 1",#3
                                          "NSC like 2",#4
                                          "cycling cells (G1S)",#5
                                          "intermediate neural progenitor (INP) like 2",#6
                                          "neuron restricted progenitor (NRP) like 2",#7
                                          "cycling cells (G2M)",#8
                                          "committed neural progenitor (CNP) like",#9
                                          "immune reactive",#10
                                          "astrocyte like",#11
                                          "mature cell type"#12
                                          ),
                             cellType4 = c("unspecified",#0
                                          "intermediate neural progenitor (INP) 1",#1
                                          "neuron restricted progenitor (NRP) 1",#2
                                          "NSC 1",#3
                                          "NSC 2",#4
                                          "cycling cells (G1S)",#5
                                          "intermediate neural progenitor (INP) 2",#6
                                          "neuron restricted progenitor (NRP) 2",#7
                                          "cycling cells (G2M)",#8
                                          "committed neural progenitor (CNP)",#9
                                          "immune reactive",#10
                                          "astrocyte like",#11
                                          "mature cell type"#12
                                          ),
                             cellType5 = c("unspecified",#0
                                          "INP 1",#1
                                          "NRP 1",#2
                                          "NSC 1",#3
                                          "NSC 2",#4
                                          "G1S",#5
                                          "INP 2",#6
                                          "NRP 2",#7
                                          "G2M",#8
                                          "CNP",#9
                                          "immune reactive",#10
                                          "astrocyte like",#11
                                          "mature cell type"#12
                                          ),
                             cellType6 = c("Unspecified",#0
                                           "Neural progenitor 1 (NP 1)",#1
                                           "Neuron restricted progenitor 1 (NRP 1)",#2
                                           "Neural stem cell like 1 (NSC-Like 1)",#3
                                           "Neural stem cell like 2 (NSC-Like 2)",#4
                                           "Cycling cells (G1S)",#5
                                           "Neural progenitor 2 (NP 2)",#6
                                           "Neuron restricted progenitor 2 (NRP 2)",#7
                                           "Cycling cells (G2M)",#8
                                           "Committed neural progenitor (CNP)",#9
                                           "Immune reactive cells",#10
                                           "Astro-glial like cells",#11
                                           "Endothelial cells"#12
                                           ),
                             cellType7 = c("Unspecified",#0
                                           "NP 1",#1
                                           "NRP 1",#2
                                           "NSC-Like 1",#3
                                           "NSC-Like 2",#4
                                           "G1S",#5
                                           "NP 2",#6
                                           "NRP 2",#7
                                           "G2M",#8
                                           "CNP",#9
                                           "Immune reactive cells",#10
                                           "Astro-glial like cells",#11
                                           "Endothelial cells"#12
                                           )
                             )
head(cellType_table)

# Convert cluster number to factor (otherwise join will not work correctly)
cellType_table$cluster <- as.factor(cellType_table$cluster)

# Create new meta data with cell type annotation
metaData_cellAnnot <- left_join(x = scRNAseq_Seurat_integrated_R26_tumCells@meta.data,
                                y = cellType_table,
                                by = c("integrated_snn_res.0.4" = "cluster")
                                )
head(metaData_cellAnnot[,c("cellType","integrated_snn_res.0.4")])

# Add cell type annotation to metadata
scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells,
                                                             metadata = metaData_cellAnnot$cellType,
                                                             col.name = "cell_type"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType2,
                                                             col.name = "cell_type2"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType3,
                                                             col.name = "cell_type3"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType4,
                                                             col.name = "cell_type4"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType5,
                                                             col.name = "cell_type5"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType6,
                                                             col.name = "cell_type6"
                                                             )

scRNAseq_Seurat_integrated_R26_tumCells_annot <- AddMetaData(object   = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                             metadata = metaData_cellAnnot$cellType7,
                                                             col.name = "cell_type7"
                                                             )

# Export cell type annotated meta data (for inferCNV analysis on individual sample script)
write.csv(x = scRNAseq_Seurat_integrated_R26_tumCells_annot@meta.data,
          file = "scRNAseq_Seurat_integrated_R26_tumCells_annot_metaData.csv"
          )

# Add cluster number before
scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data <- scRNAseq_Seurat_integrated_R26_tumCells_annot@meta.data

scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data$cell_type8 <- paste(scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data$integrated_snn_res.0.4,
scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data$cell_type6,
sep = ". "
)

levels(factor(scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data$cell_type8))


# Create legend of cell type
scRNAseq_Seurat_integrated_R26_tumCells_annot_meta.data %>%
    group_by(cell_type8) %>%
    summarize(number = n()) ->
    cell_type_name_nb
cell_type_name_nb

# Cell type named color vector
cluster_col <- c("0"   = "skyblue",#0
                 "1"   = "skyblue4",#1
                 "2"   = "springgreen",#2
                 "3"   = "springgreen4",#3
                 "4"   = "sienna4",#4
                 "5"   = "violet",#5
                 "6"   = "sienna1",#6
                 "7"   = "sienna3",#7
                 "8"   = "violetred",#8
                 "9"   = "blue4",#9
                 "10"  = "yellow",#10
                 "11"  = "blue",#11
                 "12"  =  "red" #12
                 )

cell_type_name_col <- c("unspecified"                            = "skyblue",#0
                        "intermediate neural progenitor like 1"  = "skyblue4",#1
                        "neuron restricted progenitor like 1"    = "springgreen",#2
                        "NSC like 1"                             = "springgreen4",#3
                        "NSC like 2"                             = "sienna4",#4
                        "cycling cells (G1S)"                    = "violet",#5
                        "intermediate neural progenitor like 2"  = "sienna1",#6
                        "neuron restricted progenitor like 2"    = "sienna3",#7
                        "cycling cells (G2M)"                    = "violetred",#8
                        "committed neural progenitor like"       = "blue4",#9
                        "immune reactive"                        = "yellow",#10
                        "astrocyte like"                         = "blue",#11
                        "mature cell type"                       =  "red" #12
                        )

cell_type_name2_col <- c("unspecified"      = "skyblue",#0
                        "INP like 1"        = "skyblue4",#1
                        "NRP like 1"        = "springgreen",#2
                        "NSC like 1"        = "springgreen4",#3
                        "NSC like 2"        = "sienna4",#4
                        "G1S"               = "violet",#5
                        "INP like 2"        = "sienna1",#6
                        "NRP like 2"        = "sienna3",#7
                        "G2M"               = "violetred",#8
                        "CNP like"          = "blue4",#9
                        "immune reactive"   = "yellow",#10
                        "astrocyte like"    = "blue",#11
                        "mature cell type"  = "red" #12
                        )

cell_type_name3_col <- c("unspecified"                                 = "skyblue",#0
                        "intermediate neural progenitor (INP) like 1"  = "skyblue4",#1
                        "neuron restricted progenitor (NRP) like 1"    = "springgreen",#2
                        "NSC like 1"                                   = "springgreen4",#3
                        "NSC like 2"                                   = "sienna4",#4
                        "cycling cells (G1S)"                          = "violet",#5
                        "intermediate neural progenitor (INP) like 2"  = "sienna1",#6
                        "neuron restricted progenitor (NRP) like 2"    = "sienna3",#7
                        "cycling cells (G2M)"                          = "violetred",#8
                        "committed neural progenitor (CNP) like"       = "blue4",#9
                        "immune reactive"                              = "yellow",#10
                        "astrocyte like"                               = "blue",#11
                        "mature cell type"                             =  "red" #12
                        )

cell_type_name4_col <- c("unspecified"                            = "skyblue",#0
                        "intermediate neural progenitor (INP) 1"  = "skyblue4",#1
                        "neuron restricted progenitor (NRP) 1"    = "springgreen",#2
                        "NSC 1"                                   = "springgreen4",#3
                        "NSC 2"                                   = "sienna4",#4
                        "cycling cells (G1S)"                     = "violet",#5
                        "intermediate neural progenitor (INP) 2"  = "sienna1",#6
                        "neuron restricted progenitor (NRP) 2"    = "sienna3",#7
                        "cycling cells (G2M)"                     = "violetred",#8
                        "committed neural progenitor (CNP)"       = "blue4",#9
                        "immune reactive"                         = "yellow",#10
                        "astrocyte like"                          = "blue",#11
                        "mature cell type"                        =  "red" #12
                        )

cell_type_name5_col <- c("unspecified"      = "skyblue",#0
                        "INP 1"             = "skyblue4",#1
                        "NRP 1"             = "springgreen",#2
                        "NSC 1"             = "springgreen4",#3
                        "NSC 2"             = "sienna4",#4
                        "G1S"               = "violet",#5
                        "INP 2"             = "sienna1",#6
                        "NRP 2"             = "sienna3",#7
                        "G2M"               = "violetred",#8
                        "CNP"               = "blue4",#9
                        "immune reactive"   = "yellow",#10
                        "astrocyte like"    = "blue",#11
                        "mature cell type"  = "red" #12
                        )

cell_type_name6_col <- c("Unspecified"                             = "skyblue",#0
                         "Neural progenitor 1 (NP 1)"              = "skyblue4",#1
                         "Neuron restricted progenitor 1 (NRP 1)"  = "springgreen",#2
                         "Neural stem cell like 1 (NSC-Like 1)"    = "springgreen4",#3
                         "Neural stem cell like 2 (NSC-Like 2)"    = "sienna4",#4
                         "Cycling cells (G1S)"                     = "violet",#5
                         "Neural progenitor 2 (NP 2)"              = "sienna1",#6
                         "Neuron restricted progenitor 2 (NRP 2)"  = "sienna3",#7
                         "Cycling cells (G2M)"                     = "violetred",#8
                         "Committed neural progenitor (CNP)"       = "blue4",#9
                         "Immune reactive cells"                   = "yellow",#10
                         "Astro-glial like cells"                  = "blue",#11
                         "Endothelial cells"                       = "red"#12
                         )

cell_type_name7_col <- c("Unspecified"             = "skyblue",#0
                         "NP 1"                    = "skyblue4",#1
                         "NRP 1"                   = "springgreen",#2
                         "NSC-Like 1"              = "springgreen4",#3
                         "NSC-Like 2"              = "sienna4",#4
                         "G1S"                     = "violet",#5
                         "NP 2"                    = "sienna1",#6
                         "NRP 2"                   = "sienna3",#7
                         "G2M"                     = "violetred",#8
                         "CNP"                     = "blue4",#9
                         "Immune reactive cells"   = "yellow",#10
                         "Astro-glial like cells"  = "blue",#11
                         "Endothelial cells"       = "red"#12
                         )

cell_type_name8_col <- c("0. Unspecified"                             = "skyblue",#0
                         "1. Neural progenitor 1 (NP 1)"              = "skyblue4",#1
                         "2. Neuron restricted progenitor 1 (NRP 1)"  = "springgreen",#2
                         "3. Neural stem cell like 1 (NSC-Like 1)"    = "springgreen4",#3
                         "4. Neural stem cell like 2 (NSC-Like 2)"    = "sienna4",#4
                         "5. Cycling cells (G1S)"                     = "violet",#5
                         "6. Neural progenitor 2 (NP 2)"              = "sienna1",#6
                         "7. Neuron restricted progenitor 2 (NRP 2)"  = "sienna3",#7
                         "8. Cycling cells (G2M)"                     = "violetred",#8
                         "9. Committed neural progenitor (CNP)"       = "blue4",#9
                         "10. Immune reactive cells"                   = "yellow",#10
                         "11. Astro-glial like cells"                  = "blue",#11
                         "12. Endothelial cells"                       = "red"#12
                         )

# Order cell type names
cell_type_name_nb$cell_type8 <- factor(cell_type_name_nb$cell_type8, levels = c("0. Unspecified",#0
                         "1. Neural progenitor 1 (NP 1)",#1
                         "2. Neuron restricted progenitor 1 (NRP 1)",#2
                         "3. Neural stem cell like 1 (NSC-Like 1)",#3
                         "4. Neural stem cell like 2 (NSC-Like 2)",#4
                         "5. Cycling cells (G1S)",#5
                         "6. Neural progenitor 2 (NP 2)",#6
                         "7. Neuron restricted progenitor 2 (NRP 2)",#7
                         "8. Cycling cells (G2M)",#8
                         "9. Committed neural progenitor (CNP)",#9
                         "10. Immune reactive cells",#10
                         "11. Astro-glial like cells",#11
                         "12. Endothelial cells"#12
                         )
                                        )

# Create legend
ggplot(data = cell_type_name_nb
            ) +
    geom_bar(mapping = aes(x = cell_type8,
                           y = number,
                           fill = cell_type8
                           ),
             stat = 'identity'
             ) +
    scale_fill_manual(name = "Cell Type", values = cell_type_name8_col[levels(factor(cell_type_name_nb$cell_type8))]
                      ) +
    guides(fill = guide_legend(nrow=7)) +
    theme(text = element_text(size = paperFig_fontsize*1.5),
          panel.background = element_rect(fill = "black"),
          legend.key.height = unit(0.35,"cm"),
          legend.key.width = unit(0.25,"cm")#,
          #legend.direction = "horizontal"
          ) -> p

library(cowplot)
legend_cellTypes <- get_legend(p)

# UMAP and cell type
umap_scRNAseq_Seurat_integrated_R26_tumCells_annot <- DimPlot(object     = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                              reduction   = "umap",
                                                              pt.size     = 0.25,
                                                              group.by    = "integrated_snn_res.0.4",
                                                              label       = TRUE,
                                                              label.size  = paperFig_fontsize/3,
                                                              cols        = c("skyblue",#0
                                                                              "skyblue4",#1
                                                                              "springgreen",#2
                                                                              "springgreen4",#3
                                                                              "sienna4",#4
                                                                              "violet",#5
                                                                              "sienna1",#6
                                                                              "sienna3",#7
                                                                              "violetred",#8
                                                                              "blue4",#9
                                                                              "yellow",#10
                                                                              "blue",#11
                                                                              "red"#12
                                                                              ),
                                                              label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
                                                              label.box  = TRUE
                                                              ) +
    xlab("") +
    ylab("") +
    theme(legend.position = "none",
          text = element_text(size = paperFig_fontsize),
          axis.text = element_text(size = paperFig_fontsize),
          axis.ticks = element_line(size = 0.25),
          axis.line = element_line(size = 0.25),
          panel.background = element_rect(fill = "white")
          )
umap_scRNAseq_Seurat_integrated_R26_tumCells_annot

# Export table for source data
write.csv(x = cbind(scRNAseq_Seurat_integrated_R26_tumCells_annot@reductions$umap@cell.embeddings,
                    data.frame(cluster = scRNAseq_Seurat_integrated_R26_tumCells_annot@meta.data[,"integrated_snn_res.0.4"])
                    ),
          file = "../LobonIglesias_sourceData/Figure5_A.csv",
          row.names = TRUE
     )

# Figure for presentation
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_clust_presentation.pdf"), width = 12, height = 10)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                              reduction   = "umap",
                                                              pt.size     = 1.5,
                                                              group.by    = paste0("integrated_snn_res.0.4"),
                                         #group.by    = "cell_subtype_name",
                                                              label       = TRUE,
                                                              label.size  = 20,
                                                              cols        = c("skyblue",#0
                                                                              "skyblue4",#1
                                                                              "springgreen",#2
                                                                              "springgreen4",#3
                                                                              "sienna4",#4
                                                                              "violet",#5
                                                                              "sienna1",#6
                                                                              "sienna3",#7
                                                                              "violetred",#8
                                                                              "blue4",#9
                                                                              "yellow",#10
                                                                              "blue",#11
                                                                              "red"#12
                                                                              )#,
                                                              #label.color = "white",#c("white","black","black","white","white","white","black","black","white","white"),
                                                              #label.box  = TRUE
                                                              ) +
    xlab("") +
    ylab("") +
    theme(legend.position = "none",
          text = element_text(size = 60),
          axis.text = element_text(size = 20),
          axis.ticks = element_line(size = 0.25),
          axis.line = element_line(size = 0.25),
          panel.background = element_rect(fill = "white")
          )
#dev.off()

# Create legend
p <- ggplot(data = cell_type_name_nb
            ) +
    geom_bar(mapping = aes(x = cell_type6,
                           y = number,
                           fill = cell_type6
                           ),
             stat = 'identity'
             ) +
    scale_fill_manual(name = "Cell Type", values = cell_type_name6_col[levels(factor(cell_type_name_nb$cell_type6))]
                      ) +
    guides(fill = guide_legend(ncol=1)) +
    theme(text = element_text(size = 40),
          panel.background = element_rect(fill = "black"),
          legend.key.height = unit(0.5,"cm"),
          legend.key.width = unit(2,"cm")#,
          #legend.direction = "horizontal"
          )
legend_cellTypes <- get_legend(p)

#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_clust_presentation_lgd.pdf"), width = 14, height = 10)
as_ggplot(legend_cellTypes)
#dev.off()

```

#### Stacked violon plot of cluster marker genes

```{r stackedViolon_plot}

# Import cluster marker genes
cell_type_genes <- read.table(file   = paste0(pathToInfo,"cluster_marker_genes.csv"),
                              header = TRUE,
                              sep    = ";"
                              )

cell_type_genes$gene[!is.element(cell_type_genes$gene,rownames(scRNAseq_Seurat_integrated_R26_tumCells_annot))]

# Select genes for cluster 0,1,2,3,4,6,7,9 and 11
cell_type_genes_013467911 <- filter(cell_type_genes,
                                    include == "yes",
                                    cluster %in% c(0,1,2,3,4,6,7,9,11)
                                    )
cell_type_genes_013467911 <- cell_type_genes_013467911[!is.na(cell_type_genes_013467911$genes),]
head(cell_type_genes_013467911)

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_annot_0123467911 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                                   integrated_snn_res.0.4 %in% c(0,1,2,3,4,6,7,9,11)
                                                                   )


# Order genes
#match(cell_type_genes$genes, c("Twist1","Nrgn","Nes","Nr4a3","Sox9","Sox2","Gadd45g","Dll1","Dll3","Hes6","Ascl1","Elavl4","Dlx1","Dlx2","Elavl3","Dcx","Stmn2","Tubb3","Gfap","S100b","Atp1a2"))

# Plot stacked violinplot
StackedVlnPlot_murine_RT_R26_cellTypes <- StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_0123467911,
                                                         features  = c("Twist1","Nrgn","Nes","Nr4a3","Sox9","Ttyh1","Sox2","Gadd45g","Dll1","Dll3","Hes6","Ascl1","Elavl4","Dlx1","Dlx2","Gsx1","Elavl3","Dcx","Stmn2","Tubb3","Gfap","S100b","Atp1a2"),#cell_type_genes$genes,
                                        #plot.margin                          = unit(c(0,0,0,0), "cm"),
                                                         group.by  = "cell_type5",
                                                         cols      = cell_type_name5_col,
                                                         fontsize  = paperFig_fontsize
                                                         ) +
    scale_x_discrete(limits = c("astrocyte like","CNP","NRP2","NRP1","INP1","INP2","NSC1","NSC2"))
StackedVlnPlot_murine_RT_R26_cellTypes

#pdf(file = paste0(pathToFigures,"StackedVlnPlot_murine_RT_scRNAseq_R26_integrated_cellTypes_markerGenes.pdf"), width = 14, height = 12)
StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot,
               #features                             = c(clusterMarkerGenes2,clusterMarkerGenes3),
                                           features  = cell_type_genes_013467911$genes,
               #plot.margin                          = unit(c(0,0,0,0), "cm"),
                                           group.by  = "cell_type2",
                                           cols      = cell_type_name2_col,
                                           fontsize  = 20
                                           )
#dev.off()

# Select genes for cluster 10, 11 and 12
cell_type_genes_101112 <- filter(cell_type_genes,
                                 cluster %in% c(10,11,12)
                                 )
cell_type_genes_101112 <- cell_type_genes_101112[!is.na(cell_type_genes_101112$genes),]
cell_type_genes_101112

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_annot_10_11_12 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                                 cell_type5 %in% c("immune reactive","astrocyte like","mature cell type")
                                                                 )


clusterMarkerGenes4 <- c("C1qa","Fcer1g","Ccl4","Gfap","Aqp4","Rassf9","Tm4sf1")

#pdf(file = paste0(pathToFigures,"StackedVlnPlot_murine_RT_scRNAseq_R26_integrated_cellTypes_10_11_12_markerGenes.pdf"), width = 12, height = 14)
StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_10_11_12,
               #features                             = c(clusterMarkerGenes2,clusterMarkerGenes3),
                                           features  = cell_type_genes_101112$gene,
               #plot.margin                          = unit(c(0,0,0,0), "cm"),
                                           group.by  = "cell_type2",
                                           cols      = cell_type_name2_col,
                                           fontsize  = 40
                                           )
#dev.off()

# Select genes for cluster 5, 8, 10, 11 and 12
cell_type_genes_58101112 <- filter(cell_type_genes,
                                   cluster %in% c(5,8,10,11,12),
                                   include == 'yes'
                                 )
cell_type_genes_58101112 <- cell_type_genes_58101112[!is.na(cell_type_genes_58101112$genes),]
cell_type_genes_58101112


# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                                     cell_type5 %in% c("G1S","G2M","immune reactive","astrocyte like","mature cell type")
                                                                     )


clusterMarkerGenes5 <- c("C1qa","Fcer1g","Ccl4","Gfap","Aqp4","Rassf9","Tm4sf1")

scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12@meta.data$cell_type7 <- factor(scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12@meta.data$cell_type7,
                                                                                          levels = c("Endothelial cells","Astro-glial like cells","Immune reactive cells","G1S","G2M")
                                                                                          )

#pdf(file = paste0(pathToFigures,"StackedVlnPlot_murine_RT_scRNAseq_R26_integrated_cellTypes_5_8_10_11_12_markerGenes.pdf"), width = 12, height = 14)
StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12,
               features  = cell_type_genes_58101112$gene,
               group.by  = "cell_type7",
               cols      = cell_type_name7_col,
               fontsize  = 40
               )
#dev.off()

StackedVlnPlot_cellTypes_5_8_10_11_12_markerGenes <- StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12,
                                                                    features  = cell_type_genes_58101112$gene,
                                                                    group.by  = "cell_type7",
                                                                    cols      = cell_type_name7_col,
                                                                    fontsize  = paperFig_fontsize*1.5
                                                                    )

# Export table for sourc data
write.csv(x = cbind(data.frame(cell_type = scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12@meta.data$cell_type7),
                    t(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_5_8_10_11_12@assays$RNA@data[cell_type_genes_58101112$gene,]))
                    ),
          file = "../LobonIglesias_sourceData/SuppFig6_B.csv",
          row.names = TRUE
          )

# Select genes for cluster 1,2,3,4,6,7 and 9
cell_type_genes_1234679 <- filter(cell_type_genes,
                                  cluster %in% c(1,2,3,4,6,7,9),
                                  include == "yes"
                                 )
cell_type_genes_1234679 <- cell_type_genes_58101112[!is.na(cell_type_genes_1234679$genes),]
cell_type_genes_1234679

levels(factor(scRNAseq_Seurat_integrated_R26_tumCells_annot@meta.data$integrated_snn_res.0.4))

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679 <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                                     integrated_snn_res.0.4 %in% c(1,2,3,4,6,7,9)
                                                                     )

# Order cell type names
scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679@meta.data$cell_type7 <- factor(scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679@meta.data$cell_type7, levels = c("CNP","NRP 1","NRP 2","NP 1","NP 2","NSC-Like 1","NSC-Like 2"))

# Order gene names
cell_type_genes_1234679$gene

v <- c("Ifitm2","Foxc1","Nes","Pmp22","Twist1","Col3a1",
       "Sox9","Nr4a3",
       "Ttyh1",
       "Dlx1","Dlx2","Sox2","Elavl4","Ascl1","Hes6","Gadd45g","Dll1","Elavl3","Gsx1","Dll3",
       "Tubb3",
       "Dcx","Stmn2"
       )
       
#pdf(file = paste0(pathToFigures,"StackedVlnPlot_murine_RT_scRNAseq_R26_integrated_cellTypes_1234679_markerGenes.pdf"), width = 12, height = 14)
StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679,
               features  = v,#cell_type_genes_1234679$gene,
               group.by  = "cell_type7",
               cols      = cell_type_name7_col,
               fontsize  = 10
               )
#dev.off()

# For paper figure
StackedVlnPlot_murine_RT_R26_cellTypes_1234679 <- StackedVlnPlot(obj       = scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679,
                                                                 features  = v,#cell_type_genes_1234679$gene,
                                                                 group.by  = "cell_type7",
                                                                 cols      = cell_type_name7_col,
                                                                 fontsize  = paperFig_fontsize
                                                                 )

# Export table for sourc data
write.csv(x = cbind(data.frame(cell_type = scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679@meta.data$cell_type7),
                    t(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_1234679@assays$RNA@data[v,]))
                    ),
          file = "../LobonIglesias_sourceData/Figure5_B.csv",
          row.names = TRUE
          )

```

#### Cluster specific regulons

```{r cluster_specific_regulons}

source("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/script/reorder_within.R")

scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster <- factor(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster)

# Create function for plotting regulons of interest per cell type
plotInterestedReg <- function(clusters,
                              cell_type_name,
                              intReg = NULL
                              )
{
    f <- scRNAseq_integrated_R26_scenic_rss_matrix_melt[which(scRNAseq_integrated_R26_scenic_rss_matrix_melt$cluster %in% clusters),]
    if (is.null(intReg))
    {
        intReg <- f[order(f$value, decreasing = TRUE),] %>% .[1:5,"variable"]
    }
    #intReg <- paste0(dplyr::filter(.data = interestedRegulons, Cluster %in% clusters)$Gene,"(+)")
    p <- ggplot(data = f,
                mapping = aes(x = reorder_within(variable,-value,cluster),
                              y = value
                              )
                ) +
        facet_wrap(facets = ~ cluster) +
        ggtitle(cell_type_name) +
        xlab("") +
        ylab("RSS score") +
        ylim(0,0.5) +
        geom_point(color = "grey",
                   size = 0.5
                   ) +
        geom_point(data = filter(f, variable %in% intReg),
                   color = "red",
                   size = 0.5
                   ) + 
        geom_text_repel(data            = filter(f, variable %in% intReg),
                        mapping         = aes(label = variable),
                        segment.colour  = "black",
                        segment.size    = 0.25,
                        color           = "black",#cell_type_name_col[cell_type_name],
                        size            = paperFig_fontsize/3,
                        force           = 50,
                        force_pull      = 50
                        ) + 
        theme_bw() +
        theme(text = element_text(size = paperFig_fontsize),
              axis.ticks.x = element_blank(),
              axis.text.x = element_blank(),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              strip.background = element_rect(fill = cell_type_name7_col[cell_type_name]),
              strip.text = element_text(size = paperFig_fontsize*1.5, colour = "white", face = "bold"),
              plot.title = element_text(hjust = 0.5, face = "bold", size = paperFig_fontsize*1.5)
              )
}

# Export table for sourc data
write.csv(x = scRNAseq_integrated_R26_scenic_rss_matrix_melt,
          file = "../LobonIglesias_sourceData/Figure5_C.csv",
          row.names = FALSE
          )
       
cell_type_name7_col

# Plot interested regulons for cluster 0
interestedRegulons_R26_clust0 <- plotInterestedReg(clusters = "0",
                                                   cell_type_name = "Unspecified"
                                                   )

# Plot interested regulons for cluster 1
interestedRegulons_R26_clust1 <- plotInterestedReg(clusters = "1",
                                                   cell_type_name = "NP 1",#"intermediate neural progenitor like1",
                                                   intReg = c("Sox2(+)","Pou3f2(+)","Hoxb2(+)","Emx2(+)","Gsx1(+)")
                                                   )

# Plot interested regulons for cluster 2
interestedRegulons_R26_clust2 <- plotInterestedReg(clusters = "2",
                                                   cell_type_name = "NRP 1",#"neuron restricted progenitor like 1,
                                                   intReg = c("Onecut2(+)","Pdx1(+)","Arx(+)","Eomes(+)")
                                                   )

# Plot interested regulons for cluster 3
interestedRegulons_R26_clust3 <- plotInterestedReg(clusters = "3",
                                                   cell_type_name = "NSC-Like 1",#"NSC like 1"
                                                   intReg = c("Hes5(+)","Klf10(+)","Hic1(+)","Rela(+)","Nr1h3(+)")
                                                   )

# Plot interested regulons for cluster 4
interestedRegulons_R26_clust4 <- plotInterestedReg(clusters = "4",
                                                   cell_type_name = "NSC-Like 2",#"NSC like 2"
                                                   

# Plot interested regulons for cluster 5
interestedRegulons_R26_clust5 <- plotInterestedReg(clusters = "5",
                                                   cell_type_name = "G1S",#"cycling cells (G1S)"
                                                   )

# Plot interested regulons for cluster 6
interestedRegulons_R26_clust6 <- plotInterestedReg(clusters = "6",
                                                   cell_type_name = "NP 2",#"intermediate neural progenitor like 2"
                                                   intReg = c("Hes5(+)","Etv2(+)","Dlx1(+)","Pou3f3(+)","Hoxd12(+)")
                                                   )

# Plot interested regulons for cluster 7
interestedRegulons_R26_clust7 <- plotInterestedReg(clusters = "7",
                                                   cell_type_name = "NRP 2",#"neuron restricted progenitor like 2"
                                                   intReg = c("Neurod1(+)","Sema4a(+)","Sox10(+)","Otp(+)")
                                                   )

# Plot interested regulons for cluster 8
interestedRegulons_R26_clust8 <- plotInterestedReg(clusters = "8",
                                                   cell_type_name = "G2M"#"cycling cells (G2M)"
                                                   )

# Plot interested regulons for cluster 9
interestedRegulons_R26_clust9 <- plotInterestedReg(clusters = "9",
                                                   cell_type_name = "CNP",#"committed neural progenitor like"
                                                   intReg = c("Neurod1(+)","Bhlhe23(+)","Onecut1(+)","Meis3(+)")
                                                   )

# Plot interested regulons for cluster 10
interestedRegulons_R26_clust10 <- plotInterestedReg(clusters = "10",
                                                   cell_type_name = "Immune reactive cells"#"immune reactive"
                                                   )

# Plot interested regulons for cluster 11
interestedRegulons_R26_clust11 <- plotInterestedReg(clusters = "11",
                                                   cell_type_name = "Astro-glial like cells"#"astrocyte like"
                                                   )

# Plot interested regulons for cluster 12
interestedRegulons_R26_clust12 <- plotInterestedReg(clusters = "12",
                                                   cell_type_name = "Endothelial cells"#"mature cell type"
                                                   )

# Plot interessted regulons for cluster 5 and 8
interestedRegulons_R26_clust58 <- grid.arrange(arrangeGrob(interestedRegulons_R26_clust5 +
                                                       theme(axis.title.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_R26_clust8 +
                                                       theme(axis.title.y = element_blank(),
                                                             #axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("Cycling Cells",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      ),
                                                       left = textGrob("RSS score",
                                                                       rot = 90,
                                                                       vjust = 1,
                                                                      gp = gpar(fontsize = paperFig_fontsize)
                                                                       )
                                                       )
                                           )

# Plot interessted regulons for cluster 2 and 7
interestedRegulons_R26_clust27 <- grid.arrange(arrangeGrob(interestedRegulons_R26_clust2 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_R26_clust7 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("NRP clusters",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                                       #rot = 90,
                                                                       #vjust = 1,
                                                                      #gp = gpar(fontsize = paperFig_fontsize)
                                                                       #)
                                                       )
                                           )

# Plot interessted regulons for cluster 1 and 6
interestedRegulons_R26_clust16 <- grid.arrange(arrangeGrob(interestedRegulons_R26_clust1 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       interestedRegulons_R26_clust6 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("NP clusters",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                                       #rot = 90,
                                                                       #vjust = 1,
                                                                      #gp = gpar(fontsize = paperFig_fontsize)
                                                                       #)
                                                       )
                                           )

# Plot interessted regulons for cluster 3 and 4
interestedRegulons_R26_clust34 <- grid.arrange(arrangeGrob(interestedRegulons_R26_clust3 +
                                                           theme(axis.title.y = element_blank(),
                                                                 axis.text.y = element_blank(),
                                                                 plot.title = element_blank()
                                                                 ),
                                                       interestedRegulons_R26_clust4 +
                                                       theme(axis.title.y = element_blank(),
                                                             axis.text.y = element_blank(),
                                                             #axis.ticks.y = element_blank(),
                                                             plot.title = element_blank()
                                                             ),
                                                       nrow = 1,
                                                       top = textGrob("NSC clusters",
                                                                      vjust = 1,
                                                                      gp = gpar(fontface = "bold",fontsize = paperFig_fontsize*1.5)
                                                                      )#,
                                                       #left = textGrob("RSS score",
                                                                       #rot = 90,
                                                                       #vjust = 1,
                                                                      #gp = gpar(fontsize = paperFig_fontsize)
                                                                       #)
                                                       )
                                           )

```

#### Trajectory inference

```{r trajectory_inference}

# Create TIA for paper figure
trajectory_inference_R26_monocle3_clust01234679 <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_01234679_cds_list[["PCmin"]],
                                                      color_cells_by                 = "integrated_snn_res.0.4",
                                                      cell_size                      = 0.25,
                                                      trajectory_graph_color         = "red",
                                                      trajectory_graph_segment_size  = 0.75,
                                                      label_cell_groups              = FALSE,
                                                      label_branch_points            = FALSE,
                                                      label_roots                    = FALSE,
                                                      label_leaves                   = FALSE
                                                      ) +
    scale_colour_manual(name = "Cell type", values = c("skyblue","skyblue4","springgreen","springgreen4","sienna4","sienna1","sienna3","blue4")) +
    theme_void() +
    theme(text = element_text(size = paperFig_fontsize),
          panel.background = element_rect(colour = "white", fill = "white"),
          legend.position = "none"
          )

# Create TIA for paper figure
trajectory_inference_R26_monocle3_clust1234679 <- plot_cells(cds                            = scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list[["PCmin"]],
                                                      color_cells_by                 = "integrated_snn_res.0.4",
                                                      cell_size                      = 0.25,
                                                      trajectory_graph_color         = "red",
                                                      trajectory_graph_segment_size  = 0.75,
                                                      label_cell_groups              = FALSE,
                                                      label_branch_points            = FALSE,
                                                      label_roots                    = FALSE,
                                                      label_leaves                   = FALSE
                                                      ) +
    scale_colour_manual(name = "Cell type",
                        values = cluster_col[levels(factor(scRNAseq_Seurat_integrated_R26_tumCells_1234679_cds_list$PCmin@colData$integrated_snn_res.0.4))]
                        ) +
    theme_void() +
    theme(text = element_text(size = paperFig_fontsize),
          panel.background = element_rect(colour = "white", fill = "white"),
          legend.position = "none"
          )

# Export table for sourc data
write.csv(x = mutate(.data = trajectory_inference_R26_monocle3_clust1234679$data[c("data_dim_1","data_dim_2","integrated_snn_res.0.4")],
                     cluster = integrated_snn_res.0.4,
                     .keep = "unused"
                     ),
          file = "../LobonIglesias_sourceData/Figure5_D.csv",
          row.names = TRUE
          )


```


#### RNA velocity figure 10

```{r trajectory_inference}

library(magick) # image_read
library(ggpubr) # background_image()

# Import scVelo stream
scVelo_clust01234679_DEG_monocle3 <- image_read("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/figures/scvelo_scRNAseq_integrated_R26_embeddingStream_clust01234679_DEG_monocle3_dynamical_fig.png")
scVelo_clust01234679_DEG_monocle3 <- ggplot()+
    background_image(scVelo_clust01234679_DEG_monocle3)

# Import scVelo stream
scVelo_clust1234679_DEG_monocle3 <- image_read("/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/murine_RT/svn/analyse/figures/scvelo_scRNAseq_integrated_R26_embeddingStream_clust1234679_DEG_monocle3_dynamical_fig.png")
scVelo_clust1234679_DEG_monocle3 <- ggplot()+
    background_image(scVelo_clust1234679_DEG_monocle3)

```

#### Heatmap of interesting genes on monocle3 embedding (for figure)

```{r trajectory_inference}

# Subset Seurat object
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes <- subset(x = scRNAseq_Seurat_integrated_R26_tumCells,
                                                             integrated_snn_res.0.4 %in% c(1,2,3,4,6,7,9)#,
                                                    #features = 
                                                             )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes)

# Extract expression matrix for cluster 0135 and all genes
scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes_matrix <- as(object = as.matrix(GetAssayData(object = scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes,
                                                                                       assay = "RNA",#"integrated",
                                                                                       slot = "data"#"scale.data"
                                                                                       )
                                                                          ),
   Class = "matrix"
   )
nrow(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes_matrix)

# Interesting genes 
interestingGenes <- c("Neurod1","Dcx","Rest","Id2","Id3","Id4","Dlx1","Dlx2","Ascl1","Ttyh1","Nes","Sox9","Mycn","Hes5","Hes1","Dll3","Dll1","Actl6a","Actl6b")

interestingGenes[!is.element(interestingGenes, rownames(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes_matrix))]

# Create a list of plots
plotList_heatmap_monocle3_interesingGenes <- list()
for (gene in interestingGenes)
{
    # Select genes
    scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix <- scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes_matrix[gene,]
    # Melt matrix
    scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix_melt <- melt(scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix)  
    # Merge matrix and metadata
    scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix_melt_plus <- merge(x = scRNAseq_Seurat_integrated_R26_tumCells_1234679_monocle3_UMAP_metaData,
                                                                                         y = scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix_melt,
                                                                                         by = 0
                                                                                        )
    # Plot 
    p <- ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_1234679_interestingGene_matrix_melt_plus,
                mapping = aes(x = data_dim_1,
                              y = data_dim_2
                              )
                ) +
        geom_point(mapping = aes(color = value),
                   size = 0.1
                   ) +
        ggtitle(label = gene) +
        scale_color_gradientn(name = "count", colours = c("grey","yellow","red","darkred")
                              ) +
        theme_void() +
#geom_segment(mapping = aes(x=-1.5, y=-4, xend=2, yend=1), lty=2, lwd=0.25) +
#geom_segment(mapping = aes(x=4, y=-4.5, xend=4.5, yend=0), lty=2, lwd=0.25) +
        theme(legend.position = "none",
              plot.title = element_text(hjust = 0.5, size = paperFig_fontsize*1.5),
              text = element_text(size = paperFig_fontsize)
              )
    plotList_heatmap_monocle3_interesingGenes[[length(plotList_heatmap_monocle3_interesingGenes)+1]] <- p
}
names(plotList_heatmap_monocle3_interesingGenes) <- interestingGenes

# Export table for source data
write.csv(x = cbind(trajectory_inference_R26_monocle3_clust1234679$data[c("data_dim_1","data_dim_2")],
                    t(scRNAseq_Seurat_integrated_R26_tumCells_1234679_allGenes_matrix[c("Rest","Id3","Actl6b","Dcx","Hes1","Ascl1","Dll1","Dll3"),])
                    ),
          file = "../LobonIglesias_sourceData/Figure5_F.csv",
          row.names = TRUE
          )

```

#### Patchwork figure 10

```{r patchwork_figure10}

figure10ab <- wrap_plots(boxplot_human_ATRT_primTum_GEgenes,
                         boxplot_murine_RT_R26_primTum_GEgenes,
                         umap_scRNAseq_Seurat_integrated_R26_tumCells_annot,
                         legend_cellTypes,
                         widths  = c(0.2,0.125,0.475,0.2)
                         )

figure10d1 <- wrap_plots(s = plot_spacer()
                         ,interestedRegulons_R26_clust9
                        ,s = plot_spacer()
                        ,interestedRegulons_R26_clust27
                        ,widths = c(0.1,0.25,0.1,0.6)
                         )

figure10d2 <- wrap_plots(interestedRegulons_R26_clust16
                        ,interestedRegulons_R26_clust34
                        ,widths = c(0.5,0.5)
                         )

figure10d <- wrap_plots(figure10d1,
                        figure10d2,
                        heights = c(0.5,0.5)
                        )

figure10cd <- wrap_plots(a = StackedVlnPlot_murine_RT_R26_cellTypes,
                         b = figure10d,
                         widths = c(0.35,0.65)
                         )

figure10ef <- wrap_plots(e = trajectory_inference_R26_monocle3_clust1234679,
                         #f = scVelo_clust01234679_DEG_monocle3,
                         f = scVelo_clust1234679_DEG_monocle3,
                         heights = c(0.5,0.5)
                         )

dev.new

figure10g <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["Actl6b"]],
                        plotList_heatmap_monocle3_interesingGenes[["Dcx"]],
                        plotList_heatmap_monocle3_interesingGenes[["Rest"]],
                        plotList_heatmap_monocle3_interesingGenes[["Id3"]],
                        nrow = 1
) +
    plot_annotation(title = 'Neuronal differentiation inhibitor')

figure10h <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["Id2"]],
                        plotList_heatmap_monocle3_interesingGenes[["Id4"]],
                        plotList_heatmap_monocle3_interesingGenes[["Dlx1"]],
                        plotList_heatmap_monocle3_interesingGenes[["Actl6a"]],
                        nrow = 1
                        )

figure10i <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["Ttyh1"]],
                        plotList_heatmap_monocle3_interesingGenes[["Nes"]],
                        plotList_heatmap_monocle3_interesingGenes[["Sox9"]],
                        plotList_heatmap_monocle3_interesingGenes[["Mycn"]],
                        nrow = 1
                        )

figure10j <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["Ascl1"]],
                        plotList_heatmap_monocle3_interesingGenes[["Dll1"]],
                        plotList_heatmap_monocle3_interesingGenes[["Hes1"]],
                        plotList_heatmap_monocle3_interesingGenes[["Dll3"]],
                        nrow = 1
                        )

figure10ghij <- wrap_plots(figure10g,
                            #figure10h,
                            #figure10i,
                               figure10j,
                           heights = c(0.5,0.5)
                            )
                                                       
figure10efg <- wrap_plots(e = figure10ef,
                         f = figure10ghij,
                         widths = c(0.3,0.7)
                         )

figure10 <- wrap_plots(a = figure10ab,
                       b = figure10cd,
                       c = figure10efg,
                       heights = c(0.25,0.35,0.4)
                       )

# Save image as pdf
ggsave(plot = figure10,
       file = paste0(pathToFigures,"figure10.pdf"),
       width = 210, #full a4 210
       height = 297, # full a4 297
       units = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figure10.pdf"),
              zerkaloDir = "mandrian_murine_RT"
              )

```

#### Patchwork figure 10bis (Figure 5 LobonIglesias2022)

```{r patchwork_figure10bis}

# UMAP and clusters
figure10bis_A <- wrap_plots(umap_scRNAseq_Seurat_integrated_R26_tumCells_annot) &
    labs(tag = 'A')
figure10bis_A <- wrap_plots(figure10bis_A,
                            legend_cellTypes,
                            heights =c(0.6,0.4)
                            )

# Stacked violon plots
figure10bis_B <- wrap_elements(StackedVlnPlot_murine_RT_R26_cellTypes_1234679) + labs(tag = 'B')

# UMAP and stacked violon plots
figure10bis_AB <- wrap_plots(figure10bis_A,
                             figure10bis_B,
                             widths = c(0.525,0.375)
                             )

# Regulons
figure10bis_C <- wrap_plots(interestedRegulons_R26_clust9,# + labs(tag = 'C'),
                            interestedRegulons_R26_clust27,
                            interestedRegulons_R26_clust16,
                            interestedRegulons_R26_clust34,
                            widths = c(0.115,0.275,0.275,0.275)
                            )

figure10bis_C_tag <- wrap_elements(plot_spacer() + labs(tag='C') +
                                   theme(text = element_text(face = 'bold',size = 9),
                                         plot.tag = element_text(size = 20, face = 'bold', hjust = -50)
                                         )
                                   )

# Trajectory
figure10bis_D <- wrap_plots(trajectory_inference_R26_monocle3_clust1234679) &
    labs(tag = 'D')

# RNA velocity
figure10bis_E <- wrap_plots(scVelo_clust1234679_DEG_monocle3) &
    labs(tag = 'E')

# Trajectory and RNA velocity
figure10bis_DE <- wrap_plots(figure10bis_D,
                             figure10bis_E,
                             heights = c(0.5,0.5)
                             )

# Heatmaps
figure10bis_F <- wrap_plots(plotList_heatmap_monocle3_interesingGenes[["Rest"]] &
    labs(tag = 'F'),
                            plotList_heatmap_monocle3_interesingGenes[["Id3"]],
                            plotList_heatmap_monocle3_interesingGenes[["Actl6b"]],
                            plotList_heatmap_monocle3_interesingGenes[["Dcx"]],
                            plotList_heatmap_monocle3_interesingGenes[["Hes1"]],
                            plotList_heatmap_monocle3_interesingGenes[["Ascl1"]],
                            plotList_heatmap_monocle3_interesingGenes[["Dll1"]],
                            plotList_heatmap_monocle3_interesingGenes[["Dll3"]],
                            nrow = 2
                            )

# Trajectory, RNA velocity and heatmaps
figure10bis_DEF <- wrap_plots(figure10bis_DE,
                              figure10bis_F,
                              widths = c(0.25,0.75)
                              )

# Patchwork
figure10bis <- wrap_plots(figure10bis_AB,
                          figure10bis_C_tag,
                          figure10bis_C,
                          figure10bis_DEF,
                          heights = c(0.49,0.01,0.15,0.35)
                          ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Figure 5\n",
                    theme = theme(plot.margin = unit(c(1,2,2,2),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Rename figure
figure5_LobonIglesias2022 <- figure10bis

# Save figure into pdf file
ggsave(plot   = figure5_LobonIglesias2022,
       file   = paste0(pathToFigures,"figure5_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"figure5_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_murine_RT"
              )

```

#### Patchwork Supplementary figure 5 (LobonIglesias2022)

```{r patchwork_figure10bis}

# UMAP and heatmap of SHH genes
SuppFig5_A <- wrap_elements(umap_Torchia2016_Johann2016_SHH_genes) +
    labs(tag = 'A')

# Stacked volcanoplot
SuppFig5_B <- wrap_elements(StackedVlnPlot_cellTypes_5_8_10_11_12_markerGenes)
SuppFig5_B <- wrap_plots(SuppFig5_B + labs(tag = 'B'),
                         plot_spacer(),
                         widths = c(0.5,0.5)
                         )

# Patchwork
SuppFig5_LobonIglesias2022 <- wrap_plots(SuppFig5_A,
                                         SuppFig5_B,
                                         plot_spacer(),
                                         heights = c(0.4,0.5,0.1)
                                         ) &
    theme(text = element_text(face = 'bold',size = 9),
          plot.tag = element_text(size = 20, face = 'bold')
          ) &
    plot_annotation(title = "Supplementary Fig. 6\n",
                    theme = theme(plot.margin = unit(c(1,2,2,2),'cm'),
                                  text = element_text(size = 20)
                                  )
                    )

# Save figure into pdf file
ggsave(plot   = SuppFig5_LobonIglesias2022,
       file   = paste0(pathToFigures,"SuppFig6new_LobonIglesias2022.pdf"),
       width  = 210, #full a4 210
       height = 297, # full a4 297
       units  = "mm"
       )

# Copy to Zerkalo
copyToZerkalo(file = paste0(pathToFigures,"SuppFig6new_LobonIglesias2022.pdf"),
              zerkaloDir = "mandrian_murine_RT"
              )

```


# cell-ID analysis

## Compute MCA

```{r MCA}

library(CellID)

# Copy object for cell-ID analysis (to avoid surprise when merging/adding metaData ;))
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- scRNAseq_Seurat_integrated_R26_tumCells_annot

# Compute MCA
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- RunMCA(X = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                        nmcs = 30
                                                        )

# Plot MCA
#pdf(file = paste0(pathToFigures,"mca_cellID_scRNAseq_integrated_R26.pdf"), width = 14, height = 10)
DimPlotMC(X        = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
          group.by = paste0("integrated_snn_res.",choosenResolution)
          )
#dev.off()

```

## Automatic cell type prediction using pre-established marker lists

### Import MSigDB C8 gene signatures

```{r sessionInfo}

library(msigdbr)

# Extract C8 MSigDB gene set collection
msigdb_mmusculus_c8 <- msigdbr(species = "Mus musculus",
                              category = "C8"
                              )

head(msigdb_mmusculus_c8)                                   

msigdb_mmusculus_c8_min <- msigdb_mmusculus_c8[,c("gene_symbol","gs_name")]

```

### Overlap with MSigDB C8 FAN_EMBRYONIC gene sets (Fan et al, 2018)

```{r FAN_EMBRYONIC}

# Select FAN_EMBRYONIC gene sets
msigdb_mmusculus_c8_FAN_EMBRYONIC <- filter(.data = msigdb_mmusculus_c8_min,
                                            grepl("FAN_EMBRYONIC_",gs_name)
                                            )

msigdb_mmusculus_c8_FAN_EMBRYONIC

# Convert data frame to list
msigdb_mmusculus_c8_FAN_EMBRYONIC_list <- as.list(unstack(msigdb_mmusculus_c8_FAN_EMBRYONIC))

# Run hypergeometric test
scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC <- RunCellHGT(X          = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                                                   pathways   = msigdb_mmusculus_c8_FAN_EMBRYONIC_list,
                                                                                   reduction  = "mca",
                                                                                   dim        = 30
                                                                                   )

dim(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC)
head(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC)))
enrichment_thrs <- 2

# Density distribution of enrichement score
melt(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_FAN_EMBRYONIC <- rownames(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC)[apply(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC, 2, which.max)]

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC)

# Keep cell annotation only if the score are higher than a given threshold
cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif <- ifelse(apply(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_FAN_EMBRYONIC, 2, max) > enrichment_thrs,
                                                     yes = cellTypes_by_msigdbC8_FAN_EMBRYONIC,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_integrated_R26_tumCells_annot) == names(cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                             metadata  = cellTypes_by_msigdbC8_FAN_EMBRYONIC_signif,
                                                             col.name  = "msigdb_C8_FAN_EMBRYONIC"
                                                             )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_msigdb_C8_FAN_EMBRYONIC.pdf"), width = 18, height = 10)
DimPlot(object     = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_FAN_EMBRYONIC",
        pt.size    = 1,
        cols = c("red","red","violet","cyan","mediumvioletred","yellow","black","palegoldenrod","green","grey","violet","violet","darkgreen","darkgreen","yellow","yellow","mediumblue","mediumblue","green","green","darkblue","darkblue","grey45","brown","snow3"),
        label      = FALSE
        ) #+
    #theme(legend.position = "none")
#dev.off()

```

### Overlap with MSigDB C8 DESCARTES_FETAL_CEREBELLUM gene sets (Cao et al, 2020)

```{r DESCARTES_FETAL_CEREBELLUM}

# Select DESCARTES_FETAL_CEREBELLUM gene sets
msigdb_mmusculus_c8_DESCARTES_FETAL_CEREBELLUM <- filter(.data = msigdb_mmusculus_c8_min,
                                                         grepl("DESCARTES_FETAL_CEREBELLUM_",gs_name)
                                                         )

msigdb_mmusculus_c8_DESCARTES_FETAL_CEREBELLUM

# Convert data frame to list
msigdb_mmusculus_c8_DESCARTES_FETAL_CEREBELLUM_list <- as.list(unstack(msigdb_mmusculus_c8_DESCARTES_FETAL_CEREBELLUM))

# Run hypergeometric test
scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- RunCellHGT(X          = scRNAseq_Seurat_integrated_R26_tumCells_annot,
                                                                          pathways   = msigdb_mmusculus_c8_DESCARTES_FETAL_CEREBELLUM_list,
                                                                          reduction  = "mca",
                                                                          dim        = 30
                                                                          )

dim(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)
head(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)))
enrichment_thrs <- 2

# Density distribution of enrichement score
melt(as.matrix(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# For each cell retrieve the gene set having the maximum score
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM <- rownames(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM)[apply(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, which.max)]

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM)

# Keep cell annotation only if the score are higher than a given threshold
cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif <- ifelse(apply(scRNAseq_Seurat_integrated_R26_tumCells_annot_msigdbC8_DESCARTES_FETAL_CEREBELLUM, 2, max) > enrichment_thrs,
                                                     yes = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM,
                                                     no = "unassigned"
                                                     )

head(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif)

# Check order of Seurat object metadata rownames and cell type vector names
all(colnames(scRNAseq_Seurat_integrated_R26_tumCells_annot) == names(cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif))

# Add cell type annotation to meta data
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                             metadata  = cellTypes_by_msigdbC8_DESCARTES_FETAL_CEREBELLUM_signif,
                                                             col.name  = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM"
                                                             )

# Plot UMAP with annotated cell type
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_msigdb_C8_DESCARTES_FETAL_CEREBELLUM.pdf"), width = 16, height = 10)
DimPlot(object     = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction  = "umap",
        group.by   = "msigdb_C8_DESCARTES_FETAL_CEREBELLUM",
        pt.size    = 1,
        cols = c("red","blue","mediumblue","yellow","black","darkblue","tan1","violet","darkgreen","snow3"),
        label      = FALSE
        )
#dev.off()

```

## Comparing integrated data (INI254,INI255,INI267) to integrated R26 tumor cells using Cell-ID

### Compute MCA on integrated dataset

```{r MCA_integrated}

# Import integrated Seurat object
scRNAseq_Seurat_integratedCAL <- readRDS(file = "/data/users/mandrian/rhabdoid_U830_projects/ANALYSIS_PROJECTS/cellOrigin_ATRT/svn/analyse/report/scRNAseq_Seurat_integrated.rds"
                                      )
scRNAseq_Seurat_integratedCAL

# Compute MCA
scRNAseq_Seurat_integratedCAL <- RunMCA(X     = scRNAseq_Seurat_integratedCAL,
                                        nmcs  = 30,
                                        slot  = "data"#"scale.data"
                                        )

# Plot MCA
#pdf(file = paste0(pathToFigures,"mca_cellID_scRNAseq_integratedCAL.pdf"), width = 14, height = 10)
DimPlotMC(X        = scRNAseq_Seurat_integratedCAL,
          group.by = "integrated_snn_res.0.2"
          )
#dev.off()

```

### Extract gene signatures from integrated data

```{r MCA_integrated}

nbSignatures <- 200

# Extracting per-cluster gene signatures
integratedCAL_cluster_signatures <- GetGroupGeneSet(X         = scRNAseq_Seurat_integratedCAL,
                                                    dims       = 1:30,
                                                    n.features = nbSignatures,
                                                    group.by   = "integrated_snn_res.0.2"
                                                    )

str(scRNAseq_Seurat_integratedCAL)
head(scRNAseq_Seurat_integratedCAL)

# Extracting per-cell gene signatures
integratedCAL_cell_signatures <- GetCellGeneSet(X           = scRNAseq_Seurat_integratedCAL,
                                                dims        = 1:30,
                                                n.features  = nbSignatures
                                                )

str(integratedCAL_cell_signatures)
head(integratedCAL_cell_signatures)

```

### Create human-mouse gene name conversion

```{r human_mouse_conversion}

# Load human and mouse ensembl datasets annotation
human <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl", host = "dec2021.archive.ensembl.org")
mouse <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl", host = "dec2021.archive.ensembl.org")

# Create human gene symbol and mouse orthologous gene symbol corresponding table
symbols_conversion <- getLDS(attributes  = c("hgnc_symbol"),
                             mart        = human,
                             attributesL = "mgi_symbol",
                             martL       = mouse,
                             uniqueRows  = TRUE
                             )

head(symbols_conversion)

```

### Convert human gene name to mouse gene name

```{r convert_cluster_signature}

# Convert list to data frame
integratedCAL_cluster_signatures_df <- data.frame(integratedCAL_cluster_signatures, check.names=FALSE)
head(integratedCAL_cluster_signatures_df)

# Melt data frame
integratedCAL_cluster_signatures_df_melt <- melt(integratedCAL_cluster_signatures_df, id.vars = 0)
head(integratedCAL_cluster_signatures_df_melt)
nrow(integratedCAL_cluster_signatures_df_melt)

# Merge melted data frame with mouse/human gene name conversion table
integratedCAL_cluster_signatures_df_melt_mouseSymbol <- merge(x = integratedCAL_cluster_signatures_df_melt,
                                                              y = symbols_conversion,
                                                              by.x = "value",
                                                              by.y = "HGNC.symbol",
                                                              all = FALSE
                                                              )

head(integratedCAL_cluster_signatures_df_melt_mouseSymbol)
nrow(integratedCAL_cluster_signatures_df_melt_mouseSymbol)

integratedCAL_cluster_signatures_df_melt_mouseSymbol$value <- NULL

# Convert melted data frame to list (go back to list format)
integratedCAL_cluster_signatures_mouseSymbol <- aggregate(formula = MGI.symbol ~ variable,
                                                          data = integratedCAL_cluster_signatures_df_melt_mouseSymbol,
                                                          FUN = 'c'
                                                          ) %>%
    spread(key = variable,
           value = MGI.symbol
           ) %>%
    as.list(all.names=TRUE) %>%
    unlist(recursive=FALSE)

```

```{r convert_cluster_signature}

# Convert list to data frame
integratedCAL_cell_signatures_df <- data.frame(integratedCAL_cell_signatures, check.names=FALSE)
head(integratedCAL_cell_signatures_df[,1:4])

# Melt data frame
integratedCAL_cell_signatures_df_melt <- melt(integratedCAL_cell_signatures_df, id.vars = 0)
head(integratedCAL_cell_signatures_df_melt)
nrow(integratedCAL_cell_signatures_df_melt)

# Merge melted data frame with mouse/human gene name conversion table
integratedCAL_cell_signatures_df_melt_mouseSymbol <- merge(x = integratedCAL_cell_signatures_df_melt,
                                                           y = symbols_conversion,
                                                           by.x = "value",
                                                           by.y = "HGNC.symbol",
                                                           all = FALSE
                                                           )

head(integratedCAL_cell_signatures_df_melt_mouseSymbol)
nrow(integratedCAL_cell_signatures_df_melt_mouseSymbol)

integratedCAL_cell_signatures_df_melt_mouseSymbol$value <- NULL

# Convert melted data frame to list (go back to list format)
integratedCAL_cell_signatures_mouseSymbol <- aggregate(formula = MGI.symbol ~ variable,
                                                       data = integratedCAL_cell_signatures_df_melt_mouseSymbol,
                                                       FUN = 'c'
                                                       ) %>%
    spread(key = variable,
           value = MGI.symbol
           ) %>%
    as.list(all.names=TRUE) %>%
    unlist(recursive=FALSE)

```

### Match integrated (INI254,INI255,INI267) cluster to integrated_R26_tumCells cells

```{r group_to_cell_matching}

# Group-to-cell matching
cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters <- RunCellHGT(X         = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                                            pathways  = integratedCAL_cluster_signatures_mouseSymbol,
                                                                            dims      = 30
                                                                            )

str(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters)
head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters)))
enrichment_thrs <- 2

# Density distribution of enrichement score
melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction <- rownames(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters)[apply(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters, 2, which.max)]
head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction)
levels(factor(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction))

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign <- ifelse(test = apply(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters, 2, max) > enrichment_thrs,
                                                                                        yes = cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction,
                                                                                        no = "unassigned"
                                                                                        )
head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign)

# Convert into data frame
cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df <- data.frame(integratedCAL_cluster = cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign)

cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df <- rownames_to_column(.data = cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df,
                                                                                      var = "integrated_R26_tumCells_cell"
                                                                                      )

head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data) == cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df$integrated_R26_tumCells_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                 metadata  = cellID_integrated_R26_tumCellscells_vs_integratedCAL_clusters_prediction_sign_df$integratedCAL_cluster,
                                                 col.name  = "cellID_integratedCAL_cluster"
                                                 )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_tumCells_cellID_integratedCAL_cluster.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = cellID_integratedCAL_cluster,
                             fill  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                             )
               ) +
    labs(fill = "integrated R26\ntumour cells\ncluster") +
    xlab("\nintegrated CAL-ATRT matching cluster") +
    geom_text(mapping  = aes(x     = cellID_integratedCAL_cluster,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_integratedCAL_cluster.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cellID_integratedCAL_cluster",
        label       = TRUE,
        label.size  = 15,
        cols = c("black","black","violet","darkgreen","yellow","blue","orange","cyan","red","brown","grey")
        ) +
    labs(color = "CAL-ATRT\nintegrated\nmatching\ncluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

### Match integrated (INI254,INI255,INI267) cell to integrated_R26_tumCells cells

```{r cell_to_cell_matching}

# Cell-to-cell matching
cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells <- RunCellHGT(X         = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                                         pathways  = integratedCAL_cell_signatures_mouseSymbol,
                                                                         dims      = 30
                                                                         )
str(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells)

head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells[,1:4])

# Enrichment score threshold
#summary(melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCells)))
enrichment_thrs <- 0.2

# Density distribution of enrichement score
melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCells)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction <- rownames(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells)[apply(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells, 2, which.max)]
head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction)

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign <- ifelse(test = apply(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells, 2, max) > enrichment_thrs,
                                                                       yes = cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign)

# Convert into data frame
cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df <- data.frame(integratedCell = cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign)

cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df <- rownames_to_column(.data = cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df,
                                                                                                    var = "integrated_R26_tumCells_cell"
                                                                                                    )

head(cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data) == cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df$integrated_R26_tumCells_cell )

# Add integrated cell assignment to meta data
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                              metadata  = cellID_integrated_R26_tumCellscells_vs_integratedCAL_cells_prediction_sign_df$integratedCell,
                                                              col.name  = "cellID_integratedCAL_cell"
                                                              )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Extract integrated cell cluster
scRNAseq_Seurat_integratedCAL_cellCluster <- data.frame(integratedCAL_cluster = scRNAseq_Seurat_integratedCAL@meta.data[,"integrated_snn_res.0.2"],
                                                        row.names = rownames(scRNAseq_Seurat_integratedCAL@meta.data)
                                                        )

scRNAseq_Seurat_integratedCAL_cellCluster <- rownames_to_column(.data = scRNAseq_Seurat_integratedCAL_cellCluster,
                                                                var = "integratedCAL_cell"
                                                                )

head(scRNAseq_Seurat_integratedCAL_cellCluster)

# Delete scRNAseq_Seurat_integrated object
rm(scRNAseq_Seurat_integratedCAL)

# Merge integrated cell cluster to meta data
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data <- left_join(x      = rownames_to_column(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data),
                                                                      y      = scRNAseq_Seurat_integratedCAL_cellCluster,
                                                                      by   = c("cellID_integratedCAL_cell" = "integratedCAL_cell")
                                                                      )

# Rename column
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data <- mutate(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data,
                                                                   cellID_integratedCAL_cellCluster = integratedCAL_cluster,
                                                                   .keep = "unused"
                                                                   )

# Put rownames (cell barcode)
rownames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data) <- scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data$rowname
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data$rowname <- NULL

head(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_tumCells_cellID_integratedCAL_cell.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             fill  = cellID_integratedCAL_cellCluster
                             )
               ) +
    labs(fill = "CAL-ATRT\nintegrated\nmatching\ncell cluster") +
    xlab("\n integrated R26 neuronal cluster") +
    scale_fill_manual(values = c("black","black","violet","darkgreen","yellow","blue","orange","cyan","red","brown","grey"),
                      na.value = "grey"
                      ) +
    geom_text(mapping  = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_integratedCAL_cell.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cellID_integratedCAL_cellCluster",
        label       = TRUE,
        label.size  = 15,
        cols = c("black","black","violet","darkgreen","yellow","blue","orange","cyan","red","brown","grey")
        ) +
    labs(color = "CAL-ATRT\nintegrated\nmatching\ncell cluster") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

## Comparing integrated data of CTX,LGE,MGE,CGE (Lee et al., 2022) to integrated R26 tumor cells using Cell-ID

### Compute MCA on integrated dataset

```{r MCA_integrated}

# Import integrated Seurat object
scRNAseq_Seurat_Lee2022_integrated <- readRDS(file = paste0(pathToReport,"scRNAseq_Seurat_Lee2022_integrated.RDS")
                                              )
scRNAseq_Seurat_Lee2022_integrated

# Compute MCA on integrated data
scRNAseq_Seurat_Lee2022_integrated <- RunMCA(X     = scRNAseq_Seurat_Lee2022_integrated,
                                             nmcs  = 30,
                                             slot  = "data"#"scale.data"
                                             )

# Plot MCA of integrated dataset
#pdf(file = paste0(pathToFigures,"mca_cellID_scRNAseq_Lee2022_integrated.pdf"), width = 14, height = 10)
DimPlotMC(X        = scRNAseq_Seurat_Lee2022_integrated,
          group.by = "orig.ident",
          dim      = c(1,2)
          )
#dev.off()

```

### Extract gene signatures from integrated data

```{r MCA_integrated}

nbSignatures <- 200

# Extracting per-cluster gene signatures
Lee2022_integrated_sample_signatures <- GetGroupGeneSet(X         = scRNAseq_Seurat_Lee2022_integrated,
                                                        dims       = 1:30,
                                                        n.features = nbSignatures,
                                                        group.by   = "orig.ident"
                                                        )

str(Lee2022_integrated_sample_signatures)
head(Lee2022_integrated_sample_signatures)

# Extracting per-cell gene signatures
Lee2022_integrated_cell_signatures <- GetCellGeneSet(X           = scRNAseq_Seurat_Lee2022_integrated,
                                                     dims        = 1:30,
                                                     n.features  = nbSignatures
                                                     )
str(Lee2022_integrated_cell_signatures)
head(Lee2022_integrated_cell_signatures)

```

### Match Lee et al, 2022 samples (CTX,LGE,MGE,CGE) to integrated_R26_tumCells cells

```{r group_to_cell_matching}

# Group-to-cell matching
cellID_integrated_R26_tumCellscells_vs_Lee2022integrated <- RunCellHGT(X         = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                                       pathways  = Lee2022_integrated_sample_signatures,
                                                                       dims      = 30
                                                                       )

str(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated)
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated[,1:4])

# Enrichment score threshold
summary(melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated)))
enrichment_thrs <- 2

# Density distribution of enrichement score
melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction <- rownames(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated)[apply(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated, 2, which.max)]
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction)
levels(factor(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction))

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign <- ifelse(test = apply(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated, 2, max) > enrichment_thrs,
                                                                       yes = cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign)

# Convert into data frame
cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df <- data.frame(Lee2022_sample = cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign)

cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df <- rownames_to_column(.data = cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df,
                                                                                      var = "integrated_R26_tumCells_cell"
                                                                                      )

head(cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data) == cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df$integrated_R26_tumCells_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                              metadata  = cellID_integrated_R26_tumCellscells_vs_Lee2022integrated_prediction_sign_df$Lee2022_sample,
                                                              col.name  = "cellID_Lee2022_sample"
                                                              )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_tumCells_cellID_Lee2022_sample.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = cellID_Lee2022_sample,
                             fill  = eval(as.name(paste0("integrated_snn_res.",choosenResolution)))
                             )
               ) +
    labs(fill = "integrated R26\ntumour cells\ncluster") +
    xlab("\nLee et al., 2022 E12.5 sample") +
    scale_fill_manual(values = c("skyblue",#0
                                 "skyblue4",#1
                                 "springgreen",#2
                                 "springgreen4",#3
                                 "sienna4",#4
                                 "violet",#5
                                 "sienna1",#6
                                 "sienna3",#7
                                 "violetred",#8
                                 "blue4",#9
                                 "yellow",#10
                                 "blue",#11
                                 "red"#12
                                 ),
                      na.value = "grey"
                      ) +
    geom_text(mapping  = aes(x     = cellID_Lee2022_sample,
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_Lee2022_sample.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction   = "umap",
        pt.size     = 1,
        cols        = c("violet","yellow","blue","red","grey"),
        group.by    = "cellID_Lee2022_sample",
        label       = TRUE,
        label.size  = 15
        ) +
    labs(color = "Lee et al. (2022)\nmatching\nsample") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```

### Match Lee et al, 2022 cells (annotated CTX,LGE,MGE,CGE) to integrated_R26_tumCells cells

```{r cell_to_cell_matching}

# Cell-to-cell matching
cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells <- RunCellHGT(X         = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                                            pathways  = Lee2022_integrated_cell_signatures,
                                                                            dims      = 30
                                                                            )

str(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells)
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells[,1:4])

# Enrichment score threshold
#summary(melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_integratedCells)))
enrichment_thrs <- 0.2

# Density distribution of enrichement score
melt(as.matrix(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells)) %>%
    ggplot() +
    geom_density(mapping = aes(x = log10(value+1))
                 ) +
    geom_vline(xintercept = log10(enrichment_thrs+1), color = "red", lty = 2) +
    theme_bw()

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score
cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction <- rownames(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells)[apply(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells, 2, which.max)]
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction)

# Assing integrated_R26_tumCells cell to integrated cluster according to max enrichment score with a significant thresold
cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign <- ifelse(test = apply(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells, 2, max) > enrichment_thrs,
                                                                       yes = cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction,
                                                                       no = "unassigned"
                                                                       )
head(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign)

# Convert into data frame
cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df <- data.frame(Lee2022integratedCells = cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign)

head(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df)

cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df <- rownames_to_column(.data = cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df,
                                                                                                       var = "integrated_R26_tumCells_cell"
                                                                                                       )

head(cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df)

# Check order
all(rownames(scRNAseq_Seurat_integrated_R26_tumCells@meta.data) == cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df$integrated_R26_tumCells_cell )

# Add integrated cell assignment to meta dat
scRNAseq_Seurat_integrated_R26_tumCells_cellID <- AddMetaData(object    = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
                                                              metadata  = cellID_integrated_R26_tumCellscells_vs_Lee2022integratedCells_prediction_sign_df$Lee2022integratedCells,
                                                              col.name  = "cellID_Lee2022_integratedCell"
                                                              )

colnames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Extract Lee et al, 2022 integrated cell sample
scRNAseq_Seurat_Lee2022_integrated_cellSample <- data.frame(Lee2022_integratedSample = scRNAseq_Seurat_Lee2022_integrated@meta.data[,"orig.ident"],
                                                             row.names = rownames(scRNAseq_Seurat_Lee2022_integrated@meta.data)
                                                             )

head(scRNAseq_Seurat_Lee2022_integrated_cellSample)

scRNAseq_Seurat_Lee2022_integrated_cellSample <- rownames_to_column(.data = scRNAseq_Seurat_Lee2022_integrated_cellSample,
                                                                     var = "Lee2022_integratedCell"
                                                                     )

head(scRNAseq_Seurat_Lee2022_integrated_cellSample)

# Delete scRNAseq_Seurat_integrated object
rm(scRNAseq_Seurat_Lee2022_integrated)

head(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Merge integrated cell cluster to meta data
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data <- left_join(x      = rownames_to_column(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data),
                                                                      y      = scRNAseq_Seurat_Lee2022_integrated_cellSample,
                                                                      by   = c("cellID_Lee2022_integratedCell" = "Lee2022_integratedCell")
                                                                      )

# Rename column
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data <- mutate(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data,
                                                                   cellID_Lee2022_cellSample= Lee2022_integratedSample,
                                                                   .keep = "unused"
                                                                   )

# Put rownames (cell barcode)

rownames(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data) <- scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data$rowname
scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data$rowname <- NULL

head(scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data)

# Barlot of assigned cells
#pdf(file = paste0(pathToFigures,"barplot_scRNAseq_Seurat_integrated_R26_tumCells_cellID_Lee2022_cellSample.pdf"), width = 16, height = 10)
ggplot(data = scRNAseq_Seurat_integrated_R26_tumCells_cellID@meta.data
       ) +
    stat_count(mapping = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             fill  = cellID_Lee2022_cellSample
                             )
               ) +
    labs(fill = "Lee et al, 2002\nsample") +
    xlab("\n integrated R26 neuronal cluster") +
    scale_fill_manual(values = c("violet","yellow","blue","red"),
                      na.value = "grey"
                      ) +
    geom_text(mapping  = aes(x     = eval(as.name(paste0("integrated_snn_res.",choosenResolution))),
                             label = stat(count)
                             ),
              stat     = "count",
              vjust    = -0.25,
              size     = 7.5,
              fontface     = "bold"
              ) +
    theme_bw() +
    theme(text = element_text(size = 30)#,
          #legend.position = "none"
          )
#dev.off()

# Plot UMAP
#pdf(file = paste0(pathToFigures,"umap_scRNAseq_integrated_R26_tumCells_cellID_Lee2022_cellSample.pdf"), width = 14, height = 12)
DimPlot(object      = scRNAseq_Seurat_integrated_R26_tumCells_cellID,
        reduction   = "umap",
        pt.size     = 1,
        group.by    = "cellID_Lee2022_cellSample",
        label       = FALSE,
        label.size  = 15,
        cols        = c("violet","yellow","blue","red")
        ) +
    labs(color = "Lee et al, 2022\nsample") +
    theme_bw() +
    theme(#legend.position = "none",
          text = element_text(size = 30)
          )
#dev.off()

```


# Session info

```{r sessionInfo}

# Session info
sessionInfo()

```
